<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</title>
  <meta name="description" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  
  
  

<meta name="author" content="Daan Andel" />


<meta name="date" content="2023-09-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="subclonal-dynamics.html"/>
<link rel="next" href="copy-number-pattern-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Subclonal Evolution Paper</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html"><i class="fa fa-check"></i><b>2</b> In vitro radiation response and clinical data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#progression-free-survival"><i class="fa fa-check"></i><b>2.1</b> Progression-free survival</a></li>
<li class="chapter" data-level="2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#in-vitro-dose-response"><i class="fa fa-check"></i><b>2.2</b> In vitro dose-response</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#dose-response-plot"><i class="fa fa-check"></i><b>2.2.1</b> Dose-response plot</a></li>
<li class="chapter" data-level="2.2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#rauc-across-multiple-experiments"><i class="fa fa-check"></i><b>2.2.2</b> rAUC across multiple experiments</a></li>
<li class="chapter" data-level="2.2.3" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#recovery-statistics-after-10-gy-radiation"><i class="fa fa-check"></i><b>2.2.3</b> Recovery statistics after 10 Gy radiation</a></li>
<li class="chapter" data-level="2.2.4" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#persisters-versus-shifters-dose-response-plots"><i class="fa fa-check"></i><b>2.2.4</b> Persisters versus shifters dose-response plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html"><i class="fa fa-check"></i><b>3</b> Subclonal dynamics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#aneufinder-run"><i class="fa fa-check"></i><b>3.1</b> AneuFinder run</a></li>
<li class="chapter" data-level="3.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#quality-selection"><i class="fa fa-check"></i><b>3.2</b> Quality selection</a></li>
<li class="chapter" data-level="3.3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#summary-statistics-of-copy-number-alterations"><i class="fa fa-check"></i><b>3.3</b> Summary statistics of copy number alterations</a></li>
<li class="chapter" data-level="3.4" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#pca-and-k-means-plots"><i class="fa fa-check"></i><b>3.4</b> PCA and K-means plots</a></li>
<li class="chapter" data-level="3.5" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-selection"><i class="fa fa-check"></i><b>3.5</b> Subclone selection</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#selecting-subclones"><i class="fa fa-check"></i><b>3.5.1</b> Selecting subclones</a></li>
<li class="chapter" data-level="3.5.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-statistics"><i class="fa fa-check"></i><b>3.5.2</b> Subclone statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#single-cell-copy-number-heatmap-plots"><i class="fa fa-check"></i><b>3.6</b> Single cell copy number heatmap plots</a></li>
<li class="chapter" data-level="3.7" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#fishers-exact-test"><i class="fa fa-check"></i><b>3.7</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html"><i class="fa fa-check"></i><b>4</b> Copy number analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid"><i class="fa fa-check"></i><b>4.1</b> Colorectal cancer-specific arm-level copy number alterations per organoid</a></li>
<li class="chapter" data-level="4.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-per-subclone"><i class="fa fa-check"></i><b>4.2</b> Copy number alterations per subclone</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a"><i class="fa fa-check"></i><b>4.2.1</b> Copy number alterations specific to HUB062A</a></li>
<li class="chapter" data-level="4.2.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones"><i class="fa fa-check"></i><b>4.2.2</b> Copy number alterations specific to resistant subclones</a></li>
<li class="chapter" data-level="4.2.3" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#newly-acquired-copy-number-alterations"><i class="fa fa-check"></i><b>4.2.3</b> Newly acquired copy number alterations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html"><i class="fa fa-check"></i><b>5</b> Copy number pattern analyses</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#single-cell-analyses"><i class="fa fa-check"></i><b>5.1</b> Single cell analyses</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#heterogeneity-score"><i class="fa fa-check"></i><b>5.1.1</b> Heterogeneity score</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#copy-number-length-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Copy number length analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#ccle-analysis"><i class="fa fa-check"></i><b>5.2</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#run-cin-signature"><i class="fa fa-check"></i><b>5.2.1</b> Run CIN signature</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#statistics-cin-signature"><i class="fa fa-check"></i><b>5.2.2</b> Statistics CIN signature</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#plot-cx1-cin-signature"><i class="fa fa-check"></i><b>5.2.3</b> Plot CX1 CIN signature</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="source-code.html"><a href="source-code.html"><i class="fa fa-check"></i><b>6</b> Source code</a>
<ul>
<li class="chapter" data-level="6.1" data-path="source-code.html"><a href="source-code.html#auc_rad-script-auc_fun-function"><i class="fa fa-check"></i><b>6.1</b> AUC_rad script: ‘AUC_fun’ function</a></li>
<li class="chapter" data-level="6.2" data-path="source-code.html"><a href="source-code.html#fitter_rad-script-fitter-function"><i class="fa fa-check"></i><b>6.2</b> fitter_rad script: ‘fitter’ function</a></li>
<li class="chapter" data-level="6.3" data-path="source-code.html"><a href="source-code.html#ic50_rad-script-ic50_fun-function"><i class="fa fa-check"></i><b>6.3</b> ic50_rad script: ‘IC50_fun’ function</a></li>
<li class="chapter" data-level="6.4" data-path="source-code.html"><a href="source-code.html#relativize_dr-script-relative-function"><i class="fa fa-check"></i><b>6.4</b> relativize_dr script: ‘relative’ function</a></li>
<li class="chapter" data-level="6.5" data-path="source-code.html"><a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions"><i class="fa fa-check"></i><b>6.5</b> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions</a></li>
<li class="chapter" data-level="6.6" data-path="source-code.html"><a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions"><i class="fa fa-check"></i><b>6.6</b> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions</a></li>
<li class="chapter" data-level="6.7" data-path="source-code.html"><a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions"><i class="fa fa-check"></i><b>6.7</b> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions</a></li>
<li class="chapter" data-level="6.8" data-path="source-code.html"><a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function"><i class="fa fa-check"></i><b>6.8</b> plot_genomeheatmap script: ‘genomeheatmap’ function</a></li>
<li class="chapter" data-level="6.9" data-path="source-code.html"><a href="source-code.html#fisher_test-script-perform_fisher-function"><i class="fa fa-check"></i><b>6.9</b> fisher_test script: ‘perform_fisher’ function</a></li>
<li class="chapter" data-level="6.10" data-path="source-code.html"><a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function"><i class="fa fa-check"></i><b>6.10</b> find_specific_cna_sc script: ‘find_specific_cna_sc’ function</a></li>
<li class="chapter" data-level="6.11" data-path="source-code.html"><a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function"><i class="fa fa-check"></i><b>6.11</b> get_cytoband_coverage script: ‘get_cytoband_coverage’ function</a></li>
<li class="chapter" data-level="6.12" data-path="source-code.html"><a href="source-code.html#get_genes-script-get_genes-functoin"><i class="fa fa-check"></i><b>6.12</b> get_genes script: ‘get_genes’ functoin</a></li>
<li class="chapter" data-level="6.13" data-path="source-code.html"><a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function"><i class="fa fa-check"></i><b>6.13</b> get_oncogenes scripts: ‘get_oncogenes’ function</a></li>
<li class="chapter" data-level="6.14" data-path="source-code.html"><a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function"><i class="fa fa-check"></i><b>6.14</b> plot_genes_karyotype script: ‘plot_genes_karyotype’ function</a></li>
<li class="chapter" data-level="6.15" data-path="source-code.html"><a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions"><i class="fa fa-check"></i><b>6.15</b> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions</a></li>
<li class="chapter" data-level="6.16" data-path="source-code.html"><a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function"><i class="fa fa-check"></i><b>6.16</b> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function</a></li>
<li class="chapter" data-level="6.17" data-path="source-code.html"><a href="source-code.html#cna_profiler-script-cna_profiler-function"><i class="fa fa-check"></i><b>6.17</b> cna_profiler script: ‘cna_profiler’ function</a></li>
<li class="chapter" data-level="6.18" data-path="source-code.html"><a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function"><i class="fa fa-check"></i><b>6.18</b> cin_signature_ccle script: ‘run_cin_ccle’ function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="copy-number-analyses" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Copy number analyses<a href="copy-number-analyses.html#copy-number-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="copy-number-analyses.html#cb79-1" tabindex="-1"></a><span class="co"># Libraries and sources</span></span>
<span id="cb79-2"><a href="copy-number-analyses.html#cb79-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb79-3"><a href="copy-number-analyses.html#cb79-3" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb79-4"><a href="copy-number-analyses.html#cb79-4" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb79-5"><a href="copy-number-analyses.html#cb79-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/find_specific_cna_sc.R&#39;</span>)</span>
<span id="cb79-6"><a href="copy-number-analyses.html#cb79-6" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_cytoband_coverage.R&#39;</span>)</span>
<span id="cb79-7"><a href="copy-number-analyses.html#cb79-7" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_genes.R&#39;</span>)</span>
<span id="cb79-8"><a href="copy-number-analyses.html#cb79-8" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_oncogenes.R&#39;</span>)</span>
<span id="cb79-9"><a href="copy-number-analyses.html#cb79-9" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/plot_genes_karyotype.R&#39;</span>)</span></code></pre></div>
<div id="colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Colorectal cancer-specific arm-level copy number alterations per organoid<a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="copy-number-analyses.html#cb80-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb80-2"><a href="copy-number-analyses.html#cb80-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb80-3"><a href="copy-number-analyses.html#cb80-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb80-4"><a href="copy-number-analyses.html#cb80-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/subclonal_dynamics/man_select_files_edivisive.rda&#39;</span>)</span>
<span id="cb80-5"><a href="copy-number-analyses.html#cb80-5" tabindex="-1"></a></span>
<span id="cb80-6"><a href="copy-number-analyses.html#cb80-6" tabindex="-1"></a><span class="co"># !!! GITHUB SPECIFIC CODE !!! #</span></span>
<span id="cb80-7"><a href="copy-number-analyses.html#cb80-7" tabindex="-1"></a><span class="co"># adding &#39;../cna_analysis/&#39; to each path within man_select_files_edivisive </span></span>
<span id="cb80-8"><a href="copy-number-analyses.html#cb80-8" tabindex="-1"></a>man_select_files_edivisive <span class="ot">&lt;-</span> <span class="fu">lapply</span>(man_select_files_edivisive, <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">&quot;../cna_analysis/&quot;</span>, x))</span>
<span id="cb80-9"><a href="copy-number-analyses.html#cb80-9" tabindex="-1"></a></span>
<span id="cb80-10"><a href="copy-number-analyses.html#cb80-10" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb80-11"><a href="copy-number-analyses.html#cb80-11" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb80-12"><a href="copy-number-analyses.html#cb80-12" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb80-13"><a href="copy-number-analyses.html#cb80-13" tabindex="-1"></a><span class="co"># extract only prerad clines</span></span>
<span id="cb80-14"><a href="copy-number-analyses.html#cb80-14" tabindex="-1"></a><span class="co"># prerad clines</span></span>
<span id="cb80-15"><a href="copy-number-analyses.html#cb80-15" tabindex="-1"></a>kra_list <span class="ot">&lt;-</span></span>
<span id="cb80-16"><a href="copy-number-analyses.html#cb80-16" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&quot;kra003&quot;</span>,</span>
<span id="cb80-17"><a href="copy-number-analyses.html#cb80-17" tabindex="-1"></a>    <span class="st">&quot;kra005&quot;</span>,</span>
<span id="cb80-18"><a href="copy-number-analyses.html#cb80-18" tabindex="-1"></a>    <span class="st">&quot;kra007&quot;</span>,</span>
<span id="cb80-19"><a href="copy-number-analyses.html#cb80-19" tabindex="-1"></a>    <span class="st">&quot;kra009&quot;</span>,</span>
<span id="cb80-20"><a href="copy-number-analyses.html#cb80-20" tabindex="-1"></a>    <span class="st">&quot;kra023&quot;</span>,</span>
<span id="cb80-21"><a href="copy-number-analyses.html#cb80-21" tabindex="-1"></a>    <span class="st">&quot;kra025&quot;</span>,</span>
<span id="cb80-22"><a href="copy-number-analyses.html#cb80-22" tabindex="-1"></a>    <span class="st">&quot;kra027&quot;</span>)</span>
<span id="cb80-23"><a href="copy-number-analyses.html#cb80-23" tabindex="-1"></a></span>
<span id="cb80-24"><a href="copy-number-analyses.html#cb80-24" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb80-25"><a href="copy-number-analyses.html#cb80-25" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb80-26"><a href="copy-number-analyses.html#cb80-26" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb80-27"><a href="copy-number-analyses.html#cb80-27" tabindex="-1"></a></span>
<span id="cb80-28"><a href="copy-number-analyses.html#cb80-28" tabindex="-1"></a><span class="do">## finding common CNAs within each subclone for each organoids.</span></span>
<span id="cb80-29"><a href="copy-number-analyses.html#cb80-29" tabindex="-1"></a>common_cnas <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb80-30"><a href="copy-number-analyses.html#cb80-30" tabindex="-1"></a>  <span class="at">list_resistant =</span> man_select_files_edivisive[kra_list],</span>
<span id="cb80-31"><a href="copy-number-analyses.html#cb80-31" tabindex="-1"></a>  <span class="co"># make sure to input this as a list!</span></span>
<span id="cb80-32"><a href="copy-number-analyses.html#cb80-32" tabindex="-1"></a>  <span class="at">list_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb80-33"><a href="copy-number-analyses.html#cb80-33" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> <span class="fl">0.8</span>,</span>
<span id="cb80-34"><a href="copy-number-analyses.html#cb80-34" tabindex="-1"></a>  <span class="co"># CNA needs to be in 80% of the cells or more</span></span>
<span id="cb80-35"><a href="copy-number-analyses.html#cb80-35" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb80-36"><a href="copy-number-analyses.html#cb80-36" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="cn">NULL</span>,</span>
<span id="cb80-37"><a href="copy-number-analyses.html#cb80-37" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb80-38"><a href="copy-number-analyses.html#cb80-38" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> T</span>
<span id="cb80-39"><a href="copy-number-analyses.html#cb80-39" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 80% of single cells within each resistant subclone, and by % of single cells within each sensitive subclone&quot;</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="copy-number-analyses.html#cb82-1" tabindex="-1"></a><span class="do">## mapping CNA to cytoband location</span></span>
<span id="cb82-2"><a href="copy-number-analyses.html#cb82-2" tabindex="-1"></a><span class="co"># for amplifications</span></span>
<span id="cb82-3"><a href="copy-number-analyses.html#cb82-3" tabindex="-1"></a>amplifications <span class="ot">&lt;-</span></span>
<span id="cb82-4"><a href="copy-number-analyses.html#cb82-4" tabindex="-1"></a>  <span class="fu">lapply</span>(common_cnas<span class="sc">$</span>amplifications, get_cytoband_coverage)</span>
<span id="cb82-5"><a href="copy-number-analyses.html#cb82-5" tabindex="-1"></a></span>
<span id="cb82-6"><a href="copy-number-analyses.html#cb82-6" tabindex="-1"></a><span class="co"># Extract frequency of arm level amplifications n indicates the how many</span></span>
<span id="cb82-7"><a href="copy-number-analyses.html#cb82-7" tabindex="-1"></a><span class="co"># organoids contain the CNA</span></span>
<span id="cb82-8"><a href="copy-number-analyses.html#cb82-8" tabindex="-1"></a>arm_amp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(amplifications, <span class="cf">function</span>(x) {</span>
<span id="cb82-9"><a href="copy-number-analyses.html#cb82-9" tabindex="-1"></a>  <span class="fu">return</span>(x[[<span class="dv">2</span>]])</span>
<span id="cb82-10"><a href="copy-number-analyses.html#cb82-10" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb82-11"><a href="copy-number-analyses.html#cb82-11" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;cline&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb82-12"><a href="copy-number-analyses.html#cb82-12" tabindex="-1"></a>  <span class="fu">group_by</span>(cline, chromosome, arm) <span class="sc">%&gt;%</span></span>
<span id="cb82-13"><a href="copy-number-analyses.html#cb82-13" tabindex="-1"></a>  <span class="fu">count</span>(percentage_arm_affected <span class="sc">&gt;</span> <span class="fl">0.8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb82-14"><a href="copy-number-analyses.html#cb82-14" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb82-15"><a href="copy-number-analyses.html#cb82-15" tabindex="-1"></a>  <span class="fu">ungroup</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb82-16"><a href="copy-number-analyses.html#cb82-16" tabindex="-1"></a>  <span class="fu">count</span>(chromosome) <span class="sc">%&gt;%</span></span>
<span id="cb82-17"><a href="copy-number-analyses.html#cb82-17" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb82-18"><a href="copy-number-analyses.html#cb82-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> n <span class="sc">/</span> <span class="dv">7</span>)</span>
<span id="cb82-19"><a href="copy-number-analyses.html#cb82-19" tabindex="-1"></a></span>
<span id="cb82-20"><a href="copy-number-analyses.html#cb82-20" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(arm_amp, <span class="at">caption =</span> <span class="st">&quot;Arm-level amplifications in organoids&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-26fd757ce6f594d0f643" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-26fd757ce6f594d0f643">{"x":{"filter":"none","vertical":false,"caption":"<caption>Arm-level amplifications in organoids<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44"],["1","20","20","8","3","2","16","16","13","13","12","10","10","9","8","7","7","5","2","1","19","19","17","17","12","11","9","6","6","5","3","22","22","14","14","11","X","4","21","21","X","4","18","18"],["q","q","p","q","q","q","p","q","p","q","q","p","q","q","p","p","q","p","p","p","p","q","p","q","p","q","p","p","q","q","p","p","q","p","q","p","q","q","p","q","p","p","p","q"],[7,7,6,6,6,6,5,5,5,5,5,5,5,5,5,5,5,5,5,4,4,4,4,4,4,4,4,4,4,4,4,3,3,3,3,3,3,3,2,2,2,2,1,1],[1,1,0.8571428571428571,0.8571428571428571,0.8571428571428571,0.8571428571428571,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.2857142857142857,0.2857142857142857,0.2857142857142857,0.2857142857142857,0.1428571428571428,0.1428571428571428]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chromosome<\/th>\n      <th>arm<\/th>\n      <th>n<\/th>\n      <th>freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="copy-number-analyses.html#cb83-1" tabindex="-1"></a><span class="co"># for deletions</span></span>
<span id="cb83-2"><a href="copy-number-analyses.html#cb83-2" tabindex="-1"></a>deletions <span class="ot">&lt;-</span> <span class="fu">lapply</span>(common_cnas<span class="sc">$</span>deletions, get_cytoband_coverage)</span>
<span id="cb83-3"><a href="copy-number-analyses.html#cb83-3" tabindex="-1"></a></span>
<span id="cb83-4"><a href="copy-number-analyses.html#cb83-4" tabindex="-1"></a><span class="co"># Extract frequency of arm level deletions. n indicates the how many</span></span>
<span id="cb83-5"><a href="copy-number-analyses.html#cb83-5" tabindex="-1"></a><span class="co"># organoids contain the CNA</span></span>
<span id="cb83-6"><a href="copy-number-analyses.html#cb83-6" tabindex="-1"></a>arm_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(deletions, <span class="cf">function</span>(x) {</span>
<span id="cb83-7"><a href="copy-number-analyses.html#cb83-7" tabindex="-1"></a>  <span class="fu">return</span>(x[[<span class="dv">2</span>]])</span>
<span id="cb83-8"><a href="copy-number-analyses.html#cb83-8" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb83-9"><a href="copy-number-analyses.html#cb83-9" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;cline&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb83-10"><a href="copy-number-analyses.html#cb83-10" tabindex="-1"></a>  <span class="fu">group_by</span>(cline, chromosome, arm) <span class="sc">%&gt;%</span></span>
<span id="cb83-11"><a href="copy-number-analyses.html#cb83-11" tabindex="-1"></a>  <span class="fu">count</span>(percentage_arm_affected <span class="sc">&gt;</span> <span class="fl">0.8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb83-12"><a href="copy-number-analyses.html#cb83-12" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb83-13"><a href="copy-number-analyses.html#cb83-13" tabindex="-1"></a>  <span class="fu">ungroup</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb83-14"><a href="copy-number-analyses.html#cb83-14" tabindex="-1"></a>  <span class="fu">count</span>(chromosome) <span class="sc">%&gt;%</span></span>
<span id="cb83-15"><a href="copy-number-analyses.html#cb83-15" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb83-16"><a href="copy-number-analyses.html#cb83-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> n <span class="sc">/</span> <span class="dv">7</span>)</span>
<span id="cb83-17"><a href="copy-number-analyses.html#cb83-17" tabindex="-1"></a></span>
<span id="cb83-18"><a href="copy-number-analyses.html#cb83-18" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(arm_del, <span class="at">caption =</span> <span class="st">&quot;Arm-level deletions in organoids&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-6f91ea85250de37b8271" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6f91ea85250de37b8271">{"x":{"filter":"none","vertical":false,"caption":"<caption>Arm-level deletions in organoids<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11"],["18","8","1","1","10","18","4","6","6","X","X"],["q","p","p","q","q","p","q","p","q","p","q"],[4,3,1,1,1,1,1,1,1,1,1],[0.5714285714285714,0.4285714285714285,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chromosome<\/th>\n      <th>arm<\/th>\n      <th>n<\/th>\n      <th>freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="copy-number-analyses.html#cb84-1" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb84-2"><a href="copy-number-analyses.html#cb84-2" tabindex="-1"></a><span class="co"># End #</span></span>
<span id="cb84-3"><a href="copy-number-analyses.html#cb84-3" tabindex="-1"></a><span class="do">#######</span></span></code></pre></div>
</div>
<div id="copy-number-alterations-per-subclone" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Copy number alterations per subclone<a href="copy-number-analyses.html#copy-number-alterations-per-subclone" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="copy-number-alterations-specific-to-hub062a" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Copy number alterations specific to HUB062A<a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="copy-number-analyses.html#cb85-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb85-2"><a href="copy-number-analyses.html#cb85-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb85-3"><a href="copy-number-analyses.html#cb85-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb85-4"><a href="copy-number-analyses.html#cb85-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/subclonal_dynamics/subclone_list.rda&#39;</span>)</span>
<span id="cb85-5"><a href="copy-number-analyses.html#cb85-5" tabindex="-1"></a></span>
<span id="cb85-6"><a href="copy-number-analyses.html#cb85-6" tabindex="-1"></a><span class="co"># !!! GITHUB SPECIFIC CODE !!! #</span></span>
<span id="cb85-7"><a href="copy-number-analyses.html#cb85-7" tabindex="-1"></a><span class="co"># adding &#39;../cna_analysis/&#39; to each path within man_select_files_edivisive </span></span>
<span id="cb85-8"><a href="copy-number-analyses.html#cb85-8" tabindex="-1"></a>subclone_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(subclone_list, <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">&quot;../cna_analysis/&quot;</span>, x))</span>
<span id="cb85-9"><a href="copy-number-analyses.html#cb85-9" tabindex="-1"></a></span>
<span id="cb85-10"><a href="copy-number-analyses.html#cb85-10" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb85-11"><a href="copy-number-analyses.html#cb85-11" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb85-12"><a href="copy-number-analyses.html#cb85-12" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb85-13"><a href="copy-number-analyses.html#cb85-13" tabindex="-1"></a><span class="co"># extracting resistant and sensitive subclones of hub062</span></span>
<span id="cb85-14"><a href="copy-number-analyses.html#cb85-14" tabindex="-1"></a>hub062 <span class="ot">&lt;-</span></span>
<span id="cb85-15"><a href="copy-number-analyses.html#cb85-15" tabindex="-1"></a>  <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub062&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb85-16"><a href="copy-number-analyses.html#cb85-16" tabindex="-1"></a>res <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_a&#39;</span>, hub062)]</span>
<span id="cb85-17"><a href="copy-number-analyses.html#cb85-17" tabindex="-1"></a>sens <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_b|hub062_postrad_c&#39;</span>, hub062)]</span>
<span id="cb85-18"><a href="copy-number-analyses.html#cb85-18" tabindex="-1"></a><span class="co"># remove cycle 2</span></span>
<span id="cb85-19"><a href="copy-number-analyses.html#cb85-19" tabindex="-1"></a>sens <span class="ot">&lt;-</span> sens[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&#39;c2&#39;</span>, sens)]</span>
<span id="cb85-20"><a href="copy-number-analyses.html#cb85-20" tabindex="-1"></a></span>
<span id="cb85-21"><a href="copy-number-analyses.html#cb85-21" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb85-22"><a href="copy-number-analyses.html#cb85-22" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb85-23"><a href="copy-number-analyses.html#cb85-23" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb85-24"><a href="copy-number-analyses.html#cb85-24" tabindex="-1"></a></span>
<span id="cb85-25"><a href="copy-number-analyses.html#cb85-25" tabindex="-1"></a><span class="do">## finding common CNAs within each subclone for each organoids.</span></span>
<span id="cb85-26"><a href="copy-number-analyses.html#cb85-26" tabindex="-1"></a>hub062a_specific_cnas <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb85-27"><a href="copy-number-analyses.html#cb85-27" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[res],</span>
<span id="cb85-28"><a href="copy-number-analyses.html#cb85-28" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[sens],</span>
<span id="cb85-29"><a href="copy-number-analyses.html#cb85-29" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> <span class="fl">0.6</span>,</span>
<span id="cb85-30"><a href="copy-number-analyses.html#cb85-30" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.6</span>,</span>
<span id="cb85-31"><a href="copy-number-analyses.html#cb85-31" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.2</span>,</span>
<span id="cb85-32"><a href="copy-number-analyses.html#cb85-32" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.2</span>,</span>
<span id="cb85-33"><a href="copy-number-analyses.html#cb85-33" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb85-34"><a href="copy-number-analyses.html#cb85-34" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 60 and 60% of single cells within resistant and sensitive subclones, respectively, and that are shared by 20 and 20% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="copy-number-analyses.html#cb87-1" tabindex="-1"></a><span class="do">## mapping CNA to cytoband location</span></span>
<span id="cb87-2"><a href="copy-number-analyses.html#cb87-2" tabindex="-1"></a><span class="co"># for amplifications</span></span>
<span id="cb87-3"><a href="copy-number-analyses.html#cb87-3" tabindex="-1"></a>ampl_hub062a <span class="ot">&lt;-</span></span>
<span id="cb87-4"><a href="copy-number-analyses.html#cb87-4" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(hub062a_specific_cnas<span class="sc">$</span>unique_resistant_ampl)</span>
<span id="cb87-5"><a href="copy-number-analyses.html#cb87-5" tabindex="-1"></a></span>
<span id="cb87-6"><a href="copy-number-analyses.html#cb87-6" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(ampl_hub062a[[<span class="dv">1</span>]]), <span class="at">caption =</span> <span class="st">&quot;HUB062A-specific amplifications (exact location)&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-9b3ae2f11d0b622aae99" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9b3ae2f11d0b622aae99">{"x":{"filter":"none","vertical":false,"caption":"<caption>HUB062A-specific amplifications (exact location)<\/caption>","data":[["1","2","3","4","5","6","7","8"],["2","8","9","11","17","17","17","X"],[217938033,34917837,1,1,1,21081101,27304921,52329467],[218892728,37853827,70227261,70578503,10892963,21546522,37378798,114869867],[954696,2935991,70227261,70578503,10892963,465422,10073878,62540401],["*","*","*","*","*","*","*","*"],[["2q35;15%"],["8p12;23%","8p11.23;64%"],["9p24.3;100%","9p24.2;100%","9p24.1;100%","9p23;100%","9p22.3;100%","9p22.2;100%","9p22.1;100%","9p21.3;100%","9p21.2;100%","9p21.1;100%","9p13.3;100%","9p13.2;100%","9p13.1;100%","9p12;100%","9p11.2;100%","9p11.1;100%","9q11;100%","9q12;100%","9q13;100%","9q21.11;100%","9q21.12;46%"],["11p15.5;100%","11p15.4;100%","11p15.3;100%","11p15.2;100%","11p15.1;100%","11p14.3;100%","11p14.2;100%","11p14.1;100%","11p13;100%","11p12;100%","11p11.2;100%","11p11.12;100%","11p11.11;100%","11q11;100%","11q12.1;100%","11q12.2;100%","11q12.3;100%","11q13.1;100%","11q13.2;100%","11q13.3;100%","11q13.4;2%"],["17p13.3;100%","17p13.2;100%","17p13.1;100%","17p12;2%"],["17p11.2;7%"],["17q11.1;4%","17q11.2;100%","17q12;62%"],["Xp11.22;53%","Xp11.21;100%","Xp11.1;100%","Xq11.1;100%","Xq11.2;100%","Xq12;100%","Xq13.1;100%","Xq13.2;100%","Xq13.3;100%","Xq21.1;100%","Xq21.2;100%","Xq21.31;100%","Xq21.32;100%","Xq21.33;100%","Xq22.1;100%","Xq22.2;100%","Xq22.3;100%","Xq23;68%"]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="copy-number-analyses.html#cb88-1" tabindex="-1"></a>del_hub062a <span class="ot">&lt;-</span></span>
<span id="cb88-2"><a href="copy-number-analyses.html#cb88-2" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(hub062a_specific_cnas<span class="sc">$</span>unique_resistant_del)</span>
<span id="cb88-3"><a href="copy-number-analyses.html#cb88-3" tabindex="-1"></a></span>
<span id="cb88-4"><a href="copy-number-analyses.html#cb88-4" tabindex="-1"></a><span class="co"># No HUB062A-specific deletions were found.</span></span></code></pre></div>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="copy-number-analyses.html#cb89-1" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb89-2"><a href="copy-number-analyses.html#cb89-2" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb89-3"><a href="copy-number-analyses.html#cb89-3" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb89-4"><a href="copy-number-analyses.html#cb89-4" tabindex="-1"></a><span class="co"># All genes #</span></span>
<span id="cb89-5"><a href="copy-number-analyses.html#cb89-5" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb89-6"><a href="copy-number-analyses.html#cb89-6" tabindex="-1"></a>genes_ampl_hub062a <span class="ot">&lt;-</span> <span class="fu">get_genes</span>(ampl_hub062a[[<span class="dv">1</span>]])</span>
<span id="cb89-7"><a href="copy-number-analyses.html#cb89-7" tabindex="-1"></a></span>
<span id="cb89-8"><a href="copy-number-analyses.html#cb89-8" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(genes_ampl_hub062a), <span class="at">caption =</span> <span class="st">&quot;Genes on HUB062A-specific amplifications&quot;</span>)</span>
<span id="cb89-9"><a href="copy-number-analyses.html#cb89-9" tabindex="-1"></a></span>
<span id="cb89-10"><a href="copy-number-analyses.html#cb89-10" tabindex="-1"></a><span class="co"># Oncogenes #</span></span>
<span id="cb89-11"><a href="copy-number-analyses.html#cb89-11" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb89-12"><a href="copy-number-analyses.html#cb89-12" tabindex="-1"></a>oncogenes_ampl_hub062a <span class="ot">&lt;-</span> <span class="fu">get_oncogenes</span>(ampl_hub062a[[<span class="dv">1</span>]])</span>
<span id="cb89-13"><a href="copy-number-analyses.html#cb89-13" tabindex="-1"></a></span>
<span id="cb89-14"><a href="copy-number-analyses.html#cb89-14" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(oncogenes_ampl_hub062a), <span class="at">caption =</span> <span class="st">&quot;Oncogenes on HUB062A-specific amplifications&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="copy-number-analyses.html#cb90-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb90-2"><a href="copy-number-analyses.html#cb90-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb90-3"><a href="copy-number-analyses.html#cb90-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb90-4"><a href="copy-number-analyses.html#cb90-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hub062a_specific_cnas_genes.rda&#39;</span>)</span>
<span id="cb90-5"><a href="copy-number-analyses.html#cb90-5" tabindex="-1"></a></span>
<span id="cb90-6"><a href="copy-number-analyses.html#cb90-6" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb90-7"><a href="copy-number-analyses.html#cb90-7" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb90-8"><a href="copy-number-analyses.html#cb90-8" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb90-9"><a href="copy-number-analyses.html#cb90-9" tabindex="-1"></a><span class="co"># plot_genes_karyotype needs a GRange with seqnames, ranges, strand and hgnc_symbol</span></span>
<span id="cb90-10"><a href="copy-number-analyses.html#cb90-10" tabindex="-1"></a></span>
<span id="cb90-11"><a href="copy-number-analyses.html#cb90-11" tabindex="-1"></a><span class="co"># add ampl/del column</span></span>
<span id="cb90-12"><a href="copy-number-analyses.html#cb90-12" tabindex="-1"></a>oncogenes_ampl_hub062a<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="st">&#39;amplified&#39;</span></span>
<span id="cb90-13"><a href="copy-number-analyses.html#cb90-13" tabindex="-1"></a></span>
<span id="cb90-14"><a href="copy-number-analyses.html#cb90-14" tabindex="-1"></a></span>
<span id="cb90-15"><a href="copy-number-analyses.html#cb90-15" tabindex="-1"></a><span class="co"># bind dataframes</span></span>
<span id="cb90-16"><a href="copy-number-analyses.html#cb90-16" tabindex="-1"></a>oncogenes <span class="ot">&lt;-</span> oncogenes_ampl_hub062a</span>
<span id="cb90-17"><a href="copy-number-analyses.html#cb90-17" tabindex="-1"></a></span>
<span id="cb90-18"><a href="copy-number-analyses.html#cb90-18" tabindex="-1"></a><span class="co"># extract needed columns</span></span>
<span id="cb90-19"><a href="copy-number-analyses.html#cb90-19" tabindex="-1"></a>keep <span class="ot">&lt;-</span></span>
<span id="cb90-20"><a href="copy-number-analyses.html#cb90-20" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;Gene Symbol&#39;</span>,</span>
<span id="cb90-21"><a href="copy-number-analyses.html#cb90-21" tabindex="-1"></a>    <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb90-22"><a href="copy-number-analyses.html#cb90-22" tabindex="-1"></a>    <span class="st">&#39;start_position&#39;</span>,</span>
<span id="cb90-23"><a href="copy-number-analyses.html#cb90-23" tabindex="-1"></a>    <span class="st">&#39;end_position&#39;</span>,</span>
<span id="cb90-24"><a href="copy-number-analyses.html#cb90-24" tabindex="-1"></a>    <span class="st">&#39;cn&#39;</span>)</span>
<span id="cb90-25"><a href="copy-number-analyses.html#cb90-25" tabindex="-1"></a>oncogenes <span class="ot">&lt;-</span> oncogenes[, keep]</span>
<span id="cb90-26"><a href="copy-number-analyses.html#cb90-26" tabindex="-1"></a><span class="co"># rename</span></span>
<span id="cb90-27"><a href="copy-number-analyses.html#cb90-27" tabindex="-1"></a><span class="fu">colnames</span>(oncogenes) <span class="ot">&lt;-</span></span>
<span id="cb90-28"><a href="copy-number-analyses.html#cb90-28" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;hgnc_symbol&#39;</span>, <span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;cn&#39;</span>)</span>
<span id="cb90-29"><a href="copy-number-analyses.html#cb90-29" tabindex="-1"></a></span>
<span id="cb90-30"><a href="copy-number-analyses.html#cb90-30" tabindex="-1"></a><span class="co"># convert to dataframe</span></span>
<span id="cb90-31"><a href="copy-number-analyses.html#cb90-31" tabindex="-1"></a>onco_grange <span class="ot">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span>(oncogenes)</span>
<span id="cb90-32"><a href="copy-number-analyses.html#cb90-32" tabindex="-1"></a></span>
<span id="cb90-33"><a href="copy-number-analyses.html#cb90-33" tabindex="-1"></a><span class="co"># metadata is deleted</span></span>
<span id="cb90-34"><a href="copy-number-analyses.html#cb90-34" tabindex="-1"></a>onco_grange<span class="sc">$</span>hgnc_symbol <span class="ot">&lt;-</span> oncogenes<span class="sc">$</span>hgnc_symbol</span>
<span id="cb90-35"><a href="copy-number-analyses.html#cb90-35" tabindex="-1"></a>onco_grange<span class="sc">$</span>cn <span class="ot">&lt;-</span> oncogenes<span class="sc">$</span>cn</span>
<span id="cb90-36"><a href="copy-number-analyses.html#cb90-36" tabindex="-1"></a></span>
<span id="cb90-37"><a href="copy-number-analyses.html#cb90-37" tabindex="-1"></a><span class="do">##########################</span></span>
<span id="cb90-38"><a href="copy-number-analyses.html#cb90-38" tabindex="-1"></a><span class="co"># Choose genes to depict #</span></span>
<span id="cb90-39"><a href="copy-number-analyses.html#cb90-39" tabindex="-1"></a><span class="do">##########################</span></span>
<span id="cb90-40"><a href="copy-number-analyses.html#cb90-40" tabindex="-1"></a>oncogenes_ampl_hub062a <span class="ot">&lt;-</span></span>
<span id="cb90-41"><a href="copy-number-analyses.html#cb90-41" tabindex="-1"></a>  oncogenes_ampl_hub062a[<span class="fu">order</span>(oncogenes_ampl_hub062a<span class="sc">$</span><span class="st">`</span><span class="at">Genome Location</span><span class="st">`</span>),]</span>
<span id="cb90-42"><a href="copy-number-analyses.html#cb90-42" tabindex="-1"></a></span>
<span id="cb90-43"><a href="copy-number-analyses.html#cb90-43" tabindex="-1"></a>show_oncogenes_ampl <span class="ot">&lt;-</span></span>
<span id="cb90-44"><a href="copy-number-analyses.html#cb90-44" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;FANCF&#39;</span>,</span>
<span id="cb90-45"><a href="copy-number-analyses.html#cb90-45" tabindex="-1"></a>    <span class="st">&#39;CREB3L1&#39;</span>,</span>
<span id="cb90-46"><a href="copy-number-analyses.html#cb90-46" tabindex="-1"></a>    <span class="st">&#39;DDB2&#39;</span>,</span>
<span id="cb90-47"><a href="copy-number-analyses.html#cb90-47" tabindex="-1"></a>    <span class="st">&#39;HRAS&#39;</span>,</span>
<span id="cb90-48"><a href="copy-number-analyses.html#cb90-48" tabindex="-1"></a>    <span class="st">&#39;CCND1&#39;</span>,</span>
<span id="cb90-49"><a href="copy-number-analyses.html#cb90-49" tabindex="-1"></a>    <span class="st">&#39;NF1&#39;</span>,</span>
<span id="cb90-50"><a href="copy-number-analyses.html#cb90-50" tabindex="-1"></a>    <span class="st">&#39;FANCG&#39;</span>,</span>
<span id="cb90-51"><a href="copy-number-analyses.html#cb90-51" tabindex="-1"></a>    <span class="st">&#39;JAK2&#39;</span>,</span>
<span id="cb90-52"><a href="copy-number-analyses.html#cb90-52" tabindex="-1"></a>    <span class="st">&#39;AR&#39;</span>)</span>
<span id="cb90-53"><a href="copy-number-analyses.html#cb90-53" tabindex="-1"></a></span>
<span id="cb90-54"><a href="copy-number-analyses.html#cb90-54" tabindex="-1"></a>show_oncogenes <span class="ot">&lt;-</span> <span class="fu">c</span>(show_oncogenes_ampl)</span>
<span id="cb90-55"><a href="copy-number-analyses.html#cb90-55" tabindex="-1"></a></span>
<span id="cb90-56"><a href="copy-number-analyses.html#cb90-56" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb90-57"><a href="copy-number-analyses.html#cb90-57" tabindex="-1"></a><span class="co"># PLOT #</span></span>
<span id="cb90-58"><a href="copy-number-analyses.html#cb90-58" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb90-59"><a href="copy-number-analyses.html#cb90-59" tabindex="-1"></a><span class="fu">plot_genes_karyotype</span>(</span>
<span id="cb90-60"><a href="copy-number-analyses.html#cb90-60" tabindex="-1"></a>  onco_grange,</span>
<span id="cb90-61"><a href="copy-number-analyses.html#cb90-61" tabindex="-1"></a>  <span class="at">show_genes =</span> show_oncogenes,</span>
<span id="cb90-62"><a href="copy-number-analyses.html#cb90-62" tabindex="-1"></a>  <span class="at">dist =</span> <span class="sc">-</span><span class="dv">45</span>,</span>
<span id="cb90-63"><a href="copy-number-analyses.html#cb90-63" tabindex="-1"></a>  <span class="at">show_chromosomes =</span> <span class="fu">c</span>(</span>
<span id="cb90-64"><a href="copy-number-analyses.html#cb90-64" tabindex="-1"></a>    <span class="st">&quot;chr9&quot;</span>,</span>
<span id="cb90-65"><a href="copy-number-analyses.html#cb90-65" tabindex="-1"></a>    <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb90-66"><a href="copy-number-analyses.html#cb90-66" tabindex="-1"></a>    <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb90-67"><a href="copy-number-analyses.html#cb90-67" tabindex="-1"></a>    <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb90-68"><a href="copy-number-analyses.html#cb90-68" tabindex="-1"></a>    <span class="st">&quot;chr13&quot;</span>,</span>
<span id="cb90-69"><a href="copy-number-analyses.html#cb90-69" tabindex="-1"></a>    <span class="st">&quot;chr14&quot;</span>,</span>
<span id="cb90-70"><a href="copy-number-analyses.html#cb90-70" tabindex="-1"></a>    <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb90-71"><a href="copy-number-analyses.html#cb90-71" tabindex="-1"></a>    <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb90-72"><a href="copy-number-analyses.html#cb90-72" tabindex="-1"></a>    <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb90-73"><a href="copy-number-analyses.html#cb90-73" tabindex="-1"></a>    <span class="st">&quot;chr18&quot;</span>,</span>
<span id="cb90-74"><a href="copy-number-analyses.html#cb90-74" tabindex="-1"></a>    <span class="st">&quot;chr19&quot;</span>,</span>
<span id="cb90-75"><a href="copy-number-analyses.html#cb90-75" tabindex="-1"></a>    <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb90-76"><a href="copy-number-analyses.html#cb90-76" tabindex="-1"></a>    <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb90-77"><a href="copy-number-analyses.html#cb90-77" tabindex="-1"></a>    <span class="st">&quot;chr22&quot;</span>,</span>
<span id="cb90-78"><a href="copy-number-analyses.html#cb90-78" tabindex="-1"></a>    <span class="st">&quot;chrX&quot;</span></span>
<span id="cb90-79"><a href="copy-number-analyses.html#cb90-79" tabindex="-1"></a>  )</span>
<span id="cb90-80"><a href="copy-number-analyses.html#cb90-80" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figs/hub062a_karyoplot.jpg" width="1288" /></p>
</div>
<div id="copy-number-alterations-specific-to-resistant-subclones" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Copy number alterations specific to resistant subclones<a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="copy-number-analyses.html#cb91-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb91-2"><a href="copy-number-analyses.html#cb91-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb91-3"><a href="copy-number-analyses.html#cb91-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb91-4"><a href="copy-number-analyses.html#cb91-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hub062a_specific_cnas.rda&#39;</span>)</span>
<span id="cb91-5"><a href="copy-number-analyses.html#cb91-5" tabindex="-1"></a></span>
<span id="cb91-6"><a href="copy-number-analyses.html#cb91-6" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb91-7"><a href="copy-number-analyses.html#cb91-7" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb91-8"><a href="copy-number-analyses.html#cb91-8" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb91-9"><a href="copy-number-analyses.html#cb91-9" tabindex="-1"></a><span class="co"># extracting resistant and sensitive subclones of hub062</span></span>
<span id="cb91-10"><a href="copy-number-analyses.html#cb91-10" tabindex="-1"></a><span class="co"># excluding HUB197, as here subclones were probably not selected on the basis of CNAs</span></span>
<span id="cb91-11"><a href="copy-number-analyses.html#cb91-11" tabindex="-1"></a><span class="co"># but on the basis of the presence of WGD!</span></span>
<span id="cb91-12"><a href="copy-number-analyses.html#cb91-12" tabindex="-1"></a></span>
<span id="cb91-13"><a href="copy-number-analyses.html#cb91-13" tabindex="-1"></a>resistant <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&#39;hub005_prerad_b&#39;</span>, <span class="fu">names</span>(subclone_list)),</span>
<span id="cb91-14"><a href="copy-number-analyses.html#cb91-14" tabindex="-1"></a>                                    <span class="fu">grep</span>(<span class="st">&#39;hub183_prerad&#39;</span>, <span class="fu">names</span>(subclone_list)), </span>
<span id="cb91-15"><a href="copy-number-analyses.html#cb91-15" tabindex="-1"></a>                                    <span class="fu">grep</span>(<span class="st">&#39;hub062_postrad_a&#39;</span>, <span class="fu">names</span>(subclone_list))</span>
<span id="cb91-16"><a href="copy-number-analyses.html#cb91-16" tabindex="-1"></a>                                    )]</span>
<span id="cb91-17"><a href="copy-number-analyses.html#cb91-17" tabindex="-1"></a></span>
<span id="cb91-18"><a href="copy-number-analyses.html#cb91-18" tabindex="-1"></a></span>
<span id="cb91-19"><a href="copy-number-analyses.html#cb91-19" tabindex="-1"></a><span class="co"># in resistant, </span></span>
<span id="cb91-20"><a href="copy-number-analyses.html#cb91-20" tabindex="-1"></a><span class="co"># in resistant, hub005 a.a and a.b need to be joined together, as they are no subclones</span></span>
<span id="cb91-21"><a href="copy-number-analyses.html#cb91-21" tabindex="-1"></a>hub005_prerad_a.a_a.b <span class="ot">&lt;-</span> <span class="fu">c</span>(subclone_list<span class="sc">$</span>hub005_prerad_a.a, subclone_list<span class="sc">$</span>hub005_prerad_a.b)</span>
<span id="cb91-22"><a href="copy-number-analyses.html#cb91-22" tabindex="-1"></a></span>
<span id="cb91-23"><a href="copy-number-analyses.html#cb91-23" tabindex="-1"></a><span class="co"># joining the combined clones to resistant</span></span>
<span id="cb91-24"><a href="copy-number-analyses.html#cb91-24" tabindex="-1"></a>res_list <span class="ot">&lt;-</span> subclone_list[resistant]</span>
<span id="cb91-25"><a href="copy-number-analyses.html#cb91-25" tabindex="-1"></a></span>
<span id="cb91-26"><a href="copy-number-analyses.html#cb91-26" tabindex="-1"></a>res_list[[<span class="st">&#39;hub005_prerad_a.a_a.b&#39;</span>]] <span class="ot">&lt;-</span> hub005_prerad_a.a_a.b </span>
<span id="cb91-27"><a href="copy-number-analyses.html#cb91-27" tabindex="-1"></a></span>
<span id="cb91-28"><a href="copy-number-analyses.html#cb91-28" tabindex="-1"></a><span class="co"># comparing resistant to sensitive lines yields to many results.   </span></span>
<span id="cb91-29"><a href="copy-number-analyses.html#cb91-29" tabindex="-1"></a>rest <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">c</span>(<span class="fu">grep</span>(<span class="st">&#39;hub106_prerad&#39;</span>, <span class="fu">names</span>(subclone_list)),</span>
<span id="cb91-30"><a href="copy-number-analyses.html#cb91-30" tabindex="-1"></a>                                    <span class="fu">grep</span>(<span class="st">&#39;hub062_prerad_b&#39;</span>, <span class="fu">names</span>(subclone_list)),</span>
<span id="cb91-31"><a href="copy-number-analyses.html#cb91-31" tabindex="-1"></a>                                    <span class="fu">grep</span>(<span class="st">&#39;hub062_prerad_c&#39;</span>, <span class="fu">names</span>(subclone_list)),</span>
<span id="cb91-32"><a href="copy-number-analyses.html#cb91-32" tabindex="-1"></a>                                    <span class="fu">grep</span>(<span class="st">&#39;hub197_prerad_b&#39;</span>, <span class="fu">names</span>(subclone_list))</span>
<span id="cb91-33"><a href="copy-number-analyses.html#cb91-33" tabindex="-1"></a>                               </span>
<span id="cb91-34"><a href="copy-number-analyses.html#cb91-34" tabindex="-1"></a>                               )]</span>
<span id="cb91-35"><a href="copy-number-analyses.html#cb91-35" tabindex="-1"></a></span>
<span id="cb91-36"><a href="copy-number-analyses.html#cb91-36" tabindex="-1"></a></span>
<span id="cb91-37"><a href="copy-number-analyses.html#cb91-37" tabindex="-1"></a></span>
<span id="cb91-38"><a href="copy-number-analyses.html#cb91-38" tabindex="-1"></a>rest_list <span class="ot">&lt;-</span> subclone_list[rest]</span>
<span id="cb91-39"><a href="copy-number-analyses.html#cb91-39" tabindex="-1"></a></span>
<span id="cb91-40"><a href="copy-number-analyses.html#cb91-40" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb91-41"><a href="copy-number-analyses.html#cb91-41" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb91-42"><a href="copy-number-analyses.html#cb91-42" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb91-43"><a href="copy-number-analyses.html#cb91-43" tabindex="-1"></a><span class="do">## finding common CNAs within each subclone for each organoid.</span></span>
<span id="cb91-44"><a href="copy-number-analyses.html#cb91-44" tabindex="-1"></a>resistant_specific_cnas <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb91-45"><a href="copy-number-analyses.html#cb91-45" tabindex="-1"></a>  <span class="at">list_resistant =</span> res_list,</span>
<span id="cb91-46"><a href="copy-number-analyses.html#cb91-46" tabindex="-1"></a>  <span class="at">list_sensitive =</span> rest_list,</span>
<span id="cb91-47"><a href="copy-number-analyses.html#cb91-47" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> <span class="fl">0.6</span>, <span class="co"># needs to be present in 60% or more cells per subclone</span></span>
<span id="cb91-48"><a href="copy-number-analyses.html#cb91-48" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.6</span>,</span>
<span id="cb91-49"><a href="copy-number-analyses.html#cb91-49" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.8</span>, <span class="co"># if present in 80% of the subclones, it will be called a CNA and might thus be resistant-specific (depending on if it is also found in sensitive subclones)</span></span>
<span id="cb91-50"><a href="copy-number-analyses.html#cb91-50" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.1</span>, <span class="co"># if present in 20%, it is already called a CNA in sensitive subclones and thus wont be detected as a resistant specific CNA</span></span>
<span id="cb91-51"><a href="copy-number-analyses.html#cb91-51" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb91-52"><a href="copy-number-analyses.html#cb91-52" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 60 and 60% of single cells within resistant and sensitive subclones, respectively, and that are shared by 80 and 10% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="copy-number-analyses.html#cb93-1" tabindex="-1"></a><span class="do">## mapping CNA to cytoband location</span></span>
<span id="cb93-2"><a href="copy-number-analyses.html#cb93-2" tabindex="-1"></a><span class="co"># for amplifications</span></span>
<span id="cb93-3"><a href="copy-number-analyses.html#cb93-3" tabindex="-1"></a>ampl_resistant <span class="ot">&lt;-</span></span>
<span id="cb93-4"><a href="copy-number-analyses.html#cb93-4" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(resistant_specific_cnas<span class="sc">$</span>unique_resistant_ampl)</span>
<span id="cb93-5"><a href="copy-number-analyses.html#cb93-5" tabindex="-1"></a></span>
<span id="cb93-6"><a href="copy-number-analyses.html#cb93-6" tabindex="-1"></a>del_resistant <span class="ot">&lt;-</span></span>
<span id="cb93-7"><a href="copy-number-analyses.html#cb93-7" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(resistant_specific_cnas<span class="sc">$</span>unique_resistant_del)</span>
<span id="cb93-8"><a href="copy-number-analyses.html#cb93-8" tabindex="-1"></a></span>
<span id="cb93-9"><a href="copy-number-analyses.html#cb93-9" tabindex="-1"></a><span class="co"># Both ampl_resistant and del_resistant are empty, no resistance specific cnas</span></span>
<span id="cb93-10"><a href="copy-number-analyses.html#cb93-10" tabindex="-1"></a><span class="co"># shared by subclones was found</span></span></code></pre></div>
</div>
<div id="newly-acquired-copy-number-alterations" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Newly acquired copy number alterations<a href="copy-number-analyses.html#newly-acquired-copy-number-alterations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="copy-number-analyses.html#cb94-1" tabindex="-1"></a><span class="do">###############</span></span>
<span id="cb94-2"><a href="copy-number-analyses.html#cb94-2" tabindex="-1"></a><span class="co"># Description #</span></span>
<span id="cb94-3"><a href="copy-number-analyses.html#cb94-3" tabindex="-1"></a><span class="do">###############</span></span>
<span id="cb94-4"><a href="copy-number-analyses.html#cb94-4" tabindex="-1"></a><span class="co"># We are going to define the presence of &#39;new&#39; CNAs after radiation therapy. </span></span>
<span id="cb94-5"><a href="copy-number-analyses.html#cb94-5" tabindex="-1"></a><span class="co"># I define the following rules: </span></span>
<span id="cb94-6"><a href="copy-number-analyses.html#cb94-6" tabindex="-1"></a><span class="co"># (i) Analysis will be done within subclone</span></span>
<span id="cb94-7"><a href="copy-number-analyses.html#cb94-7" tabindex="-1"></a><span class="co"># (ii) A &#39;new&#39; CNA will be called new only if it was not found before radiation</span></span>
<span id="cb94-8"><a href="copy-number-analyses.html#cb94-8" tabindex="-1"></a><span class="co"># (iii) I exclude all subclones where the numbers of cells within a subclone is lower than 5. </span></span>
<span id="cb94-9"><a href="copy-number-analyses.html#cb94-9" tabindex="-1"></a><span class="co"># (iv) Because in essence, all cells are related to each other (even those in distinct) </span></span>
<span id="cb94-10"><a href="copy-number-analyses.html#cb94-10" tabindex="-1"></a><span class="co"># subclones, I am going to compare each post-rad subclone to all pooled subclones before</span></span>
<span id="cb94-11"><a href="copy-number-analyses.html#cb94-11" tabindex="-1"></a><span class="co"># radiation. </span></span>
<span id="cb94-12"><a href="copy-number-analyses.html#cb94-12" tabindex="-1"></a><span class="co"># (v) We are looking for acquired CNAs; CNAs that could explain why cells with that </span></span>
<span id="cb94-13"><a href="copy-number-analyses.html#cb94-13" tabindex="-1"></a><span class="co"># CNA propagated. Therefore, the CNA has to be present in at least 50% of the cells. </span></span>
<span id="cb94-14"><a href="copy-number-analyses.html#cb94-14" tabindex="-1"></a></span>
<span id="cb94-15"><a href="copy-number-analyses.html#cb94-15" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb94-16"><a href="copy-number-analyses.html#cb94-16" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb94-17"><a href="copy-number-analyses.html#cb94-17" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb94-18"><a href="copy-number-analyses.html#cb94-18" tabindex="-1"></a></span>
<span id="cb94-19"><a href="copy-number-analyses.html#cb94-19" tabindex="-1"></a><span class="co"># remove subclones with fewer than 5 cells pre-or post rad. </span></span>
<span id="cb94-20"><a href="copy-number-analyses.html#cb94-20" tabindex="-1"></a><span class="fu">lapply</span>(subclone_list, length)</span></code></pre></div>
<pre><code>## $hub183_prerad_a
## [1] 127
## 
## $hub183_postrad_a
## [1] 110
## 
## $hub005_prerad_a.a
## [1] 25
## 
## $hub005_prerad_a.b
## [1] 13
## 
## $hub005_prerad_b
## [1] 118
## 
## $hub005_postrad_a.a
## [1] 17
## 
## $hub005_postrad_a.b
## [1] 2
## 
## $hub005_postrad_b
## [1] 70
## 
## $hub106_prerad_a
## [1] 64
## 
## $hub106_prerad_b
## [1] 59
## 
## $hub106_prerad_c
## [1] 4
## 
## $hub106_postrad_a
## [1] 60
## 
## $hub106_postrad_b
## [1] 103
## 
## $hub106_postrad_c
## [1] 18
## 
## $hub062_prerad_a
## [1] 2
## 
## $hub062_prerad_b
## [1] 127
## 
## $hub062_prerad_c
## [1] 7
## 
## $hub062_postrad_a
## [1] 39
## 
## $hub062_postrad_b
## [1] 38
## 
## $hub062_postrad_c
## [1] 7
## 
## $hub062_postrad_c2_a
## [1] 187
## 
## $hub062_postrad_c2_c
## [1] 1
## 
## $hub015_prerad_a.b
## [1] 276
## 
## $hub015_prerad_b
## [1] 1
## 
## $hub015_postrad_a.a
## [1] 110
## 
## $hub015_postrad_a.b
## [1] 157
## 
## $hub015_postrad_b
## [1] 14
## 
## $hub181i_prerad_a
## [1] 72
## 
## $hub181i_prerad_b
## [1] 233
## 
## $hub181i_postrad_a
## [1] 72
## 
## $hub181i_postrad_b
## [1] 239
## 
## $hub197_prerad_a
## [1] 24
## 
## $hub197_prerad_b
## [1] 17
## 
## $hub197_prerad_c
## [1] 91
## 
## $hub197_postrad_a
## [1] 252
## 
## $hub062biological_prerad_a
## [1] 14
## 
## $hub062biological_prerad_b
## [1] 205
## 
## $hub062biological_prerad_c
## [1] 13
## 
## $hub062biological_postrad_a
## [1] 178</code></pre>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="copy-number-analyses.html#cb96-1" tabindex="-1"></a>remove_me <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;hub005_prerad_a.b&#39;</span>, <span class="st">&#39;hub005_postrad_a.b&#39;</span>, <span class="co"># because only 2 cells in hub005_postrad_a.b</span></span>
<span id="cb96-2"><a href="copy-number-analyses.html#cb96-2" tabindex="-1"></a>               <span class="st">&#39;hub062_prerad_a&#39;</span>, <span class="co"># only 2 cells, here I am going to compare C1 vs C2, so </span></span>
<span id="cb96-3"><a href="copy-number-analyses.html#cb96-3" tabindex="-1"></a>               <span class="co"># I don&#39;t remove hub062_postrad_a</span></span>
<span id="cb96-4"><a href="copy-number-analyses.html#cb96-4" tabindex="-1"></a>               <span class="st">&#39;hub015_prerad_b&#39;</span>, <span class="st">&#39;hub015_postrad_b&#39;</span>, <span class="co"># only 1 cell in prerad</span></span>
<span id="cb96-5"><a href="copy-number-analyses.html#cb96-5" tabindex="-1"></a>               <span class="st">&#39;hub197_prerad_b&#39;</span>,<span class="st">&#39;hub197_prerad_b&#39;</span>, <span class="co"># extinction</span></span>
<span id="cb96-6"><a href="copy-number-analyses.html#cb96-6" tabindex="-1"></a>               <span class="st">&#39;hub197_prerad_c&#39;</span>,<span class="st">&#39;hub197_postrad_c&#39;</span> <span class="co"># extinction</span></span>
<span id="cb96-7"><a href="copy-number-analyses.html#cb96-7" tabindex="-1"></a>               )</span>
<span id="cb96-8"><a href="copy-number-analyses.html#cb96-8" tabindex="-1"></a></span>
<span id="cb96-9"><a href="copy-number-analyses.html#cb96-9" tabindex="-1"></a>subclone_list <span class="ot">&lt;-</span> subclone_list[<span class="sc">!</span>(<span class="fu">names</span>(subclone_list) <span class="sc">%in%</span> remove_me)]</span>
<span id="cb96-10"><a href="copy-number-analyses.html#cb96-10" tabindex="-1"></a></span>
<span id="cb96-11"><a href="copy-number-analyses.html#cb96-11" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb96-12"><a href="copy-number-analyses.html#cb96-12" tabindex="-1"></a><span class="co"># Variables #</span></span>
<span id="cb96-13"><a href="copy-number-analyses.html#cb96-13" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb96-14"><a href="copy-number-analyses.html#cb96-14" tabindex="-1"></a><span class="co"># Percentage of cells cut-off </span></span>
<span id="cb96-15"><a href="copy-number-analyses.html#cb96-15" tabindex="-1"></a>pcc <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb96-16"><a href="copy-number-analyses.html#cb96-16" tabindex="-1"></a></span>
<span id="cb96-17"><a href="copy-number-analyses.html#cb96-17" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb96-18"><a href="copy-number-analyses.html#cb96-18" tabindex="-1"></a><span class="co"># RUNS #</span></span>
<span id="cb96-19"><a href="copy-number-analyses.html#cb96-19" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb96-20"><a href="copy-number-analyses.html#cb96-20" tabindex="-1"></a></span>
<span id="cb96-21"><a href="copy-number-analyses.html#cb96-21" tabindex="-1"></a><span class="co"># HUB183 #</span></span>
<span id="cb96-22"><a href="copy-number-analyses.html#cb96-22" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb96-23"><a href="copy-number-analyses.html#cb96-23" tabindex="-1"></a><span class="co"># Extract HUB183</span></span>
<span id="cb96-24"><a href="copy-number-analyses.html#cb96-24" tabindex="-1"></a>hub183 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub183&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb96-25"><a href="copy-number-analyses.html#cb96-25" tabindex="-1"></a></span>
<span id="cb96-26"><a href="copy-number-analyses.html#cb96-26" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb96-27"><a href="copy-number-analyses.html#cb96-27" tabindex="-1"></a>hub183_prerad_subclones <span class="ot">&lt;-</span> hub183[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub183)]</span>
<span id="cb96-28"><a href="copy-number-analyses.html#cb96-28" tabindex="-1"></a></span>
<span id="cb96-29"><a href="copy-number-analyses.html#cb96-29" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb96-30"><a href="copy-number-analyses.html#cb96-30" tabindex="-1"></a>hub183_postrad_a <span class="ot">&lt;-</span> hub183[<span class="fu">grepl</span>(<span class="st">&#39;hub183_postrad_a&#39;</span>, hub183)]</span>
<span id="cb96-31"><a href="copy-number-analyses.html#cb96-31" tabindex="-1"></a></span>
<span id="cb96-32"><a href="copy-number-analyses.html#cb96-32" tabindex="-1"></a><span class="co"># no new genomic aberrations</span></span>
<span id="cb96-33"><a href="copy-number-analyses.html#cb96-33" tabindex="-1"></a>hub183_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb96-34"><a href="copy-number-analyses.html#cb96-34" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub183_postrad_a],</span>
<span id="cb96-35"><a href="copy-number-analyses.html#cb96-35" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub183_prerad_subclones],</span>
<span id="cb96-36"><a href="copy-number-analyses.html#cb96-36" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb96-37"><a href="copy-number-analyses.html#cb96-37" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb96-38"><a href="copy-number-analyses.html#cb96-38" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb96-39"><a href="copy-number-analyses.html#cb96-39" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb96-40"><a href="copy-number-analyses.html#cb96-40" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb96-41"><a href="copy-number-analyses.html#cb96-41" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="copy-number-analyses.html#cb98-1" tabindex="-1"></a><span class="co"># HUB005 #</span></span>
<span id="cb98-2"><a href="copy-number-analyses.html#cb98-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb98-3"><a href="copy-number-analyses.html#cb98-3" tabindex="-1"></a></span>
<span id="cb98-4"><a href="copy-number-analyses.html#cb98-4" tabindex="-1"></a><span class="co"># Extract HUB005</span></span>
<span id="cb98-5"><a href="copy-number-analyses.html#cb98-5" tabindex="-1"></a>hub005 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub005&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb98-6"><a href="copy-number-analyses.html#cb98-6" tabindex="-1"></a></span>
<span id="cb98-7"><a href="copy-number-analyses.html#cb98-7" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb98-8"><a href="copy-number-analyses.html#cb98-8" tabindex="-1"></a>hub005_prerad_subclones <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub005)]</span>
<span id="cb98-9"><a href="copy-number-analyses.html#cb98-9" tabindex="-1"></a></span>
<span id="cb98-10"><a href="copy-number-analyses.html#cb98-10" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb98-11"><a href="copy-number-analyses.html#cb98-11" tabindex="-1"></a>hub005_postrad_a.a <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;hub005_postrad_a.a&#39;</span>, hub005)]</span>
<span id="cb98-12"><a href="copy-number-analyses.html#cb98-12" tabindex="-1"></a>hub005_postrad_b <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;hub005_postrad_b&#39;</span>, hub005)]</span>
<span id="cb98-13"><a href="copy-number-analyses.html#cb98-13" tabindex="-1"></a></span>
<span id="cb98-14"><a href="copy-number-analyses.html#cb98-14" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb98-15"><a href="copy-number-analyses.html#cb98-15" tabindex="-1"></a>hub005_postrad_a.a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb98-16"><a href="copy-number-analyses.html#cb98-16" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub005_postrad_a.a],</span>
<span id="cb98-17"><a href="copy-number-analyses.html#cb98-17" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub005_prerad_subclones],</span>
<span id="cb98-18"><a href="copy-number-analyses.html#cb98-18" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb98-19"><a href="copy-number-analyses.html#cb98-19" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb98-20"><a href="copy-number-analyses.html#cb98-20" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb98-21"><a href="copy-number-analyses.html#cb98-21" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb98-22"><a href="copy-number-analyses.html#cb98-22" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb98-23"><a href="copy-number-analyses.html#cb98-23" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="copy-number-analyses.html#cb100-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb100-2"><a href="copy-number-analyses.html#cb100-2" tabindex="-1"></a>hub005_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb100-3"><a href="copy-number-analyses.html#cb100-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub005_postrad_b],</span>
<span id="cb100-4"><a href="copy-number-analyses.html#cb100-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub005_prerad_subclones],</span>
<span id="cb100-5"><a href="copy-number-analyses.html#cb100-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb100-6"><a href="copy-number-analyses.html#cb100-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb100-7"><a href="copy-number-analyses.html#cb100-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb100-8"><a href="copy-number-analyses.html#cb100-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb100-9"><a href="copy-number-analyses.html#cb100-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb100-10"><a href="copy-number-analyses.html#cb100-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="copy-number-analyses.html#cb102-1" tabindex="-1"></a><span class="co"># HUB015 #</span></span>
<span id="cb102-2"><a href="copy-number-analyses.html#cb102-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb102-3"><a href="copy-number-analyses.html#cb102-3" tabindex="-1"></a></span>
<span id="cb102-4"><a href="copy-number-analyses.html#cb102-4" tabindex="-1"></a><span class="co"># Extract HUB015</span></span>
<span id="cb102-5"><a href="copy-number-analyses.html#cb102-5" tabindex="-1"></a>hub015 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub015&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb102-6"><a href="copy-number-analyses.html#cb102-6" tabindex="-1"></a></span>
<span id="cb102-7"><a href="copy-number-analyses.html#cb102-7" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb102-8"><a href="copy-number-analyses.html#cb102-8" tabindex="-1"></a>hub015_prerad_subclones <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub015)]</span>
<span id="cb102-9"><a href="copy-number-analyses.html#cb102-9" tabindex="-1"></a></span>
<span id="cb102-10"><a href="copy-number-analyses.html#cb102-10" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb102-11"><a href="copy-number-analyses.html#cb102-11" tabindex="-1"></a>hub015_postrad_a.a <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_a.a&#39;</span>, hub015)]</span>
<span id="cb102-12"><a href="copy-number-analyses.html#cb102-12" tabindex="-1"></a>hub015_postrad_a.b <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_a.b&#39;</span>, hub015)]</span>
<span id="cb102-13"><a href="copy-number-analyses.html#cb102-13" tabindex="-1"></a>hub015_postrad_b <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_b&#39;</span>, hub015)]</span>
<span id="cb102-14"><a href="copy-number-analyses.html#cb102-14" tabindex="-1"></a></span>
<span id="cb102-15"><a href="copy-number-analyses.html#cb102-15" tabindex="-1"></a><span class="co"># hub015_postrad_a.a: new deletion on 4q</span></span>
<span id="cb102-16"><a href="copy-number-analyses.html#cb102-16" tabindex="-1"></a>hub015_postrad_a.a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb102-17"><a href="copy-number-analyses.html#cb102-17" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub015_postrad_a.a],</span>
<span id="cb102-18"><a href="copy-number-analyses.html#cb102-18" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub015_prerad_subclones],</span>
<span id="cb102-19"><a href="copy-number-analyses.html#cb102-19" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb102-20"><a href="copy-number-analyses.html#cb102-20" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb102-21"><a href="copy-number-analyses.html#cb102-21" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb102-22"><a href="copy-number-analyses.html#cb102-22" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb102-23"><a href="copy-number-analyses.html#cb102-23" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb102-24"><a href="copy-number-analyses.html#cb102-24" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="copy-number-analyses.html#cb104-1" tabindex="-1"></a><span class="co"># HUB015 has a new deletion following radiation in 4q</span></span>
<span id="cb104-2"><a href="copy-number-analyses.html#cb104-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(<span class="fu">get_cytoband_coverage</span>(hub015_postrad_a.a<span class="sc">$</span>unique_resistant_del)[[<span class="dv">1</span>]]))</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e1493842c313f0098104" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e1493842c313f0098104">{"x":{"filter":"none","vertical":false,"data":[["1"],["4"],[75676850],[95450367],[19773518],["*"],[["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="copy-number-analyses.html#cb105-1" tabindex="-1"></a><span class="co"># This deletion contains the following oncogenes</span></span>
<span id="cb105-2"><a href="copy-number-analyses.html#cb105-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">get_oncogenes</span>(<span class="fu">get_cytoband_coverage</span>(hub015_postrad_a.a<span class="sc">$</span>unique_resistant_del)[[<span class="dv">1</span>]]))</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ff77392d4671434cf6a5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ff77392d4671434cf6a5">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["AFF1","PTPN13"],["AF4/FMR2 family, member 1","protein tyrosine phosphatase, non-receptor type 13"],[4299,5783],["4:86935002-87141054","4:86594315-86815171"],[1,1],["Yes","Yes"],[22.1,21.3],["yes","yes"],[null,"yes"],["AL","lung, NSCLC, gastric, peritoneal carcinomatosis"],[null,"hepatocellular"],[null,"Familial Hepatocellular Carcinoma"],["L","E"],["Dom","Rec"],["fusion","TSG"],["T","Mis, N"],["KMT2A",null],[null,null],[null,null],["4299,AF-4,AF4,AFF1,ENSG00000172493.20,MLLT2,P51825,PBM1","5783,ENSG00000163629.12,PTP-BAS,PTP-BL,PTP1E,PTPL1,PTPN13,Q12923"],["chr4","chr4"],[75676850,75676850],[95450367,95450367],[19773518,19773518],["*","*"],[["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"],["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"]],[4,4],[86935002,86594315],[87141039,86815171]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene Symbol<\/th>\n      <th>Name<\/th>\n      <th>Entrez GeneId<\/th>\n      <th>Genome Location<\/th>\n      <th>Tier<\/th>\n      <th>Hallmark<\/th>\n      <th>Chr Band<\/th>\n      <th>Somatic<\/th>\n      <th>Germline<\/th>\n      <th>Tumour Types(Somatic)<\/th>\n      <th>Tumour Types(Germline)<\/th>\n      <th>Cancer Syndrome<\/th>\n      <th>Tissue Type<\/th>\n      <th>Molecular Genetics<\/th>\n      <th>Role in Cancer<\/th>\n      <th>Mutation Types<\/th>\n      <th>Translocation Partner<\/th>\n      <th>Other Germline Mut<\/th>\n      <th>Other Syndrome<\/th>\n      <th>Synonyms<\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n      <th>chromosome_name<\/th>\n      <th>start_position<\/th>\n      <th>end_position<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,5,7,22,23,24,27,28,29]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="copy-number-analyses.html#cb106-1" tabindex="-1"></a><span class="co"># hub015_postrad_a.a: no new amplificatoins no new deletions</span></span>
<span id="cb106-2"><a href="copy-number-analyses.html#cb106-2" tabindex="-1"></a>hub015_postrad_a.b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb106-3"><a href="copy-number-analyses.html#cb106-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub015_postrad_a.b],</span>
<span id="cb106-4"><a href="copy-number-analyses.html#cb106-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub015_prerad_subclones],</span>
<span id="cb106-5"><a href="copy-number-analyses.html#cb106-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb106-6"><a href="copy-number-analyses.html#cb106-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb106-7"><a href="copy-number-analyses.html#cb106-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb106-8"><a href="copy-number-analyses.html#cb106-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb106-9"><a href="copy-number-analyses.html#cb106-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb106-10"><a href="copy-number-analyses.html#cb106-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="copy-number-analyses.html#cb108-1" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb108-2"><a href="copy-number-analyses.html#cb108-2" tabindex="-1"></a></span>
<span id="cb108-3"><a href="copy-number-analyses.html#cb108-3" tabindex="-1"></a><span class="co"># HUB106 #</span></span>
<span id="cb108-4"><a href="copy-number-analyses.html#cb108-4" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb108-5"><a href="copy-number-analyses.html#cb108-5" tabindex="-1"></a></span>
<span id="cb108-6"><a href="copy-number-analyses.html#cb108-6" tabindex="-1"></a><span class="co"># Extract HUB106</span></span>
<span id="cb108-7"><a href="copy-number-analyses.html#cb108-7" tabindex="-1"></a>hub106 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub106&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb108-8"><a href="copy-number-analyses.html#cb108-8" tabindex="-1"></a></span>
<span id="cb108-9"><a href="copy-number-analyses.html#cb108-9" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb108-10"><a href="copy-number-analyses.html#cb108-10" tabindex="-1"></a>hub106_prerad_subclones <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub106)]</span>
<span id="cb108-11"><a href="copy-number-analyses.html#cb108-11" tabindex="-1"></a></span>
<span id="cb108-12"><a href="copy-number-analyses.html#cb108-12" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb108-13"><a href="copy-number-analyses.html#cb108-13" tabindex="-1"></a>hub106_postrad_a <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_a&#39;</span>, hub106)]</span>
<span id="cb108-14"><a href="copy-number-analyses.html#cb108-14" tabindex="-1"></a>hub106_postrad_b <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_b&#39;</span>, hub106)]</span>
<span id="cb108-15"><a href="copy-number-analyses.html#cb108-15" tabindex="-1"></a>hub106_postrad_c <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_c&#39;</span>, hub106)]</span>
<span id="cb108-16"><a href="copy-number-analyses.html#cb108-16" tabindex="-1"></a></span>
<span id="cb108-17"><a href="copy-number-analyses.html#cb108-17" tabindex="-1"></a></span>
<span id="cb108-18"><a href="copy-number-analyses.html#cb108-18" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb108-19"><a href="copy-number-analyses.html#cb108-19" tabindex="-1"></a>hub106_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb108-20"><a href="copy-number-analyses.html#cb108-20" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_a],</span>
<span id="cb108-21"><a href="copy-number-analyses.html#cb108-21" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb108-22"><a href="copy-number-analyses.html#cb108-22" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb108-23"><a href="copy-number-analyses.html#cb108-23" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb108-24"><a href="copy-number-analyses.html#cb108-24" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb108-25"><a href="copy-number-analyses.html#cb108-25" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb108-26"><a href="copy-number-analyses.html#cb108-26" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb108-27"><a href="copy-number-analyses.html#cb108-27" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="copy-number-analyses.html#cb110-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb110-2"><a href="copy-number-analyses.html#cb110-2" tabindex="-1"></a>hub106_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb110-3"><a href="copy-number-analyses.html#cb110-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_b],</span>
<span id="cb110-4"><a href="copy-number-analyses.html#cb110-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb110-5"><a href="copy-number-analyses.html#cb110-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb110-6"><a href="copy-number-analyses.html#cb110-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb110-7"><a href="copy-number-analyses.html#cb110-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb110-8"><a href="copy-number-analyses.html#cb110-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb110-9"><a href="copy-number-analyses.html#cb110-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb110-10"><a href="copy-number-analyses.html#cb110-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="copy-number-analyses.html#cb112-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb112-2"><a href="copy-number-analyses.html#cb112-2" tabindex="-1"></a>hub106_postrad_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb112-3"><a href="copy-number-analyses.html#cb112-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_c],</span>
<span id="cb112-4"><a href="copy-number-analyses.html#cb112-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb112-5"><a href="copy-number-analyses.html#cb112-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb112-6"><a href="copy-number-analyses.html#cb112-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb112-7"><a href="copy-number-analyses.html#cb112-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb112-8"><a href="copy-number-analyses.html#cb112-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb112-9"><a href="copy-number-analyses.html#cb112-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb112-10"><a href="copy-number-analyses.html#cb112-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="copy-number-analyses.html#cb114-1" tabindex="-1"></a><span class="co"># HUB062 #</span></span>
<span id="cb114-2"><a href="copy-number-analyses.html#cb114-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb114-3"><a href="copy-number-analyses.html#cb114-3" tabindex="-1"></a>hub062 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub062&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb114-4"><a href="copy-number-analyses.html#cb114-4" tabindex="-1"></a></span>
<span id="cb114-5"><a href="copy-number-analyses.html#cb114-5" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb114-6"><a href="copy-number-analyses.html#cb114-6" tabindex="-1"></a>hub062_prerad_subclones <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub062)]</span>
<span id="cb114-7"><a href="copy-number-analyses.html#cb114-7" tabindex="-1"></a></span>
<span id="cb114-8"><a href="copy-number-analyses.html#cb114-8" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb114-9"><a href="copy-number-analyses.html#cb114-9" tabindex="-1"></a>hub062_postrad_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_a&#39;</span>, hub062)]</span>
<span id="cb114-10"><a href="copy-number-analyses.html#cb114-10" tabindex="-1"></a>hub062_postrad_b <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_b&#39;</span>, hub062)]</span>
<span id="cb114-11"><a href="copy-number-analyses.html#cb114-11" tabindex="-1"></a>hub062_postrad_c <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c&#39;</span>, hub062)]</span>
<span id="cb114-12"><a href="copy-number-analyses.html#cb114-12" tabindex="-1"></a>hub062biological_postrad_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062biological_postrad_a&#39;</span>, hub062)]</span>
<span id="cb114-13"><a href="copy-number-analyses.html#cb114-13" tabindex="-1"></a>hub062_postrad_c2_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c2_a&#39;</span>, hub062)]</span>
<span id="cb114-14"><a href="copy-number-analyses.html#cb114-14" tabindex="-1"></a>hub062_postrad_c2_c <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c2_c&#39;</span>, hub062)]</span>
<span id="cb114-15"><a href="copy-number-analyses.html#cb114-15" tabindex="-1"></a></span>
<span id="cb114-16"><a href="copy-number-analyses.html#cb114-16" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb114-17"><a href="copy-number-analyses.html#cb114-17" tabindex="-1"></a>hub062_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb114-18"><a href="copy-number-analyses.html#cb114-18" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_a],</span>
<span id="cb114-19"><a href="copy-number-analyses.html#cb114-19" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb114-20"><a href="copy-number-analyses.html#cb114-20" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb114-21"><a href="copy-number-analyses.html#cb114-21" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb114-22"><a href="copy-number-analyses.html#cb114-22" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb114-23"><a href="copy-number-analyses.html#cb114-23" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb114-24"><a href="copy-number-analyses.html#cb114-24" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb114-25"><a href="copy-number-analyses.html#cb114-25" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="copy-number-analyses.html#cb116-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb116-2"><a href="copy-number-analyses.html#cb116-2" tabindex="-1"></a>hub062_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb116-3"><a href="copy-number-analyses.html#cb116-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_b],</span>
<span id="cb116-4"><a href="copy-number-analyses.html#cb116-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb116-5"><a href="copy-number-analyses.html#cb116-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb116-6"><a href="copy-number-analyses.html#cb116-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb116-7"><a href="copy-number-analyses.html#cb116-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb116-8"><a href="copy-number-analyses.html#cb116-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb116-9"><a href="copy-number-analyses.html#cb116-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb116-10"><a href="copy-number-analyses.html#cb116-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="copy-number-analyses.html#cb118-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb118-2"><a href="copy-number-analyses.html#cb118-2" tabindex="-1"></a>hub062_postrad_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb118-3"><a href="copy-number-analyses.html#cb118-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c],</span>
<span id="cb118-4"><a href="copy-number-analyses.html#cb118-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb118-5"><a href="copy-number-analyses.html#cb118-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb118-6"><a href="copy-number-analyses.html#cb118-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb118-7"><a href="copy-number-analyses.html#cb118-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb118-8"><a href="copy-number-analyses.html#cb118-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb118-9"><a href="copy-number-analyses.html#cb118-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb118-10"><a href="copy-number-analyses.html#cb118-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="copy-number-analyses.html#cb120-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb120-2"><a href="copy-number-analyses.html#cb120-2" tabindex="-1"></a>hub062biological_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb120-3"><a href="copy-number-analyses.html#cb120-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062biological_postrad_a],</span>
<span id="cb120-4"><a href="copy-number-analyses.html#cb120-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb120-5"><a href="copy-number-analyses.html#cb120-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb120-6"><a href="copy-number-analyses.html#cb120-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb120-7"><a href="copy-number-analyses.html#cb120-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb120-8"><a href="copy-number-analyses.html#cb120-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb120-9"><a href="copy-number-analyses.html#cb120-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb120-10"><a href="copy-number-analyses.html#cb120-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="copy-number-analyses.html#cb122-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb122-2"><a href="copy-number-analyses.html#cb122-2" tabindex="-1"></a>hub062_postrad_c2_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb122-3"><a href="copy-number-analyses.html#cb122-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c2_a],</span>
<span id="cb122-4"><a href="copy-number-analyses.html#cb122-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb122-5"><a href="copy-number-analyses.html#cb122-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb122-6"><a href="copy-number-analyses.html#cb122-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb122-7"><a href="copy-number-analyses.html#cb122-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb122-8"><a href="copy-number-analyses.html#cb122-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb122-9"><a href="copy-number-analyses.html#cb122-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb122-10"><a href="copy-number-analyses.html#cb122-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="copy-number-analyses.html#cb124-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb124-2"><a href="copy-number-analyses.html#cb124-2" tabindex="-1"></a>hub062_postrad_c2_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb124-3"><a href="copy-number-analyses.html#cb124-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c2_c],</span>
<span id="cb124-4"><a href="copy-number-analyses.html#cb124-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb124-5"><a href="copy-number-analyses.html#cb124-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb124-6"><a href="copy-number-analyses.html#cb124-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb124-7"><a href="copy-number-analyses.html#cb124-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb124-8"><a href="copy-number-analyses.html#cb124-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb124-9"><a href="copy-number-analyses.html#cb124-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb124-10"><a href="copy-number-analyses.html#cb124-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="subclonal-dynamics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="copy-number-pattern-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/edit/main/04-cna.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/blob/main/04-cna.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
