<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</title>
  <meta name="description" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Copy number analyses | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  
  
  

<meta name="author" content="Daan Andel" />


<meta name="date" content="2023-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="subclonal-dynamics.html"/>
<link rel="next" href="copy-number-pattern-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Subclonal Evolution Paper</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html"><i class="fa fa-check"></i><b>2</b> In vitro radiation response and clinical data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#progression-free-survival"><i class="fa fa-check"></i><b>2.1</b> Progression-free survival</a></li>
<li class="chapter" data-level="2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#in-vitro-dose-response"><i class="fa fa-check"></i><b>2.2</b> In vitro dose-response</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#dose-response-plot"><i class="fa fa-check"></i><b>2.2.1</b> Dose-response plot</a></li>
<li class="chapter" data-level="2.2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#rauc-across-multiple-experiments"><i class="fa fa-check"></i><b>2.2.2</b> rAUC across multiple experiments</a></li>
<li class="chapter" data-level="2.2.3" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#flow-cytometry-based-measurement-of-apoptotic-cells-with-sub-g1-dna-content"><i class="fa fa-check"></i><b>2.2.3</b> Flow cytometry-based measurement of (apoptotic) cells with sub-G1 DNA content</a></li>
<li class="chapter" data-level="2.2.4" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#recovery-statistics-after-10-gy-radiation"><i class="fa fa-check"></i><b>2.2.4</b> Recovery statistics after 10 Gy radiation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html"><i class="fa fa-check"></i><b>3</b> Subclonal dynamics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#aneufinder-run"><i class="fa fa-check"></i><b>3.1</b> AneuFinder run</a></li>
<li class="chapter" data-level="3.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#quality-selection"><i class="fa fa-check"></i><b>3.2</b> Quality selection</a></li>
<li class="chapter" data-level="3.3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#summary-statistics-of-copy-number-alterations"><i class="fa fa-check"></i><b>3.3</b> Summary statistics of copy number alterations</a></li>
<li class="chapter" data-level="3.4" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#pca-and-k-means-plots"><i class="fa fa-check"></i><b>3.4</b> PCA and K-means plots</a></li>
<li class="chapter" data-level="3.5" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-selection"><i class="fa fa-check"></i><b>3.5</b> Subclone selection</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#selecting-subclones"><i class="fa fa-check"></i><b>3.5.1</b> Selecting subclones</a></li>
<li class="chapter" data-level="3.5.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-statistics"><i class="fa fa-check"></i><b>3.5.2</b> Subclone statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#single-cell-copy-number-heatmap-plots"><i class="fa fa-check"></i><b>3.6</b> Single cell copy number heatmap plots</a></li>
<li class="chapter" data-level="3.7" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#fishers-exact-test"><i class="fa fa-check"></i><b>3.7</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html"><i class="fa fa-check"></i><b>4</b> Copy number analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid"><i class="fa fa-check"></i><b>4.1</b> Colorectal cancer-specific arm-level copy number alterations per organoid</a></li>
<li class="chapter" data-level="4.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-per-subclone"><i class="fa fa-check"></i><b>4.2</b> Copy number alterations per subclone</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a"><i class="fa fa-check"></i><b>4.2.1</b> Copy number alterations specific to HUB062A</a></li>
<li class="chapter" data-level="4.2.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones"><i class="fa fa-check"></i><b>4.2.2</b> Copy number alterations specific to resistant subclones</a></li>
<li class="chapter" data-level="4.2.3" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#newly-acquired-copy-number-alterations"><i class="fa fa-check"></i><b>4.2.3</b> Newly acquired copy number alterations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html"><i class="fa fa-check"></i><b>5</b> Copy number pattern analyses</a></li>
<li class="chapter" data-level="6" data-path="source-code.html"><a href="source-code.html"><i class="fa fa-check"></i><b>6</b> Source code</a>
<ul>
<li class="chapter" data-level="6.1" data-path="source-code.html"><a href="source-code.html#auc_rad-script-auc_fun-function"><i class="fa fa-check"></i><b>6.1</b> AUC_rad script: ‘AUC_fun’ function</a></li>
<li class="chapter" data-level="6.2" data-path="source-code.html"><a href="source-code.html#fitter_rad-script-fitter-function"><i class="fa fa-check"></i><b>6.2</b> fitter_rad script: ‘fitter’ function</a></li>
<li class="chapter" data-level="6.3" data-path="source-code.html"><a href="source-code.html#ic50_rad-script-ic50_fun-function"><i class="fa fa-check"></i><b>6.3</b> ic50_rad script: ‘IC50_fun’ function</a></li>
<li class="chapter" data-level="6.4" data-path="source-code.html"><a href="source-code.html#relativize_dr-script-relative-function"><i class="fa fa-check"></i><b>6.4</b> relativize_dr script: ‘relative’ function</a></li>
<li class="chapter" data-level="6.5" data-path="source-code.html"><a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions"><i class="fa fa-check"></i><b>6.5</b> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions</a></li>
<li class="chapter" data-level="6.6" data-path="source-code.html"><a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions"><i class="fa fa-check"></i><b>6.6</b> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions</a></li>
<li class="chapter" data-level="6.7" data-path="source-code.html"><a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions"><i class="fa fa-check"></i><b>6.7</b> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions</a></li>
<li class="chapter" data-level="6.8" data-path="source-code.html"><a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function"><i class="fa fa-check"></i><b>6.8</b> plot_genomeheatmap script: ‘genomeheatmap’ function</a></li>
<li class="chapter" data-level="6.9" data-path="source-code.html"><a href="source-code.html#fisher_test-script-perform_fisher-function"><i class="fa fa-check"></i><b>6.9</b> fisher_test script: ‘perform_fisher’ function</a></li>
<li class="chapter" data-level="6.10" data-path="source-code.html"><a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function"><i class="fa fa-check"></i><b>6.10</b> find_specific_cna_sc script: ‘find_specific_cna_sc’ function</a></li>
<li class="chapter" data-level="6.11" data-path="source-code.html"><a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function"><i class="fa fa-check"></i><b>6.11</b> get_cytoband_coverage script: ‘get_cytoband_coverage’ function</a></li>
<li class="chapter" data-level="6.12" data-path="source-code.html"><a href="source-code.html#get_genes-script-get_genes-functoin"><i class="fa fa-check"></i><b>6.12</b> get_genes script: ‘get_genes’ functoin</a></li>
<li class="chapter" data-level="6.13" data-path="source-code.html"><a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function"><i class="fa fa-check"></i><b>6.13</b> get_oncogenes scripts: ‘get_oncogenes’ function</a></li>
<li class="chapter" data-level="6.14" data-path="source-code.html"><a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function"><i class="fa fa-check"></i><b>6.14</b> plot_genes_karyotype script: ‘plot_genes_karyotype’ function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="copy-number-analyses" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Copy number analyses<a href="copy-number-analyses.html#copy-number-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="copy-number-analyses.html#cb1-1" tabindex="-1"></a><span class="co"># Libraries and sources</span></span>
<span id="cb1-2"><a href="copy-number-analyses.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="copy-number-analyses.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-4"><a href="copy-number-analyses.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb1-5"><a href="copy-number-analyses.html#cb1-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/find_specific_cna_sc.R&#39;</span>)</span>
<span id="cb1-6"><a href="copy-number-analyses.html#cb1-6" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_cytoband_coverage.R&#39;</span>)</span>
<span id="cb1-7"><a href="copy-number-analyses.html#cb1-7" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_genes.R&#39;</span>)</span>
<span id="cb1-8"><a href="copy-number-analyses.html#cb1-8" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/get_oncogenes.R&#39;</span>)</span>
<span id="cb1-9"><a href="copy-number-analyses.html#cb1-9" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&#39;R/plot_genes_karyotype.R&#39;</span>)</span></code></pre></div>
<div id="colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Colorectal cancer-specific arm-level copy number alterations per organoid<a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="copy-number-analyses.html#cb2-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb2-2"><a href="copy-number-analyses.html#cb2-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb2-3"><a href="copy-number-analyses.html#cb2-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb2-4"><a href="copy-number-analyses.html#cb2-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/subclonal_dynamics/man_select_files_edivisive.rda&#39;</span>)</span>
<span id="cb2-5"><a href="copy-number-analyses.html#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="copy-number-analyses.html#cb2-6" tabindex="-1"></a><span class="co"># !!! GITHUB SPECIFIC CODE !!! #</span></span>
<span id="cb2-7"><a href="copy-number-analyses.html#cb2-7" tabindex="-1"></a><span class="co"># adding &#39;../cna_analysis/&#39; to each path within man_select_files_edivisive </span></span>
<span id="cb2-8"><a href="copy-number-analyses.html#cb2-8" tabindex="-1"></a>man_select_files_edivisive <span class="ot">&lt;-</span> <span class="fu">lapply</span>(man_select_files_edivisive, <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">&quot;../cna_analysis/&quot;</span>, x))</span>
<span id="cb2-9"><a href="copy-number-analyses.html#cb2-9" tabindex="-1"></a></span>
<span id="cb2-10"><a href="copy-number-analyses.html#cb2-10" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb2-11"><a href="copy-number-analyses.html#cb2-11" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb2-12"><a href="copy-number-analyses.html#cb2-12" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb2-13"><a href="copy-number-analyses.html#cb2-13" tabindex="-1"></a><span class="co"># extract only prerad clines</span></span>
<span id="cb2-14"><a href="copy-number-analyses.html#cb2-14" tabindex="-1"></a><span class="co"># prerad clines</span></span>
<span id="cb2-15"><a href="copy-number-analyses.html#cb2-15" tabindex="-1"></a>kra_list <span class="ot">&lt;-</span></span>
<span id="cb2-16"><a href="copy-number-analyses.html#cb2-16" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&quot;kra003&quot;</span>,</span>
<span id="cb2-17"><a href="copy-number-analyses.html#cb2-17" tabindex="-1"></a>    <span class="st">&quot;kra005&quot;</span>,</span>
<span id="cb2-18"><a href="copy-number-analyses.html#cb2-18" tabindex="-1"></a>    <span class="st">&quot;kra007&quot;</span>,</span>
<span id="cb2-19"><a href="copy-number-analyses.html#cb2-19" tabindex="-1"></a>    <span class="st">&quot;kra009&quot;</span>,</span>
<span id="cb2-20"><a href="copy-number-analyses.html#cb2-20" tabindex="-1"></a>    <span class="st">&quot;kra023&quot;</span>,</span>
<span id="cb2-21"><a href="copy-number-analyses.html#cb2-21" tabindex="-1"></a>    <span class="st">&quot;kra025&quot;</span>,</span>
<span id="cb2-22"><a href="copy-number-analyses.html#cb2-22" tabindex="-1"></a>    <span class="st">&quot;kra027&quot;</span>)</span>
<span id="cb2-23"><a href="copy-number-analyses.html#cb2-23" tabindex="-1"></a></span>
<span id="cb2-24"><a href="copy-number-analyses.html#cb2-24" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb2-25"><a href="copy-number-analyses.html#cb2-25" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb2-26"><a href="copy-number-analyses.html#cb2-26" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb2-27"><a href="copy-number-analyses.html#cb2-27" tabindex="-1"></a></span>
<span id="cb2-28"><a href="copy-number-analyses.html#cb2-28" tabindex="-1"></a><span class="do">## finding common CNAs within each subclone for each organoids.</span></span>
<span id="cb2-29"><a href="copy-number-analyses.html#cb2-29" tabindex="-1"></a>common_cnas <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb2-30"><a href="copy-number-analyses.html#cb2-30" tabindex="-1"></a>  <span class="at">list_resistant =</span> man_select_files_edivisive[kra_list],</span>
<span id="cb2-31"><a href="copy-number-analyses.html#cb2-31" tabindex="-1"></a>  <span class="co"># make sure to input this as a list!</span></span>
<span id="cb2-32"><a href="copy-number-analyses.html#cb2-32" tabindex="-1"></a>  <span class="at">list_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-33"><a href="copy-number-analyses.html#cb2-33" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> <span class="fl">0.8</span>,</span>
<span id="cb2-34"><a href="copy-number-analyses.html#cb2-34" tabindex="-1"></a>  <span class="co"># CNA needs to be in 80% of the cells or more</span></span>
<span id="cb2-35"><a href="copy-number-analyses.html#cb2-35" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-36"><a href="copy-number-analyses.html#cb2-36" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-37"><a href="copy-number-analyses.html#cb2-37" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="cn">NULL</span>,</span>
<span id="cb2-38"><a href="copy-number-analyses.html#cb2-38" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> T</span>
<span id="cb2-39"><a href="copy-number-analyses.html#cb2-39" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 80% of single cells within each resistant subclone, and by % of single cells within each sensitive subclone&quot;</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="copy-number-analyses.html#cb4-1" tabindex="-1"></a><span class="do">## mapping CNA to cytoband location</span></span>
<span id="cb4-2"><a href="copy-number-analyses.html#cb4-2" tabindex="-1"></a><span class="co"># for amplifications</span></span>
<span id="cb4-3"><a href="copy-number-analyses.html#cb4-3" tabindex="-1"></a>amplifications <span class="ot">&lt;-</span></span>
<span id="cb4-4"><a href="copy-number-analyses.html#cb4-4" tabindex="-1"></a>  <span class="fu">lapply</span>(common_cnas<span class="sc">$</span>amplifications, get_cytoband_coverage)</span>
<span id="cb4-5"><a href="copy-number-analyses.html#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="copy-number-analyses.html#cb4-6" tabindex="-1"></a><span class="co"># Extract frequency of arm level amplifications n indicates the how many</span></span>
<span id="cb4-7"><a href="copy-number-analyses.html#cb4-7" tabindex="-1"></a><span class="co"># organoids contain the CNA</span></span>
<span id="cb4-8"><a href="copy-number-analyses.html#cb4-8" tabindex="-1"></a>arm_amp <span class="ot">&lt;-</span> <span class="fu">lapply</span>(amplifications, <span class="cf">function</span>(x) {</span>
<span id="cb4-9"><a href="copy-number-analyses.html#cb4-9" tabindex="-1"></a>  <span class="fu">return</span>(x[[<span class="dv">2</span>]])</span>
<span id="cb4-10"><a href="copy-number-analyses.html#cb4-10" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="copy-number-analyses.html#cb4-11" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;cline&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="copy-number-analyses.html#cb4-12" tabindex="-1"></a>  <span class="fu">group_by</span>(cline, chromosome, arm) <span class="sc">%&gt;%</span></span>
<span id="cb4-13"><a href="copy-number-analyses.html#cb4-13" tabindex="-1"></a>  <span class="fu">count</span>(percentage_arm_affected <span class="sc">&gt;</span> <span class="fl">0.8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-14"><a href="copy-number-analyses.html#cb4-14" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb4-15"><a href="copy-number-analyses.html#cb4-15" tabindex="-1"></a>  <span class="fu">ungroup</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb4-16"><a href="copy-number-analyses.html#cb4-16" tabindex="-1"></a>  <span class="fu">count</span>(chromosome) <span class="sc">%&gt;%</span></span>
<span id="cb4-17"><a href="copy-number-analyses.html#cb4-17" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb4-18"><a href="copy-number-analyses.html#cb4-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> n <span class="sc">/</span> <span class="dv">7</span>)</span>
<span id="cb4-19"><a href="copy-number-analyses.html#cb4-19" tabindex="-1"></a></span>
<span id="cb4-20"><a href="copy-number-analyses.html#cb4-20" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(arm_amp, <span class="at">caption =</span> <span class="st">&quot;Arm-level amplifications in organoids&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-39de00e5014cd1350da1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-39de00e5014cd1350da1">{"x":{"filter":"none","vertical":false,"caption":"<caption>Arm-level amplifications in organoids<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44"],["1","20","20","8","3","2","16","16","13","13","12","10","10","9","8","7","7","5","2","1","19","19","17","17","12","11","9","6","6","5","3","22","22","14","14","11","X","4","21","21","X","4","18","18"],["q","q","p","q","q","q","p","q","p","q","q","p","q","q","p","p","q","p","p","p","p","q","p","q","p","q","p","p","q","q","p","p","q","p","q","p","q","q","p","q","p","p","p","q"],[7,7,6,6,6,6,5,5,5,5,5,5,5,5,5,5,5,5,5,4,4,4,4,4,4,4,4,4,4,4,4,3,3,3,3,3,3,3,2,2,2,2,1,1],[1,1,0.8571428571428571,0.8571428571428571,0.8571428571428571,0.8571428571428571,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.7142857142857143,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.5714285714285714,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.4285714285714285,0.2857142857142857,0.2857142857142857,0.2857142857142857,0.2857142857142857,0.1428571428571428,0.1428571428571428]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chromosome<\/th>\n      <th>arm<\/th>\n      <th>n<\/th>\n      <th>freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="copy-number-analyses.html#cb5-1" tabindex="-1"></a><span class="co"># for deletions</span></span>
<span id="cb5-2"><a href="copy-number-analyses.html#cb5-2" tabindex="-1"></a>deletions <span class="ot">&lt;-</span> <span class="fu">lapply</span>(common_cnas<span class="sc">$</span>deletions, get_cytoband_coverage)</span>
<span id="cb5-3"><a href="copy-number-analyses.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="copy-number-analyses.html#cb5-4" tabindex="-1"></a><span class="co"># Extract frequency of arm level deletions. n indicates the how many</span></span>
<span id="cb5-5"><a href="copy-number-analyses.html#cb5-5" tabindex="-1"></a><span class="co"># organoids contain the CNA</span></span>
<span id="cb5-6"><a href="copy-number-analyses.html#cb5-6" tabindex="-1"></a>arm_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(deletions, <span class="cf">function</span>(x) {</span>
<span id="cb5-7"><a href="copy-number-analyses.html#cb5-7" tabindex="-1"></a>  <span class="fu">return</span>(x[[<span class="dv">2</span>]])</span>
<span id="cb5-8"><a href="copy-number-analyses.html#cb5-8" tabindex="-1"></a>}) <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="copy-number-analyses.html#cb5-9" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="at">.id =</span> <span class="st">&quot;cline&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="copy-number-analyses.html#cb5-10" tabindex="-1"></a>  <span class="fu">group_by</span>(cline, chromosome, arm) <span class="sc">%&gt;%</span></span>
<span id="cb5-11"><a href="copy-number-analyses.html#cb5-11" tabindex="-1"></a>  <span class="fu">count</span>(percentage_arm_affected <span class="sc">&gt;</span> <span class="fl">0.8</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-12"><a href="copy-number-analyses.html#cb5-12" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb5-13"><a href="copy-number-analyses.html#cb5-13" tabindex="-1"></a>  <span class="fu">ungroup</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb5-14"><a href="copy-number-analyses.html#cb5-14" tabindex="-1"></a>  <span class="fu">count</span>(chromosome) <span class="sc">%&gt;%</span></span>
<span id="cb5-15"><a href="copy-number-analyses.html#cb5-15" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb5-16"><a href="copy-number-analyses.html#cb5-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">freq =</span> n <span class="sc">/</span> <span class="dv">7</span>)</span>
<span id="cb5-17"><a href="copy-number-analyses.html#cb5-17" tabindex="-1"></a></span>
<span id="cb5-18"><a href="copy-number-analyses.html#cb5-18" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(arm_del, <span class="at">caption =</span> <span class="st">&quot;Arm-level deletions in organoids&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-48d289e05e4dd7fafbaa" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-48d289e05e4dd7fafbaa">{"x":{"filter":"none","vertical":false,"caption":"<caption>Arm-level deletions in organoids<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11"],["18","8","1","1","10","18","4","6","6","X","X"],["q","p","p","q","q","p","q","p","q","p","q"],[4,3,1,1,1,1,1,1,1,1,1],[0.5714285714285714,0.4285714285714285,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428,0.1428571428571428]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>chromosome<\/th>\n      <th>arm<\/th>\n      <th>n<\/th>\n      <th>freq<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="copy-number-analyses.html#cb6-1" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb6-2"><a href="copy-number-analyses.html#cb6-2" tabindex="-1"></a><span class="co"># End #</span></span>
<span id="cb6-3"><a href="copy-number-analyses.html#cb6-3" tabindex="-1"></a><span class="do">#######</span></span></code></pre></div>
</div>
<div id="copy-number-alterations-per-subclone" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Copy number alterations per subclone<a href="copy-number-analyses.html#copy-number-alterations-per-subclone" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="copy-number-alterations-specific-to-hub062a" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Copy number alterations specific to HUB062A<a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="copy-number-analyses.html#cb7-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb7-2"><a href="copy-number-analyses.html#cb7-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb7-3"><a href="copy-number-analyses.html#cb7-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb7-4"><a href="copy-number-analyses.html#cb7-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/subclonal_dynamics/subclone_list.rda&#39;</span>)</span>
<span id="cb7-5"><a href="copy-number-analyses.html#cb7-5" tabindex="-1"></a></span>
<span id="cb7-6"><a href="copy-number-analyses.html#cb7-6" tabindex="-1"></a><span class="co"># !!! GITHUB SPECIFIC CODE !!! #</span></span>
<span id="cb7-7"><a href="copy-number-analyses.html#cb7-7" tabindex="-1"></a><span class="co"># adding &#39;../cna_analysis/&#39; to each path within man_select_files_edivisive </span></span>
<span id="cb7-8"><a href="copy-number-analyses.html#cb7-8" tabindex="-1"></a>subclone_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(subclone_list, <span class="cf">function</span>(x) <span class="fu">paste0</span>(<span class="st">&quot;../cna_analysis/&quot;</span>, x))</span>
<span id="cb7-9"><a href="copy-number-analyses.html#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="copy-number-analyses.html#cb7-10" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb7-11"><a href="copy-number-analyses.html#cb7-11" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb7-12"><a href="copy-number-analyses.html#cb7-12" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb7-13"><a href="copy-number-analyses.html#cb7-13" tabindex="-1"></a><span class="co"># extracting resistant and sensitive subclones of hub062</span></span>
<span id="cb7-14"><a href="copy-number-analyses.html#cb7-14" tabindex="-1"></a>hub062 <span class="ot">&lt;-</span></span>
<span id="cb7-15"><a href="copy-number-analyses.html#cb7-15" tabindex="-1"></a>  <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub062&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb7-16"><a href="copy-number-analyses.html#cb7-16" tabindex="-1"></a>res <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_a&#39;</span>, hub062)]</span>
<span id="cb7-17"><a href="copy-number-analyses.html#cb7-17" tabindex="-1"></a>sens <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_b|hub062_postrad_c&#39;</span>, hub062)]</span>
<span id="cb7-18"><a href="copy-number-analyses.html#cb7-18" tabindex="-1"></a><span class="co"># remove cycle 2</span></span>
<span id="cb7-19"><a href="copy-number-analyses.html#cb7-19" tabindex="-1"></a>sens <span class="ot">&lt;-</span> sens[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&#39;c2&#39;</span>, sens)]</span>
<span id="cb7-20"><a href="copy-number-analyses.html#cb7-20" tabindex="-1"></a></span>
<span id="cb7-21"><a href="copy-number-analyses.html#cb7-21" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb7-22"><a href="copy-number-analyses.html#cb7-22" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb7-23"><a href="copy-number-analyses.html#cb7-23" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb7-24"><a href="copy-number-analyses.html#cb7-24" tabindex="-1"></a></span>
<span id="cb7-25"><a href="copy-number-analyses.html#cb7-25" tabindex="-1"></a><span class="do">## finding common CNAs within each subclone for each organoids.</span></span>
<span id="cb7-26"><a href="copy-number-analyses.html#cb7-26" tabindex="-1"></a>hub062a_specific_cnas <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb7-27"><a href="copy-number-analyses.html#cb7-27" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[res],</span>
<span id="cb7-28"><a href="copy-number-analyses.html#cb7-28" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[sens],</span>
<span id="cb7-29"><a href="copy-number-analyses.html#cb7-29" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> <span class="fl">0.6</span>,</span>
<span id="cb7-30"><a href="copy-number-analyses.html#cb7-30" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.6</span>,</span>
<span id="cb7-31"><a href="copy-number-analyses.html#cb7-31" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.2</span>,</span>
<span id="cb7-32"><a href="copy-number-analyses.html#cb7-32" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.2</span>,</span>
<span id="cb7-33"><a href="copy-number-analyses.html#cb7-33" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb7-34"><a href="copy-number-analyses.html#cb7-34" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 60 and 60% of single cells within resistant and sensitive subclones, respectively, and that are shared by 20 and 20% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="copy-number-analyses.html#cb9-1" tabindex="-1"></a><span class="do">## mapping CNA to cytoband location</span></span>
<span id="cb9-2"><a href="copy-number-analyses.html#cb9-2" tabindex="-1"></a><span class="co"># for amplifications</span></span>
<span id="cb9-3"><a href="copy-number-analyses.html#cb9-3" tabindex="-1"></a>ampl_hub062a <span class="ot">&lt;-</span></span>
<span id="cb9-4"><a href="copy-number-analyses.html#cb9-4" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(hub062a_specific_cnas<span class="sc">$</span>unique_resistant_ampl)</span>
<span id="cb9-5"><a href="copy-number-analyses.html#cb9-5" tabindex="-1"></a></span>
<span id="cb9-6"><a href="copy-number-analyses.html#cb9-6" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(ampl_hub062a[[<span class="dv">1</span>]]), <span class="at">caption =</span> <span class="st">&quot;HUB062A-specific amplifications (exact location)&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e216c6857a728874f76f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e216c6857a728874f76f">{"x":{"filter":"none","vertical":false,"caption":"<caption>HUB062A-specific amplifications (exact location)<\/caption>","data":[["1","2","3","4","5","6","7","8"],["2","8","9","11","17","17","17","X"],[217938033,34917837,1,1,1,21081101,27304921,52329467],[218892728,37853827,70227261,70578503,10892963,21546522,37378798,114869867],[954696,2935991,70227261,70578503,10892963,465422,10073878,62540401],["*","*","*","*","*","*","*","*"],[["2q35;15%"],["8p12;23%","8p11.23;64%"],["9p24.3;100%","9p24.2;100%","9p24.1;100%","9p23;100%","9p22.3;100%","9p22.2;100%","9p22.1;100%","9p21.3;100%","9p21.2;100%","9p21.1;100%","9p13.3;100%","9p13.2;100%","9p13.1;100%","9p12;100%","9p11.2;100%","9p11.1;100%","9q11;100%","9q12;100%","9q13;100%","9q21.11;100%","9q21.12;46%"],["11p15.5;100%","11p15.4;100%","11p15.3;100%","11p15.2;100%","11p15.1;100%","11p14.3;100%","11p14.2;100%","11p14.1;100%","11p13;100%","11p12;100%","11p11.2;100%","11p11.12;100%","11p11.11;100%","11q11;100%","11q12.1;100%","11q12.2;100%","11q12.3;100%","11q13.1;100%","11q13.2;100%","11q13.3;100%","11q13.4;2%"],["17p13.3;100%","17p13.2;100%","17p13.1;100%","17p12;2%"],["17p11.2;7%"],["17q11.1;4%","17q11.2;100%","17q12;62%"],["Xp11.22;53%","Xp11.21;100%","Xp11.1;100%","Xq11.1;100%","Xq11.2;100%","Xq12;100%","Xq13.1;100%","Xq13.2;100%","Xq13.3;100%","Xq21.1;100%","Xq21.2;100%","Xq21.31;100%","Xq21.32;100%","Xq21.33;100%","Xq22.1;100%","Xq22.2;100%","Xq22.3;100%","Xq23;68%"]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="copy-number-analyses.html#cb10-1" tabindex="-1"></a>del_hub062a <span class="ot">&lt;-</span></span>
<span id="cb10-2"><a href="copy-number-analyses.html#cb10-2" tabindex="-1"></a>  <span class="fu">get_cytoband_coverage</span>(hub062a_specific_cnas<span class="sc">$</span>unique_resistant_del)</span>
<span id="cb10-3"><a href="copy-number-analyses.html#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="copy-number-analyses.html#cb10-4" tabindex="-1"></a><span class="co"># No HUB062A-specific deletions were found.</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="copy-number-analyses.html#cb11-1" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb11-2"><a href="copy-number-analyses.html#cb11-2" tabindex="-1"></a><span class="co"># Run #</span></span>
<span id="cb11-3"><a href="copy-number-analyses.html#cb11-3" tabindex="-1"></a><span class="do">#######</span></span>
<span id="cb11-4"><a href="copy-number-analyses.html#cb11-4" tabindex="-1"></a><span class="co"># All genes #</span></span>
<span id="cb11-5"><a href="copy-number-analyses.html#cb11-5" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb11-6"><a href="copy-number-analyses.html#cb11-6" tabindex="-1"></a>genes_ampl_hub062a <span class="ot">&lt;-</span> <span class="fu">get_genes</span>(ampl_hub062a[[<span class="dv">1</span>]])</span>
<span id="cb11-7"><a href="copy-number-analyses.html#cb11-7" tabindex="-1"></a></span>
<span id="cb11-8"><a href="copy-number-analyses.html#cb11-8" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(genes_ampl_hub062a), <span class="at">caption =</span> <span class="st">&quot;Genes on HUB062A-specific amplifications&quot;</span>)</span>
<span id="cb11-9"><a href="copy-number-analyses.html#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="copy-number-analyses.html#cb11-10" tabindex="-1"></a><span class="co"># Oncogenes #</span></span>
<span id="cb11-11"><a href="copy-number-analyses.html#cb11-11" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb11-12"><a href="copy-number-analyses.html#cb11-12" tabindex="-1"></a>oncogenes_ampl_hub062a <span class="ot">&lt;-</span> <span class="fu">get_oncogenes</span>(ampl_hub062a[[<span class="dv">1</span>]])</span>
<span id="cb11-13"><a href="copy-number-analyses.html#cb11-13" tabindex="-1"></a></span>
<span id="cb11-14"><a href="copy-number-analyses.html#cb11-14" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(oncogenes_ampl_hub062a), <span class="at">caption =</span> <span class="st">&quot;Oncogenes on HUB062A-specific amplifications&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="copy-number-analyses.html#cb12-1" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb12-2"><a href="copy-number-analyses.html#cb12-2" tabindex="-1"></a><span class="co"># Data #</span></span>
<span id="cb12-3"><a href="copy-number-analyses.html#cb12-3" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb12-4"><a href="copy-number-analyses.html#cb12-4" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hub062a_specific_cnas_genes.rda&#39;</span>)</span>
<span id="cb12-5"><a href="copy-number-analyses.html#cb12-5" tabindex="-1"></a></span>
<span id="cb12-6"><a href="copy-number-analyses.html#cb12-6" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb12-7"><a href="copy-number-analyses.html#cb12-7" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb12-8"><a href="copy-number-analyses.html#cb12-8" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb12-9"><a href="copy-number-analyses.html#cb12-9" tabindex="-1"></a><span class="co"># plot_genes_karyotype needs a GRange with seqnames, ranges, strand and hgnc_symbol</span></span>
<span id="cb12-10"><a href="copy-number-analyses.html#cb12-10" tabindex="-1"></a></span>
<span id="cb12-11"><a href="copy-number-analyses.html#cb12-11" tabindex="-1"></a><span class="co"># add ampl/del column</span></span>
<span id="cb12-12"><a href="copy-number-analyses.html#cb12-12" tabindex="-1"></a>oncogenes_ampl_hub062a<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="st">&#39;amplified&#39;</span></span>
<span id="cb12-13"><a href="copy-number-analyses.html#cb12-13" tabindex="-1"></a></span>
<span id="cb12-14"><a href="copy-number-analyses.html#cb12-14" tabindex="-1"></a></span>
<span id="cb12-15"><a href="copy-number-analyses.html#cb12-15" tabindex="-1"></a><span class="co"># bind dataframes</span></span>
<span id="cb12-16"><a href="copy-number-analyses.html#cb12-16" tabindex="-1"></a>oncogenes <span class="ot">&lt;-</span> oncogenes_ampl_hub062a</span>
<span id="cb12-17"><a href="copy-number-analyses.html#cb12-17" tabindex="-1"></a></span>
<span id="cb12-18"><a href="copy-number-analyses.html#cb12-18" tabindex="-1"></a><span class="co"># extract needed columns</span></span>
<span id="cb12-19"><a href="copy-number-analyses.html#cb12-19" tabindex="-1"></a>keep <span class="ot">&lt;-</span></span>
<span id="cb12-20"><a href="copy-number-analyses.html#cb12-20" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;Gene Symbol&#39;</span>,</span>
<span id="cb12-21"><a href="copy-number-analyses.html#cb12-21" tabindex="-1"></a>    <span class="st">&#39;chromosome_name&#39;</span>,</span>
<span id="cb12-22"><a href="copy-number-analyses.html#cb12-22" tabindex="-1"></a>    <span class="st">&#39;start_position&#39;</span>,</span>
<span id="cb12-23"><a href="copy-number-analyses.html#cb12-23" tabindex="-1"></a>    <span class="st">&#39;end_position&#39;</span>,</span>
<span id="cb12-24"><a href="copy-number-analyses.html#cb12-24" tabindex="-1"></a>    <span class="st">&#39;cn&#39;</span>)</span>
<span id="cb12-25"><a href="copy-number-analyses.html#cb12-25" tabindex="-1"></a>oncogenes <span class="ot">&lt;-</span> oncogenes[, keep]</span>
<span id="cb12-26"><a href="copy-number-analyses.html#cb12-26" tabindex="-1"></a><span class="co"># rename</span></span>
<span id="cb12-27"><a href="copy-number-analyses.html#cb12-27" tabindex="-1"></a><span class="fu">colnames</span>(oncogenes) <span class="ot">&lt;-</span></span>
<span id="cb12-28"><a href="copy-number-analyses.html#cb12-28" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;hgnc_symbol&#39;</span>, <span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;cn&#39;</span>)</span>
<span id="cb12-29"><a href="copy-number-analyses.html#cb12-29" tabindex="-1"></a></span>
<span id="cb12-30"><a href="copy-number-analyses.html#cb12-30" tabindex="-1"></a><span class="co"># convert to dataframe</span></span>
<span id="cb12-31"><a href="copy-number-analyses.html#cb12-31" tabindex="-1"></a>onco_grange <span class="ot">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span>(oncogenes)</span>
<span id="cb12-32"><a href="copy-number-analyses.html#cb12-32" tabindex="-1"></a></span>
<span id="cb12-33"><a href="copy-number-analyses.html#cb12-33" tabindex="-1"></a><span class="co"># metadata is deleted</span></span>
<span id="cb12-34"><a href="copy-number-analyses.html#cb12-34" tabindex="-1"></a>onco_grange<span class="sc">$</span>hgnc_symbol <span class="ot">&lt;-</span> oncogenes<span class="sc">$</span>hgnc_symbol</span>
<span id="cb12-35"><a href="copy-number-analyses.html#cb12-35" tabindex="-1"></a>onco_grange<span class="sc">$</span>cn <span class="ot">&lt;-</span> oncogenes<span class="sc">$</span>cn</span>
<span id="cb12-36"><a href="copy-number-analyses.html#cb12-36" tabindex="-1"></a></span>
<span id="cb12-37"><a href="copy-number-analyses.html#cb12-37" tabindex="-1"></a><span class="do">##########################</span></span>
<span id="cb12-38"><a href="copy-number-analyses.html#cb12-38" tabindex="-1"></a><span class="co"># Choose genes to depict #</span></span>
<span id="cb12-39"><a href="copy-number-analyses.html#cb12-39" tabindex="-1"></a><span class="do">##########################</span></span>
<span id="cb12-40"><a href="copy-number-analyses.html#cb12-40" tabindex="-1"></a>oncogenes_ampl_hub062a <span class="ot">&lt;-</span></span>
<span id="cb12-41"><a href="copy-number-analyses.html#cb12-41" tabindex="-1"></a>  oncogenes_ampl_hub062a[<span class="fu">order</span>(oncogenes_ampl_hub062a<span class="sc">$</span><span class="st">`</span><span class="at">Genome Location</span><span class="st">`</span>),]</span>
<span id="cb12-42"><a href="copy-number-analyses.html#cb12-42" tabindex="-1"></a></span>
<span id="cb12-43"><a href="copy-number-analyses.html#cb12-43" tabindex="-1"></a>show_oncogenes_ampl <span class="ot">&lt;-</span></span>
<span id="cb12-44"><a href="copy-number-analyses.html#cb12-44" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">&#39;FANCF&#39;</span>,</span>
<span id="cb12-45"><a href="copy-number-analyses.html#cb12-45" tabindex="-1"></a>    <span class="st">&#39;CREB3L1&#39;</span>,</span>
<span id="cb12-46"><a href="copy-number-analyses.html#cb12-46" tabindex="-1"></a>    <span class="st">&#39;DDB2&#39;</span>,</span>
<span id="cb12-47"><a href="copy-number-analyses.html#cb12-47" tabindex="-1"></a>    <span class="st">&#39;HRAS&#39;</span>,</span>
<span id="cb12-48"><a href="copy-number-analyses.html#cb12-48" tabindex="-1"></a>    <span class="st">&#39;CCND1&#39;</span>,</span>
<span id="cb12-49"><a href="copy-number-analyses.html#cb12-49" tabindex="-1"></a>    <span class="st">&#39;NF1&#39;</span>,</span>
<span id="cb12-50"><a href="copy-number-analyses.html#cb12-50" tabindex="-1"></a>    <span class="st">&#39;FANCG&#39;</span>,</span>
<span id="cb12-51"><a href="copy-number-analyses.html#cb12-51" tabindex="-1"></a>    <span class="st">&#39;JAK2&#39;</span>,</span>
<span id="cb12-52"><a href="copy-number-analyses.html#cb12-52" tabindex="-1"></a>    <span class="st">&#39;AR&#39;</span>)</span>
<span id="cb12-53"><a href="copy-number-analyses.html#cb12-53" tabindex="-1"></a></span>
<span id="cb12-54"><a href="copy-number-analyses.html#cb12-54" tabindex="-1"></a>show_oncogenes <span class="ot">&lt;-</span> <span class="fu">c</span>(show_oncogenes_ampl)</span>
<span id="cb12-55"><a href="copy-number-analyses.html#cb12-55" tabindex="-1"></a></span>
<span id="cb12-56"><a href="copy-number-analyses.html#cb12-56" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb12-57"><a href="copy-number-analyses.html#cb12-57" tabindex="-1"></a><span class="co"># PLOT #</span></span>
<span id="cb12-58"><a href="copy-number-analyses.html#cb12-58" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb12-59"><a href="copy-number-analyses.html#cb12-59" tabindex="-1"></a><span class="fu">plot_genes_karyotype</span>(</span>
<span id="cb12-60"><a href="copy-number-analyses.html#cb12-60" tabindex="-1"></a>  onco_grange,</span>
<span id="cb12-61"><a href="copy-number-analyses.html#cb12-61" tabindex="-1"></a>  <span class="at">show_genes =</span> show_oncogenes,</span>
<span id="cb12-62"><a href="copy-number-analyses.html#cb12-62" tabindex="-1"></a>  <span class="at">dist =</span> <span class="sc">-</span><span class="dv">45</span>,</span>
<span id="cb12-63"><a href="copy-number-analyses.html#cb12-63" tabindex="-1"></a>  <span class="at">show_chromosomes =</span> <span class="fu">c</span>(</span>
<span id="cb12-64"><a href="copy-number-analyses.html#cb12-64" tabindex="-1"></a>    <span class="st">&quot;chr9&quot;</span>,</span>
<span id="cb12-65"><a href="copy-number-analyses.html#cb12-65" tabindex="-1"></a>    <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb12-66"><a href="copy-number-analyses.html#cb12-66" tabindex="-1"></a>    <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb12-67"><a href="copy-number-analyses.html#cb12-67" tabindex="-1"></a>    <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb12-68"><a href="copy-number-analyses.html#cb12-68" tabindex="-1"></a>    <span class="st">&quot;chr13&quot;</span>,</span>
<span id="cb12-69"><a href="copy-number-analyses.html#cb12-69" tabindex="-1"></a>    <span class="st">&quot;chr14&quot;</span>,</span>
<span id="cb12-70"><a href="copy-number-analyses.html#cb12-70" tabindex="-1"></a>    <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb12-71"><a href="copy-number-analyses.html#cb12-71" tabindex="-1"></a>    <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb12-72"><a href="copy-number-analyses.html#cb12-72" tabindex="-1"></a>    <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb12-73"><a href="copy-number-analyses.html#cb12-73" tabindex="-1"></a>    <span class="st">&quot;chr18&quot;</span>,</span>
<span id="cb12-74"><a href="copy-number-analyses.html#cb12-74" tabindex="-1"></a>    <span class="st">&quot;chr19&quot;</span>,</span>
<span id="cb12-75"><a href="copy-number-analyses.html#cb12-75" tabindex="-1"></a>    <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb12-76"><a href="copy-number-analyses.html#cb12-76" tabindex="-1"></a>    <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb12-77"><a href="copy-number-analyses.html#cb12-77" tabindex="-1"></a>    <span class="st">&quot;chr22&quot;</span>,</span>
<span id="cb12-78"><a href="copy-number-analyses.html#cb12-78" tabindex="-1"></a>    <span class="st">&quot;chrX&quot;</span></span>
<span id="cb12-79"><a href="copy-number-analyses.html#cb12-79" tabindex="-1"></a>  )</span>
<span id="cb12-80"><a href="copy-number-analyses.html#cb12-80" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figs/hub062a_karyoplot.jpg" width="1288" /></p>
</div>
<div id="copy-number-alterations-specific-to-resistant-subclones" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Copy number alterations specific to resistant subclones<a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>## Loading required package: AneuFinder</code></pre>
<pre><code>## Loading required package: cowplot</code></pre>
<pre><code>## 
## Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:lubridate&#39;:
## 
##     stamp</code></pre>
<pre><code>## Loading required package: AneuFinderData</code></pre>
<pre><code>## 
## Please visit https://github.com/ataudt/aneufinder for the latest bugfixes and features.</code></pre>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 60 and 60% of single cells within resistant and sensitive subclones, respectively, and that are shared by 80 and 10% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<pre><code>## Loading required package: tidyverse</code></pre>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.2     ✔ readr     2.1.4
## ✔ forcats   1.0.0     ✔ stringr   1.5.0
## ✔ ggplot2   3.4.2     ✔ tibble    3.2.1
## ✔ lubridate 1.9.2     ✔ tidyr     1.3.0
## ✔ purrr     1.0.1     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
## Loading required package: plyranges
## 
## Loading required package: BiocGenerics
## 
## 
## Attaching package: &#39;BiocGenerics&#39;
## 
## 
## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     intersect, setdiff, union
## 
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union
## 
## 
## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs
## 
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min
## 
## 
## Loading required package: IRanges
## 
## Loading required package: S4Vectors
## 
## 
## Attaching package: &#39;S4Vectors&#39;
## 
## 
## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     second, second&lt;-
## 
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename
## 
## 
## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand
## 
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname
## 
## 
## 
## Attaching package: &#39;IRanges&#39;
## 
## 
## The following object is masked from &#39;package:lubridate&#39;:
## 
##     %within%
## 
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice
## 
## 
## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce
## 
## 
## Loading required package: GenomicRanges
## 
## Loading required package: GenomeInfoDb
## 
## 
## Attaching package: &#39;plyranges&#39;
## 
## 
## The following object is masked from &#39;package:IRanges&#39;:
## 
##     slice
## 
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     between, n, n_distinct
## 
## 
## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
</div>
<div id="newly-acquired-copy-number-alterations" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Newly acquired copy number alterations<a href="copy-number-analyses.html#newly-acquired-copy-number-alterations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="copy-number-analyses.html#cb22-1" tabindex="-1"></a><span class="do">###############</span></span>
<span id="cb22-2"><a href="copy-number-analyses.html#cb22-2" tabindex="-1"></a><span class="co"># Description #</span></span>
<span id="cb22-3"><a href="copy-number-analyses.html#cb22-3" tabindex="-1"></a><span class="do">###############</span></span>
<span id="cb22-4"><a href="copy-number-analyses.html#cb22-4" tabindex="-1"></a><span class="co"># We are going to define the presence of &#39;new&#39; CNAs after radiation therapy. </span></span>
<span id="cb22-5"><a href="copy-number-analyses.html#cb22-5" tabindex="-1"></a><span class="co"># I define the following rules: </span></span>
<span id="cb22-6"><a href="copy-number-analyses.html#cb22-6" tabindex="-1"></a><span class="co"># (i) Analysis will be done within subclone</span></span>
<span id="cb22-7"><a href="copy-number-analyses.html#cb22-7" tabindex="-1"></a><span class="co"># (ii) A &#39;new&#39; CNA will be called new only if it was not found before radiation</span></span>
<span id="cb22-8"><a href="copy-number-analyses.html#cb22-8" tabindex="-1"></a><span class="co"># (iii) I exclude all subclones where the numbers of cells within a subclone is lower than 5. </span></span>
<span id="cb22-9"><a href="copy-number-analyses.html#cb22-9" tabindex="-1"></a><span class="co"># (iv) Because in essence, all cells are related to each other (even those in distinct) </span></span>
<span id="cb22-10"><a href="copy-number-analyses.html#cb22-10" tabindex="-1"></a><span class="co"># subclones, I am going to compare each post-rad subclone to all pooled subclones before</span></span>
<span id="cb22-11"><a href="copy-number-analyses.html#cb22-11" tabindex="-1"></a><span class="co"># radiation. </span></span>
<span id="cb22-12"><a href="copy-number-analyses.html#cb22-12" tabindex="-1"></a><span class="co"># (v) We are looking for acquired CNAs; CNAs that could explain why cells with that </span></span>
<span id="cb22-13"><a href="copy-number-analyses.html#cb22-13" tabindex="-1"></a><span class="co"># CNA propagated. Therefore, the CNA has to be present in at least 50% of the cells. </span></span>
<span id="cb22-14"><a href="copy-number-analyses.html#cb22-14" tabindex="-1"></a></span>
<span id="cb22-15"><a href="copy-number-analyses.html#cb22-15" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb22-16"><a href="copy-number-analyses.html#cb22-16" tabindex="-1"></a><span class="co"># Wrangle #</span></span>
<span id="cb22-17"><a href="copy-number-analyses.html#cb22-17" tabindex="-1"></a><span class="do">###########</span></span>
<span id="cb22-18"><a href="copy-number-analyses.html#cb22-18" tabindex="-1"></a></span>
<span id="cb22-19"><a href="copy-number-analyses.html#cb22-19" tabindex="-1"></a><span class="co"># remove subclones with fewer than 10 cells pre-or post rad. </span></span>
<span id="cb22-20"><a href="copy-number-analyses.html#cb22-20" tabindex="-1"></a><span class="fu">lapply</span>(subclone_list, length)</span></code></pre></div>
<pre><code>## $hub183_prerad_a
## [1] 127
## 
## $hub183_postrad_a
## [1] 110
## 
## $hub005_prerad_a.a
## [1] 25
## 
## $hub005_prerad_a.b
## [1] 13
## 
## $hub005_prerad_b
## [1] 118
## 
## $hub005_postrad_a.a
## [1] 17
## 
## $hub005_postrad_a.b
## [1] 2
## 
## $hub005_postrad_b
## [1] 70
## 
## $hub106_prerad_a
## [1] 64
## 
## $hub106_prerad_b
## [1] 59
## 
## $hub106_prerad_c
## [1] 4
## 
## $hub106_postrad_a
## [1] 60
## 
## $hub106_postrad_b
## [1] 103
## 
## $hub106_postrad_c
## [1] 18
## 
## $hub062_prerad_a
## [1] 2
## 
## $hub062_prerad_b
## [1] 127
## 
## $hub062_prerad_c
## [1] 7
## 
## $hub062_postrad_a
## [1] 39
## 
## $hub062_postrad_b
## [1] 38
## 
## $hub062_postrad_c
## [1] 7
## 
## $hub062_postrad_c2_a
## [1] 187
## 
## $hub062_postrad_c2_c
## [1] 1
## 
## $hub015_prerad_a.b
## [1] 276
## 
## $hub015_prerad_b
## [1] 1
## 
## $hub015_postrad_a.a
## [1] 110
## 
## $hub015_postrad_a.b
## [1] 157
## 
## $hub015_postrad_b
## [1] 14
## 
## $hub181i_prerad_a
## [1] 72
## 
## $hub181i_prerad_b
## [1] 233
## 
## $hub181i_postrad_a
## [1] 72
## 
## $hub181i_postrad_b
## [1] 239
## 
## $hub197_prerad_a
## [1] 24
## 
## $hub197_prerad_b
## [1] 17
## 
## $hub197_prerad_c
## [1] 91
## 
## $hub197_postrad_a
## [1] 252
## 
## $hub062biological_prerad_a
## [1] 14
## 
## $hub062biological_prerad_b
## [1] 205
## 
## $hub062biological_prerad_c
## [1] 13
## 
## $hub062biological_postrad_a
## [1] 178</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="copy-number-analyses.html#cb24-1" tabindex="-1"></a>remove_me <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;hub005_prerad_a.b&#39;</span>, <span class="st">&#39;hub005_postrad_a.b&#39;</span>, <span class="co"># because only 2 cells in hub005_postrad_a.b</span></span>
<span id="cb24-2"><a href="copy-number-analyses.html#cb24-2" tabindex="-1"></a>               <span class="st">&#39;hub062_prerad_a&#39;</span>, <span class="co"># only 2 cells, here I am going to compare C1 vs C2, so </span></span>
<span id="cb24-3"><a href="copy-number-analyses.html#cb24-3" tabindex="-1"></a>               <span class="co"># I don&#39;t remove hub062_postrad_a</span></span>
<span id="cb24-4"><a href="copy-number-analyses.html#cb24-4" tabindex="-1"></a>               <span class="st">&#39;hub015_prerad_b&#39;</span>, <span class="st">&#39;hub015_postrad_b&#39;</span>, <span class="co"># only 1 cell in prerad</span></span>
<span id="cb24-5"><a href="copy-number-analyses.html#cb24-5" tabindex="-1"></a>               <span class="st">&#39;hub197_prerad_b&#39;</span>,<span class="st">&#39;hub197_prerad_b&#39;</span>, <span class="co"># extinction</span></span>
<span id="cb24-6"><a href="copy-number-analyses.html#cb24-6" tabindex="-1"></a>               <span class="st">&#39;hub197_prerad_c&#39;</span>,<span class="st">&#39;hub197_postrad_c&#39;</span> <span class="co"># extinction</span></span>
<span id="cb24-7"><a href="copy-number-analyses.html#cb24-7" tabindex="-1"></a>               )</span>
<span id="cb24-8"><a href="copy-number-analyses.html#cb24-8" tabindex="-1"></a></span>
<span id="cb24-9"><a href="copy-number-analyses.html#cb24-9" tabindex="-1"></a>subclone_list <span class="ot">&lt;-</span> subclone_list[<span class="sc">!</span>(<span class="fu">names</span>(subclone_list) <span class="sc">%in%</span> remove_me)]</span>
<span id="cb24-10"><a href="copy-number-analyses.html#cb24-10" tabindex="-1"></a></span>
<span id="cb24-11"><a href="copy-number-analyses.html#cb24-11" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb24-12"><a href="copy-number-analyses.html#cb24-12" tabindex="-1"></a><span class="co"># Variables #</span></span>
<span id="cb24-13"><a href="copy-number-analyses.html#cb24-13" tabindex="-1"></a><span class="do">#############</span></span>
<span id="cb24-14"><a href="copy-number-analyses.html#cb24-14" tabindex="-1"></a><span class="co"># Percentage of cells cut-off </span></span>
<span id="cb24-15"><a href="copy-number-analyses.html#cb24-15" tabindex="-1"></a>pcc <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb24-16"><a href="copy-number-analyses.html#cb24-16" tabindex="-1"></a></span>
<span id="cb24-17"><a href="copy-number-analyses.html#cb24-17" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb24-18"><a href="copy-number-analyses.html#cb24-18" tabindex="-1"></a><span class="co"># RUNS #</span></span>
<span id="cb24-19"><a href="copy-number-analyses.html#cb24-19" tabindex="-1"></a><span class="do">########</span></span>
<span id="cb24-20"><a href="copy-number-analyses.html#cb24-20" tabindex="-1"></a></span>
<span id="cb24-21"><a href="copy-number-analyses.html#cb24-21" tabindex="-1"></a><span class="co"># HUB183 #</span></span>
<span id="cb24-22"><a href="copy-number-analyses.html#cb24-22" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb24-23"><a href="copy-number-analyses.html#cb24-23" tabindex="-1"></a><span class="co"># Extract HUB183</span></span>
<span id="cb24-24"><a href="copy-number-analyses.html#cb24-24" tabindex="-1"></a>hub183 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub183&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb24-25"><a href="copy-number-analyses.html#cb24-25" tabindex="-1"></a></span>
<span id="cb24-26"><a href="copy-number-analyses.html#cb24-26" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb24-27"><a href="copy-number-analyses.html#cb24-27" tabindex="-1"></a>hub183_prerad_subclones <span class="ot">&lt;-</span> hub183[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub183)]</span>
<span id="cb24-28"><a href="copy-number-analyses.html#cb24-28" tabindex="-1"></a></span>
<span id="cb24-29"><a href="copy-number-analyses.html#cb24-29" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb24-30"><a href="copy-number-analyses.html#cb24-30" tabindex="-1"></a>hub183_postrad_a <span class="ot">&lt;-</span> hub183[<span class="fu">grepl</span>(<span class="st">&#39;hub183_postrad_a&#39;</span>, hub183)]</span>
<span id="cb24-31"><a href="copy-number-analyses.html#cb24-31" tabindex="-1"></a></span>
<span id="cb24-32"><a href="copy-number-analyses.html#cb24-32" tabindex="-1"></a><span class="co"># no new genomic aberrations</span></span>
<span id="cb24-33"><a href="copy-number-analyses.html#cb24-33" tabindex="-1"></a>hub183_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb24-34"><a href="copy-number-analyses.html#cb24-34" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub183_postrad_a],</span>
<span id="cb24-35"><a href="copy-number-analyses.html#cb24-35" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub183_prerad_subclones],</span>
<span id="cb24-36"><a href="copy-number-analyses.html#cb24-36" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb24-37"><a href="copy-number-analyses.html#cb24-37" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb24-38"><a href="copy-number-analyses.html#cb24-38" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb24-39"><a href="copy-number-analyses.html#cb24-39" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb24-40"><a href="copy-number-analyses.html#cb24-40" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb24-41"><a href="copy-number-analyses.html#cb24-41" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="copy-number-analyses.html#cb26-1" tabindex="-1"></a><span class="co"># HUB005 #</span></span>
<span id="cb26-2"><a href="copy-number-analyses.html#cb26-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb26-3"><a href="copy-number-analyses.html#cb26-3" tabindex="-1"></a></span>
<span id="cb26-4"><a href="copy-number-analyses.html#cb26-4" tabindex="-1"></a><span class="co"># Extract HUB005</span></span>
<span id="cb26-5"><a href="copy-number-analyses.html#cb26-5" tabindex="-1"></a>hub005 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub005&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb26-6"><a href="copy-number-analyses.html#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="copy-number-analyses.html#cb26-7" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb26-8"><a href="copy-number-analyses.html#cb26-8" tabindex="-1"></a>hub005_prerad_subclones <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub005)]</span>
<span id="cb26-9"><a href="copy-number-analyses.html#cb26-9" tabindex="-1"></a></span>
<span id="cb26-10"><a href="copy-number-analyses.html#cb26-10" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb26-11"><a href="copy-number-analyses.html#cb26-11" tabindex="-1"></a>hub005_postrad_a.a <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;hub005_postrad_a.a&#39;</span>, hub005)]</span>
<span id="cb26-12"><a href="copy-number-analyses.html#cb26-12" tabindex="-1"></a>hub005_postrad_b <span class="ot">&lt;-</span> hub005[<span class="fu">grepl</span>(<span class="st">&#39;hub005_postrad_b&#39;</span>, hub005)]</span>
<span id="cb26-13"><a href="copy-number-analyses.html#cb26-13" tabindex="-1"></a></span>
<span id="cb26-14"><a href="copy-number-analyses.html#cb26-14" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb26-15"><a href="copy-number-analyses.html#cb26-15" tabindex="-1"></a>hub005_postrad_a.a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb26-16"><a href="copy-number-analyses.html#cb26-16" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub005_postrad_a.a],</span>
<span id="cb26-17"><a href="copy-number-analyses.html#cb26-17" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub005_prerad_subclones],</span>
<span id="cb26-18"><a href="copy-number-analyses.html#cb26-18" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb26-19"><a href="copy-number-analyses.html#cb26-19" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb26-20"><a href="copy-number-analyses.html#cb26-20" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb26-21"><a href="copy-number-analyses.html#cb26-21" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb26-22"><a href="copy-number-analyses.html#cb26-22" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb26-23"><a href="copy-number-analyses.html#cb26-23" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="copy-number-analyses.html#cb28-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb28-2"><a href="copy-number-analyses.html#cb28-2" tabindex="-1"></a>hub005_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb28-3"><a href="copy-number-analyses.html#cb28-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub005_postrad_b],</span>
<span id="cb28-4"><a href="copy-number-analyses.html#cb28-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub005_prerad_subclones],</span>
<span id="cb28-5"><a href="copy-number-analyses.html#cb28-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb28-6"><a href="copy-number-analyses.html#cb28-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb28-7"><a href="copy-number-analyses.html#cb28-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb28-8"><a href="copy-number-analyses.html#cb28-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb28-9"><a href="copy-number-analyses.html#cb28-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb28-10"><a href="copy-number-analyses.html#cb28-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="copy-number-analyses.html#cb30-1" tabindex="-1"></a><span class="co"># HUB015 #</span></span>
<span id="cb30-2"><a href="copy-number-analyses.html#cb30-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb30-3"><a href="copy-number-analyses.html#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="copy-number-analyses.html#cb30-4" tabindex="-1"></a><span class="co"># Extract HUB015</span></span>
<span id="cb30-5"><a href="copy-number-analyses.html#cb30-5" tabindex="-1"></a>hub015 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub015&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb30-6"><a href="copy-number-analyses.html#cb30-6" tabindex="-1"></a></span>
<span id="cb30-7"><a href="copy-number-analyses.html#cb30-7" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb30-8"><a href="copy-number-analyses.html#cb30-8" tabindex="-1"></a>hub015_prerad_subclones <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub015)]</span>
<span id="cb30-9"><a href="copy-number-analyses.html#cb30-9" tabindex="-1"></a></span>
<span id="cb30-10"><a href="copy-number-analyses.html#cb30-10" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb30-11"><a href="copy-number-analyses.html#cb30-11" tabindex="-1"></a>hub015_postrad_a.a <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_a.a&#39;</span>, hub015)]</span>
<span id="cb30-12"><a href="copy-number-analyses.html#cb30-12" tabindex="-1"></a>hub015_postrad_a.b <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_a.b&#39;</span>, hub015)]</span>
<span id="cb30-13"><a href="copy-number-analyses.html#cb30-13" tabindex="-1"></a>hub015_postrad_b <span class="ot">&lt;-</span> hub015[<span class="fu">grepl</span>(<span class="st">&#39;hub015_postrad_b&#39;</span>, hub015)]</span>
<span id="cb30-14"><a href="copy-number-analyses.html#cb30-14" tabindex="-1"></a></span>
<span id="cb30-15"><a href="copy-number-analyses.html#cb30-15" tabindex="-1"></a><span class="co"># hub015_postrad_a.a: new deletion on 4q</span></span>
<span id="cb30-16"><a href="copy-number-analyses.html#cb30-16" tabindex="-1"></a>hub015_postrad_a.a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb30-17"><a href="copy-number-analyses.html#cb30-17" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub015_postrad_a.a],</span>
<span id="cb30-18"><a href="copy-number-analyses.html#cb30-18" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub015_prerad_subclones],</span>
<span id="cb30-19"><a href="copy-number-analyses.html#cb30-19" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb30-20"><a href="copy-number-analyses.html#cb30-20" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb30-21"><a href="copy-number-analyses.html#cb30-21" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb30-22"><a href="copy-number-analyses.html#cb30-22" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb30-23"><a href="copy-number-analyses.html#cb30-23" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb30-24"><a href="copy-number-analyses.html#cb30-24" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="copy-number-analyses.html#cb32-1" tabindex="-1"></a><span class="co"># HUB015 has a new deletion following radiation in 4q</span></span>
<span id="cb32-2"><a href="copy-number-analyses.html#cb32-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">as.data.frame</span>(<span class="fu">get_cytoband_coverage</span>(hub015_postrad_a.a<span class="sc">$</span>unique_resistant_del)[[<span class="dv">1</span>]]), <span class="at">caption =</span> <span class="st">&quot;HUB015 has a new deletion in 4q&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-70eadeb23439fe7cb7ff" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-70eadeb23439fe7cb7ff">{"x":{"filter":"none","vertical":false,"caption":"<caption>HUB015 has a new deletion in 4q<\/caption>","data":[["1"],["4"],[75676850],[95450367],[19773518],["*"],[["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"]]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="copy-number-analyses.html#cb33-1" tabindex="-1"></a><span class="co"># This deletion contains the following oncogenes</span></span>
<span id="cb33-2"><a href="copy-number-analyses.html#cb33-2" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">get_oncogenes</span>(<span class="fu">get_cytoband_coverage</span>(hub015_postrad_a.a<span class="sc">$</span>unique_resistant_del)[[<span class="dv">1</span>]]), <span class="at">caption =</span> <span class="st">&quot;Oncogenes in HUB015 4q deletion&quot;</span>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-c7707ba31a7cff2b3cbb" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-c7707ba31a7cff2b3cbb">{"x":{"filter":"none","vertical":false,"caption":"<caption>Oncogenes in HUB015 4q deletion<\/caption>","data":[["1","2"],["AFF1","PTPN13"],["AF4/FMR2 family, member 1","protein tyrosine phosphatase, non-receptor type 13"],[4299,5783],["4:86935002-87141054","4:86594315-86815171"],[1,1],["Yes","Yes"],[22.1,21.3],["yes","yes"],[null,"yes"],["AL","lung, NSCLC, gastric, peritoneal carcinomatosis"],[null,"hepatocellular"],[null,"Familial Hepatocellular Carcinoma"],["L","E"],["Dom","Rec"],["fusion","TSG"],["T","Mis, N"],["KMT2A",null],[null,null],[null,null],["4299,AF-4,AF4,AFF1,ENSG00000172493.20,MLLT2,P51825,PBM1","5783,ENSG00000163629.12,PTP-BAS,PTP-BL,PTP1E,PTPL1,PTPN13,Q12923"],["chr4","chr4"],[75676850,75676850],[95450367,95450367],[19773518,19773518],["*","*"],[["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"],["4q21.1;86%","4q21.21;100%","4q21.22;100%","4q21.23;100%","4q21.3;100%","4q22.1;100%","4q22.2;100%","4q22.3;34%"]],[4,4],[86935002,86594315],[87141039,86815171]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene Symbol<\/th>\n      <th>Name<\/th>\n      <th>Entrez GeneId<\/th>\n      <th>Genome Location<\/th>\n      <th>Tier<\/th>\n      <th>Hallmark<\/th>\n      <th>Chr Band<\/th>\n      <th>Somatic<\/th>\n      <th>Germline<\/th>\n      <th>Tumour Types(Somatic)<\/th>\n      <th>Tumour Types(Germline)<\/th>\n      <th>Cancer Syndrome<\/th>\n      <th>Tissue Type<\/th>\n      <th>Molecular Genetics<\/th>\n      <th>Role in Cancer<\/th>\n      <th>Mutation Types<\/th>\n      <th>Translocation Partner<\/th>\n      <th>Other Germline Mut<\/th>\n      <th>Other Syndrome<\/th>\n      <th>Synonyms<\/th>\n      <th>seqnames<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>width<\/th>\n      <th>strand<\/th>\n      <th>cytobands_cov<\/th>\n      <th>chromosome_name<\/th>\n      <th>start_position<\/th>\n      <th>end_position<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,5,7,22,23,24,27,28,29]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="copy-number-analyses.html#cb34-1" tabindex="-1"></a><span class="co"># hub015_postrad_a.a: no new amplificatoins no new deletions</span></span>
<span id="cb34-2"><a href="copy-number-analyses.html#cb34-2" tabindex="-1"></a>hub015_postrad_a.b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb34-3"><a href="copy-number-analyses.html#cb34-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub015_postrad_a.b],</span>
<span id="cb34-4"><a href="copy-number-analyses.html#cb34-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub015_prerad_subclones],</span>
<span id="cb34-5"><a href="copy-number-analyses.html#cb34-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb34-6"><a href="copy-number-analyses.html#cb34-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb34-7"><a href="copy-number-analyses.html#cb34-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb34-8"><a href="copy-number-analyses.html#cb34-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb34-9"><a href="copy-number-analyses.html#cb34-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb34-10"><a href="copy-number-analyses.html#cb34-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="copy-number-analyses.html#cb36-1" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb36-2"><a href="copy-number-analyses.html#cb36-2" tabindex="-1"></a></span>
<span id="cb36-3"><a href="copy-number-analyses.html#cb36-3" tabindex="-1"></a><span class="co"># HUB106 #</span></span>
<span id="cb36-4"><a href="copy-number-analyses.html#cb36-4" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb36-5"><a href="copy-number-analyses.html#cb36-5" tabindex="-1"></a></span>
<span id="cb36-6"><a href="copy-number-analyses.html#cb36-6" tabindex="-1"></a><span class="co"># Extract HUB106</span></span>
<span id="cb36-7"><a href="copy-number-analyses.html#cb36-7" tabindex="-1"></a>hub106 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub106&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb36-8"><a href="copy-number-analyses.html#cb36-8" tabindex="-1"></a></span>
<span id="cb36-9"><a href="copy-number-analyses.html#cb36-9" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb36-10"><a href="copy-number-analyses.html#cb36-10" tabindex="-1"></a>hub106_prerad_subclones <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub106)]</span>
<span id="cb36-11"><a href="copy-number-analyses.html#cb36-11" tabindex="-1"></a></span>
<span id="cb36-12"><a href="copy-number-analyses.html#cb36-12" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb36-13"><a href="copy-number-analyses.html#cb36-13" tabindex="-1"></a>hub106_postrad_a <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_a&#39;</span>, hub106)]</span>
<span id="cb36-14"><a href="copy-number-analyses.html#cb36-14" tabindex="-1"></a>hub106_postrad_b <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_b&#39;</span>, hub106)]</span>
<span id="cb36-15"><a href="copy-number-analyses.html#cb36-15" tabindex="-1"></a>hub106_postrad_c <span class="ot">&lt;-</span> hub106[<span class="fu">grepl</span>(<span class="st">&#39;hub106_postrad_c&#39;</span>, hub106)]</span>
<span id="cb36-16"><a href="copy-number-analyses.html#cb36-16" tabindex="-1"></a></span>
<span id="cb36-17"><a href="copy-number-analyses.html#cb36-17" tabindex="-1"></a></span>
<span id="cb36-18"><a href="copy-number-analyses.html#cb36-18" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb36-19"><a href="copy-number-analyses.html#cb36-19" tabindex="-1"></a>hub106_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb36-20"><a href="copy-number-analyses.html#cb36-20" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_a],</span>
<span id="cb36-21"><a href="copy-number-analyses.html#cb36-21" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb36-22"><a href="copy-number-analyses.html#cb36-22" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb36-23"><a href="copy-number-analyses.html#cb36-23" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb36-24"><a href="copy-number-analyses.html#cb36-24" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb36-25"><a href="copy-number-analyses.html#cb36-25" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb36-26"><a href="copy-number-analyses.html#cb36-26" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb36-27"><a href="copy-number-analyses.html#cb36-27" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="copy-number-analyses.html#cb38-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb38-2"><a href="copy-number-analyses.html#cb38-2" tabindex="-1"></a>hub106_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb38-3"><a href="copy-number-analyses.html#cb38-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_b],</span>
<span id="cb38-4"><a href="copy-number-analyses.html#cb38-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb38-5"><a href="copy-number-analyses.html#cb38-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb38-6"><a href="copy-number-analyses.html#cb38-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb38-7"><a href="copy-number-analyses.html#cb38-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb38-8"><a href="copy-number-analyses.html#cb38-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb38-9"><a href="copy-number-analyses.html#cb38-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb38-10"><a href="copy-number-analyses.html#cb38-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="copy-number-analyses.html#cb40-1" tabindex="-1"></a><span class="co"># no new CNAs</span></span>
<span id="cb40-2"><a href="copy-number-analyses.html#cb40-2" tabindex="-1"></a>hub106_postrad_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb40-3"><a href="copy-number-analyses.html#cb40-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub106_postrad_c],</span>
<span id="cb40-4"><a href="copy-number-analyses.html#cb40-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub106_prerad_subclones],</span>
<span id="cb40-5"><a href="copy-number-analyses.html#cb40-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb40-6"><a href="copy-number-analyses.html#cb40-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb40-7"><a href="copy-number-analyses.html#cb40-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb40-8"><a href="copy-number-analyses.html#cb40-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb40-9"><a href="copy-number-analyses.html#cb40-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb40-10"><a href="copy-number-analyses.html#cb40-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="copy-number-analyses.html#cb42-1" tabindex="-1"></a><span class="co"># HUB062 #</span></span>
<span id="cb42-2"><a href="copy-number-analyses.html#cb42-2" tabindex="-1"></a><span class="do">######################################################################</span></span>
<span id="cb42-3"><a href="copy-number-analyses.html#cb42-3" tabindex="-1"></a>hub062 <span class="ot">&lt;-</span> <span class="fu">names</span>(subclone_list)[<span class="fu">grepl</span>(<span class="st">&#39;hub062&#39;</span>, <span class="fu">names</span>(subclone_list))]</span>
<span id="cb42-4"><a href="copy-number-analyses.html#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="copy-number-analyses.html#cb42-5" tabindex="-1"></a><span class="co"># combine prerad subclones and a.b</span></span>
<span id="cb42-6"><a href="copy-number-analyses.html#cb42-6" tabindex="-1"></a>hub062_prerad_subclones <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;prerad&#39;</span>, hub062)]</span>
<span id="cb42-7"><a href="copy-number-analyses.html#cb42-7" tabindex="-1"></a></span>
<span id="cb42-8"><a href="copy-number-analyses.html#cb42-8" tabindex="-1"></a><span class="co"># get each postrad subclone individually</span></span>
<span id="cb42-9"><a href="copy-number-analyses.html#cb42-9" tabindex="-1"></a>hub062_postrad_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_a&#39;</span>, hub062)]</span>
<span id="cb42-10"><a href="copy-number-analyses.html#cb42-10" tabindex="-1"></a>hub062_postrad_b <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_b&#39;</span>, hub062)]</span>
<span id="cb42-11"><a href="copy-number-analyses.html#cb42-11" tabindex="-1"></a>hub062_postrad_c <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c&#39;</span>, hub062)]</span>
<span id="cb42-12"><a href="copy-number-analyses.html#cb42-12" tabindex="-1"></a>hub062biological_postrad_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062biological_postrad_a&#39;</span>, hub062)]</span>
<span id="cb42-13"><a href="copy-number-analyses.html#cb42-13" tabindex="-1"></a>hub062_postrad_c2_a <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c2_a&#39;</span>, hub062)]</span>
<span id="cb42-14"><a href="copy-number-analyses.html#cb42-14" tabindex="-1"></a>hub062_postrad_c2_c <span class="ot">&lt;-</span> hub062[<span class="fu">grepl</span>(<span class="st">&#39;hub062_postrad_c2_c&#39;</span>, hub062)]</span>
<span id="cb42-15"><a href="copy-number-analyses.html#cb42-15" tabindex="-1"></a></span>
<span id="cb42-16"><a href="copy-number-analyses.html#cb42-16" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb42-17"><a href="copy-number-analyses.html#cb42-17" tabindex="-1"></a>hub062_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb42-18"><a href="copy-number-analyses.html#cb42-18" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_a],</span>
<span id="cb42-19"><a href="copy-number-analyses.html#cb42-19" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb42-20"><a href="copy-number-analyses.html#cb42-20" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb42-21"><a href="copy-number-analyses.html#cb42-21" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb42-22"><a href="copy-number-analyses.html#cb42-22" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb42-23"><a href="copy-number-analyses.html#cb42-23" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb42-24"><a href="copy-number-analyses.html#cb42-24" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb42-25"><a href="copy-number-analyses.html#cb42-25" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="copy-number-analyses.html#cb44-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb44-2"><a href="copy-number-analyses.html#cb44-2" tabindex="-1"></a>hub062_postrad_b <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb44-3"><a href="copy-number-analyses.html#cb44-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_b],</span>
<span id="cb44-4"><a href="copy-number-analyses.html#cb44-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb44-5"><a href="copy-number-analyses.html#cb44-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb44-6"><a href="copy-number-analyses.html#cb44-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb44-7"><a href="copy-number-analyses.html#cb44-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb44-8"><a href="copy-number-analyses.html#cb44-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb44-9"><a href="copy-number-analyses.html#cb44-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb44-10"><a href="copy-number-analyses.html#cb44-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="copy-number-analyses.html#cb46-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb46-2"><a href="copy-number-analyses.html#cb46-2" tabindex="-1"></a>hub062_postrad_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb46-3"><a href="copy-number-analyses.html#cb46-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c],</span>
<span id="cb46-4"><a href="copy-number-analyses.html#cb46-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb46-5"><a href="copy-number-analyses.html#cb46-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb46-6"><a href="copy-number-analyses.html#cb46-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb46-7"><a href="copy-number-analyses.html#cb46-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb46-8"><a href="copy-number-analyses.html#cb46-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb46-9"><a href="copy-number-analyses.html#cb46-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb46-10"><a href="copy-number-analyses.html#cb46-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="copy-number-analyses.html#cb48-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb48-2"><a href="copy-number-analyses.html#cb48-2" tabindex="-1"></a>hub062biological_postrad_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb48-3"><a href="copy-number-analyses.html#cb48-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062biological_postrad_a],</span>
<span id="cb48-4"><a href="copy-number-analyses.html#cb48-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb48-5"><a href="copy-number-analyses.html#cb48-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb48-6"><a href="copy-number-analyses.html#cb48-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb48-7"><a href="copy-number-analyses.html#cb48-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb48-8"><a href="copy-number-analyses.html#cb48-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb48-9"><a href="copy-number-analyses.html#cb48-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb48-10"><a href="copy-number-analyses.html#cb48-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="copy-number-analyses.html#cb50-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb50-2"><a href="copy-number-analyses.html#cb50-2" tabindex="-1"></a>hub062_postrad_c2_a <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb50-3"><a href="copy-number-analyses.html#cb50-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c2_a],</span>
<span id="cb50-4"><a href="copy-number-analyses.html#cb50-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb50-5"><a href="copy-number-analyses.html#cb50-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb50-6"><a href="copy-number-analyses.html#cb50-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb50-7"><a href="copy-number-analyses.html#cb50-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb50-8"><a href="copy-number-analyses.html#cb50-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb50-9"><a href="copy-number-analyses.html#cb50-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb50-10"><a href="copy-number-analyses.html#cb50-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="copy-number-analyses.html#cb52-1" tabindex="-1"></a><span class="co"># No new CNAs</span></span>
<span id="cb52-2"><a href="copy-number-analyses.html#cb52-2" tabindex="-1"></a>hub062_postrad_c2_c <span class="ot">&lt;-</span> <span class="fu">find_specific_cna_sc</span>(</span>
<span id="cb52-3"><a href="copy-number-analyses.html#cb52-3" tabindex="-1"></a>  <span class="at">list_resistant =</span> subclone_list[hub062_postrad_c2_c],</span>
<span id="cb52-4"><a href="copy-number-analyses.html#cb52-4" tabindex="-1"></a>  <span class="at">list_sensitive =</span> subclone_list[hub062_prerad_subclones],</span>
<span id="cb52-5"><a href="copy-number-analyses.html#cb52-5" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_resistant =</span> pcc,</span>
<span id="cb52-6"><a href="copy-number-analyses.html#cb52-6" tabindex="-1"></a>  <span class="at">perc_cutoff_within_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb52-7"><a href="copy-number-analyses.html#cb52-7" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_resistant =</span> <span class="fl">0.001</span>,</span>
<span id="cb52-8"><a href="copy-number-analyses.html#cb52-8" tabindex="-1"></a>  <span class="at">per_cutoff_across_subclone_sensitive =</span> <span class="fl">0.001</span>,</span>
<span id="cb52-9"><a href="copy-number-analyses.html#cb52-9" tabindex="-1"></a>  <span class="at">return_cnas_per_subclone =</span> F</span>
<span id="cb52-10"><a href="copy-number-analyses.html#cb52-10" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted segment info from files&quot;
## [1] &quot;Computing (this may take some minutes)...&quot;
## [1] &quot;Extracted CNAs that are shared by 50 and 0.1% of single cells within resistant and sensitive subclones, respectively, and that are shared by 0.1 and 0.1% across resistant and sensitive subclones, respectively.&quot;
## [1] &quot;Computing (few seconds)...&quot;
## [1] &quot;Identified amplifications and deletions that are unique to resistant or sensitive subclones&quot;</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="subclonal-dynamics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="copy-number-pattern-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/edit/main/04-cna.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/blob/main/04-cna.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
