<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</title>
  <meta name="description" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  
  
  

<meta name="author" content="Daan Andel" />


<meta name="date" content="2024-02-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="copy-number-pattern-analyses.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Subclonal Evolution Paper</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html"><i class="fa fa-check"></i><b>2</b> In vitro radiation response and clinical data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#progression-free-survival"><i class="fa fa-check"></i><b>2.1</b> Progression-free survival</a></li>
<li class="chapter" data-level="2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#in-vitro-dose-response"><i class="fa fa-check"></i><b>2.2</b> In vitro dose-response</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#dose-response-plot"><i class="fa fa-check"></i><b>2.2.1</b> Dose-response plot</a></li>
<li class="chapter" data-level="2.2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#rauc-across-multiple-experiments"><i class="fa fa-check"></i><b>2.2.2</b> rAUC across multiple experiments</a></li>
<li class="chapter" data-level="2.2.3" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#recovery-statistics-after-10-gy-radiation"><i class="fa fa-check"></i><b>2.2.3</b> Recovery statistics after 10 Gy radiation</a></li>
<li class="chapter" data-level="2.2.4" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#persisters-versus-shifters-dose-response-plots"><i class="fa fa-check"></i><b>2.2.4</b> Persisters versus shifters dose-response plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html"><i class="fa fa-check"></i><b>3</b> Subclonal dynamics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#aneufinder-run"><i class="fa fa-check"></i><b>3.1</b> AneuFinder run</a></li>
<li class="chapter" data-level="3.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#quality-selection"><i class="fa fa-check"></i><b>3.2</b> Quality selection</a></li>
<li class="chapter" data-level="3.3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#summary-statistics-of-copy-number-alterations"><i class="fa fa-check"></i><b>3.3</b> Summary statistics of copy number alterations</a></li>
<li class="chapter" data-level="3.4" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#pca-and-k-means-plots"><i class="fa fa-check"></i><b>3.4</b> PCA and K-means plots</a></li>
<li class="chapter" data-level="3.5" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-selection"><i class="fa fa-check"></i><b>3.5</b> Subclone selection</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#selecting-subclones"><i class="fa fa-check"></i><b>3.5.1</b> Selecting subclones</a></li>
<li class="chapter" data-level="3.5.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-statistics"><i class="fa fa-check"></i><b>3.5.2</b> Subclone statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#single-cell-copy-number-heatmap-plots"><i class="fa fa-check"></i><b>3.6</b> Single cell copy number heatmap plots</a></li>
<li class="chapter" data-level="3.7" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#fishers-exact-test"><i class="fa fa-check"></i><b>3.7</b> Fisher’s exact test</a></li>
<li class="chapter" data-level="3.8" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#medicc2"><i class="fa fa-check"></i><b>3.8</b> MEDICC2</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#wrangle"><i class="fa fa-check"></i><b>3.8.1</b> Wrangle</a></li>
<li class="chapter" data-level="3.8.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#medicc2-run"><i class="fa fa-check"></i><b>3.8.2</b> MEDICC2 run</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html"><i class="fa fa-check"></i><b>4</b> Copy number analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid"><i class="fa fa-check"></i><b>4.1</b> Colorectal cancer-specific arm-level copy number alterations per organoid</a></li>
<li class="chapter" data-level="4.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-per-subclone"><i class="fa fa-check"></i><b>4.2</b> Copy number alterations per subclone</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a"><i class="fa fa-check"></i><b>4.2.1</b> Copy number alterations specific to HUB062A</a></li>
<li class="chapter" data-level="4.2.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones"><i class="fa fa-check"></i><b>4.2.2</b> Copy number alterations specific to resistant subclones</a></li>
<li class="chapter" data-level="4.2.3" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#newly-acquired-copy-number-alterations"><i class="fa fa-check"></i><b>4.2.3</b> Newly acquired copy number alterations</a></li>
<li class="chapter" data-level="4.2.4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#ccle-ptpn13-and-aff1-analysis"><i class="fa fa-check"></i><b>4.2.4</b> CCLE PTPN13 and AFF1 analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html"><i class="fa fa-check"></i><b>5</b> Copy number pattern analyses</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#single-cell-analyses"><i class="fa fa-check"></i><b>5.1</b> Single cell analyses</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#heterogeneity-score"><i class="fa fa-check"></i><b>5.1.1</b> Heterogeneity score</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#medicc2-intraclone-variability"><i class="fa fa-check"></i><b>5.1.2</b> MEDICC2 intraclone variability</a></li>
<li class="chapter" data-level="5.1.3" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#copy-number-length-analysis"><i class="fa fa-check"></i><b>5.1.3</b> Copy number length analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#ccle-analysis"><i class="fa fa-check"></i><b>5.2</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#ploidy-and-whole-genome-duplication-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Ploidy and whole genome duplication analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#run-cin-signature"><i class="fa fa-check"></i><b>5.2.2</b> Run CIN signature</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#statistics-cin-signature"><i class="fa fa-check"></i><b>5.2.3</b> Statistics CIN signature</a></li>
<li class="chapter" data-level="5.2.4" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#plot-cx1-cin-signature"><i class="fa fa-check"></i><b>5.2.4</b> Plot CX1 CIN signature</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="source-code.html"><a href="source-code.html"><i class="fa fa-check"></i><b>6</b> Source code</a>
<ul>
<li class="chapter" data-level="6.1" data-path="source-code.html"><a href="source-code.html#auc_rad-script-auc_fun-function"><i class="fa fa-check"></i><b>6.1</b> AUC_rad script: ‘AUC_fun’ function</a></li>
<li class="chapter" data-level="6.2" data-path="source-code.html"><a href="source-code.html#fitter_rad-script-fitter-function"><i class="fa fa-check"></i><b>6.2</b> fitter_rad script: ‘fitter’ function</a></li>
<li class="chapter" data-level="6.3" data-path="source-code.html"><a href="source-code.html#ic50_rad-script-ic50_fun-function"><i class="fa fa-check"></i><b>6.3</b> ic50_rad script: ‘IC50_fun’ function</a></li>
<li class="chapter" data-level="6.4" data-path="source-code.html"><a href="source-code.html#relativize_dr-script-relative-function"><i class="fa fa-check"></i><b>6.4</b> relativize_dr script: ‘relative’ function</a></li>
<li class="chapter" data-level="6.5" data-path="source-code.html"><a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions"><i class="fa fa-check"></i><b>6.5</b> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions</a></li>
<li class="chapter" data-level="6.6" data-path="source-code.html"><a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions"><i class="fa fa-check"></i><b>6.6</b> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions</a></li>
<li class="chapter" data-level="6.7" data-path="source-code.html"><a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions"><i class="fa fa-check"></i><b>6.7</b> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions</a></li>
<li class="chapter" data-level="6.8" data-path="source-code.html"><a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function"><i class="fa fa-check"></i><b>6.8</b> plot_genomeheatmap script: ‘genomeheatmap’ function</a></li>
<li class="chapter" data-level="6.9" data-path="source-code.html"><a href="source-code.html#fisher_test-script-perform_fisher-function"><i class="fa fa-check"></i><b>6.9</b> fisher_test script: ‘perform_fisher’ function</a></li>
<li class="chapter" data-level="6.10" data-path="source-code.html"><a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function"><i class="fa fa-check"></i><b>6.10</b> find_specific_cna_sc script: ‘find_specific_cna_sc’ function</a></li>
<li class="chapter" data-level="6.11" data-path="source-code.html"><a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function"><i class="fa fa-check"></i><b>6.11</b> get_cytoband_coverage script: ‘get_cytoband_coverage’ function</a></li>
<li class="chapter" data-level="6.12" data-path="source-code.html"><a href="source-code.html#get_genes-script-get_genes-functoin"><i class="fa fa-check"></i><b>6.12</b> get_genes script: ‘get_genes’ functoin</a></li>
<li class="chapter" data-level="6.13" data-path="source-code.html"><a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function"><i class="fa fa-check"></i><b>6.13</b> get_oncogenes scripts: ‘get_oncogenes’ function</a></li>
<li class="chapter" data-level="6.14" data-path="source-code.html"><a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function"><i class="fa fa-check"></i><b>6.14</b> plot_genes_karyotype script: ‘plot_genes_karyotype’ function</a></li>
<li class="chapter" data-level="6.15" data-path="source-code.html"><a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions"><i class="fa fa-check"></i><b>6.15</b> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions</a></li>
<li class="chapter" data-level="6.16" data-path="source-code.html"><a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function"><i class="fa fa-check"></i><b>6.16</b> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function</a></li>
<li class="chapter" data-level="6.17" data-path="source-code.html"><a href="source-code.html#cna_profiler-script-cna_profiler-function"><i class="fa fa-check"></i><b>6.17</b> cna_profiler script: ‘cna_profiler’ function</a></li>
<li class="chapter" data-level="6.18" data-path="source-code.html"><a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function"><i class="fa fa-check"></i><b>6.18</b> cin_signature_ccle script: ‘run_cin_ccle’ function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="source-code" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Source code<a href="source-code.html#source-code" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The source code in this here is arranged in a loose chronological order, reflecting the sequence in which it is called in the scripts presented throughout this book.</p>
<div id="auc_rad-script-auc_fun-function" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> AUC_rad script: ‘AUC_fun’ function<a href="source-code.html#auc_rad-script-auc_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="source-code.html#cb173-1" tabindex="-1"></a><span class="co"># returns the AUC of the actual data.</span></span>
<span id="cb173-2"><a href="source-code.html#cb173-2" tabindex="-1"></a>AUC_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb173-3"><a href="source-code.html#cb173-3" tabindex="-1"></a>  </span>
<span id="cb173-4"><a href="source-code.html#cb173-4" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb173-5"><a href="source-code.html#cb173-5" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb173-6"><a href="source-code.html#cb173-6" tabindex="-1"></a>  </span>
<span id="cb173-7"><a href="source-code.html#cb173-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb173-8"><a href="source-code.html#cb173-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb173-9"><a href="source-code.html#cb173-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb173-10"><a href="source-code.html#cb173-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb173-11"><a href="source-code.html#cb173-11" tabindex="-1"></a>  </span>
<span id="cb173-12"><a href="source-code.html#cb173-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb173-13"><a href="source-code.html#cb173-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb173-14"><a href="source-code.html#cb173-14" tabindex="-1"></a>  </span>
<span id="cb173-15"><a href="source-code.html#cb173-15" tabindex="-1"></a>  <span class="co"># define dose_vetor</span></span>
<span id="cb173-16"><a href="source-code.html#cb173-16" tabindex="-1"></a>  dose_vector <span class="ot">&lt;-</span> <span class="fu">unique</span>(d<span class="sc">$</span>dose)</span>
<span id="cb173-17"><a href="source-code.html#cb173-17" tabindex="-1"></a>  </span>
<span id="cb173-18"><a href="source-code.html#cb173-18" tabindex="-1"></a>  </span>
<span id="cb173-19"><a href="source-code.html#cb173-19" tabindex="-1"></a>  <span class="co"># computes AUC for given predicted survival and dose given by dose_vector</span></span>
<span id="cb173-20"><a href="source-code.html#cb173-20" tabindex="-1"></a>  AUC_calc <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb173-21"><a href="source-code.html#cb173-21" tabindex="-1"></a>    <span class="co"># extract actual data</span></span>
<span id="cb173-22"><a href="source-code.html#cb173-22" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> df<span class="sc">$</span>relative_mean</span>
<span id="cb173-23"><a href="source-code.html#cb173-23" tabindex="-1"></a>    </span>
<span id="cb173-24"><a href="source-code.html#cb173-24" tabindex="-1"></a>    <span class="co"># extracts number of experiemnts </span></span>
<span id="cb173-25"><a href="source-code.html#cb173-25" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb173-26"><a href="source-code.html#cb173-26" tabindex="-1"></a>    <span class="co">#remove 2th and 3th number (are duplicates)</span></span>
<span id="cb173-27"><a href="source-code.html#cb173-27" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> rel_mean[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(rel_mean), n)]</span>
<span id="cb173-28"><a href="source-code.html#cb173-28" tabindex="-1"></a>    </span>
<span id="cb173-29"><a href="source-code.html#cb173-29" tabindex="-1"></a>    <span class="co"># compute maximal </span></span>
<span id="cb173-30"><a href="source-code.html#cb173-30" tabindex="-1"></a>    max_AUC <span class="ot">&lt;-</span> <span class="fu">max</span>(dose_vector)<span class="sc">*</span><span class="dv">1</span></span>
<span id="cb173-31"><a href="source-code.html#cb173-31" tabindex="-1"></a>    </span>
<span id="cb173-32"><a href="source-code.html#cb173-32" tabindex="-1"></a>    <span class="co">#compute AUC</span></span>
<span id="cb173-33"><a href="source-code.html#cb173-33" tabindex="-1"></a>    act_AUC <span class="ot">&lt;-</span> <span class="fu">AUC</span>(<span class="at">x=</span>dose_vector, <span class="at">y =</span> rel_mean)</span>
<span id="cb173-34"><a href="source-code.html#cb173-34" tabindex="-1"></a>    </span>
<span id="cb173-35"><a href="source-code.html#cb173-35" tabindex="-1"></a>    <span class="co"># make dataframe</span></span>
<span id="cb173-36"><a href="source-code.html#cb173-36" tabindex="-1"></a>    datafr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">max_AUC =</span> max_AUC,</span>
<span id="cb173-37"><a href="source-code.html#cb173-37" tabindex="-1"></a>                         <span class="at">act_AUC =</span> act_AUC, </span>
<span id="cb173-38"><a href="source-code.html#cb173-38" tabindex="-1"></a>                         <span class="at">rel_AUC =</span> act_AUC<span class="sc">/</span>max_AUC,</span>
<span id="cb173-39"><a href="source-code.html#cb173-39" tabindex="-1"></a>                         <span class="at">expcode =</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode),</span>
<span id="cb173-40"><a href="source-code.html#cb173-40" tabindex="-1"></a>                         <span class="at">cline=</span> <span class="fu">unique</span>(df<span class="sc">$</span>cline),</span>
<span id="cb173-41"><a href="source-code.html#cb173-41" tabindex="-1"></a>                         <span class="at">treatment =</span> <span class="fu">unique</span>(df<span class="sc">$</span>treatment ) )</span>
<span id="cb173-42"><a href="source-code.html#cb173-42" tabindex="-1"></a>    </span>
<span id="cb173-43"><a href="source-code.html#cb173-43" tabindex="-1"></a>    </span>
<span id="cb173-44"><a href="source-code.html#cb173-44" tabindex="-1"></a>    <span class="fu">return</span>(datafr)</span>
<span id="cb173-45"><a href="source-code.html#cb173-45" tabindex="-1"></a>  }</span>
<span id="cb173-46"><a href="source-code.html#cb173-46" tabindex="-1"></a>  </span>
<span id="cb173-47"><a href="source-code.html#cb173-47" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb173-48"><a href="source-code.html#cb173-48" tabindex="-1"></a>  AUC_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, AUC_calc) </span>
<span id="cb173-49"><a href="source-code.html#cb173-49" tabindex="-1"></a>  </span>
<span id="cb173-50"><a href="source-code.html#cb173-50" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb173-51"><a href="source-code.html#cb173-51" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, AUC_vector)</span>
<span id="cb173-52"><a href="source-code.html#cb173-52" tabindex="-1"></a>  </span>
<span id="cb173-53"><a href="source-code.html#cb173-53" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb173-54"><a href="source-code.html#cb173-54" tabindex="-1"></a>  </span>
<span id="cb173-55"><a href="source-code.html#cb173-55" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fitter_rad-script-fitter-function" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> fitter_rad script: ‘fitter’ function<a href="source-code.html#fitter_rad-script-fitter-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="source-code.html#cb174-1" tabindex="-1"></a>fitter <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb174-2"><a href="source-code.html#cb174-2" tabindex="-1"></a>  </span>
<span id="cb174-3"><a href="source-code.html#cb174-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb174-4"><a href="source-code.html#cb174-4" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb174-5"><a href="source-code.html#cb174-5" tabindex="-1"></a>  </span>
<span id="cb174-6"><a href="source-code.html#cb174-6" tabindex="-1"></a>  </span>
<span id="cb174-7"><a href="source-code.html#cb174-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb174-8"><a href="source-code.html#cb174-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb174-9"><a href="source-code.html#cb174-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb174-10"><a href="source-code.html#cb174-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb174-11"><a href="source-code.html#cb174-11" tabindex="-1"></a>  </span>
<span id="cb174-12"><a href="source-code.html#cb174-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb174-13"><a href="source-code.html#cb174-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb174-14"><a href="source-code.html#cb174-14" tabindex="-1"></a>  </span>
<span id="cb174-15"><a href="source-code.html#cb174-15" tabindex="-1"></a>  </span>
<span id="cb174-16"><a href="source-code.html#cb174-16" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb174-17"><a href="source-code.html#cb174-17" tabindex="-1"></a>  fit_model<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb174-18"><a href="source-code.html#cb174-18" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb174-19"><a href="source-code.html#cb174-19" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb174-20"><a href="source-code.html#cb174-20" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb174-21"><a href="source-code.html#cb174-21" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb174-22"><a href="source-code.html#cb174-22" tabindex="-1"></a>    </span>
<span id="cb174-23"><a href="source-code.html#cb174-23" tabindex="-1"></a>    newdata <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">dose=</span><span class="fu">exp</span>(<span class="fu">seq</span>(<span class="fu">log</span>(<span class="fl">0.00000001</span>), <span class="fu">log</span>(<span class="fu">max</span>(df<span class="sc">$</span>dose) <span class="sc">+</span> <span class="dv">1000</span>), <span class="at">length=</span><span class="dv">1000</span>))) <span class="co"># new data with doses. Note: lowest dose is not</span></span>
<span id="cb174-24"><a href="source-code.html#cb174-24" tabindex="-1"></a>    <span class="co"># log 0 but log(&#39;very small number&#39;) because otherwise this will hamper the scaling in ggplot later on. </span></span>
<span id="cb174-25"><a href="source-code.html#cb174-25" tabindex="-1"></a>    pm <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">newdata=</span>newdata, <span class="at">interval=</span><span class="st">&quot;confidence&quot;</span>) <span class="co"># new data with predictions and confidence intervals</span></span>
<span id="cb174-26"><a href="source-code.html#cb174-26" tabindex="-1"></a>    newdata<span class="sc">$</span>pred <span class="ot">&lt;-</span> pm[,<span class="dv">1</span>] <span class="co"># add prediction values to new data.</span></span>
<span id="cb174-27"><a href="source-code.html#cb174-27" tabindex="-1"></a>    newdata<span class="sc">$</span>predmin <span class="ot">&lt;-</span> pm[,<span class="dv">2</span>] <span class="co"># add lower bounderies to new data </span></span>
<span id="cb174-28"><a href="source-code.html#cb174-28" tabindex="-1"></a>    newdata<span class="sc">$</span>predmax <span class="ot">&lt;-</span> pm[,<span class="dv">3</span>] <span class="co"># add upper bounderies to new data </span></span>
<span id="cb174-29"><a href="source-code.html#cb174-29" tabindex="-1"></a>    newdata<span class="sc">$</span>cline <span class="ot">&lt;-</span> line<span class="sc">$</span>cline[<span class="dv">1</span>] <span class="co"># add column with cline.</span></span>
<span id="cb174-30"><a href="source-code.html#cb174-30" tabindex="-1"></a>    newdata<span class="sc">$</span>expcode <span class="ot">=</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode)</span>
<span id="cb174-31"><a href="source-code.html#cb174-31" tabindex="-1"></a>    <span class="fu">return</span>(newdata)</span>
<span id="cb174-32"><a href="source-code.html#cb174-32" tabindex="-1"></a>  }</span>
<span id="cb174-33"><a href="source-code.html#cb174-33" tabindex="-1"></a>  </span>
<span id="cb174-34"><a href="source-code.html#cb174-34" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb174-35"><a href="source-code.html#cb174-35" tabindex="-1"></a>  data_frames <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, fit_model) </span>
<span id="cb174-36"><a href="source-code.html#cb174-36" tabindex="-1"></a>  </span>
<span id="cb174-37"><a href="source-code.html#cb174-37" tabindex="-1"></a>  </span>
<span id="cb174-38"><a href="source-code.html#cb174-38" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb174-39"><a href="source-code.html#cb174-39" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, data_frames)</span>
<span id="cb174-40"><a href="source-code.html#cb174-40" tabindex="-1"></a>  </span>
<span id="cb174-41"><a href="source-code.html#cb174-41" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb174-42"><a href="source-code.html#cb174-42" tabindex="-1"></a>  </span>
<span id="cb174-43"><a href="source-code.html#cb174-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="ic50_rad-script-ic50_fun-function" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> ic50_rad script: ‘IC50_fun’ function<a href="source-code.html#ic50_rad-script-ic50_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="source-code.html#cb175-1" tabindex="-1"></a>IC50_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb175-2"><a href="source-code.html#cb175-2" tabindex="-1"></a>  </span>
<span id="cb175-3"><a href="source-code.html#cb175-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb175-4"><a href="source-code.html#cb175-4" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">is.null</span>(rad), d<span class="ot">&lt;-</span>df, d <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> rad))</span>
<span id="cb175-5"><a href="source-code.html#cb175-5" tabindex="-1"></a>  </span>
<span id="cb175-6"><a href="source-code.html#cb175-6" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb175-7"><a href="source-code.html#cb175-7" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb175-8"><a href="source-code.html#cb175-8" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb175-9"><a href="source-code.html#cb175-9" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb175-10"><a href="source-code.html#cb175-10" tabindex="-1"></a>  </span>
<span id="cb175-11"><a href="source-code.html#cb175-11" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb175-12"><a href="source-code.html#cb175-12" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span><span class="cn">TRUE</span>)</span>
<span id="cb175-13"><a href="source-code.html#cb175-13" tabindex="-1"></a>  </span>
<span id="cb175-14"><a href="source-code.html#cb175-14" tabindex="-1"></a>  </span>
<span id="cb175-15"><a href="source-code.html#cb175-15" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb175-16"><a href="source-code.html#cb175-16" tabindex="-1"></a>  ic50<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb175-17"><a href="source-code.html#cb175-17" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb175-18"><a href="source-code.html#cb175-18" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb175-19"><a href="source-code.html#cb175-19" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb175-20"><a href="source-code.html#cb175-20" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb175-21"><a href="source-code.html#cb175-21" tabindex="-1"></a>    </span>
<span id="cb175-22"><a href="source-code.html#cb175-22" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">coef</span>(fit)[<span class="dv">4</span>])</span>
<span id="cb175-23"><a href="source-code.html#cb175-23" tabindex="-1"></a>  }</span>
<span id="cb175-24"><a href="source-code.html#cb175-24" tabindex="-1"></a>  </span>
<span id="cb175-25"><a href="source-code.html#cb175-25" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb175-26"><a href="source-code.html#cb175-26" tabindex="-1"></a>  IC50_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, ic50) </span>
<span id="cb175-27"><a href="source-code.html#cb175-27" tabindex="-1"></a>  </span>
<span id="cb175-28"><a href="source-code.html#cb175-28" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb175-29"><a href="source-code.html#cb175-29" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, IC50_vector)</span>
<span id="cb175-30"><a href="source-code.html#cb175-30" tabindex="-1"></a>  </span>
<span id="cb175-31"><a href="source-code.html#cb175-31" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb175-32"><a href="source-code.html#cb175-32" tabindex="-1"></a>  </span>
<span id="cb175-33"><a href="source-code.html#cb175-33" tabindex="-1"></a>} </span></code></pre></div>
</div>
<div id="relativize_dr-script-relative-function" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> relativize_dr script: ‘relative’ function<a href="source-code.html#relativize_dr-script-relative-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="source-code.html#cb176-1" tabindex="-1"></a>relative <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb176-2"><a href="source-code.html#cb176-2" tabindex="-1"></a>  <span class="co">#&#39; @param df with dose response data and columns cline, dose, Exp, </span></span>
<span id="cb176-3"><a href="source-code.html#cb176-3" tabindex="-1"></a>  <span class="co">#&#39; rec_rad, expcode, ncolonies</span></span>
<span id="cb176-4"><a href="source-code.html#cb176-4" tabindex="-1"></a>  <span class="co">#&#39; @param rad if line received radiation (T or F), stored in column rec_rad</span></span>
<span id="cb176-5"><a href="source-code.html#cb176-5" tabindex="-1"></a>  <span class="co">#&#39; @return relativized data where dose = 0 is used as 100%</span></span>
<span id="cb176-6"><a href="source-code.html#cb176-6" tabindex="-1"></a>  </span>
<span id="cb176-7"><a href="source-code.html#cb176-7" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb176-8"><a href="source-code.html#cb176-8" tabindex="-1"></a>  <span class="fu">ifelse</span>(rad <span class="sc">==</span> T, d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">1</span>), d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">0</span>) )</span>
<span id="cb176-9"><a href="source-code.html#cb176-9" tabindex="-1"></a>  </span>
<span id="cb176-10"><a href="source-code.html#cb176-10" tabindex="-1"></a>  <span class="co"># define ncolonies</span></span>
<span id="cb176-11"><a href="source-code.html#cb176-11" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;ncolonies&quot;</span></span>
<span id="cb176-12"><a href="source-code.html#cb176-12" tabindex="-1"></a>  </span>
<span id="cb176-13"><a href="source-code.html#cb176-13" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb176-14"><a href="source-code.html#cb176-14" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb176-15"><a href="source-code.html#cb176-15" tabindex="-1"></a>  </span>
<span id="cb176-16"><a href="source-code.html#cb176-16" tabindex="-1"></a>  <span class="co"># extracts number of experiemnts </span></span>
<span id="cb176-17"><a href="source-code.html#cb176-17" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb176-18"><a href="source-code.html#cb176-18" tabindex="-1"></a>  </span>
<span id="cb176-19"><a href="source-code.html#cb176-19" tabindex="-1"></a>  <span class="co"># Takes mean every nth row</span></span>
<span id="cb176-20"><a href="source-code.html#cb176-20" tabindex="-1"></a>  </span>
<span id="cb176-21"><a href="source-code.html#cb176-21" tabindex="-1"></a>  mean_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb176-22"><a href="source-code.html#cb176-22" tabindex="-1"></a>    m <span class="ot">&lt;-</span> <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-23"><a href="source-code.html#cb176-23" tabindex="-1"></a>    <span class="fu">return</span>(m)</span>
<span id="cb176-24"><a href="source-code.html#cb176-24" tabindex="-1"></a>  }</span>
<span id="cb176-25"><a href="source-code.html#cb176-25" tabindex="-1"></a>  </span>
<span id="cb176-26"><a href="source-code.html#cb176-26" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))), mean_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb176-27"><a href="source-code.html#cb176-27" tabindex="-1"></a>  <span class="co"># Duplicates (n=2) or triplicates (n=3) the rows.</span></span>
<span id="cb176-28"><a href="source-code.html#cb176-28" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> mean[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(mean)), <span class="at">each =</span> n), ]</span>
<span id="cb176-29"><a href="source-code.html#cb176-29" tabindex="-1"></a>  <span class="co"># Adds everything to the dataframe. </span></span>
<span id="cb176-30"><a href="source-code.html#cb176-30" tabindex="-1"></a>  d[<span class="st">&quot;mean&quot;</span>] <span class="ot">&lt;-</span> mean</span>
<span id="cb176-31"><a href="source-code.html#cb176-31" tabindex="-1"></a>  </span>
<span id="cb176-32"><a href="source-code.html#cb176-32" tabindex="-1"></a>  <span class="co"># Select mean values of 0 concentration</span></span>
<span id="cb176-33"><a href="source-code.html#cb176-33" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb176-34"><a href="source-code.html#cb176-34" tabindex="-1"></a>    <span class="fu">group_by</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb176-35"><a href="source-code.html#cb176-35" tabindex="-1"></a>    <span class="fu">arrange</span>(cline,dose)</span>
<span id="cb176-36"><a href="source-code.html#cb176-36" tabindex="-1"></a>  </span>
<span id="cb176-37"><a href="source-code.html#cb176-37" tabindex="-1"></a>  first <span class="ot">&lt;-</span> d[d<span class="sc">$</span>dose<span class="sc">==</span><span class="dv">0</span>, ] <span class="sc">%&gt;%</span></span>
<span id="cb176-38"><a href="source-code.html#cb176-38" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(cline, <span class="at">value100=</span>mean)</span>
<span id="cb176-39"><a href="source-code.html#cb176-39" tabindex="-1"></a>  </span>
<span id="cb176-40"><a href="source-code.html#cb176-40" tabindex="-1"></a>  <span class="co"># Make cline a factor</span></span>
<span id="cb176-41"><a href="source-code.html#cb176-41" tabindex="-1"></a>  d<span class="sc">$</span>cline<span class="ot">&lt;-</span><span class="fu">factor</span>(d<span class="sc">$</span>cline)</span>
<span id="cb176-42"><a href="source-code.html#cb176-42" tabindex="-1"></a>  </span>
<span id="cb176-43"><a href="source-code.html#cb176-43" tabindex="-1"></a>  <span class="co"># Merge first original (d)</span></span>
<span id="cb176-44"><a href="source-code.html#cb176-44" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb176-45"><a href="source-code.html#cb176-45" tabindex="-1"></a>    <span class="fu">merge</span>(first,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;cline&quot;</span>))</span>
<span id="cb176-46"><a href="source-code.html#cb176-46" tabindex="-1"></a>  </span>
<span id="cb176-47"><a href="source-code.html#cb176-47" tabindex="-1"></a>  <span class="co"># Extract only every nth row.</span></span>
<span id="cb176-48"><a href="source-code.html#cb176-48" tabindex="-1"></a>  d <span class="ot">=</span> d[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(d), n), ]</span>
<span id="cb176-49"><a href="source-code.html#cb176-49" tabindex="-1"></a>  </span>
<span id="cb176-50"><a href="source-code.html#cb176-50" tabindex="-1"></a>  <span class="co"># make new column with relative, called relative</span></span>
<span id="cb176-51"><a href="source-code.html#cb176-51" tabindex="-1"></a>  d<span class="sc">$</span>relative <span class="ot">=</span> d<span class="sc">$</span>ncolonies<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb176-52"><a href="source-code.html#cb176-52" tabindex="-1"></a>  d<span class="sc">$</span>relative_mean <span class="ot">=</span> d<span class="sc">$</span>mean<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb176-53"><a href="source-code.html#cb176-53" tabindex="-1"></a>  </span>
<span id="cb176-54"><a href="source-code.html#cb176-54" tabindex="-1"></a>  d<span class="ot">&lt;-</span><span class="fu">subset</span>(d, <span class="at">select=</span><span class="sc">-</span><span class="fu">c</span>(mean,value100))</span>
<span id="cb176-55"><a href="source-code.html#cb176-55" tabindex="-1"></a>  </span>
<span id="cb176-56"><a href="source-code.html#cb176-56" tabindex="-1"></a>  <span class="co"># compute relative standard error of the mean</span></span>
<span id="cb176-57"><a href="source-code.html#cb176-57" tabindex="-1"></a>  <span class="co"># define the column with sem in it</span></span>
<span id="cb176-58"><a href="source-code.html#cb176-58" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;relative&quot;</span></span>
<span id="cb176-59"><a href="source-code.html#cb176-59" tabindex="-1"></a>  </span>
<span id="cb176-60"><a href="source-code.html#cb176-60" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb176-61"><a href="source-code.html#cb176-61" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb176-62"><a href="source-code.html#cb176-62" tabindex="-1"></a>  </span>
<span id="cb176-63"><a href="source-code.html#cb176-63" tabindex="-1"></a>  </span>
<span id="cb176-64"><a href="source-code.html#cb176-64" tabindex="-1"></a>  sem_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb176-65"><a href="source-code.html#cb176-65" tabindex="-1"></a>    std <span class="ot">&lt;-</span> <span class="fu">sd</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb176-66"><a href="source-code.html#cb176-66" tabindex="-1"></a>    vector <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(x)</span>
<span id="cb176-67"><a href="source-code.html#cb176-67" tabindex="-1"></a>    </span>
<span id="cb176-68"><a href="source-code.html#cb176-68" tabindex="-1"></a>    sem <span class="ot">&lt;-</span> std<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(vector))</span>
<span id="cb176-69"><a href="source-code.html#cb176-69" tabindex="-1"></a>    <span class="fu">return</span>(sem)</span>
<span id="cb176-70"><a href="source-code.html#cb176-70" tabindex="-1"></a>  }</span>
<span id="cb176-71"><a href="source-code.html#cb176-71" tabindex="-1"></a>  </span>
<span id="cb176-72"><a href="source-code.html#cb176-72" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))),sem_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb176-73"><a href="source-code.html#cb176-73" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> sem[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(sem)), <span class="at">each =</span> n), ]</span>
<span id="cb176-74"><a href="source-code.html#cb176-74" tabindex="-1"></a>  d[<span class="st">&quot;relative_sem&quot;</span>] <span class="ot">&lt;-</span> sem</span>
<span id="cb176-75"><a href="source-code.html#cb176-75" tabindex="-1"></a>  </span>
<span id="cb176-76"><a href="source-code.html#cb176-76" tabindex="-1"></a>  <span class="fu">return</span>(d)</span>
<span id="cb176-77"><a href="source-code.html#cb176-77" tabindex="-1"></a>  </span>
<span id="cb176-78"><a href="source-code.html#cb176-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions<a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="source-code.html#cb177-1" tabindex="-1"></a>quality_check <span class="ot">&lt;-</span> <span class="cf">function</span>(inputdir, model) {</span>
<span id="cb177-2"><a href="source-code.html#cb177-2" tabindex="-1"></a>  <span class="co">#&#39; @param inputdir is directory containing the standard aneufinder output for each plate</span></span>
<span id="cb177-3"><a href="source-code.html#cb177-3" tabindex="-1"></a>  <span class="co">#&#39; model is which model to use for the quality check: either dnacopy, edivisive or HMM</span></span>
<span id="cb177-4"><a href="source-code.html#cb177-4" tabindex="-1"></a>  <span class="co">#&#39; quality_check will make quality metrics for each plate</span></span>
<span id="cb177-5"><a href="source-code.html#cb177-5" tabindex="-1"></a>  <span class="co">#&#39; quality_check assumes that rdaBaseDirectory contains folder with models for each plate</span></span>
<span id="cb177-6"><a href="source-code.html#cb177-6" tabindex="-1"></a>  <span class="co">#&#39; each plate folder has to contain a MODEL folder and a method-&#39;dnacopy, edivisive, or HMM&#39; folder</span></span>
<span id="cb177-7"><a href="source-code.html#cb177-7" tabindex="-1"></a>  <span class="co">#&#39; @example quality_check(inputdir = rdaBaseDirectory, model = &#39;edivisive&#39;)</span></span>
<span id="cb177-8"><a href="source-code.html#cb177-8" tabindex="-1"></a>  </span>
<span id="cb177-9"><a href="source-code.html#cb177-9" tabindex="-1"></a>  <span class="co"># go to folder with appropiate files</span></span>
<span id="cb177-10"><a href="source-code.html#cb177-10" tabindex="-1"></a>  rda_folder <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputdir, <span class="st">&#39;/MODELS&#39;</span>, <span class="st">&#39;/method-&#39;</span>, model)</span>
<span id="cb177-11"><a href="source-code.html#cb177-11" tabindex="-1"></a>  </span>
<span id="cb177-12"><a href="source-code.html#cb177-12" tabindex="-1"></a>  <span class="co"># extract files</span></span>
<span id="cb177-13"><a href="source-code.html#cb177-13" tabindex="-1"></a>  rda_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(rda_folder, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb177-14"><a href="source-code.html#cb177-14" tabindex="-1"></a>  </span>
<span id="cb177-15"><a href="source-code.html#cb177-15" tabindex="-1"></a>  <span class="co"># run quality check on each file</span></span>
<span id="cb177-16"><a href="source-code.html#cb177-16" tabindex="-1"></a>  cl <span class="ot">&lt;-</span></span>
<span id="cb177-17"><a href="source-code.html#cb177-17" tabindex="-1"></a>    <span class="fu">clusterByQuality</span>(</span>
<span id="cb177-18"><a href="source-code.html#cb177-18" tabindex="-1"></a>      rda_files,</span>
<span id="cb177-19"><a href="source-code.html#cb177-19" tabindex="-1"></a>      <span class="at">measures =</span> <span class="fu">c</span>(</span>
<span id="cb177-20"><a href="source-code.html#cb177-20" tabindex="-1"></a>        <span class="st">&#39;spikiness&#39;</span>,</span>
<span id="cb177-21"><a href="source-code.html#cb177-21" tabindex="-1"></a>        <span class="st">&#39;num.segments&#39;</span>,</span>
<span id="cb177-22"><a href="source-code.html#cb177-22" tabindex="-1"></a>        <span class="st">&#39;entropy&#39;</span>,</span>
<span id="cb177-23"><a href="source-code.html#cb177-23" tabindex="-1"></a>        <span class="st">&#39;bhattacharyya&#39;</span>,</span>
<span id="cb177-24"><a href="source-code.html#cb177-24" tabindex="-1"></a>        <span class="st">&#39;sos&#39;</span></span>
<span id="cb177-25"><a href="source-code.html#cb177-25" tabindex="-1"></a>      )</span>
<span id="cb177-26"><a href="source-code.html#cb177-26" tabindex="-1"></a>    )</span>
<span id="cb177-27"><a href="source-code.html#cb177-27" tabindex="-1"></a>  </span>
<span id="cb177-28"><a href="source-code.html#cb177-28" tabindex="-1"></a>  <span class="fu">return</span>(cl)</span>
<span id="cb177-29"><a href="source-code.html#cb177-29" tabindex="-1"></a>  </span>
<span id="cb177-30"><a href="source-code.html#cb177-30" tabindex="-1"></a>}</span>
<span id="cb177-31"><a href="source-code.html#cb177-31" tabindex="-1"></a></span>
<span id="cb177-32"><a href="source-code.html#cb177-32" tabindex="-1"></a>quality_select <span class="ot">&lt;-</span> <span class="cf">function</span>(cl, spik, bhat) {</span>
<span id="cb177-33"><a href="source-code.html#cb177-33" tabindex="-1"></a>  <span class="co">#&#39; selects all files that meets the defined quality requirements</span></span>
<span id="cb177-34"><a href="source-code.html#cb177-34" tabindex="-1"></a>  <span class="co">#&#39; @param cl output of the clusterByQuality function in AneuFinder</span></span>
<span id="cb177-35"><a href="source-code.html#cb177-35" tabindex="-1"></a>  <span class="co">#&#39; @param spik the spikiness treshhold. All clusters with spikiness above</span></span>
<span id="cb177-36"><a href="source-code.html#cb177-36" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb177-37"><a href="source-code.html#cb177-37" tabindex="-1"></a>  <span class="co">#&#39; @param bhat the bhattacharrya score. All clusters with bhattacharrya below</span></span>
<span id="cb177-38"><a href="source-code.html#cb177-38" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb177-39"><a href="source-code.html#cb177-39" tabindex="-1"></a>  <span class="co">#&#39; @return returns a vector of selected files paths</span></span>
<span id="cb177-40"><a href="source-code.html#cb177-40" tabindex="-1"></a>  </span>
<span id="cb177-41"><a href="source-code.html#cb177-41" tabindex="-1"></a>  <span class="co"># convert to df to allow easy wrangling</span></span>
<span id="cb177-42"><a href="source-code.html#cb177-42" tabindex="-1"></a>  cl_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb177-43"><a href="source-code.html#cb177-43" tabindex="-1"></a>  </span>
<span id="cb177-44"><a href="source-code.html#cb177-44" tabindex="-1"></a>  <span class="co"># remove higher than spik</span></span>
<span id="cb177-45"><a href="source-code.html#cb177-45" tabindex="-1"></a>  spik <span class="ot">&lt;-</span> <span class="fu">subset</span>(cl_df, spikiness <span class="sc">&lt;</span> spik)</span>
<span id="cb177-46"><a href="source-code.html#cb177-46" tabindex="-1"></a>  </span>
<span id="cb177-47"><a href="source-code.html#cb177-47" tabindex="-1"></a>  <span class="co"># remove lower than bhat</span></span>
<span id="cb177-48"><a href="source-code.html#cb177-48" tabindex="-1"></a>  spik_bhat <span class="ot">&lt;-</span> <span class="fu">subset</span>(spik,  bhattacharyya <span class="sc">&gt;</span> bhat)</span>
<span id="cb177-49"><a href="source-code.html#cb177-49" tabindex="-1"></a>  </span>
<span id="cb177-50"><a href="source-code.html#cb177-50" tabindex="-1"></a>  cluster_n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(spik_bhat)</span>
<span id="cb177-51"><a href="source-code.html#cb177-51" tabindex="-1"></a>  </span>
<span id="cb177-52"><a href="source-code.html#cb177-52" tabindex="-1"></a>  <span class="co"># create vector of selected files</span></span>
<span id="cb177-53"><a href="source-code.html#cb177-53" tabindex="-1"></a>  selected.files <span class="ot">&lt;-</span> <span class="fu">unlist</span>(cl<span class="sc">$</span>classification[<span class="dv">0</span><span class="sc">:</span>cluster_n])</span>
<span id="cb177-54"><a href="source-code.html#cb177-54" tabindex="-1"></a>  </span>
<span id="cb177-55"><a href="source-code.html#cb177-55" tabindex="-1"></a>  <span class="fu">return</span>(selected.files)</span>
<span id="cb177-56"><a href="source-code.html#cb177-56" tabindex="-1"></a>  </span>
<span id="cb177-57"><a href="source-code.html#cb177-57" tabindex="-1"></a>}</span>
<span id="cb177-58"><a href="source-code.html#cb177-58" tabindex="-1"></a></span>
<span id="cb177-59"><a href="source-code.html#cb177-59" tabindex="-1"></a>check_quality <span class="ot">&lt;-</span> <span class="cf">function</span>(cl) {</span>
<span id="cb177-60"><a href="source-code.html#cb177-60" tabindex="-1"></a>  <span class="fu">return</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb177-61"><a href="source-code.html#cb177-61" tabindex="-1"></a>}</span>
<span id="cb177-62"><a href="source-code.html#cb177-62" tabindex="-1"></a>  </span>
<span id="cb177-63"><a href="source-code.html#cb177-63" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions<a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="source-code.html#cb178-1" tabindex="-1"></a>draw_pca <span class="ot">&lt;-</span></span>
<span id="cb178-2"><a href="source-code.html#cb178-2" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb178-3"><a href="source-code.html#cb178-3" tabindex="-1"></a>           list_files2,</span>
<span id="cb178-4"><a href="source-code.html#cb178-4" tabindex="-1"></a>           size,</span>
<span id="cb178-5"><a href="source-code.html#cb178-5" tabindex="-1"></a>           legend_position,</span>
<span id="cb178-6"><a href="source-code.html#cb178-6" tabindex="-1"></a>           baseline_kra,</span>
<span id="cb178-7"><a href="source-code.html#cb178-7" tabindex="-1"></a>           recurrence_kra,</span>
<span id="cb178-8"><a href="source-code.html#cb178-8" tabindex="-1"></a>           <span class="at">ssdna004 =</span> F) {</span>
<span id="cb178-9"><a href="source-code.html#cb178-9" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb178-10"><a href="source-code.html#cb178-10" tabindex="-1"></a>    </span>
<span id="cb178-11"><a href="source-code.html#cb178-11" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-12"><a href="source-code.html#cb178-12" tabindex="-1"></a>      cells,</span>
<span id="cb178-13"><a href="source-code.html#cb178-13" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb178-14"><a href="source-code.html#cb178-14" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb178-15"><a href="source-code.html#cb178-15" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb178-16"><a href="source-code.html#cb178-16" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb178-17"><a href="source-code.html#cb178-17" tabindex="-1"></a>    )</span>
<span id="cb178-18"><a href="source-code.html#cb178-18" tabindex="-1"></a>    </span>
<span id="cb178-19"><a href="source-code.html#cb178-19" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb178-20"><a href="source-code.html#cb178-20" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-21"><a href="source-code.html#cb178-21" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-22"><a href="source-code.html#cb178-22" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb178-23"><a href="source-code.html#cb178-23" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb178-24"><a href="source-code.html#cb178-24" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb178-25"><a href="source-code.html#cb178-25" tabindex="-1"></a>    </span>
<span id="cb178-26"><a href="source-code.html#cb178-26" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb178-27"><a href="source-code.html#cb178-27" tabindex="-1"></a>    </span>
<span id="cb178-28"><a href="source-code.html#cb178-28" tabindex="-1"></a>    col_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&quot;#999999&quot;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>)))</span>
<span id="cb178-29"><a href="source-code.html#cb178-29" tabindex="-1"></a>    label_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Recurrence&#39;</span>, <span class="st">&#39;Baseline&#39;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence&#39;</span>)))</span>
<span id="cb178-30"><a href="source-code.html#cb178-30" tabindex="-1"></a>    </span>
<span id="cb178-31"><a href="source-code.html#cb178-31" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb178-32"><a href="source-code.html#cb178-32" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb178-33"><a href="source-code.html#cb178-33" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb178-34"><a href="source-code.html#cb178-34" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb178-35"><a href="source-code.html#cb178-35" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb178-36"><a href="source-code.html#cb178-36" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb178-37"><a href="source-code.html#cb178-37" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb178-38"><a href="source-code.html#cb178-38" tabindex="-1"></a>        <span class="at">values =</span> col_vals[[<span class="dv">1</span>]],</span>
<span id="cb178-39"><a href="source-code.html#cb178-39" tabindex="-1"></a>        <span class="at">labels =</span> label_vals[[<span class="dv">1</span>]]</span>
<span id="cb178-40"><a href="source-code.html#cb178-40" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb178-41"><a href="source-code.html#cb178-41" tabindex="-1"></a>      </span>
<span id="cb178-42"><a href="source-code.html#cb178-42" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb178-43"><a href="source-code.html#cb178-43" tabindex="-1"></a>      </span>
<span id="cb178-44"><a href="source-code.html#cb178-44" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb178-45"><a href="source-code.html#cb178-45" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb178-46"><a href="source-code.html#cb178-46" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb178-47"><a href="source-code.html#cb178-47" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb178-48"><a href="source-code.html#cb178-48" tabindex="-1"></a>      </span>
<span id="cb178-49"><a href="source-code.html#cb178-49" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb178-50"><a href="source-code.html#cb178-50" tabindex="-1"></a>    p</span>
<span id="cb178-51"><a href="source-code.html#cb178-51" tabindex="-1"></a>    </span>
<span id="cb178-52"><a href="source-code.html#cb178-52" tabindex="-1"></a>  </span>
<span id="cb178-53"><a href="source-code.html#cb178-53" tabindex="-1"></a>    </span>
<span id="cb178-54"><a href="source-code.html#cb178-54" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb178-55"><a href="source-code.html#cb178-55" tabindex="-1"></a>    </span>
<span id="cb178-56"><a href="source-code.html#cb178-56" tabindex="-1"></a>  }</span>
<span id="cb178-57"><a href="source-code.html#cb178-57" tabindex="-1"></a></span>
<span id="cb178-58"><a href="source-code.html#cb178-58" tabindex="-1"></a></span>
<span id="cb178-59"><a href="source-code.html#cb178-59" tabindex="-1"></a>k_cluster <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb178-60"><a href="source-code.html#cb178-60" tabindex="-1"></a>                      list_files2,</span>
<span id="cb178-61"><a href="source-code.html#cb178-61" tabindex="-1"></a>                      cluster_n,</span>
<span id="cb178-62"><a href="source-code.html#cb178-62" tabindex="-1"></a>                      cols,</span>
<span id="cb178-63"><a href="source-code.html#cb178-63" tabindex="-1"></a>                      labels,</span>
<span id="cb178-64"><a href="source-code.html#cb178-64" tabindex="-1"></a>                      legend_position,</span>
<span id="cb178-65"><a href="source-code.html#cb178-65" tabindex="-1"></a>                      normalize,</span>
<span id="cb178-66"><a href="source-code.html#cb178-66" tabindex="-1"></a>                      <span class="at">return_elbow =</span> F) {</span>
<span id="cb178-67"><a href="source-code.html#cb178-67" tabindex="-1"></a>  </span>
<span id="cb178-68"><a href="source-code.html#cb178-68" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb178-69"><a href="source-code.html#cb178-69" tabindex="-1"></a>  </span>
<span id="cb178-70"><a href="source-code.html#cb178-70" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb178-71"><a href="source-code.html#cb178-71" tabindex="-1"></a>  </span>
<span id="cb178-72"><a href="source-code.html#cb178-72" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-73"><a href="source-code.html#cb178-73" tabindex="-1"></a>    cells,</span>
<span id="cb178-74"><a href="source-code.html#cb178-74" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-75"><a href="source-code.html#cb178-75" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb178-76"><a href="source-code.html#cb178-76" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb178-77"><a href="source-code.html#cb178-77" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-78"><a href="source-code.html#cb178-78" tabindex="-1"></a>  )</span>
<span id="cb178-79"><a href="source-code.html#cb178-79" tabindex="-1"></a></span>
<span id="cb178-80"><a href="source-code.html#cb178-80" tabindex="-1"></a>  </span>
<span id="cb178-81"><a href="source-code.html#cb178-81" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-82"><a href="source-code.html#cb178-82" tabindex="-1"></a>    cells,</span>
<span id="cb178-83"><a href="source-code.html#cb178-83" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-84"><a href="source-code.html#cb178-84" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb178-85"><a href="source-code.html#cb178-85" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb178-86"><a href="source-code.html#cb178-86" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-87"><a href="source-code.html#cb178-87" tabindex="-1"></a>  )</span>
<span id="cb178-88"><a href="source-code.html#cb178-88" tabindex="-1"></a>  </span>
<span id="cb178-89"><a href="source-code.html#cb178-89" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-90"><a href="source-code.html#cb178-90" tabindex="-1"></a>    cells,</span>
<span id="cb178-91"><a href="source-code.html#cb178-91" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-92"><a href="source-code.html#cb178-92" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb178-93"><a href="source-code.html#cb178-93" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb178-94"><a href="source-code.html#cb178-94" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-95"><a href="source-code.html#cb178-95" tabindex="-1"></a>  )</span>
<span id="cb178-96"><a href="source-code.html#cb178-96" tabindex="-1"></a>  </span>
<span id="cb178-97"><a href="source-code.html#cb178-97" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-98"><a href="source-code.html#cb178-98" tabindex="-1"></a>    cells,</span>
<span id="cb178-99"><a href="source-code.html#cb178-99" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-100"><a href="source-code.html#cb178-100" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb178-101"><a href="source-code.html#cb178-101" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb178-102"><a href="source-code.html#cb178-102" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-103"><a href="source-code.html#cb178-103" tabindex="-1"></a>  )</span>
<span id="cb178-104"><a href="source-code.html#cb178-104" tabindex="-1"></a>  </span>
<span id="cb178-105"><a href="source-code.html#cb178-105" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-106"><a href="source-code.html#cb178-106" tabindex="-1"></a>    cells,</span>
<span id="cb178-107"><a href="source-code.html#cb178-107" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-108"><a href="source-code.html#cb178-108" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb178-109"><a href="source-code.html#cb178-109" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb178-110"><a href="source-code.html#cb178-110" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-111"><a href="source-code.html#cb178-111" tabindex="-1"></a>  )</span>
<span id="cb178-112"><a href="source-code.html#cb178-112" tabindex="-1"></a>  </span>
<span id="cb178-113"><a href="source-code.html#cb178-113" tabindex="-1"></a>  </span>
<span id="cb178-114"><a href="source-code.html#cb178-114" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb178-115"><a href="source-code.html#cb178-115" tabindex="-1"></a>  </span>
<span id="cb178-116"><a href="source-code.html#cb178-116" tabindex="-1"></a>  </span>
<span id="cb178-117"><a href="source-code.html#cb178-117" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb178-118"><a href="source-code.html#cb178-118" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb178-119"><a href="source-code.html#cb178-119" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb178-120"><a href="source-code.html#cb178-120" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb178-121"><a href="source-code.html#cb178-121" tabindex="-1"></a>    </span>
<span id="cb178-122"><a href="source-code.html#cb178-122" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb178-123"><a href="source-code.html#cb178-123" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb178-124"><a href="source-code.html#cb178-124" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb178-125"><a href="source-code.html#cb178-125" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb178-126"><a href="source-code.html#cb178-126" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb178-127"><a href="source-code.html#cb178-127" tabindex="-1"></a>    }</span>
<span id="cb178-128"><a href="source-code.html#cb178-128" tabindex="-1"></a>    </span>
<span id="cb178-129"><a href="source-code.html#cb178-129" tabindex="-1"></a>    </span>
<span id="cb178-130"><a href="source-code.html#cb178-130" tabindex="-1"></a>    </span>
<span id="cb178-131"><a href="source-code.html#cb178-131" tabindex="-1"></a>    </span>
<span id="cb178-132"><a href="source-code.html#cb178-132" tabindex="-1"></a>    </span>
<span id="cb178-133"><a href="source-code.html#cb178-133" tabindex="-1"></a>  }</span>
<span id="cb178-134"><a href="source-code.html#cb178-134" tabindex="-1"></a>  </span>
<span id="cb178-135"><a href="source-code.html#cb178-135" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb178-136"><a href="source-code.html#cb178-136" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb178-137"><a href="source-code.html#cb178-137" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb178-138"><a href="source-code.html#cb178-138" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Elbow method&quot;</span>))</span>
<span id="cb178-139"><a href="source-code.html#cb178-139" tabindex="-1"></a>    </span>
<span id="cb178-140"><a href="source-code.html#cb178-140" tabindex="-1"></a>  }</span>
<span id="cb178-141"><a href="source-code.html#cb178-141" tabindex="-1"></a>  </span>
<span id="cb178-142"><a href="source-code.html#cb178-142" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb178-143"><a href="source-code.html#cb178-143" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb178-144"><a href="source-code.html#cb178-144" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb178-145"><a href="source-code.html#cb178-145" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb178-146"><a href="source-code.html#cb178-146" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb178-147"><a href="source-code.html#cb178-147" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb178-148"><a href="source-code.html#cb178-148" tabindex="-1"></a>  </span>
<span id="cb178-149"><a href="source-code.html#cb178-149" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb178-150"><a href="source-code.html#cb178-150" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb178-151"><a href="source-code.html#cb178-151" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb178-152"><a href="source-code.html#cb178-152" tabindex="-1"></a>  </span>
<span id="cb178-153"><a href="source-code.html#cb178-153" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb178-154"><a href="source-code.html#cb178-154" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb178-155"><a href="source-code.html#cb178-155" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-156"><a href="source-code.html#cb178-156" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-157"><a href="source-code.html#cb178-157" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb178-158"><a href="source-code.html#cb178-158" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb178-159"><a href="source-code.html#cb178-159" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb178-160"><a href="source-code.html#cb178-160" tabindex="-1"></a>  </span>
<span id="cb178-161"><a href="source-code.html#cb178-161" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb178-162"><a href="source-code.html#cb178-162" tabindex="-1"></a>  </span>
<span id="cb178-163"><a href="source-code.html#cb178-163" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb178-164"><a href="source-code.html#cb178-164" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb178-165"><a href="source-code.html#cb178-165" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb178-166"><a href="source-code.html#cb178-166" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb178-167"><a href="source-code.html#cb178-167" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb178-168"><a href="source-code.html#cb178-168" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb178-169"><a href="source-code.html#cb178-169" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb178-170"><a href="source-code.html#cb178-170" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb178-171"><a href="source-code.html#cb178-171" tabindex="-1"></a>    </span>
<span id="cb178-172"><a href="source-code.html#cb178-172" tabindex="-1"></a>    </span>
<span id="cb178-173"><a href="source-code.html#cb178-173" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb178-174"><a href="source-code.html#cb178-174" tabindex="-1"></a>    </span>
<span id="cb178-175"><a href="source-code.html#cb178-175" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb178-176"><a href="source-code.html#cb178-176" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb178-177"><a href="source-code.html#cb178-177" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb178-178"><a href="source-code.html#cb178-178" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb178-179"><a href="source-code.html#cb178-179" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb178-180"><a href="source-code.html#cb178-180" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb178-181"><a href="source-code.html#cb178-181" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb178-182"><a href="source-code.html#cb178-182" tabindex="-1"></a>  </span>
<span id="cb178-183"><a href="source-code.html#cb178-183" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb178-184"><a href="source-code.html#cb178-184" tabindex="-1"></a>  </span>
<span id="cb178-185"><a href="source-code.html#cb178-185" tabindex="-1"></a>  <span class="co">#ggplotly(p)</span></span>
<span id="cb178-186"><a href="source-code.html#cb178-186" tabindex="-1"></a>  </span>
<span id="cb178-187"><a href="source-code.html#cb178-187" tabindex="-1"></a>}</span>
<span id="cb178-188"><a href="source-code.html#cb178-188" tabindex="-1"></a></span>
<span id="cb178-189"><a href="source-code.html#cb178-189" tabindex="-1"></a>draw_pca_double <span class="ot">&lt;-</span></span>
<span id="cb178-190"><a href="source-code.html#cb178-190" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb178-191"><a href="source-code.html#cb178-191" tabindex="-1"></a>           list_files2,</span>
<span id="cb178-192"><a href="source-code.html#cb178-192" tabindex="-1"></a>           list_files3,</span>
<span id="cb178-193"><a href="source-code.html#cb178-193" tabindex="-1"></a>           size,</span>
<span id="cb178-194"><a href="source-code.html#cb178-194" tabindex="-1"></a>           legend_position) {</span>
<span id="cb178-195"><a href="source-code.html#cb178-195" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb178-196"><a href="source-code.html#cb178-196" tabindex="-1"></a>    </span>
<span id="cb178-197"><a href="source-code.html#cb178-197" tabindex="-1"></a>    </span>
<span id="cb178-198"><a href="source-code.html#cb178-198" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-199"><a href="source-code.html#cb178-199" tabindex="-1"></a>      cells,</span>
<span id="cb178-200"><a href="source-code.html#cb178-200" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb178-201"><a href="source-code.html#cb178-201" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb178-202"><a href="source-code.html#cb178-202" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb178-203"><a href="source-code.html#cb178-203" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb178-204"><a href="source-code.html#cb178-204" tabindex="-1"></a>    )</span>
<span id="cb178-205"><a href="source-code.html#cb178-205" tabindex="-1"></a>    </span>
<span id="cb178-206"><a href="source-code.html#cb178-206" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb178-207"><a href="source-code.html#cb178-207" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-208"><a href="source-code.html#cb178-208" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-209"><a href="source-code.html#cb178-209" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb178-210"><a href="source-code.html#cb178-210" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb178-211"><a href="source-code.html#cb178-211" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb178-212"><a href="source-code.html#cb178-212" tabindex="-1"></a>    </span>
<span id="cb178-213"><a href="source-code.html#cb178-213" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb178-214"><a href="source-code.html#cb178-214" tabindex="-1"></a>    </span>
<span id="cb178-215"><a href="source-code.html#cb178-215" tabindex="-1"></a>    </span>
<span id="cb178-216"><a href="source-code.html#cb178-216" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb178-217"><a href="source-code.html#cb178-217" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb178-218"><a href="source-code.html#cb178-218" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb178-219"><a href="source-code.html#cb178-219" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb178-220"><a href="source-code.html#cb178-220" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb178-221"><a href="source-code.html#cb178-221" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb178-222"><a href="source-code.html#cb178-222" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb178-223"><a href="source-code.html#cb178-223" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&#39;#d65c4e&#39;</span>),</span>
<span id="cb178-224"><a href="source-code.html#cb178-224" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence Cycle 1&#39;</span>, <span class="st">&#39;Recurrence Cycle 2&#39;</span>)</span>
<span id="cb178-225"><a href="source-code.html#cb178-225" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb178-226"><a href="source-code.html#cb178-226" tabindex="-1"></a>      </span>
<span id="cb178-227"><a href="source-code.html#cb178-227" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb178-228"><a href="source-code.html#cb178-228" tabindex="-1"></a>      </span>
<span id="cb178-229"><a href="source-code.html#cb178-229" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb178-230"><a href="source-code.html#cb178-230" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb178-231"><a href="source-code.html#cb178-231" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb178-232"><a href="source-code.html#cb178-232" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb178-233"><a href="source-code.html#cb178-233" tabindex="-1"></a>      </span>
<span id="cb178-234"><a href="source-code.html#cb178-234" tabindex="-1"></a>      </span>
<span id="cb178-235"><a href="source-code.html#cb178-235" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb178-236"><a href="source-code.html#cb178-236" tabindex="-1"></a>    <span class="fu">return</span>(p)</span>
<span id="cb178-237"><a href="source-code.html#cb178-237" tabindex="-1"></a>    </span>
<span id="cb178-238"><a href="source-code.html#cb178-238" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb178-239"><a href="source-code.html#cb178-239" tabindex="-1"></a>    </span>
<span id="cb178-240"><a href="source-code.html#cb178-240" tabindex="-1"></a>  }</span>
<span id="cb178-241"><a href="source-code.html#cb178-241" tabindex="-1"></a></span>
<span id="cb178-242"><a href="source-code.html#cb178-242" tabindex="-1"></a></span>
<span id="cb178-243"><a href="source-code.html#cb178-243" tabindex="-1"></a>k_cluster_double <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb178-244"><a href="source-code.html#cb178-244" tabindex="-1"></a>                             list_files2,</span>
<span id="cb178-245"><a href="source-code.html#cb178-245" tabindex="-1"></a>                             list_files3,</span>
<span id="cb178-246"><a href="source-code.html#cb178-246" tabindex="-1"></a>                             </span>
<span id="cb178-247"><a href="source-code.html#cb178-247" tabindex="-1"></a>                             cluster_n,</span>
<span id="cb178-248"><a href="source-code.html#cb178-248" tabindex="-1"></a>                             cols,</span>
<span id="cb178-249"><a href="source-code.html#cb178-249" tabindex="-1"></a>                             labels,</span>
<span id="cb178-250"><a href="source-code.html#cb178-250" tabindex="-1"></a>                             legend_position,</span>
<span id="cb178-251"><a href="source-code.html#cb178-251" tabindex="-1"></a>                             normalize,</span>
<span id="cb178-252"><a href="source-code.html#cb178-252" tabindex="-1"></a>                             <span class="at">return_elbow =</span> F) {</span>
<span id="cb178-253"><a href="source-code.html#cb178-253" tabindex="-1"></a>  </span>
<span id="cb178-254"><a href="source-code.html#cb178-254" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb178-255"><a href="source-code.html#cb178-255" tabindex="-1"></a>  </span>
<span id="cb178-256"><a href="source-code.html#cb178-256" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb178-257"><a href="source-code.html#cb178-257" tabindex="-1"></a>  </span>
<span id="cb178-258"><a href="source-code.html#cb178-258" tabindex="-1"></a>  </span>
<span id="cb178-259"><a href="source-code.html#cb178-259" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-260"><a href="source-code.html#cb178-260" tabindex="-1"></a>    cells,</span>
<span id="cb178-261"><a href="source-code.html#cb178-261" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-262"><a href="source-code.html#cb178-262" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb178-263"><a href="source-code.html#cb178-263" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb178-264"><a href="source-code.html#cb178-264" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-265"><a href="source-code.html#cb178-265" tabindex="-1"></a>  )</span>
<span id="cb178-266"><a href="source-code.html#cb178-266" tabindex="-1"></a>  </span>
<span id="cb178-267"><a href="source-code.html#cb178-267" tabindex="-1"></a>  </span>
<span id="cb178-268"><a href="source-code.html#cb178-268" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-269"><a href="source-code.html#cb178-269" tabindex="-1"></a>    cells,</span>
<span id="cb178-270"><a href="source-code.html#cb178-270" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-271"><a href="source-code.html#cb178-271" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb178-272"><a href="source-code.html#cb178-272" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb178-273"><a href="source-code.html#cb178-273" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-274"><a href="source-code.html#cb178-274" tabindex="-1"></a>  )</span>
<span id="cb178-275"><a href="source-code.html#cb178-275" tabindex="-1"></a>  </span>
<span id="cb178-276"><a href="source-code.html#cb178-276" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-277"><a href="source-code.html#cb178-277" tabindex="-1"></a>    cells,</span>
<span id="cb178-278"><a href="source-code.html#cb178-278" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-279"><a href="source-code.html#cb178-279" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb178-280"><a href="source-code.html#cb178-280" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb178-281"><a href="source-code.html#cb178-281" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-282"><a href="source-code.html#cb178-282" tabindex="-1"></a>  )</span>
<span id="cb178-283"><a href="source-code.html#cb178-283" tabindex="-1"></a>  </span>
<span id="cb178-284"><a href="source-code.html#cb178-284" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-285"><a href="source-code.html#cb178-285" tabindex="-1"></a>    cells,</span>
<span id="cb178-286"><a href="source-code.html#cb178-286" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-287"><a href="source-code.html#cb178-287" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb178-288"><a href="source-code.html#cb178-288" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb178-289"><a href="source-code.html#cb178-289" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-290"><a href="source-code.html#cb178-290" tabindex="-1"></a>  )</span>
<span id="cb178-291"><a href="source-code.html#cb178-291" tabindex="-1"></a>  </span>
<span id="cb178-292"><a href="source-code.html#cb178-292" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb178-293"><a href="source-code.html#cb178-293" tabindex="-1"></a>    cells,</span>
<span id="cb178-294"><a href="source-code.html#cb178-294" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb178-295"><a href="source-code.html#cb178-295" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb178-296"><a href="source-code.html#cb178-296" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb178-297"><a href="source-code.html#cb178-297" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb178-298"><a href="source-code.html#cb178-298" tabindex="-1"></a>  )</span>
<span id="cb178-299"><a href="source-code.html#cb178-299" tabindex="-1"></a>  </span>
<span id="cb178-300"><a href="source-code.html#cb178-300" tabindex="-1"></a>  </span>
<span id="cb178-301"><a href="source-code.html#cb178-301" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb178-302"><a href="source-code.html#cb178-302" tabindex="-1"></a>  </span>
<span id="cb178-303"><a href="source-code.html#cb178-303" tabindex="-1"></a>  </span>
<span id="cb178-304"><a href="source-code.html#cb178-304" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb178-305"><a href="source-code.html#cb178-305" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb178-306"><a href="source-code.html#cb178-306" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb178-307"><a href="source-code.html#cb178-307" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb178-308"><a href="source-code.html#cb178-308" tabindex="-1"></a>    </span>
<span id="cb178-309"><a href="source-code.html#cb178-309" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb178-310"><a href="source-code.html#cb178-310" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb178-311"><a href="source-code.html#cb178-311" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb178-312"><a href="source-code.html#cb178-312" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb178-313"><a href="source-code.html#cb178-313" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb178-314"><a href="source-code.html#cb178-314" tabindex="-1"></a>    }</span>
<span id="cb178-315"><a href="source-code.html#cb178-315" tabindex="-1"></a>    </span>
<span id="cb178-316"><a href="source-code.html#cb178-316" tabindex="-1"></a>    </span>
<span id="cb178-317"><a href="source-code.html#cb178-317" tabindex="-1"></a>    </span>
<span id="cb178-318"><a href="source-code.html#cb178-318" tabindex="-1"></a>    </span>
<span id="cb178-319"><a href="source-code.html#cb178-319" tabindex="-1"></a>    </span>
<span id="cb178-320"><a href="source-code.html#cb178-320" tabindex="-1"></a>  }</span>
<span id="cb178-321"><a href="source-code.html#cb178-321" tabindex="-1"></a>  </span>
<span id="cb178-322"><a href="source-code.html#cb178-322" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb178-323"><a href="source-code.html#cb178-323" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb178-324"><a href="source-code.html#cb178-324" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb178-325"><a href="source-code.html#cb178-325" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Silhouette method&quot;</span>))</span>
<span id="cb178-326"><a href="source-code.html#cb178-326" tabindex="-1"></a>    </span>
<span id="cb178-327"><a href="source-code.html#cb178-327" tabindex="-1"></a>  }</span>
<span id="cb178-328"><a href="source-code.html#cb178-328" tabindex="-1"></a>  </span>
<span id="cb178-329"><a href="source-code.html#cb178-329" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb178-330"><a href="source-code.html#cb178-330" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb178-331"><a href="source-code.html#cb178-331" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb178-332"><a href="source-code.html#cb178-332" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb178-333"><a href="source-code.html#cb178-333" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb178-334"><a href="source-code.html#cb178-334" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb178-335"><a href="source-code.html#cb178-335" tabindex="-1"></a>  </span>
<span id="cb178-336"><a href="source-code.html#cb178-336" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb178-337"><a href="source-code.html#cb178-337" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb178-338"><a href="source-code.html#cb178-338" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb178-339"><a href="source-code.html#cb178-339" tabindex="-1"></a>  </span>
<span id="cb178-340"><a href="source-code.html#cb178-340" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb178-341"><a href="source-code.html#cb178-341" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb178-342"><a href="source-code.html#cb178-342" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-343"><a href="source-code.html#cb178-343" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb178-344"><a href="source-code.html#cb178-344" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb178-345"><a href="source-code.html#cb178-345" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb178-346"><a href="source-code.html#cb178-346" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb178-347"><a href="source-code.html#cb178-347" tabindex="-1"></a>  </span>
<span id="cb178-348"><a href="source-code.html#cb178-348" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb178-349"><a href="source-code.html#cb178-349" tabindex="-1"></a>  </span>
<span id="cb178-350"><a href="source-code.html#cb178-350" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb178-351"><a href="source-code.html#cb178-351" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb178-352"><a href="source-code.html#cb178-352" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb178-353"><a href="source-code.html#cb178-353" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb178-354"><a href="source-code.html#cb178-354" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb178-355"><a href="source-code.html#cb178-355" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb178-356"><a href="source-code.html#cb178-356" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb178-357"><a href="source-code.html#cb178-357" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb178-358"><a href="source-code.html#cb178-358" tabindex="-1"></a>    </span>
<span id="cb178-359"><a href="source-code.html#cb178-359" tabindex="-1"></a>    </span>
<span id="cb178-360"><a href="source-code.html#cb178-360" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb178-361"><a href="source-code.html#cb178-361" tabindex="-1"></a>    </span>
<span id="cb178-362"><a href="source-code.html#cb178-362" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb178-363"><a href="source-code.html#cb178-363" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb178-364"><a href="source-code.html#cb178-364" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb178-365"><a href="source-code.html#cb178-365" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb178-366"><a href="source-code.html#cb178-366" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb178-367"><a href="source-code.html#cb178-367" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb178-368"><a href="source-code.html#cb178-368" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb178-369"><a href="source-code.html#cb178-369" tabindex="-1"></a>  </span>
<span id="cb178-370"><a href="source-code.html#cb178-370" tabindex="-1"></a>  p</span>
<span id="cb178-371"><a href="source-code.html#cb178-371" tabindex="-1"></a>  </span>
<span id="cb178-372"><a href="source-code.html#cb178-372" tabindex="-1"></a>}</span>
<span id="cb178-373"><a href="source-code.html#cb178-373" tabindex="-1"></a></span>
<span id="cb178-374"><a href="source-code.html#cb178-374" tabindex="-1"></a>  </span>
<span id="cb178-375"><a href="source-code.html#cb178-375" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions<a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="source-code.html#cb179-1" tabindex="-1"></a>extract_other_cells <span class="ot">&lt;-</span> <span class="cf">function</span>(kra_name, list_of_subclones) {</span>
<span id="cb179-2"><a href="source-code.html#cb179-2" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(man_select_files_edivisive) <span class="sc">==</span> kra_name)</span>
<span id="cb179-3"><a href="source-code.html#cb179-3" tabindex="-1"></a>  all <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(man_select_files_edivisive[[i]],  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>) </span>
<span id="cb179-4"><a href="source-code.html#cb179-4" tabindex="-1"></a>  rest <span class="ot">&lt;-</span> all[<span class="fu">which</span>(<span class="sc">!</span>all <span class="sc">%in%</span> list_of_subclones)]</span>
<span id="cb179-5"><a href="source-code.html#cb179-5" tabindex="-1"></a>  <span class="fu">return</span>(rest)</span>
<span id="cb179-6"><a href="source-code.html#cb179-6" tabindex="-1"></a>}</span>
<span id="cb179-7"><a href="source-code.html#cb179-7" tabindex="-1"></a></span>
<span id="cb179-8"><a href="source-code.html#cb179-8" tabindex="-1"></a></span>
<span id="cb179-9"><a href="source-code.html#cb179-9" tabindex="-1"></a>extract_square <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1, list_files2, square, x, y) {</span>
<span id="cb179-10"><a href="source-code.html#cb179-10" tabindex="-1"></a>  <span class="co"># extracts all cells that lie within a square of a pca plot defined by </span></span>
<span id="cb179-11"><a href="source-code.html#cb179-11" tabindex="-1"></a>  <span class="co"># user. For example square c(0.1, 0.0) will extract all cells with</span></span>
<span id="cb179-12"><a href="source-code.html#cb179-12" tabindex="-1"></a>  <span class="co"># pc1 lower or higher that 0.1 and pc2 lower or higher than 0.0. User has to rearange functon</span></span>
<span id="cb179-13"><a href="source-code.html#cb179-13" tabindex="-1"></a>  <span class="co"># if he wants x lower or higher than 0.1</span></span>
<span id="cb179-14"><a href="source-code.html#cb179-14" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb179-15"><a href="source-code.html#cb179-15" tabindex="-1"></a>  </span>
<span id="cb179-16"><a href="source-code.html#cb179-16" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(cells,</span>
<span id="cb179-17"><a href="source-code.html#cb179-17" tabindex="-1"></a>                   <span class="at">colorBy =</span> classes, </span>
<span id="cb179-18"><a href="source-code.html#cb179-18" tabindex="-1"></a>                   <span class="at">PC1=</span><span class="dv">1</span>, </span>
<span id="cb179-19"><a href="source-code.html#cb179-19" tabindex="-1"></a>                   <span class="at">PC2=</span><span class="dv">2</span>,</span>
<span id="cb179-20"><a href="source-code.html#cb179-20" tabindex="-1"></a>                   <span class="at">plot =</span> F)</span>
<span id="cb179-21"><a href="source-code.html#cb179-21" tabindex="-1"></a>  </span>
<span id="cb179-22"><a href="source-code.html#cb179-22" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>)</span>
<span id="cb179-23"><a href="source-code.html#cb179-23" tabindex="-1"></a>  </span>
<span id="cb179-24"><a href="source-code.html#cb179-24" tabindex="-1"></a>  </span>
<span id="cb179-25"><a href="source-code.html#cb179-25" tabindex="-1"></a>  <span class="co"># extracting dataframe with correct square</span></span>
<span id="cb179-26"><a href="source-code.html#cb179-26" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb179-27"><a href="source-code.html#cb179-27" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb179-28"><a href="source-code.html#cb179-28" tabindex="-1"></a>  }</span>
<span id="cb179-29"><a href="source-code.html#cb179-29" tabindex="-1"></a>  </span>
<span id="cb179-30"><a href="source-code.html#cb179-30" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb179-31"><a href="source-code.html#cb179-31" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb179-32"><a href="source-code.html#cb179-32" tabindex="-1"></a>  }</span>
<span id="cb179-33"><a href="source-code.html#cb179-33" tabindex="-1"></a>  </span>
<span id="cb179-34"><a href="source-code.html#cb179-34" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb179-35"><a href="source-code.html#cb179-35" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb179-36"><a href="source-code.html#cb179-36" tabindex="-1"></a>  }</span>
<span id="cb179-37"><a href="source-code.html#cb179-37" tabindex="-1"></a>  </span>
<span id="cb179-38"><a href="source-code.html#cb179-38" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb179-39"><a href="source-code.html#cb179-39" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb179-40"><a href="source-code.html#cb179-40" tabindex="-1"></a>  }</span>
<span id="cb179-41"><a href="source-code.html#cb179-41" tabindex="-1"></a>  </span>
<span id="cb179-42"><a href="source-code.html#cb179-42" tabindex="-1"></a>  <span class="co"># numbers is: </span></span>
<span id="cb179-43"><a href="source-code.html#cb179-43" tabindex="-1"></a>  cell_id <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)  </span>
<span id="cb179-44"><a href="source-code.html#cb179-44" tabindex="-1"></a>  </span>
<span id="cb179-45"><a href="source-code.html#cb179-45" tabindex="-1"></a>  <span class="fu">return</span>(cell_id)</span>
<span id="cb179-46"><a href="source-code.html#cb179-46" tabindex="-1"></a>  </span>
<span id="cb179-47"><a href="source-code.html#cb179-47" tabindex="-1"></a>}</span>
<span id="cb179-48"><a href="source-code.html#cb179-48" tabindex="-1"></a></span>
<span id="cb179-49"><a href="source-code.html#cb179-49" tabindex="-1"></a>l_g <span class="ot">&lt;-</span> <span class="cf">function</span>(x, kra_num, list_cells) {</span>
<span id="cb179-50"><a href="source-code.html#cb179-50" tabindex="-1"></a>  select <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="fu">paste0</span>(kra_num,<span class="st">&#39;_&#39;</span>,x,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>), list_cells)</span>
<span id="cb179-51"><a href="source-code.html#cb179-51" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(select)</span>
<span id="cb179-52"><a href="source-code.html#cb179-52" tabindex="-1"></a>  <span class="fu">return</span>(list_cells[i])</span>
<span id="cb179-53"><a href="source-code.html#cb179-53" tabindex="-1"></a>  </span>
<span id="cb179-54"><a href="source-code.html#cb179-54" tabindex="-1"></a>  </span>
<span id="cb179-55"><a href="source-code.html#cb179-55" tabindex="-1"></a>}</span>
<span id="cb179-56"><a href="source-code.html#cb179-56" tabindex="-1"></a></span>
<span id="cb179-57"><a href="source-code.html#cb179-57" tabindex="-1"></a></span>
<span id="cb179-58"><a href="source-code.html#cb179-58" tabindex="-1"></a>subclone_classes_for_genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(list_of_subclones, cline) {</span>
<span id="cb179-59"><a href="source-code.html#cb179-59" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb179-60"><a href="source-code.html#cb179-60" tabindex="-1"></a>  <span class="co">#&#39; list of subclones is the full list of rda files that have been assigned a subclone_classes_for_genomeheatmap</span></span>
<span id="cb179-61"><a href="source-code.html#cb179-61" tabindex="-1"></a>  <span class="co">#&#39; cline is the line (hub005, hub183 etc.)</span></span>
<span id="cb179-62"><a href="source-code.html#cb179-62" tabindex="-1"></a>  <span class="co">#&#39; rad is either prerad or rad</span></span>
<span id="cb179-63"><a href="source-code.html#cb179-63" tabindex="-1"></a>  <span class="co">#&#39; @return </span></span>
<span id="cb179-64"><a href="source-code.html#cb179-64" tabindex="-1"></a>  <span class="co">#&#39; returns a vector with that assigns a subclone</span></span>
<span id="cb179-65"><a href="source-code.html#cb179-65" tabindex="-1"></a>  </span>
<span id="cb179-66"><a href="source-code.html#cb179-66" tabindex="-1"></a>  extract_subclones <span class="ot">&lt;-</span> list_of_subclones[<span class="fu">grepl</span>(<span class="fu">names</span>(list_of_subclones), <span class="at">pattern =</span> <span class="fu">paste0</span>(cline))]</span>
<span id="cb179-67"><a href="source-code.html#cb179-67" tabindex="-1"></a>  </span>
<span id="cb179-68"><a href="source-code.html#cb179-68" tabindex="-1"></a>  size_subclones <span class="ot">&lt;-</span> <span class="fu">lapply</span>(extract_subclones, length)</span>
<span id="cb179-69"><a href="source-code.html#cb179-69" tabindex="-1"></a>  </span>
<span id="cb179-70"><a href="source-code.html#cb179-70" tabindex="-1"></a>  classes <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">names</span>(size_subclones), size_subclones)</span>
<span id="cb179-71"><a href="source-code.html#cb179-71" tabindex="-1"></a>  </span>
<span id="cb179-72"><a href="source-code.html#cb179-72" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">path =</span> <span class="fu">unlist</span>(extract_subclones),</span>
<span id="cb179-73"><a href="source-code.html#cb179-73" tabindex="-1"></a>                   <span class="at">class =</span> classes)</span>
<span id="cb179-74"><a href="source-code.html#cb179-74" tabindex="-1"></a>  </span>
<span id="cb179-75"><a href="source-code.html#cb179-75" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb179-76"><a href="source-code.html#cb179-76" tabindex="-1"></a>  </span>
<span id="cb179-77"><a href="source-code.html#cb179-77" tabindex="-1"></a>  </span>
<span id="cb179-78"><a href="source-code.html#cb179-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genomeheatmap-script-genomeheatmap-function" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> plot_genomeheatmap script: ‘genomeheatmap’ function<a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="source-code.html#cb180-1" tabindex="-1"></a>genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(selected.files, </span>
<span id="cb180-2"><a href="source-code.html#cb180-2" tabindex="-1"></a>                          path, </span>
<span id="cb180-3"><a href="source-code.html#cb180-3" tabindex="-1"></a>                          <span class="at">classes_daan =</span> <span class="cn">NULL</span>, </span>
<span id="cb180-4"><a href="source-code.html#cb180-4" tabindex="-1"></a>                          <span class="at">class.col =</span> <span class="cn">NULL</span>, </span>
<span id="cb180-5"><a href="source-code.html#cb180-5" tabindex="-1"></a>                          <span class="at">dendogram =</span> F,</span>
<span id="cb180-6"><a href="source-code.html#cb180-6" tabindex="-1"></a>                          <span class="at">daan_colours =</span> T,</span>
<span id="cb180-7"><a href="source-code.html#cb180-7" tabindex="-1"></a>                          name) {</span>
<span id="cb180-8"><a href="source-code.html#cb180-8" tabindex="-1"></a>  <span class="co"># This code was adapted from the AneuFinder package</span></span>
<span id="cb180-9"><a href="source-code.html#cb180-9" tabindex="-1"></a>  <span class="co"># plots and saves genome heatmap of selected.files</span></span>
<span id="cb180-10"><a href="source-code.html#cb180-10" tabindex="-1"></a>  <span class="co"># @param </span></span>
<span id="cb180-11"><a href="source-code.html#cb180-11" tabindex="-1"></a>  <span class="co"># selected.files: vector of location of selected files.</span></span>
<span id="cb180-12"><a href="source-code.html#cb180-12" tabindex="-1"></a>  <span class="co"># output path: path were plots should be saved. </span></span>
<span id="cb180-13"><a href="source-code.html#cb180-13" tabindex="-1"></a>  <span class="co"># name: how you want to name the plot</span></span>
<span id="cb180-14"><a href="source-code.html#cb180-14" tabindex="-1"></a>  </span>
<span id="cb180-15"><a href="source-code.html#cb180-15" tabindex="-1"></a>  <span class="co"># get platename</span></span>
<span id="cb180-16"><a href="source-code.html#cb180-16" tabindex="-1"></a>  platename <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(selected.files[<span class="dv">1</span>],  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb180-17"><a href="source-code.html#cb180-17" tabindex="-1"></a>  </span>
<span id="cb180-18"><a href="source-code.html#cb180-18" tabindex="-1"></a>  <span class="co"># AneuFinder functions: </span></span>
<span id="cb180-19"><a href="source-code.html#cb180-19" tabindex="-1"></a>  startTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb180-20"><a href="source-code.html#cb180-20" tabindex="-1"></a>    </span>
<span id="cb180-21"><a href="source-code.html#cb180-21" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">paste0</span>(..., <span class="at">collapse=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb180-22"><a href="source-code.html#cb180-22" tabindex="-1"></a>    <span class="fu">message</span>(x, <span class="at">appendLF=</span><span class="cn">FALSE</span>)</span>
<span id="cb180-23"><a href="source-code.html#cb180-23" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">proc.time</span>()</span>
<span id="cb180-24"><a href="source-code.html#cb180-24" tabindex="-1"></a>    <span class="fu">return</span>(ptm)</span>
<span id="cb180-25"><a href="source-code.html#cb180-25" tabindex="-1"></a>    </span>
<span id="cb180-26"><a href="source-code.html#cb180-26" tabindex="-1"></a>  }</span>
<span id="cb180-27"><a href="source-code.html#cb180-27" tabindex="-1"></a>  </span>
<span id="cb180-28"><a href="source-code.html#cb180-28" tabindex="-1"></a>  transCoord <span class="ot">&lt;-</span> <span class="cf">function</span> (gr) </span>
<span id="cb180-29"><a href="source-code.html#cb180-29" tabindex="-1"></a>  {</span>
<span id="cb180-30"><a href="source-code.html#cb180-30" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(gr)))</span>
<span id="cb180-31"><a href="source-code.html#cb180-31" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb180-32"><a href="source-code.html#cb180-32" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(gr)</span>
<span id="cb180-33"><a href="source-code.html#cb180-33" tabindex="-1"></a>    gr<span class="sc">$</span>start.genome <span class="ot">&lt;-</span> <span class="fu">start</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb180-34"><a href="source-code.html#cb180-34" tabindex="-1"></a>    gr<span class="sc">$</span>end.genome <span class="ot">&lt;-</span> <span class="fu">end</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb180-35"><a href="source-code.html#cb180-35" tabindex="-1"></a>    <span class="fu">return</span>(gr)</span>
<span id="cb180-36"><a href="source-code.html#cb180-36" tabindex="-1"></a>  }</span>
<span id="cb180-37"><a href="source-code.html#cb180-37" tabindex="-1"></a>  </span>
<span id="cb180-38"><a href="source-code.html#cb180-38" tabindex="-1"></a>  stopTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(ptm) {</span>
<span id="cb180-39"><a href="source-code.html#cb180-39" tabindex="-1"></a>    </span>
<span id="cb180-40"><a href="source-code.html#cb180-40" tabindex="-1"></a>    time <span class="ot">&lt;-</span> <span class="fu">proc.time</span>() <span class="sc">-</span> ptm</span>
<span id="cb180-41"><a href="source-code.html#cb180-41" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot; &quot;</span>, <span class="fu">round</span>(time[<span class="dv">3</span>],<span class="dv">2</span>), <span class="st">&quot;s&quot;</span>)</span>
<span id="cb180-42"><a href="source-code.html#cb180-42" tabindex="-1"></a>    </span>
<span id="cb180-43"><a href="source-code.html#cb180-43" tabindex="-1"></a>  }</span>
<span id="cb180-44"><a href="source-code.html#cb180-44" tabindex="-1"></a>  </span>
<span id="cb180-45"><a href="source-code.html#cb180-45" tabindex="-1"></a>  initializeStates <span class="ot">&lt;-</span> <span class="cf">function</span> (states) </span>
<span id="cb180-46"><a href="source-code.html#cb180-46" tabindex="-1"></a>  {</span>
<span id="cb180-47"><a href="source-code.html#cb180-47" tabindex="-1"></a>    somy.states <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;somy&quot;</span>, states, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb180-48"><a href="source-code.html#cb180-48" tabindex="-1"></a>    somy.numbers <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(somy.states, <span class="st">&quot;-somy&quot;</span>), </span>
<span id="cb180-49"><a href="source-code.html#cb180-49" tabindex="-1"></a>                                      <span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>))</span>
<span id="cb180-50"><a href="source-code.html#cb180-50" tabindex="-1"></a>    <span class="fu">names</span>(somy.numbers) <span class="ot">&lt;-</span> somy.states</span>
<span id="cb180-51"><a href="source-code.html#cb180-51" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">&quot;zero-inflation&quot;</span> <span class="sc">%in%</span> states) {</span>
<span id="cb180-52"><a href="source-code.html#cb180-52" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">0</span>, somy.numbers)</span>
<span id="cb180-53"><a href="source-code.html#cb180-53" tabindex="-1"></a>    }</span>
<span id="cb180-54"><a href="source-code.html#cb180-54" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb180-55"><a href="source-code.html#cb180-55" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> somy.numbers</span>
<span id="cb180-56"><a href="source-code.html#cb180-56" tabindex="-1"></a>    }</span>
<span id="cb180-57"><a href="source-code.html#cb180-57" tabindex="-1"></a>    levels.distributions <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;delta&quot;</span>, <span class="st">&quot;dgeom&quot;</span>, <span class="st">&quot;dnbinom&quot;</span>, <span class="st">&quot;dbinom&quot;</span>)</span>
<span id="cb180-58"><a href="source-code.html#cb180-58" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(states))</span>
<span id="cb180-59"><a href="source-code.html#cb180-59" tabindex="-1"></a>    <span class="fu">names</span>(distributions) <span class="ot">&lt;-</span> states</span>
<span id="cb180-60"><a href="source-code.html#cb180-60" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;zero-inflation&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;delta&quot;</span></span>
<span id="cb180-61"><a href="source-code.html#cb180-61" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;0-somy&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;dgeom&quot;</span></span>
<span id="cb180-62"><a href="source-code.html#cb180-62" tabindex="-1"></a>    distributions[(states <span class="sc">!=</span> <span class="st">&quot;zero-inflation&quot;</span>) <span class="sc">&amp;</span> (states <span class="sc">!=</span> <span class="st">&quot;0-somy&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;dnbinom&quot;</span></span>
<span id="cb180-63"><a href="source-code.html#cb180-63" tabindex="-1"></a>    states <span class="ot">&lt;-</span> <span class="fu">factor</span>(states, <span class="at">levels =</span> states)</span>
<span id="cb180-64"><a href="source-code.html#cb180-64" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">factor</span>(distributions, <span class="at">levels =</span> levels.distributions)</span>
<span id="cb180-65"><a href="source-code.html#cb180-65" tabindex="-1"></a>    l <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">states =</span> states, <span class="at">distributions =</span> distributions, </span>
<span id="cb180-66"><a href="source-code.html#cb180-66" tabindex="-1"></a>              <span class="at">multiplicity =</span> multiplicity)</span>
<span id="cb180-67"><a href="source-code.html#cb180-67" tabindex="-1"></a>    <span class="fu">return</span>(l)</span>
<span id="cb180-68"><a href="source-code.html#cb180-68" tabindex="-1"></a>  }</span>
<span id="cb180-69"><a href="source-code.html#cb180-69" tabindex="-1"></a>  </span>
<span id="cb180-70"><a href="source-code.html#cb180-70" tabindex="-1"></a>  <span class="co">#  colours</span></span>
<span id="cb180-71"><a href="source-code.html#cb180-71" tabindex="-1"></a>  <span class="cf">if</span> (daan_colours) {</span>
<span id="cb180-72"><a href="source-code.html#cb180-72" tabindex="-1"></a>    stateColors <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">states =</span> <span class="fu">c</span>(<span class="st">&quot;zero-inflation&quot;</span>, <span class="fu">paste0</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>, <span class="st">&quot;-somy&quot;</span>),</span>
<span id="cb180-73"><a href="source-code.html#cb180-73" tabindex="-1"></a>                                       <span class="st">&quot;total&quot;</span>)) {</span>
<span id="cb180-74"><a href="source-code.html#cb180-74" tabindex="-1"></a>      </span>
<span id="cb180-75"><a href="source-code.html#cb180-75" tabindex="-1"></a>      state.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>, <span class="st">`</span><span class="at">0-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>,</span>
<span id="cb180-76"><a href="source-code.html#cb180-76" tabindex="-1"></a>                        <span class="st">`</span><span class="at">1-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3787BA&quot;</span>, <span class="st">`</span><span class="at">2-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#95B8C5&quot;</span>,</span>
<span id="cb180-77"><a href="source-code.html#cb180-77" tabindex="-1"></a>                        <span class="st">`</span><span class="at">3-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#F0ECEB&quot;</span>, <span class="st">`</span><span class="at">4-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#D7A290&quot;</span>, <span class="st">`</span><span class="at">5-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#BF583B&quot;</span>,</span>
<span id="cb180-78"><a href="source-code.html#cb180-78" tabindex="-1"></a>                        <span class="st">`</span><span class="at">6-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#8D1128&quot;</span>, <span class="st">`</span><span class="at">7-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3C0912&quot;</span>, <span class="st">`</span><span class="at">8-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb180-79"><a href="source-code.html#cb180-79" tabindex="-1"></a>                        <span class="at">total =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb180-80"><a href="source-code.html#cb180-80" tabindex="-1"></a>      </span>
<span id="cb180-81"><a href="source-code.html#cb180-81" tabindex="-1"></a>      states.with.color <span class="ot">&lt;-</span> <span class="fu">intersect</span>(states, <span class="fu">names</span>(state.colors))</span>
<span id="cb180-82"><a href="source-code.html#cb180-82" tabindex="-1"></a>      cols <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;black&quot;</span>, <span class="fu">length</span>(states))</span>
<span id="cb180-83"><a href="source-code.html#cb180-83" tabindex="-1"></a>      <span class="fu">names</span>(cols) <span class="ot">&lt;-</span> states</span>
<span id="cb180-84"><a href="source-code.html#cb180-84" tabindex="-1"></a>      cols[states.with.color] <span class="ot">&lt;-</span> state.colors[states.with.color]</span>
<span id="cb180-85"><a href="source-code.html#cb180-85" tabindex="-1"></a>      <span class="fu">return</span>(cols)</span>
<span id="cb180-86"><a href="source-code.html#cb180-86" tabindex="-1"></a>    }</span>
<span id="cb180-87"><a href="source-code.html#cb180-87" tabindex="-1"></a>    </span>
<span id="cb180-88"><a href="source-code.html#cb180-88" tabindex="-1"></a>  }</span>
<span id="cb180-89"><a href="source-code.html#cb180-89" tabindex="-1"></a></span>
<span id="cb180-90"><a href="source-code.html#cb180-90" tabindex="-1"></a>  </span>
<span id="cb180-91"><a href="source-code.html#cb180-91" tabindex="-1"></a>  </span>
<span id="cb180-92"><a href="source-code.html#cb180-92" tabindex="-1"></a>  <span class="co"># heatmapgenomewide adapted form AneuFinder package</span></span>
<span id="cb180-93"><a href="source-code.html#cb180-93" tabindex="-1"></a>  heatmapGenomewide_daan <span class="ot">&lt;-</span> <span class="cf">function</span> (hmms, <span class="at">ylabels =</span> <span class="cn">NULL</span>, classes, <span class="at">reorder.by.class =</span> <span class="cn">TRUE</span>, </span>
<span id="cb180-94"><a href="source-code.html#cb180-94" tabindex="-1"></a>                                      classes.color, <span class="at">file =</span> <span class="cn">NULL</span>, <span class="at">cluster =</span> <span class="cn">TRUE</span>, <span class="at">plot.breakpoints =</span> <span class="cn">FALSE</span>, </span>
<span id="cb180-95"><a href="source-code.html#cb180-95" tabindex="-1"></a>                                      <span class="at">hotspots =</span> <span class="cn">NULL</span>, <span class="at">exclude.regions =</span> <span class="cn">NULL</span>) </span>
<span id="cb180-96"><a href="source-code.html#cb180-96" tabindex="-1"></a>  {</span>
<span id="cb180-97"><a href="source-code.html#cb180-97" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb180-98"><a href="source-code.html#cb180-98" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(ylabels) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb180-99"><a href="source-code.html#cb180-99" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(ylabels) must equal length(hmms)&quot;</span>)</span>
<span id="cb180-100"><a href="source-code.html#cb180-100" tabindex="-1"></a>      }</span>
<span id="cb180-101"><a href="source-code.html#cb180-101" tabindex="-1"></a>    }</span>
<span id="cb180-102"><a href="source-code.html#cb180-102" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb180-103"><a href="source-code.html#cb180-103" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(classes) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb180-104"><a href="source-code.html#cb180-104" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(classes) must equal length(hmms)&quot;</span>)</span>
<span id="cb180-105"><a href="source-code.html#cb180-105" tabindex="-1"></a>      }</span>
<span id="cb180-106"><a href="source-code.html#cb180-106" tabindex="-1"></a>    }</span>
<span id="cb180-107"><a href="source-code.html#cb180-107" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(classes.color) <span class="sc">!=</span> <span class="fu">length</span>(<span class="fu">unique</span>(classes))) {</span>
<span id="cb180-108"><a href="source-code.html#cb180-108" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;&#39;classes.color&#39; must have the same length as unique(classes)&quot;</span>)</span>
<span id="cb180-109"><a href="source-code.html#cb180-109" tabindex="-1"></a>    }</span>
<span id="cb180-110"><a href="source-code.html#cb180-110" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">names</span>(classes.color))) {</span>
<span id="cb180-111"><a href="source-code.html#cb180-111" tabindex="-1"></a>      <span class="fu">names</span>(classes.color) <span class="ot">&lt;-</span> <span class="fu">unique</span>(classes)</span>
<span id="cb180-112"><a href="source-code.html#cb180-112" tabindex="-1"></a>    }</span>
<span id="cb180-113"><a href="source-code.html#cb180-113" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">setequal</span>(<span class="fu">names</span>(classes.color), <span class="fu">unique</span>(classes))) {</span>
<span id="cb180-114"><a href="source-code.html#cb180-114" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;The names of &#39;classes.color&#39; must be equal to the unique elements in &#39;classes&#39;&quot;</span>)</span>
<span id="cb180-115"><a href="source-code.html#cb180-115" tabindex="-1"></a>    }</span>
<span id="cb180-116"><a href="source-code.html#cb180-116" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(hmms) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> cluster <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb180-117"><a href="source-code.html#cb180-117" tabindex="-1"></a>      cluster <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb180-118"><a href="source-code.html#cb180-118" tabindex="-1"></a>      <span class="fu">warning</span>(<span class="st">&quot;Cannot do clustering because only one object was given.&quot;</span>)</span>
<span id="cb180-119"><a href="source-code.html#cb180-119" tabindex="-1"></a>    }</span>
<span id="cb180-120"><a href="source-code.html#cb180-120" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> <span class="fu">loadFromFiles</span>(hmms, <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb180-121"><a href="source-code.html#cb180-121" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">sapply</span>(hmms, <span class="st">&quot;[[&quot;</span>, <span class="st">&quot;ID&quot;</span>))</span>
<span id="cb180-122"><a href="source-code.html#cb180-122" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb180-123"><a href="source-code.html#cb180-123" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb180-124"><a href="source-code.html#cb180-124" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb180-125"><a href="source-code.html#cb180-125" tabindex="-1"></a>    }</span>
<span id="cb180-126"><a href="source-code.html#cb180-126" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb180-127"><a href="source-code.html#cb180-127" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ylabels)</span>
<span id="cb180-128"><a href="source-code.html#cb180-128" tabindex="-1"></a>    }</span>
<span id="cb180-129"><a href="source-code.html#cb180-129" tabindex="-1"></a>    class.data<span class="sc">$</span>class <span class="ot">&lt;-</span> classes</span>
<span id="cb180-130"><a href="source-code.html#cb180-130" tabindex="-1"></a>    mapping <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ylabel</span>
<span id="cb180-131"><a href="source-code.html#cb180-131" tabindex="-1"></a>    <span class="fu">names</span>(mapping) <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ID</span>
<span id="cb180-132"><a href="source-code.html#cb180-132" tabindex="-1"></a>    <span class="cf">if</span> (reorder.by.class) {</span>
<span id="cb180-133"><a href="source-code.html#cb180-133" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">classes =</span> classes, </span>
<span id="cb180-134"><a href="source-code.html#cb180-134" tabindex="-1"></a>                        <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb180-135"><a href="source-code.html#cb180-135" tabindex="-1"></a>    }</span>
<span id="cb180-136"><a href="source-code.html#cb180-136" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb180-137"><a href="source-code.html#cb180-137" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb180-138"><a href="source-code.html#cb180-138" tabindex="-1"></a>    }</span>
<span id="cb180-139"><a href="source-code.html#cb180-139" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> hmms[cl<span class="sc">$</span>IDorder]</span>
<span id="cb180-140"><a href="source-code.html#cb180-140" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> class.data[cl<span class="sc">$</span>IDorder, ]</span>
<span id="cb180-141"><a href="source-code.html#cb180-141" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb180-142"><a href="source-code.html#cb180-142" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb180-143"><a href="source-code.html#cb180-143" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb180-144"><a href="source-code.html#cb180-144" tabindex="-1"></a>      hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb180-145"><a href="source-code.html#cb180-145" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>segments)) {</span>
<span id="cb180-146"><a href="source-code.html#cb180-146" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb180-147"><a href="source-code.html#cb180-147" tabindex="-1"></a>      }</span>
<span id="cb180-148"><a href="source-code.html#cb180-148" tabindex="-1"></a>      <span class="cf">else</span> {</span>
<span id="cb180-149"><a href="source-code.html#cb180-149" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>segments</span>
<span id="cb180-150"><a href="source-code.html#cb180-150" tabindex="-1"></a>      }</span>
<span id="cb180-151"><a href="source-code.html#cb180-151" tabindex="-1"></a>    }</span>
<span id="cb180-152"><a href="source-code.html#cb180-152" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb180-153"><a href="source-code.html#cb180-153" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb180-154"><a href="source-code.html#cb180-154" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb180-155"><a href="source-code.html#cb180-155" tabindex="-1"></a>        hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb180-156"><a href="source-code.html#cb180-156" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>breakpoints)) {</span>
<span id="cb180-157"><a href="source-code.html#cb180-157" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb180-158"><a href="source-code.html#cb180-158" tabindex="-1"></a>        }</span>
<span id="cb180-159"><a href="source-code.html#cb180-159" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb180-160"><a href="source-code.html#cb180-160" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>breakpoints</span>
<span id="cb180-161"><a href="source-code.html#cb180-161" tabindex="-1"></a>        }</span>
<span id="cb180-162"><a href="source-code.html#cb180-162" tabindex="-1"></a>      }</span>
<span id="cb180-163"><a href="source-code.html#cb180-163" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(breakpoints) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb180-164"><a href="source-code.html#cb180-164" tabindex="-1"></a>        plot.breakpoints <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb180-165"><a href="source-code.html#cb180-165" tabindex="-1"></a>      }</span>
<span id="cb180-166"><a href="source-code.html#cb180-166" tabindex="-1"></a>    }</span>
<span id="cb180-167"><a href="source-code.html#cb180-167" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Transforming coordinates ...&quot;</span>)</span>
<span id="cb180-168"><a href="source-code.html#cb180-168" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(segments.list, transCoord)</span>
<span id="cb180-169"><a href="source-code.html#cb180-169" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb180-170"><a href="source-code.html#cb180-170" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(breakpoints, transCoord)</span>
<span id="cb180-171"><a href="source-code.html#cb180-171" tabindex="-1"></a>    }</span>
<span id="cb180-172"><a href="source-code.html#cb180-172" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb180-173"><a href="source-code.html#cb180-173" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Making the plot ...&quot;</span>)</span>
<span id="cb180-174"><a href="source-code.html#cb180-174" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb180-175"><a href="source-code.html#cb180-175" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(segments.list)) {</span>
<span id="cb180-176"><a href="source-code.html#cb180-176" tabindex="-1"></a>      df[[<span class="fu">length</span>(df) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> segments.list[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb180-177"><a href="source-code.html#cb180-177" tabindex="-1"></a>                                         <span class="at">end =</span> segments.list[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(segments.list[[i1]]), </span>
<span id="cb180-178"><a href="source-code.html#cb180-178" tabindex="-1"></a>                                         <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">state =</span> segments.list[[i1]]<span class="sc">$</span>state)</span>
<span id="cb180-179"><a href="source-code.html#cb180-179" tabindex="-1"></a>    }</span>
<span id="cb180-180"><a href="source-code.html#cb180-180" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df)</span>
<span id="cb180-181"><a href="source-code.html#cb180-181" tabindex="-1"></a>    df<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb180-182"><a href="source-code.html#cb180-182" tabindex="-1"></a>    df<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df<span class="sc">$</span>ID)]</span>
<span id="cb180-183"><a href="source-code.html#cb180-183" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb180-184"><a href="source-code.html#cb180-184" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb180-185"><a href="source-code.html#cb180-185" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(breakpoints)) {</span>
<span id="cb180-186"><a href="source-code.html#cb180-186" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(breakpoints[[i1]]) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb180-187"><a href="source-code.html#cb180-187" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> breakpoints[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb180-188"><a href="source-code.html#cb180-188" tabindex="-1"></a>                                                                     <span class="at">end =</span> breakpoints[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(breakpoints[[i1]]), </span>
<span id="cb180-189"><a href="source-code.html#cb180-189" tabindex="-1"></a>                                                                     <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">mid =</span> (breakpoints[[i1]]<span class="sc">$</span>start.genome <span class="sc">+</span> </span>
<span id="cb180-190"><a href="source-code.html#cb180-190" tabindex="-1"></a>                                                                                                             breakpoints[[i1]]<span class="sc">$</span>end.genome)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb180-191"><a href="source-code.html#cb180-191" tabindex="-1"></a>        }</span>
<span id="cb180-192"><a href="source-code.html#cb180-192" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb180-193"><a href="source-code.html#cb180-193" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> <span class="fu">numeric</span>(), </span>
<span id="cb180-194"><a href="source-code.html#cb180-194" tabindex="-1"></a>                                                                     <span class="at">end =</span> <span class="fu">numeric</span>(), <span class="at">seqnames =</span> <span class="fu">character</span>(), <span class="at">ID =</span> <span class="fu">character</span>(), </span>
<span id="cb180-195"><a href="source-code.html#cb180-195" tabindex="-1"></a>                                                                     <span class="at">mid =</span> <span class="fu">numeric</span>())</span>
<span id="cb180-196"><a href="source-code.html#cb180-196" tabindex="-1"></a>        }</span>
<span id="cb180-197"><a href="source-code.html#cb180-197" tabindex="-1"></a>      }</span>
<span id="cb180-198"><a href="source-code.html#cb180-198" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df.breakpoints)</span>
<span id="cb180-199"><a href="source-code.html#cb180-199" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df.breakpoints<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb180-200"><a href="source-code.html#cb180-200" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df.breakpoints<span class="sc">$</span>ID)]</span>
<span id="cb180-201"><a href="source-code.html#cb180-201" tabindex="-1"></a>    }</span>
<span id="cb180-202"><a href="source-code.html#cb180-202" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]])))</span>
<span id="cb180-203"><a href="source-code.html#cb180-203" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb180-204"><a href="source-code.html#cb180-204" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb180-205"><a href="source-code.html#cb180-205" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb180-206"><a href="source-code.html#cb180-206" tabindex="-1"></a>    label.pos <span class="ot">&lt;-</span> <span class="fu">round</span>(cum.seqlengths<span class="fl">.0</span> <span class="sc">+</span> <span class="fl">0.5</span> <span class="sc">*</span> <span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]]))</span>
<span id="cb180-207"><a href="source-code.html#cb180-207" tabindex="-1"></a>    df.chroms <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths), <span class="at">x =</span> <span class="dv">1</span>, </span>
<span id="cb180-208"><a href="source-code.html#cb180-208" tabindex="-1"></a>                            <span class="at">xend =</span> <span class="fu">length</span>(segments.list))</span>
<span id="cb180-209"><a href="source-code.html#cb180-209" tabindex="-1"></a>    pltlist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb180-210"><a href="source-code.html#cb180-210" tabindex="-1"></a>    widths <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb180-211"><a href="source-code.html#cb180-211" tabindex="-1"></a>    df<span class="sc">$</span>state <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>state, <span class="at">levels =</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">initializeStates</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))<span class="sc">$</span>multiplicity)))</span>
<span id="cb180-212"><a href="source-code.html#cb180-212" tabindex="-1"></a>    df<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>ID)</span>
<span id="cb180-213"><a href="source-code.html#cb180-213" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, </span>
<span id="cb180-214"><a href="source-code.html#cb180-214" tabindex="-1"></a>                                                    <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;state&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> label.pos, </span>
<span id="cb180-215"><a href="source-code.html#cb180-215" tabindex="-1"></a>                                                                                                                          <span class="at">labels =</span> <span class="fu">names</span>(label.pos))</span>
<span id="cb180-216"><a href="source-code.html#cb180-216" tabindex="-1"></a>    </span>
<span id="cb180-217"><a href="source-code.html#cb180-217" tabindex="-1"></a></span>
<span id="cb180-218"><a href="source-code.html#cb180-218" tabindex="-1"></a>    <span class="co"># ggplt &lt;- ggplt + scale_x_continuous(name = &quot;&quot;,</span></span>
<span id="cb180-219"><a href="source-code.html#cb180-219" tabindex="-1"></a>    <span class="co">#                      breaks = 1:length(unique(df$ylabel)),</span></span>
<span id="cb180-220"><a href="source-code.html#cb180-220" tabindex="-1"></a>    <span class="co">#                      labels = unique(df$ylabel))</span></span>
<span id="cb180-221"><a href="source-code.html#cb180-221" tabindex="-1"></a>    </span>
<span id="cb180-222"><a href="source-code.html#cb180-222" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">stateColors</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))) <span class="co"># adding custom colours</span></span>
<span id="cb180-223"><a href="source-code.html#cb180-223" tabindex="-1"></a>    <span class="co"># adjusintg x axis</span></span>
<span id="cb180-224"><a href="source-code.html#cb180-224" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-225"><a href="source-code.html#cb180-225" tabindex="-1"></a>                           <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-226"><a href="source-code.html#cb180-226" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb180-227"><a href="source-code.html#cb180-227" tabindex="-1"></a>    </span>
<span id="cb180-228"><a href="source-code.html#cb180-228" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>,</span>
<span id="cb180-229"><a href="source-code.html#cb180-229" tabindex="-1"></a>                                             <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;y&quot;</span>), <span class="at">data =</span> df.chroms, <span class="at">col =</span> <span class="st">&quot;grey13&quot;</span>)</span>
<span id="cb180-230"><a href="source-code.html#cb180-230" tabindex="-1"></a>    </span>
<span id="cb180-231"><a href="source-code.html#cb180-231" tabindex="-1"></a>    </span>
<span id="cb180-232"><a href="source-code.html#cb180-232" tabindex="-1"></a>    </span>
<span id="cb180-233"><a href="source-code.html#cb180-233" tabindex="-1"></a>    <span class="co"># adjusting y axis</span></span>
<span id="cb180-234"><a href="source-code.html#cb180-234" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-235"><a href="source-code.html#cb180-235" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb180-236"><a href="source-code.html#cb180-236" tabindex="-1"></a>    </span>
<span id="cb180-237"><a href="source-code.html#cb180-237" tabindex="-1"></a>    <span class="co"># removing legend</span></span>
<span id="cb180-238"><a href="source-code.html#cb180-238" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb180-239"><a href="source-code.html#cb180-239" tabindex="-1"></a>    </span>
<span id="cb180-240"><a href="source-code.html#cb180-240" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb180-241"><a href="source-code.html#cb180-241" tabindex="-1"></a>    </span>
<span id="cb180-242"><a href="source-code.html#cb180-242" tabindex="-1"></a>    <span class="co"># removing all axis names</span></span>
<span id="cb180-243"><a href="source-code.html#cb180-243" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb180-244"><a href="source-code.html#cb180-244" tabindex="-1"></a>    </span>
<span id="cb180-245"><a href="source-code.html#cb180-245" tabindex="-1"></a>    <span class="co"># add numeber of cells sequenced text</span></span>
<span id="cb180-246"><a href="source-code.html#cb180-246" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, </span>
<span id="cb180-247"><a href="source-code.html#cb180-247" tabindex="-1"></a>                              <span class="at">label =</span> <span class="fu">paste</span>(<span class="fu">length</span>(hmms), <span class="st">&#39;cells&#39;</span>), </span>
<span id="cb180-248"><a href="source-code.html#cb180-248" tabindex="-1"></a>                              <span class="at">x =</span> <span class="fu">length</span>(hmms)<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb180-249"><a href="source-code.html#cb180-249" tabindex="-1"></a>                              <span class="at">y =</span> <span class="dv">3100000000</span>,</span>
<span id="cb180-250"><a href="source-code.html#cb180-250" tabindex="-1"></a>                              <span class="at">angle =</span> <span class="dv">270</span>,</span>
<span id="cb180-251"><a href="source-code.html#cb180-251" tabindex="-1"></a>                              <span class="at">size =</span> <span class="dv">50</span>)</span>
<span id="cb180-252"><a href="source-code.html#cb180-252" tabindex="-1"></a>    </span>
<span id="cb180-253"><a href="source-code.html#cb180-253" tabindex="-1"></a>    <span class="co"># decreasing plot margin</span></span>
<span id="cb180-254"><a href="source-code.html#cb180-254" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb180-255"><a href="source-code.html#cb180-255" tabindex="-1"></a>    </span>
<span id="cb180-256"><a href="source-code.html#cb180-256" tabindex="-1"></a>    </span>
<span id="cb180-257"><a href="source-code.html#cb180-257" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb180-258"><a href="source-code.html#cb180-258" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df.breakpoints<span class="sc">$</span>ID)</span>
<span id="cb180-259"><a href="source-code.html#cb180-259" tabindex="-1"></a>      ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb180-260"><a href="source-code.html#cb180-260" tabindex="-1"></a>                                      <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>), </span>
<span id="cb180-261"><a href="source-code.html#cb180-261" tabindex="-1"></a>                                      <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb180-262"><a href="source-code.html#cb180-262" tabindex="-1"></a>                                                                        <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">y =</span> <span class="st">&quot;mid&quot;</span>))</span>
<span id="cb180-263"><a href="source-code.html#cb180-263" tabindex="-1"></a>    }</span>
<span id="cb180-264"><a href="source-code.html#cb180-264" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(hotspots)) {</span>
<span id="cb180-265"><a href="source-code.html#cb180-265" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(hotspots) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb180-266"><a href="source-code.html#cb180-266" tabindex="-1"></a>        df.hot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">transCoord</span>(hotspots))</span>
<span id="cb180-267"><a href="source-code.html#cb180-267" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmin <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb180-268"><a href="source-code.html#cb180-268" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmax <span class="ot">&lt;-</span> <span class="fu">length</span>(class.data<span class="sc">$</span>ID) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb180-269"><a href="source-code.html#cb180-269" tabindex="-1"></a>        ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_rect</span>(<span class="at">data =</span> df.hot, <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">xmin =</span> <span class="st">&quot;xmin&quot;</span>, </span>
<span id="cb180-270"><a href="source-code.html#cb180-270" tabindex="-1"></a>                                                                       <span class="at">xmax =</span> <span class="st">&quot;xmax&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start.genome&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end.genome&quot;</span>, </span>
<span id="cb180-271"><a href="source-code.html#cb180-271" tabindex="-1"></a>                                                                       <span class="at">alpha =</span> <span class="st">&quot;num.events&quot;</span>), <span class="at">fill =</span> <span class="st">&quot;hotpink4&quot;</span>) <span class="sc">+</span> <span class="fu">scale_alpha_continuous</span>(<span class="at">name =</span> <span class="st">&quot;breakpoints&quot;</span>, </span>
<span id="cb180-272"><a href="source-code.html#cb180-272" tabindex="-1"></a>                                                                                                                                          <span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.8</span>))</span>
<span id="cb180-273"><a href="source-code.html#cb180-273" tabindex="-1"></a>      }</span>
<span id="cb180-274"><a href="source-code.html#cb180-274" tabindex="-1"></a>    }</span>
<span id="cb180-275"><a href="source-code.html#cb180-275" tabindex="-1"></a>    </span>
<span id="cb180-276"><a href="source-code.html#cb180-276" tabindex="-1"></a>    </span>
<span id="cb180-277"><a href="source-code.html#cb180-277" tabindex="-1"></a>    width.heatmap <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(hmms[[<span class="dv">1</span>]]<span class="sc">$</span>bins)))<span class="sc">/</span><span class="fl">3e+09</span> <span class="sc">*</span> </span>
<span id="cb180-278"><a href="source-code.html#cb180-278" tabindex="-1"></a>      <span class="dv">150</span></span>
<span id="cb180-279"><a href="source-code.html#cb180-279" tabindex="-1"></a>    height <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">length</span>(hmms) <span class="sc">*</span> <span class="fl">0.5</span>, <span class="dv">2</span>)</span>
<span id="cb180-280"><a href="source-code.html#cb180-280" tabindex="-1"></a>    pltlist[[<span class="st">&quot;heatmap&quot;</span>]] <span class="ot">&lt;-</span> ggplt</span>
<span id="cb180-281"><a href="source-code.html#cb180-281" tabindex="-1"></a>    widths[<span class="st">&quot;heatmap&quot;</span>] <span class="ot">&lt;-</span> width.heatmap</span>
<span id="cb180-282"><a href="source-code.html#cb180-282" tabindex="-1"></a>    <span class="co"># adding class colors</span></span>
<span id="cb180-283"><a href="source-code.html#cb180-283" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb180-284"><a href="source-code.html#cb180-284" tabindex="-1"></a>      width.classes <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb180-285"><a href="source-code.html#cb180-285" tabindex="-1"></a>      class.data<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb180-286"><a href="source-code.html#cb180-286" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(class.data) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="dv">0</span>, </span>
<span id="cb180-287"><a href="source-code.html#cb180-287" tabindex="-1"></a>                                                                <span class="at">ymax =</span> <span class="dv">1</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;class&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">col =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb180-288"><a href="source-code.html#cb180-288" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb180-289"><a href="source-code.html#cb180-289" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-290"><a href="source-code.html#cb180-290" tabindex="-1"></a>                                 <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-291"><a href="source-code.html#cb180-291" tabindex="-1"></a>                                 <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb180-292"><a href="source-code.html#cb180-292" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb180-293"><a href="source-code.html#cb180-293" tabindex="-1"></a>      </span>
<span id="cb180-294"><a href="source-code.html#cb180-294" tabindex="-1"></a>      <span class="co"># decreasing plot margin</span></span>
<span id="cb180-295"><a href="source-code.html#cb180-295" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb180-296"><a href="source-code.html#cb180-296" tabindex="-1"></a>      </span>
<span id="cb180-297"><a href="source-code.html#cb180-297" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes.color)) {</span>
<span id="cb180-298"><a href="source-code.html#cb180-298" tabindex="-1"></a>        ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">breaks =</span> <span class="fu">names</span>(classes.color), </span>
<span id="cb180-299"><a href="source-code.html#cb180-299" tabindex="-1"></a>                                                <span class="at">values =</span> classes.color)</span>
<span id="cb180-300"><a href="source-code.html#cb180-300" tabindex="-1"></a>      }</span>
<span id="cb180-301"><a href="source-code.html#cb180-301" tabindex="-1"></a>      pltlist[[<span class="st">&quot;classbar&quot;</span>]] <span class="ot">&lt;-</span> ggclass</span>
<span id="cb180-302"><a href="source-code.html#cb180-302" tabindex="-1"></a>      widths[<span class="st">&quot;classbar&quot;</span>] <span class="ot">&lt;-</span> width.classes</span>
<span id="cb180-303"><a href="source-code.html#cb180-303" tabindex="-1"></a>    }</span>
<span id="cb180-304"><a href="source-code.html#cb180-304" tabindex="-1"></a>    <span class="co"># adding dendogram</span></span>
<span id="cb180-305"><a href="source-code.html#cb180-305" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(cl<span class="sc">$</span>hclust) <span class="sc">&amp;</span> dendogram) {</span>
<span id="cb180-306"><a href="source-code.html#cb180-306" tabindex="-1"></a>      dhc <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">as.dendrogram</span>(cl<span class="sc">$</span>hclust)</span>
<span id="cb180-307"><a href="source-code.html#cb180-307" tabindex="-1"></a>      ddata <span class="ot">&lt;-</span> ggdendro<span class="sc">::</span><span class="fu">dendro_data</span>(dhc, <span class="at">type =</span> <span class="st">&quot;rectangle&quot;</span>)</span>
<span id="cb180-308"><a href="source-code.html#cb180-308" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ddata<span class="sc">$</span>segments) <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, </span>
<span id="cb180-309"><a href="source-code.html#cb180-309" tabindex="-1"></a>                                                                 <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>, <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;yend&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_y_reverse</span>()</span>
<span id="cb180-310"><a href="source-code.html#cb180-310" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb180-311"><a href="source-code.html#cb180-311" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-312"><a href="source-code.html#cb180-312" tabindex="-1"></a>                               <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb180-313"><a href="source-code.html#cb180-313" tabindex="-1"></a>                               <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb180-314"><a href="source-code.html#cb180-314" tabindex="-1"></a>      width.dendro <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb180-315"><a href="source-code.html#cb180-315" tabindex="-1"></a>      pltlist[[<span class="st">&quot;dendro&quot;</span>]] <span class="ot">&lt;-</span> ggdndr</span>
<span id="cb180-316"><a href="source-code.html#cb180-316" tabindex="-1"></a>      widths[<span class="st">&quot;dendro&quot;</span>] <span class="ot">&lt;-</span> width.dendro</span>
<span id="cb180-317"><a href="source-code.html#cb180-317" tabindex="-1"></a>    }</span>
<span id="cb180-318"><a href="source-code.html#cb180-318" tabindex="-1"></a>    <span class="co"># alligning ggpllt with dendogram and classes</span></span>
<span id="cb180-319"><a href="source-code.html#cb180-319" tabindex="-1"></a>    cowplt <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">rev</span>(pltlist), <span class="at">align =</span> <span class="st">&quot;h&quot;</span>, </span>
<span id="cb180-320"><a href="source-code.html#cb180-320" tabindex="-1"></a>                                 <span class="at">ncol =</span> <span class="fu">length</span>(pltlist), <span class="at">rel_widths =</span> <span class="fu">rev</span>(widths))</span>
<span id="cb180-321"><a href="source-code.html#cb180-321" tabindex="-1"></a>    </span>
<span id="cb180-322"><a href="source-code.html#cb180-322" tabindex="-1"></a>    </span>
<span id="cb180-323"><a href="source-code.html#cb180-323" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb180-324"><a href="source-code.html#cb180-324" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(file)) {</span>
<span id="cb180-325"><a href="source-code.html#cb180-325" tabindex="-1"></a>      ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Plotting to file &quot;</span>, file, <span class="st">&quot; ...&quot;</span>)</span>
<span id="cb180-326"><a href="source-code.html#cb180-326" tabindex="-1"></a>      <span class="fu">ggsave</span>(file, cowplt, <span class="at">width =</span> <span class="fu">sum</span>(widths), <span class="at">height =</span> height, </span>
<span id="cb180-327"><a href="source-code.html#cb180-327" tabindex="-1"></a>             <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>)</span>
<span id="cb180-328"><a href="source-code.html#cb180-328" tabindex="-1"></a>      <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb180-329"><a href="source-code.html#cb180-329" tabindex="-1"></a>    }</span>
<span id="cb180-330"><a href="source-code.html#cb180-330" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb180-331"><a href="source-code.html#cb180-331" tabindex="-1"></a>      <span class="fu">return</span>(cowplt)</span>
<span id="cb180-332"><a href="source-code.html#cb180-332" tabindex="-1"></a>    }</span>
<span id="cb180-333"><a href="source-code.html#cb180-333" tabindex="-1"></a>  }</span>
<span id="cb180-334"><a href="source-code.html#cb180-334" tabindex="-1"></a>  </span>
<span id="cb180-335"><a href="source-code.html#cb180-335" tabindex="-1"></a>  <span class="co">#make heatmap and safe</span></span>
<span id="cb180-336"><a href="source-code.html#cb180-336" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(<span class="fu">heatmapGenomewide_daan</span>(selected.files, <span class="at">classes =</span> classes_daan, <span class="at">classes.color =</span> class.col,</span>
<span id="cb180-337"><a href="source-code.html#cb180-337" tabindex="-1"></a>                         <span class="at">file =</span> <span class="fu">paste0</span>(path, <span class="st">&#39;/&#39;</span>, name, <span class="st">&#39;_&#39;</span>, platename, <span class="st">&#39;.pdf&#39;</span>)))</span>
<span id="cb180-338"><a href="source-code.html#cb180-338" tabindex="-1"></a>  </span>
<span id="cb180-339"><a href="source-code.html#cb180-339" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">print</span>(<span class="fu">paste0</span>(name, <span class="st">&#39; done.&#39;</span>)))</span>
<span id="cb180-340"><a href="source-code.html#cb180-340" tabindex="-1"></a>  </span>
<span id="cb180-341"><a href="source-code.html#cb180-341" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fisher_test-script-perform_fisher-function" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> fisher_test script: ‘perform_fisher’ function<a href="source-code.html#fisher_test-script-perform_fisher-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="source-code.html#cb181-1" tabindex="-1"></a>perform_fisher <span class="ot">&lt;-</span> <span class="cf">function</span>(dataframe,</span>
<span id="cb181-2"><a href="source-code.html#cb181-2" tabindex="-1"></a>                           population1,</span>
<span id="cb181-3"><a href="source-code.html#cb181-3" tabindex="-1"></a>                           population2) {</span>
<span id="cb181-4"><a href="source-code.html#cb181-4" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb181-5"><a href="source-code.html#cb181-5" tabindex="-1"></a>  <span class="co">#&#39; dataframe is a df with columns paths and class with values &#39;hub015_rad_a&#39;, &#39;hub015_prerad_a&#39; etc.</span></span>
<span id="cb181-6"><a href="source-code.html#cb181-6" tabindex="-1"></a>  <span class="co">#&#39; population 1 and population two are the two pops you want to compare, for example prerad vs rad</span></span>
<span id="cb181-7"><a href="source-code.html#cb181-7" tabindex="-1"></a>  <span class="co">#&#39; or prerad versus rad cycle 2</span></span>
<span id="cb181-8"><a href="source-code.html#cb181-8" tabindex="-1"></a>  <span class="co">#&#39; @return</span></span>
<span id="cb181-9"><a href="source-code.html#cb181-9" tabindex="-1"></a>  <span class="co">#&#39; returns the p value of a Fisher&#39;s exact test. If the p value is above 0.05, the distributions</span></span>
<span id="cb181-10"><a href="source-code.html#cb181-10" tabindex="-1"></a>  <span class="co">#&#39; pre and post rad are the same. </span></span>
<span id="cb181-11"><a href="source-code.html#cb181-11" tabindex="-1"></a>  <span class="co">#&#39; </span></span>
<span id="cb181-12"><a href="source-code.html#cb181-12" tabindex="-1"></a>  </span>
<span id="cb181-13"><a href="source-code.html#cb181-13" tabindex="-1"></a>  dataframe<span class="sc">$</span>rad <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population1, <span class="st">&#39;pop1&#39;</span>,</span>
<span id="cb181-14"><a href="source-code.html#cb181-14" tabindex="-1"></a>                          <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population2, <span class="st">&#39;pop2&#39;</span>,</span>
<span id="cb181-15"><a href="source-code.html#cb181-15" tabindex="-1"></a>                                 <span class="cn">NA</span>))</span>
<span id="cb181-16"><a href="source-code.html#cb181-16" tabindex="-1"></a>  </span>
<span id="cb181-17"><a href="source-code.html#cb181-17" tabindex="-1"></a>  org_id <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">str_extract</span>(<span class="fu">unique</span>(dataframe<span class="sc">$</span>class), <span class="st">&#39;hub</span><span class="sc">\\</span><span class="st">d{3}&#39;</span>))</span>
<span id="cb181-18"><a href="source-code.html#cb181-18" tabindex="-1"></a>  </span>
<span id="cb181-19"><a href="source-code.html#cb181-19" tabindex="-1"></a>  <span class="co"># remove na</span></span>
<span id="cb181-20"><a href="source-code.html#cb181-20" tabindex="-1"></a>  dataframe <span class="ot">&lt;-</span> dataframe[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(dataframe<span class="sc">$</span>rad)),]</span>
<span id="cb181-21"><a href="source-code.html#cb181-21" tabindex="-1"></a>  </span>
<span id="cb181-22"><a href="source-code.html#cb181-22" tabindex="-1"></a>  simple_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(string) {</span>
<span id="cb181-23"><a href="source-code.html#cb181-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">tail</span>(string,<span class="dv">1</span>))</span>
<span id="cb181-24"><a href="source-code.html#cb181-24" tabindex="-1"></a>  }</span>
<span id="cb181-25"><a href="source-code.html#cb181-25" tabindex="-1"></a>  </span>
<span id="cb181-26"><a href="source-code.html#cb181-26" tabindex="-1"></a>  dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stringr<span class="sc">::</span><span class="fu">str_split</span>(dataframe<span class="sc">$</span>class, <span class="at">pattern =</span> <span class="st">&#39;_&#39;</span>), simple_fun))</span>
<span id="cb181-27"><a href="source-code.html#cb181-27" tabindex="-1"></a>  </span>
<span id="cb181-28"><a href="source-code.html#cb181-28" tabindex="-1"></a>  <span class="co"># in hub015 and hub005, we defined subclones. These are actually part of the same clone, so should be </span></span>
<span id="cb181-29"><a href="source-code.html#cb181-29" tabindex="-1"></a>  <span class="co"># analysed together</span></span>
<span id="cb181-30"><a href="source-code.html#cb181-30" tabindex="-1"></a>  <span class="cf">if</span> (org_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;hub005&#39;</span>, <span class="st">&#39;hub015&#39;</span>)){</span>
<span id="cb181-31"><a href="source-code.html#cb181-31" tabindex="-1"></a>    dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(dataframe<span class="sc">$</span>clone, <span class="at">pattern =</span> <span class="st">&#39;a.a|a.b&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;a&#39;</span>)</span>
<span id="cb181-32"><a href="source-code.html#cb181-32" tabindex="-1"></a>    </span>
<span id="cb181-33"><a href="source-code.html#cb181-33" tabindex="-1"></a>  }</span>
<span id="cb181-34"><a href="source-code.html#cb181-34" tabindex="-1"></a>  </span>
<span id="cb181-35"><a href="source-code.html#cb181-35" tabindex="-1"></a>  tab <span class="ot">&lt;-</span> <span class="fu">table</span>(dataframe<span class="sc">$</span>clone, dataframe<span class="sc">$</span>rad)</span>
<span id="cb181-36"><a href="source-code.html#cb181-36" tabindex="-1"></a>  </span>
<span id="cb181-37"><a href="source-code.html#cb181-37" tabindex="-1"></a>  </span>
<span id="cb181-38"><a href="source-code.html#cb181-38" tabindex="-1"></a>  <span class="co"># df &lt;- data.frame(cline &lt;- org_id, p_val &lt;- fisher.test(tab)$p.value)</span></span>
<span id="cb181-39"><a href="source-code.html#cb181-39" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb181-40"><a href="source-code.html#cb181-40" tabindex="-1"></a>  <span class="co"># return(df) </span></span>
<span id="cb181-41"><a href="source-code.html#cb181-41" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">fisher.test</span>(tab)<span class="sc">$</span>p.value)</span>
<span id="cb181-42"><a href="source-code.html#cb181-42" tabindex="-1"></a>  </span>
<span id="cb181-43"><a href="source-code.html#cb181-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="find_specific_cna_sc-script-find_specific_cna_sc-function" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> find_specific_cna_sc script: ‘find_specific_cna_sc’ function<a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="source-code.html#cb182-1" tabindex="-1"></a>find_specific_cna_sc <span class="ot">&lt;-</span></span>
<span id="cb182-2"><a href="source-code.html#cb182-2" tabindex="-1"></a>  <span class="cf">function</span>(list_resistant,</span>
<span id="cb182-3"><a href="source-code.html#cb182-3" tabindex="-1"></a>           list_sensitive,</span>
<span id="cb182-4"><a href="source-code.html#cb182-4" tabindex="-1"></a>           perc_cutoff_within_subclone_resistant,</span>
<span id="cb182-5"><a href="source-code.html#cb182-5" tabindex="-1"></a>           perc_cutoff_within_subclone_sensitive,</span>
<span id="cb182-6"><a href="source-code.html#cb182-6" tabindex="-1"></a>           per_cutoff_across_subclone_resistant,</span>
<span id="cb182-7"><a href="source-code.html#cb182-7" tabindex="-1"></a>           per_cutoff_across_subclone_sensitive,</span>
<span id="cb182-8"><a href="source-code.html#cb182-8" tabindex="-1"></a>           <span class="at">return_cnas_per_subclone =</span> F) {</span>
<span id="cb182-9"><a href="source-code.html#cb182-9" tabindex="-1"></a>    <span class="co">#&#39; @param list_resistant a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb182-10"><a href="source-code.html#cb182-10" tabindex="-1"></a>    <span class="co">#&#39; @param list_sensitive a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb182-11"><a href="source-code.html#cb182-11" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_resistant user defined percentage cutoff to use (e.g. 0.6 means a certain CNA is shared by 60% of the cells</span></span>
<span id="cb182-12"><a href="source-code.html#cb182-12" tabindex="-1"></a>    <span class="co">#&#39; within a subclone).</span></span>
<span id="cb182-13"><a href="source-code.html#cb182-13" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_senstive same as in perc_cutoff_within_subclone_resistant but for the sensitive group</span></span>
<span id="cb182-14"><a href="source-code.html#cb182-14" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_resistant user defined percentage cutoff to use (e.g. 0.7 means a certain CNA is shared in 70% of all subclones)</span></span>
<span id="cb182-15"><a href="source-code.html#cb182-15" tabindex="-1"></a>    <span class="co">#&#39; @param return_cnas_per_subclone boolean. If TRUE will return the shared by perc_cutoff_within_subclone CNAs for each list of each item within </span></span>
<span id="cb182-16"><a href="source-code.html#cb182-16" tabindex="-1"></a>    <span class="co">#&#39; list_resistant. Note list_sensitive should be set to NULL for proper results!!</span></span>
<span id="cb182-17"><a href="source-code.html#cb182-17" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_sensitive same as resistant but for the sensitive group</span></span>
<span id="cb182-18"><a href="source-code.html#cb182-18" tabindex="-1"></a>    <span class="co">#&#39; @example perc_cutoff_within_subclone = 0.6 and per_cutoff_across_subclone = 0.7 means the algorithm will find CNAs that are shared </span></span>
<span id="cb182-19"><a href="source-code.html#cb182-19" tabindex="-1"></a>    <span class="co">#&#39; by 60% of single cells within each subclone, and by 70% of all subclones. Important, a cutoff of 50% is the same as computing the median.</span></span>
<span id="cb182-20"><a href="source-code.html#cb182-20" tabindex="-1"></a>    <span class="co">#&#39; @return a list of GRanges showing unique to list_resistant amplifications and deletions and unique to list_sensitive</span></span>
<span id="cb182-21"><a href="source-code.html#cb182-21" tabindex="-1"></a>    <span class="co">#&#39; amplifications and deletions or, if return_cnas_per_subclone see param return_cnas_per_subclone above.</span></span>
<span id="cb182-22"><a href="source-code.html#cb182-22" tabindex="-1"></a>    <span class="co">#&#39; @details Note that this algorithm is NOT sensitive to the number of single cells within each subclone. </span></span>
<span id="cb182-23"><a href="source-code.html#cb182-23" tabindex="-1"></a>    </span>
<span id="cb182-24"><a href="source-code.html#cb182-24" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb182-25"><a href="source-code.html#cb182-25" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb182-26"><a href="source-code.html#cb182-26" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb182-27"><a href="source-code.html#cb182-27" tabindex="-1"></a>    <span class="fu">require</span>(plyranges)</span>
<span id="cb182-28"><a href="source-code.html#cb182-28" tabindex="-1"></a>    <span class="fu">require</span>(AneuFinder)</span>
<span id="cb182-29"><a href="source-code.html#cb182-29" tabindex="-1"></a>    </span>
<span id="cb182-30"><a href="source-code.html#cb182-30" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-31"><a href="source-code.html#cb182-31" tabindex="-1"></a>    <span class="co"># Part 1. #</span></span>
<span id="cb182-32"><a href="source-code.html#cb182-32" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-33"><a href="source-code.html#cb182-33" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb182-34"><a href="source-code.html#cb182-34" tabindex="-1"></a>    </span>
<span id="cb182-35"><a href="source-code.html#cb182-35" tabindex="-1"></a>    resistant_sc <span class="ot">&lt;-</span></span>
<span id="cb182-36"><a href="source-code.html#cb182-36" tabindex="-1"></a>      <span class="fu">lapply</span>(list_resistant,</span>
<span id="cb182-37"><a href="source-code.html#cb182-37" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb182-38"><a href="source-code.html#cb182-38" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb182-39"><a href="source-code.html#cb182-39" tabindex="-1"></a>    sensitive_sc <span class="ot">&lt;-</span></span>
<span id="cb182-40"><a href="source-code.html#cb182-40" tabindex="-1"></a>      <span class="fu">lapply</span>(list_sensitive,</span>
<span id="cb182-41"><a href="source-code.html#cb182-41" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb182-42"><a href="source-code.html#cb182-42" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb182-43"><a href="source-code.html#cb182-43" tabindex="-1"></a>    </span>
<span id="cb182-44"><a href="source-code.html#cb182-44" tabindex="-1"></a>    <span class="co"># find specific CNAs needs a list of GRanges where each GRange has columns</span></span>
<span id="cb182-45"><a href="source-code.html#cb182-45" tabindex="-1"></a>    <span class="co"># seqnames, ranges, strand (not necessary) and cn.</span></span>
<span id="cb182-46"><a href="source-code.html#cb182-46" tabindex="-1"></a>    wrangle_sc <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff) {</span>
<span id="cb182-47"><a href="source-code.html#cb182-47" tabindex="-1"></a>      <span class="co"># segments is the reduced bins GRange (so all equal CN bins put together)</span></span>
<span id="cb182-48"><a href="source-code.html#cb182-48" tabindex="-1"></a>      wrangle_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff_within) {</span>
<span id="cb182-49"><a href="source-code.html#cb182-49" tabindex="-1"></a>        segments <span class="ot">&lt;-</span> list_lff_within<span class="sc">$</span>segments</span>
<span id="cb182-50"><a href="source-code.html#cb182-50" tabindex="-1"></a>        <span class="co"># ugly way of adding a cn column</span></span>
<span id="cb182-51"><a href="source-code.html#cb182-51" tabindex="-1"></a>        segments<span class="sc">$</span>cn <span class="ot">&lt;-</span> segments<span class="sc">$</span>copy.number</span>
<span id="cb182-52"><a href="source-code.html#cb182-52" tabindex="-1"></a>        </span>
<span id="cb182-53"><a href="source-code.html#cb182-53" tabindex="-1"></a>        <span class="fu">return</span>(segments)</span>
<span id="cb182-54"><a href="source-code.html#cb182-54" tabindex="-1"></a>        </span>
<span id="cb182-55"><a href="source-code.html#cb182-55" tabindex="-1"></a>        <span class="co"># </span><span class="re">END</span><span class="co"> of wrange_within function</span></span>
<span id="cb182-56"><a href="source-code.html#cb182-56" tabindex="-1"></a>        </span>
<span id="cb182-57"><a href="source-code.html#cb182-57" tabindex="-1"></a>      }</span>
<span id="cb182-58"><a href="source-code.html#cb182-58" tabindex="-1"></a>      within_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_lff, wrangle_within)</span>
<span id="cb182-59"><a href="source-code.html#cb182-59" tabindex="-1"></a>      </span>
<span id="cb182-60"><a href="source-code.html#cb182-60" tabindex="-1"></a>      <span class="co"># </span><span class="re">END</span><span class="co"> OF wrangle_sc function</span></span>
<span id="cb182-61"><a href="source-code.html#cb182-61" tabindex="-1"></a>      </span>
<span id="cb182-62"><a href="source-code.html#cb182-62" tabindex="-1"></a>    }</span>
<span id="cb182-63"><a href="source-code.html#cb182-63" tabindex="-1"></a></span>
<span id="cb182-64"><a href="source-code.html#cb182-64" tabindex="-1"></a>    list_resistant <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_sc, wrangle_sc)</span>
<span id="cb182-65"><a href="source-code.html#cb182-65" tabindex="-1"></a>    list_sensitive <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_sc, wrangle_sc)</span>
<span id="cb182-66"><a href="source-code.html#cb182-66" tabindex="-1"></a>    </span>
<span id="cb182-67"><a href="source-code.html#cb182-67" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Extracted segment info from files&#39;</span>)</span>
<span id="cb182-68"><a href="source-code.html#cb182-68" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb182-69"><a href="source-code.html#cb182-69" tabindex="-1"></a>    </span>
<span id="cb182-70"><a href="source-code.html#cb182-70" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-71"><a href="source-code.html#cb182-71" tabindex="-1"></a>    <span class="co"># Part 2. #</span></span>
<span id="cb182-72"><a href="source-code.html#cb182-72" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-73"><a href="source-code.html#cb182-73" tabindex="-1"></a>    <span class="co"># The Granges are now still arranged in bins. We are first going to</span></span>
<span id="cb182-74"><a href="source-code.html#cb182-74" tabindex="-1"></a>    <span class="co"># combine bins with similar copy number.</span></span>
<span id="cb182-75"><a href="source-code.html#cb182-75" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb182-76"><a href="source-code.html#cb182-76" tabindex="-1"></a>    combine_indels <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus) {</span>
<span id="cb182-77"><a href="source-code.html#cb182-77" tabindex="-1"></a>      <span class="co">#&#39; @param subclone_consensus_list is a GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb182-78"><a href="source-code.html#cb182-78" tabindex="-1"></a>      <span class="co">#&#39; @returns list of Granges where each Granges has only CNA of cn2 or cn&gt;2</span></span>
<span id="cb182-79"><a href="source-code.html#cb182-79" tabindex="-1"></a>      </span>
<span id="cb182-80"><a href="source-code.html#cb182-80" tabindex="-1"></a>      combine_within <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus_within) {</span>
<span id="cb182-81"><a href="source-code.html#cb182-81" tabindex="-1"></a>        <span class="co"># subset subclone_consensus in cn &lt;2, cn = 2 and cn &gt; 2</span></span>
<span id="cb182-82"><a href="source-code.html#cb182-82" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span></span>
<span id="cb182-83"><a href="source-code.html#cb182-83" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&lt;</span> <span class="dv">2</span>,]</span>
<span id="cb182-84"><a href="source-code.html#cb182-84" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span></span>
<span id="cb182-85"><a href="source-code.html#cb182-85" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&gt;</span> <span class="dv">2</span>,]</span>
<span id="cb182-86"><a href="source-code.html#cb182-86" tabindex="-1"></a>        </span>
<span id="cb182-87"><a href="source-code.html#cb182-87" tabindex="-1"></a>        <span class="co"># reduce each GRanges (joins consecutive ranges together)</span></span>
<span id="cb182-88"><a href="source-code.html#cb182-88" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_less_2)</span>
<span id="cb182-89"><a href="source-code.html#cb182-89" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_more_2)</span>
<span id="cb182-90"><a href="source-code.html#cb182-90" tabindex="-1"></a>        </span>
<span id="cb182-91"><a href="source-code.html#cb182-91" tabindex="-1"></a>        <span class="co"># reduce deletes metadata.</span></span>
<span id="cb182-92"><a href="source-code.html#cb182-92" tabindex="-1"></a>        <span class="co"># adding metadata</span></span>
<span id="cb182-93"><a href="source-code.html#cb182-93" tabindex="-1"></a>        cn_less_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_less_2))</span>
<span id="cb182-94"><a href="source-code.html#cb182-94" tabindex="-1"></a>        </span>
<span id="cb182-95"><a href="source-code.html#cb182-95" tabindex="-1"></a>        cn_more_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_more_2))</span>
<span id="cb182-96"><a href="source-code.html#cb182-96" tabindex="-1"></a>        </span>
<span id="cb182-97"><a href="source-code.html#cb182-97" tabindex="-1"></a>        <span class="co"># make list and rename items</span></span>
<span id="cb182-98"><a href="source-code.html#cb182-98" tabindex="-1"></a>        GR_list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(cn_less_2, cn_more_2)</span>
<span id="cb182-99"><a href="source-code.html#cb182-99" tabindex="-1"></a>        <span class="fu">names</span>(GR_list) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb182-100"><a href="source-code.html#cb182-100" tabindex="-1"></a>        </span>
<span id="cb182-101"><a href="source-code.html#cb182-101" tabindex="-1"></a>        <span class="co"># Return #</span></span>
<span id="cb182-102"><a href="source-code.html#cb182-102" tabindex="-1"></a>        <span class="fu">return</span>(GR_list)</span>
<span id="cb182-103"><a href="source-code.html#cb182-103" tabindex="-1"></a>      }</span>
<span id="cb182-104"><a href="source-code.html#cb182-104" tabindex="-1"></a>      </span>
<span id="cb182-105"><a href="source-code.html#cb182-105" tabindex="-1"></a>      combine_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(subclone_consensus, combine_within)</span>
<span id="cb182-106"><a href="source-code.html#cb182-106" tabindex="-1"></a>      </span>
<span id="cb182-107"><a href="source-code.html#cb182-107" tabindex="-1"></a>    }</span>
<span id="cb182-108"><a href="source-code.html#cb182-108" tabindex="-1"></a>    </span>
<span id="cb182-109"><a href="source-code.html#cb182-109" tabindex="-1"></a>    <span class="co"># run an apply function here on each Granges in list_resistant versus list_sensitive</span></span>
<span id="cb182-110"><a href="source-code.html#cb182-110" tabindex="-1"></a>    resistant_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_resistant, combine_indels)</span>
<span id="cb182-111"><a href="source-code.html#cb182-111" tabindex="-1"></a>    sensitive_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_sensitive, combine_indels)</span>
<span id="cb182-112"><a href="source-code.html#cb182-112" tabindex="-1"></a>    </span>
<span id="cb182-113"><a href="source-code.html#cb182-113" tabindex="-1"></a>    <span class="co"># splits deletions together within each subclone</span></span>
<span id="cb182-114"><a href="source-code.html#cb182-114" tabindex="-1"></a>    grouper_del <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb182-115"><a href="source-code.html#cb182-115" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb182-116"><a href="source-code.html#cb182-116" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&lt;2&#39;</span>)</span>
<span id="cb182-117"><a href="source-code.html#cb182-117" tabindex="-1"></a>      }</span>
<span id="cb182-118"><a href="source-code.html#cb182-118" tabindex="-1"></a>      </span>
<span id="cb182-119"><a href="source-code.html#cb182-119" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb182-120"><a href="source-code.html#cb182-120" tabindex="-1"></a>    }</span>
<span id="cb182-121"><a href="source-code.html#cb182-121" tabindex="-1"></a>    </span>
<span id="cb182-122"><a href="source-code.html#cb182-122" tabindex="-1"></a>    <span class="co"># splits amplifciations together within each subclone</span></span>
<span id="cb182-123"><a href="source-code.html#cb182-123" tabindex="-1"></a>    grouper_ampl <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb182-124"><a href="source-code.html#cb182-124" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb182-125"><a href="source-code.html#cb182-125" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb182-126"><a href="source-code.html#cb182-126" tabindex="-1"></a>      }</span>
<span id="cb182-127"><a href="source-code.html#cb182-127" tabindex="-1"></a>      </span>
<span id="cb182-128"><a href="source-code.html#cb182-128" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb182-129"><a href="source-code.html#cb182-129" tabindex="-1"></a>    }</span>
<span id="cb182-130"><a href="source-code.html#cb182-130" tabindex="-1"></a>    </span>
<span id="cb182-131"><a href="source-code.html#cb182-131" tabindex="-1"></a>    <span class="co"># run apply on each subclone</span></span>
<span id="cb182-132"><a href="source-code.html#cb182-132" tabindex="-1"></a>    resistant_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_del)</span>
<span id="cb182-133"><a href="source-code.html#cb182-133" tabindex="-1"></a>    resistant_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_ampl)</span>
<span id="cb182-134"><a href="source-code.html#cb182-134" tabindex="-1"></a>    </span>
<span id="cb182-135"><a href="source-code.html#cb182-135" tabindex="-1"></a>    sensitive_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_del)</span>
<span id="cb182-136"><a href="source-code.html#cb182-136" tabindex="-1"></a>    sensitive_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_ampl)</span>
<span id="cb182-137"><a href="source-code.html#cb182-137" tabindex="-1"></a>    </span>
<span id="cb182-138"><a href="source-code.html#cb182-138" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-139"><a href="source-code.html#cb182-139" tabindex="-1"></a>    <span class="co"># Part 3. #</span></span>
<span id="cb182-140"><a href="source-code.html#cb182-140" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-141"><a href="source-code.html#cb182-141" tabindex="-1"></a>    <span class="co"># Part 3A #</span></span>
<span id="cb182-142"><a href="source-code.html#cb182-142" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-143"><a href="source-code.html#cb182-143" tabindex="-1"></a>    <span class="co"># Finding common CNAs within each subclone</span></span>
<span id="cb182-144"><a href="source-code.html#cb182-144" tabindex="-1"></a>    shared_cnas <span class="ot">&lt;-</span></span>
<span id="cb182-145"><a href="source-code.html#cb182-145" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb182-146"><a href="source-code.html#cb182-146" tabindex="-1"></a>               perc_cutoff_within_subclone) {</span>
<span id="cb182-147"><a href="source-code.html#cb182-147" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb182-148"><a href="source-code.html#cb182-148" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff_within_subclone user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb182-149"><a href="source-code.html#cb182-149" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb182-150"><a href="source-code.html#cb182-150" tabindex="-1"></a>        </span>
<span id="cb182-151"><a href="source-code.html#cb182-151" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb182-152"><a href="source-code.html#cb182-152" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb182-153"><a href="source-code.html#cb182-153" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb182-154"><a href="source-code.html#cb182-154" tabindex="-1"></a>        </span>
<span id="cb182-155"><a href="source-code.html#cb182-155" tabindex="-1"></a>        shared_cnas_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_GRanges_within) {</span>
<span id="cb182-156"><a href="source-code.html#cb182-156" tabindex="-1"></a>          n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges_within) <span class="co"># number of range sets</span></span>
<span id="cb182-157"><a href="source-code.html#cb182-157" tabindex="-1"></a>          shared_cna <span class="ot">&lt;-</span></span>
<span id="cb182-158"><a href="source-code.html#cb182-158" tabindex="-1"></a>            <span class="fu">bind_ranges</span>(list_GRanges_within, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb182-159"><a href="source-code.html#cb182-159" tabindex="-1"></a>            <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb182-160"><a href="source-code.html#cb182-160" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb182-161"><a href="source-code.html#cb182-161" tabindex="-1"></a>            <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> perc_cutoff_within_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb182-162"><a href="source-code.html#cb182-162" tabindex="-1"></a>            <span class="fu">reduce_ranges</span>()</span>
<span id="cb182-163"><a href="source-code.html#cb182-163" tabindex="-1"></a>          </span>
<span id="cb182-164"><a href="source-code.html#cb182-164" tabindex="-1"></a>          <span class="fu">return</span>(shared_cna)</span>
<span id="cb182-165"><a href="source-code.html#cb182-165" tabindex="-1"></a>        }</span>
<span id="cb182-166"><a href="source-code.html#cb182-166" tabindex="-1"></a>        </span>
<span id="cb182-167"><a href="source-code.html#cb182-167" tabindex="-1"></a>        <span class="fu">lapply</span>(list_GRanges, shared_cnas_within)</span>
<span id="cb182-168"><a href="source-code.html#cb182-168" tabindex="-1"></a>      }</span>
<span id="cb182-169"><a href="source-code.html#cb182-169" tabindex="-1"></a>    </span>
<span id="cb182-170"><a href="source-code.html#cb182-170" tabindex="-1"></a>    resistant_shared_del <span class="ot">&lt;-</span></span>
<span id="cb182-171"><a href="source-code.html#cb182-171" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb182-172"><a href="source-code.html#cb182-172" tabindex="-1"></a>    resistant_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb182-173"><a href="source-code.html#cb182-173" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb182-174"><a href="source-code.html#cb182-174" tabindex="-1"></a>    sensitive_shared_del <span class="ot">&lt;-</span></span>
<span id="cb182-175"><a href="source-code.html#cb182-175" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb182-176"><a href="source-code.html#cb182-176" tabindex="-1"></a>    sensitive_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb182-177"><a href="source-code.html#cb182-177" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb182-178"><a href="source-code.html#cb182-178" tabindex="-1"></a>    </span>
<span id="cb182-179"><a href="source-code.html#cb182-179" tabindex="-1"></a>    <span class="co"># </span><span class="re">END</span><span class="co"> of function if return_cnas_per_subclone == T</span></span>
<span id="cb182-180"><a href="source-code.html#cb182-180" tabindex="-1"></a>    <span class="cf">if</span> (return_cnas_per_subclone <span class="sc">==</span> T) {</span>
<span id="cb182-181"><a href="source-code.html#cb182-181" tabindex="-1"></a>      cna_list <span class="ot">&lt;-</span> <span class="fu">list</span> (<span class="at">deletions =</span> <span class="fu">c</span>(resistant_shared_del, sensitive_shared_del), </span>
<span id="cb182-182"><a href="source-code.html#cb182-182" tabindex="-1"></a>                        <span class="at">amplifications =</span> <span class="fu">c</span>(resistant_shared_ampl, sensitive_shared_del))</span>
<span id="cb182-183"><a href="source-code.html#cb182-183" tabindex="-1"></a>      </span>
<span id="cb182-184"><a href="source-code.html#cb182-184" tabindex="-1"></a>      <span class="fu">print</span>(</span>
<span id="cb182-185"><a href="source-code.html#cb182-185" tabindex="-1"></a>        <span class="fu">paste0</span>(</span>
<span id="cb182-186"><a href="source-code.html#cb182-186" tabindex="-1"></a>          <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb182-187"><a href="source-code.html#cb182-187" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb182-188"><a href="source-code.html#cb182-188" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each resistant subclone, and by &#39;</span>,</span>
<span id="cb182-189"><a href="source-code.html#cb182-189" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb182-190"><a href="source-code.html#cb182-190" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each sensitive subclone&#39;</span></span>
<span id="cb182-191"><a href="source-code.html#cb182-191" tabindex="-1"></a>        )</span>
<span id="cb182-192"><a href="source-code.html#cb182-192" tabindex="-1"></a>      )</span>
<span id="cb182-193"><a href="source-code.html#cb182-193" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb182-194"><a href="source-code.html#cb182-194" tabindex="-1"></a>      <span class="co"># Unload #</span></span>
<span id="cb182-195"><a href="source-code.html#cb182-195" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb182-196"><a href="source-code.html#cb182-196" tabindex="-1"></a>      <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb182-197"><a href="source-code.html#cb182-197" tabindex="-1"></a>                                        detach,</span>
<span id="cb182-198"><a href="source-code.html#cb182-198" tabindex="-1"></a>                                        <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb182-199"><a href="source-code.html#cb182-199" tabindex="-1"></a>      </span>
<span id="cb182-200"><a href="source-code.html#cb182-200" tabindex="-1"></a>      <span class="fu">return</span>(cna_list)</span>
<span id="cb182-201"><a href="source-code.html#cb182-201" tabindex="-1"></a>    }</span>
<span id="cb182-202"><a href="source-code.html#cb182-202" tabindex="-1"></a>    </span>
<span id="cb182-203"><a href="source-code.html#cb182-203" tabindex="-1"></a>    <span class="co"># Part 3B #</span></span>
<span id="cb182-204"><a href="source-code.html#cb182-204" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-205"><a href="source-code.html#cb182-205" tabindex="-1"></a>    <span class="co"># Finding common CNAs across subclones</span></span>
<span id="cb182-206"><a href="source-code.html#cb182-206" tabindex="-1"></a>    shared_cnas_across_subclones <span class="ot">&lt;-</span></span>
<span id="cb182-207"><a href="source-code.html#cb182-207" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb182-208"><a href="source-code.html#cb182-208" tabindex="-1"></a>               per_cutoff_across_subclone) {</span>
<span id="cb182-209"><a href="source-code.html#cb182-209" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb182-210"><a href="source-code.html#cb182-210" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb182-211"><a href="source-code.html#cb182-211" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb182-212"><a href="source-code.html#cb182-212" tabindex="-1"></a>        </span>
<span id="cb182-213"><a href="source-code.html#cb182-213" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb182-214"><a href="source-code.html#cb182-214" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb182-215"><a href="source-code.html#cb182-215" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb182-216"><a href="source-code.html#cb182-216" tabindex="-1"></a>        </span>
<span id="cb182-217"><a href="source-code.html#cb182-217" tabindex="-1"></a>        n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges) <span class="co"># number of range sets</span></span>
<span id="cb182-218"><a href="source-code.html#cb182-218" tabindex="-1"></a>        shared_cna <span class="ot">&lt;-</span></span>
<span id="cb182-219"><a href="source-code.html#cb182-219" tabindex="-1"></a>          <span class="fu">bind_ranges</span>(list_GRanges, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb182-220"><a href="source-code.html#cb182-220" tabindex="-1"></a>          <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb182-221"><a href="source-code.html#cb182-221" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb182-222"><a href="source-code.html#cb182-222" tabindex="-1"></a>          <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> per_cutoff_across_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb182-223"><a href="source-code.html#cb182-223" tabindex="-1"></a>          <span class="fu">reduce_ranges</span>()</span>
<span id="cb182-224"><a href="source-code.html#cb182-224" tabindex="-1"></a>        </span>
<span id="cb182-225"><a href="source-code.html#cb182-225" tabindex="-1"></a>        <span class="fu">return</span>(shared_cna)</span>
<span id="cb182-226"><a href="source-code.html#cb182-226" tabindex="-1"></a>        </span>
<span id="cb182-227"><a href="source-code.html#cb182-227" tabindex="-1"></a>      }</span>
<span id="cb182-228"><a href="source-code.html#cb182-228" tabindex="-1"></a>    </span>
<span id="cb182-229"><a href="source-code.html#cb182-229" tabindex="-1"></a>    resistant_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb182-230"><a href="source-code.html#cb182-230" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb182-231"><a href="source-code.html#cb182-231" tabindex="-1"></a>    resistant_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb182-232"><a href="source-code.html#cb182-232" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb182-233"><a href="source-code.html#cb182-233" tabindex="-1"></a>    sensitive_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb182-234"><a href="source-code.html#cb182-234" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb182-235"><a href="source-code.html#cb182-235" tabindex="-1"></a>    sensitive_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb182-236"><a href="source-code.html#cb182-236" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb182-237"><a href="source-code.html#cb182-237" tabindex="-1"></a>    </span>
<span id="cb182-238"><a href="source-code.html#cb182-238" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb182-239"><a href="source-code.html#cb182-239" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb182-240"><a href="source-code.html#cb182-240" tabindex="-1"></a>        <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb182-241"><a href="source-code.html#cb182-241" tabindex="-1"></a>        <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb182-242"><a href="source-code.html#cb182-242" tabindex="-1"></a>        <span class="st">&#39;% of single cells within resistant and sensitive subclones, respectively, and that are shared by &#39;</span>,</span>
<span id="cb182-243"><a href="source-code.html#cb182-243" tabindex="-1"></a>        <span class="fu">as.numeric</span>(per_cutoff_across_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(per_cutoff_across_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb182-244"><a href="source-code.html#cb182-244" tabindex="-1"></a>        <span class="st">&#39;% across resistant and sensitive subclones, respectively.&#39;</span></span>
<span id="cb182-245"><a href="source-code.html#cb182-245" tabindex="-1"></a>      )</span>
<span id="cb182-246"><a href="source-code.html#cb182-246" tabindex="-1"></a>    )</span>
<span id="cb182-247"><a href="source-code.html#cb182-247" tabindex="-1"></a>    </span>
<span id="cb182-248"><a href="source-code.html#cb182-248" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb182-249"><a href="source-code.html#cb182-249" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (few seconds)...&#39;</span>)</span>
<span id="cb182-250"><a href="source-code.html#cb182-250" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">2</span>) </span>
<span id="cb182-251"><a href="source-code.html#cb182-251" tabindex="-1"></a>    </span>
<span id="cb182-252"><a href="source-code.html#cb182-252" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-253"><a href="source-code.html#cb182-253" tabindex="-1"></a>    <span class="co"># Part 4. #</span></span>
<span id="cb182-254"><a href="source-code.html#cb182-254" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb182-255"><a href="source-code.html#cb182-255" tabindex="-1"></a>    <span class="co"># Now that we have shared CNAs across subclones with the resistant and sensitive group, we need to extract</span></span>
<span id="cb182-256"><a href="source-code.html#cb182-256" tabindex="-1"></a>    <span class="co"># only those CNAs that are unique to the resistant group.</span></span>
<span id="cb182-257"><a href="source-code.html#cb182-257" tabindex="-1"></a>    </span>
<span id="cb182-258"><a href="source-code.html#cb182-258" tabindex="-1"></a>    <span class="co"># unique to resistant lines deletions:</span></span>
<span id="cb182-259"><a href="source-code.html#cb182-259" tabindex="-1"></a>    unique_resistant_del <span class="ot">&lt;-</span></span>
<span id="cb182-260"><a href="source-code.html#cb182-260" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_del_across, sensitive_shared_del_across)</span>
<span id="cb182-261"><a href="source-code.html#cb182-261" tabindex="-1"></a>    <span class="co"># unique to resistant lines amplifications</span></span>
<span id="cb182-262"><a href="source-code.html#cb182-262" tabindex="-1"></a>    unique_resistant_ampl <span class="ot">&lt;-</span></span>
<span id="cb182-263"><a href="source-code.html#cb182-263" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_ampl_across,</span>
<span id="cb182-264"><a href="source-code.html#cb182-264" tabindex="-1"></a>              sensitive_shared_ampl_across)</span>
<span id="cb182-265"><a href="source-code.html#cb182-265" tabindex="-1"></a>    </span>
<span id="cb182-266"><a href="source-code.html#cb182-266" tabindex="-1"></a>    <span class="co"># unique to sensitive lines deletions:</span></span>
<span id="cb182-267"><a href="source-code.html#cb182-267" tabindex="-1"></a>    unique_sensitive_del <span class="ot">&lt;-</span></span>
<span id="cb182-268"><a href="source-code.html#cb182-268" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_del_across, resistant_shared_del_across)</span>
<span id="cb182-269"><a href="source-code.html#cb182-269" tabindex="-1"></a>    <span class="co"># unique to sensitive lines amplifications:</span></span>
<span id="cb182-270"><a href="source-code.html#cb182-270" tabindex="-1"></a>    unique_sensitive_ampl <span class="ot">&lt;-</span></span>
<span id="cb182-271"><a href="source-code.html#cb182-271" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_ampl_across,</span>
<span id="cb182-272"><a href="source-code.html#cb182-272" tabindex="-1"></a>              resistant_shared_ampl_across)</span>
<span id="cb182-273"><a href="source-code.html#cb182-273" tabindex="-1"></a>    </span>
<span id="cb182-274"><a href="source-code.html#cb182-274" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb182-275"><a href="source-code.html#cb182-275" tabindex="-1"></a>      <span class="st">&#39;Identified amplifications and deletions that are unique to resistant or sensitive subclones&#39;</span></span>
<span id="cb182-276"><a href="source-code.html#cb182-276" tabindex="-1"></a>    )</span>
<span id="cb182-277"><a href="source-code.html#cb182-277" tabindex="-1"></a>    </span>
<span id="cb182-278"><a href="source-code.html#cb182-278" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb182-279"><a href="source-code.html#cb182-279" tabindex="-1"></a>    <span class="co"># Return #</span></span>
<span id="cb182-280"><a href="source-code.html#cb182-280" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb182-281"><a href="source-code.html#cb182-281" tabindex="-1"></a>    returner <span class="ot">&lt;-</span></span>
<span id="cb182-282"><a href="source-code.html#cb182-282" tabindex="-1"></a>      <span class="fu">list</span>(</span>
<span id="cb182-283"><a href="source-code.html#cb182-283" tabindex="-1"></a>        unique_resistant_del,</span>
<span id="cb182-284"><a href="source-code.html#cb182-284" tabindex="-1"></a>        unique_resistant_ampl,</span>
<span id="cb182-285"><a href="source-code.html#cb182-285" tabindex="-1"></a>        unique_sensitive_del,</span>
<span id="cb182-286"><a href="source-code.html#cb182-286" tabindex="-1"></a>        unique_sensitive_ampl</span>
<span id="cb182-287"><a href="source-code.html#cb182-287" tabindex="-1"></a>      )</span>
<span id="cb182-288"><a href="source-code.html#cb182-288" tabindex="-1"></a>    <span class="fu">names</span>(returner) <span class="ot">&lt;-</span></span>
<span id="cb182-289"><a href="source-code.html#cb182-289" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb182-290"><a href="source-code.html#cb182-290" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_del&#39;</span>,</span>
<span id="cb182-291"><a href="source-code.html#cb182-291" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_ampl&#39;</span>,</span>
<span id="cb182-292"><a href="source-code.html#cb182-292" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_del&#39;</span>,</span>
<span id="cb182-293"><a href="source-code.html#cb182-293" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_ampl&#39;</span></span>
<span id="cb182-294"><a href="source-code.html#cb182-294" tabindex="-1"></a>      )</span>
<span id="cb182-295"><a href="source-code.html#cb182-295" tabindex="-1"></a>    </span>
<span id="cb182-296"><a href="source-code.html#cb182-296" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb182-297"><a href="source-code.html#cb182-297" tabindex="-1"></a>    <span class="co"># Unload #</span></span>
<span id="cb182-298"><a href="source-code.html#cb182-298" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb182-299"><a href="source-code.html#cb182-299" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb182-300"><a href="source-code.html#cb182-300" tabindex="-1"></a>                     detach,</span>
<span id="cb182-301"><a href="source-code.html#cb182-301" tabindex="-1"></a>                     <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb182-302"><a href="source-code.html#cb182-302" tabindex="-1"></a>    </span>
<span id="cb182-303"><a href="source-code.html#cb182-303" tabindex="-1"></a>    <span class="fu">return</span>(returner)</span>
<span id="cb182-304"><a href="source-code.html#cb182-304" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="get_cytoband_coverage-script-get_cytoband_coverage-function" class="section level2 hasAnchor" number="6.11">
<h2><span class="header-section-number">6.11</span> get_cytoband_coverage script: ‘get_cytoband_coverage’ function<a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="source-code.html#cb183-1" tabindex="-1"></a>get_cytoband_coverage <span class="ot">&lt;-</span> <span class="cf">function</span>(Grange) {</span>
<span id="cb183-2"><a href="source-code.html#cb183-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges.</span></span>
<span id="cb183-3"><a href="source-code.html#cb183-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns a GRange with cytogenetic locations including</span></span>
<span id="cb183-4"><a href="source-code.html#cb183-4" tabindex="-1"></a>  <span class="co">#&#39; percentage of overlap and a dataframe containing information on how much of</span></span>
<span id="cb183-5"><a href="source-code.html#cb183-5" tabindex="-1"></a>  <span class="co">#&#39; the total arm was affected.</span></span>
<span id="cb183-6"><a href="source-code.html#cb183-6" tabindex="-1"></a>  </span>
<span id="cb183-7"><a href="source-code.html#cb183-7" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb183-8"><a href="source-code.html#cb183-8" tabindex="-1"></a>  <span class="co"># Libraries #</span></span>
<span id="cb183-9"><a href="source-code.html#cb183-9" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb183-10"><a href="source-code.html#cb183-10" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb183-11"><a href="source-code.html#cb183-11" tabindex="-1"></a>  <span class="fu">require</span>(plyranges)</span>
<span id="cb183-12"><a href="source-code.html#cb183-12" tabindex="-1"></a>  </span>
<span id="cb183-13"><a href="source-code.html#cb183-13" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb183-14"><a href="source-code.html#cb183-14" tabindex="-1"></a>  <span class="co"># Data #</span></span>
<span id="cb183-15"><a href="source-code.html#cb183-15" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb183-16"><a href="source-code.html#cb183-16" tabindex="-1"></a>  <span class="co"># Uncomment me, this is to download cytogenetic coordinates</span></span>
<span id="cb183-17"><a href="source-code.html#cb183-17" tabindex="-1"></a>  <span class="co"># library(biovizBase)</span></span>
<span id="cb183-18"><a href="source-code.html#cb183-18" tabindex="-1"></a>  <span class="co"># hg38IdeogramCyto &lt;- getIdeogram(&quot;hg38&quot;, cytobands = TRUE)</span></span>
<span id="cb183-19"><a href="source-code.html#cb183-19" tabindex="-1"></a>  <span class="co"># save(hg38IdeogramCyto, file = &#39;rda/cna_analysis/hg38IdeogramCyto.rda&#39;)</span></span>
<span id="cb183-20"><a href="source-code.html#cb183-20" tabindex="-1"></a>  </span>
<span id="cb183-21"><a href="source-code.html#cb183-21" tabindex="-1"></a>  <span class="co"># !!! ADDING ../cna_analysis for publication in GitHub !!!</span></span>
<span id="cb183-22"><a href="source-code.html#cb183-22" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hg38IdeogramCyto.rda&#39;</span>)</span>
<span id="cb183-23"><a href="source-code.html#cb183-23" tabindex="-1"></a>  </span>
<span id="cb183-24"><a href="source-code.html#cb183-24" tabindex="-1"></a>  </span>
<span id="cb183-25"><a href="source-code.html#cb183-25" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb183-26"><a href="source-code.html#cb183-26" tabindex="-1"></a>  <span class="co"># Wrangle #</span></span>
<span id="cb183-27"><a href="source-code.html#cb183-27" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb183-28"><a href="source-code.html#cb183-28" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb183-29"><a href="source-code.html#cb183-29" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(Grange) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb183-30"><a href="source-code.html#cb183-30" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(hg38IdeogramCyto) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb183-31"><a href="source-code.html#cb183-31" tabindex="-1"></a>  </span>
<span id="cb183-32"><a href="source-code.html#cb183-32" tabindex="-1"></a>  Grange_list_cytoband <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb183-33"><a href="source-code.html#cb183-33" tabindex="-1"></a>  Grange_list_pq_affected <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb183-34"><a href="source-code.html#cb183-34" tabindex="-1"></a>  </span>
<span id="cb183-35"><a href="source-code.html#cb183-35" tabindex="-1"></a>  <span class="co"># return empty Grange and empty data.frame if Grange is empty</span></span>
<span id="cb183-36"><a href="source-code.html#cb183-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(Grange) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb183-37"><a href="source-code.html#cb183-37" tabindex="-1"></a>    <span class="co"># return empty Grange</span></span>
<span id="cb183-38"><a href="source-code.html#cb183-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="fu">GRanges</span>(</span>
<span id="cb183-39"><a href="source-code.html#cb183-39" tabindex="-1"></a>      <span class="at">seqnames =</span> <span class="fu">character</span>(<span class="dv">0</span>),</span>
<span id="cb183-40"><a href="source-code.html#cb183-40" tabindex="-1"></a>      <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">integer</span>(<span class="dv">0</span>), <span class="at">end =</span> <span class="fu">integer</span>(<span class="dv">0</span>)),</span>
<span id="cb183-41"><a href="source-code.html#cb183-41" tabindex="-1"></a>      <span class="at">cytobands_cov =</span>  <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb183-42"><a href="source-code.html#cb183-42" tabindex="-1"></a>    ),</span>
<span id="cb183-43"><a href="source-code.html#cb183-43" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">seqnames =</span> <span class="fu">character</span>(), </span>
<span id="cb183-44"><a href="source-code.html#cb183-44" tabindex="-1"></a>               <span class="at">start_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb183-45"><a href="source-code.html#cb183-45" tabindex="-1"></a>               <span class="at">end_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb183-46"><a href="source-code.html#cb183-46" tabindex="-1"></a>               <span class="at">width_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb183-47"><a href="source-code.html#cb183-47" tabindex="-1"></a>               <span class="at">strand =</span> <span class="fu">character</span>(), </span>
<span id="cb183-48"><a href="source-code.html#cb183-48" tabindex="-1"></a>               <span class="at">chromosome =</span> <span class="fu">character</span>(), </span>
<span id="cb183-49"><a href="source-code.html#cb183-49" tabindex="-1"></a>               <span class="at">arm =</span> <span class="fu">character</span>(), </span>
<span id="cb183-50"><a href="source-code.html#cb183-50" tabindex="-1"></a>               <span class="at">percentage_arm_affected =</span> <span class="fu">numeric</span>(), </span>
<span id="cb183-51"><a href="source-code.html#cb183-51" tabindex="-1"></a>               <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb183-52"><a href="source-code.html#cb183-52" tabindex="-1"></a>    ))</span>
<span id="cb183-53"><a href="source-code.html#cb183-53" tabindex="-1"></a>  }</span>
<span id="cb183-54"><a href="source-code.html#cb183-54" tabindex="-1"></a>  </span>
<span id="cb183-55"><a href="source-code.html#cb183-55" tabindex="-1"></a>  <span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Grange)) {</span>
<span id="cb183-56"><a href="source-code.html#cb183-56" tabindex="-1"></a>    for_range <span class="ot">&lt;-</span> Grange[row]</span>
<span id="cb183-57"><a href="source-code.html#cb183-57" tabindex="-1"></a>    </span>
<span id="cb183-58"><a href="source-code.html#cb183-58" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb183-59"><a href="source-code.html#cb183-59" tabindex="-1"></a>    <span class="co"># Part 1 #</span></span>
<span id="cb183-60"><a href="source-code.html#cb183-60" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb183-61"><a href="source-code.html#cb183-61" tabindex="-1"></a>    <span class="do">## First find specific cytoband locations</span></span>
<span id="cb183-62"><a href="source-code.html#cb183-62" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb183-63"><a href="source-code.html#cb183-63" tabindex="-1"></a>    hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, hg38IdeogramCyto)</span>
<span id="cb183-64"><a href="source-code.html#cb183-64" tabindex="-1"></a>    </span>
<span id="cb183-65"><a href="source-code.html#cb183-65" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb183-66"><a href="source-code.html#cb183-66" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb183-67"><a href="source-code.html#cb183-67" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits)], hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb183-68"><a href="source-code.html#cb183-68" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb183-69"><a href="source-code.html#cb183-69" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb183-70"><a href="source-code.html#cb183-70" tabindex="-1"></a>    </span>
<span id="cb183-71"><a href="source-code.html#cb183-71" tabindex="-1"></a>    <span class="co"># extracting cytobands</span></span>
<span id="cb183-72"><a href="source-code.html#cb183-72" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb183-73"><a href="source-code.html#cb183-73" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="fu">seqnames</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]), hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]<span class="sc">$</span>name)</span>
<span id="cb183-74"><a href="source-code.html#cb183-74" tabindex="-1"></a>    <span class="co"># replace &#39;chr&#39;</span></span>
<span id="cb183-75"><a href="source-code.html#cb183-75" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb183-76"><a href="source-code.html#cb183-76" tabindex="-1"></a>      <span class="fu">str_replace</span>(cytobands, <span class="at">pattern =</span> <span class="st">&#39;chr&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb183-77"><a href="source-code.html#cb183-77" tabindex="-1"></a>    </span>
<span id="cb183-78"><a href="source-code.html#cb183-78" tabindex="-1"></a>    cytobands_cov <span class="ot">&lt;-</span></span>
<span id="cb183-79"><a href="source-code.html#cb183-79" tabindex="-1"></a>      <span class="fu">list</span>(<span class="fu">paste0</span>(cytobands, <span class="st">&#39;;&#39;</span>, <span class="fu">round</span>(percentOverlap, <span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39;%&#39;</span>))</span>
<span id="cb183-80"><a href="source-code.html#cb183-80" tabindex="-1"></a>    </span>
<span id="cb183-81"><a href="source-code.html#cb183-81" tabindex="-1"></a>    for_range<span class="sc">$</span>cytobands_cov <span class="ot">&lt;-</span> cytobands_cov</span>
<span id="cb183-82"><a href="source-code.html#cb183-82" tabindex="-1"></a>    </span>
<span id="cb183-83"><a href="source-code.html#cb183-83" tabindex="-1"></a>    Grange_list_cytoband[[row]] <span class="ot">&lt;-</span> for_range</span>
<span id="cb183-84"><a href="source-code.html#cb183-84" tabindex="-1"></a>    </span>
<span id="cb183-85"><a href="source-code.html#cb183-85" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb183-86"><a href="source-code.html#cb183-86" tabindex="-1"></a>    <span class="co"># Part 2 #</span></span>
<span id="cb183-87"><a href="source-code.html#cb183-87" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb183-88"><a href="source-code.html#cb183-88" tabindex="-1"></a>    <span class="co"># make dataframe that shows how much of an arm is affected </span></span>
<span id="cb183-89"><a href="source-code.html#cb183-89" tabindex="-1"></a>    </span>
<span id="cb183-90"><a href="source-code.html#cb183-90" tabindex="-1"></a>    <span class="co"># Grange with p and q IRanges</span></span>
<span id="cb183-91"><a href="source-code.html#cb183-91" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> hg38IdeogramCyto</span>
<span id="cb183-92"><a href="source-code.html#cb183-92" tabindex="-1"></a>    </span>
<span id="cb183-93"><a href="source-code.html#cb183-93" tabindex="-1"></a>    p_q<span class="sc">$</span>arm <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(p_q<span class="sc">$</span>name, <span class="st">&#39;p|q&#39;</span>)</span>
<span id="cb183-94"><a href="source-code.html#cb183-94" tabindex="-1"></a>    </span>
<span id="cb183-95"><a href="source-code.html#cb183-95" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> <span class="fu">subset</span>(p_q, seqnames <span class="sc">%in%</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">&#39;X&#39;</span>, <span class="st">&#39;Y&#39;</span>))</span>
<span id="cb183-96"><a href="source-code.html#cb183-96" tabindex="-1"></a>    </span>
<span id="cb183-97"><a href="source-code.html#cb183-97" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> p_q <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames, arm) <span class="sc">%&gt;%</span> <span class="fu">reduce_ranges</span>()</span>
<span id="cb183-98"><a href="source-code.html#cb183-98" tabindex="-1"></a>    </span>
<span id="cb183-99"><a href="source-code.html#cb183-99" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb183-100"><a href="source-code.html#cb183-100" tabindex="-1"></a>    hits_pq <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, p_q)</span>
<span id="cb183-101"><a href="source-code.html#cb183-101" tabindex="-1"></a>    </span>
<span id="cb183-102"><a href="source-code.html#cb183-102" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb183-103"><a href="source-code.html#cb183-103" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb183-104"><a href="source-code.html#cb183-104" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits_pq)], p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb183-105"><a href="source-code.html#cb183-105" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb183-106"><a href="source-code.html#cb183-106" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb183-107"><a href="source-code.html#cb183-107" tabindex="-1"></a>    </span>
<span id="cb183-108"><a href="source-code.html#cb183-108" tabindex="-1"></a>    p_q_percentage <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb183-109"><a href="source-code.html#cb183-109" tabindex="-1"></a>    </span>
<span id="cb183-110"><a href="source-code.html#cb183-110" tabindex="-1"></a>    p_q_percentage<span class="sc">$</span>percentage_arm_affected <span class="ot">&lt;-</span> percentOverlap <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb183-111"><a href="source-code.html#cb183-111" tabindex="-1"></a>    </span>
<span id="cb183-112"><a href="source-code.html#cb183-112" tabindex="-1"></a>    <span class="fu">colnames</span>(p_q_percentage) <span class="ot">&lt;-</span></span>
<span id="cb183-113"><a href="source-code.html#cb183-113" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb183-114"><a href="source-code.html#cb183-114" tabindex="-1"></a>        <span class="st">&#39;seqnames&#39;</span>,</span>
<span id="cb183-115"><a href="source-code.html#cb183-115" tabindex="-1"></a>        <span class="st">&#39;start_of_arm&#39;</span>,</span>
<span id="cb183-116"><a href="source-code.html#cb183-116" tabindex="-1"></a>        <span class="st">&#39;end_of_arm&#39;</span>,</span>
<span id="cb183-117"><a href="source-code.html#cb183-117" tabindex="-1"></a>        <span class="st">&#39;width_of_arm&#39;</span>,</span>
<span id="cb183-118"><a href="source-code.html#cb183-118" tabindex="-1"></a>        <span class="st">&#39;strand&#39;</span>,</span>
<span id="cb183-119"><a href="source-code.html#cb183-119" tabindex="-1"></a>        <span class="st">&#39;chromosome&#39;</span>,</span>
<span id="cb183-120"><a href="source-code.html#cb183-120" tabindex="-1"></a>        <span class="st">&#39;arm&#39;</span>,</span>
<span id="cb183-121"><a href="source-code.html#cb183-121" tabindex="-1"></a>        <span class="st">&#39;percentage_arm_affected&#39;</span></span>
<span id="cb183-122"><a href="source-code.html#cb183-122" tabindex="-1"></a>      )</span>
<span id="cb183-123"><a href="source-code.html#cb183-123" tabindex="-1"></a>    </span>
<span id="cb183-124"><a href="source-code.html#cb183-124" tabindex="-1"></a>    </span>
<span id="cb183-125"><a href="source-code.html#cb183-125" tabindex="-1"></a>    </span>
<span id="cb183-126"><a href="source-code.html#cb183-126" tabindex="-1"></a>    </span>
<span id="cb183-127"><a href="source-code.html#cb183-127" tabindex="-1"></a>    Grange_list_pq_affected[[row]] <span class="ot">&lt;-</span> p_q_percentage</span>
<span id="cb183-128"><a href="source-code.html#cb183-128" tabindex="-1"></a>    </span>
<span id="cb183-129"><a href="source-code.html#cb183-129" tabindex="-1"></a>    </span>
<span id="cb183-130"><a href="source-code.html#cb183-130" tabindex="-1"></a>  }</span>
<span id="cb183-131"><a href="source-code.html#cb183-131" tabindex="-1"></a>  </span>
<span id="cb183-132"><a href="source-code.html#cb183-132" tabindex="-1"></a>  Grange <span class="ot">&lt;-</span> <span class="fu">bind_ranges</span>(Grange_list_cytoband)</span>
<span id="cb183-133"><a href="source-code.html#cb183-133" tabindex="-1"></a>  pq_arm_affect <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(Grange_list_pq_affected)</span>
<span id="cb183-134"><a href="source-code.html#cb183-134" tabindex="-1"></a>  </span>
<span id="cb183-135"><a href="source-code.html#cb183-135" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(Grange, pq_arm_affect))</span>
<span id="cb183-136"><a href="source-code.html#cb183-136" tabindex="-1"></a>  </span>
<span id="cb183-137"><a href="source-code.html#cb183-137" tabindex="-1"></a>  <span class="do">### </span><span class="re">END</span><span class="do"> of function </span><span class="al">###</span></span>
<span id="cb183-138"><a href="source-code.html#cb183-138" tabindex="-1"></a>  </span>
<span id="cb183-139"><a href="source-code.html#cb183-139" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_genes-script-get_genes-functoin" class="section level2 hasAnchor" number="6.12">
<h2><span class="header-section-number">6.12</span> get_genes script: ‘get_genes’ functoin<a href="source-code.html#get_genes-script-get_genes-functoin" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="source-code.html#cb184-1" tabindex="-1"></a>get_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) { </span>
<span id="cb184-2"><a href="source-code.html#cb184-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb184-3"><a href="source-code.html#cb184-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb184-4"><a href="source-code.html#cb184-4" tabindex="-1"></a>  </span>
<span id="cb184-5"><a href="source-code.html#cb184-5" tabindex="-1"></a>  <span class="fu">require</span>(GenomicRanges)</span>
<span id="cb184-6"><a href="source-code.html#cb184-6" tabindex="-1"></a>  <span class="fu">require</span>(biomaRt)</span>
<span id="cb184-7"><a href="source-code.html#cb184-7" tabindex="-1"></a>  </span>
<span id="cb184-8"><a href="source-code.html#cb184-8" tabindex="-1"></a>  <span class="co"># This code should be run only once and is therefore commented. </span></span>
<span id="cb184-9"><a href="source-code.html#cb184-9" tabindex="-1"></a>  <span class="co"># code from: https://www.biostars.org/p/311199/</span></span>
<span id="cb184-10"><a href="source-code.html#cb184-10" tabindex="-1"></a>  </span>
<span id="cb184-11"><a href="source-code.html#cb184-11" tabindex="-1"></a>  <span class="co"># Set up an gene annotation template to use</span></span>
<span id="cb184-12"><a href="source-code.html#cb184-12" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb184-13"><a href="source-code.html#cb184-13" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, host=&quot;www.ensembl.org&quot;, path=&quot;/biomart/martservice&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb184-14"><a href="source-code.html#cb184-14" tabindex="-1"></a>  <span class="co"># genes &lt;- getBM(attributes=c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;), mart=mart)</span></span>
<span id="cb184-15"><a href="source-code.html#cb184-15" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[genes[,1]!=&quot;&quot; &amp; genes[,2] %in% c(1:22,&quot;X&quot;,&quot;Y&quot;),]</span></span>
<span id="cb184-16"><a href="source-code.html#cb184-16" tabindex="-1"></a>  <span class="co"># xidx &lt;- which(genes[,2]==&quot;X&quot;)</span></span>
<span id="cb184-17"><a href="source-code.html#cb184-17" tabindex="-1"></a>  <span class="co"># yidx &lt;- which(genes[,2]==&quot;Y&quot;)</span></span>
<span id="cb184-18"><a href="source-code.html#cb184-18" tabindex="-1"></a>  <span class="co"># genes[xidx, 2] &lt;- 23</span></span>
<span id="cb184-19"><a href="source-code.html#cb184-19" tabindex="-1"></a>  <span class="co"># genes[yidx, 2] &lt;- 24</span></span>
<span id="cb184-20"><a href="source-code.html#cb184-20" tabindex="-1"></a>  <span class="co"># genes[,2] &lt;- sapply(genes[,2],as.integer)</span></span>
<span id="cb184-21"><a href="source-code.html#cb184-21" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,3]),]</span></span>
<span id="cb184-22"><a href="source-code.html#cb184-22" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,2]),]</span></span>
<span id="cb184-23"><a href="source-code.html#cb184-23" tabindex="-1"></a>  <span class="co"># save(genes, file = &#39;rda/cna_analysis/genes.rda&#39;)</span></span>
<span id="cb184-24"><a href="source-code.html#cb184-24" tabindex="-1"></a>  <span class="co"># colnames(genes) &lt;- c(&quot;GeneSymbol&quot;,&quot;Chr&quot;,&quot;Start&quot;,&quot;End&quot;)</span></span>
<span id="cb184-25"><a href="source-code.html#cb184-25" tabindex="-1"></a>  <span class="co"># genes_GR &lt;- makeGRangesFromDataFrame(genes,keep.extra.columns = TRUE)</span></span>
<span id="cb184-26"><a href="source-code.html#cb184-26" tabindex="-1"></a>  <span class="co"># save(genes_GR, file = &#39;rda/cna_analysis/genes_GR.rda&#39;)</span></span>
<span id="cb184-27"><a href="source-code.html#cb184-27" tabindex="-1"></a>  </span>
<span id="cb184-28"><a href="source-code.html#cb184-28" tabindex="-1"></a>  <span class="co"># loading genelist and genomic location</span></span>
<span id="cb184-29"><a href="source-code.html#cb184-29" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&quot;../cna_analysis/rda/cna_analysis/genes_GR.rda&quot;</span>, <span class="at">envir =</span> .GlobalEnv)</span>
<span id="cb184-30"><a href="source-code.html#cb184-30" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/genes.rda&#39;</span>, <span class="at">envir =</span> .GlobalEnv )</span>
<span id="cb184-31"><a href="source-code.html#cb184-31" tabindex="-1"></a>  </span>
<span id="cb184-32"><a href="source-code.html#cb184-32" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb184-33"><a href="source-code.html#cb184-33" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb184-34"><a href="source-code.html#cb184-34" tabindex="-1"></a>  GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="fu">paste0</span>(<span class="st">&#39;chr&#39;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">23</span>)))</span>
<span id="cb184-35"><a href="source-code.html#cb184-35" tabindex="-1"></a>  </span>
<span id="cb184-36"><a href="source-code.html#cb184-36" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(genes_GR) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb184-37"><a href="source-code.html#cb184-37" tabindex="-1"></a>  </span>
<span id="cb184-38"><a href="source-code.html#cb184-38" tabindex="-1"></a>  <span class="co"># Finding hits with genes_GR and GRanges</span></span>
<span id="cb184-39"><a href="source-code.html#cb184-39" tabindex="-1"></a>  hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(genes_GR, GRange, <span class="at">type=</span><span class="st">&quot;within&quot;</span>)</span>
<span id="cb184-40"><a href="source-code.html#cb184-40" tabindex="-1"></a>  </span>
<span id="cb184-41"><a href="source-code.html#cb184-41" tabindex="-1"></a>  <span class="co"># constructing</span></span>
<span id="cb184-42"><a href="source-code.html#cb184-42" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(GRange)[<span class="fu">subjectHits</span>(hits),],genes[<span class="fu">queryHits</span>(hits),])</span>
<span id="cb184-43"><a href="source-code.html#cb184-43" tabindex="-1"></a>  </span>
<span id="cb184-44"><a href="source-code.html#cb184-44" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb184-45"><a href="source-code.html#cb184-45" tabindex="-1"></a>  </span>
<span id="cb184-46"><a href="source-code.html#cb184-46" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_oncogenes-scripts-get_oncogenes-function" class="section level2 hasAnchor" number="6.13">
<h2><span class="header-section-number">6.13</span> get_oncogenes scripts: ‘get_oncogenes’ function<a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="source-code.html#cb185-1" tabindex="-1"></a>get_oncogenes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) {</span>
<span id="cb185-2"><a href="source-code.html#cb185-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb185-3"><a href="source-code.html#cb185-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb185-4"><a href="source-code.html#cb185-4" tabindex="-1"></a>  <span class="fu">source</span>(<span class="st">&#39;R/get_genes.R&#39;</span>)</span>
<span id="cb185-5"><a href="source-code.html#cb185-5" tabindex="-1"></a>  </span>
<span id="cb185-6"><a href="source-code.html#cb185-6" tabindex="-1"></a>  <span class="co"># first finding all genes within GRange</span></span>
<span id="cb185-7"><a href="source-code.html#cb185-7" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">get_genes</span>(GRange) </span>
<span id="cb185-8"><a href="source-code.html#cb185-8" tabindex="-1"></a>  </span>
<span id="cb185-9"><a href="source-code.html#cb185-9" tabindex="-1"></a>  <span class="co"># change colnames so dataframes match</span></span>
<span id="cb185-10"><a href="source-code.html#cb185-10" tabindex="-1"></a>  <span class="fu">colnames</span>(all_genes)[<span class="fu">which</span>(<span class="fu">names</span>(all_genes) <span class="sc">==</span> <span class="st">&#39;hgnc_symbol&#39;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Gene Symbol&quot;</span>   </span>
<span id="cb185-11"><a href="source-code.html#cb185-11" tabindex="-1"></a>  </span>
<span id="cb185-12"><a href="source-code.html#cb185-12" tabindex="-1"></a>  <span class="co"># downloading oncogenes</span></span>
<span id="cb185-13"><a href="source-code.html#cb185-13" tabindex="-1"></a>  <span class="fu">require</span>(readr)</span>
<span id="cb185-14"><a href="source-code.html#cb185-14" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../cna_analysis/data/cna_analysis/Census_allMon May 30 12_43_01 2022.csv&quot;</span>)</span>
<span id="cb185-15"><a href="source-code.html#cb185-15" tabindex="-1"></a>  </span>
<span id="cb185-16"><a href="source-code.html#cb185-16" tabindex="-1"></a>  </span>
<span id="cb185-17"><a href="source-code.html#cb185-17" tabindex="-1"></a>  <span class="co"># subset df_genes so to only have oncogenes</span></span>
<span id="cb185-18"><a href="source-code.html#cb185-18" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb185-19"><a href="source-code.html#cb185-19" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(oncogenes, all_genes)</span>
<span id="cb185-20"><a href="source-code.html#cb185-20" tabindex="-1"></a>  </span>
<span id="cb185-21"><a href="source-code.html#cb185-21" tabindex="-1"></a>  <span class="fu">return</span>(oncogenes)</span>
<span id="cb185-22"><a href="source-code.html#cb185-22" tabindex="-1"></a>  </span>
<span id="cb185-23"><a href="source-code.html#cb185-23" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genes_karyotype-script-plot_genes_karyotype-function" class="section level2 hasAnchor" number="6.14">
<h2><span class="header-section-number">6.14</span> plot_genes_karyotype script: ‘plot_genes_karyotype’ function<a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="source-code.html#cb186-1" tabindex="-1"></a>plot_genes_karyotype <span class="ot">&lt;-</span></span>
<span id="cb186-2"><a href="source-code.html#cb186-2" tabindex="-1"></a>  <span class="cf">function</span>(GRange,</span>
<span id="cb186-3"><a href="source-code.html#cb186-3" tabindex="-1"></a>           <span class="at">dist =</span> <span class="sc">-</span><span class="dv">30</span>,</span>
<span id="cb186-4"><a href="source-code.html#cb186-4" tabindex="-1"></a>           show_genes,</span>
<span id="cb186-5"><a href="source-code.html#cb186-5" tabindex="-1"></a>           <span class="at">show_chromosomes =</span> <span class="fu">c</span>(</span>
<span id="cb186-6"><a href="source-code.html#cb186-6" tabindex="-1"></a>             <span class="st">&quot;chr1&quot;</span>,</span>
<span id="cb186-7"><a href="source-code.html#cb186-7" tabindex="-1"></a>             <span class="st">&quot;chr2&quot;</span>,</span>
<span id="cb186-8"><a href="source-code.html#cb186-8" tabindex="-1"></a>             <span class="st">&quot;chr3&quot;</span>,</span>
<span id="cb186-9"><a href="source-code.html#cb186-9" tabindex="-1"></a>             <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb186-10"><a href="source-code.html#cb186-10" tabindex="-1"></a>             <span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb186-11"><a href="source-code.html#cb186-11" tabindex="-1"></a>             <span class="st">&quot;chr6&quot;</span>,</span>
<span id="cb186-12"><a href="source-code.html#cb186-12" tabindex="-1"></a>             <span class="st">&quot;chr7&quot;</span>,</span>
<span id="cb186-13"><a href="source-code.html#cb186-13" tabindex="-1"></a>             <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb186-14"><a href="source-code.html#cb186-14" tabindex="-1"></a>             <span class="st">&quot;chr9&quot;</span>,</span>
<span id="cb186-15"><a href="source-code.html#cb186-15" tabindex="-1"></a>             <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb186-16"><a href="source-code.html#cb186-16" tabindex="-1"></a>             <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb186-17"><a href="source-code.html#cb186-17" tabindex="-1"></a>             <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb186-18"><a href="source-code.html#cb186-18" tabindex="-1"></a>             <span class="st">&quot;chr13&quot;</span>,</span>
<span id="cb186-19"><a href="source-code.html#cb186-19" tabindex="-1"></a>             <span class="st">&quot;chr14&quot;</span>,</span>
<span id="cb186-20"><a href="source-code.html#cb186-20" tabindex="-1"></a>             <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb186-21"><a href="source-code.html#cb186-21" tabindex="-1"></a>             <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb186-22"><a href="source-code.html#cb186-22" tabindex="-1"></a>             <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb186-23"><a href="source-code.html#cb186-23" tabindex="-1"></a>             <span class="st">&quot;chr18&quot;</span>,</span>
<span id="cb186-24"><a href="source-code.html#cb186-24" tabindex="-1"></a>             <span class="st">&quot;chr19&quot;</span>,</span>
<span id="cb186-25"><a href="source-code.html#cb186-25" tabindex="-1"></a>             <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb186-26"><a href="source-code.html#cb186-26" tabindex="-1"></a>             <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb186-27"><a href="source-code.html#cb186-27" tabindex="-1"></a>             <span class="st">&quot;chr22&quot;</span>,</span>
<span id="cb186-28"><a href="source-code.html#cb186-28" tabindex="-1"></a>             <span class="st">&quot;chrX&quot;</span></span>
<span id="cb186-29"><a href="source-code.html#cb186-29" tabindex="-1"></a>           )) {</span>
<span id="cb186-30"><a href="source-code.html#cb186-30" tabindex="-1"></a>    <span class="co">#&#39; @param GRange a GRange with seqnames, ranges, strand and hgnc_symbol</span></span>
<span id="cb186-31"><a href="source-code.html#cb186-31" tabindex="-1"></a>    <span class="co">#&#39; @param dist How far away the genes should be plotted from the lines.</span></span>
<span id="cb186-32"><a href="source-code.html#cb186-32" tabindex="-1"></a>    <span class="co">#&#39; Between -40 and -70 seems to do the trick most of the times.</span></span>
<span id="cb186-33"><a href="source-code.html#cb186-33" tabindex="-1"></a>    <span class="co">#&#39; @param show_genes is a vector of gene symbols to show.</span></span>
<span id="cb186-34"><a href="source-code.html#cb186-34" tabindex="-1"></a>    <span class="co">#&#39; @param show_chromosomes is a vector indicating which chromosomes to show.</span></span>
<span id="cb186-35"><a href="source-code.html#cb186-35" tabindex="-1"></a>    <span class="co">#&#39; @return a karyotype with genes labeled</span></span>
<span id="cb186-36"><a href="source-code.html#cb186-36" tabindex="-1"></a>    <span class="co">#&#39; @details. Throws an error when the GRange does not contain either</span></span>
<span id="cb186-37"><a href="source-code.html#cb186-37" tabindex="-1"></a>    <span class="co">#&#39; ampflications or deletions. This can be ignored, the resulting plot</span></span>
<span id="cb186-38"><a href="source-code.html#cb186-38" tabindex="-1"></a>    <span class="co">#&#39; is correct</span></span>
<span id="cb186-39"><a href="source-code.html#cb186-39" tabindex="-1"></a>    </span>
<span id="cb186-40"><a href="source-code.html#cb186-40" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb186-41"><a href="source-code.html#cb186-41" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb186-42"><a href="source-code.html#cb186-42" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb186-43"><a href="source-code.html#cb186-43" tabindex="-1"></a>    <span class="fu">require</span>(karyoploteR)</span>
<span id="cb186-44"><a href="source-code.html#cb186-44" tabindex="-1"></a>    <span class="fu">require</span>(stringr)</span>
<span id="cb186-45"><a href="source-code.html#cb186-45" tabindex="-1"></a>    </span>
<span id="cb186-46"><a href="source-code.html#cb186-46" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb186-47"><a href="source-code.html#cb186-47" tabindex="-1"></a>    <span class="co"># Wrangle #</span></span>
<span id="cb186-48"><a href="source-code.html#cb186-48" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb186-49"><a href="source-code.html#cb186-49" tabindex="-1"></a>    <span class="co"># making sure seqlevels match</span></span>
<span id="cb186-50"><a href="source-code.html#cb186-50" tabindex="-1"></a>    <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&#39;UCSC&#39;</span></span>
<span id="cb186-51"><a href="source-code.html#cb186-51" tabindex="-1"></a>    </span>
<span id="cb186-52"><a href="source-code.html#cb186-52" tabindex="-1"></a>    kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(</span>
<span id="cb186-53"><a href="source-code.html#cb186-53" tabindex="-1"></a>      <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb186-54"><a href="source-code.html#cb186-54" tabindex="-1"></a>      <span class="at">plot.type =</span> <span class="dv">5</span>,</span>
<span id="cb186-55"><a href="source-code.html#cb186-55" tabindex="-1"></a>      <span class="at">labels.plotter =</span> <span class="cn">NULL</span>,</span>
<span id="cb186-56"><a href="source-code.html#cb186-56" tabindex="-1"></a>      <span class="at">main =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb186-57"><a href="source-code.html#cb186-57" tabindex="-1"></a>      <span class="at">cex =</span> <span class="dv">4</span>,</span>
<span id="cb186-58"><a href="source-code.html#cb186-58" tabindex="-1"></a>      <span class="at">chromosomes =</span> show_chromosomes</span>
<span id="cb186-59"><a href="source-code.html#cb186-59" tabindex="-1"></a>    )</span>
<span id="cb186-60"><a href="source-code.html#cb186-60" tabindex="-1"></a>    </span>
<span id="cb186-61"><a href="source-code.html#cb186-61" tabindex="-1"></a>    <span class="co">#renaming seqlevels</span></span>
<span id="cb186-62"><a href="source-code.html#cb186-62" tabindex="-1"></a>    seqlevels_grange_change <span class="ot">&lt;-</span></span>
<span id="cb186-63"><a href="source-code.html#cb186-63" tabindex="-1"></a>      <span class="fu">str_replace_all</span>(<span class="fu">seqlevels</span>(GRange),</span>
<span id="cb186-64"><a href="source-code.html#cb186-64" tabindex="-1"></a>                      <span class="at">pattern =</span> <span class="st">&#39;chr23&#39;</span>,</span>
<span id="cb186-65"><a href="source-code.html#cb186-65" tabindex="-1"></a>                      <span class="at">replacement =</span> <span class="st">&#39;chrX&#39;</span>)</span>
<span id="cb186-66"><a href="source-code.html#cb186-66" tabindex="-1"></a>    </span>
<span id="cb186-67"><a href="source-code.html#cb186-67" tabindex="-1"></a>    GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="at">value =</span> seqlevels_grange_change)</span>
<span id="cb186-68"><a href="source-code.html#cb186-68" tabindex="-1"></a>    </span>
<span id="cb186-69"><a href="source-code.html#cb186-69" tabindex="-1"></a>    <span class="fu">kpAddChromosomeNames</span>(</span>
<span id="cb186-70"><a href="source-code.html#cb186-70" tabindex="-1"></a>      kp,</span>
<span id="cb186-71"><a href="source-code.html#cb186-71" tabindex="-1"></a>      <span class="at">chr.names =</span> <span class="fu">str_remove_all</span>(show_chromosomes, <span class="st">&#39;chr&#39;</span>),</span>
<span id="cb186-72"><a href="source-code.html#cb186-72" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.3</span></span>
<span id="cb186-73"><a href="source-code.html#cb186-73" tabindex="-1"></a>    )</span>
<span id="cb186-74"><a href="source-code.html#cb186-74" tabindex="-1"></a>    </span>
<span id="cb186-75"><a href="source-code.html#cb186-75" tabindex="-1"></a>    <span class="co"># split into amplifications and deletions</span></span>
<span id="cb186-76"><a href="source-code.html#cb186-76" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;amplified&#39;</span>]</span>
<span id="cb186-77"><a href="source-code.html#cb186-77" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> ampl[ampl<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb186-78"><a href="source-code.html#cb186-78" tabindex="-1"></a>    </span>
<span id="cb186-79"><a href="source-code.html#cb186-79" tabindex="-1"></a>    del <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;deleted&#39;</span>]</span>
<span id="cb186-80"><a href="source-code.html#cb186-80" tabindex="-1"></a>    del <span class="ot">&lt;-</span> del[del<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb186-81"><a href="source-code.html#cb186-81" tabindex="-1"></a>    </span>
<span id="cb186-82"><a href="source-code.html#cb186-82" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb186-83"><a href="source-code.html#cb186-83" tabindex="-1"></a>      cor <span class="ot">&lt;-</span></span>
<span id="cb186-84"><a href="source-code.html#cb186-84" tabindex="-1"></a>        <span class="fu">cor.test</span>(df<span class="sc">$</span>AUC, df<span class="sc">$</span>dependency, <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>),</span>
<span id="cb186-85"><a href="source-code.html#cb186-85" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb186-86"><a href="source-code.html#cb186-86" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb186-87"><a href="source-code.html#cb186-87" tabindex="-1"></a>    )</span>
<span id="cb186-88"><a href="source-code.html#cb186-88" tabindex="-1"></a>    </span>
<span id="cb186-89"><a href="source-code.html#cb186-89" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb186-90"><a href="source-code.html#cb186-90" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb186-91"><a href="source-code.html#cb186-91" tabindex="-1"></a>        kp,</span>
<span id="cb186-92"><a href="source-code.html#cb186-92" tabindex="-1"></a>        <span class="at">data =</span> ampl,</span>
<span id="cb186-93"><a href="source-code.html#cb186-93" tabindex="-1"></a>        <span class="at">labels =</span> ampl<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb186-94"><a href="source-code.html#cb186-94" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb186-95"><a href="source-code.html#cb186-95" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb186-96"><a href="source-code.html#cb186-96" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.85</span>,</span>
<span id="cb186-97"><a href="source-code.html#cb186-97" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb186-98"><a href="source-code.html#cb186-98" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb186-99"><a href="source-code.html#cb186-99" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb186-100"><a href="source-code.html#cb186-100" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;firebrick&#39;</span>,</span>
<span id="cb186-101"><a href="source-code.html#cb186-101" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;firebrick&#39;</span></span>
<span id="cb186-102"><a href="source-code.html#cb186-102" tabindex="-1"></a>      ),</span>
<span id="cb186-103"><a href="source-code.html#cb186-103" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb186-104"><a href="source-code.html#cb186-104" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb186-105"><a href="source-code.html#cb186-105" tabindex="-1"></a>    )</span>
<span id="cb186-106"><a href="source-code.html#cb186-106" tabindex="-1"></a>    </span>
<span id="cb186-107"><a href="source-code.html#cb186-107" tabindex="-1"></a>    </span>
<span id="cb186-108"><a href="source-code.html#cb186-108" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb186-109"><a href="source-code.html#cb186-109" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb186-110"><a href="source-code.html#cb186-110" tabindex="-1"></a>        kp,</span>
<span id="cb186-111"><a href="source-code.html#cb186-111" tabindex="-1"></a>        <span class="at">data =</span> del,</span>
<span id="cb186-112"><a href="source-code.html#cb186-112" tabindex="-1"></a>        <span class="at">labels =</span> del<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb186-113"><a href="source-code.html#cb186-113" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb186-114"><a href="source-code.html#cb186-114" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb186-115"><a href="source-code.html#cb186-115" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.4</span>,</span>
<span id="cb186-116"><a href="source-code.html#cb186-116" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb186-117"><a href="source-code.html#cb186-117" tabindex="-1"></a>        <span class="at">cex =</span> <span class="dv">8</span>,</span>
<span id="cb186-118"><a href="source-code.html#cb186-118" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb186-119"><a href="source-code.html#cb186-119" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb186-120"><a href="source-code.html#cb186-120" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;steelblue&#39;</span>,</span>
<span id="cb186-121"><a href="source-code.html#cb186-121" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;steelblue&#39;</span></span>
<span id="cb186-122"><a href="source-code.html#cb186-122" tabindex="-1"></a>      ),</span>
<span id="cb186-123"><a href="source-code.html#cb186-123" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb186-124"><a href="source-code.html#cb186-124" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb186-125"><a href="source-code.html#cb186-125" tabindex="-1"></a>    )</span>
<span id="cb186-126"><a href="source-code.html#cb186-126" tabindex="-1"></a>    </span>
<span id="cb186-127"><a href="source-code.html#cb186-127" tabindex="-1"></a>    </span>
<span id="cb186-128"><a href="source-code.html#cb186-128" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="section level2 hasAnchor" number="6.15">
<h2><span class="header-section-number">6.15</span> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions<a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="section level2 hasAnchor" number="6.16">
<h2><span class="header-section-number">6.16</span> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function<a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘calculate_heterogeneity_score’ function, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="cna_profiler-script-cna_profiler-function" class="section level2 hasAnchor" number="6.17">
<h2><span class="header-section-number">6.17</span> cna_profiler script: ‘cna_profiler’ function<a href="source-code.html#cna_profiler-script-cna_profiler-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="source-code.html#cb187-1" tabindex="-1"></a>cna_profiler <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb187-2"><a href="source-code.html#cb187-2" tabindex="-1"></a>  <span class="co">#&#39; @param df is a datafame containing all the copy number calls of all the files.</span></span>
<span id="cb187-3"><a href="source-code.html#cb187-3" tabindex="-1"></a>  <span class="co">#&#39;                       Columns: 01) seqnames</span></span>
<span id="cb187-4"><a href="source-code.html#cb187-4" tabindex="-1"></a>  <span class="co">#&#39;                                02) start</span></span>
<span id="cb187-5"><a href="source-code.html#cb187-5" tabindex="-1"></a>  <span class="co">#&#39;                                03) end</span></span>
<span id="cb187-6"><a href="source-code.html#cb187-6" tabindex="-1"></a>  <span class="co">#&#39;                                04) width</span></span>
<span id="cb187-7"><a href="source-code.html#cb187-7" tabindex="-1"></a>  <span class="co">#&#39;                                05) strand</span></span>
<span id="cb187-8"><a href="source-code.html#cb187-8" tabindex="-1"></a>  <span class="co">#&#39;                                06) [copy.number cell 1]</span></span>
<span id="cb187-9"><a href="source-code.html#cb187-9" tabindex="-1"></a>  <span class="co">#&#39;                                07) [copy.number cell 2], etc.</span></span>
<span id="cb187-10"><a href="source-code.html#cb187-10" tabindex="-1"></a>  <span class="co">#&#39; @return Outputs a dataframe that states the ratio (normalized to total cells) of</span></span>
<span id="cb187-11"><a href="source-code.html#cb187-11" tabindex="-1"></a>  <span class="co">#&#39; small, medium, large and whole chromosome CNAs</span></span>
<span id="cb187-12"><a href="source-code.html#cb187-12" tabindex="-1"></a></span>
<span id="cb187-13"><a href="source-code.html#cb187-13" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb187-14"><a href="source-code.html#cb187-14" tabindex="-1"></a>  <span class="fu">require</span>(GWASTools)</span>
<span id="cb187-15"><a href="source-code.html#cb187-15" tabindex="-1"></a>  </span>
<span id="cb187-16"><a href="source-code.html#cb187-16" tabindex="-1"></a>  <span class="co"># defining some parameters</span></span>
<span id="cb187-17"><a href="source-code.html#cb187-17" tabindex="-1"></a>  <span class="co"># number of cells </span></span>
<span id="cb187-18"><a href="source-code.html#cb187-18" tabindex="-1"></a>  n_cells <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df) <span class="sc">-</span> <span class="dv">5</span></span>
<span id="cb187-19"><a href="source-code.html#cb187-19" tabindex="-1"></a>  </span>
<span id="cb187-20"><a href="source-code.html#cb187-20" tabindex="-1"></a>  <span class="co"># extract centromere positions</span></span>
<span id="cb187-21"><a href="source-code.html#cb187-21" tabindex="-1"></a>  <span class="fu">data</span>(centromeres.hg38)</span>
<span id="cb187-22"><a href="source-code.html#cb187-22" tabindex="-1"></a>  </span>
<span id="cb187-23"><a href="source-code.html#cb187-23" tabindex="-1"></a>  centromeres.hg38 <span class="ot">&lt;-</span> <span class="fu">subset</span>(centromeres.hg38, chrom <span class="sc">!=</span> <span class="st">&#39;Y&#39;</span>)</span>
<span id="cb187-24"><a href="source-code.html#cb187-24" tabindex="-1"></a>  <span class="fu">names</span>(centromeres.hg38) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>)</span>
<span id="cb187-25"><a href="source-code.html#cb187-25" tabindex="-1"></a>  </span>
<span id="cb187-26"><a href="source-code.html#cb187-26" tabindex="-1"></a>  </span>
<span id="cb187-27"><a href="source-code.html#cb187-27" tabindex="-1"></a>  <span class="co"># extract total length chromosomes</span></span>
<span id="cb187-28"><a href="source-code.html#cb187-28" tabindex="-1"></a>  total_length <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">sum</span>(width))</span>
<span id="cb187-29"><a href="source-code.html#cb187-29" tabindex="-1"></a>  </span>
<span id="cb187-30"><a href="source-code.html#cb187-30" tabindex="-1"></a>  <span class="co"># construct dataframe containing chromosome name, centromere position (start and end) and total length chromosome</span></span>
<span id="cb187-31"><a href="source-code.html#cb187-31" tabindex="-1"></a>  chrom_centro_total <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(total_length, centromeres.hg38)</span>
<span id="cb187-32"><a href="source-code.html#cb187-32" tabindex="-1"></a>  </span>
<span id="cb187-33"><a href="source-code.html#cb187-33" tabindex="-1"></a>  <span class="co"># compute arm lengths</span></span>
<span id="cb187-34"><a href="source-code.html#cb187-34" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>p_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>start <span class="sc">+</span> (chrom_centro_total<span class="sc">$</span>end<span class="sc">-</span>chrom_centro_total<span class="sc">$</span>start)</span>
<span id="cb187-35"><a href="source-code.html#cb187-35" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>q_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span><span class="st">`</span><span class="at">sum(width)</span><span class="st">`</span> <span class="sc">-</span>  chrom_centro_total<span class="sc">$</span>p_arm</span>
<span id="cb187-36"><a href="source-code.html#cb187-36" tabindex="-1"></a>  </span>
<span id="cb187-37"><a href="source-code.html#cb187-37" tabindex="-1"></a>  <span class="do">##### !!!!!!!! ######</span></span>
<span id="cb187-38"><a href="source-code.html#cb187-38" tabindex="-1"></a>  <span class="co"># Check if this is still correct when using new assemblies !</span></span>
<span id="cb187-39"><a href="source-code.html#cb187-39" tabindex="-1"></a>  </span>
<span id="cb187-40"><a href="source-code.html#cb187-40" tabindex="-1"></a>  <span class="co"># now all p_arms except for chromosome 1 are correct (for some reason the p-arm is the larger arm here). </span></span>
<span id="cb187-41"><a href="source-code.html#cb187-41" tabindex="-1"></a>  <span class="co"># I am going to swithc it. </span></span>
<span id="cb187-42"><a href="source-code.html#cb187-42" tabindex="-1"></a>  p_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>q_arm[<span class="dv">1</span>]</span>
<span id="cb187-43"><a href="source-code.html#cb187-43" tabindex="-1"></a>  q_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>p_arm[<span class="dv">1</span>]</span>
<span id="cb187-44"><a href="source-code.html#cb187-44" tabindex="-1"></a>  </span>
<span id="cb187-45"><a href="source-code.html#cb187-45" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">5</span>] <span class="ot">&lt;-</span> p_arm_1</span>
<span id="cb187-46"><a href="source-code.html#cb187-46" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">6</span>] <span class="ot">&lt;-</span> q_arm_1</span>
<span id="cb187-47"><a href="source-code.html#cb187-47" tabindex="-1"></a>  </span>
<span id="cb187-48"><a href="source-code.html#cb187-48" tabindex="-1"></a>  </span>
<span id="cb187-49"><a href="source-code.html#cb187-49" tabindex="-1"></a>  <span class="co"># function to extract CNA for each chromosome, run this function for each chromosome</span></span>
<span id="cb187-50"><a href="source-code.html#cb187-50" tabindex="-1"></a>  cna_extract <span class="ot">&lt;-</span> <span class="cf">function</span>(chromosome) {</span>
<span id="cb187-51"><a href="source-code.html#cb187-51" tabindex="-1"></a>    <span class="co"># @parama: chromosome is the chromosome (i.e., 1, 2,3, X, etc) </span></span>
<span id="cb187-52"><a href="source-code.html#cb187-52" tabindex="-1"></a>    df_chrom <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb187-53"><a href="source-code.html#cb187-53" tabindex="-1"></a>    </span>
<span id="cb187-54"><a href="source-code.html#cb187-54" tabindex="-1"></a>    <span class="co"># run this function for each column in df_chrom</span></span>
<span id="cb187-55"><a href="source-code.html#cb187-55" tabindex="-1"></a>    cna_extract_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(column_i) {</span>
<span id="cb187-56"><a href="source-code.html#cb187-56" tabindex="-1"></a>      <span class="co"># @ param column i is the integer of the column in df_chrom </span></span>
<span id="cb187-57"><a href="source-code.html#cb187-57" tabindex="-1"></a>      <span class="co"># from which to extract CNAs</span></span>
<span id="cb187-58"><a href="source-code.html#cb187-58" tabindex="-1"></a>      </span>
<span id="cb187-59"><a href="source-code.html#cb187-59" tabindex="-1"></a>      <span class="co"># extract rows that switch CNA value (for example from 2 to 3)</span></span>
<span id="cb187-60"><a href="source-code.html#cb187-60" tabindex="-1"></a>      CNA_change <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="fu">tail</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>) <span class="sc">!=</span> <span class="fu">head</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>)))</span>
<span id="cb187-61"><a href="source-code.html#cb187-61" tabindex="-1"></a>      </span>
<span id="cb187-62"><a href="source-code.html#cb187-62" tabindex="-1"></a>      <span class="co"># extract indices</span></span>
<span id="cb187-63"><a href="source-code.html#cb187-63" tabindex="-1"></a>      CNA_change_indices <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">sort</span>(<span class="fu">c</span>(<span class="dv">1</span>,CNA_change, (CNA_change<span class="dv">-1</span>), <span class="fu">nrow</span>(df_chrom))))</span>
<span id="cb187-64"><a href="source-code.html#cb187-64" tabindex="-1"></a>      </span>
<span id="cb187-65"><a href="source-code.html#cb187-65" tabindex="-1"></a>      <span class="co"># construct dataframe containing end and start of each CNA</span></span>
<span id="cb187-66"><a href="source-code.html#cb187-66" tabindex="-1"></a>      CNA_change_df <span class="ot">&lt;-</span> df_chrom[CNA_change_indices,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>, column_i)]</span>
<span id="cb187-67"><a href="source-code.html#cb187-67" tabindex="-1"></a>      <span class="fu">names</span>(CNA_change_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;width&#39;</span>, <span class="st">&#39;strand&#39;</span>, <span class="st">&#39;cna&#39;</span>)</span>
<span id="cb187-68"><a href="source-code.html#cb187-68" tabindex="-1"></a>      </span>
<span id="cb187-69"><a href="source-code.html#cb187-69" tabindex="-1"></a>      start_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb187-70"><a href="source-code.html#cb187-70" tabindex="-1"></a>      end_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb187-71"><a href="source-code.html#cb187-71" tabindex="-1"></a>      </span>
<span id="cb187-72"><a href="source-code.html#cb187-72" tabindex="-1"></a>      i <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb187-73"><a href="source-code.html#cb187-73" tabindex="-1"></a>      </span>
<span id="cb187-74"><a href="source-code.html#cb187-74" tabindex="-1"></a>      <span class="cf">while</span>(i<span class="sc">&lt;=</span><span class="fu">nrow</span>(CNA_change_df)){</span>
<span id="cb187-75"><a href="source-code.html#cb187-75" tabindex="-1"></a>        </span>
<span id="cb187-76"><a href="source-code.html#cb187-76" tabindex="-1"></a>        <span class="fu">try</span>(<span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">==</span> CNA_change_df[i, <span class="dv">6</span>]){</span>
<span id="cb187-77"><a href="source-code.html#cb187-77" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb187-78"><a href="source-code.html#cb187-78" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span>]</span>
<span id="cb187-79"><a href="source-code.html#cb187-79" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">2</span></span>
<span id="cb187-80"><a href="source-code.html#cb187-80" tabindex="-1"></a>        })</span>
<span id="cb187-81"><a href="source-code.html#cb187-81" tabindex="-1"></a>        </span>
<span id="cb187-82"><a href="source-code.html#cb187-82" tabindex="-1"></a>        <span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">!=</span> CNA_change_df[i, <span class="dv">6</span>] <span class="sc">||</span> <span class="fu">is.na</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>])){</span>
<span id="cb187-83"><a href="source-code.html#cb187-83" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb187-84"><a href="source-code.html#cb187-84" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i, <span class="dv">3</span>]</span>
<span id="cb187-85"><a href="source-code.html#cb187-85" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb187-86"><a href="source-code.html#cb187-86" tabindex="-1"></a>        }</span>
<span id="cb187-87"><a href="source-code.html#cb187-87" tabindex="-1"></a>        </span>
<span id="cb187-88"><a href="source-code.html#cb187-88" tabindex="-1"></a>      }</span>
<span id="cb187-89"><a href="source-code.html#cb187-89" tabindex="-1"></a>      </span>
<span id="cb187-90"><a href="source-code.html#cb187-90" tabindex="-1"></a>      <span class="co"># constructing a dataframe</span></span>
<span id="cb187-91"><a href="source-code.html#cb187-91" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">seqnames =</span> chromosome, </span>
<span id="cb187-92"><a href="source-code.html#cb187-92" tabindex="-1"></a>                                  <span class="at">start =</span> <span class="fu">unlist</span>(start_list), </span>
<span id="cb187-93"><a href="source-code.html#cb187-93" tabindex="-1"></a>                                  <span class="at">end =</span> <span class="fu">unlist</span>(end_list))</span>
<span id="cb187-94"><a href="source-code.html#cb187-94" tabindex="-1"></a>      </span>
<span id="cb187-95"><a href="source-code.html#cb187-95" tabindex="-1"></a>      <span class="co"># remove NA</span></span>
<span id="cb187-96"><a href="source-code.html#cb187-96" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(cna_dataframe)</span>
<span id="cb187-97"><a href="source-code.html#cb187-97" tabindex="-1"></a>      </span>
<span id="cb187-98"><a href="source-code.html#cb187-98" tabindex="-1"></a>      <span class="co"># add column CNA</span></span>
<span id="cb187-99"><a href="source-code.html#cb187-99" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>cna <span class="ot">&lt;-</span> (CNA_change_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cna<span class="sc">!=</span><span class="fu">lag</span>(cna) <span class="sc">|</span> <span class="fu">is.na</span>(<span class="fu">lag</span>(cna))))<span class="sc">$</span>cna</span>
<span id="cb187-100"><a href="source-code.html#cb187-100" tabindex="-1"></a>      </span>
<span id="cb187-101"><a href="source-code.html#cb187-101" tabindex="-1"></a>      <span class="co"># remove columsn with cna == 2 (this is no CNA but normal diploid)</span></span>
<span id="cb187-102"><a href="source-code.html#cb187-102" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">subset</span>(cna_dataframe, cna <span class="sc">!=</span> <span class="dv">2</span>)</span>
<span id="cb187-103"><a href="source-code.html#cb187-103" tabindex="-1"></a>      </span>
<span id="cb187-104"><a href="source-code.html#cb187-104" tabindex="-1"></a>      <span class="co"># add total number of CNAs in this cell as colunn num_cna</span></span>
<span id="cb187-105"><a href="source-code.html#cb187-105" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>num_cna <span class="ot">&lt;-</span> <span class="fu">length</span>(cna_dataframe<span class="sc">$</span>cna), <span class="at">silent =</span> T)</span>
<span id="cb187-106"><a href="source-code.html#cb187-106" tabindex="-1"></a>      </span>
<span id="cb187-107"><a href="source-code.html#cb187-107" tabindex="-1"></a>      <span class="co"># add a width column of the CNA</span></span>
<span id="cb187-108"><a href="source-code.html#cb187-108" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>width <span class="ot">&lt;-</span> cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb187-109"><a href="source-code.html#cb187-109" tabindex="-1"></a>      </span>
<span id="cb187-110"><a href="source-code.html#cb187-110" tabindex="-1"></a>      <span class="co"># extract chromosome specifics</span></span>
<span id="cb187-111"><a href="source-code.html#cb187-111" tabindex="-1"></a>      chrom_df <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrom_centro_total, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb187-112"><a href="source-code.html#cb187-112" tabindex="-1"></a>      </span>
<span id="cb187-113"><a href="source-code.html#cb187-113" tabindex="-1"></a>      <span class="co"># in the next of code, we are going to compute the fraction of SCNA</span></span>
<span id="cb187-114"><a href="source-code.html#cb187-114" tabindex="-1"></a>      <span class="co"># Important: using this measure, a fraction of 1 means that a region the size of one arm</span></span>
<span id="cb187-115"><a href="source-code.html#cb187-115" tabindex="-1"></a>      <span class="co"># was affected. A fraction of 2 means the whole chromosome is affected. Note that these CNA</span></span>
<span id="cb187-116"><a href="source-code.html#cb187-116" tabindex="-1"></a>      <span class="co"># can cross centromeres, so we are unable to easily assign if it is the p-arm or q-arm that is </span></span>
<span id="cb187-117"><a href="source-code.html#cb187-117" tabindex="-1"></a>      <span class="co"># affected. </span></span>
<span id="cb187-118"><a href="source-code.html#cb187-118" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>fraction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>end <span class="sc">&lt;</span> chrom_df<span class="sc">$</span>p_arm, (cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm), </span>
<span id="cb187-119"><a href="source-code.html#cb187-119" tabindex="-1"></a>                                       <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>start <span class="sc">&gt;</span> chrom_df<span class="sc">$</span>p_arm, cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm,</span>
<span id="cb187-120"><a href="source-code.html#cb187-120" tabindex="-1"></a>                                              (((chrom_df<span class="sc">$</span>p_arm <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start) <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">+</span> ((cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm))))</span>
<span id="cb187-121"><a href="source-code.html#cb187-121" tabindex="-1"></a>      </span>
<span id="cb187-122"><a href="source-code.html#cb187-122" tabindex="-1"></a>      <span class="co"># adding cellname</span></span>
<span id="cb187-123"><a href="source-code.html#cb187-123" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>cell_id <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df_chrom)[column_i], <span class="at">silent =</span> T)</span>
<span id="cb187-124"><a href="source-code.html#cb187-124" tabindex="-1"></a>      </span>
<span id="cb187-125"><a href="source-code.html#cb187-125" tabindex="-1"></a>      <span class="fu">return</span>(cna_dataframe)</span>
<span id="cb187-126"><a href="source-code.html#cb187-126" tabindex="-1"></a>      </span>
<span id="cb187-127"><a href="source-code.html#cb187-127" tabindex="-1"></a>    }</span>
<span id="cb187-128"><a href="source-code.html#cb187-128" tabindex="-1"></a>    </span>
<span id="cb187-129"><a href="source-code.html#cb187-129" tabindex="-1"></a>    <span class="co"># running for each cell. </span></span>
<span id="cb187-130"><a href="source-code.html#cb187-130" tabindex="-1"></a>    cna_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">6</span><span class="sc">:</span><span class="fu">ncol</span>(df), cna_extract_cell)</span>
<span id="cb187-131"><a href="source-code.html#cb187-131" tabindex="-1"></a>    </span>
<span id="cb187-132"><a href="source-code.html#cb187-132" tabindex="-1"></a>    <span class="co"># constructing single dataframe </span></span>
<span id="cb187-133"><a href="source-code.html#cb187-133" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_list, <span class="at">.id =</span> <span class="st">&quot;cell_num_scar&quot;</span>)</span>
<span id="cb187-134"><a href="source-code.html#cb187-134" tabindex="-1"></a>    </span>
<span id="cb187-135"><a href="source-code.html#cb187-135" tabindex="-1"></a>    <span class="co"># removing cell_num scar</span></span>
<span id="cb187-136"><a href="source-code.html#cb187-136" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> cna_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df)]</span>
<span id="cb187-137"><a href="source-code.html#cb187-137" tabindex="-1"></a>    </span>
<span id="cb187-138"><a href="source-code.html#cb187-138" tabindex="-1"></a>    <span class="fu">return</span>(cna_df)</span>
<span id="cb187-139"><a href="source-code.html#cb187-139" tabindex="-1"></a>    </span>
<span id="cb187-140"><a href="source-code.html#cb187-140" tabindex="-1"></a>  }</span>
<span id="cb187-141"><a href="source-code.html#cb187-141" tabindex="-1"></a>  </span>
<span id="cb187-142"><a href="source-code.html#cb187-142" tabindex="-1"></a>  cna_df_per_chrom_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb187-143"><a href="source-code.html#cb187-143" tabindex="-1"></a>  </span>
<span id="cb187-144"><a href="source-code.html#cb187-144" tabindex="-1"></a>  <span class="cf">for</span>(chrom <span class="cf">in</span> <span class="fu">unique</span>(df<span class="sc">$</span>seqnames)) {</span>
<span id="cb187-145"><a href="source-code.html#cb187-145" tabindex="-1"></a>    cna_df_per_chrom_list[[chrom]] <span class="ot">&lt;-</span> <span class="fu">cna_extract</span>(chrom)</span>
<span id="cb187-146"><a href="source-code.html#cb187-146" tabindex="-1"></a>  }</span>
<span id="cb187-147"><a href="source-code.html#cb187-147" tabindex="-1"></a>  </span>
<span id="cb187-148"><a href="source-code.html#cb187-148" tabindex="-1"></a>  <span class="co"># constructing single dataframe </span></span>
<span id="cb187-149"><a href="source-code.html#cb187-149" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_df_per_chrom_list, <span class="at">.id =</span> <span class="st">&quot;chrom_scar&quot;</span>)</span>
<span id="cb187-150"><a href="source-code.html#cb187-150" tabindex="-1"></a>  </span>
<span id="cb187-151"><a href="source-code.html#cb187-151" tabindex="-1"></a>  <span class="co"># removing cell_num scar</span></span>
<span id="cb187-152"><a href="source-code.html#cb187-152" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> cna_df_per_chrom_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df_per_chrom_df)]</span>
<span id="cb187-153"><a href="source-code.html#cb187-153" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="cin_signature_ccle-script-run_cin_ccle-function" class="section level2 hasAnchor" number="6.18">
<h2><span class="header-section-number">6.18</span> cin_signature_ccle script: ‘run_cin_ccle’ function<a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="run_cin_ccle"><code>run_cin_ccle &lt;- function(cnv, d) { 
  #&#39; @param cnv is segmental data to input in CINSignatureQuantification
  #&#39; @param d is a dataframe with sample metadata that is joined with cnv
  #&#39; @return outputs CIN signature values.
  
  #############
  # Libraries #
  #############
  require(CINSignatureQuantification)
  require(tidyverse)
  
  ###########
  # Wrangle #
  ###########
  wrangler_cin_signature_ccle &lt;- function(df) {
    #&#39; @param  df: A dataframe containing all the copy number calls of all the files.
    #&#39; @return a dataframe in the format https://github.com/markowetzlab/CINSignatureQuantification
    
    # remove unwanted column names
    keep_columns &lt;-
      c(&#39;Chromosome&#39;,
        &#39;Start&#39;,
        &#39;End&#39;,
        &#39;Segment_Mean&#39;,
        &#39;DepMap_ID&#39;)
    
    new_df &lt;- subset(df, select = keep_columns)
    
    # reorder columns
    col_order &lt;- c(&#39;Chromosome&#39;,
                   &#39;Start&#39;,
                   &#39;End&#39;,
                   &#39;Segment_Mean&#39;,
                   &#39;DepMap_ID&#39;)
    
    new_df &lt;- new_df[, col_order]
    
    # define new column names
    column_names &lt;-
      c(&#39;chromosome&#39;,
        &#39;start&#39;,
        &#39;end&#39;,
        &#39;segVal&#39;,
        &#39;sample&#39;)
    
    colnames(new_df) &lt;- column_names
    
    return(new_df)
    
  }
  
  # Subset cnv to only include samples where radiation sensitivity is known
  cnv &lt;-
    cnv[cnv$DepMap_ID %in% d$DepMap_ID,]
  
  # Wrangle
  cnv &lt;-
    wrangler_cin_signature_ccle(cnv)
  
  # The CIN algorithm (run with quantifyCNSignatures) does not support Y chromosome, we have to get rid of those.
  cnv &lt;-
    cnv[cnv$chromosome != &#39;Y&#39;, ]
  
  #######
  # Run #
  #######
  # Running the CIN scores
  cin_scores &lt;-
    quantifyCNSignatures(cnv, build = &#39;hg38&#39;)
  
  # Extracting the values.
  cin_values &lt;- getActivities(cin_scores)

  ###########
  # Wrangle #
  ###########
  cin_df &lt;- as.data.frame(cin_values)
  
  cin_df$DepMap_ID &lt;- rownames(cin_df)
  
  cin_df &lt;-
    gather(cin_df, cx_signature, cx_value, CX1:CX17, factor_key = T)
  
  # add some metadata
  cin_df &lt;- inner_join(cin_df, d)

  ##########
  # Return #
  ##########
  return(cin_df)
  
}
</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="copy-number-pattern-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/edit/main/06-source.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/blob/main/06-source.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
