<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</title>
  <meta name="description" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  
  
  

<meta name="author" content="Daan Andel" />


<meta name="date" content="2023-09-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="copy-number-pattern-analyses.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Subclonal Evolution Paper</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html"><i class="fa fa-check"></i><b>2</b> In vitro radiation response and clinical data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#progression-free-survival"><i class="fa fa-check"></i><b>2.1</b> Progression-free survival</a></li>
<li class="chapter" data-level="2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#in-vitro-dose-response"><i class="fa fa-check"></i><b>2.2</b> In vitro dose-response</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#dose-response-plot"><i class="fa fa-check"></i><b>2.2.1</b> Dose-response plot</a></li>
<li class="chapter" data-level="2.2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#rauc-across-multiple-experiments"><i class="fa fa-check"></i><b>2.2.2</b> rAUC across multiple experiments</a></li>
<li class="chapter" data-level="2.2.3" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#recovery-statistics-after-10-gy-radiation"><i class="fa fa-check"></i><b>2.2.3</b> Recovery statistics after 10 Gy radiation</a></li>
<li class="chapter" data-level="2.2.4" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#persisters-versus-shifters-dose-response-plots"><i class="fa fa-check"></i><b>2.2.4</b> Persisters versus shifters dose-response plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html"><i class="fa fa-check"></i><b>3</b> Subclonal dynamics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#aneufinder-run"><i class="fa fa-check"></i><b>3.1</b> AneuFinder run</a></li>
<li class="chapter" data-level="3.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#quality-selection"><i class="fa fa-check"></i><b>3.2</b> Quality selection</a></li>
<li class="chapter" data-level="3.3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#summary-statistics-of-copy-number-alterations"><i class="fa fa-check"></i><b>3.3</b> Summary statistics of copy number alterations</a></li>
<li class="chapter" data-level="3.4" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#pca-and-k-means-plots"><i class="fa fa-check"></i><b>3.4</b> PCA and K-means plots</a></li>
<li class="chapter" data-level="3.5" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-selection"><i class="fa fa-check"></i><b>3.5</b> Subclone selection</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#selecting-subclones"><i class="fa fa-check"></i><b>3.5.1</b> Selecting subclones</a></li>
<li class="chapter" data-level="3.5.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-statistics"><i class="fa fa-check"></i><b>3.5.2</b> Subclone statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#single-cell-copy-number-heatmap-plots"><i class="fa fa-check"></i><b>3.6</b> Single cell copy number heatmap plots</a></li>
<li class="chapter" data-level="3.7" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#fishers-exact-test"><i class="fa fa-check"></i><b>3.7</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html"><i class="fa fa-check"></i><b>4</b> Copy number analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid"><i class="fa fa-check"></i><b>4.1</b> Colorectal cancer-specific arm-level copy number alterations per organoid</a></li>
<li class="chapter" data-level="4.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-per-subclone"><i class="fa fa-check"></i><b>4.2</b> Copy number alterations per subclone</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a"><i class="fa fa-check"></i><b>4.2.1</b> Copy number alterations specific to HUB062A</a></li>
<li class="chapter" data-level="4.2.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones"><i class="fa fa-check"></i><b>4.2.2</b> Copy number alterations specific to resistant subclones</a></li>
<li class="chapter" data-level="4.2.3" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#newly-acquired-copy-number-alterations"><i class="fa fa-check"></i><b>4.2.3</b> Newly acquired copy number alterations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html"><i class="fa fa-check"></i><b>5</b> Copy number pattern analyses</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#single-cell-analyses"><i class="fa fa-check"></i><b>5.1</b> Single cell analyses</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#heterogeneity-score"><i class="fa fa-check"></i><b>5.1.1</b> Heterogeneity score</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#copy-number-length-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Copy number length analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#ccle-analysis"><i class="fa fa-check"></i><b>5.2</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#run-cin-signature"><i class="fa fa-check"></i><b>5.2.1</b> Run CIN signature</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#statistics-cin-signature"><i class="fa fa-check"></i><b>5.2.2</b> Statistics CIN signature</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#plot-cx1-cin-signature"><i class="fa fa-check"></i><b>5.2.3</b> Plot CX1 CIN signature</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="source-code.html"><a href="source-code.html"><i class="fa fa-check"></i><b>6</b> Source code</a>
<ul>
<li class="chapter" data-level="6.1" data-path="source-code.html"><a href="source-code.html#auc_rad-script-auc_fun-function"><i class="fa fa-check"></i><b>6.1</b> AUC_rad script: ‘AUC_fun’ function</a></li>
<li class="chapter" data-level="6.2" data-path="source-code.html"><a href="source-code.html#fitter_rad-script-fitter-function"><i class="fa fa-check"></i><b>6.2</b> fitter_rad script: ‘fitter’ function</a></li>
<li class="chapter" data-level="6.3" data-path="source-code.html"><a href="source-code.html#ic50_rad-script-ic50_fun-function"><i class="fa fa-check"></i><b>6.3</b> ic50_rad script: ‘IC50_fun’ function</a></li>
<li class="chapter" data-level="6.4" data-path="source-code.html"><a href="source-code.html#relativize_dr-script-relative-function"><i class="fa fa-check"></i><b>6.4</b> relativize_dr script: ‘relative’ function</a></li>
<li class="chapter" data-level="6.5" data-path="source-code.html"><a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions"><i class="fa fa-check"></i><b>6.5</b> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions</a></li>
<li class="chapter" data-level="6.6" data-path="source-code.html"><a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions"><i class="fa fa-check"></i><b>6.6</b> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions</a></li>
<li class="chapter" data-level="6.7" data-path="source-code.html"><a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions"><i class="fa fa-check"></i><b>6.7</b> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions</a></li>
<li class="chapter" data-level="6.8" data-path="source-code.html"><a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function"><i class="fa fa-check"></i><b>6.8</b> plot_genomeheatmap script: ‘genomeheatmap’ function</a></li>
<li class="chapter" data-level="6.9" data-path="source-code.html"><a href="source-code.html#fisher_test-script-perform_fisher-function"><i class="fa fa-check"></i><b>6.9</b> fisher_test script: ‘perform_fisher’ function</a></li>
<li class="chapter" data-level="6.10" data-path="source-code.html"><a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function"><i class="fa fa-check"></i><b>6.10</b> find_specific_cna_sc script: ‘find_specific_cna_sc’ function</a></li>
<li class="chapter" data-level="6.11" data-path="source-code.html"><a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function"><i class="fa fa-check"></i><b>6.11</b> get_cytoband_coverage script: ‘get_cytoband_coverage’ function</a></li>
<li class="chapter" data-level="6.12" data-path="source-code.html"><a href="source-code.html#get_genes-script-get_genes-functoin"><i class="fa fa-check"></i><b>6.12</b> get_genes script: ‘get_genes’ functoin</a></li>
<li class="chapter" data-level="6.13" data-path="source-code.html"><a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function"><i class="fa fa-check"></i><b>6.13</b> get_oncogenes scripts: ‘get_oncogenes’ function</a></li>
<li class="chapter" data-level="6.14" data-path="source-code.html"><a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function"><i class="fa fa-check"></i><b>6.14</b> plot_genes_karyotype script: ‘plot_genes_karyotype’ function</a></li>
<li class="chapter" data-level="6.15" data-path="source-code.html"><a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions"><i class="fa fa-check"></i><b>6.15</b> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions</a></li>
<li class="chapter" data-level="6.16" data-path="source-code.html"><a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function"><i class="fa fa-check"></i><b>6.16</b> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function</a></li>
<li class="chapter" data-level="6.17" data-path="source-code.html"><a href="source-code.html#cna_profiler-script-cna_profiler-function"><i class="fa fa-check"></i><b>6.17</b> cna_profiler script: ‘cna_profiler’ function</a></li>
<li class="chapter" data-level="6.18" data-path="source-code.html"><a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function"><i class="fa fa-check"></i><b>6.18</b> cin_signature_ccle script: ‘run_cin_ccle’ function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="source-code" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Source code<a href="source-code.html#source-code" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The source code in this here is arranged in a loose chronological order, reflecting the sequence in which it is called in the scripts presented throughout this book.</p>
<div id="auc_rad-script-auc_fun-function" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> AUC_rad script: ‘AUC_fun’ function<a href="source-code.html#auc_rad-script-auc_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="source-code.html#cb144-1" tabindex="-1"></a><span class="co"># returns the AUC of the actual data.</span></span>
<span id="cb144-2"><a href="source-code.html#cb144-2" tabindex="-1"></a>AUC_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb144-3"><a href="source-code.html#cb144-3" tabindex="-1"></a>  </span>
<span id="cb144-4"><a href="source-code.html#cb144-4" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb144-5"><a href="source-code.html#cb144-5" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb144-6"><a href="source-code.html#cb144-6" tabindex="-1"></a>  </span>
<span id="cb144-7"><a href="source-code.html#cb144-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb144-8"><a href="source-code.html#cb144-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb144-9"><a href="source-code.html#cb144-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb144-10"><a href="source-code.html#cb144-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb144-11"><a href="source-code.html#cb144-11" tabindex="-1"></a>  </span>
<span id="cb144-12"><a href="source-code.html#cb144-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb144-13"><a href="source-code.html#cb144-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb144-14"><a href="source-code.html#cb144-14" tabindex="-1"></a>  </span>
<span id="cb144-15"><a href="source-code.html#cb144-15" tabindex="-1"></a>  <span class="co"># define dose_vetor</span></span>
<span id="cb144-16"><a href="source-code.html#cb144-16" tabindex="-1"></a>  dose_vector <span class="ot">&lt;-</span> <span class="fu">unique</span>(d<span class="sc">$</span>dose)</span>
<span id="cb144-17"><a href="source-code.html#cb144-17" tabindex="-1"></a>  </span>
<span id="cb144-18"><a href="source-code.html#cb144-18" tabindex="-1"></a>  </span>
<span id="cb144-19"><a href="source-code.html#cb144-19" tabindex="-1"></a>  <span class="co"># computes AUC for given predicted survival and dose given by dose_vector</span></span>
<span id="cb144-20"><a href="source-code.html#cb144-20" tabindex="-1"></a>  AUC_calc <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb144-21"><a href="source-code.html#cb144-21" tabindex="-1"></a>    <span class="co"># extract actual data</span></span>
<span id="cb144-22"><a href="source-code.html#cb144-22" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> df<span class="sc">$</span>relative_mean</span>
<span id="cb144-23"><a href="source-code.html#cb144-23" tabindex="-1"></a>    </span>
<span id="cb144-24"><a href="source-code.html#cb144-24" tabindex="-1"></a>    <span class="co"># extracts number of experiemnts </span></span>
<span id="cb144-25"><a href="source-code.html#cb144-25" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb144-26"><a href="source-code.html#cb144-26" tabindex="-1"></a>    <span class="co">#remove 2th and 3th number (are duplicates)</span></span>
<span id="cb144-27"><a href="source-code.html#cb144-27" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> rel_mean[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(rel_mean), n)]</span>
<span id="cb144-28"><a href="source-code.html#cb144-28" tabindex="-1"></a>    </span>
<span id="cb144-29"><a href="source-code.html#cb144-29" tabindex="-1"></a>    <span class="co"># compute maximal </span></span>
<span id="cb144-30"><a href="source-code.html#cb144-30" tabindex="-1"></a>    max_AUC <span class="ot">&lt;-</span> <span class="fu">max</span>(dose_vector)<span class="sc">*</span><span class="dv">1</span></span>
<span id="cb144-31"><a href="source-code.html#cb144-31" tabindex="-1"></a>    </span>
<span id="cb144-32"><a href="source-code.html#cb144-32" tabindex="-1"></a>    <span class="co">#compute AUC</span></span>
<span id="cb144-33"><a href="source-code.html#cb144-33" tabindex="-1"></a>    act_AUC <span class="ot">&lt;-</span> <span class="fu">AUC</span>(<span class="at">x=</span>dose_vector, <span class="at">y =</span> rel_mean)</span>
<span id="cb144-34"><a href="source-code.html#cb144-34" tabindex="-1"></a>    </span>
<span id="cb144-35"><a href="source-code.html#cb144-35" tabindex="-1"></a>    <span class="co"># make dataframe</span></span>
<span id="cb144-36"><a href="source-code.html#cb144-36" tabindex="-1"></a>    datafr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">max_AUC =</span> max_AUC,</span>
<span id="cb144-37"><a href="source-code.html#cb144-37" tabindex="-1"></a>                         <span class="at">act_AUC =</span> act_AUC, </span>
<span id="cb144-38"><a href="source-code.html#cb144-38" tabindex="-1"></a>                         <span class="at">rel_AUC =</span> act_AUC<span class="sc">/</span>max_AUC,</span>
<span id="cb144-39"><a href="source-code.html#cb144-39" tabindex="-1"></a>                         <span class="at">expcode =</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode),</span>
<span id="cb144-40"><a href="source-code.html#cb144-40" tabindex="-1"></a>                         <span class="at">cline=</span> <span class="fu">unique</span>(df<span class="sc">$</span>cline),</span>
<span id="cb144-41"><a href="source-code.html#cb144-41" tabindex="-1"></a>                         <span class="at">treatment =</span> <span class="fu">unique</span>(df<span class="sc">$</span>treatment ) )</span>
<span id="cb144-42"><a href="source-code.html#cb144-42" tabindex="-1"></a>    </span>
<span id="cb144-43"><a href="source-code.html#cb144-43" tabindex="-1"></a>    </span>
<span id="cb144-44"><a href="source-code.html#cb144-44" tabindex="-1"></a>    <span class="fu">return</span>(datafr)</span>
<span id="cb144-45"><a href="source-code.html#cb144-45" tabindex="-1"></a>  }</span>
<span id="cb144-46"><a href="source-code.html#cb144-46" tabindex="-1"></a>  </span>
<span id="cb144-47"><a href="source-code.html#cb144-47" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb144-48"><a href="source-code.html#cb144-48" tabindex="-1"></a>  AUC_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, AUC_calc) </span>
<span id="cb144-49"><a href="source-code.html#cb144-49" tabindex="-1"></a>  </span>
<span id="cb144-50"><a href="source-code.html#cb144-50" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb144-51"><a href="source-code.html#cb144-51" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, AUC_vector)</span>
<span id="cb144-52"><a href="source-code.html#cb144-52" tabindex="-1"></a>  </span>
<span id="cb144-53"><a href="source-code.html#cb144-53" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb144-54"><a href="source-code.html#cb144-54" tabindex="-1"></a>  </span>
<span id="cb144-55"><a href="source-code.html#cb144-55" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fitter_rad-script-fitter-function" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> fitter_rad script: ‘fitter’ function<a href="source-code.html#fitter_rad-script-fitter-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="source-code.html#cb145-1" tabindex="-1"></a>fitter <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb145-2"><a href="source-code.html#cb145-2" tabindex="-1"></a>  </span>
<span id="cb145-3"><a href="source-code.html#cb145-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb145-4"><a href="source-code.html#cb145-4" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb145-5"><a href="source-code.html#cb145-5" tabindex="-1"></a>  </span>
<span id="cb145-6"><a href="source-code.html#cb145-6" tabindex="-1"></a>  </span>
<span id="cb145-7"><a href="source-code.html#cb145-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb145-8"><a href="source-code.html#cb145-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb145-9"><a href="source-code.html#cb145-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb145-10"><a href="source-code.html#cb145-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb145-11"><a href="source-code.html#cb145-11" tabindex="-1"></a>  </span>
<span id="cb145-12"><a href="source-code.html#cb145-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb145-13"><a href="source-code.html#cb145-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb145-14"><a href="source-code.html#cb145-14" tabindex="-1"></a>  </span>
<span id="cb145-15"><a href="source-code.html#cb145-15" tabindex="-1"></a>  </span>
<span id="cb145-16"><a href="source-code.html#cb145-16" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb145-17"><a href="source-code.html#cb145-17" tabindex="-1"></a>  fit_model<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb145-18"><a href="source-code.html#cb145-18" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb145-19"><a href="source-code.html#cb145-19" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb145-20"><a href="source-code.html#cb145-20" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb145-21"><a href="source-code.html#cb145-21" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb145-22"><a href="source-code.html#cb145-22" tabindex="-1"></a>    </span>
<span id="cb145-23"><a href="source-code.html#cb145-23" tabindex="-1"></a>    newdata <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">dose=</span><span class="fu">exp</span>(<span class="fu">seq</span>(<span class="fu">log</span>(<span class="fl">0.00000001</span>), <span class="fu">log</span>(<span class="fu">max</span>(df<span class="sc">$</span>dose) <span class="sc">+</span> <span class="dv">1000</span>), <span class="at">length=</span><span class="dv">1000</span>))) <span class="co"># new data with doses. Note: lowest dose is not</span></span>
<span id="cb145-24"><a href="source-code.html#cb145-24" tabindex="-1"></a>    <span class="co"># log 0 but log(&#39;very small number&#39;) because otherwise this will hamper the scaling in ggplot later on. </span></span>
<span id="cb145-25"><a href="source-code.html#cb145-25" tabindex="-1"></a>    pm <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">newdata=</span>newdata, <span class="at">interval=</span><span class="st">&quot;confidence&quot;</span>) <span class="co"># new data with predictions and confidence intervals</span></span>
<span id="cb145-26"><a href="source-code.html#cb145-26" tabindex="-1"></a>    newdata<span class="sc">$</span>pred <span class="ot">&lt;-</span> pm[,<span class="dv">1</span>] <span class="co"># add prediction values to new data.</span></span>
<span id="cb145-27"><a href="source-code.html#cb145-27" tabindex="-1"></a>    newdata<span class="sc">$</span>predmin <span class="ot">&lt;-</span> pm[,<span class="dv">2</span>] <span class="co"># add lower bounderies to new data </span></span>
<span id="cb145-28"><a href="source-code.html#cb145-28" tabindex="-1"></a>    newdata<span class="sc">$</span>predmax <span class="ot">&lt;-</span> pm[,<span class="dv">3</span>] <span class="co"># add upper bounderies to new data </span></span>
<span id="cb145-29"><a href="source-code.html#cb145-29" tabindex="-1"></a>    newdata<span class="sc">$</span>cline <span class="ot">&lt;-</span> line<span class="sc">$</span>cline[<span class="dv">1</span>] <span class="co"># add column with cline.</span></span>
<span id="cb145-30"><a href="source-code.html#cb145-30" tabindex="-1"></a>    newdata<span class="sc">$</span>expcode <span class="ot">=</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode)</span>
<span id="cb145-31"><a href="source-code.html#cb145-31" tabindex="-1"></a>    <span class="fu">return</span>(newdata)</span>
<span id="cb145-32"><a href="source-code.html#cb145-32" tabindex="-1"></a>  }</span>
<span id="cb145-33"><a href="source-code.html#cb145-33" tabindex="-1"></a>  </span>
<span id="cb145-34"><a href="source-code.html#cb145-34" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb145-35"><a href="source-code.html#cb145-35" tabindex="-1"></a>  data_frames <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, fit_model) </span>
<span id="cb145-36"><a href="source-code.html#cb145-36" tabindex="-1"></a>  </span>
<span id="cb145-37"><a href="source-code.html#cb145-37" tabindex="-1"></a>  </span>
<span id="cb145-38"><a href="source-code.html#cb145-38" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb145-39"><a href="source-code.html#cb145-39" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, data_frames)</span>
<span id="cb145-40"><a href="source-code.html#cb145-40" tabindex="-1"></a>  </span>
<span id="cb145-41"><a href="source-code.html#cb145-41" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb145-42"><a href="source-code.html#cb145-42" tabindex="-1"></a>  </span>
<span id="cb145-43"><a href="source-code.html#cb145-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="ic50_rad-script-ic50_fun-function" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> ic50_rad script: ‘IC50_fun’ function<a href="source-code.html#ic50_rad-script-ic50_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="source-code.html#cb146-1" tabindex="-1"></a>IC50_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb146-2"><a href="source-code.html#cb146-2" tabindex="-1"></a>  </span>
<span id="cb146-3"><a href="source-code.html#cb146-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb146-4"><a href="source-code.html#cb146-4" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">is.null</span>(rad), d<span class="ot">&lt;-</span>df, d <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> rad))</span>
<span id="cb146-5"><a href="source-code.html#cb146-5" tabindex="-1"></a>  </span>
<span id="cb146-6"><a href="source-code.html#cb146-6" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb146-7"><a href="source-code.html#cb146-7" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb146-8"><a href="source-code.html#cb146-8" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb146-9"><a href="source-code.html#cb146-9" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb146-10"><a href="source-code.html#cb146-10" tabindex="-1"></a>  </span>
<span id="cb146-11"><a href="source-code.html#cb146-11" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb146-12"><a href="source-code.html#cb146-12" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span><span class="cn">TRUE</span>)</span>
<span id="cb146-13"><a href="source-code.html#cb146-13" tabindex="-1"></a>  </span>
<span id="cb146-14"><a href="source-code.html#cb146-14" tabindex="-1"></a>  </span>
<span id="cb146-15"><a href="source-code.html#cb146-15" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb146-16"><a href="source-code.html#cb146-16" tabindex="-1"></a>  ic50<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb146-17"><a href="source-code.html#cb146-17" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb146-18"><a href="source-code.html#cb146-18" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb146-19"><a href="source-code.html#cb146-19" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb146-20"><a href="source-code.html#cb146-20" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb146-21"><a href="source-code.html#cb146-21" tabindex="-1"></a>    </span>
<span id="cb146-22"><a href="source-code.html#cb146-22" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">coef</span>(fit)[<span class="dv">4</span>])</span>
<span id="cb146-23"><a href="source-code.html#cb146-23" tabindex="-1"></a>  }</span>
<span id="cb146-24"><a href="source-code.html#cb146-24" tabindex="-1"></a>  </span>
<span id="cb146-25"><a href="source-code.html#cb146-25" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb146-26"><a href="source-code.html#cb146-26" tabindex="-1"></a>  IC50_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, ic50) </span>
<span id="cb146-27"><a href="source-code.html#cb146-27" tabindex="-1"></a>  </span>
<span id="cb146-28"><a href="source-code.html#cb146-28" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb146-29"><a href="source-code.html#cb146-29" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, IC50_vector)</span>
<span id="cb146-30"><a href="source-code.html#cb146-30" tabindex="-1"></a>  </span>
<span id="cb146-31"><a href="source-code.html#cb146-31" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb146-32"><a href="source-code.html#cb146-32" tabindex="-1"></a>  </span>
<span id="cb146-33"><a href="source-code.html#cb146-33" tabindex="-1"></a>} </span></code></pre></div>
</div>
<div id="relativize_dr-script-relative-function" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> relativize_dr script: ‘relative’ function<a href="source-code.html#relativize_dr-script-relative-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="source-code.html#cb147-1" tabindex="-1"></a>relative <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb147-2"><a href="source-code.html#cb147-2" tabindex="-1"></a>  <span class="co">#&#39; @param df with dose response data and columns cline, dose, Exp, </span></span>
<span id="cb147-3"><a href="source-code.html#cb147-3" tabindex="-1"></a>  <span class="co">#&#39; rec_rad, expcode, ncolonies</span></span>
<span id="cb147-4"><a href="source-code.html#cb147-4" tabindex="-1"></a>  <span class="co">#&#39; @param rad if line received radiation (T or F), stored in column rec_rad</span></span>
<span id="cb147-5"><a href="source-code.html#cb147-5" tabindex="-1"></a>  <span class="co">#&#39; @return relativized data where dose = 0 is used as 100%</span></span>
<span id="cb147-6"><a href="source-code.html#cb147-6" tabindex="-1"></a>  </span>
<span id="cb147-7"><a href="source-code.html#cb147-7" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb147-8"><a href="source-code.html#cb147-8" tabindex="-1"></a>  <span class="fu">ifelse</span>(rad <span class="sc">==</span> T, d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">1</span>), d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">0</span>) )</span>
<span id="cb147-9"><a href="source-code.html#cb147-9" tabindex="-1"></a>  </span>
<span id="cb147-10"><a href="source-code.html#cb147-10" tabindex="-1"></a>  <span class="co"># define ncolonies</span></span>
<span id="cb147-11"><a href="source-code.html#cb147-11" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;ncolonies&quot;</span></span>
<span id="cb147-12"><a href="source-code.html#cb147-12" tabindex="-1"></a>  </span>
<span id="cb147-13"><a href="source-code.html#cb147-13" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb147-14"><a href="source-code.html#cb147-14" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb147-15"><a href="source-code.html#cb147-15" tabindex="-1"></a>  </span>
<span id="cb147-16"><a href="source-code.html#cb147-16" tabindex="-1"></a>  <span class="co"># extracts number of experiemnts </span></span>
<span id="cb147-17"><a href="source-code.html#cb147-17" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb147-18"><a href="source-code.html#cb147-18" tabindex="-1"></a>  </span>
<span id="cb147-19"><a href="source-code.html#cb147-19" tabindex="-1"></a>  <span class="co"># Takes mean every nth row</span></span>
<span id="cb147-20"><a href="source-code.html#cb147-20" tabindex="-1"></a>  </span>
<span id="cb147-21"><a href="source-code.html#cb147-21" tabindex="-1"></a>  mean_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb147-22"><a href="source-code.html#cb147-22" tabindex="-1"></a>    m <span class="ot">&lt;-</span> <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb147-23"><a href="source-code.html#cb147-23" tabindex="-1"></a>    <span class="fu">return</span>(m)</span>
<span id="cb147-24"><a href="source-code.html#cb147-24" tabindex="-1"></a>  }</span>
<span id="cb147-25"><a href="source-code.html#cb147-25" tabindex="-1"></a>  </span>
<span id="cb147-26"><a href="source-code.html#cb147-26" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))), mean_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb147-27"><a href="source-code.html#cb147-27" tabindex="-1"></a>  <span class="co"># Duplicates (n=2) or triplicates (n=3) the rows.</span></span>
<span id="cb147-28"><a href="source-code.html#cb147-28" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> mean[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(mean)), <span class="at">each =</span> n), ]</span>
<span id="cb147-29"><a href="source-code.html#cb147-29" tabindex="-1"></a>  <span class="co"># Adds everything to the dataframe. </span></span>
<span id="cb147-30"><a href="source-code.html#cb147-30" tabindex="-1"></a>  d[<span class="st">&quot;mean&quot;</span>] <span class="ot">&lt;-</span> mean</span>
<span id="cb147-31"><a href="source-code.html#cb147-31" tabindex="-1"></a>  </span>
<span id="cb147-32"><a href="source-code.html#cb147-32" tabindex="-1"></a>  <span class="co"># Select mean values of 0 concentration</span></span>
<span id="cb147-33"><a href="source-code.html#cb147-33" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb147-34"><a href="source-code.html#cb147-34" tabindex="-1"></a>    <span class="fu">group_by</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb147-35"><a href="source-code.html#cb147-35" tabindex="-1"></a>    <span class="fu">arrange</span>(cline,dose)</span>
<span id="cb147-36"><a href="source-code.html#cb147-36" tabindex="-1"></a>  </span>
<span id="cb147-37"><a href="source-code.html#cb147-37" tabindex="-1"></a>  first <span class="ot">&lt;-</span> d[d<span class="sc">$</span>dose<span class="sc">==</span><span class="dv">0</span>, ] <span class="sc">%&gt;%</span></span>
<span id="cb147-38"><a href="source-code.html#cb147-38" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(cline, <span class="at">value100=</span>mean)</span>
<span id="cb147-39"><a href="source-code.html#cb147-39" tabindex="-1"></a>  </span>
<span id="cb147-40"><a href="source-code.html#cb147-40" tabindex="-1"></a>  <span class="co"># Make cline a factor</span></span>
<span id="cb147-41"><a href="source-code.html#cb147-41" tabindex="-1"></a>  d<span class="sc">$</span>cline<span class="ot">&lt;-</span><span class="fu">factor</span>(d<span class="sc">$</span>cline)</span>
<span id="cb147-42"><a href="source-code.html#cb147-42" tabindex="-1"></a>  </span>
<span id="cb147-43"><a href="source-code.html#cb147-43" tabindex="-1"></a>  <span class="co"># Merge first original (d)</span></span>
<span id="cb147-44"><a href="source-code.html#cb147-44" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb147-45"><a href="source-code.html#cb147-45" tabindex="-1"></a>    <span class="fu">merge</span>(first,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;cline&quot;</span>))</span>
<span id="cb147-46"><a href="source-code.html#cb147-46" tabindex="-1"></a>  </span>
<span id="cb147-47"><a href="source-code.html#cb147-47" tabindex="-1"></a>  <span class="co"># Extract only every nth row.</span></span>
<span id="cb147-48"><a href="source-code.html#cb147-48" tabindex="-1"></a>  d <span class="ot">=</span> d[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(d), n), ]</span>
<span id="cb147-49"><a href="source-code.html#cb147-49" tabindex="-1"></a>  </span>
<span id="cb147-50"><a href="source-code.html#cb147-50" tabindex="-1"></a>  <span class="co"># make new column with relative, called relative</span></span>
<span id="cb147-51"><a href="source-code.html#cb147-51" tabindex="-1"></a>  d<span class="sc">$</span>relative <span class="ot">=</span> d<span class="sc">$</span>ncolonies<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb147-52"><a href="source-code.html#cb147-52" tabindex="-1"></a>  d<span class="sc">$</span>relative_mean <span class="ot">=</span> d<span class="sc">$</span>mean<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb147-53"><a href="source-code.html#cb147-53" tabindex="-1"></a>  </span>
<span id="cb147-54"><a href="source-code.html#cb147-54" tabindex="-1"></a>  d<span class="ot">&lt;-</span><span class="fu">subset</span>(d, <span class="at">select=</span><span class="sc">-</span><span class="fu">c</span>(mean,value100))</span>
<span id="cb147-55"><a href="source-code.html#cb147-55" tabindex="-1"></a>  </span>
<span id="cb147-56"><a href="source-code.html#cb147-56" tabindex="-1"></a>  <span class="co"># compute relative standard error of the mean</span></span>
<span id="cb147-57"><a href="source-code.html#cb147-57" tabindex="-1"></a>  <span class="co"># define the column with sem in it</span></span>
<span id="cb147-58"><a href="source-code.html#cb147-58" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;relative&quot;</span></span>
<span id="cb147-59"><a href="source-code.html#cb147-59" tabindex="-1"></a>  </span>
<span id="cb147-60"><a href="source-code.html#cb147-60" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb147-61"><a href="source-code.html#cb147-61" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb147-62"><a href="source-code.html#cb147-62" tabindex="-1"></a>  </span>
<span id="cb147-63"><a href="source-code.html#cb147-63" tabindex="-1"></a>  </span>
<span id="cb147-64"><a href="source-code.html#cb147-64" tabindex="-1"></a>  sem_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb147-65"><a href="source-code.html#cb147-65" tabindex="-1"></a>    std <span class="ot">&lt;-</span> <span class="fu">sd</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb147-66"><a href="source-code.html#cb147-66" tabindex="-1"></a>    vector <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(x)</span>
<span id="cb147-67"><a href="source-code.html#cb147-67" tabindex="-1"></a>    </span>
<span id="cb147-68"><a href="source-code.html#cb147-68" tabindex="-1"></a>    sem <span class="ot">&lt;-</span> std<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(vector))</span>
<span id="cb147-69"><a href="source-code.html#cb147-69" tabindex="-1"></a>    <span class="fu">return</span>(sem)</span>
<span id="cb147-70"><a href="source-code.html#cb147-70" tabindex="-1"></a>  }</span>
<span id="cb147-71"><a href="source-code.html#cb147-71" tabindex="-1"></a>  </span>
<span id="cb147-72"><a href="source-code.html#cb147-72" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))),sem_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb147-73"><a href="source-code.html#cb147-73" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> sem[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(sem)), <span class="at">each =</span> n), ]</span>
<span id="cb147-74"><a href="source-code.html#cb147-74" tabindex="-1"></a>  d[<span class="st">&quot;relative_sem&quot;</span>] <span class="ot">&lt;-</span> sem</span>
<span id="cb147-75"><a href="source-code.html#cb147-75" tabindex="-1"></a>  </span>
<span id="cb147-76"><a href="source-code.html#cb147-76" tabindex="-1"></a>  <span class="fu">return</span>(d)</span>
<span id="cb147-77"><a href="source-code.html#cb147-77" tabindex="-1"></a>  </span>
<span id="cb147-78"><a href="source-code.html#cb147-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions<a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="source-code.html#cb148-1" tabindex="-1"></a>quality_check <span class="ot">&lt;-</span> <span class="cf">function</span>(inputdir, model) {</span>
<span id="cb148-2"><a href="source-code.html#cb148-2" tabindex="-1"></a>  <span class="co">#&#39; @param inputdir is directory containing the standard aneufinder output for each plate</span></span>
<span id="cb148-3"><a href="source-code.html#cb148-3" tabindex="-1"></a>  <span class="co">#&#39; model is which model to use for the quality check: either dnacopy, edivisive or HMM</span></span>
<span id="cb148-4"><a href="source-code.html#cb148-4" tabindex="-1"></a>  <span class="co">#&#39; quality_check will make quality metrics for each plate</span></span>
<span id="cb148-5"><a href="source-code.html#cb148-5" tabindex="-1"></a>  <span class="co">#&#39; quality_check assumes that rdaBaseDirectory contains folder with models for each plate</span></span>
<span id="cb148-6"><a href="source-code.html#cb148-6" tabindex="-1"></a>  <span class="co">#&#39; each plate folder has to contain a MODEL folder and a method-&#39;dnacopy, edivisive, or HMM&#39; folder</span></span>
<span id="cb148-7"><a href="source-code.html#cb148-7" tabindex="-1"></a>  <span class="co">#&#39; @example quality_check(inputdir = rdaBaseDirectory, model = &#39;edivisive&#39;)</span></span>
<span id="cb148-8"><a href="source-code.html#cb148-8" tabindex="-1"></a>  </span>
<span id="cb148-9"><a href="source-code.html#cb148-9" tabindex="-1"></a>  <span class="co"># go to folder with appropiate files</span></span>
<span id="cb148-10"><a href="source-code.html#cb148-10" tabindex="-1"></a>  rda_folder <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputdir, <span class="st">&#39;/MODELS&#39;</span>, <span class="st">&#39;/method-&#39;</span>, model)</span>
<span id="cb148-11"><a href="source-code.html#cb148-11" tabindex="-1"></a>  </span>
<span id="cb148-12"><a href="source-code.html#cb148-12" tabindex="-1"></a>  <span class="co"># extract files</span></span>
<span id="cb148-13"><a href="source-code.html#cb148-13" tabindex="-1"></a>  rda_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(rda_folder, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb148-14"><a href="source-code.html#cb148-14" tabindex="-1"></a>  </span>
<span id="cb148-15"><a href="source-code.html#cb148-15" tabindex="-1"></a>  <span class="co"># run quality check on each file</span></span>
<span id="cb148-16"><a href="source-code.html#cb148-16" tabindex="-1"></a>  cl <span class="ot">&lt;-</span></span>
<span id="cb148-17"><a href="source-code.html#cb148-17" tabindex="-1"></a>    <span class="fu">clusterByQuality</span>(</span>
<span id="cb148-18"><a href="source-code.html#cb148-18" tabindex="-1"></a>      rda_files,</span>
<span id="cb148-19"><a href="source-code.html#cb148-19" tabindex="-1"></a>      <span class="at">measures =</span> <span class="fu">c</span>(</span>
<span id="cb148-20"><a href="source-code.html#cb148-20" tabindex="-1"></a>        <span class="st">&#39;spikiness&#39;</span>,</span>
<span id="cb148-21"><a href="source-code.html#cb148-21" tabindex="-1"></a>        <span class="st">&#39;num.segments&#39;</span>,</span>
<span id="cb148-22"><a href="source-code.html#cb148-22" tabindex="-1"></a>        <span class="st">&#39;entropy&#39;</span>,</span>
<span id="cb148-23"><a href="source-code.html#cb148-23" tabindex="-1"></a>        <span class="st">&#39;bhattacharyya&#39;</span>,</span>
<span id="cb148-24"><a href="source-code.html#cb148-24" tabindex="-1"></a>        <span class="st">&#39;sos&#39;</span></span>
<span id="cb148-25"><a href="source-code.html#cb148-25" tabindex="-1"></a>      )</span>
<span id="cb148-26"><a href="source-code.html#cb148-26" tabindex="-1"></a>    )</span>
<span id="cb148-27"><a href="source-code.html#cb148-27" tabindex="-1"></a>  </span>
<span id="cb148-28"><a href="source-code.html#cb148-28" tabindex="-1"></a>  <span class="fu">return</span>(cl)</span>
<span id="cb148-29"><a href="source-code.html#cb148-29" tabindex="-1"></a>  </span>
<span id="cb148-30"><a href="source-code.html#cb148-30" tabindex="-1"></a>}</span>
<span id="cb148-31"><a href="source-code.html#cb148-31" tabindex="-1"></a></span>
<span id="cb148-32"><a href="source-code.html#cb148-32" tabindex="-1"></a>quality_select <span class="ot">&lt;-</span> <span class="cf">function</span>(cl, spik, bhat) {</span>
<span id="cb148-33"><a href="source-code.html#cb148-33" tabindex="-1"></a>  <span class="co">#&#39; selects all files that meets the defined quality requirements</span></span>
<span id="cb148-34"><a href="source-code.html#cb148-34" tabindex="-1"></a>  <span class="co">#&#39; @param cl output of the clusterByQuality function in AneuFinder</span></span>
<span id="cb148-35"><a href="source-code.html#cb148-35" tabindex="-1"></a>  <span class="co">#&#39; @param spik the spikiness treshhold. All clusters with spikiness above</span></span>
<span id="cb148-36"><a href="source-code.html#cb148-36" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb148-37"><a href="source-code.html#cb148-37" tabindex="-1"></a>  <span class="co">#&#39; @param bhat the bhattacharrya score. All clusters with bhattacharrya below</span></span>
<span id="cb148-38"><a href="source-code.html#cb148-38" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb148-39"><a href="source-code.html#cb148-39" tabindex="-1"></a>  <span class="co">#&#39; @return returns a vector of selected files paths</span></span>
<span id="cb148-40"><a href="source-code.html#cb148-40" tabindex="-1"></a>  </span>
<span id="cb148-41"><a href="source-code.html#cb148-41" tabindex="-1"></a>  <span class="co"># convert to df to allow easy wrangling</span></span>
<span id="cb148-42"><a href="source-code.html#cb148-42" tabindex="-1"></a>  cl_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb148-43"><a href="source-code.html#cb148-43" tabindex="-1"></a>  </span>
<span id="cb148-44"><a href="source-code.html#cb148-44" tabindex="-1"></a>  <span class="co"># remove higher than spik</span></span>
<span id="cb148-45"><a href="source-code.html#cb148-45" tabindex="-1"></a>  spik <span class="ot">&lt;-</span> <span class="fu">subset</span>(cl_df, spikiness <span class="sc">&lt;</span> spik)</span>
<span id="cb148-46"><a href="source-code.html#cb148-46" tabindex="-1"></a>  </span>
<span id="cb148-47"><a href="source-code.html#cb148-47" tabindex="-1"></a>  <span class="co"># remove lower than bhat</span></span>
<span id="cb148-48"><a href="source-code.html#cb148-48" tabindex="-1"></a>  spik_bhat <span class="ot">&lt;-</span> <span class="fu">subset</span>(spik,  bhattacharyya <span class="sc">&gt;</span> bhat)</span>
<span id="cb148-49"><a href="source-code.html#cb148-49" tabindex="-1"></a>  </span>
<span id="cb148-50"><a href="source-code.html#cb148-50" tabindex="-1"></a>  cluster_n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(spik_bhat)</span>
<span id="cb148-51"><a href="source-code.html#cb148-51" tabindex="-1"></a>  </span>
<span id="cb148-52"><a href="source-code.html#cb148-52" tabindex="-1"></a>  <span class="co"># create vector of selected files</span></span>
<span id="cb148-53"><a href="source-code.html#cb148-53" tabindex="-1"></a>  selected.files <span class="ot">&lt;-</span> <span class="fu">unlist</span>(cl<span class="sc">$</span>classification[<span class="dv">0</span><span class="sc">:</span>cluster_n])</span>
<span id="cb148-54"><a href="source-code.html#cb148-54" tabindex="-1"></a>  </span>
<span id="cb148-55"><a href="source-code.html#cb148-55" tabindex="-1"></a>  <span class="fu">return</span>(selected.files)</span>
<span id="cb148-56"><a href="source-code.html#cb148-56" tabindex="-1"></a>  </span>
<span id="cb148-57"><a href="source-code.html#cb148-57" tabindex="-1"></a>}</span>
<span id="cb148-58"><a href="source-code.html#cb148-58" tabindex="-1"></a></span>
<span id="cb148-59"><a href="source-code.html#cb148-59" tabindex="-1"></a>check_quality <span class="ot">&lt;-</span> <span class="cf">function</span>(cl) {</span>
<span id="cb148-60"><a href="source-code.html#cb148-60" tabindex="-1"></a>  <span class="fu">return</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb148-61"><a href="source-code.html#cb148-61" tabindex="-1"></a>}</span>
<span id="cb148-62"><a href="source-code.html#cb148-62" tabindex="-1"></a>  </span>
<span id="cb148-63"><a href="source-code.html#cb148-63" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions<a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="source-code.html#cb149-1" tabindex="-1"></a>draw_pca <span class="ot">&lt;-</span></span>
<span id="cb149-2"><a href="source-code.html#cb149-2" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb149-3"><a href="source-code.html#cb149-3" tabindex="-1"></a>           list_files2,</span>
<span id="cb149-4"><a href="source-code.html#cb149-4" tabindex="-1"></a>           size,</span>
<span id="cb149-5"><a href="source-code.html#cb149-5" tabindex="-1"></a>           legend_position,</span>
<span id="cb149-6"><a href="source-code.html#cb149-6" tabindex="-1"></a>           baseline_kra,</span>
<span id="cb149-7"><a href="source-code.html#cb149-7" tabindex="-1"></a>           recurrence_kra,</span>
<span id="cb149-8"><a href="source-code.html#cb149-8" tabindex="-1"></a>           <span class="at">ssdna004 =</span> F) {</span>
<span id="cb149-9"><a href="source-code.html#cb149-9" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb149-10"><a href="source-code.html#cb149-10" tabindex="-1"></a>    </span>
<span id="cb149-11"><a href="source-code.html#cb149-11" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-12"><a href="source-code.html#cb149-12" tabindex="-1"></a>      cells,</span>
<span id="cb149-13"><a href="source-code.html#cb149-13" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb149-14"><a href="source-code.html#cb149-14" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb149-15"><a href="source-code.html#cb149-15" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb149-16"><a href="source-code.html#cb149-16" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb149-17"><a href="source-code.html#cb149-17" tabindex="-1"></a>    )</span>
<span id="cb149-18"><a href="source-code.html#cb149-18" tabindex="-1"></a>    </span>
<span id="cb149-19"><a href="source-code.html#cb149-19" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb149-20"><a href="source-code.html#cb149-20" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-21"><a href="source-code.html#cb149-21" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-22"><a href="source-code.html#cb149-22" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb149-23"><a href="source-code.html#cb149-23" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb149-24"><a href="source-code.html#cb149-24" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb149-25"><a href="source-code.html#cb149-25" tabindex="-1"></a>    </span>
<span id="cb149-26"><a href="source-code.html#cb149-26" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb149-27"><a href="source-code.html#cb149-27" tabindex="-1"></a>    </span>
<span id="cb149-28"><a href="source-code.html#cb149-28" tabindex="-1"></a>    col_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&quot;#999999&quot;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>)))</span>
<span id="cb149-29"><a href="source-code.html#cb149-29" tabindex="-1"></a>    label_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Recurrence&#39;</span>, <span class="st">&#39;Baseline&#39;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence&#39;</span>)))</span>
<span id="cb149-30"><a href="source-code.html#cb149-30" tabindex="-1"></a>    </span>
<span id="cb149-31"><a href="source-code.html#cb149-31" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb149-32"><a href="source-code.html#cb149-32" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb149-33"><a href="source-code.html#cb149-33" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb149-34"><a href="source-code.html#cb149-34" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb149-35"><a href="source-code.html#cb149-35" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb149-36"><a href="source-code.html#cb149-36" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb149-37"><a href="source-code.html#cb149-37" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb149-38"><a href="source-code.html#cb149-38" tabindex="-1"></a>        <span class="at">values =</span> col_vals[[<span class="dv">1</span>]],</span>
<span id="cb149-39"><a href="source-code.html#cb149-39" tabindex="-1"></a>        <span class="at">labels =</span> label_vals[[<span class="dv">1</span>]]</span>
<span id="cb149-40"><a href="source-code.html#cb149-40" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb149-41"><a href="source-code.html#cb149-41" tabindex="-1"></a>      </span>
<span id="cb149-42"><a href="source-code.html#cb149-42" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb149-43"><a href="source-code.html#cb149-43" tabindex="-1"></a>      </span>
<span id="cb149-44"><a href="source-code.html#cb149-44" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb149-45"><a href="source-code.html#cb149-45" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb149-46"><a href="source-code.html#cb149-46" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb149-47"><a href="source-code.html#cb149-47" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb149-48"><a href="source-code.html#cb149-48" tabindex="-1"></a>      </span>
<span id="cb149-49"><a href="source-code.html#cb149-49" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb149-50"><a href="source-code.html#cb149-50" tabindex="-1"></a>    p</span>
<span id="cb149-51"><a href="source-code.html#cb149-51" tabindex="-1"></a>    </span>
<span id="cb149-52"><a href="source-code.html#cb149-52" tabindex="-1"></a>  </span>
<span id="cb149-53"><a href="source-code.html#cb149-53" tabindex="-1"></a>    </span>
<span id="cb149-54"><a href="source-code.html#cb149-54" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb149-55"><a href="source-code.html#cb149-55" tabindex="-1"></a>    </span>
<span id="cb149-56"><a href="source-code.html#cb149-56" tabindex="-1"></a>  }</span>
<span id="cb149-57"><a href="source-code.html#cb149-57" tabindex="-1"></a></span>
<span id="cb149-58"><a href="source-code.html#cb149-58" tabindex="-1"></a></span>
<span id="cb149-59"><a href="source-code.html#cb149-59" tabindex="-1"></a>k_cluster <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb149-60"><a href="source-code.html#cb149-60" tabindex="-1"></a>                      list_files2,</span>
<span id="cb149-61"><a href="source-code.html#cb149-61" tabindex="-1"></a>                      cluster_n,</span>
<span id="cb149-62"><a href="source-code.html#cb149-62" tabindex="-1"></a>                      cols,</span>
<span id="cb149-63"><a href="source-code.html#cb149-63" tabindex="-1"></a>                      labels,</span>
<span id="cb149-64"><a href="source-code.html#cb149-64" tabindex="-1"></a>                      legend_position,</span>
<span id="cb149-65"><a href="source-code.html#cb149-65" tabindex="-1"></a>                      normalize,</span>
<span id="cb149-66"><a href="source-code.html#cb149-66" tabindex="-1"></a>                      <span class="at">return_elbow =</span> F) {</span>
<span id="cb149-67"><a href="source-code.html#cb149-67" tabindex="-1"></a>  </span>
<span id="cb149-68"><a href="source-code.html#cb149-68" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb149-69"><a href="source-code.html#cb149-69" tabindex="-1"></a>  </span>
<span id="cb149-70"><a href="source-code.html#cb149-70" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb149-71"><a href="source-code.html#cb149-71" tabindex="-1"></a>  </span>
<span id="cb149-72"><a href="source-code.html#cb149-72" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-73"><a href="source-code.html#cb149-73" tabindex="-1"></a>    cells,</span>
<span id="cb149-74"><a href="source-code.html#cb149-74" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-75"><a href="source-code.html#cb149-75" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb149-76"><a href="source-code.html#cb149-76" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb149-77"><a href="source-code.html#cb149-77" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-78"><a href="source-code.html#cb149-78" tabindex="-1"></a>  )</span>
<span id="cb149-79"><a href="source-code.html#cb149-79" tabindex="-1"></a></span>
<span id="cb149-80"><a href="source-code.html#cb149-80" tabindex="-1"></a>  </span>
<span id="cb149-81"><a href="source-code.html#cb149-81" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-82"><a href="source-code.html#cb149-82" tabindex="-1"></a>    cells,</span>
<span id="cb149-83"><a href="source-code.html#cb149-83" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-84"><a href="source-code.html#cb149-84" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb149-85"><a href="source-code.html#cb149-85" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb149-86"><a href="source-code.html#cb149-86" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-87"><a href="source-code.html#cb149-87" tabindex="-1"></a>  )</span>
<span id="cb149-88"><a href="source-code.html#cb149-88" tabindex="-1"></a>  </span>
<span id="cb149-89"><a href="source-code.html#cb149-89" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-90"><a href="source-code.html#cb149-90" tabindex="-1"></a>    cells,</span>
<span id="cb149-91"><a href="source-code.html#cb149-91" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-92"><a href="source-code.html#cb149-92" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb149-93"><a href="source-code.html#cb149-93" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb149-94"><a href="source-code.html#cb149-94" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-95"><a href="source-code.html#cb149-95" tabindex="-1"></a>  )</span>
<span id="cb149-96"><a href="source-code.html#cb149-96" tabindex="-1"></a>  </span>
<span id="cb149-97"><a href="source-code.html#cb149-97" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-98"><a href="source-code.html#cb149-98" tabindex="-1"></a>    cells,</span>
<span id="cb149-99"><a href="source-code.html#cb149-99" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-100"><a href="source-code.html#cb149-100" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb149-101"><a href="source-code.html#cb149-101" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb149-102"><a href="source-code.html#cb149-102" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-103"><a href="source-code.html#cb149-103" tabindex="-1"></a>  )</span>
<span id="cb149-104"><a href="source-code.html#cb149-104" tabindex="-1"></a>  </span>
<span id="cb149-105"><a href="source-code.html#cb149-105" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-106"><a href="source-code.html#cb149-106" tabindex="-1"></a>    cells,</span>
<span id="cb149-107"><a href="source-code.html#cb149-107" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-108"><a href="source-code.html#cb149-108" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb149-109"><a href="source-code.html#cb149-109" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb149-110"><a href="source-code.html#cb149-110" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-111"><a href="source-code.html#cb149-111" tabindex="-1"></a>  )</span>
<span id="cb149-112"><a href="source-code.html#cb149-112" tabindex="-1"></a>  </span>
<span id="cb149-113"><a href="source-code.html#cb149-113" tabindex="-1"></a>  </span>
<span id="cb149-114"><a href="source-code.html#cb149-114" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb149-115"><a href="source-code.html#cb149-115" tabindex="-1"></a>  </span>
<span id="cb149-116"><a href="source-code.html#cb149-116" tabindex="-1"></a>  </span>
<span id="cb149-117"><a href="source-code.html#cb149-117" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb149-118"><a href="source-code.html#cb149-118" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb149-119"><a href="source-code.html#cb149-119" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb149-120"><a href="source-code.html#cb149-120" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb149-121"><a href="source-code.html#cb149-121" tabindex="-1"></a>    </span>
<span id="cb149-122"><a href="source-code.html#cb149-122" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb149-123"><a href="source-code.html#cb149-123" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb149-124"><a href="source-code.html#cb149-124" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb149-125"><a href="source-code.html#cb149-125" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb149-126"><a href="source-code.html#cb149-126" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb149-127"><a href="source-code.html#cb149-127" tabindex="-1"></a>    }</span>
<span id="cb149-128"><a href="source-code.html#cb149-128" tabindex="-1"></a>    </span>
<span id="cb149-129"><a href="source-code.html#cb149-129" tabindex="-1"></a>    </span>
<span id="cb149-130"><a href="source-code.html#cb149-130" tabindex="-1"></a>    </span>
<span id="cb149-131"><a href="source-code.html#cb149-131" tabindex="-1"></a>    </span>
<span id="cb149-132"><a href="source-code.html#cb149-132" tabindex="-1"></a>    </span>
<span id="cb149-133"><a href="source-code.html#cb149-133" tabindex="-1"></a>  }</span>
<span id="cb149-134"><a href="source-code.html#cb149-134" tabindex="-1"></a>  </span>
<span id="cb149-135"><a href="source-code.html#cb149-135" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb149-136"><a href="source-code.html#cb149-136" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb149-137"><a href="source-code.html#cb149-137" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-138"><a href="source-code.html#cb149-138" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Elbow method&quot;</span>))</span>
<span id="cb149-139"><a href="source-code.html#cb149-139" tabindex="-1"></a>    </span>
<span id="cb149-140"><a href="source-code.html#cb149-140" tabindex="-1"></a>  }</span>
<span id="cb149-141"><a href="source-code.html#cb149-141" tabindex="-1"></a>  </span>
<span id="cb149-142"><a href="source-code.html#cb149-142" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb149-143"><a href="source-code.html#cb149-143" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb149-144"><a href="source-code.html#cb149-144" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb149-145"><a href="source-code.html#cb149-145" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb149-146"><a href="source-code.html#cb149-146" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb149-147"><a href="source-code.html#cb149-147" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb149-148"><a href="source-code.html#cb149-148" tabindex="-1"></a>  </span>
<span id="cb149-149"><a href="source-code.html#cb149-149" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb149-150"><a href="source-code.html#cb149-150" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb149-151"><a href="source-code.html#cb149-151" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb149-152"><a href="source-code.html#cb149-152" tabindex="-1"></a>  </span>
<span id="cb149-153"><a href="source-code.html#cb149-153" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb149-154"><a href="source-code.html#cb149-154" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb149-155"><a href="source-code.html#cb149-155" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-156"><a href="source-code.html#cb149-156" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-157"><a href="source-code.html#cb149-157" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb149-158"><a href="source-code.html#cb149-158" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb149-159"><a href="source-code.html#cb149-159" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb149-160"><a href="source-code.html#cb149-160" tabindex="-1"></a>  </span>
<span id="cb149-161"><a href="source-code.html#cb149-161" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb149-162"><a href="source-code.html#cb149-162" tabindex="-1"></a>  </span>
<span id="cb149-163"><a href="source-code.html#cb149-163" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb149-164"><a href="source-code.html#cb149-164" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb149-165"><a href="source-code.html#cb149-165" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb149-166"><a href="source-code.html#cb149-166" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb149-167"><a href="source-code.html#cb149-167" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb149-168"><a href="source-code.html#cb149-168" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb149-169"><a href="source-code.html#cb149-169" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb149-170"><a href="source-code.html#cb149-170" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb149-171"><a href="source-code.html#cb149-171" tabindex="-1"></a>    </span>
<span id="cb149-172"><a href="source-code.html#cb149-172" tabindex="-1"></a>    </span>
<span id="cb149-173"><a href="source-code.html#cb149-173" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb149-174"><a href="source-code.html#cb149-174" tabindex="-1"></a>    </span>
<span id="cb149-175"><a href="source-code.html#cb149-175" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb149-176"><a href="source-code.html#cb149-176" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb149-177"><a href="source-code.html#cb149-177" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb149-178"><a href="source-code.html#cb149-178" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-179"><a href="source-code.html#cb149-179" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb149-180"><a href="source-code.html#cb149-180" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb149-181"><a href="source-code.html#cb149-181" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb149-182"><a href="source-code.html#cb149-182" tabindex="-1"></a>  </span>
<span id="cb149-183"><a href="source-code.html#cb149-183" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb149-184"><a href="source-code.html#cb149-184" tabindex="-1"></a>  </span>
<span id="cb149-185"><a href="source-code.html#cb149-185" tabindex="-1"></a>  <span class="co">#ggplotly(p)</span></span>
<span id="cb149-186"><a href="source-code.html#cb149-186" tabindex="-1"></a>  </span>
<span id="cb149-187"><a href="source-code.html#cb149-187" tabindex="-1"></a>}</span>
<span id="cb149-188"><a href="source-code.html#cb149-188" tabindex="-1"></a></span>
<span id="cb149-189"><a href="source-code.html#cb149-189" tabindex="-1"></a>draw_pca_double <span class="ot">&lt;-</span></span>
<span id="cb149-190"><a href="source-code.html#cb149-190" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb149-191"><a href="source-code.html#cb149-191" tabindex="-1"></a>           list_files2,</span>
<span id="cb149-192"><a href="source-code.html#cb149-192" tabindex="-1"></a>           list_files3,</span>
<span id="cb149-193"><a href="source-code.html#cb149-193" tabindex="-1"></a>           size,</span>
<span id="cb149-194"><a href="source-code.html#cb149-194" tabindex="-1"></a>           legend_position) {</span>
<span id="cb149-195"><a href="source-code.html#cb149-195" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb149-196"><a href="source-code.html#cb149-196" tabindex="-1"></a>    </span>
<span id="cb149-197"><a href="source-code.html#cb149-197" tabindex="-1"></a>    </span>
<span id="cb149-198"><a href="source-code.html#cb149-198" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-199"><a href="source-code.html#cb149-199" tabindex="-1"></a>      cells,</span>
<span id="cb149-200"><a href="source-code.html#cb149-200" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb149-201"><a href="source-code.html#cb149-201" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb149-202"><a href="source-code.html#cb149-202" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb149-203"><a href="source-code.html#cb149-203" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb149-204"><a href="source-code.html#cb149-204" tabindex="-1"></a>    )</span>
<span id="cb149-205"><a href="source-code.html#cb149-205" tabindex="-1"></a>    </span>
<span id="cb149-206"><a href="source-code.html#cb149-206" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb149-207"><a href="source-code.html#cb149-207" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-208"><a href="source-code.html#cb149-208" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-209"><a href="source-code.html#cb149-209" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb149-210"><a href="source-code.html#cb149-210" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb149-211"><a href="source-code.html#cb149-211" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb149-212"><a href="source-code.html#cb149-212" tabindex="-1"></a>    </span>
<span id="cb149-213"><a href="source-code.html#cb149-213" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb149-214"><a href="source-code.html#cb149-214" tabindex="-1"></a>    </span>
<span id="cb149-215"><a href="source-code.html#cb149-215" tabindex="-1"></a>    </span>
<span id="cb149-216"><a href="source-code.html#cb149-216" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb149-217"><a href="source-code.html#cb149-217" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb149-218"><a href="source-code.html#cb149-218" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb149-219"><a href="source-code.html#cb149-219" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb149-220"><a href="source-code.html#cb149-220" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb149-221"><a href="source-code.html#cb149-221" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb149-222"><a href="source-code.html#cb149-222" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb149-223"><a href="source-code.html#cb149-223" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&#39;#d65c4e&#39;</span>),</span>
<span id="cb149-224"><a href="source-code.html#cb149-224" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence Cycle 1&#39;</span>, <span class="st">&#39;Recurrence Cycle 2&#39;</span>)</span>
<span id="cb149-225"><a href="source-code.html#cb149-225" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb149-226"><a href="source-code.html#cb149-226" tabindex="-1"></a>      </span>
<span id="cb149-227"><a href="source-code.html#cb149-227" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb149-228"><a href="source-code.html#cb149-228" tabindex="-1"></a>      </span>
<span id="cb149-229"><a href="source-code.html#cb149-229" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb149-230"><a href="source-code.html#cb149-230" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb149-231"><a href="source-code.html#cb149-231" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb149-232"><a href="source-code.html#cb149-232" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb149-233"><a href="source-code.html#cb149-233" tabindex="-1"></a>      </span>
<span id="cb149-234"><a href="source-code.html#cb149-234" tabindex="-1"></a>      </span>
<span id="cb149-235"><a href="source-code.html#cb149-235" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb149-236"><a href="source-code.html#cb149-236" tabindex="-1"></a>    <span class="fu">return</span>(p)</span>
<span id="cb149-237"><a href="source-code.html#cb149-237" tabindex="-1"></a>    </span>
<span id="cb149-238"><a href="source-code.html#cb149-238" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb149-239"><a href="source-code.html#cb149-239" tabindex="-1"></a>    </span>
<span id="cb149-240"><a href="source-code.html#cb149-240" tabindex="-1"></a>  }</span>
<span id="cb149-241"><a href="source-code.html#cb149-241" tabindex="-1"></a></span>
<span id="cb149-242"><a href="source-code.html#cb149-242" tabindex="-1"></a></span>
<span id="cb149-243"><a href="source-code.html#cb149-243" tabindex="-1"></a>k_cluster_double <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb149-244"><a href="source-code.html#cb149-244" tabindex="-1"></a>                             list_files2,</span>
<span id="cb149-245"><a href="source-code.html#cb149-245" tabindex="-1"></a>                             list_files3,</span>
<span id="cb149-246"><a href="source-code.html#cb149-246" tabindex="-1"></a>                             </span>
<span id="cb149-247"><a href="source-code.html#cb149-247" tabindex="-1"></a>                             cluster_n,</span>
<span id="cb149-248"><a href="source-code.html#cb149-248" tabindex="-1"></a>                             cols,</span>
<span id="cb149-249"><a href="source-code.html#cb149-249" tabindex="-1"></a>                             labels,</span>
<span id="cb149-250"><a href="source-code.html#cb149-250" tabindex="-1"></a>                             legend_position,</span>
<span id="cb149-251"><a href="source-code.html#cb149-251" tabindex="-1"></a>                             normalize,</span>
<span id="cb149-252"><a href="source-code.html#cb149-252" tabindex="-1"></a>                             <span class="at">return_elbow =</span> F) {</span>
<span id="cb149-253"><a href="source-code.html#cb149-253" tabindex="-1"></a>  </span>
<span id="cb149-254"><a href="source-code.html#cb149-254" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb149-255"><a href="source-code.html#cb149-255" tabindex="-1"></a>  </span>
<span id="cb149-256"><a href="source-code.html#cb149-256" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb149-257"><a href="source-code.html#cb149-257" tabindex="-1"></a>  </span>
<span id="cb149-258"><a href="source-code.html#cb149-258" tabindex="-1"></a>  </span>
<span id="cb149-259"><a href="source-code.html#cb149-259" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-260"><a href="source-code.html#cb149-260" tabindex="-1"></a>    cells,</span>
<span id="cb149-261"><a href="source-code.html#cb149-261" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-262"><a href="source-code.html#cb149-262" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb149-263"><a href="source-code.html#cb149-263" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb149-264"><a href="source-code.html#cb149-264" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-265"><a href="source-code.html#cb149-265" tabindex="-1"></a>  )</span>
<span id="cb149-266"><a href="source-code.html#cb149-266" tabindex="-1"></a>  </span>
<span id="cb149-267"><a href="source-code.html#cb149-267" tabindex="-1"></a>  </span>
<span id="cb149-268"><a href="source-code.html#cb149-268" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-269"><a href="source-code.html#cb149-269" tabindex="-1"></a>    cells,</span>
<span id="cb149-270"><a href="source-code.html#cb149-270" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-271"><a href="source-code.html#cb149-271" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb149-272"><a href="source-code.html#cb149-272" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb149-273"><a href="source-code.html#cb149-273" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-274"><a href="source-code.html#cb149-274" tabindex="-1"></a>  )</span>
<span id="cb149-275"><a href="source-code.html#cb149-275" tabindex="-1"></a>  </span>
<span id="cb149-276"><a href="source-code.html#cb149-276" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-277"><a href="source-code.html#cb149-277" tabindex="-1"></a>    cells,</span>
<span id="cb149-278"><a href="source-code.html#cb149-278" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-279"><a href="source-code.html#cb149-279" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb149-280"><a href="source-code.html#cb149-280" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb149-281"><a href="source-code.html#cb149-281" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-282"><a href="source-code.html#cb149-282" tabindex="-1"></a>  )</span>
<span id="cb149-283"><a href="source-code.html#cb149-283" tabindex="-1"></a>  </span>
<span id="cb149-284"><a href="source-code.html#cb149-284" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-285"><a href="source-code.html#cb149-285" tabindex="-1"></a>    cells,</span>
<span id="cb149-286"><a href="source-code.html#cb149-286" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-287"><a href="source-code.html#cb149-287" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb149-288"><a href="source-code.html#cb149-288" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb149-289"><a href="source-code.html#cb149-289" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-290"><a href="source-code.html#cb149-290" tabindex="-1"></a>  )</span>
<span id="cb149-291"><a href="source-code.html#cb149-291" tabindex="-1"></a>  </span>
<span id="cb149-292"><a href="source-code.html#cb149-292" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb149-293"><a href="source-code.html#cb149-293" tabindex="-1"></a>    cells,</span>
<span id="cb149-294"><a href="source-code.html#cb149-294" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb149-295"><a href="source-code.html#cb149-295" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb149-296"><a href="source-code.html#cb149-296" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb149-297"><a href="source-code.html#cb149-297" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb149-298"><a href="source-code.html#cb149-298" tabindex="-1"></a>  )</span>
<span id="cb149-299"><a href="source-code.html#cb149-299" tabindex="-1"></a>  </span>
<span id="cb149-300"><a href="source-code.html#cb149-300" tabindex="-1"></a>  </span>
<span id="cb149-301"><a href="source-code.html#cb149-301" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb149-302"><a href="source-code.html#cb149-302" tabindex="-1"></a>  </span>
<span id="cb149-303"><a href="source-code.html#cb149-303" tabindex="-1"></a>  </span>
<span id="cb149-304"><a href="source-code.html#cb149-304" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb149-305"><a href="source-code.html#cb149-305" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb149-306"><a href="source-code.html#cb149-306" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb149-307"><a href="source-code.html#cb149-307" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb149-308"><a href="source-code.html#cb149-308" tabindex="-1"></a>    </span>
<span id="cb149-309"><a href="source-code.html#cb149-309" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb149-310"><a href="source-code.html#cb149-310" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb149-311"><a href="source-code.html#cb149-311" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb149-312"><a href="source-code.html#cb149-312" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb149-313"><a href="source-code.html#cb149-313" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb149-314"><a href="source-code.html#cb149-314" tabindex="-1"></a>    }</span>
<span id="cb149-315"><a href="source-code.html#cb149-315" tabindex="-1"></a>    </span>
<span id="cb149-316"><a href="source-code.html#cb149-316" tabindex="-1"></a>    </span>
<span id="cb149-317"><a href="source-code.html#cb149-317" tabindex="-1"></a>    </span>
<span id="cb149-318"><a href="source-code.html#cb149-318" tabindex="-1"></a>    </span>
<span id="cb149-319"><a href="source-code.html#cb149-319" tabindex="-1"></a>    </span>
<span id="cb149-320"><a href="source-code.html#cb149-320" tabindex="-1"></a>  }</span>
<span id="cb149-321"><a href="source-code.html#cb149-321" tabindex="-1"></a>  </span>
<span id="cb149-322"><a href="source-code.html#cb149-322" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb149-323"><a href="source-code.html#cb149-323" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb149-324"><a href="source-code.html#cb149-324" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-325"><a href="source-code.html#cb149-325" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Silhouette method&quot;</span>))</span>
<span id="cb149-326"><a href="source-code.html#cb149-326" tabindex="-1"></a>    </span>
<span id="cb149-327"><a href="source-code.html#cb149-327" tabindex="-1"></a>  }</span>
<span id="cb149-328"><a href="source-code.html#cb149-328" tabindex="-1"></a>  </span>
<span id="cb149-329"><a href="source-code.html#cb149-329" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb149-330"><a href="source-code.html#cb149-330" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb149-331"><a href="source-code.html#cb149-331" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb149-332"><a href="source-code.html#cb149-332" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb149-333"><a href="source-code.html#cb149-333" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb149-334"><a href="source-code.html#cb149-334" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb149-335"><a href="source-code.html#cb149-335" tabindex="-1"></a>  </span>
<span id="cb149-336"><a href="source-code.html#cb149-336" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb149-337"><a href="source-code.html#cb149-337" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb149-338"><a href="source-code.html#cb149-338" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb149-339"><a href="source-code.html#cb149-339" tabindex="-1"></a>  </span>
<span id="cb149-340"><a href="source-code.html#cb149-340" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb149-341"><a href="source-code.html#cb149-341" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb149-342"><a href="source-code.html#cb149-342" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-343"><a href="source-code.html#cb149-343" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb149-344"><a href="source-code.html#cb149-344" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb149-345"><a href="source-code.html#cb149-345" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb149-346"><a href="source-code.html#cb149-346" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb149-347"><a href="source-code.html#cb149-347" tabindex="-1"></a>  </span>
<span id="cb149-348"><a href="source-code.html#cb149-348" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb149-349"><a href="source-code.html#cb149-349" tabindex="-1"></a>  </span>
<span id="cb149-350"><a href="source-code.html#cb149-350" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb149-351"><a href="source-code.html#cb149-351" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb149-352"><a href="source-code.html#cb149-352" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb149-353"><a href="source-code.html#cb149-353" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb149-354"><a href="source-code.html#cb149-354" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb149-355"><a href="source-code.html#cb149-355" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb149-356"><a href="source-code.html#cb149-356" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb149-357"><a href="source-code.html#cb149-357" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb149-358"><a href="source-code.html#cb149-358" tabindex="-1"></a>    </span>
<span id="cb149-359"><a href="source-code.html#cb149-359" tabindex="-1"></a>    </span>
<span id="cb149-360"><a href="source-code.html#cb149-360" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb149-361"><a href="source-code.html#cb149-361" tabindex="-1"></a>    </span>
<span id="cb149-362"><a href="source-code.html#cb149-362" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb149-363"><a href="source-code.html#cb149-363" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb149-364"><a href="source-code.html#cb149-364" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb149-365"><a href="source-code.html#cb149-365" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-366"><a href="source-code.html#cb149-366" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb149-367"><a href="source-code.html#cb149-367" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb149-368"><a href="source-code.html#cb149-368" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb149-369"><a href="source-code.html#cb149-369" tabindex="-1"></a>  </span>
<span id="cb149-370"><a href="source-code.html#cb149-370" tabindex="-1"></a>  p</span>
<span id="cb149-371"><a href="source-code.html#cb149-371" tabindex="-1"></a>  </span>
<span id="cb149-372"><a href="source-code.html#cb149-372" tabindex="-1"></a>}</span>
<span id="cb149-373"><a href="source-code.html#cb149-373" tabindex="-1"></a></span>
<span id="cb149-374"><a href="source-code.html#cb149-374" tabindex="-1"></a>  </span>
<span id="cb149-375"><a href="source-code.html#cb149-375" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions<a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="source-code.html#cb150-1" tabindex="-1"></a>extract_other_cells <span class="ot">&lt;-</span> <span class="cf">function</span>(kra_name, list_of_subclones) {</span>
<span id="cb150-2"><a href="source-code.html#cb150-2" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(man_select_files_edivisive) <span class="sc">==</span> kra_name)</span>
<span id="cb150-3"><a href="source-code.html#cb150-3" tabindex="-1"></a>  all <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(man_select_files_edivisive[[i]],  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>) </span>
<span id="cb150-4"><a href="source-code.html#cb150-4" tabindex="-1"></a>  rest <span class="ot">&lt;-</span> all[<span class="fu">which</span>(<span class="sc">!</span>all <span class="sc">%in%</span> list_of_subclones)]</span>
<span id="cb150-5"><a href="source-code.html#cb150-5" tabindex="-1"></a>  <span class="fu">return</span>(rest)</span>
<span id="cb150-6"><a href="source-code.html#cb150-6" tabindex="-1"></a>}</span>
<span id="cb150-7"><a href="source-code.html#cb150-7" tabindex="-1"></a></span>
<span id="cb150-8"><a href="source-code.html#cb150-8" tabindex="-1"></a></span>
<span id="cb150-9"><a href="source-code.html#cb150-9" tabindex="-1"></a>extract_square <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1, list_files2, square, x, y) {</span>
<span id="cb150-10"><a href="source-code.html#cb150-10" tabindex="-1"></a>  <span class="co"># extracts all cells that lie within a square of a pca plot defined by </span></span>
<span id="cb150-11"><a href="source-code.html#cb150-11" tabindex="-1"></a>  <span class="co"># user. For example square c(0.1, 0.0) will extract all cells with</span></span>
<span id="cb150-12"><a href="source-code.html#cb150-12" tabindex="-1"></a>  <span class="co"># pc1 lower or higher that 0.1 and pc2 lower or higher than 0.0. User has to rearange functon</span></span>
<span id="cb150-13"><a href="source-code.html#cb150-13" tabindex="-1"></a>  <span class="co"># if he wants x lower or higher than 0.1</span></span>
<span id="cb150-14"><a href="source-code.html#cb150-14" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb150-15"><a href="source-code.html#cb150-15" tabindex="-1"></a>  </span>
<span id="cb150-16"><a href="source-code.html#cb150-16" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(cells,</span>
<span id="cb150-17"><a href="source-code.html#cb150-17" tabindex="-1"></a>                   <span class="at">colorBy =</span> classes, </span>
<span id="cb150-18"><a href="source-code.html#cb150-18" tabindex="-1"></a>                   <span class="at">PC1=</span><span class="dv">1</span>, </span>
<span id="cb150-19"><a href="source-code.html#cb150-19" tabindex="-1"></a>                   <span class="at">PC2=</span><span class="dv">2</span>,</span>
<span id="cb150-20"><a href="source-code.html#cb150-20" tabindex="-1"></a>                   <span class="at">plot =</span> F)</span>
<span id="cb150-21"><a href="source-code.html#cb150-21" tabindex="-1"></a>  </span>
<span id="cb150-22"><a href="source-code.html#cb150-22" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>)</span>
<span id="cb150-23"><a href="source-code.html#cb150-23" tabindex="-1"></a>  </span>
<span id="cb150-24"><a href="source-code.html#cb150-24" tabindex="-1"></a>  </span>
<span id="cb150-25"><a href="source-code.html#cb150-25" tabindex="-1"></a>  <span class="co"># extracting dataframe with correct square</span></span>
<span id="cb150-26"><a href="source-code.html#cb150-26" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb150-27"><a href="source-code.html#cb150-27" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb150-28"><a href="source-code.html#cb150-28" tabindex="-1"></a>  }</span>
<span id="cb150-29"><a href="source-code.html#cb150-29" tabindex="-1"></a>  </span>
<span id="cb150-30"><a href="source-code.html#cb150-30" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb150-31"><a href="source-code.html#cb150-31" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb150-32"><a href="source-code.html#cb150-32" tabindex="-1"></a>  }</span>
<span id="cb150-33"><a href="source-code.html#cb150-33" tabindex="-1"></a>  </span>
<span id="cb150-34"><a href="source-code.html#cb150-34" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb150-35"><a href="source-code.html#cb150-35" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb150-36"><a href="source-code.html#cb150-36" tabindex="-1"></a>  }</span>
<span id="cb150-37"><a href="source-code.html#cb150-37" tabindex="-1"></a>  </span>
<span id="cb150-38"><a href="source-code.html#cb150-38" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb150-39"><a href="source-code.html#cb150-39" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb150-40"><a href="source-code.html#cb150-40" tabindex="-1"></a>  }</span>
<span id="cb150-41"><a href="source-code.html#cb150-41" tabindex="-1"></a>  </span>
<span id="cb150-42"><a href="source-code.html#cb150-42" tabindex="-1"></a>  <span class="co"># numbers is: </span></span>
<span id="cb150-43"><a href="source-code.html#cb150-43" tabindex="-1"></a>  cell_id <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)  </span>
<span id="cb150-44"><a href="source-code.html#cb150-44" tabindex="-1"></a>  </span>
<span id="cb150-45"><a href="source-code.html#cb150-45" tabindex="-1"></a>  <span class="fu">return</span>(cell_id)</span>
<span id="cb150-46"><a href="source-code.html#cb150-46" tabindex="-1"></a>  </span>
<span id="cb150-47"><a href="source-code.html#cb150-47" tabindex="-1"></a>}</span>
<span id="cb150-48"><a href="source-code.html#cb150-48" tabindex="-1"></a></span>
<span id="cb150-49"><a href="source-code.html#cb150-49" tabindex="-1"></a>l_g <span class="ot">&lt;-</span> <span class="cf">function</span>(x, kra_num, list_cells) {</span>
<span id="cb150-50"><a href="source-code.html#cb150-50" tabindex="-1"></a>  select <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="fu">paste0</span>(kra_num,<span class="st">&#39;_&#39;</span>,x,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>), list_cells)</span>
<span id="cb150-51"><a href="source-code.html#cb150-51" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(select)</span>
<span id="cb150-52"><a href="source-code.html#cb150-52" tabindex="-1"></a>  <span class="fu">return</span>(list_cells[i])</span>
<span id="cb150-53"><a href="source-code.html#cb150-53" tabindex="-1"></a>  </span>
<span id="cb150-54"><a href="source-code.html#cb150-54" tabindex="-1"></a>  </span>
<span id="cb150-55"><a href="source-code.html#cb150-55" tabindex="-1"></a>}</span>
<span id="cb150-56"><a href="source-code.html#cb150-56" tabindex="-1"></a></span>
<span id="cb150-57"><a href="source-code.html#cb150-57" tabindex="-1"></a></span>
<span id="cb150-58"><a href="source-code.html#cb150-58" tabindex="-1"></a>subclone_classes_for_genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(list_of_subclones, cline) {</span>
<span id="cb150-59"><a href="source-code.html#cb150-59" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb150-60"><a href="source-code.html#cb150-60" tabindex="-1"></a>  <span class="co">#&#39; list of subclones is the full list of rda files that have been assigned a subclone_classes_for_genomeheatmap</span></span>
<span id="cb150-61"><a href="source-code.html#cb150-61" tabindex="-1"></a>  <span class="co">#&#39; cline is the line (hub005, hub183 etc.)</span></span>
<span id="cb150-62"><a href="source-code.html#cb150-62" tabindex="-1"></a>  <span class="co">#&#39; rad is either prerad or rad</span></span>
<span id="cb150-63"><a href="source-code.html#cb150-63" tabindex="-1"></a>  <span class="co">#&#39; @return </span></span>
<span id="cb150-64"><a href="source-code.html#cb150-64" tabindex="-1"></a>  <span class="co">#&#39; returns a vector with that assigns a subclone</span></span>
<span id="cb150-65"><a href="source-code.html#cb150-65" tabindex="-1"></a>  </span>
<span id="cb150-66"><a href="source-code.html#cb150-66" tabindex="-1"></a>  extract_subclones <span class="ot">&lt;-</span> list_of_subclones[<span class="fu">grepl</span>(<span class="fu">names</span>(list_of_subclones), <span class="at">pattern =</span> <span class="fu">paste0</span>(cline))]</span>
<span id="cb150-67"><a href="source-code.html#cb150-67" tabindex="-1"></a>  </span>
<span id="cb150-68"><a href="source-code.html#cb150-68" tabindex="-1"></a>  size_subclones <span class="ot">&lt;-</span> <span class="fu">lapply</span>(extract_subclones, length)</span>
<span id="cb150-69"><a href="source-code.html#cb150-69" tabindex="-1"></a>  </span>
<span id="cb150-70"><a href="source-code.html#cb150-70" tabindex="-1"></a>  classes <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">names</span>(size_subclones), size_subclones)</span>
<span id="cb150-71"><a href="source-code.html#cb150-71" tabindex="-1"></a>  </span>
<span id="cb150-72"><a href="source-code.html#cb150-72" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">path =</span> <span class="fu">unlist</span>(extract_subclones),</span>
<span id="cb150-73"><a href="source-code.html#cb150-73" tabindex="-1"></a>                   <span class="at">class =</span> classes)</span>
<span id="cb150-74"><a href="source-code.html#cb150-74" tabindex="-1"></a>  </span>
<span id="cb150-75"><a href="source-code.html#cb150-75" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb150-76"><a href="source-code.html#cb150-76" tabindex="-1"></a>  </span>
<span id="cb150-77"><a href="source-code.html#cb150-77" tabindex="-1"></a>  </span>
<span id="cb150-78"><a href="source-code.html#cb150-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genomeheatmap-script-genomeheatmap-function" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> plot_genomeheatmap script: ‘genomeheatmap’ function<a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="source-code.html#cb151-1" tabindex="-1"></a>genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(selected.files, </span>
<span id="cb151-2"><a href="source-code.html#cb151-2" tabindex="-1"></a>                          path, </span>
<span id="cb151-3"><a href="source-code.html#cb151-3" tabindex="-1"></a>                          <span class="at">classes_daan =</span> <span class="cn">NULL</span>, </span>
<span id="cb151-4"><a href="source-code.html#cb151-4" tabindex="-1"></a>                          <span class="at">class.col =</span> <span class="cn">NULL</span>, </span>
<span id="cb151-5"><a href="source-code.html#cb151-5" tabindex="-1"></a>                          <span class="at">dendogram =</span> F,</span>
<span id="cb151-6"><a href="source-code.html#cb151-6" tabindex="-1"></a>                          <span class="at">daan_colours =</span> T,</span>
<span id="cb151-7"><a href="source-code.html#cb151-7" tabindex="-1"></a>                          name) {</span>
<span id="cb151-8"><a href="source-code.html#cb151-8" tabindex="-1"></a>  <span class="co"># This code was adapted from the AneuFinder package</span></span>
<span id="cb151-9"><a href="source-code.html#cb151-9" tabindex="-1"></a>  <span class="co"># plots and saves genome heatmap of selected.files</span></span>
<span id="cb151-10"><a href="source-code.html#cb151-10" tabindex="-1"></a>  <span class="co"># @param </span></span>
<span id="cb151-11"><a href="source-code.html#cb151-11" tabindex="-1"></a>  <span class="co"># selected.files: vector of location of selected files.</span></span>
<span id="cb151-12"><a href="source-code.html#cb151-12" tabindex="-1"></a>  <span class="co"># output path: path were plots should be saved. </span></span>
<span id="cb151-13"><a href="source-code.html#cb151-13" tabindex="-1"></a>  <span class="co"># name: how you want to name the plot</span></span>
<span id="cb151-14"><a href="source-code.html#cb151-14" tabindex="-1"></a>  </span>
<span id="cb151-15"><a href="source-code.html#cb151-15" tabindex="-1"></a>  <span class="co"># get platename</span></span>
<span id="cb151-16"><a href="source-code.html#cb151-16" tabindex="-1"></a>  platename <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(selected.files[<span class="dv">1</span>],  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb151-17"><a href="source-code.html#cb151-17" tabindex="-1"></a>  </span>
<span id="cb151-18"><a href="source-code.html#cb151-18" tabindex="-1"></a>  <span class="co"># AneuFinder functions: </span></span>
<span id="cb151-19"><a href="source-code.html#cb151-19" tabindex="-1"></a>  startTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb151-20"><a href="source-code.html#cb151-20" tabindex="-1"></a>    </span>
<span id="cb151-21"><a href="source-code.html#cb151-21" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">paste0</span>(..., <span class="at">collapse=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb151-22"><a href="source-code.html#cb151-22" tabindex="-1"></a>    <span class="fu">message</span>(x, <span class="at">appendLF=</span><span class="cn">FALSE</span>)</span>
<span id="cb151-23"><a href="source-code.html#cb151-23" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">proc.time</span>()</span>
<span id="cb151-24"><a href="source-code.html#cb151-24" tabindex="-1"></a>    <span class="fu">return</span>(ptm)</span>
<span id="cb151-25"><a href="source-code.html#cb151-25" tabindex="-1"></a>    </span>
<span id="cb151-26"><a href="source-code.html#cb151-26" tabindex="-1"></a>  }</span>
<span id="cb151-27"><a href="source-code.html#cb151-27" tabindex="-1"></a>  </span>
<span id="cb151-28"><a href="source-code.html#cb151-28" tabindex="-1"></a>  transCoord <span class="ot">&lt;-</span> <span class="cf">function</span> (gr) </span>
<span id="cb151-29"><a href="source-code.html#cb151-29" tabindex="-1"></a>  {</span>
<span id="cb151-30"><a href="source-code.html#cb151-30" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(gr)))</span>
<span id="cb151-31"><a href="source-code.html#cb151-31" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb151-32"><a href="source-code.html#cb151-32" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(gr)</span>
<span id="cb151-33"><a href="source-code.html#cb151-33" tabindex="-1"></a>    gr<span class="sc">$</span>start.genome <span class="ot">&lt;-</span> <span class="fu">start</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb151-34"><a href="source-code.html#cb151-34" tabindex="-1"></a>    gr<span class="sc">$</span>end.genome <span class="ot">&lt;-</span> <span class="fu">end</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb151-35"><a href="source-code.html#cb151-35" tabindex="-1"></a>    <span class="fu">return</span>(gr)</span>
<span id="cb151-36"><a href="source-code.html#cb151-36" tabindex="-1"></a>  }</span>
<span id="cb151-37"><a href="source-code.html#cb151-37" tabindex="-1"></a>  </span>
<span id="cb151-38"><a href="source-code.html#cb151-38" tabindex="-1"></a>  stopTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(ptm) {</span>
<span id="cb151-39"><a href="source-code.html#cb151-39" tabindex="-1"></a>    </span>
<span id="cb151-40"><a href="source-code.html#cb151-40" tabindex="-1"></a>    time <span class="ot">&lt;-</span> <span class="fu">proc.time</span>() <span class="sc">-</span> ptm</span>
<span id="cb151-41"><a href="source-code.html#cb151-41" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot; &quot;</span>, <span class="fu">round</span>(time[<span class="dv">3</span>],<span class="dv">2</span>), <span class="st">&quot;s&quot;</span>)</span>
<span id="cb151-42"><a href="source-code.html#cb151-42" tabindex="-1"></a>    </span>
<span id="cb151-43"><a href="source-code.html#cb151-43" tabindex="-1"></a>  }</span>
<span id="cb151-44"><a href="source-code.html#cb151-44" tabindex="-1"></a>  </span>
<span id="cb151-45"><a href="source-code.html#cb151-45" tabindex="-1"></a>  initializeStates <span class="ot">&lt;-</span> <span class="cf">function</span> (states) </span>
<span id="cb151-46"><a href="source-code.html#cb151-46" tabindex="-1"></a>  {</span>
<span id="cb151-47"><a href="source-code.html#cb151-47" tabindex="-1"></a>    somy.states <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;somy&quot;</span>, states, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb151-48"><a href="source-code.html#cb151-48" tabindex="-1"></a>    somy.numbers <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(somy.states, <span class="st">&quot;-somy&quot;</span>), </span>
<span id="cb151-49"><a href="source-code.html#cb151-49" tabindex="-1"></a>                                      <span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>))</span>
<span id="cb151-50"><a href="source-code.html#cb151-50" tabindex="-1"></a>    <span class="fu">names</span>(somy.numbers) <span class="ot">&lt;-</span> somy.states</span>
<span id="cb151-51"><a href="source-code.html#cb151-51" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">&quot;zero-inflation&quot;</span> <span class="sc">%in%</span> states) {</span>
<span id="cb151-52"><a href="source-code.html#cb151-52" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">0</span>, somy.numbers)</span>
<span id="cb151-53"><a href="source-code.html#cb151-53" tabindex="-1"></a>    }</span>
<span id="cb151-54"><a href="source-code.html#cb151-54" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb151-55"><a href="source-code.html#cb151-55" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> somy.numbers</span>
<span id="cb151-56"><a href="source-code.html#cb151-56" tabindex="-1"></a>    }</span>
<span id="cb151-57"><a href="source-code.html#cb151-57" tabindex="-1"></a>    levels.distributions <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;delta&quot;</span>, <span class="st">&quot;dgeom&quot;</span>, <span class="st">&quot;dnbinom&quot;</span>, <span class="st">&quot;dbinom&quot;</span>)</span>
<span id="cb151-58"><a href="source-code.html#cb151-58" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(states))</span>
<span id="cb151-59"><a href="source-code.html#cb151-59" tabindex="-1"></a>    <span class="fu">names</span>(distributions) <span class="ot">&lt;-</span> states</span>
<span id="cb151-60"><a href="source-code.html#cb151-60" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;zero-inflation&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;delta&quot;</span></span>
<span id="cb151-61"><a href="source-code.html#cb151-61" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;0-somy&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;dgeom&quot;</span></span>
<span id="cb151-62"><a href="source-code.html#cb151-62" tabindex="-1"></a>    distributions[(states <span class="sc">!=</span> <span class="st">&quot;zero-inflation&quot;</span>) <span class="sc">&amp;</span> (states <span class="sc">!=</span> <span class="st">&quot;0-somy&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;dnbinom&quot;</span></span>
<span id="cb151-63"><a href="source-code.html#cb151-63" tabindex="-1"></a>    states <span class="ot">&lt;-</span> <span class="fu">factor</span>(states, <span class="at">levels =</span> states)</span>
<span id="cb151-64"><a href="source-code.html#cb151-64" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">factor</span>(distributions, <span class="at">levels =</span> levels.distributions)</span>
<span id="cb151-65"><a href="source-code.html#cb151-65" tabindex="-1"></a>    l <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">states =</span> states, <span class="at">distributions =</span> distributions, </span>
<span id="cb151-66"><a href="source-code.html#cb151-66" tabindex="-1"></a>              <span class="at">multiplicity =</span> multiplicity)</span>
<span id="cb151-67"><a href="source-code.html#cb151-67" tabindex="-1"></a>    <span class="fu">return</span>(l)</span>
<span id="cb151-68"><a href="source-code.html#cb151-68" tabindex="-1"></a>  }</span>
<span id="cb151-69"><a href="source-code.html#cb151-69" tabindex="-1"></a>  </span>
<span id="cb151-70"><a href="source-code.html#cb151-70" tabindex="-1"></a>  <span class="co">#  colours</span></span>
<span id="cb151-71"><a href="source-code.html#cb151-71" tabindex="-1"></a>  <span class="cf">if</span> (daan_colours) {</span>
<span id="cb151-72"><a href="source-code.html#cb151-72" tabindex="-1"></a>    stateColors <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">states =</span> <span class="fu">c</span>(<span class="st">&quot;zero-inflation&quot;</span>, <span class="fu">paste0</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>, <span class="st">&quot;-somy&quot;</span>),</span>
<span id="cb151-73"><a href="source-code.html#cb151-73" tabindex="-1"></a>                                       <span class="st">&quot;total&quot;</span>)) {</span>
<span id="cb151-74"><a href="source-code.html#cb151-74" tabindex="-1"></a>      </span>
<span id="cb151-75"><a href="source-code.html#cb151-75" tabindex="-1"></a>      state.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>, <span class="st">`</span><span class="at">0-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>,</span>
<span id="cb151-76"><a href="source-code.html#cb151-76" tabindex="-1"></a>                        <span class="st">`</span><span class="at">1-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3787BA&quot;</span>, <span class="st">`</span><span class="at">2-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#95B8C5&quot;</span>,</span>
<span id="cb151-77"><a href="source-code.html#cb151-77" tabindex="-1"></a>                        <span class="st">`</span><span class="at">3-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#F0ECEB&quot;</span>, <span class="st">`</span><span class="at">4-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#D7A290&quot;</span>, <span class="st">`</span><span class="at">5-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#BF583B&quot;</span>,</span>
<span id="cb151-78"><a href="source-code.html#cb151-78" tabindex="-1"></a>                        <span class="st">`</span><span class="at">6-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#8D1128&quot;</span>, <span class="st">`</span><span class="at">7-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3C0912&quot;</span>, <span class="st">`</span><span class="at">8-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb151-79"><a href="source-code.html#cb151-79" tabindex="-1"></a>                        <span class="at">total =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb151-80"><a href="source-code.html#cb151-80" tabindex="-1"></a>      </span>
<span id="cb151-81"><a href="source-code.html#cb151-81" tabindex="-1"></a>      states.with.color <span class="ot">&lt;-</span> <span class="fu">intersect</span>(states, <span class="fu">names</span>(state.colors))</span>
<span id="cb151-82"><a href="source-code.html#cb151-82" tabindex="-1"></a>      cols <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;black&quot;</span>, <span class="fu">length</span>(states))</span>
<span id="cb151-83"><a href="source-code.html#cb151-83" tabindex="-1"></a>      <span class="fu">names</span>(cols) <span class="ot">&lt;-</span> states</span>
<span id="cb151-84"><a href="source-code.html#cb151-84" tabindex="-1"></a>      cols[states.with.color] <span class="ot">&lt;-</span> state.colors[states.with.color]</span>
<span id="cb151-85"><a href="source-code.html#cb151-85" tabindex="-1"></a>      <span class="fu">return</span>(cols)</span>
<span id="cb151-86"><a href="source-code.html#cb151-86" tabindex="-1"></a>    }</span>
<span id="cb151-87"><a href="source-code.html#cb151-87" tabindex="-1"></a>    </span>
<span id="cb151-88"><a href="source-code.html#cb151-88" tabindex="-1"></a>  }</span>
<span id="cb151-89"><a href="source-code.html#cb151-89" tabindex="-1"></a></span>
<span id="cb151-90"><a href="source-code.html#cb151-90" tabindex="-1"></a>  </span>
<span id="cb151-91"><a href="source-code.html#cb151-91" tabindex="-1"></a>  </span>
<span id="cb151-92"><a href="source-code.html#cb151-92" tabindex="-1"></a>  <span class="co"># heatmapgenomewide adapted form AneuFinder package</span></span>
<span id="cb151-93"><a href="source-code.html#cb151-93" tabindex="-1"></a>  heatmapGenomewide_daan <span class="ot">&lt;-</span> <span class="cf">function</span> (hmms, <span class="at">ylabels =</span> <span class="cn">NULL</span>, classes, <span class="at">reorder.by.class =</span> <span class="cn">TRUE</span>, </span>
<span id="cb151-94"><a href="source-code.html#cb151-94" tabindex="-1"></a>                                      classes.color, <span class="at">file =</span> <span class="cn">NULL</span>, <span class="at">cluster =</span> <span class="cn">TRUE</span>, <span class="at">plot.breakpoints =</span> <span class="cn">FALSE</span>, </span>
<span id="cb151-95"><a href="source-code.html#cb151-95" tabindex="-1"></a>                                      <span class="at">hotspots =</span> <span class="cn">NULL</span>, <span class="at">exclude.regions =</span> <span class="cn">NULL</span>) </span>
<span id="cb151-96"><a href="source-code.html#cb151-96" tabindex="-1"></a>  {</span>
<span id="cb151-97"><a href="source-code.html#cb151-97" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb151-98"><a href="source-code.html#cb151-98" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(ylabels) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb151-99"><a href="source-code.html#cb151-99" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(ylabels) must equal length(hmms)&quot;</span>)</span>
<span id="cb151-100"><a href="source-code.html#cb151-100" tabindex="-1"></a>      }</span>
<span id="cb151-101"><a href="source-code.html#cb151-101" tabindex="-1"></a>    }</span>
<span id="cb151-102"><a href="source-code.html#cb151-102" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb151-103"><a href="source-code.html#cb151-103" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(classes) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb151-104"><a href="source-code.html#cb151-104" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(classes) must equal length(hmms)&quot;</span>)</span>
<span id="cb151-105"><a href="source-code.html#cb151-105" tabindex="-1"></a>      }</span>
<span id="cb151-106"><a href="source-code.html#cb151-106" tabindex="-1"></a>    }</span>
<span id="cb151-107"><a href="source-code.html#cb151-107" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(classes.color) <span class="sc">!=</span> <span class="fu">length</span>(<span class="fu">unique</span>(classes))) {</span>
<span id="cb151-108"><a href="source-code.html#cb151-108" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;&#39;classes.color&#39; must have the same length as unique(classes)&quot;</span>)</span>
<span id="cb151-109"><a href="source-code.html#cb151-109" tabindex="-1"></a>    }</span>
<span id="cb151-110"><a href="source-code.html#cb151-110" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">names</span>(classes.color))) {</span>
<span id="cb151-111"><a href="source-code.html#cb151-111" tabindex="-1"></a>      <span class="fu">names</span>(classes.color) <span class="ot">&lt;-</span> <span class="fu">unique</span>(classes)</span>
<span id="cb151-112"><a href="source-code.html#cb151-112" tabindex="-1"></a>    }</span>
<span id="cb151-113"><a href="source-code.html#cb151-113" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">setequal</span>(<span class="fu">names</span>(classes.color), <span class="fu">unique</span>(classes))) {</span>
<span id="cb151-114"><a href="source-code.html#cb151-114" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;The names of &#39;classes.color&#39; must be equal to the unique elements in &#39;classes&#39;&quot;</span>)</span>
<span id="cb151-115"><a href="source-code.html#cb151-115" tabindex="-1"></a>    }</span>
<span id="cb151-116"><a href="source-code.html#cb151-116" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(hmms) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> cluster <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb151-117"><a href="source-code.html#cb151-117" tabindex="-1"></a>      cluster <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb151-118"><a href="source-code.html#cb151-118" tabindex="-1"></a>      <span class="fu">warning</span>(<span class="st">&quot;Cannot do clustering because only one object was given.&quot;</span>)</span>
<span id="cb151-119"><a href="source-code.html#cb151-119" tabindex="-1"></a>    }</span>
<span id="cb151-120"><a href="source-code.html#cb151-120" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> <span class="fu">loadFromFiles</span>(hmms, <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb151-121"><a href="source-code.html#cb151-121" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">sapply</span>(hmms, <span class="st">&quot;[[&quot;</span>, <span class="st">&quot;ID&quot;</span>))</span>
<span id="cb151-122"><a href="source-code.html#cb151-122" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb151-123"><a href="source-code.html#cb151-123" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb151-124"><a href="source-code.html#cb151-124" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb151-125"><a href="source-code.html#cb151-125" tabindex="-1"></a>    }</span>
<span id="cb151-126"><a href="source-code.html#cb151-126" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb151-127"><a href="source-code.html#cb151-127" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ylabels)</span>
<span id="cb151-128"><a href="source-code.html#cb151-128" tabindex="-1"></a>    }</span>
<span id="cb151-129"><a href="source-code.html#cb151-129" tabindex="-1"></a>    class.data<span class="sc">$</span>class <span class="ot">&lt;-</span> classes</span>
<span id="cb151-130"><a href="source-code.html#cb151-130" tabindex="-1"></a>    mapping <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ylabel</span>
<span id="cb151-131"><a href="source-code.html#cb151-131" tabindex="-1"></a>    <span class="fu">names</span>(mapping) <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ID</span>
<span id="cb151-132"><a href="source-code.html#cb151-132" tabindex="-1"></a>    <span class="cf">if</span> (reorder.by.class) {</span>
<span id="cb151-133"><a href="source-code.html#cb151-133" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">classes =</span> classes, </span>
<span id="cb151-134"><a href="source-code.html#cb151-134" tabindex="-1"></a>                        <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb151-135"><a href="source-code.html#cb151-135" tabindex="-1"></a>    }</span>
<span id="cb151-136"><a href="source-code.html#cb151-136" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb151-137"><a href="source-code.html#cb151-137" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb151-138"><a href="source-code.html#cb151-138" tabindex="-1"></a>    }</span>
<span id="cb151-139"><a href="source-code.html#cb151-139" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> hmms[cl<span class="sc">$</span>IDorder]</span>
<span id="cb151-140"><a href="source-code.html#cb151-140" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> class.data[cl<span class="sc">$</span>IDorder, ]</span>
<span id="cb151-141"><a href="source-code.html#cb151-141" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb151-142"><a href="source-code.html#cb151-142" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb151-143"><a href="source-code.html#cb151-143" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb151-144"><a href="source-code.html#cb151-144" tabindex="-1"></a>      hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb151-145"><a href="source-code.html#cb151-145" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>segments)) {</span>
<span id="cb151-146"><a href="source-code.html#cb151-146" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb151-147"><a href="source-code.html#cb151-147" tabindex="-1"></a>      }</span>
<span id="cb151-148"><a href="source-code.html#cb151-148" tabindex="-1"></a>      <span class="cf">else</span> {</span>
<span id="cb151-149"><a href="source-code.html#cb151-149" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>segments</span>
<span id="cb151-150"><a href="source-code.html#cb151-150" tabindex="-1"></a>      }</span>
<span id="cb151-151"><a href="source-code.html#cb151-151" tabindex="-1"></a>    }</span>
<span id="cb151-152"><a href="source-code.html#cb151-152" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb151-153"><a href="source-code.html#cb151-153" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb151-154"><a href="source-code.html#cb151-154" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb151-155"><a href="source-code.html#cb151-155" tabindex="-1"></a>        hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb151-156"><a href="source-code.html#cb151-156" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>breakpoints)) {</span>
<span id="cb151-157"><a href="source-code.html#cb151-157" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb151-158"><a href="source-code.html#cb151-158" tabindex="-1"></a>        }</span>
<span id="cb151-159"><a href="source-code.html#cb151-159" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb151-160"><a href="source-code.html#cb151-160" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>breakpoints</span>
<span id="cb151-161"><a href="source-code.html#cb151-161" tabindex="-1"></a>        }</span>
<span id="cb151-162"><a href="source-code.html#cb151-162" tabindex="-1"></a>      }</span>
<span id="cb151-163"><a href="source-code.html#cb151-163" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(breakpoints) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb151-164"><a href="source-code.html#cb151-164" tabindex="-1"></a>        plot.breakpoints <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb151-165"><a href="source-code.html#cb151-165" tabindex="-1"></a>      }</span>
<span id="cb151-166"><a href="source-code.html#cb151-166" tabindex="-1"></a>    }</span>
<span id="cb151-167"><a href="source-code.html#cb151-167" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Transforming coordinates ...&quot;</span>)</span>
<span id="cb151-168"><a href="source-code.html#cb151-168" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(segments.list, transCoord)</span>
<span id="cb151-169"><a href="source-code.html#cb151-169" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb151-170"><a href="source-code.html#cb151-170" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(breakpoints, transCoord)</span>
<span id="cb151-171"><a href="source-code.html#cb151-171" tabindex="-1"></a>    }</span>
<span id="cb151-172"><a href="source-code.html#cb151-172" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb151-173"><a href="source-code.html#cb151-173" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Making the plot ...&quot;</span>)</span>
<span id="cb151-174"><a href="source-code.html#cb151-174" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb151-175"><a href="source-code.html#cb151-175" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(segments.list)) {</span>
<span id="cb151-176"><a href="source-code.html#cb151-176" tabindex="-1"></a>      df[[<span class="fu">length</span>(df) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> segments.list[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb151-177"><a href="source-code.html#cb151-177" tabindex="-1"></a>                                         <span class="at">end =</span> segments.list[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(segments.list[[i1]]), </span>
<span id="cb151-178"><a href="source-code.html#cb151-178" tabindex="-1"></a>                                         <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">state =</span> segments.list[[i1]]<span class="sc">$</span>state)</span>
<span id="cb151-179"><a href="source-code.html#cb151-179" tabindex="-1"></a>    }</span>
<span id="cb151-180"><a href="source-code.html#cb151-180" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df)</span>
<span id="cb151-181"><a href="source-code.html#cb151-181" tabindex="-1"></a>    df<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb151-182"><a href="source-code.html#cb151-182" tabindex="-1"></a>    df<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df<span class="sc">$</span>ID)]</span>
<span id="cb151-183"><a href="source-code.html#cb151-183" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb151-184"><a href="source-code.html#cb151-184" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb151-185"><a href="source-code.html#cb151-185" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(breakpoints)) {</span>
<span id="cb151-186"><a href="source-code.html#cb151-186" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(breakpoints[[i1]]) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb151-187"><a href="source-code.html#cb151-187" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> breakpoints[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb151-188"><a href="source-code.html#cb151-188" tabindex="-1"></a>                                                                     <span class="at">end =</span> breakpoints[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(breakpoints[[i1]]), </span>
<span id="cb151-189"><a href="source-code.html#cb151-189" tabindex="-1"></a>                                                                     <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">mid =</span> (breakpoints[[i1]]<span class="sc">$</span>start.genome <span class="sc">+</span> </span>
<span id="cb151-190"><a href="source-code.html#cb151-190" tabindex="-1"></a>                                                                                                             breakpoints[[i1]]<span class="sc">$</span>end.genome)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb151-191"><a href="source-code.html#cb151-191" tabindex="-1"></a>        }</span>
<span id="cb151-192"><a href="source-code.html#cb151-192" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb151-193"><a href="source-code.html#cb151-193" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> <span class="fu">numeric</span>(), </span>
<span id="cb151-194"><a href="source-code.html#cb151-194" tabindex="-1"></a>                                                                     <span class="at">end =</span> <span class="fu">numeric</span>(), <span class="at">seqnames =</span> <span class="fu">character</span>(), <span class="at">ID =</span> <span class="fu">character</span>(), </span>
<span id="cb151-195"><a href="source-code.html#cb151-195" tabindex="-1"></a>                                                                     <span class="at">mid =</span> <span class="fu">numeric</span>())</span>
<span id="cb151-196"><a href="source-code.html#cb151-196" tabindex="-1"></a>        }</span>
<span id="cb151-197"><a href="source-code.html#cb151-197" tabindex="-1"></a>      }</span>
<span id="cb151-198"><a href="source-code.html#cb151-198" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df.breakpoints)</span>
<span id="cb151-199"><a href="source-code.html#cb151-199" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df.breakpoints<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb151-200"><a href="source-code.html#cb151-200" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df.breakpoints<span class="sc">$</span>ID)]</span>
<span id="cb151-201"><a href="source-code.html#cb151-201" tabindex="-1"></a>    }</span>
<span id="cb151-202"><a href="source-code.html#cb151-202" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]])))</span>
<span id="cb151-203"><a href="source-code.html#cb151-203" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb151-204"><a href="source-code.html#cb151-204" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb151-205"><a href="source-code.html#cb151-205" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb151-206"><a href="source-code.html#cb151-206" tabindex="-1"></a>    label.pos <span class="ot">&lt;-</span> <span class="fu">round</span>(cum.seqlengths<span class="fl">.0</span> <span class="sc">+</span> <span class="fl">0.5</span> <span class="sc">*</span> <span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]]))</span>
<span id="cb151-207"><a href="source-code.html#cb151-207" tabindex="-1"></a>    df.chroms <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths), <span class="at">x =</span> <span class="dv">1</span>, </span>
<span id="cb151-208"><a href="source-code.html#cb151-208" tabindex="-1"></a>                            <span class="at">xend =</span> <span class="fu">length</span>(segments.list))</span>
<span id="cb151-209"><a href="source-code.html#cb151-209" tabindex="-1"></a>    pltlist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb151-210"><a href="source-code.html#cb151-210" tabindex="-1"></a>    widths <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb151-211"><a href="source-code.html#cb151-211" tabindex="-1"></a>    df<span class="sc">$</span>state <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>state, <span class="at">levels =</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">initializeStates</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))<span class="sc">$</span>multiplicity)))</span>
<span id="cb151-212"><a href="source-code.html#cb151-212" tabindex="-1"></a>    df<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>ID)</span>
<span id="cb151-213"><a href="source-code.html#cb151-213" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, </span>
<span id="cb151-214"><a href="source-code.html#cb151-214" tabindex="-1"></a>                                                    <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;state&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> label.pos, </span>
<span id="cb151-215"><a href="source-code.html#cb151-215" tabindex="-1"></a>                                                                                                                          <span class="at">labels =</span> <span class="fu">names</span>(label.pos))</span>
<span id="cb151-216"><a href="source-code.html#cb151-216" tabindex="-1"></a>    </span>
<span id="cb151-217"><a href="source-code.html#cb151-217" tabindex="-1"></a></span>
<span id="cb151-218"><a href="source-code.html#cb151-218" tabindex="-1"></a>    <span class="co"># ggplt &lt;- ggplt + scale_x_continuous(name = &quot;&quot;,</span></span>
<span id="cb151-219"><a href="source-code.html#cb151-219" tabindex="-1"></a>    <span class="co">#                      breaks = 1:length(unique(df$ylabel)),</span></span>
<span id="cb151-220"><a href="source-code.html#cb151-220" tabindex="-1"></a>    <span class="co">#                      labels = unique(df$ylabel))</span></span>
<span id="cb151-221"><a href="source-code.html#cb151-221" tabindex="-1"></a>    </span>
<span id="cb151-222"><a href="source-code.html#cb151-222" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">stateColors</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))) <span class="co"># adding custom colours</span></span>
<span id="cb151-223"><a href="source-code.html#cb151-223" tabindex="-1"></a>    <span class="co"># adjusintg x axis</span></span>
<span id="cb151-224"><a href="source-code.html#cb151-224" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-225"><a href="source-code.html#cb151-225" tabindex="-1"></a>                           <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-226"><a href="source-code.html#cb151-226" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb151-227"><a href="source-code.html#cb151-227" tabindex="-1"></a>    </span>
<span id="cb151-228"><a href="source-code.html#cb151-228" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>,</span>
<span id="cb151-229"><a href="source-code.html#cb151-229" tabindex="-1"></a>                                             <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;y&quot;</span>), <span class="at">data =</span> df.chroms, <span class="at">col =</span> <span class="st">&quot;grey13&quot;</span>)</span>
<span id="cb151-230"><a href="source-code.html#cb151-230" tabindex="-1"></a>    </span>
<span id="cb151-231"><a href="source-code.html#cb151-231" tabindex="-1"></a>    </span>
<span id="cb151-232"><a href="source-code.html#cb151-232" tabindex="-1"></a>    </span>
<span id="cb151-233"><a href="source-code.html#cb151-233" tabindex="-1"></a>    <span class="co"># adjusting y axis</span></span>
<span id="cb151-234"><a href="source-code.html#cb151-234" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-235"><a href="source-code.html#cb151-235" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb151-236"><a href="source-code.html#cb151-236" tabindex="-1"></a>    </span>
<span id="cb151-237"><a href="source-code.html#cb151-237" tabindex="-1"></a>    <span class="co"># removing legend</span></span>
<span id="cb151-238"><a href="source-code.html#cb151-238" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb151-239"><a href="source-code.html#cb151-239" tabindex="-1"></a>    </span>
<span id="cb151-240"><a href="source-code.html#cb151-240" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb151-241"><a href="source-code.html#cb151-241" tabindex="-1"></a>    </span>
<span id="cb151-242"><a href="source-code.html#cb151-242" tabindex="-1"></a>    <span class="co"># removing all axis names</span></span>
<span id="cb151-243"><a href="source-code.html#cb151-243" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-244"><a href="source-code.html#cb151-244" tabindex="-1"></a>    </span>
<span id="cb151-245"><a href="source-code.html#cb151-245" tabindex="-1"></a>    <span class="co"># add numeber of cells sequenced text</span></span>
<span id="cb151-246"><a href="source-code.html#cb151-246" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, </span>
<span id="cb151-247"><a href="source-code.html#cb151-247" tabindex="-1"></a>                              <span class="at">label =</span> <span class="fu">paste</span>(<span class="fu">length</span>(hmms), <span class="st">&#39;cells&#39;</span>), </span>
<span id="cb151-248"><a href="source-code.html#cb151-248" tabindex="-1"></a>                              <span class="at">x =</span> <span class="fu">length</span>(hmms)<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb151-249"><a href="source-code.html#cb151-249" tabindex="-1"></a>                              <span class="at">y =</span> <span class="dv">3100000000</span>,</span>
<span id="cb151-250"><a href="source-code.html#cb151-250" tabindex="-1"></a>                              <span class="at">angle =</span> <span class="dv">270</span>,</span>
<span id="cb151-251"><a href="source-code.html#cb151-251" tabindex="-1"></a>                              <span class="at">size =</span> <span class="dv">50</span>)</span>
<span id="cb151-252"><a href="source-code.html#cb151-252" tabindex="-1"></a>    </span>
<span id="cb151-253"><a href="source-code.html#cb151-253" tabindex="-1"></a>    <span class="co"># decreasing plot margin</span></span>
<span id="cb151-254"><a href="source-code.html#cb151-254" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb151-255"><a href="source-code.html#cb151-255" tabindex="-1"></a>    </span>
<span id="cb151-256"><a href="source-code.html#cb151-256" tabindex="-1"></a>    </span>
<span id="cb151-257"><a href="source-code.html#cb151-257" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb151-258"><a href="source-code.html#cb151-258" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df.breakpoints<span class="sc">$</span>ID)</span>
<span id="cb151-259"><a href="source-code.html#cb151-259" tabindex="-1"></a>      ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb151-260"><a href="source-code.html#cb151-260" tabindex="-1"></a>                                      <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>), </span>
<span id="cb151-261"><a href="source-code.html#cb151-261" tabindex="-1"></a>                                      <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb151-262"><a href="source-code.html#cb151-262" tabindex="-1"></a>                                                                        <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">y =</span> <span class="st">&quot;mid&quot;</span>))</span>
<span id="cb151-263"><a href="source-code.html#cb151-263" tabindex="-1"></a>    }</span>
<span id="cb151-264"><a href="source-code.html#cb151-264" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(hotspots)) {</span>
<span id="cb151-265"><a href="source-code.html#cb151-265" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(hotspots) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb151-266"><a href="source-code.html#cb151-266" tabindex="-1"></a>        df.hot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">transCoord</span>(hotspots))</span>
<span id="cb151-267"><a href="source-code.html#cb151-267" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmin <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb151-268"><a href="source-code.html#cb151-268" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmax <span class="ot">&lt;-</span> <span class="fu">length</span>(class.data<span class="sc">$</span>ID) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb151-269"><a href="source-code.html#cb151-269" tabindex="-1"></a>        ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_rect</span>(<span class="at">data =</span> df.hot, <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">xmin =</span> <span class="st">&quot;xmin&quot;</span>, </span>
<span id="cb151-270"><a href="source-code.html#cb151-270" tabindex="-1"></a>                                                                       <span class="at">xmax =</span> <span class="st">&quot;xmax&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start.genome&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end.genome&quot;</span>, </span>
<span id="cb151-271"><a href="source-code.html#cb151-271" tabindex="-1"></a>                                                                       <span class="at">alpha =</span> <span class="st">&quot;num.events&quot;</span>), <span class="at">fill =</span> <span class="st">&quot;hotpink4&quot;</span>) <span class="sc">+</span> <span class="fu">scale_alpha_continuous</span>(<span class="at">name =</span> <span class="st">&quot;breakpoints&quot;</span>, </span>
<span id="cb151-272"><a href="source-code.html#cb151-272" tabindex="-1"></a>                                                                                                                                          <span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.8</span>))</span>
<span id="cb151-273"><a href="source-code.html#cb151-273" tabindex="-1"></a>      }</span>
<span id="cb151-274"><a href="source-code.html#cb151-274" tabindex="-1"></a>    }</span>
<span id="cb151-275"><a href="source-code.html#cb151-275" tabindex="-1"></a>    </span>
<span id="cb151-276"><a href="source-code.html#cb151-276" tabindex="-1"></a>    </span>
<span id="cb151-277"><a href="source-code.html#cb151-277" tabindex="-1"></a>    width.heatmap <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(hmms[[<span class="dv">1</span>]]<span class="sc">$</span>bins)))<span class="sc">/</span><span class="fl">3e+09</span> <span class="sc">*</span> </span>
<span id="cb151-278"><a href="source-code.html#cb151-278" tabindex="-1"></a>      <span class="dv">150</span></span>
<span id="cb151-279"><a href="source-code.html#cb151-279" tabindex="-1"></a>    height <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">length</span>(hmms) <span class="sc">*</span> <span class="fl">0.5</span>, <span class="dv">2</span>)</span>
<span id="cb151-280"><a href="source-code.html#cb151-280" tabindex="-1"></a>    pltlist[[<span class="st">&quot;heatmap&quot;</span>]] <span class="ot">&lt;-</span> ggplt</span>
<span id="cb151-281"><a href="source-code.html#cb151-281" tabindex="-1"></a>    widths[<span class="st">&quot;heatmap&quot;</span>] <span class="ot">&lt;-</span> width.heatmap</span>
<span id="cb151-282"><a href="source-code.html#cb151-282" tabindex="-1"></a>    <span class="co"># adding class colors</span></span>
<span id="cb151-283"><a href="source-code.html#cb151-283" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb151-284"><a href="source-code.html#cb151-284" tabindex="-1"></a>      width.classes <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb151-285"><a href="source-code.html#cb151-285" tabindex="-1"></a>      class.data<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb151-286"><a href="source-code.html#cb151-286" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(class.data) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="dv">0</span>, </span>
<span id="cb151-287"><a href="source-code.html#cb151-287" tabindex="-1"></a>                                                                <span class="at">ymax =</span> <span class="dv">1</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;class&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">col =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb151-288"><a href="source-code.html#cb151-288" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-289"><a href="source-code.html#cb151-289" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-290"><a href="source-code.html#cb151-290" tabindex="-1"></a>                                 <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-291"><a href="source-code.html#cb151-291" tabindex="-1"></a>                                 <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb151-292"><a href="source-code.html#cb151-292" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb151-293"><a href="source-code.html#cb151-293" tabindex="-1"></a>      </span>
<span id="cb151-294"><a href="source-code.html#cb151-294" tabindex="-1"></a>      <span class="co"># decreasing plot margin</span></span>
<span id="cb151-295"><a href="source-code.html#cb151-295" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb151-296"><a href="source-code.html#cb151-296" tabindex="-1"></a>      </span>
<span id="cb151-297"><a href="source-code.html#cb151-297" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes.color)) {</span>
<span id="cb151-298"><a href="source-code.html#cb151-298" tabindex="-1"></a>        ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">breaks =</span> <span class="fu">names</span>(classes.color), </span>
<span id="cb151-299"><a href="source-code.html#cb151-299" tabindex="-1"></a>                                                <span class="at">values =</span> classes.color)</span>
<span id="cb151-300"><a href="source-code.html#cb151-300" tabindex="-1"></a>      }</span>
<span id="cb151-301"><a href="source-code.html#cb151-301" tabindex="-1"></a>      pltlist[[<span class="st">&quot;classbar&quot;</span>]] <span class="ot">&lt;-</span> ggclass</span>
<span id="cb151-302"><a href="source-code.html#cb151-302" tabindex="-1"></a>      widths[<span class="st">&quot;classbar&quot;</span>] <span class="ot">&lt;-</span> width.classes</span>
<span id="cb151-303"><a href="source-code.html#cb151-303" tabindex="-1"></a>    }</span>
<span id="cb151-304"><a href="source-code.html#cb151-304" tabindex="-1"></a>    <span class="co"># adding dendogram</span></span>
<span id="cb151-305"><a href="source-code.html#cb151-305" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(cl<span class="sc">$</span>hclust) <span class="sc">&amp;</span> dendogram) {</span>
<span id="cb151-306"><a href="source-code.html#cb151-306" tabindex="-1"></a>      dhc <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">as.dendrogram</span>(cl<span class="sc">$</span>hclust)</span>
<span id="cb151-307"><a href="source-code.html#cb151-307" tabindex="-1"></a>      ddata <span class="ot">&lt;-</span> ggdendro<span class="sc">::</span><span class="fu">dendro_data</span>(dhc, <span class="at">type =</span> <span class="st">&quot;rectangle&quot;</span>)</span>
<span id="cb151-308"><a href="source-code.html#cb151-308" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ddata<span class="sc">$</span>segments) <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, </span>
<span id="cb151-309"><a href="source-code.html#cb151-309" tabindex="-1"></a>                                                                 <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>, <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;yend&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_y_reverse</span>()</span>
<span id="cb151-310"><a href="source-code.html#cb151-310" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb151-311"><a href="source-code.html#cb151-311" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-312"><a href="source-code.html#cb151-312" tabindex="-1"></a>                               <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb151-313"><a href="source-code.html#cb151-313" tabindex="-1"></a>                               <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb151-314"><a href="source-code.html#cb151-314" tabindex="-1"></a>      width.dendro <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb151-315"><a href="source-code.html#cb151-315" tabindex="-1"></a>      pltlist[[<span class="st">&quot;dendro&quot;</span>]] <span class="ot">&lt;-</span> ggdndr</span>
<span id="cb151-316"><a href="source-code.html#cb151-316" tabindex="-1"></a>      widths[<span class="st">&quot;dendro&quot;</span>] <span class="ot">&lt;-</span> width.dendro</span>
<span id="cb151-317"><a href="source-code.html#cb151-317" tabindex="-1"></a>    }</span>
<span id="cb151-318"><a href="source-code.html#cb151-318" tabindex="-1"></a>    <span class="co"># alligning ggpllt with dendogram and classes</span></span>
<span id="cb151-319"><a href="source-code.html#cb151-319" tabindex="-1"></a>    cowplt <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">rev</span>(pltlist), <span class="at">align =</span> <span class="st">&quot;h&quot;</span>, </span>
<span id="cb151-320"><a href="source-code.html#cb151-320" tabindex="-1"></a>                                 <span class="at">ncol =</span> <span class="fu">length</span>(pltlist), <span class="at">rel_widths =</span> <span class="fu">rev</span>(widths))</span>
<span id="cb151-321"><a href="source-code.html#cb151-321" tabindex="-1"></a>    </span>
<span id="cb151-322"><a href="source-code.html#cb151-322" tabindex="-1"></a>    </span>
<span id="cb151-323"><a href="source-code.html#cb151-323" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb151-324"><a href="source-code.html#cb151-324" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(file)) {</span>
<span id="cb151-325"><a href="source-code.html#cb151-325" tabindex="-1"></a>      ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Plotting to file &quot;</span>, file, <span class="st">&quot; ...&quot;</span>)</span>
<span id="cb151-326"><a href="source-code.html#cb151-326" tabindex="-1"></a>      <span class="fu">ggsave</span>(file, cowplt, <span class="at">width =</span> <span class="fu">sum</span>(widths), <span class="at">height =</span> height, </span>
<span id="cb151-327"><a href="source-code.html#cb151-327" tabindex="-1"></a>             <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>)</span>
<span id="cb151-328"><a href="source-code.html#cb151-328" tabindex="-1"></a>      <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb151-329"><a href="source-code.html#cb151-329" tabindex="-1"></a>    }</span>
<span id="cb151-330"><a href="source-code.html#cb151-330" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb151-331"><a href="source-code.html#cb151-331" tabindex="-1"></a>      <span class="fu">return</span>(cowplt)</span>
<span id="cb151-332"><a href="source-code.html#cb151-332" tabindex="-1"></a>    }</span>
<span id="cb151-333"><a href="source-code.html#cb151-333" tabindex="-1"></a>  }</span>
<span id="cb151-334"><a href="source-code.html#cb151-334" tabindex="-1"></a>  </span>
<span id="cb151-335"><a href="source-code.html#cb151-335" tabindex="-1"></a>  <span class="co">#make heatmap and safe</span></span>
<span id="cb151-336"><a href="source-code.html#cb151-336" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(<span class="fu">heatmapGenomewide_daan</span>(selected.files, <span class="at">classes =</span> classes_daan, <span class="at">classes.color =</span> class.col,</span>
<span id="cb151-337"><a href="source-code.html#cb151-337" tabindex="-1"></a>                         <span class="at">file =</span> <span class="fu">paste0</span>(path, <span class="st">&#39;/&#39;</span>, name, <span class="st">&#39;_&#39;</span>, platename, <span class="st">&#39;.pdf&#39;</span>)))</span>
<span id="cb151-338"><a href="source-code.html#cb151-338" tabindex="-1"></a>  </span>
<span id="cb151-339"><a href="source-code.html#cb151-339" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">print</span>(<span class="fu">paste0</span>(name, <span class="st">&#39; done.&#39;</span>)))</span>
<span id="cb151-340"><a href="source-code.html#cb151-340" tabindex="-1"></a>  </span>
<span id="cb151-341"><a href="source-code.html#cb151-341" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fisher_test-script-perform_fisher-function" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> fisher_test script: ‘perform_fisher’ function<a href="source-code.html#fisher_test-script-perform_fisher-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="source-code.html#cb152-1" tabindex="-1"></a>perform_fisher <span class="ot">&lt;-</span> <span class="cf">function</span>(dataframe,</span>
<span id="cb152-2"><a href="source-code.html#cb152-2" tabindex="-1"></a>                           population1,</span>
<span id="cb152-3"><a href="source-code.html#cb152-3" tabindex="-1"></a>                           population2) {</span>
<span id="cb152-4"><a href="source-code.html#cb152-4" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb152-5"><a href="source-code.html#cb152-5" tabindex="-1"></a>  <span class="co">#&#39; dataframe is a df with columns paths and class with values &#39;hub015_rad_a&#39;, &#39;hub015_prerad_a&#39; etc.</span></span>
<span id="cb152-6"><a href="source-code.html#cb152-6" tabindex="-1"></a>  <span class="co">#&#39; population 1 and population two are the two pops you want to compare, for example prerad vs rad</span></span>
<span id="cb152-7"><a href="source-code.html#cb152-7" tabindex="-1"></a>  <span class="co">#&#39; or prerad versus rad cycle 2</span></span>
<span id="cb152-8"><a href="source-code.html#cb152-8" tabindex="-1"></a>  <span class="co">#&#39; @return</span></span>
<span id="cb152-9"><a href="source-code.html#cb152-9" tabindex="-1"></a>  <span class="co">#&#39; returns the p value of a Fisher&#39;s exact test. If the p value is above 0.05, the distributions</span></span>
<span id="cb152-10"><a href="source-code.html#cb152-10" tabindex="-1"></a>  <span class="co">#&#39; pre and post rad are the same. </span></span>
<span id="cb152-11"><a href="source-code.html#cb152-11" tabindex="-1"></a>  <span class="co">#&#39; </span></span>
<span id="cb152-12"><a href="source-code.html#cb152-12" tabindex="-1"></a>  </span>
<span id="cb152-13"><a href="source-code.html#cb152-13" tabindex="-1"></a>  dataframe<span class="sc">$</span>rad <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population1, <span class="st">&#39;pop1&#39;</span>,</span>
<span id="cb152-14"><a href="source-code.html#cb152-14" tabindex="-1"></a>                          <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population2, <span class="st">&#39;pop2&#39;</span>,</span>
<span id="cb152-15"><a href="source-code.html#cb152-15" tabindex="-1"></a>                                 <span class="cn">NA</span>))</span>
<span id="cb152-16"><a href="source-code.html#cb152-16" tabindex="-1"></a>  </span>
<span id="cb152-17"><a href="source-code.html#cb152-17" tabindex="-1"></a>  org_id <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">str_extract</span>(<span class="fu">unique</span>(dataframe<span class="sc">$</span>class), <span class="st">&#39;hub</span><span class="sc">\\</span><span class="st">d{3}&#39;</span>))</span>
<span id="cb152-18"><a href="source-code.html#cb152-18" tabindex="-1"></a>  </span>
<span id="cb152-19"><a href="source-code.html#cb152-19" tabindex="-1"></a>  <span class="co"># remove na</span></span>
<span id="cb152-20"><a href="source-code.html#cb152-20" tabindex="-1"></a>  dataframe <span class="ot">&lt;-</span> dataframe[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(dataframe<span class="sc">$</span>rad)),]</span>
<span id="cb152-21"><a href="source-code.html#cb152-21" tabindex="-1"></a>  </span>
<span id="cb152-22"><a href="source-code.html#cb152-22" tabindex="-1"></a>  simple_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(string) {</span>
<span id="cb152-23"><a href="source-code.html#cb152-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">tail</span>(string,<span class="dv">1</span>))</span>
<span id="cb152-24"><a href="source-code.html#cb152-24" tabindex="-1"></a>  }</span>
<span id="cb152-25"><a href="source-code.html#cb152-25" tabindex="-1"></a>  </span>
<span id="cb152-26"><a href="source-code.html#cb152-26" tabindex="-1"></a>  dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stringr<span class="sc">::</span><span class="fu">str_split</span>(dataframe<span class="sc">$</span>class, <span class="at">pattern =</span> <span class="st">&#39;_&#39;</span>), simple_fun))</span>
<span id="cb152-27"><a href="source-code.html#cb152-27" tabindex="-1"></a>  </span>
<span id="cb152-28"><a href="source-code.html#cb152-28" tabindex="-1"></a>  <span class="co"># in hub015 and hub005, we defined subclones. These are actually part of the same clone, so should be </span></span>
<span id="cb152-29"><a href="source-code.html#cb152-29" tabindex="-1"></a>  <span class="co"># analysed together</span></span>
<span id="cb152-30"><a href="source-code.html#cb152-30" tabindex="-1"></a>  <span class="cf">if</span> (org_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;hub005&#39;</span>, <span class="st">&#39;hub015&#39;</span>)){</span>
<span id="cb152-31"><a href="source-code.html#cb152-31" tabindex="-1"></a>    dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(dataframe<span class="sc">$</span>clone, <span class="at">pattern =</span> <span class="st">&#39;a.a|a.b&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;a&#39;</span>)</span>
<span id="cb152-32"><a href="source-code.html#cb152-32" tabindex="-1"></a>    </span>
<span id="cb152-33"><a href="source-code.html#cb152-33" tabindex="-1"></a>  }</span>
<span id="cb152-34"><a href="source-code.html#cb152-34" tabindex="-1"></a>  </span>
<span id="cb152-35"><a href="source-code.html#cb152-35" tabindex="-1"></a>  tab <span class="ot">&lt;-</span> <span class="fu">table</span>(dataframe<span class="sc">$</span>clone, dataframe<span class="sc">$</span>rad)</span>
<span id="cb152-36"><a href="source-code.html#cb152-36" tabindex="-1"></a>  </span>
<span id="cb152-37"><a href="source-code.html#cb152-37" tabindex="-1"></a>  </span>
<span id="cb152-38"><a href="source-code.html#cb152-38" tabindex="-1"></a>  <span class="co"># df &lt;- data.frame(cline &lt;- org_id, p_val &lt;- fisher.test(tab)$p.value)</span></span>
<span id="cb152-39"><a href="source-code.html#cb152-39" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb152-40"><a href="source-code.html#cb152-40" tabindex="-1"></a>  <span class="co"># return(df) </span></span>
<span id="cb152-41"><a href="source-code.html#cb152-41" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">fisher.test</span>(tab)<span class="sc">$</span>p.value)</span>
<span id="cb152-42"><a href="source-code.html#cb152-42" tabindex="-1"></a>  </span>
<span id="cb152-43"><a href="source-code.html#cb152-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="find_specific_cna_sc-script-find_specific_cna_sc-function" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> find_specific_cna_sc script: ‘find_specific_cna_sc’ function<a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="source-code.html#cb153-1" tabindex="-1"></a>find_specific_cna_sc <span class="ot">&lt;-</span></span>
<span id="cb153-2"><a href="source-code.html#cb153-2" tabindex="-1"></a>  <span class="cf">function</span>(list_resistant,</span>
<span id="cb153-3"><a href="source-code.html#cb153-3" tabindex="-1"></a>           list_sensitive,</span>
<span id="cb153-4"><a href="source-code.html#cb153-4" tabindex="-1"></a>           perc_cutoff_within_subclone_resistant,</span>
<span id="cb153-5"><a href="source-code.html#cb153-5" tabindex="-1"></a>           perc_cutoff_within_subclone_sensitive,</span>
<span id="cb153-6"><a href="source-code.html#cb153-6" tabindex="-1"></a>           per_cutoff_across_subclone_resistant,</span>
<span id="cb153-7"><a href="source-code.html#cb153-7" tabindex="-1"></a>           per_cutoff_across_subclone_sensitive,</span>
<span id="cb153-8"><a href="source-code.html#cb153-8" tabindex="-1"></a>           <span class="at">return_cnas_per_subclone =</span> F) {</span>
<span id="cb153-9"><a href="source-code.html#cb153-9" tabindex="-1"></a>    <span class="co">#&#39; @param list_resistant a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb153-10"><a href="source-code.html#cb153-10" tabindex="-1"></a>    <span class="co">#&#39; @param list_sensitive a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb153-11"><a href="source-code.html#cb153-11" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_resistant user defined percentage cutoff to use (e.g. 0.6 means a certain CNA is shared by 60% of the cells</span></span>
<span id="cb153-12"><a href="source-code.html#cb153-12" tabindex="-1"></a>    <span class="co">#&#39; within a subclone).</span></span>
<span id="cb153-13"><a href="source-code.html#cb153-13" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_senstive same as in perc_cutoff_within_subclone_resistant but for the sensitive group</span></span>
<span id="cb153-14"><a href="source-code.html#cb153-14" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_resistant user defined percentage cutoff to use (e.g. 0.7 means a certain CNA is shared in 70% of all subclones)</span></span>
<span id="cb153-15"><a href="source-code.html#cb153-15" tabindex="-1"></a>    <span class="co">#&#39; @param return_cnas_per_subclone boolean. If TRUE will return the shared by perc_cutoff_within_subclone CNAs for each list of each item within </span></span>
<span id="cb153-16"><a href="source-code.html#cb153-16" tabindex="-1"></a>    <span class="co">#&#39; list_resistant. Note list_sensitive should be set to NULL for proper results!!</span></span>
<span id="cb153-17"><a href="source-code.html#cb153-17" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_sensitive same as resistant but for the sensitive group</span></span>
<span id="cb153-18"><a href="source-code.html#cb153-18" tabindex="-1"></a>    <span class="co">#&#39; @example perc_cutoff_within_subclone = 0.6 and per_cutoff_across_subclone = 0.7 means the algorithm will find CNAs that are shared </span></span>
<span id="cb153-19"><a href="source-code.html#cb153-19" tabindex="-1"></a>    <span class="co">#&#39; by 60% of single cells within each subclone, and by 70% of all subclones. Important, a cutoff of 50% is the same as computing the median.</span></span>
<span id="cb153-20"><a href="source-code.html#cb153-20" tabindex="-1"></a>    <span class="co">#&#39; @return a list of GRanges showing unique to list_resistant amplifications and deletions and unique to list_sensitive</span></span>
<span id="cb153-21"><a href="source-code.html#cb153-21" tabindex="-1"></a>    <span class="co">#&#39; amplifications and deletions or, if return_cnas_per_subclone see param return_cnas_per_subclone above.</span></span>
<span id="cb153-22"><a href="source-code.html#cb153-22" tabindex="-1"></a>    <span class="co">#&#39; @details Note that this algorithm is NOT sensitive to the number of single cells within each subclone. </span></span>
<span id="cb153-23"><a href="source-code.html#cb153-23" tabindex="-1"></a>    </span>
<span id="cb153-24"><a href="source-code.html#cb153-24" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb153-25"><a href="source-code.html#cb153-25" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb153-26"><a href="source-code.html#cb153-26" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb153-27"><a href="source-code.html#cb153-27" tabindex="-1"></a>    <span class="fu">require</span>(plyranges)</span>
<span id="cb153-28"><a href="source-code.html#cb153-28" tabindex="-1"></a>    <span class="fu">require</span>(AneuFinder)</span>
<span id="cb153-29"><a href="source-code.html#cb153-29" tabindex="-1"></a>    </span>
<span id="cb153-30"><a href="source-code.html#cb153-30" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-31"><a href="source-code.html#cb153-31" tabindex="-1"></a>    <span class="co"># Part 1. #</span></span>
<span id="cb153-32"><a href="source-code.html#cb153-32" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-33"><a href="source-code.html#cb153-33" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb153-34"><a href="source-code.html#cb153-34" tabindex="-1"></a>    </span>
<span id="cb153-35"><a href="source-code.html#cb153-35" tabindex="-1"></a>    resistant_sc <span class="ot">&lt;-</span></span>
<span id="cb153-36"><a href="source-code.html#cb153-36" tabindex="-1"></a>      <span class="fu">lapply</span>(list_resistant,</span>
<span id="cb153-37"><a href="source-code.html#cb153-37" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb153-38"><a href="source-code.html#cb153-38" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb153-39"><a href="source-code.html#cb153-39" tabindex="-1"></a>    sensitive_sc <span class="ot">&lt;-</span></span>
<span id="cb153-40"><a href="source-code.html#cb153-40" tabindex="-1"></a>      <span class="fu">lapply</span>(list_sensitive,</span>
<span id="cb153-41"><a href="source-code.html#cb153-41" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb153-42"><a href="source-code.html#cb153-42" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb153-43"><a href="source-code.html#cb153-43" tabindex="-1"></a>    </span>
<span id="cb153-44"><a href="source-code.html#cb153-44" tabindex="-1"></a>    <span class="co"># find specific CNAs needs a list of GRanges where each GRange has columns</span></span>
<span id="cb153-45"><a href="source-code.html#cb153-45" tabindex="-1"></a>    <span class="co"># seqnames, ranges, strand (not necessary) and cn.</span></span>
<span id="cb153-46"><a href="source-code.html#cb153-46" tabindex="-1"></a>    wrangle_sc <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff) {</span>
<span id="cb153-47"><a href="source-code.html#cb153-47" tabindex="-1"></a>      <span class="co"># segments is the reduced bins GRange (so all equal CN bins put together)</span></span>
<span id="cb153-48"><a href="source-code.html#cb153-48" tabindex="-1"></a>      wrangle_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff_within) {</span>
<span id="cb153-49"><a href="source-code.html#cb153-49" tabindex="-1"></a>        segments <span class="ot">&lt;-</span> list_lff_within<span class="sc">$</span>segments</span>
<span id="cb153-50"><a href="source-code.html#cb153-50" tabindex="-1"></a>        <span class="co"># ugly way of adding a cn column</span></span>
<span id="cb153-51"><a href="source-code.html#cb153-51" tabindex="-1"></a>        segments<span class="sc">$</span>cn <span class="ot">&lt;-</span> segments<span class="sc">$</span>copy.number</span>
<span id="cb153-52"><a href="source-code.html#cb153-52" tabindex="-1"></a>        </span>
<span id="cb153-53"><a href="source-code.html#cb153-53" tabindex="-1"></a>        <span class="fu">return</span>(segments)</span>
<span id="cb153-54"><a href="source-code.html#cb153-54" tabindex="-1"></a>        </span>
<span id="cb153-55"><a href="source-code.html#cb153-55" tabindex="-1"></a>        <span class="co"># </span><span class="re">END</span><span class="co"> of wrange_within function</span></span>
<span id="cb153-56"><a href="source-code.html#cb153-56" tabindex="-1"></a>        </span>
<span id="cb153-57"><a href="source-code.html#cb153-57" tabindex="-1"></a>      }</span>
<span id="cb153-58"><a href="source-code.html#cb153-58" tabindex="-1"></a>      within_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_lff, wrangle_within)</span>
<span id="cb153-59"><a href="source-code.html#cb153-59" tabindex="-1"></a>      </span>
<span id="cb153-60"><a href="source-code.html#cb153-60" tabindex="-1"></a>      <span class="co"># </span><span class="re">END</span><span class="co"> OF wrangle_sc function</span></span>
<span id="cb153-61"><a href="source-code.html#cb153-61" tabindex="-1"></a>      </span>
<span id="cb153-62"><a href="source-code.html#cb153-62" tabindex="-1"></a>    }</span>
<span id="cb153-63"><a href="source-code.html#cb153-63" tabindex="-1"></a></span>
<span id="cb153-64"><a href="source-code.html#cb153-64" tabindex="-1"></a>    list_resistant <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_sc, wrangle_sc)</span>
<span id="cb153-65"><a href="source-code.html#cb153-65" tabindex="-1"></a>    list_sensitive <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_sc, wrangle_sc)</span>
<span id="cb153-66"><a href="source-code.html#cb153-66" tabindex="-1"></a>    </span>
<span id="cb153-67"><a href="source-code.html#cb153-67" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Extracted segment info from files&#39;</span>)</span>
<span id="cb153-68"><a href="source-code.html#cb153-68" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb153-69"><a href="source-code.html#cb153-69" tabindex="-1"></a>    </span>
<span id="cb153-70"><a href="source-code.html#cb153-70" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-71"><a href="source-code.html#cb153-71" tabindex="-1"></a>    <span class="co"># Part 2. #</span></span>
<span id="cb153-72"><a href="source-code.html#cb153-72" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-73"><a href="source-code.html#cb153-73" tabindex="-1"></a>    <span class="co"># The Granges are now still arranged in bins. We are first going to</span></span>
<span id="cb153-74"><a href="source-code.html#cb153-74" tabindex="-1"></a>    <span class="co"># combine bins with similar copy number.</span></span>
<span id="cb153-75"><a href="source-code.html#cb153-75" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb153-76"><a href="source-code.html#cb153-76" tabindex="-1"></a>    combine_indels <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus) {</span>
<span id="cb153-77"><a href="source-code.html#cb153-77" tabindex="-1"></a>      <span class="co">#&#39; @param subclone_consensus_list is a GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb153-78"><a href="source-code.html#cb153-78" tabindex="-1"></a>      <span class="co">#&#39; @returns list of Granges where each Granges has only CNA of cn2 or cn&gt;2</span></span>
<span id="cb153-79"><a href="source-code.html#cb153-79" tabindex="-1"></a>      </span>
<span id="cb153-80"><a href="source-code.html#cb153-80" tabindex="-1"></a>      combine_within <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus_within) {</span>
<span id="cb153-81"><a href="source-code.html#cb153-81" tabindex="-1"></a>        <span class="co"># subset subclone_consensus in cn &lt;2, cn = 2 and cn &gt; 2</span></span>
<span id="cb153-82"><a href="source-code.html#cb153-82" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span></span>
<span id="cb153-83"><a href="source-code.html#cb153-83" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&lt;</span> <span class="dv">2</span>,]</span>
<span id="cb153-84"><a href="source-code.html#cb153-84" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span></span>
<span id="cb153-85"><a href="source-code.html#cb153-85" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&gt;</span> <span class="dv">2</span>,]</span>
<span id="cb153-86"><a href="source-code.html#cb153-86" tabindex="-1"></a>        </span>
<span id="cb153-87"><a href="source-code.html#cb153-87" tabindex="-1"></a>        <span class="co"># reduce each GRanges (joins consecutive ranges together)</span></span>
<span id="cb153-88"><a href="source-code.html#cb153-88" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_less_2)</span>
<span id="cb153-89"><a href="source-code.html#cb153-89" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_more_2)</span>
<span id="cb153-90"><a href="source-code.html#cb153-90" tabindex="-1"></a>        </span>
<span id="cb153-91"><a href="source-code.html#cb153-91" tabindex="-1"></a>        <span class="co"># reduce deletes metadata.</span></span>
<span id="cb153-92"><a href="source-code.html#cb153-92" tabindex="-1"></a>        <span class="co"># adding metadata</span></span>
<span id="cb153-93"><a href="source-code.html#cb153-93" tabindex="-1"></a>        cn_less_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_less_2))</span>
<span id="cb153-94"><a href="source-code.html#cb153-94" tabindex="-1"></a>        </span>
<span id="cb153-95"><a href="source-code.html#cb153-95" tabindex="-1"></a>        cn_more_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_more_2))</span>
<span id="cb153-96"><a href="source-code.html#cb153-96" tabindex="-1"></a>        </span>
<span id="cb153-97"><a href="source-code.html#cb153-97" tabindex="-1"></a>        <span class="co"># make list and rename items</span></span>
<span id="cb153-98"><a href="source-code.html#cb153-98" tabindex="-1"></a>        GR_list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(cn_less_2, cn_more_2)</span>
<span id="cb153-99"><a href="source-code.html#cb153-99" tabindex="-1"></a>        <span class="fu">names</span>(GR_list) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb153-100"><a href="source-code.html#cb153-100" tabindex="-1"></a>        </span>
<span id="cb153-101"><a href="source-code.html#cb153-101" tabindex="-1"></a>        <span class="co"># Return #</span></span>
<span id="cb153-102"><a href="source-code.html#cb153-102" tabindex="-1"></a>        <span class="fu">return</span>(GR_list)</span>
<span id="cb153-103"><a href="source-code.html#cb153-103" tabindex="-1"></a>      }</span>
<span id="cb153-104"><a href="source-code.html#cb153-104" tabindex="-1"></a>      </span>
<span id="cb153-105"><a href="source-code.html#cb153-105" tabindex="-1"></a>      combine_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(subclone_consensus, combine_within)</span>
<span id="cb153-106"><a href="source-code.html#cb153-106" tabindex="-1"></a>      </span>
<span id="cb153-107"><a href="source-code.html#cb153-107" tabindex="-1"></a>    }</span>
<span id="cb153-108"><a href="source-code.html#cb153-108" tabindex="-1"></a>    </span>
<span id="cb153-109"><a href="source-code.html#cb153-109" tabindex="-1"></a>    <span class="co"># run an apply function here on each Granges in list_resistant versus list_sensitive</span></span>
<span id="cb153-110"><a href="source-code.html#cb153-110" tabindex="-1"></a>    resistant_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_resistant, combine_indels)</span>
<span id="cb153-111"><a href="source-code.html#cb153-111" tabindex="-1"></a>    sensitive_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_sensitive, combine_indels)</span>
<span id="cb153-112"><a href="source-code.html#cb153-112" tabindex="-1"></a>    </span>
<span id="cb153-113"><a href="source-code.html#cb153-113" tabindex="-1"></a>    <span class="co"># splits deletions together within each subclone</span></span>
<span id="cb153-114"><a href="source-code.html#cb153-114" tabindex="-1"></a>    grouper_del <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb153-115"><a href="source-code.html#cb153-115" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb153-116"><a href="source-code.html#cb153-116" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&lt;2&#39;</span>)</span>
<span id="cb153-117"><a href="source-code.html#cb153-117" tabindex="-1"></a>      }</span>
<span id="cb153-118"><a href="source-code.html#cb153-118" tabindex="-1"></a>      </span>
<span id="cb153-119"><a href="source-code.html#cb153-119" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb153-120"><a href="source-code.html#cb153-120" tabindex="-1"></a>    }</span>
<span id="cb153-121"><a href="source-code.html#cb153-121" tabindex="-1"></a>    </span>
<span id="cb153-122"><a href="source-code.html#cb153-122" tabindex="-1"></a>    <span class="co"># splits amplifciations together within each subclone</span></span>
<span id="cb153-123"><a href="source-code.html#cb153-123" tabindex="-1"></a>    grouper_ampl <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb153-124"><a href="source-code.html#cb153-124" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb153-125"><a href="source-code.html#cb153-125" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb153-126"><a href="source-code.html#cb153-126" tabindex="-1"></a>      }</span>
<span id="cb153-127"><a href="source-code.html#cb153-127" tabindex="-1"></a>      </span>
<span id="cb153-128"><a href="source-code.html#cb153-128" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb153-129"><a href="source-code.html#cb153-129" tabindex="-1"></a>    }</span>
<span id="cb153-130"><a href="source-code.html#cb153-130" tabindex="-1"></a>    </span>
<span id="cb153-131"><a href="source-code.html#cb153-131" tabindex="-1"></a>    <span class="co"># run apply on each subclone</span></span>
<span id="cb153-132"><a href="source-code.html#cb153-132" tabindex="-1"></a>    resistant_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_del)</span>
<span id="cb153-133"><a href="source-code.html#cb153-133" tabindex="-1"></a>    resistant_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_ampl)</span>
<span id="cb153-134"><a href="source-code.html#cb153-134" tabindex="-1"></a>    </span>
<span id="cb153-135"><a href="source-code.html#cb153-135" tabindex="-1"></a>    sensitive_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_del)</span>
<span id="cb153-136"><a href="source-code.html#cb153-136" tabindex="-1"></a>    sensitive_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_ampl)</span>
<span id="cb153-137"><a href="source-code.html#cb153-137" tabindex="-1"></a>    </span>
<span id="cb153-138"><a href="source-code.html#cb153-138" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-139"><a href="source-code.html#cb153-139" tabindex="-1"></a>    <span class="co"># Part 3. #</span></span>
<span id="cb153-140"><a href="source-code.html#cb153-140" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-141"><a href="source-code.html#cb153-141" tabindex="-1"></a>    <span class="co"># Part 3A #</span></span>
<span id="cb153-142"><a href="source-code.html#cb153-142" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-143"><a href="source-code.html#cb153-143" tabindex="-1"></a>    <span class="co"># Finding common CNAs within each subclone</span></span>
<span id="cb153-144"><a href="source-code.html#cb153-144" tabindex="-1"></a>    shared_cnas <span class="ot">&lt;-</span></span>
<span id="cb153-145"><a href="source-code.html#cb153-145" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb153-146"><a href="source-code.html#cb153-146" tabindex="-1"></a>               perc_cutoff_within_subclone) {</span>
<span id="cb153-147"><a href="source-code.html#cb153-147" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb153-148"><a href="source-code.html#cb153-148" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff_within_subclone user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb153-149"><a href="source-code.html#cb153-149" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb153-150"><a href="source-code.html#cb153-150" tabindex="-1"></a>        </span>
<span id="cb153-151"><a href="source-code.html#cb153-151" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb153-152"><a href="source-code.html#cb153-152" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb153-153"><a href="source-code.html#cb153-153" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb153-154"><a href="source-code.html#cb153-154" tabindex="-1"></a>        </span>
<span id="cb153-155"><a href="source-code.html#cb153-155" tabindex="-1"></a>        shared_cnas_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_GRanges_within) {</span>
<span id="cb153-156"><a href="source-code.html#cb153-156" tabindex="-1"></a>          n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges_within) <span class="co"># number of range sets</span></span>
<span id="cb153-157"><a href="source-code.html#cb153-157" tabindex="-1"></a>          shared_cna <span class="ot">&lt;-</span></span>
<span id="cb153-158"><a href="source-code.html#cb153-158" tabindex="-1"></a>            <span class="fu">bind_ranges</span>(list_GRanges_within, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb153-159"><a href="source-code.html#cb153-159" tabindex="-1"></a>            <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb153-160"><a href="source-code.html#cb153-160" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb153-161"><a href="source-code.html#cb153-161" tabindex="-1"></a>            <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> perc_cutoff_within_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb153-162"><a href="source-code.html#cb153-162" tabindex="-1"></a>            <span class="fu">reduce_ranges</span>()</span>
<span id="cb153-163"><a href="source-code.html#cb153-163" tabindex="-1"></a>          </span>
<span id="cb153-164"><a href="source-code.html#cb153-164" tabindex="-1"></a>          <span class="fu">return</span>(shared_cna)</span>
<span id="cb153-165"><a href="source-code.html#cb153-165" tabindex="-1"></a>        }</span>
<span id="cb153-166"><a href="source-code.html#cb153-166" tabindex="-1"></a>        </span>
<span id="cb153-167"><a href="source-code.html#cb153-167" tabindex="-1"></a>        <span class="fu">lapply</span>(list_GRanges, shared_cnas_within)</span>
<span id="cb153-168"><a href="source-code.html#cb153-168" tabindex="-1"></a>      }</span>
<span id="cb153-169"><a href="source-code.html#cb153-169" tabindex="-1"></a>    </span>
<span id="cb153-170"><a href="source-code.html#cb153-170" tabindex="-1"></a>    resistant_shared_del <span class="ot">&lt;-</span></span>
<span id="cb153-171"><a href="source-code.html#cb153-171" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb153-172"><a href="source-code.html#cb153-172" tabindex="-1"></a>    resistant_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb153-173"><a href="source-code.html#cb153-173" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb153-174"><a href="source-code.html#cb153-174" tabindex="-1"></a>    sensitive_shared_del <span class="ot">&lt;-</span></span>
<span id="cb153-175"><a href="source-code.html#cb153-175" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb153-176"><a href="source-code.html#cb153-176" tabindex="-1"></a>    sensitive_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb153-177"><a href="source-code.html#cb153-177" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb153-178"><a href="source-code.html#cb153-178" tabindex="-1"></a>    </span>
<span id="cb153-179"><a href="source-code.html#cb153-179" tabindex="-1"></a>    <span class="co"># </span><span class="re">END</span><span class="co"> of function if return_cnas_per_subclone == T</span></span>
<span id="cb153-180"><a href="source-code.html#cb153-180" tabindex="-1"></a>    <span class="cf">if</span> (return_cnas_per_subclone <span class="sc">==</span> T) {</span>
<span id="cb153-181"><a href="source-code.html#cb153-181" tabindex="-1"></a>      cna_list <span class="ot">&lt;-</span> <span class="fu">list</span> (<span class="at">deletions =</span> <span class="fu">c</span>(resistant_shared_del, sensitive_shared_del), </span>
<span id="cb153-182"><a href="source-code.html#cb153-182" tabindex="-1"></a>                        <span class="at">amplifications =</span> <span class="fu">c</span>(resistant_shared_ampl, sensitive_shared_del))</span>
<span id="cb153-183"><a href="source-code.html#cb153-183" tabindex="-1"></a>      </span>
<span id="cb153-184"><a href="source-code.html#cb153-184" tabindex="-1"></a>      <span class="fu">print</span>(</span>
<span id="cb153-185"><a href="source-code.html#cb153-185" tabindex="-1"></a>        <span class="fu">paste0</span>(</span>
<span id="cb153-186"><a href="source-code.html#cb153-186" tabindex="-1"></a>          <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb153-187"><a href="source-code.html#cb153-187" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb153-188"><a href="source-code.html#cb153-188" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each resistant subclone, and by &#39;</span>,</span>
<span id="cb153-189"><a href="source-code.html#cb153-189" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb153-190"><a href="source-code.html#cb153-190" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each sensitive subclone&#39;</span></span>
<span id="cb153-191"><a href="source-code.html#cb153-191" tabindex="-1"></a>        )</span>
<span id="cb153-192"><a href="source-code.html#cb153-192" tabindex="-1"></a>      )</span>
<span id="cb153-193"><a href="source-code.html#cb153-193" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb153-194"><a href="source-code.html#cb153-194" tabindex="-1"></a>      <span class="co"># Unload #</span></span>
<span id="cb153-195"><a href="source-code.html#cb153-195" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb153-196"><a href="source-code.html#cb153-196" tabindex="-1"></a>      <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb153-197"><a href="source-code.html#cb153-197" tabindex="-1"></a>                                        detach,</span>
<span id="cb153-198"><a href="source-code.html#cb153-198" tabindex="-1"></a>                                        <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb153-199"><a href="source-code.html#cb153-199" tabindex="-1"></a>      </span>
<span id="cb153-200"><a href="source-code.html#cb153-200" tabindex="-1"></a>      <span class="fu">return</span>(cna_list)</span>
<span id="cb153-201"><a href="source-code.html#cb153-201" tabindex="-1"></a>    }</span>
<span id="cb153-202"><a href="source-code.html#cb153-202" tabindex="-1"></a>    </span>
<span id="cb153-203"><a href="source-code.html#cb153-203" tabindex="-1"></a>    <span class="co"># Part 3B #</span></span>
<span id="cb153-204"><a href="source-code.html#cb153-204" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-205"><a href="source-code.html#cb153-205" tabindex="-1"></a>    <span class="co"># Finding common CNAs across subclones</span></span>
<span id="cb153-206"><a href="source-code.html#cb153-206" tabindex="-1"></a>    shared_cnas_across_subclones <span class="ot">&lt;-</span></span>
<span id="cb153-207"><a href="source-code.html#cb153-207" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb153-208"><a href="source-code.html#cb153-208" tabindex="-1"></a>               per_cutoff_across_subclone) {</span>
<span id="cb153-209"><a href="source-code.html#cb153-209" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb153-210"><a href="source-code.html#cb153-210" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb153-211"><a href="source-code.html#cb153-211" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb153-212"><a href="source-code.html#cb153-212" tabindex="-1"></a>        </span>
<span id="cb153-213"><a href="source-code.html#cb153-213" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb153-214"><a href="source-code.html#cb153-214" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb153-215"><a href="source-code.html#cb153-215" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb153-216"><a href="source-code.html#cb153-216" tabindex="-1"></a>        </span>
<span id="cb153-217"><a href="source-code.html#cb153-217" tabindex="-1"></a>        n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges) <span class="co"># number of range sets</span></span>
<span id="cb153-218"><a href="source-code.html#cb153-218" tabindex="-1"></a>        shared_cna <span class="ot">&lt;-</span></span>
<span id="cb153-219"><a href="source-code.html#cb153-219" tabindex="-1"></a>          <span class="fu">bind_ranges</span>(list_GRanges, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb153-220"><a href="source-code.html#cb153-220" tabindex="-1"></a>          <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb153-221"><a href="source-code.html#cb153-221" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb153-222"><a href="source-code.html#cb153-222" tabindex="-1"></a>          <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> per_cutoff_across_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb153-223"><a href="source-code.html#cb153-223" tabindex="-1"></a>          <span class="fu">reduce_ranges</span>()</span>
<span id="cb153-224"><a href="source-code.html#cb153-224" tabindex="-1"></a>        </span>
<span id="cb153-225"><a href="source-code.html#cb153-225" tabindex="-1"></a>        <span class="fu">return</span>(shared_cna)</span>
<span id="cb153-226"><a href="source-code.html#cb153-226" tabindex="-1"></a>        </span>
<span id="cb153-227"><a href="source-code.html#cb153-227" tabindex="-1"></a>      }</span>
<span id="cb153-228"><a href="source-code.html#cb153-228" tabindex="-1"></a>    </span>
<span id="cb153-229"><a href="source-code.html#cb153-229" tabindex="-1"></a>    resistant_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb153-230"><a href="source-code.html#cb153-230" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb153-231"><a href="source-code.html#cb153-231" tabindex="-1"></a>    resistant_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb153-232"><a href="source-code.html#cb153-232" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb153-233"><a href="source-code.html#cb153-233" tabindex="-1"></a>    sensitive_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb153-234"><a href="source-code.html#cb153-234" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb153-235"><a href="source-code.html#cb153-235" tabindex="-1"></a>    sensitive_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb153-236"><a href="source-code.html#cb153-236" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb153-237"><a href="source-code.html#cb153-237" tabindex="-1"></a>    </span>
<span id="cb153-238"><a href="source-code.html#cb153-238" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb153-239"><a href="source-code.html#cb153-239" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb153-240"><a href="source-code.html#cb153-240" tabindex="-1"></a>        <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb153-241"><a href="source-code.html#cb153-241" tabindex="-1"></a>        <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb153-242"><a href="source-code.html#cb153-242" tabindex="-1"></a>        <span class="st">&#39;% of single cells within resistant and sensitive subclones, respectively, and that are shared by &#39;</span>,</span>
<span id="cb153-243"><a href="source-code.html#cb153-243" tabindex="-1"></a>        <span class="fu">as.numeric</span>(per_cutoff_across_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(per_cutoff_across_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb153-244"><a href="source-code.html#cb153-244" tabindex="-1"></a>        <span class="st">&#39;% across resistant and sensitive subclones, respectively.&#39;</span></span>
<span id="cb153-245"><a href="source-code.html#cb153-245" tabindex="-1"></a>      )</span>
<span id="cb153-246"><a href="source-code.html#cb153-246" tabindex="-1"></a>    )</span>
<span id="cb153-247"><a href="source-code.html#cb153-247" tabindex="-1"></a>    </span>
<span id="cb153-248"><a href="source-code.html#cb153-248" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb153-249"><a href="source-code.html#cb153-249" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (few seconds)...&#39;</span>)</span>
<span id="cb153-250"><a href="source-code.html#cb153-250" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">2</span>) </span>
<span id="cb153-251"><a href="source-code.html#cb153-251" tabindex="-1"></a>    </span>
<span id="cb153-252"><a href="source-code.html#cb153-252" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-253"><a href="source-code.html#cb153-253" tabindex="-1"></a>    <span class="co"># Part 4. #</span></span>
<span id="cb153-254"><a href="source-code.html#cb153-254" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb153-255"><a href="source-code.html#cb153-255" tabindex="-1"></a>    <span class="co"># Now that we have shared CNAs across subclones with the resistant and sensitive group, we need to extract</span></span>
<span id="cb153-256"><a href="source-code.html#cb153-256" tabindex="-1"></a>    <span class="co"># only those CNAs that are unique to the resistant group.</span></span>
<span id="cb153-257"><a href="source-code.html#cb153-257" tabindex="-1"></a>    </span>
<span id="cb153-258"><a href="source-code.html#cb153-258" tabindex="-1"></a>    <span class="co"># unique to resistant lines deletions:</span></span>
<span id="cb153-259"><a href="source-code.html#cb153-259" tabindex="-1"></a>    unique_resistant_del <span class="ot">&lt;-</span></span>
<span id="cb153-260"><a href="source-code.html#cb153-260" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_del_across, sensitive_shared_del_across)</span>
<span id="cb153-261"><a href="source-code.html#cb153-261" tabindex="-1"></a>    <span class="co"># unique to resistant lines amplifications</span></span>
<span id="cb153-262"><a href="source-code.html#cb153-262" tabindex="-1"></a>    unique_resistant_ampl <span class="ot">&lt;-</span></span>
<span id="cb153-263"><a href="source-code.html#cb153-263" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_ampl_across,</span>
<span id="cb153-264"><a href="source-code.html#cb153-264" tabindex="-1"></a>              sensitive_shared_ampl_across)</span>
<span id="cb153-265"><a href="source-code.html#cb153-265" tabindex="-1"></a>    </span>
<span id="cb153-266"><a href="source-code.html#cb153-266" tabindex="-1"></a>    <span class="co"># unique to sensitive lines deletions:</span></span>
<span id="cb153-267"><a href="source-code.html#cb153-267" tabindex="-1"></a>    unique_sensitive_del <span class="ot">&lt;-</span></span>
<span id="cb153-268"><a href="source-code.html#cb153-268" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_del_across, resistant_shared_del_across)</span>
<span id="cb153-269"><a href="source-code.html#cb153-269" tabindex="-1"></a>    <span class="co"># unique to sensitive lines amplifications:</span></span>
<span id="cb153-270"><a href="source-code.html#cb153-270" tabindex="-1"></a>    unique_sensitive_ampl <span class="ot">&lt;-</span></span>
<span id="cb153-271"><a href="source-code.html#cb153-271" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_ampl_across,</span>
<span id="cb153-272"><a href="source-code.html#cb153-272" tabindex="-1"></a>              resistant_shared_ampl_across)</span>
<span id="cb153-273"><a href="source-code.html#cb153-273" tabindex="-1"></a>    </span>
<span id="cb153-274"><a href="source-code.html#cb153-274" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb153-275"><a href="source-code.html#cb153-275" tabindex="-1"></a>      <span class="st">&#39;Identified amplifications and deletions that are unique to resistant or sensitive subclones&#39;</span></span>
<span id="cb153-276"><a href="source-code.html#cb153-276" tabindex="-1"></a>    )</span>
<span id="cb153-277"><a href="source-code.html#cb153-277" tabindex="-1"></a>    </span>
<span id="cb153-278"><a href="source-code.html#cb153-278" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb153-279"><a href="source-code.html#cb153-279" tabindex="-1"></a>    <span class="co"># Return #</span></span>
<span id="cb153-280"><a href="source-code.html#cb153-280" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb153-281"><a href="source-code.html#cb153-281" tabindex="-1"></a>    returner <span class="ot">&lt;-</span></span>
<span id="cb153-282"><a href="source-code.html#cb153-282" tabindex="-1"></a>      <span class="fu">list</span>(</span>
<span id="cb153-283"><a href="source-code.html#cb153-283" tabindex="-1"></a>        unique_resistant_del,</span>
<span id="cb153-284"><a href="source-code.html#cb153-284" tabindex="-1"></a>        unique_resistant_ampl,</span>
<span id="cb153-285"><a href="source-code.html#cb153-285" tabindex="-1"></a>        unique_sensitive_del,</span>
<span id="cb153-286"><a href="source-code.html#cb153-286" tabindex="-1"></a>        unique_sensitive_ampl</span>
<span id="cb153-287"><a href="source-code.html#cb153-287" tabindex="-1"></a>      )</span>
<span id="cb153-288"><a href="source-code.html#cb153-288" tabindex="-1"></a>    <span class="fu">names</span>(returner) <span class="ot">&lt;-</span></span>
<span id="cb153-289"><a href="source-code.html#cb153-289" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb153-290"><a href="source-code.html#cb153-290" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_del&#39;</span>,</span>
<span id="cb153-291"><a href="source-code.html#cb153-291" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_ampl&#39;</span>,</span>
<span id="cb153-292"><a href="source-code.html#cb153-292" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_del&#39;</span>,</span>
<span id="cb153-293"><a href="source-code.html#cb153-293" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_ampl&#39;</span></span>
<span id="cb153-294"><a href="source-code.html#cb153-294" tabindex="-1"></a>      )</span>
<span id="cb153-295"><a href="source-code.html#cb153-295" tabindex="-1"></a>    </span>
<span id="cb153-296"><a href="source-code.html#cb153-296" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb153-297"><a href="source-code.html#cb153-297" tabindex="-1"></a>    <span class="co"># Unload #</span></span>
<span id="cb153-298"><a href="source-code.html#cb153-298" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb153-299"><a href="source-code.html#cb153-299" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb153-300"><a href="source-code.html#cb153-300" tabindex="-1"></a>                     detach,</span>
<span id="cb153-301"><a href="source-code.html#cb153-301" tabindex="-1"></a>                     <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb153-302"><a href="source-code.html#cb153-302" tabindex="-1"></a>    </span>
<span id="cb153-303"><a href="source-code.html#cb153-303" tabindex="-1"></a>    <span class="fu">return</span>(returner)</span>
<span id="cb153-304"><a href="source-code.html#cb153-304" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="get_cytoband_coverage-script-get_cytoband_coverage-function" class="section level2 hasAnchor" number="6.11">
<h2><span class="header-section-number">6.11</span> get_cytoband_coverage script: ‘get_cytoband_coverage’ function<a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="source-code.html#cb154-1" tabindex="-1"></a>get_cytoband_coverage <span class="ot">&lt;-</span> <span class="cf">function</span>(Grange) {</span>
<span id="cb154-2"><a href="source-code.html#cb154-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges.</span></span>
<span id="cb154-3"><a href="source-code.html#cb154-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns a GRange with cytogenetic locations including</span></span>
<span id="cb154-4"><a href="source-code.html#cb154-4" tabindex="-1"></a>  <span class="co">#&#39; percentage of overlap and a dataframe containing information on how much of</span></span>
<span id="cb154-5"><a href="source-code.html#cb154-5" tabindex="-1"></a>  <span class="co">#&#39; the total arm was affected.</span></span>
<span id="cb154-6"><a href="source-code.html#cb154-6" tabindex="-1"></a>  </span>
<span id="cb154-7"><a href="source-code.html#cb154-7" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb154-8"><a href="source-code.html#cb154-8" tabindex="-1"></a>  <span class="co"># Libraries #</span></span>
<span id="cb154-9"><a href="source-code.html#cb154-9" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb154-10"><a href="source-code.html#cb154-10" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb154-11"><a href="source-code.html#cb154-11" tabindex="-1"></a>  <span class="fu">require</span>(plyranges)</span>
<span id="cb154-12"><a href="source-code.html#cb154-12" tabindex="-1"></a>  </span>
<span id="cb154-13"><a href="source-code.html#cb154-13" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb154-14"><a href="source-code.html#cb154-14" tabindex="-1"></a>  <span class="co"># Data #</span></span>
<span id="cb154-15"><a href="source-code.html#cb154-15" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb154-16"><a href="source-code.html#cb154-16" tabindex="-1"></a>  <span class="co"># Uncomment me, this is to download cytogenetic coordinates</span></span>
<span id="cb154-17"><a href="source-code.html#cb154-17" tabindex="-1"></a>  <span class="co"># library(biovizBase)</span></span>
<span id="cb154-18"><a href="source-code.html#cb154-18" tabindex="-1"></a>  <span class="co"># hg38IdeogramCyto &lt;- getIdeogram(&quot;hg38&quot;, cytobands = TRUE)</span></span>
<span id="cb154-19"><a href="source-code.html#cb154-19" tabindex="-1"></a>  <span class="co"># save(hg38IdeogramCyto, file = &#39;rda/cna_analysis/hg38IdeogramCyto.rda&#39;)</span></span>
<span id="cb154-20"><a href="source-code.html#cb154-20" tabindex="-1"></a>  </span>
<span id="cb154-21"><a href="source-code.html#cb154-21" tabindex="-1"></a>  <span class="co"># !!! ADDING ../cna_analysis for publication in GitHub !!!</span></span>
<span id="cb154-22"><a href="source-code.html#cb154-22" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hg38IdeogramCyto.rda&#39;</span>)</span>
<span id="cb154-23"><a href="source-code.html#cb154-23" tabindex="-1"></a>  </span>
<span id="cb154-24"><a href="source-code.html#cb154-24" tabindex="-1"></a>  </span>
<span id="cb154-25"><a href="source-code.html#cb154-25" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb154-26"><a href="source-code.html#cb154-26" tabindex="-1"></a>  <span class="co"># Wrangle #</span></span>
<span id="cb154-27"><a href="source-code.html#cb154-27" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb154-28"><a href="source-code.html#cb154-28" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb154-29"><a href="source-code.html#cb154-29" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(Grange) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb154-30"><a href="source-code.html#cb154-30" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(hg38IdeogramCyto) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb154-31"><a href="source-code.html#cb154-31" tabindex="-1"></a>  </span>
<span id="cb154-32"><a href="source-code.html#cb154-32" tabindex="-1"></a>  Grange_list_cytoband <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb154-33"><a href="source-code.html#cb154-33" tabindex="-1"></a>  Grange_list_pq_affected <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb154-34"><a href="source-code.html#cb154-34" tabindex="-1"></a>  </span>
<span id="cb154-35"><a href="source-code.html#cb154-35" tabindex="-1"></a>  <span class="co"># return empty Grange and empty data.frame if Grange is empty</span></span>
<span id="cb154-36"><a href="source-code.html#cb154-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(Grange) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb154-37"><a href="source-code.html#cb154-37" tabindex="-1"></a>    <span class="co"># return empty Grange</span></span>
<span id="cb154-38"><a href="source-code.html#cb154-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="fu">GRanges</span>(</span>
<span id="cb154-39"><a href="source-code.html#cb154-39" tabindex="-1"></a>      <span class="at">seqnames =</span> <span class="fu">character</span>(<span class="dv">0</span>),</span>
<span id="cb154-40"><a href="source-code.html#cb154-40" tabindex="-1"></a>      <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">integer</span>(<span class="dv">0</span>), <span class="at">end =</span> <span class="fu">integer</span>(<span class="dv">0</span>)),</span>
<span id="cb154-41"><a href="source-code.html#cb154-41" tabindex="-1"></a>      <span class="at">cytobands_cov =</span>  <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb154-42"><a href="source-code.html#cb154-42" tabindex="-1"></a>    ),</span>
<span id="cb154-43"><a href="source-code.html#cb154-43" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">seqnames =</span> <span class="fu">character</span>(), </span>
<span id="cb154-44"><a href="source-code.html#cb154-44" tabindex="-1"></a>               <span class="at">start_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb154-45"><a href="source-code.html#cb154-45" tabindex="-1"></a>               <span class="at">end_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb154-46"><a href="source-code.html#cb154-46" tabindex="-1"></a>               <span class="at">width_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb154-47"><a href="source-code.html#cb154-47" tabindex="-1"></a>               <span class="at">strand =</span> <span class="fu">character</span>(), </span>
<span id="cb154-48"><a href="source-code.html#cb154-48" tabindex="-1"></a>               <span class="at">chromosome =</span> <span class="fu">character</span>(), </span>
<span id="cb154-49"><a href="source-code.html#cb154-49" tabindex="-1"></a>               <span class="at">arm =</span> <span class="fu">character</span>(), </span>
<span id="cb154-50"><a href="source-code.html#cb154-50" tabindex="-1"></a>               <span class="at">percentage_arm_affected =</span> <span class="fu">numeric</span>(), </span>
<span id="cb154-51"><a href="source-code.html#cb154-51" tabindex="-1"></a>               <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb154-52"><a href="source-code.html#cb154-52" tabindex="-1"></a>    ))</span>
<span id="cb154-53"><a href="source-code.html#cb154-53" tabindex="-1"></a>  }</span>
<span id="cb154-54"><a href="source-code.html#cb154-54" tabindex="-1"></a>  </span>
<span id="cb154-55"><a href="source-code.html#cb154-55" tabindex="-1"></a>  <span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Grange)) {</span>
<span id="cb154-56"><a href="source-code.html#cb154-56" tabindex="-1"></a>    for_range <span class="ot">&lt;-</span> Grange[row]</span>
<span id="cb154-57"><a href="source-code.html#cb154-57" tabindex="-1"></a>    </span>
<span id="cb154-58"><a href="source-code.html#cb154-58" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb154-59"><a href="source-code.html#cb154-59" tabindex="-1"></a>    <span class="co"># Part 1 #</span></span>
<span id="cb154-60"><a href="source-code.html#cb154-60" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb154-61"><a href="source-code.html#cb154-61" tabindex="-1"></a>    <span class="do">## First find specific cytoband locations</span></span>
<span id="cb154-62"><a href="source-code.html#cb154-62" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb154-63"><a href="source-code.html#cb154-63" tabindex="-1"></a>    hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, hg38IdeogramCyto)</span>
<span id="cb154-64"><a href="source-code.html#cb154-64" tabindex="-1"></a>    </span>
<span id="cb154-65"><a href="source-code.html#cb154-65" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb154-66"><a href="source-code.html#cb154-66" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb154-67"><a href="source-code.html#cb154-67" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits)], hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb154-68"><a href="source-code.html#cb154-68" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb154-69"><a href="source-code.html#cb154-69" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb154-70"><a href="source-code.html#cb154-70" tabindex="-1"></a>    </span>
<span id="cb154-71"><a href="source-code.html#cb154-71" tabindex="-1"></a>    <span class="co"># extracting cytobands</span></span>
<span id="cb154-72"><a href="source-code.html#cb154-72" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb154-73"><a href="source-code.html#cb154-73" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="fu">seqnames</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]), hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]<span class="sc">$</span>name)</span>
<span id="cb154-74"><a href="source-code.html#cb154-74" tabindex="-1"></a>    <span class="co"># replace &#39;chr&#39;</span></span>
<span id="cb154-75"><a href="source-code.html#cb154-75" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb154-76"><a href="source-code.html#cb154-76" tabindex="-1"></a>      <span class="fu">str_replace</span>(cytobands, <span class="at">pattern =</span> <span class="st">&#39;chr&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb154-77"><a href="source-code.html#cb154-77" tabindex="-1"></a>    </span>
<span id="cb154-78"><a href="source-code.html#cb154-78" tabindex="-1"></a>    cytobands_cov <span class="ot">&lt;-</span></span>
<span id="cb154-79"><a href="source-code.html#cb154-79" tabindex="-1"></a>      <span class="fu">list</span>(<span class="fu">paste0</span>(cytobands, <span class="st">&#39;;&#39;</span>, <span class="fu">round</span>(percentOverlap, <span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39;%&#39;</span>))</span>
<span id="cb154-80"><a href="source-code.html#cb154-80" tabindex="-1"></a>    </span>
<span id="cb154-81"><a href="source-code.html#cb154-81" tabindex="-1"></a>    for_range<span class="sc">$</span>cytobands_cov <span class="ot">&lt;-</span> cytobands_cov</span>
<span id="cb154-82"><a href="source-code.html#cb154-82" tabindex="-1"></a>    </span>
<span id="cb154-83"><a href="source-code.html#cb154-83" tabindex="-1"></a>    Grange_list_cytoband[[row]] <span class="ot">&lt;-</span> for_range</span>
<span id="cb154-84"><a href="source-code.html#cb154-84" tabindex="-1"></a>    </span>
<span id="cb154-85"><a href="source-code.html#cb154-85" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb154-86"><a href="source-code.html#cb154-86" tabindex="-1"></a>    <span class="co"># Part 2 #</span></span>
<span id="cb154-87"><a href="source-code.html#cb154-87" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb154-88"><a href="source-code.html#cb154-88" tabindex="-1"></a>    <span class="co"># make dataframe that shows how much of an arm is affected </span></span>
<span id="cb154-89"><a href="source-code.html#cb154-89" tabindex="-1"></a>    </span>
<span id="cb154-90"><a href="source-code.html#cb154-90" tabindex="-1"></a>    <span class="co"># Grange with p and q IRanges</span></span>
<span id="cb154-91"><a href="source-code.html#cb154-91" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> hg38IdeogramCyto</span>
<span id="cb154-92"><a href="source-code.html#cb154-92" tabindex="-1"></a>    </span>
<span id="cb154-93"><a href="source-code.html#cb154-93" tabindex="-1"></a>    p_q<span class="sc">$</span>arm <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(p_q<span class="sc">$</span>name, <span class="st">&#39;p|q&#39;</span>)</span>
<span id="cb154-94"><a href="source-code.html#cb154-94" tabindex="-1"></a>    </span>
<span id="cb154-95"><a href="source-code.html#cb154-95" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> <span class="fu">subset</span>(p_q, seqnames <span class="sc">%in%</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">&#39;X&#39;</span>, <span class="st">&#39;Y&#39;</span>))</span>
<span id="cb154-96"><a href="source-code.html#cb154-96" tabindex="-1"></a>    </span>
<span id="cb154-97"><a href="source-code.html#cb154-97" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> p_q <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames, arm) <span class="sc">%&gt;%</span> <span class="fu">reduce_ranges</span>()</span>
<span id="cb154-98"><a href="source-code.html#cb154-98" tabindex="-1"></a>    </span>
<span id="cb154-99"><a href="source-code.html#cb154-99" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb154-100"><a href="source-code.html#cb154-100" tabindex="-1"></a>    hits_pq <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, p_q)</span>
<span id="cb154-101"><a href="source-code.html#cb154-101" tabindex="-1"></a>    </span>
<span id="cb154-102"><a href="source-code.html#cb154-102" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb154-103"><a href="source-code.html#cb154-103" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb154-104"><a href="source-code.html#cb154-104" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits_pq)], p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb154-105"><a href="source-code.html#cb154-105" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb154-106"><a href="source-code.html#cb154-106" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb154-107"><a href="source-code.html#cb154-107" tabindex="-1"></a>    </span>
<span id="cb154-108"><a href="source-code.html#cb154-108" tabindex="-1"></a>    p_q_percentage <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb154-109"><a href="source-code.html#cb154-109" tabindex="-1"></a>    </span>
<span id="cb154-110"><a href="source-code.html#cb154-110" tabindex="-1"></a>    p_q_percentage<span class="sc">$</span>percentage_arm_affected <span class="ot">&lt;-</span> percentOverlap <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb154-111"><a href="source-code.html#cb154-111" tabindex="-1"></a>    </span>
<span id="cb154-112"><a href="source-code.html#cb154-112" tabindex="-1"></a>    <span class="fu">colnames</span>(p_q_percentage) <span class="ot">&lt;-</span></span>
<span id="cb154-113"><a href="source-code.html#cb154-113" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb154-114"><a href="source-code.html#cb154-114" tabindex="-1"></a>        <span class="st">&#39;seqnames&#39;</span>,</span>
<span id="cb154-115"><a href="source-code.html#cb154-115" tabindex="-1"></a>        <span class="st">&#39;start_of_arm&#39;</span>,</span>
<span id="cb154-116"><a href="source-code.html#cb154-116" tabindex="-1"></a>        <span class="st">&#39;end_of_arm&#39;</span>,</span>
<span id="cb154-117"><a href="source-code.html#cb154-117" tabindex="-1"></a>        <span class="st">&#39;width_of_arm&#39;</span>,</span>
<span id="cb154-118"><a href="source-code.html#cb154-118" tabindex="-1"></a>        <span class="st">&#39;strand&#39;</span>,</span>
<span id="cb154-119"><a href="source-code.html#cb154-119" tabindex="-1"></a>        <span class="st">&#39;chromosome&#39;</span>,</span>
<span id="cb154-120"><a href="source-code.html#cb154-120" tabindex="-1"></a>        <span class="st">&#39;arm&#39;</span>,</span>
<span id="cb154-121"><a href="source-code.html#cb154-121" tabindex="-1"></a>        <span class="st">&#39;percentage_arm_affected&#39;</span></span>
<span id="cb154-122"><a href="source-code.html#cb154-122" tabindex="-1"></a>      )</span>
<span id="cb154-123"><a href="source-code.html#cb154-123" tabindex="-1"></a>    </span>
<span id="cb154-124"><a href="source-code.html#cb154-124" tabindex="-1"></a>    </span>
<span id="cb154-125"><a href="source-code.html#cb154-125" tabindex="-1"></a>    </span>
<span id="cb154-126"><a href="source-code.html#cb154-126" tabindex="-1"></a>    </span>
<span id="cb154-127"><a href="source-code.html#cb154-127" tabindex="-1"></a>    Grange_list_pq_affected[[row]] <span class="ot">&lt;-</span> p_q_percentage</span>
<span id="cb154-128"><a href="source-code.html#cb154-128" tabindex="-1"></a>    </span>
<span id="cb154-129"><a href="source-code.html#cb154-129" tabindex="-1"></a>    </span>
<span id="cb154-130"><a href="source-code.html#cb154-130" tabindex="-1"></a>  }</span>
<span id="cb154-131"><a href="source-code.html#cb154-131" tabindex="-1"></a>  </span>
<span id="cb154-132"><a href="source-code.html#cb154-132" tabindex="-1"></a>  Grange <span class="ot">&lt;-</span> <span class="fu">bind_ranges</span>(Grange_list_cytoband)</span>
<span id="cb154-133"><a href="source-code.html#cb154-133" tabindex="-1"></a>  pq_arm_affect <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(Grange_list_pq_affected)</span>
<span id="cb154-134"><a href="source-code.html#cb154-134" tabindex="-1"></a>  </span>
<span id="cb154-135"><a href="source-code.html#cb154-135" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(Grange, pq_arm_affect))</span>
<span id="cb154-136"><a href="source-code.html#cb154-136" tabindex="-1"></a>  </span>
<span id="cb154-137"><a href="source-code.html#cb154-137" tabindex="-1"></a>  <span class="do">### </span><span class="re">END</span><span class="do"> of function </span><span class="al">###</span></span>
<span id="cb154-138"><a href="source-code.html#cb154-138" tabindex="-1"></a>  </span>
<span id="cb154-139"><a href="source-code.html#cb154-139" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_genes-script-get_genes-functoin" class="section level2 hasAnchor" number="6.12">
<h2><span class="header-section-number">6.12</span> get_genes script: ‘get_genes’ functoin<a href="source-code.html#get_genes-script-get_genes-functoin" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="source-code.html#cb155-1" tabindex="-1"></a>get_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) { </span>
<span id="cb155-2"><a href="source-code.html#cb155-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb155-3"><a href="source-code.html#cb155-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb155-4"><a href="source-code.html#cb155-4" tabindex="-1"></a>  </span>
<span id="cb155-5"><a href="source-code.html#cb155-5" tabindex="-1"></a>  <span class="fu">require</span>(GenomicRanges)</span>
<span id="cb155-6"><a href="source-code.html#cb155-6" tabindex="-1"></a>  <span class="fu">require</span>(biomaRt)</span>
<span id="cb155-7"><a href="source-code.html#cb155-7" tabindex="-1"></a>  </span>
<span id="cb155-8"><a href="source-code.html#cb155-8" tabindex="-1"></a>  <span class="co"># This code should be run only once and is therefore commented. </span></span>
<span id="cb155-9"><a href="source-code.html#cb155-9" tabindex="-1"></a>  <span class="co"># code from: https://www.biostars.org/p/311199/</span></span>
<span id="cb155-10"><a href="source-code.html#cb155-10" tabindex="-1"></a>  </span>
<span id="cb155-11"><a href="source-code.html#cb155-11" tabindex="-1"></a>  <span class="co"># Set up an gene annotation template to use</span></span>
<span id="cb155-12"><a href="source-code.html#cb155-12" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb155-13"><a href="source-code.html#cb155-13" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, host=&quot;www.ensembl.org&quot;, path=&quot;/biomart/martservice&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb155-14"><a href="source-code.html#cb155-14" tabindex="-1"></a>  <span class="co"># genes &lt;- getBM(attributes=c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;), mart=mart)</span></span>
<span id="cb155-15"><a href="source-code.html#cb155-15" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[genes[,1]!=&quot;&quot; &amp; genes[,2] %in% c(1:22,&quot;X&quot;,&quot;Y&quot;),]</span></span>
<span id="cb155-16"><a href="source-code.html#cb155-16" tabindex="-1"></a>  <span class="co"># xidx &lt;- which(genes[,2]==&quot;X&quot;)</span></span>
<span id="cb155-17"><a href="source-code.html#cb155-17" tabindex="-1"></a>  <span class="co"># yidx &lt;- which(genes[,2]==&quot;Y&quot;)</span></span>
<span id="cb155-18"><a href="source-code.html#cb155-18" tabindex="-1"></a>  <span class="co"># genes[xidx, 2] &lt;- 23</span></span>
<span id="cb155-19"><a href="source-code.html#cb155-19" tabindex="-1"></a>  <span class="co"># genes[yidx, 2] &lt;- 24</span></span>
<span id="cb155-20"><a href="source-code.html#cb155-20" tabindex="-1"></a>  <span class="co"># genes[,2] &lt;- sapply(genes[,2],as.integer)</span></span>
<span id="cb155-21"><a href="source-code.html#cb155-21" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,3]),]</span></span>
<span id="cb155-22"><a href="source-code.html#cb155-22" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,2]),]</span></span>
<span id="cb155-23"><a href="source-code.html#cb155-23" tabindex="-1"></a>  <span class="co"># save(genes, file = &#39;rda/cna_analysis/genes.rda&#39;)</span></span>
<span id="cb155-24"><a href="source-code.html#cb155-24" tabindex="-1"></a>  <span class="co"># colnames(genes) &lt;- c(&quot;GeneSymbol&quot;,&quot;Chr&quot;,&quot;Start&quot;,&quot;End&quot;)</span></span>
<span id="cb155-25"><a href="source-code.html#cb155-25" tabindex="-1"></a>  <span class="co"># genes_GR &lt;- makeGRangesFromDataFrame(genes,keep.extra.columns = TRUE)</span></span>
<span id="cb155-26"><a href="source-code.html#cb155-26" tabindex="-1"></a>  <span class="co"># save(genes_GR, file = &#39;rda/cna_analysis/genes_GR.rda&#39;)</span></span>
<span id="cb155-27"><a href="source-code.html#cb155-27" tabindex="-1"></a>  </span>
<span id="cb155-28"><a href="source-code.html#cb155-28" tabindex="-1"></a>  <span class="co"># loading genelist and genomic location</span></span>
<span id="cb155-29"><a href="source-code.html#cb155-29" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&quot;../cna_analysis/rda/cna_analysis/genes_GR.rda&quot;</span>, <span class="at">envir =</span> .GlobalEnv)</span>
<span id="cb155-30"><a href="source-code.html#cb155-30" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/genes.rda&#39;</span>, <span class="at">envir =</span> .GlobalEnv )</span>
<span id="cb155-31"><a href="source-code.html#cb155-31" tabindex="-1"></a>  </span>
<span id="cb155-32"><a href="source-code.html#cb155-32" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb155-33"><a href="source-code.html#cb155-33" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb155-34"><a href="source-code.html#cb155-34" tabindex="-1"></a>  GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="fu">paste0</span>(<span class="st">&#39;chr&#39;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">23</span>)))</span>
<span id="cb155-35"><a href="source-code.html#cb155-35" tabindex="-1"></a>  </span>
<span id="cb155-36"><a href="source-code.html#cb155-36" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(genes_GR) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb155-37"><a href="source-code.html#cb155-37" tabindex="-1"></a>  </span>
<span id="cb155-38"><a href="source-code.html#cb155-38" tabindex="-1"></a>  <span class="co"># Finding hits with genes_GR and GRanges</span></span>
<span id="cb155-39"><a href="source-code.html#cb155-39" tabindex="-1"></a>  hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(genes_GR, GRange, <span class="at">type=</span><span class="st">&quot;within&quot;</span>)</span>
<span id="cb155-40"><a href="source-code.html#cb155-40" tabindex="-1"></a>  </span>
<span id="cb155-41"><a href="source-code.html#cb155-41" tabindex="-1"></a>  <span class="co"># constructing</span></span>
<span id="cb155-42"><a href="source-code.html#cb155-42" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(GRange)[<span class="fu">subjectHits</span>(hits),],genes[<span class="fu">queryHits</span>(hits),])</span>
<span id="cb155-43"><a href="source-code.html#cb155-43" tabindex="-1"></a>  </span>
<span id="cb155-44"><a href="source-code.html#cb155-44" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb155-45"><a href="source-code.html#cb155-45" tabindex="-1"></a>  </span>
<span id="cb155-46"><a href="source-code.html#cb155-46" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_oncogenes-scripts-get_oncogenes-function" class="section level2 hasAnchor" number="6.13">
<h2><span class="header-section-number">6.13</span> get_oncogenes scripts: ‘get_oncogenes’ function<a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="source-code.html#cb156-1" tabindex="-1"></a>get_oncogenes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) {</span>
<span id="cb156-2"><a href="source-code.html#cb156-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb156-3"><a href="source-code.html#cb156-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb156-4"><a href="source-code.html#cb156-4" tabindex="-1"></a>  <span class="fu">source</span>(<span class="st">&#39;R/get_genes.R&#39;</span>)</span>
<span id="cb156-5"><a href="source-code.html#cb156-5" tabindex="-1"></a>  </span>
<span id="cb156-6"><a href="source-code.html#cb156-6" tabindex="-1"></a>  <span class="co"># first finding all genes within GRange</span></span>
<span id="cb156-7"><a href="source-code.html#cb156-7" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">get_genes</span>(GRange) </span>
<span id="cb156-8"><a href="source-code.html#cb156-8" tabindex="-1"></a>  </span>
<span id="cb156-9"><a href="source-code.html#cb156-9" tabindex="-1"></a>  <span class="co"># change colnames so dataframes match</span></span>
<span id="cb156-10"><a href="source-code.html#cb156-10" tabindex="-1"></a>  <span class="fu">colnames</span>(all_genes)[<span class="fu">which</span>(<span class="fu">names</span>(all_genes) <span class="sc">==</span> <span class="st">&#39;hgnc_symbol&#39;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Gene Symbol&quot;</span>   </span>
<span id="cb156-11"><a href="source-code.html#cb156-11" tabindex="-1"></a>  </span>
<span id="cb156-12"><a href="source-code.html#cb156-12" tabindex="-1"></a>  <span class="co"># downloading oncogenes</span></span>
<span id="cb156-13"><a href="source-code.html#cb156-13" tabindex="-1"></a>  <span class="fu">require</span>(readr)</span>
<span id="cb156-14"><a href="source-code.html#cb156-14" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../cna_analysis/data/cna_analysis/Census_allMon May 30 12_43_01 2022.csv&quot;</span>)</span>
<span id="cb156-15"><a href="source-code.html#cb156-15" tabindex="-1"></a>  </span>
<span id="cb156-16"><a href="source-code.html#cb156-16" tabindex="-1"></a>  </span>
<span id="cb156-17"><a href="source-code.html#cb156-17" tabindex="-1"></a>  <span class="co"># subset df_genes so to only have oncogenes</span></span>
<span id="cb156-18"><a href="source-code.html#cb156-18" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb156-19"><a href="source-code.html#cb156-19" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(oncogenes, all_genes)</span>
<span id="cb156-20"><a href="source-code.html#cb156-20" tabindex="-1"></a>  </span>
<span id="cb156-21"><a href="source-code.html#cb156-21" tabindex="-1"></a>  <span class="fu">return</span>(oncogenes)</span>
<span id="cb156-22"><a href="source-code.html#cb156-22" tabindex="-1"></a>  </span>
<span id="cb156-23"><a href="source-code.html#cb156-23" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genes_karyotype-script-plot_genes_karyotype-function" class="section level2 hasAnchor" number="6.14">
<h2><span class="header-section-number">6.14</span> plot_genes_karyotype script: ‘plot_genes_karyotype’ function<a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="source-code.html#cb157-1" tabindex="-1"></a>plot_genes_karyotype <span class="ot">&lt;-</span></span>
<span id="cb157-2"><a href="source-code.html#cb157-2" tabindex="-1"></a>  <span class="cf">function</span>(GRange,</span>
<span id="cb157-3"><a href="source-code.html#cb157-3" tabindex="-1"></a>           <span class="at">dist =</span> <span class="sc">-</span><span class="dv">30</span>,</span>
<span id="cb157-4"><a href="source-code.html#cb157-4" tabindex="-1"></a>           show_genes,</span>
<span id="cb157-5"><a href="source-code.html#cb157-5" tabindex="-1"></a>           <span class="at">show_chromosomes =</span> <span class="fu">c</span>(</span>
<span id="cb157-6"><a href="source-code.html#cb157-6" tabindex="-1"></a>             <span class="st">&quot;chr1&quot;</span>,</span>
<span id="cb157-7"><a href="source-code.html#cb157-7" tabindex="-1"></a>             <span class="st">&quot;chr2&quot;</span>,</span>
<span id="cb157-8"><a href="source-code.html#cb157-8" tabindex="-1"></a>             <span class="st">&quot;chr3&quot;</span>,</span>
<span id="cb157-9"><a href="source-code.html#cb157-9" tabindex="-1"></a>             <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb157-10"><a href="source-code.html#cb157-10" tabindex="-1"></a>             <span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb157-11"><a href="source-code.html#cb157-11" tabindex="-1"></a>             <span class="st">&quot;chr6&quot;</span>,</span>
<span id="cb157-12"><a href="source-code.html#cb157-12" tabindex="-1"></a>             <span class="st">&quot;chr7&quot;</span>,</span>
<span id="cb157-13"><a href="source-code.html#cb157-13" tabindex="-1"></a>             <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb157-14"><a href="source-code.html#cb157-14" tabindex="-1"></a>             <span class="st">&quot;chr9&quot;</span>,</span>
<span id="cb157-15"><a href="source-code.html#cb157-15" tabindex="-1"></a>             <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb157-16"><a href="source-code.html#cb157-16" tabindex="-1"></a>             <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb157-17"><a href="source-code.html#cb157-17" tabindex="-1"></a>             <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb157-18"><a href="source-code.html#cb157-18" tabindex="-1"></a>             <span class="st">&quot;chr13&quot;</span>,</span>
<span id="cb157-19"><a href="source-code.html#cb157-19" tabindex="-1"></a>             <span class="st">&quot;chr14&quot;</span>,</span>
<span id="cb157-20"><a href="source-code.html#cb157-20" tabindex="-1"></a>             <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb157-21"><a href="source-code.html#cb157-21" tabindex="-1"></a>             <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb157-22"><a href="source-code.html#cb157-22" tabindex="-1"></a>             <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb157-23"><a href="source-code.html#cb157-23" tabindex="-1"></a>             <span class="st">&quot;chr18&quot;</span>,</span>
<span id="cb157-24"><a href="source-code.html#cb157-24" tabindex="-1"></a>             <span class="st">&quot;chr19&quot;</span>,</span>
<span id="cb157-25"><a href="source-code.html#cb157-25" tabindex="-1"></a>             <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb157-26"><a href="source-code.html#cb157-26" tabindex="-1"></a>             <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb157-27"><a href="source-code.html#cb157-27" tabindex="-1"></a>             <span class="st">&quot;chr22&quot;</span>,</span>
<span id="cb157-28"><a href="source-code.html#cb157-28" tabindex="-1"></a>             <span class="st">&quot;chrX&quot;</span></span>
<span id="cb157-29"><a href="source-code.html#cb157-29" tabindex="-1"></a>           )) {</span>
<span id="cb157-30"><a href="source-code.html#cb157-30" tabindex="-1"></a>    <span class="co">#&#39; @param GRange a GRange with seqnames, ranges, strand and hgnc_symbol</span></span>
<span id="cb157-31"><a href="source-code.html#cb157-31" tabindex="-1"></a>    <span class="co">#&#39; @param dist How far away the genes should be plotted from the lines.</span></span>
<span id="cb157-32"><a href="source-code.html#cb157-32" tabindex="-1"></a>    <span class="co">#&#39; Between -40 and -70 seems to do the trick most of the times.</span></span>
<span id="cb157-33"><a href="source-code.html#cb157-33" tabindex="-1"></a>    <span class="co">#&#39; @param show_genes is a vector of gene symbols to show.</span></span>
<span id="cb157-34"><a href="source-code.html#cb157-34" tabindex="-1"></a>    <span class="co">#&#39; @param show_chromosomes is a vector indicating which chromosomes to show.</span></span>
<span id="cb157-35"><a href="source-code.html#cb157-35" tabindex="-1"></a>    <span class="co">#&#39; @return a karyotype with genes labeled</span></span>
<span id="cb157-36"><a href="source-code.html#cb157-36" tabindex="-1"></a>    <span class="co">#&#39; @details. Throws an error when the GRange does not contain either</span></span>
<span id="cb157-37"><a href="source-code.html#cb157-37" tabindex="-1"></a>    <span class="co">#&#39; ampflications or deletions. This can be ignored, the resulting plot</span></span>
<span id="cb157-38"><a href="source-code.html#cb157-38" tabindex="-1"></a>    <span class="co">#&#39; is correct</span></span>
<span id="cb157-39"><a href="source-code.html#cb157-39" tabindex="-1"></a>    </span>
<span id="cb157-40"><a href="source-code.html#cb157-40" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb157-41"><a href="source-code.html#cb157-41" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb157-42"><a href="source-code.html#cb157-42" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb157-43"><a href="source-code.html#cb157-43" tabindex="-1"></a>    <span class="fu">require</span>(karyoploteR)</span>
<span id="cb157-44"><a href="source-code.html#cb157-44" tabindex="-1"></a>    <span class="fu">require</span>(stringr)</span>
<span id="cb157-45"><a href="source-code.html#cb157-45" tabindex="-1"></a>    </span>
<span id="cb157-46"><a href="source-code.html#cb157-46" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb157-47"><a href="source-code.html#cb157-47" tabindex="-1"></a>    <span class="co"># Wrangle #</span></span>
<span id="cb157-48"><a href="source-code.html#cb157-48" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb157-49"><a href="source-code.html#cb157-49" tabindex="-1"></a>    <span class="co"># making sure seqlevels match</span></span>
<span id="cb157-50"><a href="source-code.html#cb157-50" tabindex="-1"></a>    <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&#39;UCSC&#39;</span></span>
<span id="cb157-51"><a href="source-code.html#cb157-51" tabindex="-1"></a>    </span>
<span id="cb157-52"><a href="source-code.html#cb157-52" tabindex="-1"></a>    kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(</span>
<span id="cb157-53"><a href="source-code.html#cb157-53" tabindex="-1"></a>      <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb157-54"><a href="source-code.html#cb157-54" tabindex="-1"></a>      <span class="at">plot.type =</span> <span class="dv">5</span>,</span>
<span id="cb157-55"><a href="source-code.html#cb157-55" tabindex="-1"></a>      <span class="at">labels.plotter =</span> <span class="cn">NULL</span>,</span>
<span id="cb157-56"><a href="source-code.html#cb157-56" tabindex="-1"></a>      <span class="at">main =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb157-57"><a href="source-code.html#cb157-57" tabindex="-1"></a>      <span class="at">cex =</span> <span class="dv">4</span>,</span>
<span id="cb157-58"><a href="source-code.html#cb157-58" tabindex="-1"></a>      <span class="at">chromosomes =</span> show_chromosomes</span>
<span id="cb157-59"><a href="source-code.html#cb157-59" tabindex="-1"></a>    )</span>
<span id="cb157-60"><a href="source-code.html#cb157-60" tabindex="-1"></a>    </span>
<span id="cb157-61"><a href="source-code.html#cb157-61" tabindex="-1"></a>    <span class="co">#renaming seqlevels</span></span>
<span id="cb157-62"><a href="source-code.html#cb157-62" tabindex="-1"></a>    seqlevels_grange_change <span class="ot">&lt;-</span></span>
<span id="cb157-63"><a href="source-code.html#cb157-63" tabindex="-1"></a>      <span class="fu">str_replace_all</span>(<span class="fu">seqlevels</span>(GRange),</span>
<span id="cb157-64"><a href="source-code.html#cb157-64" tabindex="-1"></a>                      <span class="at">pattern =</span> <span class="st">&#39;chr23&#39;</span>,</span>
<span id="cb157-65"><a href="source-code.html#cb157-65" tabindex="-1"></a>                      <span class="at">replacement =</span> <span class="st">&#39;chrX&#39;</span>)</span>
<span id="cb157-66"><a href="source-code.html#cb157-66" tabindex="-1"></a>    </span>
<span id="cb157-67"><a href="source-code.html#cb157-67" tabindex="-1"></a>    GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="at">value =</span> seqlevels_grange_change)</span>
<span id="cb157-68"><a href="source-code.html#cb157-68" tabindex="-1"></a>    </span>
<span id="cb157-69"><a href="source-code.html#cb157-69" tabindex="-1"></a>    <span class="fu">kpAddChromosomeNames</span>(</span>
<span id="cb157-70"><a href="source-code.html#cb157-70" tabindex="-1"></a>      kp,</span>
<span id="cb157-71"><a href="source-code.html#cb157-71" tabindex="-1"></a>      <span class="at">chr.names =</span> <span class="fu">str_remove_all</span>(show_chromosomes, <span class="st">&#39;chr&#39;</span>),</span>
<span id="cb157-72"><a href="source-code.html#cb157-72" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.3</span></span>
<span id="cb157-73"><a href="source-code.html#cb157-73" tabindex="-1"></a>    )</span>
<span id="cb157-74"><a href="source-code.html#cb157-74" tabindex="-1"></a>    </span>
<span id="cb157-75"><a href="source-code.html#cb157-75" tabindex="-1"></a>    <span class="co"># split into amplifications and deletions</span></span>
<span id="cb157-76"><a href="source-code.html#cb157-76" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;amplified&#39;</span>]</span>
<span id="cb157-77"><a href="source-code.html#cb157-77" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> ampl[ampl<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb157-78"><a href="source-code.html#cb157-78" tabindex="-1"></a>    </span>
<span id="cb157-79"><a href="source-code.html#cb157-79" tabindex="-1"></a>    del <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;deleted&#39;</span>]</span>
<span id="cb157-80"><a href="source-code.html#cb157-80" tabindex="-1"></a>    del <span class="ot">&lt;-</span> del[del<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb157-81"><a href="source-code.html#cb157-81" tabindex="-1"></a>    </span>
<span id="cb157-82"><a href="source-code.html#cb157-82" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb157-83"><a href="source-code.html#cb157-83" tabindex="-1"></a>      cor <span class="ot">&lt;-</span></span>
<span id="cb157-84"><a href="source-code.html#cb157-84" tabindex="-1"></a>        <span class="fu">cor.test</span>(df<span class="sc">$</span>AUC, df<span class="sc">$</span>dependency, <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>),</span>
<span id="cb157-85"><a href="source-code.html#cb157-85" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb157-86"><a href="source-code.html#cb157-86" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb157-87"><a href="source-code.html#cb157-87" tabindex="-1"></a>    )</span>
<span id="cb157-88"><a href="source-code.html#cb157-88" tabindex="-1"></a>    </span>
<span id="cb157-89"><a href="source-code.html#cb157-89" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb157-90"><a href="source-code.html#cb157-90" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb157-91"><a href="source-code.html#cb157-91" tabindex="-1"></a>        kp,</span>
<span id="cb157-92"><a href="source-code.html#cb157-92" tabindex="-1"></a>        <span class="at">data =</span> ampl,</span>
<span id="cb157-93"><a href="source-code.html#cb157-93" tabindex="-1"></a>        <span class="at">labels =</span> ampl<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb157-94"><a href="source-code.html#cb157-94" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb157-95"><a href="source-code.html#cb157-95" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb157-96"><a href="source-code.html#cb157-96" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.85</span>,</span>
<span id="cb157-97"><a href="source-code.html#cb157-97" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb157-98"><a href="source-code.html#cb157-98" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb157-99"><a href="source-code.html#cb157-99" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb157-100"><a href="source-code.html#cb157-100" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;firebrick&#39;</span>,</span>
<span id="cb157-101"><a href="source-code.html#cb157-101" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;firebrick&#39;</span></span>
<span id="cb157-102"><a href="source-code.html#cb157-102" tabindex="-1"></a>      ),</span>
<span id="cb157-103"><a href="source-code.html#cb157-103" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb157-104"><a href="source-code.html#cb157-104" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb157-105"><a href="source-code.html#cb157-105" tabindex="-1"></a>    )</span>
<span id="cb157-106"><a href="source-code.html#cb157-106" tabindex="-1"></a>    </span>
<span id="cb157-107"><a href="source-code.html#cb157-107" tabindex="-1"></a>    </span>
<span id="cb157-108"><a href="source-code.html#cb157-108" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb157-109"><a href="source-code.html#cb157-109" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb157-110"><a href="source-code.html#cb157-110" tabindex="-1"></a>        kp,</span>
<span id="cb157-111"><a href="source-code.html#cb157-111" tabindex="-1"></a>        <span class="at">data =</span> del,</span>
<span id="cb157-112"><a href="source-code.html#cb157-112" tabindex="-1"></a>        <span class="at">labels =</span> del<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb157-113"><a href="source-code.html#cb157-113" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb157-114"><a href="source-code.html#cb157-114" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb157-115"><a href="source-code.html#cb157-115" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.4</span>,</span>
<span id="cb157-116"><a href="source-code.html#cb157-116" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb157-117"><a href="source-code.html#cb157-117" tabindex="-1"></a>        <span class="at">cex =</span> <span class="dv">8</span>,</span>
<span id="cb157-118"><a href="source-code.html#cb157-118" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb157-119"><a href="source-code.html#cb157-119" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb157-120"><a href="source-code.html#cb157-120" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;steelblue&#39;</span>,</span>
<span id="cb157-121"><a href="source-code.html#cb157-121" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;steelblue&#39;</span></span>
<span id="cb157-122"><a href="source-code.html#cb157-122" tabindex="-1"></a>      ),</span>
<span id="cb157-123"><a href="source-code.html#cb157-123" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb157-124"><a href="source-code.html#cb157-124" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb157-125"><a href="source-code.html#cb157-125" tabindex="-1"></a>    )</span>
<span id="cb157-126"><a href="source-code.html#cb157-126" tabindex="-1"></a>    </span>
<span id="cb157-127"><a href="source-code.html#cb157-127" tabindex="-1"></a>    </span>
<span id="cb157-128"><a href="source-code.html#cb157-128" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="section level2 hasAnchor" number="6.15">
<h2><span class="header-section-number">6.15</span> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions<a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="section level2 hasAnchor" number="6.16">
<h2><span class="header-section-number">6.16</span> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function<a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘calculate_heterogeneity_score’ function, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="cna_profiler-script-cna_profiler-function" class="section level2 hasAnchor" number="6.17">
<h2><span class="header-section-number">6.17</span> cna_profiler script: ‘cna_profiler’ function<a href="source-code.html#cna_profiler-script-cna_profiler-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="source-code.html#cb158-1" tabindex="-1"></a>cna_profiler <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb158-2"><a href="source-code.html#cb158-2" tabindex="-1"></a>  <span class="co">#&#39; @param df is a datafame containing all the copy number calls of all the files.</span></span>
<span id="cb158-3"><a href="source-code.html#cb158-3" tabindex="-1"></a>  <span class="co">#&#39;                       Columns: 01) seqnames</span></span>
<span id="cb158-4"><a href="source-code.html#cb158-4" tabindex="-1"></a>  <span class="co">#&#39;                                02) start</span></span>
<span id="cb158-5"><a href="source-code.html#cb158-5" tabindex="-1"></a>  <span class="co">#&#39;                                03) end</span></span>
<span id="cb158-6"><a href="source-code.html#cb158-6" tabindex="-1"></a>  <span class="co">#&#39;                                04) width</span></span>
<span id="cb158-7"><a href="source-code.html#cb158-7" tabindex="-1"></a>  <span class="co">#&#39;                                05) strand</span></span>
<span id="cb158-8"><a href="source-code.html#cb158-8" tabindex="-1"></a>  <span class="co">#&#39;                                06) [copy.number cell 1]</span></span>
<span id="cb158-9"><a href="source-code.html#cb158-9" tabindex="-1"></a>  <span class="co">#&#39;                                07) [copy.number cell 2], etc.</span></span>
<span id="cb158-10"><a href="source-code.html#cb158-10" tabindex="-1"></a>  <span class="co">#&#39; @return Outputs a dataframe that states the ratio (normalized to total cells) of</span></span>
<span id="cb158-11"><a href="source-code.html#cb158-11" tabindex="-1"></a>  <span class="co">#&#39; small, medium, large and whole chromosome CNAs</span></span>
<span id="cb158-12"><a href="source-code.html#cb158-12" tabindex="-1"></a></span>
<span id="cb158-13"><a href="source-code.html#cb158-13" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb158-14"><a href="source-code.html#cb158-14" tabindex="-1"></a>  <span class="fu">require</span>(GWASTools)</span>
<span id="cb158-15"><a href="source-code.html#cb158-15" tabindex="-1"></a>  </span>
<span id="cb158-16"><a href="source-code.html#cb158-16" tabindex="-1"></a>  <span class="co"># defining some parameters</span></span>
<span id="cb158-17"><a href="source-code.html#cb158-17" tabindex="-1"></a>  <span class="co"># number of cells </span></span>
<span id="cb158-18"><a href="source-code.html#cb158-18" tabindex="-1"></a>  n_cells <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df) <span class="sc">-</span> <span class="dv">5</span></span>
<span id="cb158-19"><a href="source-code.html#cb158-19" tabindex="-1"></a>  </span>
<span id="cb158-20"><a href="source-code.html#cb158-20" tabindex="-1"></a>  <span class="co"># extract centromere positions</span></span>
<span id="cb158-21"><a href="source-code.html#cb158-21" tabindex="-1"></a>  <span class="fu">data</span>(centromeres.hg38)</span>
<span id="cb158-22"><a href="source-code.html#cb158-22" tabindex="-1"></a>  </span>
<span id="cb158-23"><a href="source-code.html#cb158-23" tabindex="-1"></a>  centromeres.hg38 <span class="ot">&lt;-</span> <span class="fu">subset</span>(centromeres.hg38, chrom <span class="sc">!=</span> <span class="st">&#39;Y&#39;</span>)</span>
<span id="cb158-24"><a href="source-code.html#cb158-24" tabindex="-1"></a>  <span class="fu">names</span>(centromeres.hg38) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>)</span>
<span id="cb158-25"><a href="source-code.html#cb158-25" tabindex="-1"></a>  </span>
<span id="cb158-26"><a href="source-code.html#cb158-26" tabindex="-1"></a>  </span>
<span id="cb158-27"><a href="source-code.html#cb158-27" tabindex="-1"></a>  <span class="co"># extract total length chromosomes</span></span>
<span id="cb158-28"><a href="source-code.html#cb158-28" tabindex="-1"></a>  total_length <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">sum</span>(width))</span>
<span id="cb158-29"><a href="source-code.html#cb158-29" tabindex="-1"></a>  </span>
<span id="cb158-30"><a href="source-code.html#cb158-30" tabindex="-1"></a>  <span class="co"># construct dataframe containing chromosome name, centromere position (start and end) and total length chromosome</span></span>
<span id="cb158-31"><a href="source-code.html#cb158-31" tabindex="-1"></a>  chrom_centro_total <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(total_length, centromeres.hg38)</span>
<span id="cb158-32"><a href="source-code.html#cb158-32" tabindex="-1"></a>  </span>
<span id="cb158-33"><a href="source-code.html#cb158-33" tabindex="-1"></a>  <span class="co"># compute arm lengths</span></span>
<span id="cb158-34"><a href="source-code.html#cb158-34" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>p_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>start <span class="sc">+</span> (chrom_centro_total<span class="sc">$</span>end<span class="sc">-</span>chrom_centro_total<span class="sc">$</span>start)</span>
<span id="cb158-35"><a href="source-code.html#cb158-35" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>q_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span><span class="st">`</span><span class="at">sum(width)</span><span class="st">`</span> <span class="sc">-</span>  chrom_centro_total<span class="sc">$</span>p_arm</span>
<span id="cb158-36"><a href="source-code.html#cb158-36" tabindex="-1"></a>  </span>
<span id="cb158-37"><a href="source-code.html#cb158-37" tabindex="-1"></a>  <span class="do">##### !!!!!!!! ######</span></span>
<span id="cb158-38"><a href="source-code.html#cb158-38" tabindex="-1"></a>  <span class="co"># Check if this is still correct when using new assemblies !</span></span>
<span id="cb158-39"><a href="source-code.html#cb158-39" tabindex="-1"></a>  </span>
<span id="cb158-40"><a href="source-code.html#cb158-40" tabindex="-1"></a>  <span class="co"># now all p_arms except for chromosome 1 are correct (for some reason the p-arm is the larger arm here). </span></span>
<span id="cb158-41"><a href="source-code.html#cb158-41" tabindex="-1"></a>  <span class="co"># I am going to swithc it. </span></span>
<span id="cb158-42"><a href="source-code.html#cb158-42" tabindex="-1"></a>  p_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>q_arm[<span class="dv">1</span>]</span>
<span id="cb158-43"><a href="source-code.html#cb158-43" tabindex="-1"></a>  q_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>p_arm[<span class="dv">1</span>]</span>
<span id="cb158-44"><a href="source-code.html#cb158-44" tabindex="-1"></a>  </span>
<span id="cb158-45"><a href="source-code.html#cb158-45" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">5</span>] <span class="ot">&lt;-</span> p_arm_1</span>
<span id="cb158-46"><a href="source-code.html#cb158-46" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">6</span>] <span class="ot">&lt;-</span> q_arm_1</span>
<span id="cb158-47"><a href="source-code.html#cb158-47" tabindex="-1"></a>  </span>
<span id="cb158-48"><a href="source-code.html#cb158-48" tabindex="-1"></a>  </span>
<span id="cb158-49"><a href="source-code.html#cb158-49" tabindex="-1"></a>  <span class="co"># function to extract CNA for each chromosome, run this function for each chromosome</span></span>
<span id="cb158-50"><a href="source-code.html#cb158-50" tabindex="-1"></a>  cna_extract <span class="ot">&lt;-</span> <span class="cf">function</span>(chromosome) {</span>
<span id="cb158-51"><a href="source-code.html#cb158-51" tabindex="-1"></a>    <span class="co"># @parama: chromosome is the chromosome (i.e., 1, 2,3, X, etc) </span></span>
<span id="cb158-52"><a href="source-code.html#cb158-52" tabindex="-1"></a>    df_chrom <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb158-53"><a href="source-code.html#cb158-53" tabindex="-1"></a>    </span>
<span id="cb158-54"><a href="source-code.html#cb158-54" tabindex="-1"></a>    <span class="co"># run this function for each column in df_chrom</span></span>
<span id="cb158-55"><a href="source-code.html#cb158-55" tabindex="-1"></a>    cna_extract_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(column_i) {</span>
<span id="cb158-56"><a href="source-code.html#cb158-56" tabindex="-1"></a>      <span class="co"># @ param column i is the integer of the column in df_chrom </span></span>
<span id="cb158-57"><a href="source-code.html#cb158-57" tabindex="-1"></a>      <span class="co"># from which to extract CNAs</span></span>
<span id="cb158-58"><a href="source-code.html#cb158-58" tabindex="-1"></a>      </span>
<span id="cb158-59"><a href="source-code.html#cb158-59" tabindex="-1"></a>      <span class="co"># extract rows that switch CNA value (for example from 2 to 3)</span></span>
<span id="cb158-60"><a href="source-code.html#cb158-60" tabindex="-1"></a>      CNA_change <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="fu">tail</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>) <span class="sc">!=</span> <span class="fu">head</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>)))</span>
<span id="cb158-61"><a href="source-code.html#cb158-61" tabindex="-1"></a>      </span>
<span id="cb158-62"><a href="source-code.html#cb158-62" tabindex="-1"></a>      <span class="co"># extract indices</span></span>
<span id="cb158-63"><a href="source-code.html#cb158-63" tabindex="-1"></a>      CNA_change_indices <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">sort</span>(<span class="fu">c</span>(<span class="dv">1</span>,CNA_change, (CNA_change<span class="dv">-1</span>), <span class="fu">nrow</span>(df_chrom))))</span>
<span id="cb158-64"><a href="source-code.html#cb158-64" tabindex="-1"></a>      </span>
<span id="cb158-65"><a href="source-code.html#cb158-65" tabindex="-1"></a>      <span class="co"># construct dataframe containing end and start of each CNA</span></span>
<span id="cb158-66"><a href="source-code.html#cb158-66" tabindex="-1"></a>      CNA_change_df <span class="ot">&lt;-</span> df_chrom[CNA_change_indices,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>, column_i)]</span>
<span id="cb158-67"><a href="source-code.html#cb158-67" tabindex="-1"></a>      <span class="fu">names</span>(CNA_change_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;width&#39;</span>, <span class="st">&#39;strand&#39;</span>, <span class="st">&#39;cna&#39;</span>)</span>
<span id="cb158-68"><a href="source-code.html#cb158-68" tabindex="-1"></a>      </span>
<span id="cb158-69"><a href="source-code.html#cb158-69" tabindex="-1"></a>      start_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb158-70"><a href="source-code.html#cb158-70" tabindex="-1"></a>      end_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb158-71"><a href="source-code.html#cb158-71" tabindex="-1"></a>      </span>
<span id="cb158-72"><a href="source-code.html#cb158-72" tabindex="-1"></a>      i <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb158-73"><a href="source-code.html#cb158-73" tabindex="-1"></a>      </span>
<span id="cb158-74"><a href="source-code.html#cb158-74" tabindex="-1"></a>      <span class="cf">while</span>(i<span class="sc">&lt;=</span><span class="fu">nrow</span>(CNA_change_df)){</span>
<span id="cb158-75"><a href="source-code.html#cb158-75" tabindex="-1"></a>        </span>
<span id="cb158-76"><a href="source-code.html#cb158-76" tabindex="-1"></a>        <span class="fu">try</span>(<span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">==</span> CNA_change_df[i, <span class="dv">6</span>]){</span>
<span id="cb158-77"><a href="source-code.html#cb158-77" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb158-78"><a href="source-code.html#cb158-78" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span>]</span>
<span id="cb158-79"><a href="source-code.html#cb158-79" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">2</span></span>
<span id="cb158-80"><a href="source-code.html#cb158-80" tabindex="-1"></a>        })</span>
<span id="cb158-81"><a href="source-code.html#cb158-81" tabindex="-1"></a>        </span>
<span id="cb158-82"><a href="source-code.html#cb158-82" tabindex="-1"></a>        <span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">!=</span> CNA_change_df[i, <span class="dv">6</span>] <span class="sc">||</span> <span class="fu">is.na</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>])){</span>
<span id="cb158-83"><a href="source-code.html#cb158-83" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb158-84"><a href="source-code.html#cb158-84" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i, <span class="dv">3</span>]</span>
<span id="cb158-85"><a href="source-code.html#cb158-85" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb158-86"><a href="source-code.html#cb158-86" tabindex="-1"></a>        }</span>
<span id="cb158-87"><a href="source-code.html#cb158-87" tabindex="-1"></a>        </span>
<span id="cb158-88"><a href="source-code.html#cb158-88" tabindex="-1"></a>      }</span>
<span id="cb158-89"><a href="source-code.html#cb158-89" tabindex="-1"></a>      </span>
<span id="cb158-90"><a href="source-code.html#cb158-90" tabindex="-1"></a>      <span class="co"># constructing a dataframe</span></span>
<span id="cb158-91"><a href="source-code.html#cb158-91" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">seqnames =</span> chromosome, </span>
<span id="cb158-92"><a href="source-code.html#cb158-92" tabindex="-1"></a>                                  <span class="at">start =</span> <span class="fu">unlist</span>(start_list), </span>
<span id="cb158-93"><a href="source-code.html#cb158-93" tabindex="-1"></a>                                  <span class="at">end =</span> <span class="fu">unlist</span>(end_list))</span>
<span id="cb158-94"><a href="source-code.html#cb158-94" tabindex="-1"></a>      </span>
<span id="cb158-95"><a href="source-code.html#cb158-95" tabindex="-1"></a>      <span class="co"># remove NA</span></span>
<span id="cb158-96"><a href="source-code.html#cb158-96" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(cna_dataframe)</span>
<span id="cb158-97"><a href="source-code.html#cb158-97" tabindex="-1"></a>      </span>
<span id="cb158-98"><a href="source-code.html#cb158-98" tabindex="-1"></a>      <span class="co"># add column CNA</span></span>
<span id="cb158-99"><a href="source-code.html#cb158-99" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>cna <span class="ot">&lt;-</span> (CNA_change_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cna<span class="sc">!=</span><span class="fu">lag</span>(cna) <span class="sc">|</span> <span class="fu">is.na</span>(<span class="fu">lag</span>(cna))))<span class="sc">$</span>cna</span>
<span id="cb158-100"><a href="source-code.html#cb158-100" tabindex="-1"></a>      </span>
<span id="cb158-101"><a href="source-code.html#cb158-101" tabindex="-1"></a>      <span class="co"># remove columsn with cna == 2 (this is no CNA but normal diploid)</span></span>
<span id="cb158-102"><a href="source-code.html#cb158-102" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">subset</span>(cna_dataframe, cna <span class="sc">!=</span> <span class="dv">2</span>)</span>
<span id="cb158-103"><a href="source-code.html#cb158-103" tabindex="-1"></a>      </span>
<span id="cb158-104"><a href="source-code.html#cb158-104" tabindex="-1"></a>      <span class="co"># add total number of CNAs in this cell as colunn num_cna</span></span>
<span id="cb158-105"><a href="source-code.html#cb158-105" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>num_cna <span class="ot">&lt;-</span> <span class="fu">length</span>(cna_dataframe<span class="sc">$</span>cna), <span class="at">silent =</span> T)</span>
<span id="cb158-106"><a href="source-code.html#cb158-106" tabindex="-1"></a>      </span>
<span id="cb158-107"><a href="source-code.html#cb158-107" tabindex="-1"></a>      <span class="co"># add a width column of the CNA</span></span>
<span id="cb158-108"><a href="source-code.html#cb158-108" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>width <span class="ot">&lt;-</span> cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb158-109"><a href="source-code.html#cb158-109" tabindex="-1"></a>      </span>
<span id="cb158-110"><a href="source-code.html#cb158-110" tabindex="-1"></a>      <span class="co"># extract chromosome specifics</span></span>
<span id="cb158-111"><a href="source-code.html#cb158-111" tabindex="-1"></a>      chrom_df <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrom_centro_total, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb158-112"><a href="source-code.html#cb158-112" tabindex="-1"></a>      </span>
<span id="cb158-113"><a href="source-code.html#cb158-113" tabindex="-1"></a>      <span class="co"># in the next of code, we are going to compute the fraction of SCNA</span></span>
<span id="cb158-114"><a href="source-code.html#cb158-114" tabindex="-1"></a>      <span class="co"># Important: using this measure, a fraction of 1 means that a region the size of one arm</span></span>
<span id="cb158-115"><a href="source-code.html#cb158-115" tabindex="-1"></a>      <span class="co"># was affected. A fraction of 2 means the whole chromosome is affected. Note that these CNA</span></span>
<span id="cb158-116"><a href="source-code.html#cb158-116" tabindex="-1"></a>      <span class="co"># can cross centromeres, so we are unable to easily assign if it is the p-arm or q-arm that is </span></span>
<span id="cb158-117"><a href="source-code.html#cb158-117" tabindex="-1"></a>      <span class="co"># affected. </span></span>
<span id="cb158-118"><a href="source-code.html#cb158-118" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>fraction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>end <span class="sc">&lt;</span> chrom_df<span class="sc">$</span>p_arm, (cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm), </span>
<span id="cb158-119"><a href="source-code.html#cb158-119" tabindex="-1"></a>                                       <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>start <span class="sc">&gt;</span> chrom_df<span class="sc">$</span>p_arm, cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm,</span>
<span id="cb158-120"><a href="source-code.html#cb158-120" tabindex="-1"></a>                                              (((chrom_df<span class="sc">$</span>p_arm <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start) <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">+</span> ((cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm))))</span>
<span id="cb158-121"><a href="source-code.html#cb158-121" tabindex="-1"></a>      </span>
<span id="cb158-122"><a href="source-code.html#cb158-122" tabindex="-1"></a>      <span class="co"># adding cellname</span></span>
<span id="cb158-123"><a href="source-code.html#cb158-123" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>cell_id <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df_chrom)[column_i], <span class="at">silent =</span> T)</span>
<span id="cb158-124"><a href="source-code.html#cb158-124" tabindex="-1"></a>      </span>
<span id="cb158-125"><a href="source-code.html#cb158-125" tabindex="-1"></a>      <span class="fu">return</span>(cna_dataframe)</span>
<span id="cb158-126"><a href="source-code.html#cb158-126" tabindex="-1"></a>      </span>
<span id="cb158-127"><a href="source-code.html#cb158-127" tabindex="-1"></a>    }</span>
<span id="cb158-128"><a href="source-code.html#cb158-128" tabindex="-1"></a>    </span>
<span id="cb158-129"><a href="source-code.html#cb158-129" tabindex="-1"></a>    <span class="co"># running for each cell. </span></span>
<span id="cb158-130"><a href="source-code.html#cb158-130" tabindex="-1"></a>    cna_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">6</span><span class="sc">:</span><span class="fu">ncol</span>(df), cna_extract_cell)</span>
<span id="cb158-131"><a href="source-code.html#cb158-131" tabindex="-1"></a>    </span>
<span id="cb158-132"><a href="source-code.html#cb158-132" tabindex="-1"></a>    <span class="co"># constructing single dataframe </span></span>
<span id="cb158-133"><a href="source-code.html#cb158-133" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_list, <span class="at">.id =</span> <span class="st">&quot;cell_num_scar&quot;</span>)</span>
<span id="cb158-134"><a href="source-code.html#cb158-134" tabindex="-1"></a>    </span>
<span id="cb158-135"><a href="source-code.html#cb158-135" tabindex="-1"></a>    <span class="co"># removing cell_num scar</span></span>
<span id="cb158-136"><a href="source-code.html#cb158-136" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> cna_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df)]</span>
<span id="cb158-137"><a href="source-code.html#cb158-137" tabindex="-1"></a>    </span>
<span id="cb158-138"><a href="source-code.html#cb158-138" tabindex="-1"></a>    <span class="fu">return</span>(cna_df)</span>
<span id="cb158-139"><a href="source-code.html#cb158-139" tabindex="-1"></a>    </span>
<span id="cb158-140"><a href="source-code.html#cb158-140" tabindex="-1"></a>  }</span>
<span id="cb158-141"><a href="source-code.html#cb158-141" tabindex="-1"></a>  </span>
<span id="cb158-142"><a href="source-code.html#cb158-142" tabindex="-1"></a>  cna_df_per_chrom_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb158-143"><a href="source-code.html#cb158-143" tabindex="-1"></a>  </span>
<span id="cb158-144"><a href="source-code.html#cb158-144" tabindex="-1"></a>  <span class="cf">for</span>(chrom <span class="cf">in</span> <span class="fu">unique</span>(df<span class="sc">$</span>seqnames)) {</span>
<span id="cb158-145"><a href="source-code.html#cb158-145" tabindex="-1"></a>    cna_df_per_chrom_list[[chrom]] <span class="ot">&lt;-</span> <span class="fu">cna_extract</span>(chrom)</span>
<span id="cb158-146"><a href="source-code.html#cb158-146" tabindex="-1"></a>  }</span>
<span id="cb158-147"><a href="source-code.html#cb158-147" tabindex="-1"></a>  </span>
<span id="cb158-148"><a href="source-code.html#cb158-148" tabindex="-1"></a>  <span class="co"># constructing single dataframe </span></span>
<span id="cb158-149"><a href="source-code.html#cb158-149" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_df_per_chrom_list, <span class="at">.id =</span> <span class="st">&quot;chrom_scar&quot;</span>)</span>
<span id="cb158-150"><a href="source-code.html#cb158-150" tabindex="-1"></a>  </span>
<span id="cb158-151"><a href="source-code.html#cb158-151" tabindex="-1"></a>  <span class="co"># removing cell_num scar</span></span>
<span id="cb158-152"><a href="source-code.html#cb158-152" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> cna_df_per_chrom_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df_per_chrom_df)]</span>
<span id="cb158-153"><a href="source-code.html#cb158-153" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="cin_signature_ccle-script-run_cin_ccle-function" class="section level2 hasAnchor" number="6.18">
<h2><span class="header-section-number">6.18</span> cin_signature_ccle script: ‘run_cin_ccle’ function<a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="run_cin_ccle"><code>run_cin_ccle &lt;- function(cnv, d) { 
  #&#39; @param cnv is segmental data to input in CINSignatureQuantification
  #&#39; @param d is a dataframe with sample metadata that is joined with cnv
  #&#39; @return outputs CIN signature values.
  
  #############
  # Libraries #
  #############
  require(CINSignatureQuantification)
  require(tidyverse)
  
  ###########
  # Wrangle #
  ###########
  wrangler_cin_signature_ccle &lt;- function(df) {
    #&#39; @param  df: A dataframe containing all the copy number calls of all the files.
    #&#39; @return a dataframe in the format https://github.com/markowetzlab/CINSignatureQuantification
    
    # remove unwanted column names
    keep_columns &lt;-
      c(&#39;Chromosome&#39;,
        &#39;Start&#39;,
        &#39;End&#39;,
        &#39;Segment_Mean&#39;,
        &#39;DepMap_ID&#39;)
    
    new_df &lt;- subset(df, select = keep_columns)
    
    # reorder columns
    col_order &lt;- c(&#39;Chromosome&#39;,
                   &#39;Start&#39;,
                   &#39;End&#39;,
                   &#39;Segment_Mean&#39;,
                   &#39;DepMap_ID&#39;)
    
    new_df &lt;- new_df[, col_order]
    
    # define new column names
    column_names &lt;-
      c(&#39;chromosome&#39;,
        &#39;start&#39;,
        &#39;end&#39;,
        &#39;segVal&#39;,
        &#39;sample&#39;)
    
    colnames(new_df) &lt;- column_names
    
    return(new_df)
    
  }
  
  # Subset cnv to only include samples where radiation sensitivity is known
  cnv &lt;-
    cnv[cnv$DepMap_ID %in% d$DepMap_ID,]
  
  # Wrangle
  cnv &lt;-
    wrangler_cin_signature_ccle(cnv)
  
  # The CIN algorithm (run with quantifyCNSignatures) does not support Y chromosome, we have to get rid of those.
  cnv &lt;-
    cnv[cnv$chromosome != &#39;Y&#39;, ]
  
  #######
  # Run #
  #######
  # Running the CIN scores
  cin_scores &lt;-
    quantifyCNSignatures(cnv, build = &#39;hg38&#39;)
  
  # Extracting the values.
  cin_values &lt;- getActivities(cin_scores)

  ###########
  # Wrangle #
  ###########
  cin_df &lt;- as.data.frame(cin_values)
  
  cin_df$DepMap_ID &lt;- rownames(cin_df)
  
  cin_df &lt;-
    gather(cin_df, cx_signature, cx_value, CX1:CX17, factor_key = T)
  
  # add some metadata
  cin_df &lt;- inner_join(cin_df, d)

  ##########
  # Return #
  ##########
  return(cin_df)
  
}
</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="copy-number-pattern-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/edit/main/06-source.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/blob/main/06-source.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
