<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</title>
  <meta name="description" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Source code | Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids" />
  
  
  

<meta name="author" content="Daan Andel" />


<meta name="date" content="2023-07-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="copy-number-pattern-analyses.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Subclonal Evolution Paper</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html"><i class="fa fa-check"></i><b>2</b> In vitro radiation response and clinical data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#progression-free-survival"><i class="fa fa-check"></i><b>2.1</b> Progression-free survival</a></li>
<li class="chapter" data-level="2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#in-vitro-dose-response"><i class="fa fa-check"></i><b>2.2</b> In vitro dose-response</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#dose-response-plot"><i class="fa fa-check"></i><b>2.2.1</b> Dose-response plot</a></li>
<li class="chapter" data-level="2.2.2" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#rauc-across-multiple-experiments"><i class="fa fa-check"></i><b>2.2.2</b> rAUC across multiple experiments</a></li>
<li class="chapter" data-level="2.2.3" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#flow-cytometry-based-measurement-of-apoptotic-cells-with-sub-g1-dna-content"><i class="fa fa-check"></i><b>2.2.3</b> Flow cytometry-based measurement of (apoptotic) cells with sub-G1 DNA content</a></li>
<li class="chapter" data-level="2.2.4" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#recovery-statistics-after-10-gy-radiation"><i class="fa fa-check"></i><b>2.2.4</b> Recovery statistics after 10 Gy radiation</a></li>
<li class="chapter" data-level="2.2.5" data-path="in-vitro-radiation-response-and-clinical-data.html"><a href="in-vitro-radiation-response-and-clinical-data.html#persisters-versus-shifters-dose-response-plots"><i class="fa fa-check"></i><b>2.2.5</b> Persisters versus shifters dose-response plots</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html"><i class="fa fa-check"></i><b>3</b> Subclonal dynamics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#aneufinder-run"><i class="fa fa-check"></i><b>3.1</b> AneuFinder run</a></li>
<li class="chapter" data-level="3.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#quality-selection"><i class="fa fa-check"></i><b>3.2</b> Quality selection</a></li>
<li class="chapter" data-level="3.3" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#summary-statistics-of-copy-number-alterations"><i class="fa fa-check"></i><b>3.3</b> Summary statistics of copy number alterations</a></li>
<li class="chapter" data-level="3.4" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#pca-and-k-means-plots"><i class="fa fa-check"></i><b>3.4</b> PCA and K-means plots</a></li>
<li class="chapter" data-level="3.5" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-selection"><i class="fa fa-check"></i><b>3.5</b> Subclone selection</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#selecting-subclones"><i class="fa fa-check"></i><b>3.5.1</b> Selecting subclones</a></li>
<li class="chapter" data-level="3.5.2" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#subclone-statistics"><i class="fa fa-check"></i><b>3.5.2</b> Subclone statistics</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#single-cell-copy-number-heatmap-plots"><i class="fa fa-check"></i><b>3.6</b> Single cell copy number heatmap plots</a></li>
<li class="chapter" data-level="3.7" data-path="subclonal-dynamics.html"><a href="subclonal-dynamics.html#fishers-exact-test"><i class="fa fa-check"></i><b>3.7</b> Fisher’s exact test</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html"><i class="fa fa-check"></i><b>4</b> Copy number analyses</a>
<ul>
<li class="chapter" data-level="4.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#colorectal-cancer-specific-arm-level-copy-number-alterations-per-organoid"><i class="fa fa-check"></i><b>4.1</b> Colorectal cancer-specific arm-level copy number alterations per organoid</a></li>
<li class="chapter" data-level="4.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-per-subclone"><i class="fa fa-check"></i><b>4.2</b> Copy number alterations per subclone</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-hub062a"><i class="fa fa-check"></i><b>4.2.1</b> Copy number alterations specific to HUB062A</a></li>
<li class="chapter" data-level="4.2.2" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#copy-number-alterations-specific-to-resistant-subclones"><i class="fa fa-check"></i><b>4.2.2</b> Copy number alterations specific to resistant subclones</a></li>
<li class="chapter" data-level="4.2.3" data-path="copy-number-analyses.html"><a href="copy-number-analyses.html#newly-acquired-copy-number-alterations"><i class="fa fa-check"></i><b>4.2.3</b> Newly acquired copy number alterations</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html"><i class="fa fa-check"></i><b>5</b> Copy number pattern analyses</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#single-cell-analyses"><i class="fa fa-check"></i><b>5.1</b> Single cell analyses</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#heterogeneity-score"><i class="fa fa-check"></i><b>5.1.1</b> Heterogeneity score</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#copy-number-length-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Copy number length analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#ccle-analysis"><i class="fa fa-check"></i><b>5.2</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#run-cin-signature"><i class="fa fa-check"></i><b>5.2.1</b> Run CIN signature</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#statistics-cin-signature"><i class="fa fa-check"></i><b>5.2.2</b> Statistics CIN signature</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-pattern-analyses.html"><a href="copy-number-pattern-analyses.html#plot-cx1-cin-signature"><i class="fa fa-check"></i><b>5.2.3</b> PLot CX1 CIN signature</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="source-code.html"><a href="source-code.html"><i class="fa fa-check"></i><b>6</b> Source code</a>
<ul>
<li class="chapter" data-level="6.1" data-path="source-code.html"><a href="source-code.html#auc_rad-script-auc_fun-function"><i class="fa fa-check"></i><b>6.1</b> AUC_rad script: ‘AUC_fun’ function</a></li>
<li class="chapter" data-level="6.2" data-path="source-code.html"><a href="source-code.html#fitter_rad-script-fitter-function"><i class="fa fa-check"></i><b>6.2</b> fitter_rad script: ‘fitter’ function</a></li>
<li class="chapter" data-level="6.3" data-path="source-code.html"><a href="source-code.html#ic50_rad-script-ic50_fun-function"><i class="fa fa-check"></i><b>6.3</b> ic50_rad script: ‘IC50_fun’ function</a></li>
<li class="chapter" data-level="6.4" data-path="source-code.html"><a href="source-code.html#relativize_dr-script-relative-function"><i class="fa fa-check"></i><b>6.4</b> relativize_dr script: ‘relative’ function</a></li>
<li class="chapter" data-level="6.5" data-path="source-code.html"><a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions"><i class="fa fa-check"></i><b>6.5</b> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions</a></li>
<li class="chapter" data-level="6.6" data-path="source-code.html"><a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions"><i class="fa fa-check"></i><b>6.6</b> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions</a></li>
<li class="chapter" data-level="6.7" data-path="source-code.html"><a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions"><i class="fa fa-check"></i><b>6.7</b> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions</a></li>
<li class="chapter" data-level="6.8" data-path="source-code.html"><a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function"><i class="fa fa-check"></i><b>6.8</b> plot_genomeheatmap script: ‘genomeheatmap’ function</a></li>
<li class="chapter" data-level="6.9" data-path="source-code.html"><a href="source-code.html#fisher_test-script-perform_fisher-function"><i class="fa fa-check"></i><b>6.9</b> fisher_test script: ‘perform_fisher’ function</a></li>
<li class="chapter" data-level="6.10" data-path="source-code.html"><a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function"><i class="fa fa-check"></i><b>6.10</b> find_specific_cna_sc script: ‘find_specific_cna_sc’ function</a></li>
<li class="chapter" data-level="6.11" data-path="source-code.html"><a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function"><i class="fa fa-check"></i><b>6.11</b> get_cytoband_coverage script: ‘get_cytoband_coverage’ function</a></li>
<li class="chapter" data-level="6.12" data-path="source-code.html"><a href="source-code.html#get_genes-script-get_genes-functoin"><i class="fa fa-check"></i><b>6.12</b> get_genes script: ‘get_genes’ functoin</a></li>
<li class="chapter" data-level="6.13" data-path="source-code.html"><a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function"><i class="fa fa-check"></i><b>6.13</b> get_oncogenes scripts: ‘get_oncogenes’ function</a></li>
<li class="chapter" data-level="6.14" data-path="source-code.html"><a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function"><i class="fa fa-check"></i><b>6.14</b> plot_genes_karyotype script: ‘plot_genes_karyotype’ function</a></li>
<li class="chapter" data-level="6.15" data-path="source-code.html"><a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions"><i class="fa fa-check"></i><b>6.15</b> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions</a></li>
<li class="chapter" data-level="6.16" data-path="source-code.html"><a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function"><i class="fa fa-check"></i><b>6.16</b> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function</a></li>
<li class="chapter" data-level="6.17" data-path="source-code.html"><a href="source-code.html#cna_profiler-script-cna_profiler-function"><i class="fa fa-check"></i><b>6.17</b> cna_profiler script: ‘cna_profiler’ function</a></li>
<li class="chapter" data-level="6.18" data-path="source-code.html"><a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function"><i class="fa fa-check"></i><b>6.18</b> cin_signature_ccle script: ‘run_cin_ccle’ function</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pre-existing Radioresistant Subclones Determine Radioresistance in Rectal Cancer Organoids</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="source-code" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Source code<a href="source-code.html#source-code" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The source code in this here is arranged in a loose chronological order, reflecting the sequence in which it is called in the scripts presented throughout this book.</p>
<div id="auc_rad-script-auc_fun-function" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> AUC_rad script: ‘AUC_fun’ function<a href="source-code.html#auc_rad-script-auc_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="source-code.html#cb1-1" tabindex="-1"></a><span class="co"># returns the AUC of the actual data.</span></span>
<span id="cb1-2"><a href="source-code.html#cb1-2" tabindex="-1"></a>AUC_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb1-3"><a href="source-code.html#cb1-3" tabindex="-1"></a>  </span>
<span id="cb1-4"><a href="source-code.html#cb1-4" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb1-5"><a href="source-code.html#cb1-5" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb1-6"><a href="source-code.html#cb1-6" tabindex="-1"></a>  </span>
<span id="cb1-7"><a href="source-code.html#cb1-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb1-8"><a href="source-code.html#cb1-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb1-9"><a href="source-code.html#cb1-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb1-10"><a href="source-code.html#cb1-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb1-11"><a href="source-code.html#cb1-11" tabindex="-1"></a>  </span>
<span id="cb1-12"><a href="source-code.html#cb1-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb1-13"><a href="source-code.html#cb1-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-14"><a href="source-code.html#cb1-14" tabindex="-1"></a>  </span>
<span id="cb1-15"><a href="source-code.html#cb1-15" tabindex="-1"></a>  <span class="co"># define dose_vetor</span></span>
<span id="cb1-16"><a href="source-code.html#cb1-16" tabindex="-1"></a>  dose_vector <span class="ot">&lt;-</span> <span class="fu">unique</span>(d<span class="sc">$</span>dose)</span>
<span id="cb1-17"><a href="source-code.html#cb1-17" tabindex="-1"></a>  </span>
<span id="cb1-18"><a href="source-code.html#cb1-18" tabindex="-1"></a>  </span>
<span id="cb1-19"><a href="source-code.html#cb1-19" tabindex="-1"></a>  <span class="co"># computes AUC for given predicted survival and dose given by dose_vector</span></span>
<span id="cb1-20"><a href="source-code.html#cb1-20" tabindex="-1"></a>  AUC_calc <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb1-21"><a href="source-code.html#cb1-21" tabindex="-1"></a>    <span class="co"># extract actual data</span></span>
<span id="cb1-22"><a href="source-code.html#cb1-22" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> df<span class="sc">$</span>relative_mean</span>
<span id="cb1-23"><a href="source-code.html#cb1-23" tabindex="-1"></a>    </span>
<span id="cb1-24"><a href="source-code.html#cb1-24" tabindex="-1"></a>    <span class="co"># extracts number of experiemnts </span></span>
<span id="cb1-25"><a href="source-code.html#cb1-25" tabindex="-1"></a>    n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb1-26"><a href="source-code.html#cb1-26" tabindex="-1"></a>    <span class="co">#remove 2th and 3th number (are duplicates)</span></span>
<span id="cb1-27"><a href="source-code.html#cb1-27" tabindex="-1"></a>    rel_mean <span class="ot">&lt;-</span> rel_mean[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">length</span>(rel_mean), n)]</span>
<span id="cb1-28"><a href="source-code.html#cb1-28" tabindex="-1"></a>    </span>
<span id="cb1-29"><a href="source-code.html#cb1-29" tabindex="-1"></a>    <span class="co"># compute maximal </span></span>
<span id="cb1-30"><a href="source-code.html#cb1-30" tabindex="-1"></a>    max_AUC <span class="ot">&lt;-</span> <span class="fu">max</span>(dose_vector)<span class="sc">*</span><span class="dv">1</span></span>
<span id="cb1-31"><a href="source-code.html#cb1-31" tabindex="-1"></a>    </span>
<span id="cb1-32"><a href="source-code.html#cb1-32" tabindex="-1"></a>    <span class="co">#compute AUC</span></span>
<span id="cb1-33"><a href="source-code.html#cb1-33" tabindex="-1"></a>    act_AUC <span class="ot">&lt;-</span> <span class="fu">AUC</span>(<span class="at">x=</span>dose_vector, <span class="at">y =</span> rel_mean)</span>
<span id="cb1-34"><a href="source-code.html#cb1-34" tabindex="-1"></a>    </span>
<span id="cb1-35"><a href="source-code.html#cb1-35" tabindex="-1"></a>    <span class="co"># make dataframe</span></span>
<span id="cb1-36"><a href="source-code.html#cb1-36" tabindex="-1"></a>    datafr <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">max_AUC =</span> max_AUC,</span>
<span id="cb1-37"><a href="source-code.html#cb1-37" tabindex="-1"></a>                         <span class="at">act_AUC =</span> act_AUC, </span>
<span id="cb1-38"><a href="source-code.html#cb1-38" tabindex="-1"></a>                         <span class="at">rel_AUC =</span> act_AUC<span class="sc">/</span>max_AUC,</span>
<span id="cb1-39"><a href="source-code.html#cb1-39" tabindex="-1"></a>                         <span class="at">expcode =</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode),</span>
<span id="cb1-40"><a href="source-code.html#cb1-40" tabindex="-1"></a>                         <span class="at">cline=</span> <span class="fu">unique</span>(df<span class="sc">$</span>cline),</span>
<span id="cb1-41"><a href="source-code.html#cb1-41" tabindex="-1"></a>                         <span class="at">treatment =</span> <span class="fu">unique</span>(df<span class="sc">$</span>treatment ) )</span>
<span id="cb1-42"><a href="source-code.html#cb1-42" tabindex="-1"></a>    </span>
<span id="cb1-43"><a href="source-code.html#cb1-43" tabindex="-1"></a>    </span>
<span id="cb1-44"><a href="source-code.html#cb1-44" tabindex="-1"></a>    <span class="fu">return</span>(datafr)</span>
<span id="cb1-45"><a href="source-code.html#cb1-45" tabindex="-1"></a>  }</span>
<span id="cb1-46"><a href="source-code.html#cb1-46" tabindex="-1"></a>  </span>
<span id="cb1-47"><a href="source-code.html#cb1-47" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb1-48"><a href="source-code.html#cb1-48" tabindex="-1"></a>  AUC_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, AUC_calc) </span>
<span id="cb1-49"><a href="source-code.html#cb1-49" tabindex="-1"></a>  </span>
<span id="cb1-50"><a href="source-code.html#cb1-50" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb1-51"><a href="source-code.html#cb1-51" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, AUC_vector)</span>
<span id="cb1-52"><a href="source-code.html#cb1-52" tabindex="-1"></a>  </span>
<span id="cb1-53"><a href="source-code.html#cb1-53" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb1-54"><a href="source-code.html#cb1-54" tabindex="-1"></a>  </span>
<span id="cb1-55"><a href="source-code.html#cb1-55" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fitter_rad-script-fitter-function" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> fitter_rad script: ‘fitter’ function<a href="source-code.html#fitter_rad-script-fitter-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="source-code.html#cb2-1" tabindex="-1"></a>fitter <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb2-2"><a href="source-code.html#cb2-2" tabindex="-1"></a>  </span>
<span id="cb2-3"><a href="source-code.html#cb2-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb2-4"><a href="source-code.html#cb2-4" tabindex="-1"></a>  d <span class="ot">&lt;-</span> df</span>
<span id="cb2-5"><a href="source-code.html#cb2-5" tabindex="-1"></a>  </span>
<span id="cb2-6"><a href="source-code.html#cb2-6" tabindex="-1"></a>  </span>
<span id="cb2-7"><a href="source-code.html#cb2-7" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb2-8"><a href="source-code.html#cb2-8" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb2-9"><a href="source-code.html#cb2-9" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb2-10"><a href="source-code.html#cb2-10" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb2-11"><a href="source-code.html#cb2-11" tabindex="-1"></a>  </span>
<span id="cb2-12"><a href="source-code.html#cb2-12" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb2-13"><a href="source-code.html#cb2-13" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-14"><a href="source-code.html#cb2-14" tabindex="-1"></a>  </span>
<span id="cb2-15"><a href="source-code.html#cb2-15" tabindex="-1"></a>  </span>
<span id="cb2-16"><a href="source-code.html#cb2-16" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb2-17"><a href="source-code.html#cb2-17" tabindex="-1"></a>  fit_model<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb2-18"><a href="source-code.html#cb2-18" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb2-19"><a href="source-code.html#cb2-19" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb2-20"><a href="source-code.html#cb2-20" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb2-21"><a href="source-code.html#cb2-21" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb2-22"><a href="source-code.html#cb2-22" tabindex="-1"></a>    </span>
<span id="cb2-23"><a href="source-code.html#cb2-23" tabindex="-1"></a>    newdata <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">dose=</span><span class="fu">exp</span>(<span class="fu">seq</span>(<span class="fu">log</span>(<span class="fl">0.00000001</span>), <span class="fu">log</span>(<span class="fu">max</span>(df<span class="sc">$</span>dose) <span class="sc">+</span> <span class="dv">1000</span>), <span class="at">length=</span><span class="dv">1000</span>))) <span class="co"># new data with doses. Note: lowest dose is not</span></span>
<span id="cb2-24"><a href="source-code.html#cb2-24" tabindex="-1"></a>    <span class="co"># log 0 but log(&#39;very small number&#39;) because otherwise this will hamper the scaling in ggplot later on. </span></span>
<span id="cb2-25"><a href="source-code.html#cb2-25" tabindex="-1"></a>    pm <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit, <span class="at">newdata=</span>newdata, <span class="at">interval=</span><span class="st">&quot;confidence&quot;</span>) <span class="co"># new data with predictions and confidence intervals</span></span>
<span id="cb2-26"><a href="source-code.html#cb2-26" tabindex="-1"></a>    newdata<span class="sc">$</span>pred <span class="ot">&lt;-</span> pm[,<span class="dv">1</span>] <span class="co"># add prediction values to new data.</span></span>
<span id="cb2-27"><a href="source-code.html#cb2-27" tabindex="-1"></a>    newdata<span class="sc">$</span>predmin <span class="ot">&lt;-</span> pm[,<span class="dv">2</span>] <span class="co"># add lower bounderies to new data </span></span>
<span id="cb2-28"><a href="source-code.html#cb2-28" tabindex="-1"></a>    newdata<span class="sc">$</span>predmax <span class="ot">&lt;-</span> pm[,<span class="dv">3</span>] <span class="co"># add upper bounderies to new data </span></span>
<span id="cb2-29"><a href="source-code.html#cb2-29" tabindex="-1"></a>    newdata<span class="sc">$</span>cline <span class="ot">&lt;-</span> line<span class="sc">$</span>cline[<span class="dv">1</span>] <span class="co"># add column with cline.</span></span>
<span id="cb2-30"><a href="source-code.html#cb2-30" tabindex="-1"></a>    newdata<span class="sc">$</span>expcode <span class="ot">=</span> <span class="fu">unique</span>(df<span class="sc">$</span>expcode)</span>
<span id="cb2-31"><a href="source-code.html#cb2-31" tabindex="-1"></a>    <span class="fu">return</span>(newdata)</span>
<span id="cb2-32"><a href="source-code.html#cb2-32" tabindex="-1"></a>  }</span>
<span id="cb2-33"><a href="source-code.html#cb2-33" tabindex="-1"></a>  </span>
<span id="cb2-34"><a href="source-code.html#cb2-34" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb2-35"><a href="source-code.html#cb2-35" tabindex="-1"></a>  data_frames <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, fit_model) </span>
<span id="cb2-36"><a href="source-code.html#cb2-36" tabindex="-1"></a>  </span>
<span id="cb2-37"><a href="source-code.html#cb2-37" tabindex="-1"></a>  </span>
<span id="cb2-38"><a href="source-code.html#cb2-38" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb2-39"><a href="source-code.html#cb2-39" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, data_frames)</span>
<span id="cb2-40"><a href="source-code.html#cb2-40" tabindex="-1"></a>  </span>
<span id="cb2-41"><a href="source-code.html#cb2-41" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb2-42"><a href="source-code.html#cb2-42" tabindex="-1"></a>  </span>
<span id="cb2-43"><a href="source-code.html#cb2-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="ic50_rad-script-ic50_fun-function" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> ic50_rad script: ‘IC50_fun’ function<a href="source-code.html#ic50_rad-script-ic50_fun-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="source-code.html#cb3-1" tabindex="-1"></a>IC50_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb3-2"><a href="source-code.html#cb3-2" tabindex="-1"></a>  </span>
<span id="cb3-3"><a href="source-code.html#cb3-3" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb3-4"><a href="source-code.html#cb3-4" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">is.null</span>(rad), d<span class="ot">&lt;-</span>df, d <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> rad))</span>
<span id="cb3-5"><a href="source-code.html#cb3-5" tabindex="-1"></a>  </span>
<span id="cb3-6"><a href="source-code.html#cb3-6" tabindex="-1"></a>  lines <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb3-7"><a href="source-code.html#cb3-7" tabindex="-1"></a>  <span class="co"># for loop, checks if CF$cline is in vector lines, if so, do nothing, else append to vector. </span></span>
<span id="cb3-8"><a href="source-code.html#cb3-8" tabindex="-1"></a>  <span class="cf">for</span> (line <span class="cf">in</span> d<span class="sc">$</span>cline) (</span>
<span id="cb3-9"><a href="source-code.html#cb3-9" tabindex="-1"></a>    <span class="fu">ifelse</span>(line <span class="sc">%in%</span> lines, <span class="cn">NA</span>, lines <span class="ot">&lt;-</span> <span class="fu">c</span>(lines, line)))</span>
<span id="cb3-10"><a href="source-code.html#cb3-10" tabindex="-1"></a>  </span>
<span id="cb3-11"><a href="source-code.html#cb3-11" tabindex="-1"></a>  <span class="co"># separate into dataframes according to cline</span></span>
<span id="cb3-12"><a href="source-code.html#cb3-12" tabindex="-1"></a>  per_line <span class="ot">&lt;-</span> <span class="fu">split</span>(d, <span class="at">f =</span> d<span class="sc">$</span>cline, <span class="at">drop =</span><span class="cn">TRUE</span>)</span>
<span id="cb3-13"><a href="source-code.html#cb3-13" tabindex="-1"></a>  </span>
<span id="cb3-14"><a href="source-code.html#cb3-14" tabindex="-1"></a>  </span>
<span id="cb3-15"><a href="source-code.html#cb3-15" tabindex="-1"></a>  <span class="co"># Function to make model for every line and dataframe with predicted data for each line. </span></span>
<span id="cb3-16"><a href="source-code.html#cb3-16" tabindex="-1"></a>  ic50<span class="ot">=</span> <span class="cf">function</span>(line) {</span>
<span id="cb3-17"><a href="source-code.html#cb3-17" tabindex="-1"></a>    <span class="co"># line is a dataframe. Outputs the fits of norm and dose</span></span>
<span id="cb3-18"><a href="source-code.html#cb3-18" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">drm</span>(relative <span class="sc">~</span> dose, <span class="co"># define y -axis (ncolonies) and x-axis (dose)</span></span>
<span id="cb3-19"><a href="source-code.html#cb3-19" tabindex="-1"></a>               <span class="at">data =</span> line, <span class="co"># defines dataframe</span></span>
<span id="cb3-20"><a href="source-code.html#cb3-20" tabindex="-1"></a>               <span class="at">fct =</span> <span class="fu">LL.4</span> (<span class="at">names =</span> <span class="fu">c</span>(<span class="st">&#39;Slope&#39;</span>, <span class="st">&quot;Lower Limit&quot;</span>, <span class="st">&quot;Upper Limit&quot;</span>, <span class="st">&quot;IC50&quot;</span>))) <span class="co"># defines to fit a log-losistic model)</span></span>
<span id="cb3-21"><a href="source-code.html#cb3-21" tabindex="-1"></a>    </span>
<span id="cb3-22"><a href="source-code.html#cb3-22" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">coef</span>(fit)[<span class="dv">4</span>])</span>
<span id="cb3-23"><a href="source-code.html#cb3-23" tabindex="-1"></a>  }</span>
<span id="cb3-24"><a href="source-code.html#cb3-24" tabindex="-1"></a>  </span>
<span id="cb3-25"><a href="source-code.html#cb3-25" tabindex="-1"></a>  <span class="co"># Store in vector</span></span>
<span id="cb3-26"><a href="source-code.html#cb3-26" tabindex="-1"></a>  IC50_vector <span class="ot">&lt;-</span> <span class="fu">lapply</span>(per_line, ic50) </span>
<span id="cb3-27"><a href="source-code.html#cb3-27" tabindex="-1"></a>  </span>
<span id="cb3-28"><a href="source-code.html#cb3-28" tabindex="-1"></a>  <span class="co">#merge dataframes</span></span>
<span id="cb3-29"><a href="source-code.html#cb3-29" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">do.call</span>(rbind, IC50_vector)</span>
<span id="cb3-30"><a href="source-code.html#cb3-30" tabindex="-1"></a>  </span>
<span id="cb3-31"><a href="source-code.html#cb3-31" tabindex="-1"></a>  <span class="fu">return</span>(df) </span>
<span id="cb3-32"><a href="source-code.html#cb3-32" tabindex="-1"></a>  </span>
<span id="cb3-33"><a href="source-code.html#cb3-33" tabindex="-1"></a>} </span></code></pre></div>
</div>
<div id="relativize_dr-script-relative-function" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> relativize_dr script: ‘relative’ function<a href="source-code.html#relativize_dr-script-relative-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="source-code.html#cb4-1" tabindex="-1"></a>relative <span class="ot">&lt;-</span> <span class="cf">function</span>(df, rad) {</span>
<span id="cb4-2"><a href="source-code.html#cb4-2" tabindex="-1"></a>  <span class="co">#&#39; @param df with dose response data and columns cline, dose, Exp, </span></span>
<span id="cb4-3"><a href="source-code.html#cb4-3" tabindex="-1"></a>  <span class="co">#&#39; rec_rad, expcode, ncolonies</span></span>
<span id="cb4-4"><a href="source-code.html#cb4-4" tabindex="-1"></a>  <span class="co">#&#39; @param rad if line received radiation (T or F), stored in column rec_rad</span></span>
<span id="cb4-5"><a href="source-code.html#cb4-5" tabindex="-1"></a>  <span class="co">#&#39; @return relativized data where dose = 0 is used as 100%</span></span>
<span id="cb4-6"><a href="source-code.html#cb4-6" tabindex="-1"></a>  </span>
<span id="cb4-7"><a href="source-code.html#cb4-7" tabindex="-1"></a>  <span class="co"># defining variables to work with</span></span>
<span id="cb4-8"><a href="source-code.html#cb4-8" tabindex="-1"></a>  <span class="fu">ifelse</span>(rad <span class="sc">==</span> T, d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">1</span>), d<span class="ot">&lt;-</span><span class="fu">subset</span>(df, rec_rad <span class="sc">==</span> <span class="dv">0</span>) )</span>
<span id="cb4-9"><a href="source-code.html#cb4-9" tabindex="-1"></a>  </span>
<span id="cb4-10"><a href="source-code.html#cb4-10" tabindex="-1"></a>  <span class="co"># define ncolonies</span></span>
<span id="cb4-11"><a href="source-code.html#cb4-11" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;ncolonies&quot;</span></span>
<span id="cb4-12"><a href="source-code.html#cb4-12" tabindex="-1"></a>  </span>
<span id="cb4-13"><a href="source-code.html#cb4-13" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb4-14"><a href="source-code.html#cb4-14" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb4-15"><a href="source-code.html#cb4-15" tabindex="-1"></a>  </span>
<span id="cb4-16"><a href="source-code.html#cb4-16" tabindex="-1"></a>  <span class="co"># extracts number of experiemnts </span></span>
<span id="cb4-17"><a href="source-code.html#cb4-17" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(d<span class="sc">$</span>Exp))</span>
<span id="cb4-18"><a href="source-code.html#cb4-18" tabindex="-1"></a>  </span>
<span id="cb4-19"><a href="source-code.html#cb4-19" tabindex="-1"></a>  <span class="co"># Takes mean every nth row</span></span>
<span id="cb4-20"><a href="source-code.html#cb4-20" tabindex="-1"></a>  </span>
<span id="cb4-21"><a href="source-code.html#cb4-21" tabindex="-1"></a>  mean_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-22"><a href="source-code.html#cb4-22" tabindex="-1"></a>    m <span class="ot">&lt;-</span> <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-23"><a href="source-code.html#cb4-23" tabindex="-1"></a>    <span class="fu">return</span>(m)</span>
<span id="cb4-24"><a href="source-code.html#cb4-24" tabindex="-1"></a>  }</span>
<span id="cb4-25"><a href="source-code.html#cb4-25" tabindex="-1"></a>  </span>
<span id="cb4-26"><a href="source-code.html#cb4-26" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))), mean_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb4-27"><a href="source-code.html#cb4-27" tabindex="-1"></a>  <span class="co"># Duplicates (n=2) or triplicates (n=3) the rows.</span></span>
<span id="cb4-28"><a href="source-code.html#cb4-28" tabindex="-1"></a>  mean <span class="ot">&lt;-</span> mean[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(mean)), <span class="at">each =</span> n), ]</span>
<span id="cb4-29"><a href="source-code.html#cb4-29" tabindex="-1"></a>  <span class="co"># Adds everything to the dataframe. </span></span>
<span id="cb4-30"><a href="source-code.html#cb4-30" tabindex="-1"></a>  d[<span class="st">&quot;mean&quot;</span>] <span class="ot">&lt;-</span> mean</span>
<span id="cb4-31"><a href="source-code.html#cb4-31" tabindex="-1"></a>  </span>
<span id="cb4-32"><a href="source-code.html#cb4-32" tabindex="-1"></a>  <span class="co"># Select mean values of 0 concentration</span></span>
<span id="cb4-33"><a href="source-code.html#cb4-33" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb4-34"><a href="source-code.html#cb4-34" tabindex="-1"></a>    <span class="fu">group_by</span>(cline) <span class="sc">%&gt;%</span></span>
<span id="cb4-35"><a href="source-code.html#cb4-35" tabindex="-1"></a>    <span class="fu">arrange</span>(cline,dose)</span>
<span id="cb4-36"><a href="source-code.html#cb4-36" tabindex="-1"></a>  </span>
<span id="cb4-37"><a href="source-code.html#cb4-37" tabindex="-1"></a>  first <span class="ot">&lt;-</span> d[d<span class="sc">$</span>dose<span class="sc">==</span><span class="dv">0</span>, ] <span class="sc">%&gt;%</span></span>
<span id="cb4-38"><a href="source-code.html#cb4-38" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(cline, <span class="at">value100=</span>mean)</span>
<span id="cb4-39"><a href="source-code.html#cb4-39" tabindex="-1"></a>  </span>
<span id="cb4-40"><a href="source-code.html#cb4-40" tabindex="-1"></a>  <span class="co"># Make cline a factor</span></span>
<span id="cb4-41"><a href="source-code.html#cb4-41" tabindex="-1"></a>  d<span class="sc">$</span>cline<span class="ot">&lt;-</span><span class="fu">factor</span>(d<span class="sc">$</span>cline)</span>
<span id="cb4-42"><a href="source-code.html#cb4-42" tabindex="-1"></a>  </span>
<span id="cb4-43"><a href="source-code.html#cb4-43" tabindex="-1"></a>  <span class="co"># Merge first original (d)</span></span>
<span id="cb4-44"><a href="source-code.html#cb4-44" tabindex="-1"></a>  d <span class="ot">&lt;-</span> d <span class="sc">%&gt;%</span></span>
<span id="cb4-45"><a href="source-code.html#cb4-45" tabindex="-1"></a>    <span class="fu">merge</span>(first,<span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;cline&quot;</span>))</span>
<span id="cb4-46"><a href="source-code.html#cb4-46" tabindex="-1"></a>  </span>
<span id="cb4-47"><a href="source-code.html#cb4-47" tabindex="-1"></a>  <span class="co"># Extract only every nth row.</span></span>
<span id="cb4-48"><a href="source-code.html#cb4-48" tabindex="-1"></a>  d <span class="ot">=</span> d[<span class="fu">seq</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(d), n), ]</span>
<span id="cb4-49"><a href="source-code.html#cb4-49" tabindex="-1"></a>  </span>
<span id="cb4-50"><a href="source-code.html#cb4-50" tabindex="-1"></a>  <span class="co"># make new column with relative, called relative</span></span>
<span id="cb4-51"><a href="source-code.html#cb4-51" tabindex="-1"></a>  d<span class="sc">$</span>relative <span class="ot">=</span> d<span class="sc">$</span>ncolonies<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb4-52"><a href="source-code.html#cb4-52" tabindex="-1"></a>  d<span class="sc">$</span>relative_mean <span class="ot">=</span> d<span class="sc">$</span>mean<span class="sc">/</span>d<span class="sc">$</span>value100</span>
<span id="cb4-53"><a href="source-code.html#cb4-53" tabindex="-1"></a>  </span>
<span id="cb4-54"><a href="source-code.html#cb4-54" tabindex="-1"></a>  d<span class="ot">&lt;-</span><span class="fu">subset</span>(d, <span class="at">select=</span><span class="sc">-</span><span class="fu">c</span>(mean,value100))</span>
<span id="cb4-55"><a href="source-code.html#cb4-55" tabindex="-1"></a>  </span>
<span id="cb4-56"><a href="source-code.html#cb4-56" tabindex="-1"></a>  <span class="co"># compute relative standard error of the mean</span></span>
<span id="cb4-57"><a href="source-code.html#cb4-57" tabindex="-1"></a>  <span class="co"># define the column with sem in it</span></span>
<span id="cb4-58"><a href="source-code.html#cb4-58" tabindex="-1"></a>  column <span class="ot">&lt;-</span> <span class="st">&quot;relative&quot;</span></span>
<span id="cb4-59"><a href="source-code.html#cb4-59" tabindex="-1"></a>  </span>
<span id="cb4-60"><a href="source-code.html#cb4-60" tabindex="-1"></a>  <span class="co"># Extracts the column name.</span></span>
<span id="cb4-61"><a href="source-code.html#cb4-61" tabindex="-1"></a>  col <span class="ot">&lt;-</span> d[<span class="fu">c</span>(column)]</span>
<span id="cb4-62"><a href="source-code.html#cb4-62" tabindex="-1"></a>  </span>
<span id="cb4-63"><a href="source-code.html#cb4-63" tabindex="-1"></a>  </span>
<span id="cb4-64"><a href="source-code.html#cb4-64" tabindex="-1"></a>  sem_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb4-65"><a href="source-code.html#cb4-65" tabindex="-1"></a>    std <span class="ot">&lt;-</span> <span class="fu">sd</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-66"><a href="source-code.html#cb4-66" tabindex="-1"></a>    vector <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(x)</span>
<span id="cb4-67"><a href="source-code.html#cb4-67" tabindex="-1"></a>    </span>
<span id="cb4-68"><a href="source-code.html#cb4-68" tabindex="-1"></a>    sem <span class="ot">&lt;-</span> std<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">length</span>(vector))</span>
<span id="cb4-69"><a href="source-code.html#cb4-69" tabindex="-1"></a>    <span class="fu">return</span>(sem)</span>
<span id="cb4-70"><a href="source-code.html#cb4-70" tabindex="-1"></a>  }</span>
<span id="cb4-71"><a href="source-code.html#cb4-71" tabindex="-1"></a>  </span>
<span id="cb4-72"><a href="source-code.html#cb4-72" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(col,<span class="fu">list</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(col)<span class="sc">%/%</span>n<span class="sc">+</span><span class="dv">1</span>),<span class="at">each=</span>n,<span class="at">len=</span><span class="fu">nrow</span>(col))),sem_fun)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb4-73"><a href="source-code.html#cb4-73" tabindex="-1"></a>  sem <span class="ot">&lt;-</span> sem[<span class="fu">rep</span>(<span class="fu">seq_len</span>(<span class="fu">nrow</span>(sem)), <span class="at">each =</span> n), ]</span>
<span id="cb4-74"><a href="source-code.html#cb4-74" tabindex="-1"></a>  d[<span class="st">&quot;relative_sem&quot;</span>] <span class="ot">&lt;-</span> sem</span>
<span id="cb4-75"><a href="source-code.html#cb4-75" tabindex="-1"></a>  </span>
<span id="cb4-76"><a href="source-code.html#cb4-76" tabindex="-1"></a>  <span class="fu">return</span>(d)</span>
<span id="cb4-77"><a href="source-code.html#cb4-77" tabindex="-1"></a>  </span>
<span id="cb4-78"><a href="source-code.html#cb4-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> quality_functions script: ‘quality_check’, ‘quality_select’ and ‘check_quality’ functions<a href="source-code.html#quality_functions-script-quality_check-quality_select-and-check_quality-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="source-code.html#cb5-1" tabindex="-1"></a>quality_check <span class="ot">&lt;-</span> <span class="cf">function</span>(inputdir, model) {</span>
<span id="cb5-2"><a href="source-code.html#cb5-2" tabindex="-1"></a>  <span class="co">#&#39; @param inputdir is directory containing the standard aneufinder output for each plate</span></span>
<span id="cb5-3"><a href="source-code.html#cb5-3" tabindex="-1"></a>  <span class="co">#&#39; model is which model to use for the quality check: either dnacopy, edivisive or HMM</span></span>
<span id="cb5-4"><a href="source-code.html#cb5-4" tabindex="-1"></a>  <span class="co">#&#39; quality_check will make quality metrics for each plate</span></span>
<span id="cb5-5"><a href="source-code.html#cb5-5" tabindex="-1"></a>  <span class="co">#&#39; quality_check assumes that rdaBaseDirectory contains folder with models for each plate</span></span>
<span id="cb5-6"><a href="source-code.html#cb5-6" tabindex="-1"></a>  <span class="co">#&#39; each plate folder has to contain a MODEL folder and a method-&#39;dnacopy, edivisive, or HMM&#39; folder</span></span>
<span id="cb5-7"><a href="source-code.html#cb5-7" tabindex="-1"></a>  <span class="co">#&#39; @example quality_check(inputdir = rdaBaseDirectory, model = &#39;edivisive&#39;)</span></span>
<span id="cb5-8"><a href="source-code.html#cb5-8" tabindex="-1"></a>  </span>
<span id="cb5-9"><a href="source-code.html#cb5-9" tabindex="-1"></a>  <span class="co"># go to folder with appropiate files</span></span>
<span id="cb5-10"><a href="source-code.html#cb5-10" tabindex="-1"></a>  rda_folder <span class="ot">&lt;-</span> <span class="fu">paste0</span>(inputdir, <span class="st">&#39;/MODELS&#39;</span>, <span class="st">&#39;/method-&#39;</span>, model)</span>
<span id="cb5-11"><a href="source-code.html#cb5-11" tabindex="-1"></a>  </span>
<span id="cb5-12"><a href="source-code.html#cb5-12" tabindex="-1"></a>  <span class="co"># extract files</span></span>
<span id="cb5-13"><a href="source-code.html#cb5-13" tabindex="-1"></a>  rda_files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(rda_folder, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-14"><a href="source-code.html#cb5-14" tabindex="-1"></a>  </span>
<span id="cb5-15"><a href="source-code.html#cb5-15" tabindex="-1"></a>  <span class="co"># run quality check on each file</span></span>
<span id="cb5-16"><a href="source-code.html#cb5-16" tabindex="-1"></a>  cl <span class="ot">&lt;-</span></span>
<span id="cb5-17"><a href="source-code.html#cb5-17" tabindex="-1"></a>    <span class="fu">clusterByQuality</span>(</span>
<span id="cb5-18"><a href="source-code.html#cb5-18" tabindex="-1"></a>      rda_files,</span>
<span id="cb5-19"><a href="source-code.html#cb5-19" tabindex="-1"></a>      <span class="at">measures =</span> <span class="fu">c</span>(</span>
<span id="cb5-20"><a href="source-code.html#cb5-20" tabindex="-1"></a>        <span class="st">&#39;spikiness&#39;</span>,</span>
<span id="cb5-21"><a href="source-code.html#cb5-21" tabindex="-1"></a>        <span class="st">&#39;num.segments&#39;</span>,</span>
<span id="cb5-22"><a href="source-code.html#cb5-22" tabindex="-1"></a>        <span class="st">&#39;entropy&#39;</span>,</span>
<span id="cb5-23"><a href="source-code.html#cb5-23" tabindex="-1"></a>        <span class="st">&#39;bhattacharyya&#39;</span>,</span>
<span id="cb5-24"><a href="source-code.html#cb5-24" tabindex="-1"></a>        <span class="st">&#39;sos&#39;</span></span>
<span id="cb5-25"><a href="source-code.html#cb5-25" tabindex="-1"></a>      )</span>
<span id="cb5-26"><a href="source-code.html#cb5-26" tabindex="-1"></a>    )</span>
<span id="cb5-27"><a href="source-code.html#cb5-27" tabindex="-1"></a>  </span>
<span id="cb5-28"><a href="source-code.html#cb5-28" tabindex="-1"></a>  <span class="fu">return</span>(cl)</span>
<span id="cb5-29"><a href="source-code.html#cb5-29" tabindex="-1"></a>  </span>
<span id="cb5-30"><a href="source-code.html#cb5-30" tabindex="-1"></a>}</span>
<span id="cb5-31"><a href="source-code.html#cb5-31" tabindex="-1"></a></span>
<span id="cb5-32"><a href="source-code.html#cb5-32" tabindex="-1"></a>quality_select <span class="ot">&lt;-</span> <span class="cf">function</span>(cl, spik, bhat) {</span>
<span id="cb5-33"><a href="source-code.html#cb5-33" tabindex="-1"></a>  <span class="co">#&#39; selects all files that meets the defined quality requirements</span></span>
<span id="cb5-34"><a href="source-code.html#cb5-34" tabindex="-1"></a>  <span class="co">#&#39; @param cl output of the clusterByQuality function in AneuFinder</span></span>
<span id="cb5-35"><a href="source-code.html#cb5-35" tabindex="-1"></a>  <span class="co">#&#39; @param spik the spikiness treshhold. All clusters with spikiness above</span></span>
<span id="cb5-36"><a href="source-code.html#cb5-36" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb5-37"><a href="source-code.html#cb5-37" tabindex="-1"></a>  <span class="co">#&#39; @param bhat the bhattacharrya score. All clusters with bhattacharrya below</span></span>
<span id="cb5-38"><a href="source-code.html#cb5-38" tabindex="-1"></a>  <span class="co">#&#39; the treshold will be removed</span></span>
<span id="cb5-39"><a href="source-code.html#cb5-39" tabindex="-1"></a>  <span class="co">#&#39; @return returns a vector of selected files paths</span></span>
<span id="cb5-40"><a href="source-code.html#cb5-40" tabindex="-1"></a>  </span>
<span id="cb5-41"><a href="source-code.html#cb5-41" tabindex="-1"></a>  <span class="co"># convert to df to allow easy wrangling</span></span>
<span id="cb5-42"><a href="source-code.html#cb5-42" tabindex="-1"></a>  cl_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb5-43"><a href="source-code.html#cb5-43" tabindex="-1"></a>  </span>
<span id="cb5-44"><a href="source-code.html#cb5-44" tabindex="-1"></a>  <span class="co"># remove higher than spik</span></span>
<span id="cb5-45"><a href="source-code.html#cb5-45" tabindex="-1"></a>  spik <span class="ot">&lt;-</span> <span class="fu">subset</span>(cl_df, spikiness <span class="sc">&lt;</span> spik)</span>
<span id="cb5-46"><a href="source-code.html#cb5-46" tabindex="-1"></a>  </span>
<span id="cb5-47"><a href="source-code.html#cb5-47" tabindex="-1"></a>  <span class="co"># remove lower than bhat</span></span>
<span id="cb5-48"><a href="source-code.html#cb5-48" tabindex="-1"></a>  spik_bhat <span class="ot">&lt;-</span> <span class="fu">subset</span>(spik,  bhattacharyya <span class="sc">&gt;</span> bhat)</span>
<span id="cb5-49"><a href="source-code.html#cb5-49" tabindex="-1"></a>  </span>
<span id="cb5-50"><a href="source-code.html#cb5-50" tabindex="-1"></a>  cluster_n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(spik_bhat)</span>
<span id="cb5-51"><a href="source-code.html#cb5-51" tabindex="-1"></a>  </span>
<span id="cb5-52"><a href="source-code.html#cb5-52" tabindex="-1"></a>  <span class="co"># create vector of selected files</span></span>
<span id="cb5-53"><a href="source-code.html#cb5-53" tabindex="-1"></a>  selected.files <span class="ot">&lt;-</span> <span class="fu">unlist</span>(cl<span class="sc">$</span>classification[<span class="dv">0</span><span class="sc">:</span>cluster_n])</span>
<span id="cb5-54"><a href="source-code.html#cb5-54" tabindex="-1"></a>  </span>
<span id="cb5-55"><a href="source-code.html#cb5-55" tabindex="-1"></a>  <span class="fu">return</span>(selected.files)</span>
<span id="cb5-56"><a href="source-code.html#cb5-56" tabindex="-1"></a>  </span>
<span id="cb5-57"><a href="source-code.html#cb5-57" tabindex="-1"></a>}</span>
<span id="cb5-58"><a href="source-code.html#cb5-58" tabindex="-1"></a></span>
<span id="cb5-59"><a href="source-code.html#cb5-59" tabindex="-1"></a>check_quality <span class="ot">&lt;-</span> <span class="cf">function</span>(cl) {</span>
<span id="cb5-60"><a href="source-code.html#cb5-60" tabindex="-1"></a>  <span class="fu">return</span>(cl<span class="sc">$</span>parameters)</span>
<span id="cb5-61"><a href="source-code.html#cb5-61" tabindex="-1"></a>}</span>
<span id="cb5-62"><a href="source-code.html#cb5-62" tabindex="-1"></a>  </span>
<span id="cb5-63"><a href="source-code.html#cb5-63" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> plot_pca_kmeans script: ‘draw_pca’, ‘k_cluster’, ‘draw_pca_double’ and ‘k_cluster_double’ functions<a href="source-code.html#plot_pca_kmeans-script-draw_pca-k_cluster-draw_pca_double-and-k_cluster_double-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="source-code.html#cb6-1" tabindex="-1"></a>draw_pca <span class="ot">&lt;-</span></span>
<span id="cb6-2"><a href="source-code.html#cb6-2" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb6-3"><a href="source-code.html#cb6-3" tabindex="-1"></a>           list_files2,</span>
<span id="cb6-4"><a href="source-code.html#cb6-4" tabindex="-1"></a>           size,</span>
<span id="cb6-5"><a href="source-code.html#cb6-5" tabindex="-1"></a>           legend_position,</span>
<span id="cb6-6"><a href="source-code.html#cb6-6" tabindex="-1"></a>           baseline_kra,</span>
<span id="cb6-7"><a href="source-code.html#cb6-7" tabindex="-1"></a>           recurrence_kra,</span>
<span id="cb6-8"><a href="source-code.html#cb6-8" tabindex="-1"></a>           <span class="at">ssdna004 =</span> F) {</span>
<span id="cb6-9"><a href="source-code.html#cb6-9" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb6-10"><a href="source-code.html#cb6-10" tabindex="-1"></a>    </span>
<span id="cb6-11"><a href="source-code.html#cb6-11" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-12"><a href="source-code.html#cb6-12" tabindex="-1"></a>      cells,</span>
<span id="cb6-13"><a href="source-code.html#cb6-13" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb6-14"><a href="source-code.html#cb6-14" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb6-15"><a href="source-code.html#cb6-15" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb6-16"><a href="source-code.html#cb6-16" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb6-17"><a href="source-code.html#cb6-17" tabindex="-1"></a>    )</span>
<span id="cb6-18"><a href="source-code.html#cb6-18" tabindex="-1"></a>    </span>
<span id="cb6-19"><a href="source-code.html#cb6-19" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb6-20"><a href="source-code.html#cb6-20" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-21"><a href="source-code.html#cb6-21" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-22"><a href="source-code.html#cb6-22" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb6-23"><a href="source-code.html#cb6-23" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb6-24"><a href="source-code.html#cb6-24" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb6-25"><a href="source-code.html#cb6-25" tabindex="-1"></a>    </span>
<span id="cb6-26"><a href="source-code.html#cb6-26" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb6-27"><a href="source-code.html#cb6-27" tabindex="-1"></a>    </span>
<span id="cb6-28"><a href="source-code.html#cb6-28" tabindex="-1"></a>    col_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&quot;#999999&quot;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>)))</span>
<span id="cb6-29"><a href="source-code.html#cb6-29" tabindex="-1"></a>    label_vals <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(ssdna004 <span class="sc">==</span> T, <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Recurrence&#39;</span>, <span class="st">&#39;Baseline&#39;</span>)), <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence&#39;</span>)))</span>
<span id="cb6-30"><a href="source-code.html#cb6-30" tabindex="-1"></a>    </span>
<span id="cb6-31"><a href="source-code.html#cb6-31" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb6-32"><a href="source-code.html#cb6-32" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb6-33"><a href="source-code.html#cb6-33" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb6-34"><a href="source-code.html#cb6-34" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb6-35"><a href="source-code.html#cb6-35" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb6-36"><a href="source-code.html#cb6-36" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb6-37"><a href="source-code.html#cb6-37" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb6-38"><a href="source-code.html#cb6-38" tabindex="-1"></a>        <span class="at">values =</span> col_vals[[<span class="dv">1</span>]],</span>
<span id="cb6-39"><a href="source-code.html#cb6-39" tabindex="-1"></a>        <span class="at">labels =</span> label_vals[[<span class="dv">1</span>]]</span>
<span id="cb6-40"><a href="source-code.html#cb6-40" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb6-41"><a href="source-code.html#cb6-41" tabindex="-1"></a>      </span>
<span id="cb6-42"><a href="source-code.html#cb6-42" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb6-43"><a href="source-code.html#cb6-43" tabindex="-1"></a>      </span>
<span id="cb6-44"><a href="source-code.html#cb6-44" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb6-45"><a href="source-code.html#cb6-45" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb6-46"><a href="source-code.html#cb6-46" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb6-47"><a href="source-code.html#cb6-47" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb6-48"><a href="source-code.html#cb6-48" tabindex="-1"></a>      </span>
<span id="cb6-49"><a href="source-code.html#cb6-49" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb6-50"><a href="source-code.html#cb6-50" tabindex="-1"></a>    p</span>
<span id="cb6-51"><a href="source-code.html#cb6-51" tabindex="-1"></a>    </span>
<span id="cb6-52"><a href="source-code.html#cb6-52" tabindex="-1"></a>  </span>
<span id="cb6-53"><a href="source-code.html#cb6-53" tabindex="-1"></a>    </span>
<span id="cb6-54"><a href="source-code.html#cb6-54" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb6-55"><a href="source-code.html#cb6-55" tabindex="-1"></a>    </span>
<span id="cb6-56"><a href="source-code.html#cb6-56" tabindex="-1"></a>  }</span>
<span id="cb6-57"><a href="source-code.html#cb6-57" tabindex="-1"></a></span>
<span id="cb6-58"><a href="source-code.html#cb6-58" tabindex="-1"></a></span>
<span id="cb6-59"><a href="source-code.html#cb6-59" tabindex="-1"></a>k_cluster <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb6-60"><a href="source-code.html#cb6-60" tabindex="-1"></a>                      list_files2,</span>
<span id="cb6-61"><a href="source-code.html#cb6-61" tabindex="-1"></a>                      cluster_n,</span>
<span id="cb6-62"><a href="source-code.html#cb6-62" tabindex="-1"></a>                      cols,</span>
<span id="cb6-63"><a href="source-code.html#cb6-63" tabindex="-1"></a>                      labels,</span>
<span id="cb6-64"><a href="source-code.html#cb6-64" tabindex="-1"></a>                      legend_position,</span>
<span id="cb6-65"><a href="source-code.html#cb6-65" tabindex="-1"></a>                      normalize,</span>
<span id="cb6-66"><a href="source-code.html#cb6-66" tabindex="-1"></a>                      <span class="at">return_elbow =</span> F) {</span>
<span id="cb6-67"><a href="source-code.html#cb6-67" tabindex="-1"></a>  </span>
<span id="cb6-68"><a href="source-code.html#cb6-68" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb6-69"><a href="source-code.html#cb6-69" tabindex="-1"></a>  </span>
<span id="cb6-70"><a href="source-code.html#cb6-70" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb6-71"><a href="source-code.html#cb6-71" tabindex="-1"></a>  </span>
<span id="cb6-72"><a href="source-code.html#cb6-72" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-73"><a href="source-code.html#cb6-73" tabindex="-1"></a>    cells,</span>
<span id="cb6-74"><a href="source-code.html#cb6-74" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-75"><a href="source-code.html#cb6-75" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb6-76"><a href="source-code.html#cb6-76" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb6-77"><a href="source-code.html#cb6-77" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-78"><a href="source-code.html#cb6-78" tabindex="-1"></a>  )</span>
<span id="cb6-79"><a href="source-code.html#cb6-79" tabindex="-1"></a></span>
<span id="cb6-80"><a href="source-code.html#cb6-80" tabindex="-1"></a>  </span>
<span id="cb6-81"><a href="source-code.html#cb6-81" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-82"><a href="source-code.html#cb6-82" tabindex="-1"></a>    cells,</span>
<span id="cb6-83"><a href="source-code.html#cb6-83" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-84"><a href="source-code.html#cb6-84" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb6-85"><a href="source-code.html#cb6-85" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb6-86"><a href="source-code.html#cb6-86" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-87"><a href="source-code.html#cb6-87" tabindex="-1"></a>  )</span>
<span id="cb6-88"><a href="source-code.html#cb6-88" tabindex="-1"></a>  </span>
<span id="cb6-89"><a href="source-code.html#cb6-89" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-90"><a href="source-code.html#cb6-90" tabindex="-1"></a>    cells,</span>
<span id="cb6-91"><a href="source-code.html#cb6-91" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-92"><a href="source-code.html#cb6-92" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb6-93"><a href="source-code.html#cb6-93" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb6-94"><a href="source-code.html#cb6-94" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-95"><a href="source-code.html#cb6-95" tabindex="-1"></a>  )</span>
<span id="cb6-96"><a href="source-code.html#cb6-96" tabindex="-1"></a>  </span>
<span id="cb6-97"><a href="source-code.html#cb6-97" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-98"><a href="source-code.html#cb6-98" tabindex="-1"></a>    cells,</span>
<span id="cb6-99"><a href="source-code.html#cb6-99" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-100"><a href="source-code.html#cb6-100" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb6-101"><a href="source-code.html#cb6-101" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb6-102"><a href="source-code.html#cb6-102" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-103"><a href="source-code.html#cb6-103" tabindex="-1"></a>  )</span>
<span id="cb6-104"><a href="source-code.html#cb6-104" tabindex="-1"></a>  </span>
<span id="cb6-105"><a href="source-code.html#cb6-105" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-106"><a href="source-code.html#cb6-106" tabindex="-1"></a>    cells,</span>
<span id="cb6-107"><a href="source-code.html#cb6-107" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-108"><a href="source-code.html#cb6-108" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb6-109"><a href="source-code.html#cb6-109" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb6-110"><a href="source-code.html#cb6-110" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-111"><a href="source-code.html#cb6-111" tabindex="-1"></a>  )</span>
<span id="cb6-112"><a href="source-code.html#cb6-112" tabindex="-1"></a>  </span>
<span id="cb6-113"><a href="source-code.html#cb6-113" tabindex="-1"></a>  </span>
<span id="cb6-114"><a href="source-code.html#cb6-114" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb6-115"><a href="source-code.html#cb6-115" tabindex="-1"></a>  </span>
<span id="cb6-116"><a href="source-code.html#cb6-116" tabindex="-1"></a>  </span>
<span id="cb6-117"><a href="source-code.html#cb6-117" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb6-118"><a href="source-code.html#cb6-118" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb6-119"><a href="source-code.html#cb6-119" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb6-120"><a href="source-code.html#cb6-120" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb6-121"><a href="source-code.html#cb6-121" tabindex="-1"></a>    </span>
<span id="cb6-122"><a href="source-code.html#cb6-122" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb6-123"><a href="source-code.html#cb6-123" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb6-124"><a href="source-code.html#cb6-124" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb6-125"><a href="source-code.html#cb6-125" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb6-126"><a href="source-code.html#cb6-126" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb6-127"><a href="source-code.html#cb6-127" tabindex="-1"></a>    }</span>
<span id="cb6-128"><a href="source-code.html#cb6-128" tabindex="-1"></a>    </span>
<span id="cb6-129"><a href="source-code.html#cb6-129" tabindex="-1"></a>    </span>
<span id="cb6-130"><a href="source-code.html#cb6-130" tabindex="-1"></a>    </span>
<span id="cb6-131"><a href="source-code.html#cb6-131" tabindex="-1"></a>    </span>
<span id="cb6-132"><a href="source-code.html#cb6-132" tabindex="-1"></a>    </span>
<span id="cb6-133"><a href="source-code.html#cb6-133" tabindex="-1"></a>  }</span>
<span id="cb6-134"><a href="source-code.html#cb6-134" tabindex="-1"></a>  </span>
<span id="cb6-135"><a href="source-code.html#cb6-135" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb6-136"><a href="source-code.html#cb6-136" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb6-137"><a href="source-code.html#cb6-137" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-138"><a href="source-code.html#cb6-138" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Elbow method&quot;</span>))</span>
<span id="cb6-139"><a href="source-code.html#cb6-139" tabindex="-1"></a>    </span>
<span id="cb6-140"><a href="source-code.html#cb6-140" tabindex="-1"></a>  }</span>
<span id="cb6-141"><a href="source-code.html#cb6-141" tabindex="-1"></a>  </span>
<span id="cb6-142"><a href="source-code.html#cb6-142" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb6-143"><a href="source-code.html#cb6-143" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb6-144"><a href="source-code.html#cb6-144" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb6-145"><a href="source-code.html#cb6-145" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb6-146"><a href="source-code.html#cb6-146" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb6-147"><a href="source-code.html#cb6-147" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb6-148"><a href="source-code.html#cb6-148" tabindex="-1"></a>  </span>
<span id="cb6-149"><a href="source-code.html#cb6-149" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb6-150"><a href="source-code.html#cb6-150" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb6-151"><a href="source-code.html#cb6-151" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb6-152"><a href="source-code.html#cb6-152" tabindex="-1"></a>  </span>
<span id="cb6-153"><a href="source-code.html#cb6-153" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb6-154"><a href="source-code.html#cb6-154" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb6-155"><a href="source-code.html#cb6-155" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-156"><a href="source-code.html#cb6-156" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-157"><a href="source-code.html#cb6-157" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb6-158"><a href="source-code.html#cb6-158" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb6-159"><a href="source-code.html#cb6-159" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb6-160"><a href="source-code.html#cb6-160" tabindex="-1"></a>  </span>
<span id="cb6-161"><a href="source-code.html#cb6-161" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb6-162"><a href="source-code.html#cb6-162" tabindex="-1"></a>  </span>
<span id="cb6-163"><a href="source-code.html#cb6-163" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb6-164"><a href="source-code.html#cb6-164" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb6-165"><a href="source-code.html#cb6-165" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb6-166"><a href="source-code.html#cb6-166" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb6-167"><a href="source-code.html#cb6-167" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb6-168"><a href="source-code.html#cb6-168" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb6-169"><a href="source-code.html#cb6-169" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb6-170"><a href="source-code.html#cb6-170" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb6-171"><a href="source-code.html#cb6-171" tabindex="-1"></a>    </span>
<span id="cb6-172"><a href="source-code.html#cb6-172" tabindex="-1"></a>    </span>
<span id="cb6-173"><a href="source-code.html#cb6-173" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb6-174"><a href="source-code.html#cb6-174" tabindex="-1"></a>    </span>
<span id="cb6-175"><a href="source-code.html#cb6-175" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb6-176"><a href="source-code.html#cb6-176" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb6-177"><a href="source-code.html#cb6-177" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb6-178"><a href="source-code.html#cb6-178" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-179"><a href="source-code.html#cb6-179" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb6-180"><a href="source-code.html#cb6-180" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb6-181"><a href="source-code.html#cb6-181" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb6-182"><a href="source-code.html#cb6-182" tabindex="-1"></a>  </span>
<span id="cb6-183"><a href="source-code.html#cb6-183" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb6-184"><a href="source-code.html#cb6-184" tabindex="-1"></a>  </span>
<span id="cb6-185"><a href="source-code.html#cb6-185" tabindex="-1"></a>  <span class="co">#ggplotly(p)</span></span>
<span id="cb6-186"><a href="source-code.html#cb6-186" tabindex="-1"></a>  </span>
<span id="cb6-187"><a href="source-code.html#cb6-187" tabindex="-1"></a>}</span>
<span id="cb6-188"><a href="source-code.html#cb6-188" tabindex="-1"></a></span>
<span id="cb6-189"><a href="source-code.html#cb6-189" tabindex="-1"></a>draw_pca_double <span class="ot">&lt;-</span></span>
<span id="cb6-190"><a href="source-code.html#cb6-190" tabindex="-1"></a>  <span class="cf">function</span>(list_files1,</span>
<span id="cb6-191"><a href="source-code.html#cb6-191" tabindex="-1"></a>           list_files2,</span>
<span id="cb6-192"><a href="source-code.html#cb6-192" tabindex="-1"></a>           list_files3,</span>
<span id="cb6-193"><a href="source-code.html#cb6-193" tabindex="-1"></a>           size,</span>
<span id="cb6-194"><a href="source-code.html#cb6-194" tabindex="-1"></a>           legend_position) {</span>
<span id="cb6-195"><a href="source-code.html#cb6-195" tabindex="-1"></a>    cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb6-196"><a href="source-code.html#cb6-196" tabindex="-1"></a>    </span>
<span id="cb6-197"><a href="source-code.html#cb6-197" tabindex="-1"></a>    </span>
<span id="cb6-198"><a href="source-code.html#cb6-198" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-199"><a href="source-code.html#cb6-199" tabindex="-1"></a>      cells,</span>
<span id="cb6-200"><a href="source-code.html#cb6-200" tabindex="-1"></a>      <span class="at">colorBy =</span> classes,</span>
<span id="cb6-201"><a href="source-code.html#cb6-201" tabindex="-1"></a>      <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb6-202"><a href="source-code.html#cb6-202" tabindex="-1"></a>      <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb6-203"><a href="source-code.html#cb6-203" tabindex="-1"></a>      <span class="at">plot =</span> F</span>
<span id="cb6-204"><a href="source-code.html#cb6-204" tabindex="-1"></a>    )</span>
<span id="cb6-205"><a href="source-code.html#cb6-205" tabindex="-1"></a>    </span>
<span id="cb6-206"><a href="source-code.html#cb6-206" tabindex="-1"></a>    df<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb6-207"><a href="source-code.html#cb6-207" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-208"><a href="source-code.html#cb6-208" tabindex="-1"></a>    num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-209"><a href="source-code.html#cb6-209" tabindex="-1"></a>    df<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df<span class="sc">$</span>class, num)</span>
<span id="cb6-210"><a href="source-code.html#cb6-210" tabindex="-1"></a>    pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">1</span>]</span>
<span id="cb6-211"><a href="source-code.html#cb6-211" tabindex="-1"></a>    pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df)[<span class="dv">2</span>]</span>
<span id="cb6-212"><a href="source-code.html#cb6-212" tabindex="-1"></a>    </span>
<span id="cb6-213"><a href="source-code.html#cb6-213" tabindex="-1"></a>    <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb6-214"><a href="source-code.html#cb6-214" tabindex="-1"></a>    </span>
<span id="cb6-215"><a href="source-code.html#cb6-215" tabindex="-1"></a>    </span>
<span id="cb6-216"><a href="source-code.html#cb6-216" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb6-217"><a href="source-code.html#cb6-217" tabindex="-1"></a>      <span class="fu">geom_point</span>(<span class="at">size =</span> size, <span class="fu">aes</span>(<span class="at">x =</span> PC1,</span>
<span id="cb6-218"><a href="source-code.html#cb6-218" tabindex="-1"></a>                                  <span class="at">y =</span> PC2,</span>
<span id="cb6-219"><a href="source-code.html#cb6-219" tabindex="-1"></a>                                  <span class="at">col =</span> class)) <span class="sc">+</span></span>
<span id="cb6-220"><a href="source-code.html#cb6-220" tabindex="-1"></a>      <span class="co"># scale_color_manual(values = c(&quot;#FFDB6D&quot;, &quot;#00AFBB&quot;)) +</span></span>
<span id="cb6-221"><a href="source-code.html#cb6-221" tabindex="-1"></a>      <span class="fu">scale_color_manual</span>(</span>
<span id="cb6-222"><a href="source-code.html#cb6-222" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&#39;&#39;</span>,</span>
<span id="cb6-223"><a href="source-code.html#cb6-223" tabindex="-1"></a>        <span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#D69C4E&quot;</span>, <span class="st">&#39;#d65c4e&#39;</span>),</span>
<span id="cb6-224"><a href="source-code.html#cb6-224" tabindex="-1"></a>        <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&#39;Baseline&#39;</span>, <span class="st">&#39;Recurrence Cycle 1&#39;</span>, <span class="st">&#39;Recurrence Cycle 2&#39;</span>)</span>
<span id="cb6-225"><a href="source-code.html#cb6-225" tabindex="-1"></a>      ) <span class="sc">+</span></span>
<span id="cb6-226"><a href="source-code.html#cb6-226" tabindex="-1"></a>      </span>
<span id="cb6-227"><a href="source-code.html#cb6-227" tabindex="-1"></a>      <span class="co"># or consider &#39;#9e1303&#39; for second cycle of recurrence</span></span>
<span id="cb6-228"><a href="source-code.html#cb6-228" tabindex="-1"></a>      </span>
<span id="cb6-229"><a href="source-code.html#cb6-229" tabindex="-1"></a>      <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb6-230"><a href="source-code.html#cb6-230" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb6-231"><a href="source-code.html#cb6-231" tabindex="-1"></a>            <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb6-232"><a href="source-code.html#cb6-232" tabindex="-1"></a>            <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb6-233"><a href="source-code.html#cb6-233" tabindex="-1"></a>      </span>
<span id="cb6-234"><a href="source-code.html#cb6-234" tabindex="-1"></a>      </span>
<span id="cb6-235"><a href="source-code.html#cb6-235" tabindex="-1"></a>      <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2)</span>
<span id="cb6-236"><a href="source-code.html#cb6-236" tabindex="-1"></a>    <span class="fu">return</span>(p)</span>
<span id="cb6-237"><a href="source-code.html#cb6-237" tabindex="-1"></a>    </span>
<span id="cb6-238"><a href="source-code.html#cb6-238" tabindex="-1"></a>    <span class="co"># ggplotly(p)</span></span>
<span id="cb6-239"><a href="source-code.html#cb6-239" tabindex="-1"></a>    </span>
<span id="cb6-240"><a href="source-code.html#cb6-240" tabindex="-1"></a>  }</span>
<span id="cb6-241"><a href="source-code.html#cb6-241" tabindex="-1"></a></span>
<span id="cb6-242"><a href="source-code.html#cb6-242" tabindex="-1"></a></span>
<span id="cb6-243"><a href="source-code.html#cb6-243" tabindex="-1"></a>k_cluster_double <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1,</span>
<span id="cb6-244"><a href="source-code.html#cb6-244" tabindex="-1"></a>                             list_files2,</span>
<span id="cb6-245"><a href="source-code.html#cb6-245" tabindex="-1"></a>                             list_files3,</span>
<span id="cb6-246"><a href="source-code.html#cb6-246" tabindex="-1"></a>                             </span>
<span id="cb6-247"><a href="source-code.html#cb6-247" tabindex="-1"></a>                             cluster_n,</span>
<span id="cb6-248"><a href="source-code.html#cb6-248" tabindex="-1"></a>                             cols,</span>
<span id="cb6-249"><a href="source-code.html#cb6-249" tabindex="-1"></a>                             labels,</span>
<span id="cb6-250"><a href="source-code.html#cb6-250" tabindex="-1"></a>                             legend_position,</span>
<span id="cb6-251"><a href="source-code.html#cb6-251" tabindex="-1"></a>                             normalize,</span>
<span id="cb6-252"><a href="source-code.html#cb6-252" tabindex="-1"></a>                             <span class="at">return_elbow =</span> F) {</span>
<span id="cb6-253"><a href="source-code.html#cb6-253" tabindex="-1"></a>  </span>
<span id="cb6-254"><a href="source-code.html#cb6-254" tabindex="-1"></a>  <span class="fu">require</span>(factoextra)</span>
<span id="cb6-255"><a href="source-code.html#cb6-255" tabindex="-1"></a>  </span>
<span id="cb6-256"><a href="source-code.html#cb6-256" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2, list_files3)</span>
<span id="cb6-257"><a href="source-code.html#cb6-257" tabindex="-1"></a>  </span>
<span id="cb6-258"><a href="source-code.html#cb6-258" tabindex="-1"></a>  </span>
<span id="cb6-259"><a href="source-code.html#cb6-259" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-260"><a href="source-code.html#cb6-260" tabindex="-1"></a>    cells,</span>
<span id="cb6-261"><a href="source-code.html#cb6-261" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-262"><a href="source-code.html#cb6-262" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">1</span>,</span>
<span id="cb6-263"><a href="source-code.html#cb6-263" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">2</span>,</span>
<span id="cb6-264"><a href="source-code.html#cb6-264" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-265"><a href="source-code.html#cb6-265" tabindex="-1"></a>  )</span>
<span id="cb6-266"><a href="source-code.html#cb6-266" tabindex="-1"></a>  </span>
<span id="cb6-267"><a href="source-code.html#cb6-267" tabindex="-1"></a>  </span>
<span id="cb6-268"><a href="source-code.html#cb6-268" tabindex="-1"></a>  df34 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-269"><a href="source-code.html#cb6-269" tabindex="-1"></a>    cells,</span>
<span id="cb6-270"><a href="source-code.html#cb6-270" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-271"><a href="source-code.html#cb6-271" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">3</span>,</span>
<span id="cb6-272"><a href="source-code.html#cb6-272" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">4</span>,</span>
<span id="cb6-273"><a href="source-code.html#cb6-273" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-274"><a href="source-code.html#cb6-274" tabindex="-1"></a>  )</span>
<span id="cb6-275"><a href="source-code.html#cb6-275" tabindex="-1"></a>  </span>
<span id="cb6-276"><a href="source-code.html#cb6-276" tabindex="-1"></a>  df56 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-277"><a href="source-code.html#cb6-277" tabindex="-1"></a>    cells,</span>
<span id="cb6-278"><a href="source-code.html#cb6-278" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-279"><a href="source-code.html#cb6-279" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">5</span>,</span>
<span id="cb6-280"><a href="source-code.html#cb6-280" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">6</span>,</span>
<span id="cb6-281"><a href="source-code.html#cb6-281" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-282"><a href="source-code.html#cb6-282" tabindex="-1"></a>  )</span>
<span id="cb6-283"><a href="source-code.html#cb6-283" tabindex="-1"></a>  </span>
<span id="cb6-284"><a href="source-code.html#cb6-284" tabindex="-1"></a>  df78 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-285"><a href="source-code.html#cb6-285" tabindex="-1"></a>    cells,</span>
<span id="cb6-286"><a href="source-code.html#cb6-286" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-287"><a href="source-code.html#cb6-287" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">7</span>,</span>
<span id="cb6-288"><a href="source-code.html#cb6-288" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">8</span>,</span>
<span id="cb6-289"><a href="source-code.html#cb6-289" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-290"><a href="source-code.html#cb6-290" tabindex="-1"></a>  )</span>
<span id="cb6-291"><a href="source-code.html#cb6-291" tabindex="-1"></a>  </span>
<span id="cb6-292"><a href="source-code.html#cb6-292" tabindex="-1"></a>  df910 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(</span>
<span id="cb6-293"><a href="source-code.html#cb6-293" tabindex="-1"></a>    cells,</span>
<span id="cb6-294"><a href="source-code.html#cb6-294" tabindex="-1"></a>    <span class="at">colorBy =</span> classes,</span>
<span id="cb6-295"><a href="source-code.html#cb6-295" tabindex="-1"></a>    <span class="at">PC1 =</span> <span class="dv">9</span>,</span>
<span id="cb6-296"><a href="source-code.html#cb6-296" tabindex="-1"></a>    <span class="at">PC2 =</span> <span class="dv">10</span>,</span>
<span id="cb6-297"><a href="source-code.html#cb6-297" tabindex="-1"></a>    <span class="at">plot =</span> F</span>
<span id="cb6-298"><a href="source-code.html#cb6-298" tabindex="-1"></a>  )</span>
<span id="cb6-299"><a href="source-code.html#cb6-299" tabindex="-1"></a>  </span>
<span id="cb6-300"><a href="source-code.html#cb6-300" tabindex="-1"></a>  </span>
<span id="cb6-301"><a href="source-code.html#cb6-301" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df12, df34, df56, df78, df910)</span>
<span id="cb6-302"><a href="source-code.html#cb6-302" tabindex="-1"></a>  </span>
<span id="cb6-303"><a href="source-code.html#cb6-303" tabindex="-1"></a>  </span>
<span id="cb6-304"><a href="source-code.html#cb6-304" tabindex="-1"></a>  <span class="cf">if</span> (normalize) {</span>
<span id="cb6-305"><a href="source-code.html#cb6-305" tabindex="-1"></a>    <span class="co"># extract variance explained for each pca</span></span>
<span id="cb6-306"><a href="source-code.html#cb6-306" tabindex="-1"></a>    pca_var <span class="ot">&lt;-</span></span>
<span id="cb6-307"><a href="source-code.html#cb6-307" tabindex="-1"></a>      <span class="fu">as.numeric</span>(<span class="fu">str_match</span>(<span class="fu">colnames</span>(df), <span class="st">&quot;</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">s*(.*?)</span><span class="sc">\\</span><span class="st">s*%&quot;</span>)[, <span class="dv">2</span>])</span>
<span id="cb6-308"><a href="source-code.html#cb6-308" tabindex="-1"></a>    </span>
<span id="cb6-309"><a href="source-code.html#cb6-309" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(pca_var)) {</span>
<span id="cb6-310"><a href="source-code.html#cb6-310" tabindex="-1"></a>      <span class="co"># calculate normalized pca values</span></span>
<span id="cb6-311"><a href="source-code.html#cb6-311" tabindex="-1"></a>      norm_pca <span class="ot">&lt;-</span> df[, i] <span class="sc">/</span> <span class="dv">100</span> <span class="sc">*</span> pca_var[i]</span>
<span id="cb6-312"><a href="source-code.html#cb6-312" tabindex="-1"></a>      <span class="co">#repopulate the dataframe</span></span>
<span id="cb6-313"><a href="source-code.html#cb6-313" tabindex="-1"></a>      df[, i] <span class="ot">&lt;-</span> norm_pca</span>
<span id="cb6-314"><a href="source-code.html#cb6-314" tabindex="-1"></a>    }</span>
<span id="cb6-315"><a href="source-code.html#cb6-315" tabindex="-1"></a>    </span>
<span id="cb6-316"><a href="source-code.html#cb6-316" tabindex="-1"></a>    </span>
<span id="cb6-317"><a href="source-code.html#cb6-317" tabindex="-1"></a>    </span>
<span id="cb6-318"><a href="source-code.html#cb6-318" tabindex="-1"></a>    </span>
<span id="cb6-319"><a href="source-code.html#cb6-319" tabindex="-1"></a>    </span>
<span id="cb6-320"><a href="source-code.html#cb6-320" tabindex="-1"></a>  }</span>
<span id="cb6-321"><a href="source-code.html#cb6-321" tabindex="-1"></a>  </span>
<span id="cb6-322"><a href="source-code.html#cb6-322" tabindex="-1"></a>  <span class="co">#check best k for k-means clustering method</span></span>
<span id="cb6-323"><a href="source-code.html#cb6-323" tabindex="-1"></a>  <span class="cf">if</span> (return_elbow) {</span>
<span id="cb6-324"><a href="source-code.html#cb6-324" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">fviz_nbclust</span>(df, kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-325"><a href="source-code.html#cb6-325" tabindex="-1"></a>             <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Silhouette method&quot;</span>))</span>
<span id="cb6-326"><a href="source-code.html#cb6-326" tabindex="-1"></a>    </span>
<span id="cb6-327"><a href="source-code.html#cb6-327" tabindex="-1"></a>  }</span>
<span id="cb6-328"><a href="source-code.html#cb6-328" tabindex="-1"></a>  </span>
<span id="cb6-329"><a href="source-code.html#cb6-329" tabindex="-1"></a>  <span class="co"># #</span></span>
<span id="cb6-330"><a href="source-code.html#cb6-330" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, method = &quot;silhouette&quot;)+</span></span>
<span id="cb6-331"><a href="source-code.html#cb6-331" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Silhouette method&quot;)</span></span>
<span id="cb6-332"><a href="source-code.html#cb6-332" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb6-333"><a href="source-code.html#cb6-333" tabindex="-1"></a>  <span class="co"># fviz_nbclust(df, kmeans, nstart = 25,  method = &quot;gap_stat&quot;, nboot = 50)+</span></span>
<span id="cb6-334"><a href="source-code.html#cb6-334" tabindex="-1"></a>  <span class="co">#   labs(subtitle = &quot;Gap statistic method&quot;)</span></span>
<span id="cb6-335"><a href="source-code.html#cb6-335" tabindex="-1"></a>  </span>
<span id="cb6-336"><a href="source-code.html#cb6-336" tabindex="-1"></a>  <span class="co"># compute k-means clustering</span></span>
<span id="cb6-337"><a href="source-code.html#cb6-337" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb6-338"><a href="source-code.html#cb6-338" tabindex="-1"></a>  res.km <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, cluster_n, <span class="at">nstart =</span> <span class="dv">1000</span>)</span>
<span id="cb6-339"><a href="source-code.html#cb6-339" tabindex="-1"></a>  </span>
<span id="cb6-340"><a href="source-code.html#cb6-340" tabindex="-1"></a>  <span class="co"># change colnames so it works for our plot</span></span>
<span id="cb6-341"><a href="source-code.html#cb6-341" tabindex="-1"></a>  df12<span class="sc">$</span>class <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb6-342"><a href="source-code.html#cb6-342" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-343"><a href="source-code.html#cb6-343" tabindex="-1"></a>  num <span class="ot">&lt;-</span> <span class="fu">str_remove_all</span>(num, <span class="at">pattern =</span> <span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)</span>
<span id="cb6-344"><a href="source-code.html#cb6-344" tabindex="-1"></a>  df12<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste0</span>(df12<span class="sc">$</span>class, num)</span>
<span id="cb6-345"><a href="source-code.html#cb6-345" tabindex="-1"></a>  pc1 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">1</span>]</span>
<span id="cb6-346"><a href="source-code.html#cb6-346" tabindex="-1"></a>  pc2 <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df12)[<span class="dv">2</span>]</span>
<span id="cb6-347"><a href="source-code.html#cb6-347" tabindex="-1"></a>  </span>
<span id="cb6-348"><a href="source-code.html#cb6-348" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>, <span class="st">&#39;class&#39;</span>, <span class="st">&#39;label&#39;</span>)</span>
<span id="cb6-349"><a href="source-code.html#cb6-349" tabindex="-1"></a>  </span>
<span id="cb6-350"><a href="source-code.html#cb6-350" tabindex="-1"></a>  <span class="co"># plot nicely</span></span>
<span id="cb6-351"><a href="source-code.html#cb6-351" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> df12, <span class="fu">aes</span>(<span class="at">label =</span> label)) <span class="sc">+</span></span>
<span id="cb6-352"><a href="source-code.html#cb6-352" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">1.0</span>,</span>
<span id="cb6-353"><a href="source-code.html#cb6-353" tabindex="-1"></a>               <span class="fu">aes</span>(</span>
<span id="cb6-354"><a href="source-code.html#cb6-354" tabindex="-1"></a>                 <span class="at">x =</span> PC1,</span>
<span id="cb6-355"><a href="source-code.html#cb6-355" tabindex="-1"></a>                 <span class="at">y =</span> PC2,</span>
<span id="cb6-356"><a href="source-code.html#cb6-356" tabindex="-1"></a>                 <span class="at">col =</span> <span class="fu">as.factor</span>(res.km<span class="sc">$</span>cluster)</span>
<span id="cb6-357"><a href="source-code.html#cb6-357" tabindex="-1"></a>               )) <span class="sc">+</span></span>
<span id="cb6-358"><a href="source-code.html#cb6-358" tabindex="-1"></a>    </span>
<span id="cb6-359"><a href="source-code.html#cb6-359" tabindex="-1"></a>    </span>
<span id="cb6-360"><a href="source-code.html#cb6-360" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb6-361"><a href="source-code.html#cb6-361" tabindex="-1"></a>    </span>
<span id="cb6-362"><a href="source-code.html#cb6-362" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> pc1, <span class="at">y =</span> pc2) <span class="sc">+</span></span>
<span id="cb6-363"><a href="source-code.html#cb6-363" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">labels =</span> labels,</span>
<span id="cb6-364"><a href="source-code.html#cb6-364" tabindex="-1"></a>                       <span class="at">values =</span> cols,</span>
<span id="cb6-365"><a href="source-code.html#cb6-365" tabindex="-1"></a>                       <span class="at">name =</span> <span class="st">&quot;Clone&quot;</span>) <span class="sc">+</span></span>
<span id="cb6-366"><a href="source-code.html#cb6-366" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> legend_position,</span>
<span id="cb6-367"><a href="source-code.html#cb6-367" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">17</span>),</span>
<span id="cb6-368"><a href="source-code.html#cb6-368" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) </span>
<span id="cb6-369"><a href="source-code.html#cb6-369" tabindex="-1"></a>  </span>
<span id="cb6-370"><a href="source-code.html#cb6-370" tabindex="-1"></a>  p</span>
<span id="cb6-371"><a href="source-code.html#cb6-371" tabindex="-1"></a>  </span>
<span id="cb6-372"><a href="source-code.html#cb6-372" tabindex="-1"></a>}</span>
<span id="cb6-373"><a href="source-code.html#cb6-373" tabindex="-1"></a></span>
<span id="cb6-374"><a href="source-code.html#cb6-374" tabindex="-1"></a>  </span>
<span id="cb6-375"><a href="source-code.html#cb6-375" tabindex="-1"></a><span class="er">}</span></span></code></pre></div>
</div>
<div id="divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> divide_subclone_functions script: ‘extract_other_cells’, ‘extract_square’, ‘l_g’ and ‘subclone_classes_for_genomeheatmap’ functions<a href="source-code.html#divide_subclone_functions-script-extract_other_cells-extract_square-l_g-and-subclone_classes_for_genomeheatmap-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="source-code.html#cb7-1" tabindex="-1"></a>extract_other_cells <span class="ot">&lt;-</span> <span class="cf">function</span>(kra_name, list_of_subclones) {</span>
<span id="cb7-2"><a href="source-code.html#cb7-2" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(man_select_files_edivisive) <span class="sc">==</span> kra_name)</span>
<span id="cb7-3"><a href="source-code.html#cb7-3" tabindex="-1"></a>  all <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(man_select_files_edivisive[[i]],  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>) </span>
<span id="cb7-4"><a href="source-code.html#cb7-4" tabindex="-1"></a>  rest <span class="ot">&lt;-</span> all[<span class="fu">which</span>(<span class="sc">!</span>all <span class="sc">%in%</span> list_of_subclones)]</span>
<span id="cb7-5"><a href="source-code.html#cb7-5" tabindex="-1"></a>  <span class="fu">return</span>(rest)</span>
<span id="cb7-6"><a href="source-code.html#cb7-6" tabindex="-1"></a>}</span>
<span id="cb7-7"><a href="source-code.html#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="source-code.html#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="source-code.html#cb7-9" tabindex="-1"></a>extract_square <span class="ot">&lt;-</span> <span class="cf">function</span>(list_files1, list_files2, square, x, y) {</span>
<span id="cb7-10"><a href="source-code.html#cb7-10" tabindex="-1"></a>  <span class="co"># extracts all cells that lie within a square of a pca plot defined by </span></span>
<span id="cb7-11"><a href="source-code.html#cb7-11" tabindex="-1"></a>  <span class="co"># user. For example square c(0.1, 0.0) will extract all cells with</span></span>
<span id="cb7-12"><a href="source-code.html#cb7-12" tabindex="-1"></a>  <span class="co"># pc1 lower or higher that 0.1 and pc2 lower or higher than 0.0. User has to rearange functon</span></span>
<span id="cb7-13"><a href="source-code.html#cb7-13" tabindex="-1"></a>  <span class="co"># if he wants x lower or higher than 0.1</span></span>
<span id="cb7-14"><a href="source-code.html#cb7-14" tabindex="-1"></a>  cells <span class="ot">&lt;-</span> <span class="fu">c</span>(list_files1, list_files2)</span>
<span id="cb7-15"><a href="source-code.html#cb7-15" tabindex="-1"></a>  </span>
<span id="cb7-16"><a href="source-code.html#cb7-16" tabindex="-1"></a>  df12 <span class="ot">&lt;-</span> <span class="fu">plot_pca</span>(cells,</span>
<span id="cb7-17"><a href="source-code.html#cb7-17" tabindex="-1"></a>                   <span class="at">colorBy =</span> classes, </span>
<span id="cb7-18"><a href="source-code.html#cb7-18" tabindex="-1"></a>                   <span class="at">PC1=</span><span class="dv">1</span>, </span>
<span id="cb7-19"><a href="source-code.html#cb7-19" tabindex="-1"></a>                   <span class="at">PC2=</span><span class="dv">2</span>,</span>
<span id="cb7-20"><a href="source-code.html#cb7-20" tabindex="-1"></a>                   <span class="at">plot =</span> F)</span>
<span id="cb7-21"><a href="source-code.html#cb7-21" tabindex="-1"></a>  </span>
<span id="cb7-22"><a href="source-code.html#cb7-22" tabindex="-1"></a>  <span class="fu">colnames</span>(df12) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;PC1&#39;</span>, <span class="st">&#39;PC2&#39;</span>)</span>
<span id="cb7-23"><a href="source-code.html#cb7-23" tabindex="-1"></a>  </span>
<span id="cb7-24"><a href="source-code.html#cb7-24" tabindex="-1"></a>  </span>
<span id="cb7-25"><a href="source-code.html#cb7-25" tabindex="-1"></a>  <span class="co"># extracting dataframe with correct square</span></span>
<span id="cb7-26"><a href="source-code.html#cb7-26" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb7-27"><a href="source-code.html#cb7-27" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb7-28"><a href="source-code.html#cb7-28" tabindex="-1"></a>  }</span>
<span id="cb7-29"><a href="source-code.html#cb7-29" tabindex="-1"></a>  </span>
<span id="cb7-30"><a href="source-code.html#cb7-30" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb7-31"><a href="source-code.html#cb7-31" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&lt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb7-32"><a href="source-code.html#cb7-32" tabindex="-1"></a>  }</span>
<span id="cb7-33"><a href="source-code.html#cb7-33" tabindex="-1"></a>  </span>
<span id="cb7-34"><a href="source-code.html#cb7-34" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&lt;&#39;</span>) {</span>
<span id="cb7-35"><a href="source-code.html#cb7-35" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&lt;</span> square[<span class="dv">2</span>])</span>
<span id="cb7-36"><a href="source-code.html#cb7-36" tabindex="-1"></a>  }</span>
<span id="cb7-37"><a href="source-code.html#cb7-37" tabindex="-1"></a>  </span>
<span id="cb7-38"><a href="source-code.html#cb7-38" tabindex="-1"></a>  <span class="cf">if</span>(x <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span> <span class="sc">&amp;</span> y <span class="sc">==</span> <span class="st">&#39;&gt;&#39;</span>) {</span>
<span id="cb7-39"><a href="source-code.html#cb7-39" tabindex="-1"></a>    df12 <span class="ot">&lt;-</span> <span class="fu">subset</span>(df12, PC1 <span class="sc">&gt;</span> square[<span class="dv">1</span>] <span class="sc">&amp;</span> PC2 <span class="sc">&gt;</span> square[<span class="dv">2</span>])</span>
<span id="cb7-40"><a href="source-code.html#cb7-40" tabindex="-1"></a>  }</span>
<span id="cb7-41"><a href="source-code.html#cb7-41" tabindex="-1"></a>  </span>
<span id="cb7-42"><a href="source-code.html#cb7-42" tabindex="-1"></a>  <span class="co"># numbers is: </span></span>
<span id="cb7-43"><a href="source-code.html#cb7-43" tabindex="-1"></a>  cell_id <span class="ot">&lt;-</span>  <span class="fu">str_extract</span>(<span class="fu">rownames</span>(df12),  <span class="st">&#39;_</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">.&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span> <span class="fu">str_remove</span>(<span class="st">&#39;</span><span class="sc">\\</span><span class="st">.&#39;</span>)  </span>
<span id="cb7-44"><a href="source-code.html#cb7-44" tabindex="-1"></a>  </span>
<span id="cb7-45"><a href="source-code.html#cb7-45" tabindex="-1"></a>  <span class="fu">return</span>(cell_id)</span>
<span id="cb7-46"><a href="source-code.html#cb7-46" tabindex="-1"></a>  </span>
<span id="cb7-47"><a href="source-code.html#cb7-47" tabindex="-1"></a>}</span>
<span id="cb7-48"><a href="source-code.html#cb7-48" tabindex="-1"></a></span>
<span id="cb7-49"><a href="source-code.html#cb7-49" tabindex="-1"></a>l_g <span class="ot">&lt;-</span> <span class="cf">function</span>(x, kra_num, list_cells) {</span>
<span id="cb7-50"><a href="source-code.html#cb7-50" tabindex="-1"></a>  select <span class="ot">&lt;-</span> <span class="fu">grepl</span>(<span class="fu">paste0</span>(kra_num,<span class="st">&#39;_&#39;</span>,x,<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>), list_cells)</span>
<span id="cb7-51"><a href="source-code.html#cb7-51" tabindex="-1"></a>  i <span class="ot">&lt;-</span> <span class="fu">which</span>(select)</span>
<span id="cb7-52"><a href="source-code.html#cb7-52" tabindex="-1"></a>  <span class="fu">return</span>(list_cells[i])</span>
<span id="cb7-53"><a href="source-code.html#cb7-53" tabindex="-1"></a>  </span>
<span id="cb7-54"><a href="source-code.html#cb7-54" tabindex="-1"></a>  </span>
<span id="cb7-55"><a href="source-code.html#cb7-55" tabindex="-1"></a>}</span>
<span id="cb7-56"><a href="source-code.html#cb7-56" tabindex="-1"></a></span>
<span id="cb7-57"><a href="source-code.html#cb7-57" tabindex="-1"></a></span>
<span id="cb7-58"><a href="source-code.html#cb7-58" tabindex="-1"></a>subclone_classes_for_genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(list_of_subclones, cline) {</span>
<span id="cb7-59"><a href="source-code.html#cb7-59" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb7-60"><a href="source-code.html#cb7-60" tabindex="-1"></a>  <span class="co">#&#39; list of subclones is the full list of rda files that have been assigned a subclone_classes_for_genomeheatmap</span></span>
<span id="cb7-61"><a href="source-code.html#cb7-61" tabindex="-1"></a>  <span class="co">#&#39; cline is the line (hub005, hub183 etc.)</span></span>
<span id="cb7-62"><a href="source-code.html#cb7-62" tabindex="-1"></a>  <span class="co">#&#39; rad is either prerad or rad</span></span>
<span id="cb7-63"><a href="source-code.html#cb7-63" tabindex="-1"></a>  <span class="co">#&#39; @return </span></span>
<span id="cb7-64"><a href="source-code.html#cb7-64" tabindex="-1"></a>  <span class="co">#&#39; returns a vector with that assigns a subclone</span></span>
<span id="cb7-65"><a href="source-code.html#cb7-65" tabindex="-1"></a>  </span>
<span id="cb7-66"><a href="source-code.html#cb7-66" tabindex="-1"></a>  extract_subclones <span class="ot">&lt;-</span> list_of_subclones[<span class="fu">grepl</span>(<span class="fu">names</span>(list_of_subclones), <span class="at">pattern =</span> <span class="fu">paste0</span>(cline))]</span>
<span id="cb7-67"><a href="source-code.html#cb7-67" tabindex="-1"></a>  </span>
<span id="cb7-68"><a href="source-code.html#cb7-68" tabindex="-1"></a>  size_subclones <span class="ot">&lt;-</span> <span class="fu">lapply</span>(extract_subclones, length)</span>
<span id="cb7-69"><a href="source-code.html#cb7-69" tabindex="-1"></a>  </span>
<span id="cb7-70"><a href="source-code.html#cb7-70" tabindex="-1"></a>  classes <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">names</span>(size_subclones), size_subclones)</span>
<span id="cb7-71"><a href="source-code.html#cb7-71" tabindex="-1"></a>  </span>
<span id="cb7-72"><a href="source-code.html#cb7-72" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">path =</span> <span class="fu">unlist</span>(extract_subclones),</span>
<span id="cb7-73"><a href="source-code.html#cb7-73" tabindex="-1"></a>                   <span class="at">class =</span> classes)</span>
<span id="cb7-74"><a href="source-code.html#cb7-74" tabindex="-1"></a>  </span>
<span id="cb7-75"><a href="source-code.html#cb7-75" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb7-76"><a href="source-code.html#cb7-76" tabindex="-1"></a>  </span>
<span id="cb7-77"><a href="source-code.html#cb7-77" tabindex="-1"></a>  </span>
<span id="cb7-78"><a href="source-code.html#cb7-78" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genomeheatmap-script-genomeheatmap-function" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> plot_genomeheatmap script: ‘genomeheatmap’ function<a href="source-code.html#plot_genomeheatmap-script-genomeheatmap-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="source-code.html#cb8-1" tabindex="-1"></a>genomeheatmap <span class="ot">&lt;-</span> <span class="cf">function</span>(selected.files, </span>
<span id="cb8-2"><a href="source-code.html#cb8-2" tabindex="-1"></a>                          path, </span>
<span id="cb8-3"><a href="source-code.html#cb8-3" tabindex="-1"></a>                          <span class="at">classes_daan =</span> <span class="cn">NULL</span>, </span>
<span id="cb8-4"><a href="source-code.html#cb8-4" tabindex="-1"></a>                          <span class="at">class.col =</span> <span class="cn">NULL</span>, </span>
<span id="cb8-5"><a href="source-code.html#cb8-5" tabindex="-1"></a>                          <span class="at">dendogram =</span> F,</span>
<span id="cb8-6"><a href="source-code.html#cb8-6" tabindex="-1"></a>                          <span class="at">daan_colours =</span> T,</span>
<span id="cb8-7"><a href="source-code.html#cb8-7" tabindex="-1"></a>                          name) {</span>
<span id="cb8-8"><a href="source-code.html#cb8-8" tabindex="-1"></a>  <span class="co"># This code was adapted from the AneuFinder package</span></span>
<span id="cb8-9"><a href="source-code.html#cb8-9" tabindex="-1"></a>  <span class="co"># plots and saves genome heatmap of selected.files</span></span>
<span id="cb8-10"><a href="source-code.html#cb8-10" tabindex="-1"></a>  <span class="co"># @param </span></span>
<span id="cb8-11"><a href="source-code.html#cb8-11" tabindex="-1"></a>  <span class="co"># selected.files: vector of location of selected files.</span></span>
<span id="cb8-12"><a href="source-code.html#cb8-12" tabindex="-1"></a>  <span class="co"># output path: path were plots should be saved. </span></span>
<span id="cb8-13"><a href="source-code.html#cb8-13" tabindex="-1"></a>  <span class="co"># name: how you want to name the plot</span></span>
<span id="cb8-14"><a href="source-code.html#cb8-14" tabindex="-1"></a>  </span>
<span id="cb8-15"><a href="source-code.html#cb8-15" tabindex="-1"></a>  <span class="co"># get platename</span></span>
<span id="cb8-16"><a href="source-code.html#cb8-16" tabindex="-1"></a>  platename <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(selected.files[<span class="dv">1</span>],  <span class="st">&#39;KRA-0</span><span class="sc">\\</span><span class="st">d+&#39;</span>)</span>
<span id="cb8-17"><a href="source-code.html#cb8-17" tabindex="-1"></a>  </span>
<span id="cb8-18"><a href="source-code.html#cb8-18" tabindex="-1"></a>  <span class="co"># AneuFinder functions: </span></span>
<span id="cb8-19"><a href="source-code.html#cb8-19" tabindex="-1"></a>  startTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(...) {</span>
<span id="cb8-20"><a href="source-code.html#cb8-20" tabindex="-1"></a>    </span>
<span id="cb8-21"><a href="source-code.html#cb8-21" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">paste0</span>(..., <span class="at">collapse=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb8-22"><a href="source-code.html#cb8-22" tabindex="-1"></a>    <span class="fu">message</span>(x, <span class="at">appendLF=</span><span class="cn">FALSE</span>)</span>
<span id="cb8-23"><a href="source-code.html#cb8-23" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">proc.time</span>()</span>
<span id="cb8-24"><a href="source-code.html#cb8-24" tabindex="-1"></a>    <span class="fu">return</span>(ptm)</span>
<span id="cb8-25"><a href="source-code.html#cb8-25" tabindex="-1"></a>    </span>
<span id="cb8-26"><a href="source-code.html#cb8-26" tabindex="-1"></a>  }</span>
<span id="cb8-27"><a href="source-code.html#cb8-27" tabindex="-1"></a>  </span>
<span id="cb8-28"><a href="source-code.html#cb8-28" tabindex="-1"></a>  transCoord <span class="ot">&lt;-</span> <span class="cf">function</span> (gr) </span>
<span id="cb8-29"><a href="source-code.html#cb8-29" tabindex="-1"></a>  {</span>
<span id="cb8-30"><a href="source-code.html#cb8-30" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(gr)))</span>
<span id="cb8-31"><a href="source-code.html#cb8-31" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb8-32"><a href="source-code.html#cb8-32" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(gr)</span>
<span id="cb8-33"><a href="source-code.html#cb8-33" tabindex="-1"></a>    gr<span class="sc">$</span>start.genome <span class="ot">&lt;-</span> <span class="fu">start</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb8-34"><a href="source-code.html#cb8-34" tabindex="-1"></a>    gr<span class="sc">$</span>end.genome <span class="ot">&lt;-</span> <span class="fu">end</span>(gr) <span class="sc">+</span> cum.seqlengths<span class="fl">.0</span>[<span class="fu">as.character</span>(<span class="fu">seqnames</span>(gr))]</span>
<span id="cb8-35"><a href="source-code.html#cb8-35" tabindex="-1"></a>    <span class="fu">return</span>(gr)</span>
<span id="cb8-36"><a href="source-code.html#cb8-36" tabindex="-1"></a>  }</span>
<span id="cb8-37"><a href="source-code.html#cb8-37" tabindex="-1"></a>  </span>
<span id="cb8-38"><a href="source-code.html#cb8-38" tabindex="-1"></a>  stopTimedMessage <span class="ot">&lt;-</span> <span class="cf">function</span>(ptm) {</span>
<span id="cb8-39"><a href="source-code.html#cb8-39" tabindex="-1"></a>    </span>
<span id="cb8-40"><a href="source-code.html#cb8-40" tabindex="-1"></a>    time <span class="ot">&lt;-</span> <span class="fu">proc.time</span>() <span class="sc">-</span> ptm</span>
<span id="cb8-41"><a href="source-code.html#cb8-41" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot; &quot;</span>, <span class="fu">round</span>(time[<span class="dv">3</span>],<span class="dv">2</span>), <span class="st">&quot;s&quot;</span>)</span>
<span id="cb8-42"><a href="source-code.html#cb8-42" tabindex="-1"></a>    </span>
<span id="cb8-43"><a href="source-code.html#cb8-43" tabindex="-1"></a>  }</span>
<span id="cb8-44"><a href="source-code.html#cb8-44" tabindex="-1"></a>  </span>
<span id="cb8-45"><a href="source-code.html#cb8-45" tabindex="-1"></a>  initializeStates <span class="ot">&lt;-</span> <span class="cf">function</span> (states) </span>
<span id="cb8-46"><a href="source-code.html#cb8-46" tabindex="-1"></a>  {</span>
<span id="cb8-47"><a href="source-code.html#cb8-47" tabindex="-1"></a>    somy.states <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;somy&quot;</span>, states, <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-48"><a href="source-code.html#cb8-48" tabindex="-1"></a>    somy.numbers <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(<span class="fu">sapply</span>(<span class="fu">strsplit</span>(somy.states, <span class="st">&quot;-somy&quot;</span>), </span>
<span id="cb8-49"><a href="source-code.html#cb8-49" tabindex="-1"></a>                                      <span class="st">&quot;[[&quot;</span>, <span class="dv">1</span>))</span>
<span id="cb8-50"><a href="source-code.html#cb8-50" tabindex="-1"></a>    <span class="fu">names</span>(somy.numbers) <span class="ot">&lt;-</span> somy.states</span>
<span id="cb8-51"><a href="source-code.html#cb8-51" tabindex="-1"></a>    <span class="cf">if</span> (<span class="st">&quot;zero-inflation&quot;</span> <span class="sc">%in%</span> states) {</span>
<span id="cb8-52"><a href="source-code.html#cb8-52" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">0</span>, somy.numbers)</span>
<span id="cb8-53"><a href="source-code.html#cb8-53" tabindex="-1"></a>    }</span>
<span id="cb8-54"><a href="source-code.html#cb8-54" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb8-55"><a href="source-code.html#cb8-55" tabindex="-1"></a>      multiplicity <span class="ot">&lt;-</span> somy.numbers</span>
<span id="cb8-56"><a href="source-code.html#cb8-56" tabindex="-1"></a>    }</span>
<span id="cb8-57"><a href="source-code.html#cb8-57" tabindex="-1"></a>    levels.distributions <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;delta&quot;</span>, <span class="st">&quot;dgeom&quot;</span>, <span class="st">&quot;dnbinom&quot;</span>, <span class="st">&quot;dbinom&quot;</span>)</span>
<span id="cb8-58"><a href="source-code.html#cb8-58" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(states))</span>
<span id="cb8-59"><a href="source-code.html#cb8-59" tabindex="-1"></a>    <span class="fu">names</span>(distributions) <span class="ot">&lt;-</span> states</span>
<span id="cb8-60"><a href="source-code.html#cb8-60" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;zero-inflation&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;delta&quot;</span></span>
<span id="cb8-61"><a href="source-code.html#cb8-61" tabindex="-1"></a>    distributions[states <span class="sc">==</span> <span class="st">&quot;0-somy&quot;</span>] <span class="ot">&lt;-</span> <span class="st">&quot;dgeom&quot;</span></span>
<span id="cb8-62"><a href="source-code.html#cb8-62" tabindex="-1"></a>    distributions[(states <span class="sc">!=</span> <span class="st">&quot;zero-inflation&quot;</span>) <span class="sc">&amp;</span> (states <span class="sc">!=</span> <span class="st">&quot;0-somy&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;dnbinom&quot;</span></span>
<span id="cb8-63"><a href="source-code.html#cb8-63" tabindex="-1"></a>    states <span class="ot">&lt;-</span> <span class="fu">factor</span>(states, <span class="at">levels =</span> states)</span>
<span id="cb8-64"><a href="source-code.html#cb8-64" tabindex="-1"></a>    distributions <span class="ot">&lt;-</span> <span class="fu">factor</span>(distributions, <span class="at">levels =</span> levels.distributions)</span>
<span id="cb8-65"><a href="source-code.html#cb8-65" tabindex="-1"></a>    l <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">states =</span> states, <span class="at">distributions =</span> distributions, </span>
<span id="cb8-66"><a href="source-code.html#cb8-66" tabindex="-1"></a>              <span class="at">multiplicity =</span> multiplicity)</span>
<span id="cb8-67"><a href="source-code.html#cb8-67" tabindex="-1"></a>    <span class="fu">return</span>(l)</span>
<span id="cb8-68"><a href="source-code.html#cb8-68" tabindex="-1"></a>  }</span>
<span id="cb8-69"><a href="source-code.html#cb8-69" tabindex="-1"></a>  </span>
<span id="cb8-70"><a href="source-code.html#cb8-70" tabindex="-1"></a>  <span class="co">#  colours</span></span>
<span id="cb8-71"><a href="source-code.html#cb8-71" tabindex="-1"></a>  <span class="cf">if</span> (daan_colours) {</span>
<span id="cb8-72"><a href="source-code.html#cb8-72" tabindex="-1"></a>    stateColors <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">states =</span> <span class="fu">c</span>(<span class="st">&quot;zero-inflation&quot;</span>, <span class="fu">paste0</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>, <span class="st">&quot;-somy&quot;</span>),</span>
<span id="cb8-73"><a href="source-code.html#cb8-73" tabindex="-1"></a>                                       <span class="st">&quot;total&quot;</span>)) {</span>
<span id="cb8-74"><a href="source-code.html#cb8-74" tabindex="-1"></a>      </span>
<span id="cb8-75"><a href="source-code.html#cb8-75" tabindex="-1"></a>      state.colors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">zero-inflation</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>, <span class="st">`</span><span class="at">0-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#1d4661&quot;</span>,</span>
<span id="cb8-76"><a href="source-code.html#cb8-76" tabindex="-1"></a>                        <span class="st">`</span><span class="at">1-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3787BA&quot;</span>, <span class="st">`</span><span class="at">2-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#95B8C5&quot;</span>,</span>
<span id="cb8-77"><a href="source-code.html#cb8-77" tabindex="-1"></a>                        <span class="st">`</span><span class="at">3-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#F0ECEB&quot;</span>, <span class="st">`</span><span class="at">4-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#D7A290&quot;</span>, <span class="st">`</span><span class="at">5-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#BF583B&quot;</span>,</span>
<span id="cb8-78"><a href="source-code.html#cb8-78" tabindex="-1"></a>                        <span class="st">`</span><span class="at">6-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#8D1128&quot;</span>, <span class="st">`</span><span class="at">7-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;#3C0912&quot;</span>, <span class="st">`</span><span class="at">8-somy</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb8-79"><a href="source-code.html#cb8-79" tabindex="-1"></a>                        <span class="at">total =</span> <span class="st">&quot;black&quot;</span>)</span>
<span id="cb8-80"><a href="source-code.html#cb8-80" tabindex="-1"></a>      </span>
<span id="cb8-81"><a href="source-code.html#cb8-81" tabindex="-1"></a>      states.with.color <span class="ot">&lt;-</span> <span class="fu">intersect</span>(states, <span class="fu">names</span>(state.colors))</span>
<span id="cb8-82"><a href="source-code.html#cb8-82" tabindex="-1"></a>      cols <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&quot;black&quot;</span>, <span class="fu">length</span>(states))</span>
<span id="cb8-83"><a href="source-code.html#cb8-83" tabindex="-1"></a>      <span class="fu">names</span>(cols) <span class="ot">&lt;-</span> states</span>
<span id="cb8-84"><a href="source-code.html#cb8-84" tabindex="-1"></a>      cols[states.with.color] <span class="ot">&lt;-</span> state.colors[states.with.color]</span>
<span id="cb8-85"><a href="source-code.html#cb8-85" tabindex="-1"></a>      <span class="fu">return</span>(cols)</span>
<span id="cb8-86"><a href="source-code.html#cb8-86" tabindex="-1"></a>    }</span>
<span id="cb8-87"><a href="source-code.html#cb8-87" tabindex="-1"></a>    </span>
<span id="cb8-88"><a href="source-code.html#cb8-88" tabindex="-1"></a>  }</span>
<span id="cb8-89"><a href="source-code.html#cb8-89" tabindex="-1"></a></span>
<span id="cb8-90"><a href="source-code.html#cb8-90" tabindex="-1"></a>  </span>
<span id="cb8-91"><a href="source-code.html#cb8-91" tabindex="-1"></a>  </span>
<span id="cb8-92"><a href="source-code.html#cb8-92" tabindex="-1"></a>  <span class="co"># heatmapgenomewide adapted form AneuFinder package</span></span>
<span id="cb8-93"><a href="source-code.html#cb8-93" tabindex="-1"></a>  heatmapGenomewide_daan <span class="ot">&lt;-</span> <span class="cf">function</span> (hmms, <span class="at">ylabels =</span> <span class="cn">NULL</span>, classes, <span class="at">reorder.by.class =</span> <span class="cn">TRUE</span>, </span>
<span id="cb8-94"><a href="source-code.html#cb8-94" tabindex="-1"></a>                                      classes.color, <span class="at">file =</span> <span class="cn">NULL</span>, <span class="at">cluster =</span> <span class="cn">TRUE</span>, <span class="at">plot.breakpoints =</span> <span class="cn">FALSE</span>, </span>
<span id="cb8-95"><a href="source-code.html#cb8-95" tabindex="-1"></a>                                      <span class="at">hotspots =</span> <span class="cn">NULL</span>, <span class="at">exclude.regions =</span> <span class="cn">NULL</span>) </span>
<span id="cb8-96"><a href="source-code.html#cb8-96" tabindex="-1"></a>  {</span>
<span id="cb8-97"><a href="source-code.html#cb8-97" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb8-98"><a href="source-code.html#cb8-98" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(ylabels) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb8-99"><a href="source-code.html#cb8-99" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(ylabels) must equal length(hmms)&quot;</span>)</span>
<span id="cb8-100"><a href="source-code.html#cb8-100" tabindex="-1"></a>      }</span>
<span id="cb8-101"><a href="source-code.html#cb8-101" tabindex="-1"></a>    }</span>
<span id="cb8-102"><a href="source-code.html#cb8-102" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb8-103"><a href="source-code.html#cb8-103" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(classes) <span class="sc">!=</span> <span class="fu">length</span>(hmms)) {</span>
<span id="cb8-104"><a href="source-code.html#cb8-104" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">&quot;length(classes) must equal length(hmms)&quot;</span>)</span>
<span id="cb8-105"><a href="source-code.html#cb8-105" tabindex="-1"></a>      }</span>
<span id="cb8-106"><a href="source-code.html#cb8-106" tabindex="-1"></a>    }</span>
<span id="cb8-107"><a href="source-code.html#cb8-107" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(classes.color) <span class="sc">!=</span> <span class="fu">length</span>(<span class="fu">unique</span>(classes))) {</span>
<span id="cb8-108"><a href="source-code.html#cb8-108" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;&#39;classes.color&#39; must have the same length as unique(classes)&quot;</span>)</span>
<span id="cb8-109"><a href="source-code.html#cb8-109" tabindex="-1"></a>    }</span>
<span id="cb8-110"><a href="source-code.html#cb8-110" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">names</span>(classes.color))) {</span>
<span id="cb8-111"><a href="source-code.html#cb8-111" tabindex="-1"></a>      <span class="fu">names</span>(classes.color) <span class="ot">&lt;-</span> <span class="fu">unique</span>(classes)</span>
<span id="cb8-112"><a href="source-code.html#cb8-112" tabindex="-1"></a>    }</span>
<span id="cb8-113"><a href="source-code.html#cb8-113" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">setequal</span>(<span class="fu">names</span>(classes.color), <span class="fu">unique</span>(classes))) {</span>
<span id="cb8-114"><a href="source-code.html#cb8-114" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">&quot;The names of &#39;classes.color&#39; must be equal to the unique elements in &#39;classes&#39;&quot;</span>)</span>
<span id="cb8-115"><a href="source-code.html#cb8-115" tabindex="-1"></a>    }</span>
<span id="cb8-116"><a href="source-code.html#cb8-116" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(hmms) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> cluster <span class="sc">==</span> <span class="cn">TRUE</span>) {</span>
<span id="cb8-117"><a href="source-code.html#cb8-117" tabindex="-1"></a>      cluster <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb8-118"><a href="source-code.html#cb8-118" tabindex="-1"></a>      <span class="fu">warning</span>(<span class="st">&quot;Cannot do clustering because only one object was given.&quot;</span>)</span>
<span id="cb8-119"><a href="source-code.html#cb8-119" tabindex="-1"></a>    }</span>
<span id="cb8-120"><a href="source-code.html#cb8-120" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> <span class="fu">loadFromFiles</span>(hmms, <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb8-121"><a href="source-code.html#cb8-121" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID =</span> <span class="fu">sapply</span>(hmms, <span class="st">&quot;[[&quot;</span>, <span class="st">&quot;ID&quot;</span>))</span>
<span id="cb8-122"><a href="source-code.html#cb8-122" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb8-123"><a href="source-code.html#cb8-123" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(ylabels)) {</span>
<span id="cb8-124"><a href="source-code.html#cb8-124" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb8-125"><a href="source-code.html#cb8-125" tabindex="-1"></a>    }</span>
<span id="cb8-126"><a href="source-code.html#cb8-126" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb8-127"><a href="source-code.html#cb8-127" tabindex="-1"></a>      class.data<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ylabels)</span>
<span id="cb8-128"><a href="source-code.html#cb8-128" tabindex="-1"></a>    }</span>
<span id="cb8-129"><a href="source-code.html#cb8-129" tabindex="-1"></a>    class.data<span class="sc">$</span>class <span class="ot">&lt;-</span> classes</span>
<span id="cb8-130"><a href="source-code.html#cb8-130" tabindex="-1"></a>    mapping <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ylabel</span>
<span id="cb8-131"><a href="source-code.html#cb8-131" tabindex="-1"></a>    <span class="fu">names</span>(mapping) <span class="ot">&lt;-</span> class.data<span class="sc">$</span>ID</span>
<span id="cb8-132"><a href="source-code.html#cb8-132" tabindex="-1"></a>    <span class="cf">if</span> (reorder.by.class) {</span>
<span id="cb8-133"><a href="source-code.html#cb8-133" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">classes =</span> classes, </span>
<span id="cb8-134"><a href="source-code.html#cb8-134" tabindex="-1"></a>                        <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb8-135"><a href="source-code.html#cb8-135" tabindex="-1"></a>    }</span>
<span id="cb8-136"><a href="source-code.html#cb8-136" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb8-137"><a href="source-code.html#cb8-137" tabindex="-1"></a>      cl <span class="ot">&lt;-</span> <span class="fu">clusterHMMs</span>(hmms, <span class="at">cluster =</span> cluster, <span class="at">exclude.regions =</span> exclude.regions)</span>
<span id="cb8-138"><a href="source-code.html#cb8-138" tabindex="-1"></a>    }</span>
<span id="cb8-139"><a href="source-code.html#cb8-139" tabindex="-1"></a>    hmms <span class="ot">&lt;-</span> hmms[cl<span class="sc">$</span>IDorder]</span>
<span id="cb8-140"><a href="source-code.html#cb8-140" tabindex="-1"></a>    class.data <span class="ot">&lt;-</span> class.data[cl<span class="sc">$</span>IDorder, ]</span>
<span id="cb8-141"><a href="source-code.html#cb8-141" tabindex="-1"></a>    class.data<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(class.data<span class="sc">$</span>ID, <span class="at">levels =</span> class.data<span class="sc">$</span>ID)</span>
<span id="cb8-142"><a href="source-code.html#cb8-142" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb8-143"><a href="source-code.html#cb8-143" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb8-144"><a href="source-code.html#cb8-144" tabindex="-1"></a>      hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb8-145"><a href="source-code.html#cb8-145" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>segments)) {</span>
<span id="cb8-146"><a href="source-code.html#cb8-146" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb8-147"><a href="source-code.html#cb8-147" tabindex="-1"></a>      }</span>
<span id="cb8-148"><a href="source-code.html#cb8-148" tabindex="-1"></a>      <span class="cf">else</span> {</span>
<span id="cb8-149"><a href="source-code.html#cb8-149" tabindex="-1"></a>        segments.list[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>segments</span>
<span id="cb8-150"><a href="source-code.html#cb8-150" tabindex="-1"></a>      }</span>
<span id="cb8-151"><a href="source-code.html#cb8-151" tabindex="-1"></a>    }</span>
<span id="cb8-152"><a href="source-code.html#cb8-152" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb8-153"><a href="source-code.html#cb8-153" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>()</span>
<span id="cb8-154"><a href="source-code.html#cb8-154" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hmms)) {</span>
<span id="cb8-155"><a href="source-code.html#cb8-155" tabindex="-1"></a>        hmm <span class="ot">&lt;-</span> hmms[[i1]]</span>
<span id="cb8-156"><a href="source-code.html#cb8-156" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.null</span>(hmm<span class="sc">$</span>breakpoints)) {</span>
<span id="cb8-157"><a href="source-code.html#cb8-157" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> <span class="fu">GRanges</span>()</span>
<span id="cb8-158"><a href="source-code.html#cb8-158" tabindex="-1"></a>        }</span>
<span id="cb8-159"><a href="source-code.html#cb8-159" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb8-160"><a href="source-code.html#cb8-160" tabindex="-1"></a>          breakpoints[[hmm<span class="sc">$</span>ID]] <span class="ot">&lt;-</span> hmm<span class="sc">$</span>breakpoints</span>
<span id="cb8-161"><a href="source-code.html#cb8-161" tabindex="-1"></a>        }</span>
<span id="cb8-162"><a href="source-code.html#cb8-162" tabindex="-1"></a>      }</span>
<span id="cb8-163"><a href="source-code.html#cb8-163" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(breakpoints) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb8-164"><a href="source-code.html#cb8-164" tabindex="-1"></a>        plot.breakpoints <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb8-165"><a href="source-code.html#cb8-165" tabindex="-1"></a>      }</span>
<span id="cb8-166"><a href="source-code.html#cb8-166" tabindex="-1"></a>    }</span>
<span id="cb8-167"><a href="source-code.html#cb8-167" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Transforming coordinates ...&quot;</span>)</span>
<span id="cb8-168"><a href="source-code.html#cb8-168" tabindex="-1"></a>    segments.list <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(segments.list, transCoord)</span>
<span id="cb8-169"><a href="source-code.html#cb8-169" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb8-170"><a href="source-code.html#cb8-170" tabindex="-1"></a>      breakpoints <span class="ot">&lt;-</span> <span class="fu">endoapply</span>(breakpoints, transCoord)</span>
<span id="cb8-171"><a href="source-code.html#cb8-171" tabindex="-1"></a>    }</span>
<span id="cb8-172"><a href="source-code.html#cb8-172" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb8-173"><a href="source-code.html#cb8-173" tabindex="-1"></a>    ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Making the plot ...&quot;</span>)</span>
<span id="cb8-174"><a href="source-code.html#cb8-174" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-175"><a href="source-code.html#cb8-175" tabindex="-1"></a>    <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(segments.list)) {</span>
<span id="cb8-176"><a href="source-code.html#cb8-176" tabindex="-1"></a>      df[[<span class="fu">length</span>(df) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> segments.list[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb8-177"><a href="source-code.html#cb8-177" tabindex="-1"></a>                                         <span class="at">end =</span> segments.list[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(segments.list[[i1]]), </span>
<span id="cb8-178"><a href="source-code.html#cb8-178" tabindex="-1"></a>                                         <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">state =</span> segments.list[[i1]]<span class="sc">$</span>state)</span>
<span id="cb8-179"><a href="source-code.html#cb8-179" tabindex="-1"></a>    }</span>
<span id="cb8-180"><a href="source-code.html#cb8-180" tabindex="-1"></a>    df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df)</span>
<span id="cb8-181"><a href="source-code.html#cb8-181" tabindex="-1"></a>    df<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb8-182"><a href="source-code.html#cb8-182" tabindex="-1"></a>    df<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df<span class="sc">$</span>ID)]</span>
<span id="cb8-183"><a href="source-code.html#cb8-183" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb8-184"><a href="source-code.html#cb8-184" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-185"><a href="source-code.html#cb8-185" tabindex="-1"></a>      <span class="cf">for</span> (i1 <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(breakpoints)) {</span>
<span id="cb8-186"><a href="source-code.html#cb8-186" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">length</span>(breakpoints[[i1]]) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb8-187"><a href="source-code.html#cb8-187" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> breakpoints[[i1]]<span class="sc">$</span>start.genome, </span>
<span id="cb8-188"><a href="source-code.html#cb8-188" tabindex="-1"></a>                                                                     <span class="at">end =</span> breakpoints[[i1]]<span class="sc">$</span>end.genome, <span class="at">seqnames =</span> <span class="fu">seqnames</span>(breakpoints[[i1]]), </span>
<span id="cb8-189"><a href="source-code.html#cb8-189" tabindex="-1"></a>                                                                     <span class="at">ID =</span> <span class="fu">names</span>(segments.list)[i1], <span class="at">mid =</span> (breakpoints[[i1]]<span class="sc">$</span>start.genome <span class="sc">+</span> </span>
<span id="cb8-190"><a href="source-code.html#cb8-190" tabindex="-1"></a>                                                                                                             breakpoints[[i1]]<span class="sc">$</span>end.genome)<span class="sc">/</span><span class="dv">2</span>)</span>
<span id="cb8-191"><a href="source-code.html#cb8-191" tabindex="-1"></a>        }</span>
<span id="cb8-192"><a href="source-code.html#cb8-192" tabindex="-1"></a>        <span class="cf">else</span> {</span>
<span id="cb8-193"><a href="source-code.html#cb8-193" tabindex="-1"></a>          df.breakpoints[[<span class="fu">length</span>(df.breakpoints) <span class="sc">+</span> <span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">start =</span> <span class="fu">numeric</span>(), </span>
<span id="cb8-194"><a href="source-code.html#cb8-194" tabindex="-1"></a>                                                                     <span class="at">end =</span> <span class="fu">numeric</span>(), <span class="at">seqnames =</span> <span class="fu">character</span>(), <span class="at">ID =</span> <span class="fu">character</span>(), </span>
<span id="cb8-195"><a href="source-code.html#cb8-195" tabindex="-1"></a>                                                                     <span class="at">mid =</span> <span class="fu">numeric</span>())</span>
<span id="cb8-196"><a href="source-code.html#cb8-196" tabindex="-1"></a>        }</span>
<span id="cb8-197"><a href="source-code.html#cb8-197" tabindex="-1"></a>      }</span>
<span id="cb8-198"><a href="source-code.html#cb8-198" tabindex="-1"></a>      df.breakpoints <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, df.breakpoints)</span>
<span id="cb8-199"><a href="source-code.html#cb8-199" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(df.breakpoints<span class="sc">$</span>ID, <span class="at">levels =</span> <span class="fu">levels</span>(class.data<span class="sc">$</span>ID))</span>
<span id="cb8-200"><a href="source-code.html#cb8-200" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>ylabel <span class="ot">&lt;-</span> mapping[<span class="fu">as.character</span>(df.breakpoints<span class="sc">$</span>ID)]</span>
<span id="cb8-201"><a href="source-code.html#cb8-201" tabindex="-1"></a>    }</span>
<span id="cb8-202"><a href="source-code.html#cb8-202" tabindex="-1"></a>    cum.seqlengths <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]])))</span>
<span id="cb8-203"><a href="source-code.html#cb8-203" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb8-204"><a href="source-code.html#cb8-204" tabindex="-1"></a>    cum.seqlengths<span class="fl">.0</span> <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths[<span class="sc">-</span><span class="fu">length</span>(cum.seqlengths)])</span>
<span id="cb8-205"><a href="source-code.html#cb8-205" tabindex="-1"></a>    <span class="fu">names</span>(cum.seqlengths<span class="fl">.0</span>) <span class="ot">&lt;-</span> <span class="fu">seqlevels</span>(segments.list[[<span class="dv">1</span>]])</span>
<span id="cb8-206"><a href="source-code.html#cb8-206" tabindex="-1"></a>    label.pos <span class="ot">&lt;-</span> <span class="fu">round</span>(cum.seqlengths<span class="fl">.0</span> <span class="sc">+</span> <span class="fl">0.5</span> <span class="sc">*</span> <span class="fu">seqlengths</span>(segments.list[[<span class="dv">1</span>]]))</span>
<span id="cb8-207"><a href="source-code.html#cb8-207" tabindex="-1"></a>    df.chroms <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">y =</span> <span class="fu">c</span>(<span class="dv">0</span>, cum.seqlengths), <span class="at">x =</span> <span class="dv">1</span>, </span>
<span id="cb8-208"><a href="source-code.html#cb8-208" tabindex="-1"></a>                            <span class="at">xend =</span> <span class="fu">length</span>(segments.list))</span>
<span id="cb8-209"><a href="source-code.html#cb8-209" tabindex="-1"></a>    pltlist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb8-210"><a href="source-code.html#cb8-210" tabindex="-1"></a>    widths <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb8-211"><a href="source-code.html#cb8-211" tabindex="-1"></a>    df<span class="sc">$</span>state <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>state, <span class="at">levels =</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">initializeStates</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))<span class="sc">$</span>multiplicity)))</span>
<span id="cb8-212"><a href="source-code.html#cb8-212" tabindex="-1"></a>    df<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df<span class="sc">$</span>ID)</span>
<span id="cb8-213"><a href="source-code.html#cb8-213" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, </span>
<span id="cb8-214"><a href="source-code.html#cb8-214" tabindex="-1"></a>                                                    <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;state&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> label.pos, </span>
<span id="cb8-215"><a href="source-code.html#cb8-215" tabindex="-1"></a>                                                                                                                          <span class="at">labels =</span> <span class="fu">names</span>(label.pos))</span>
<span id="cb8-216"><a href="source-code.html#cb8-216" tabindex="-1"></a>    </span>
<span id="cb8-217"><a href="source-code.html#cb8-217" tabindex="-1"></a></span>
<span id="cb8-218"><a href="source-code.html#cb8-218" tabindex="-1"></a>    <span class="co"># ggplt &lt;- ggplt + scale_x_continuous(name = &quot;&quot;,</span></span>
<span id="cb8-219"><a href="source-code.html#cb8-219" tabindex="-1"></a>    <span class="co">#                      breaks = 1:length(unique(df$ylabel)),</span></span>
<span id="cb8-220"><a href="source-code.html#cb8-220" tabindex="-1"></a>    <span class="co">#                      labels = unique(df$ylabel))</span></span>
<span id="cb8-221"><a href="source-code.html#cb8-221" tabindex="-1"></a>    </span>
<span id="cb8-222"><a href="source-code.html#cb8-222" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">stateColors</span>(<span class="fu">levels</span>(df<span class="sc">$</span>state))) <span class="co"># adding custom colours</span></span>
<span id="cb8-223"><a href="source-code.html#cb8-223" tabindex="-1"></a>    <span class="co"># adjusintg x axis</span></span>
<span id="cb8-224"><a href="source-code.html#cb8-224" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-225"><a href="source-code.html#cb8-225" tabindex="-1"></a>                           <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-226"><a href="source-code.html#cb8-226" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb8-227"><a href="source-code.html#cb8-227" tabindex="-1"></a>    </span>
<span id="cb8-228"><a href="source-code.html#cb8-228" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>,</span>
<span id="cb8-229"><a href="source-code.html#cb8-229" tabindex="-1"></a>                                             <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;y&quot;</span>), <span class="at">data =</span> df.chroms, <span class="at">col =</span> <span class="st">&quot;grey13&quot;</span>)</span>
<span id="cb8-230"><a href="source-code.html#cb8-230" tabindex="-1"></a>    </span>
<span id="cb8-231"><a href="source-code.html#cb8-231" tabindex="-1"></a>    </span>
<span id="cb8-232"><a href="source-code.html#cb8-232" tabindex="-1"></a>    </span>
<span id="cb8-233"><a href="source-code.html#cb8-233" tabindex="-1"></a>    <span class="co"># adjusting y axis</span></span>
<span id="cb8-234"><a href="source-code.html#cb8-234" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-235"><a href="source-code.html#cb8-235" tabindex="-1"></a>                           <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb8-236"><a href="source-code.html#cb8-236" tabindex="-1"></a>    </span>
<span id="cb8-237"><a href="source-code.html#cb8-237" tabindex="-1"></a>    <span class="co"># removing legend</span></span>
<span id="cb8-238"><a href="source-code.html#cb8-238" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb8-239"><a href="source-code.html#cb8-239" tabindex="-1"></a>    </span>
<span id="cb8-240"><a href="source-code.html#cb8-240" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb8-241"><a href="source-code.html#cb8-241" tabindex="-1"></a>    </span>
<span id="cb8-242"><a href="source-code.html#cb8-242" tabindex="-1"></a>    <span class="co"># removing all axis names</span></span>
<span id="cb8-243"><a href="source-code.html#cb8-243" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb8-244"><a href="source-code.html#cb8-244" tabindex="-1"></a>    </span>
<span id="cb8-245"><a href="source-code.html#cb8-245" tabindex="-1"></a>    <span class="co"># add numeber of cells sequenced text</span></span>
<span id="cb8-246"><a href="source-code.html#cb8-246" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">&quot;text&quot;</span>, </span>
<span id="cb8-247"><a href="source-code.html#cb8-247" tabindex="-1"></a>                              <span class="at">label =</span> <span class="fu">paste</span>(<span class="fu">length</span>(hmms), <span class="st">&#39;cells&#39;</span>), </span>
<span id="cb8-248"><a href="source-code.html#cb8-248" tabindex="-1"></a>                              <span class="at">x =</span> <span class="fu">length</span>(hmms)<span class="sc">/</span><span class="dv">2</span>, </span>
<span id="cb8-249"><a href="source-code.html#cb8-249" tabindex="-1"></a>                              <span class="at">y =</span> <span class="dv">3100000000</span>,</span>
<span id="cb8-250"><a href="source-code.html#cb8-250" tabindex="-1"></a>                              <span class="at">angle =</span> <span class="dv">270</span>,</span>
<span id="cb8-251"><a href="source-code.html#cb8-251" tabindex="-1"></a>                              <span class="at">size =</span> <span class="dv">50</span>)</span>
<span id="cb8-252"><a href="source-code.html#cb8-252" tabindex="-1"></a>    </span>
<span id="cb8-253"><a href="source-code.html#cb8-253" tabindex="-1"></a>    <span class="co"># decreasing plot margin</span></span>
<span id="cb8-254"><a href="source-code.html#cb8-254" tabindex="-1"></a>    ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb8-255"><a href="source-code.html#cb8-255" tabindex="-1"></a>    </span>
<span id="cb8-256"><a href="source-code.html#cb8-256" tabindex="-1"></a>    </span>
<span id="cb8-257"><a href="source-code.html#cb8-257" tabindex="-1"></a>    <span class="cf">if</span> (plot.breakpoints) {</span>
<span id="cb8-258"><a href="source-code.html#cb8-258" tabindex="-1"></a>      df.breakpoints<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(df.breakpoints<span class="sc">$</span>ID)</span>
<span id="cb8-259"><a href="source-code.html#cb8-259" tabindex="-1"></a>      ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb8-260"><a href="source-code.html#cb8-260" tabindex="-1"></a>                                      <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end&quot;</span>), </span>
<span id="cb8-261"><a href="source-code.html#cb8-261" tabindex="-1"></a>                                      <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;&quot;</span>) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">data =</span> df.breakpoints, </span>
<span id="cb8-262"><a href="source-code.html#cb8-262" tabindex="-1"></a>                                                                        <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">y =</span> <span class="st">&quot;mid&quot;</span>))</span>
<span id="cb8-263"><a href="source-code.html#cb8-263" tabindex="-1"></a>    }</span>
<span id="cb8-264"><a href="source-code.html#cb8-264" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(hotspots)) {</span>
<span id="cb8-265"><a href="source-code.html#cb8-265" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">length</span>(hotspots) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb8-266"><a href="source-code.html#cb8-266" tabindex="-1"></a>        df.hot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">transCoord</span>(hotspots))</span>
<span id="cb8-267"><a href="source-code.html#cb8-267" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmin <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb8-268"><a href="source-code.html#cb8-268" tabindex="-1"></a>        df.hot<span class="sc">$</span>xmax <span class="ot">&lt;-</span> <span class="fu">length</span>(class.data<span class="sc">$</span>ID) <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb8-269"><a href="source-code.html#cb8-269" tabindex="-1"></a>        ggplt <span class="ot">&lt;-</span> ggplt <span class="sc">+</span> <span class="fu">geom_rect</span>(<span class="at">data =</span> df.hot, <span class="at">mapping =</span> <span class="fu">aes_string</span>(<span class="at">xmin =</span> <span class="st">&quot;xmin&quot;</span>, </span>
<span id="cb8-270"><a href="source-code.html#cb8-270" tabindex="-1"></a>                                                                       <span class="at">xmax =</span> <span class="st">&quot;xmax&quot;</span>, <span class="at">ymin =</span> <span class="st">&quot;start.genome&quot;</span>, <span class="at">ymax =</span> <span class="st">&quot;end.genome&quot;</span>, </span>
<span id="cb8-271"><a href="source-code.html#cb8-271" tabindex="-1"></a>                                                                       <span class="at">alpha =</span> <span class="st">&quot;num.events&quot;</span>), <span class="at">fill =</span> <span class="st">&quot;hotpink4&quot;</span>) <span class="sc">+</span> <span class="fu">scale_alpha_continuous</span>(<span class="at">name =</span> <span class="st">&quot;breakpoints&quot;</span>, </span>
<span id="cb8-272"><a href="source-code.html#cb8-272" tabindex="-1"></a>                                                                                                                                          <span class="at">range =</span> <span class="fu">c</span>(<span class="fl">0.4</span>, <span class="fl">0.8</span>))</span>
<span id="cb8-273"><a href="source-code.html#cb8-273" tabindex="-1"></a>      }</span>
<span id="cb8-274"><a href="source-code.html#cb8-274" tabindex="-1"></a>    }</span>
<span id="cb8-275"><a href="source-code.html#cb8-275" tabindex="-1"></a>    </span>
<span id="cb8-276"><a href="source-code.html#cb8-276" tabindex="-1"></a>    </span>
<span id="cb8-277"><a href="source-code.html#cb8-277" tabindex="-1"></a>    width.heatmap <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">as.numeric</span>(<span class="fu">seqlengths</span>(hmms[[<span class="dv">1</span>]]<span class="sc">$</span>bins)))<span class="sc">/</span><span class="fl">3e+09</span> <span class="sc">*</span> </span>
<span id="cb8-278"><a href="source-code.html#cb8-278" tabindex="-1"></a>      <span class="dv">150</span></span>
<span id="cb8-279"><a href="source-code.html#cb8-279" tabindex="-1"></a>    height <span class="ot">&lt;-</span> <span class="fu">max</span>(<span class="fu">length</span>(hmms) <span class="sc">*</span> <span class="fl">0.5</span>, <span class="dv">2</span>)</span>
<span id="cb8-280"><a href="source-code.html#cb8-280" tabindex="-1"></a>    pltlist[[<span class="st">&quot;heatmap&quot;</span>]] <span class="ot">&lt;-</span> ggplt</span>
<span id="cb8-281"><a href="source-code.html#cb8-281" tabindex="-1"></a>    widths[<span class="st">&quot;heatmap&quot;</span>] <span class="ot">&lt;-</span> width.heatmap</span>
<span id="cb8-282"><a href="source-code.html#cb8-282" tabindex="-1"></a>    <span class="co"># adding class colors</span></span>
<span id="cb8-283"><a href="source-code.html#cb8-283" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes)) {</span>
<span id="cb8-284"><a href="source-code.html#cb8-284" tabindex="-1"></a>      width.classes <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb8-285"><a href="source-code.html#cb8-285" tabindex="-1"></a>      class.data<span class="sc">$</span>x <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(class.data<span class="sc">$</span>ID)</span>
<span id="cb8-286"><a href="source-code.html#cb8-286" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(class.data) <span class="sc">+</span> <span class="fu">geom_linerange</span>(<span class="fu">aes_string</span>(<span class="at">ymin =</span> <span class="dv">0</span>, </span>
<span id="cb8-287"><a href="source-code.html#cb8-287" tabindex="-1"></a>                                                                <span class="at">ymax =</span> <span class="dv">1</span>, <span class="at">x =</span> <span class="st">&quot;x&quot;</span>, <span class="at">col =</span> <span class="st">&quot;class&quot;</span>), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">col =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb8-288"><a href="source-code.html#cb8-288" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;&quot;</span>)</span>
<span id="cb8-289"><a href="source-code.html#cb8-289" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-290"><a href="source-code.html#cb8-290" tabindex="-1"></a>                                 <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-291"><a href="source-code.html#cb8-291" tabindex="-1"></a>                                 <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb8-292"><a href="source-code.html#cb8-292" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb8-293"><a href="source-code.html#cb8-293" tabindex="-1"></a>      </span>
<span id="cb8-294"><a href="source-code.html#cb8-294" tabindex="-1"></a>      <span class="co"># decreasing plot margin</span></span>
<span id="cb8-295"><a href="source-code.html#cb8-295" tabindex="-1"></a>      ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>), <span class="st">&quot;cm&quot;</span>))</span>
<span id="cb8-296"><a href="source-code.html#cb8-296" tabindex="-1"></a>      </span>
<span id="cb8-297"><a href="source-code.html#cb8-297" tabindex="-1"></a>      <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(classes.color)) {</span>
<span id="cb8-298"><a href="source-code.html#cb8-298" tabindex="-1"></a>        ggclass <span class="ot">&lt;-</span> ggclass <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">breaks =</span> <span class="fu">names</span>(classes.color), </span>
<span id="cb8-299"><a href="source-code.html#cb8-299" tabindex="-1"></a>                                                <span class="at">values =</span> classes.color)</span>
<span id="cb8-300"><a href="source-code.html#cb8-300" tabindex="-1"></a>      }</span>
<span id="cb8-301"><a href="source-code.html#cb8-301" tabindex="-1"></a>      pltlist[[<span class="st">&quot;classbar&quot;</span>]] <span class="ot">&lt;-</span> ggclass</span>
<span id="cb8-302"><a href="source-code.html#cb8-302" tabindex="-1"></a>      widths[<span class="st">&quot;classbar&quot;</span>] <span class="ot">&lt;-</span> width.classes</span>
<span id="cb8-303"><a href="source-code.html#cb8-303" tabindex="-1"></a>    }</span>
<span id="cb8-304"><a href="source-code.html#cb8-304" tabindex="-1"></a>    <span class="co"># adding dendogram</span></span>
<span id="cb8-305"><a href="source-code.html#cb8-305" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(cl<span class="sc">$</span>hclust) <span class="sc">&amp;</span> dendogram) {</span>
<span id="cb8-306"><a href="source-code.html#cb8-306" tabindex="-1"></a>      dhc <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">as.dendrogram</span>(cl<span class="sc">$</span>hclust)</span>
<span id="cb8-307"><a href="source-code.html#cb8-307" tabindex="-1"></a>      ddata <span class="ot">&lt;-</span> ggdendro<span class="sc">::</span><span class="fu">dendro_data</span>(dhc, <span class="at">type =</span> <span class="st">&quot;rectangle&quot;</span>)</span>
<span id="cb8-308"><a href="source-code.html#cb8-308" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(ddata<span class="sc">$</span>segments) <span class="sc">+</span> <span class="fu">geom_segment</span>(<span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">&quot;x&quot;</span>, </span>
<span id="cb8-309"><a href="source-code.html#cb8-309" tabindex="-1"></a>                                                                 <span class="at">xend =</span> <span class="st">&quot;xend&quot;</span>, <span class="at">y =</span> <span class="st">&quot;y&quot;</span>, <span class="at">yend =</span> <span class="st">&quot;yend&quot;</span>)) <span class="sc">+</span> <span class="fu">scale_y_reverse</span>()</span>
<span id="cb8-310"><a href="source-code.html#cb8-310" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">coord_flip</span>()</span>
<span id="cb8-311"><a href="source-code.html#cb8-311" tabindex="-1"></a>      ggdndr <span class="ot">&lt;-</span> ggdndr <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">panel.background =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-312"><a href="source-code.html#cb8-312" tabindex="-1"></a>                               <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb8-313"><a href="source-code.html#cb8-313" tabindex="-1"></a>                               <span class="at">axis.line =</span> <span class="fu">element_blank</span>(), <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb8-314"><a href="source-code.html#cb8-314" tabindex="-1"></a>      width.dendro <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb8-315"><a href="source-code.html#cb8-315" tabindex="-1"></a>      pltlist[[<span class="st">&quot;dendro&quot;</span>]] <span class="ot">&lt;-</span> ggdndr</span>
<span id="cb8-316"><a href="source-code.html#cb8-316" tabindex="-1"></a>      widths[<span class="st">&quot;dendro&quot;</span>] <span class="ot">&lt;-</span> width.dendro</span>
<span id="cb8-317"><a href="source-code.html#cb8-317" tabindex="-1"></a>    }</span>
<span id="cb8-318"><a href="source-code.html#cb8-318" tabindex="-1"></a>    <span class="co"># alligning ggpllt with dendogram and classes</span></span>
<span id="cb8-319"><a href="source-code.html#cb8-319" tabindex="-1"></a>    cowplt <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> <span class="fu">rev</span>(pltlist), <span class="at">align =</span> <span class="st">&quot;h&quot;</span>, </span>
<span id="cb8-320"><a href="source-code.html#cb8-320" tabindex="-1"></a>                                 <span class="at">ncol =</span> <span class="fu">length</span>(pltlist), <span class="at">rel_widths =</span> <span class="fu">rev</span>(widths))</span>
<span id="cb8-321"><a href="source-code.html#cb8-321" tabindex="-1"></a>    </span>
<span id="cb8-322"><a href="source-code.html#cb8-322" tabindex="-1"></a>    </span>
<span id="cb8-323"><a href="source-code.html#cb8-323" tabindex="-1"></a>    <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb8-324"><a href="source-code.html#cb8-324" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(file)) {</span>
<span id="cb8-325"><a href="source-code.html#cb8-325" tabindex="-1"></a>      ptm <span class="ot">&lt;-</span> <span class="fu">startTimedMessage</span>(<span class="st">&quot;Plotting to file &quot;</span>, file, <span class="st">&quot; ...&quot;</span>)</span>
<span id="cb8-326"><a href="source-code.html#cb8-326" tabindex="-1"></a>      <span class="fu">ggsave</span>(file, cowplt, <span class="at">width =</span> <span class="fu">sum</span>(widths), <span class="at">height =</span> height, </span>
<span id="cb8-327"><a href="source-code.html#cb8-327" tabindex="-1"></a>             <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>, <span class="at">limitsize =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-328"><a href="source-code.html#cb8-328" tabindex="-1"></a>      <span class="fu">stopTimedMessage</span>(ptm)</span>
<span id="cb8-329"><a href="source-code.html#cb8-329" tabindex="-1"></a>    }</span>
<span id="cb8-330"><a href="source-code.html#cb8-330" tabindex="-1"></a>    <span class="cf">else</span> {</span>
<span id="cb8-331"><a href="source-code.html#cb8-331" tabindex="-1"></a>      <span class="fu">return</span>(cowplt)</span>
<span id="cb8-332"><a href="source-code.html#cb8-332" tabindex="-1"></a>    }</span>
<span id="cb8-333"><a href="source-code.html#cb8-333" tabindex="-1"></a>  }</span>
<span id="cb8-334"><a href="source-code.html#cb8-334" tabindex="-1"></a>  </span>
<span id="cb8-335"><a href="source-code.html#cb8-335" tabindex="-1"></a>  <span class="co">#make heatmap and safe</span></span>
<span id="cb8-336"><a href="source-code.html#cb8-336" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(<span class="fu">heatmapGenomewide_daan</span>(selected.files, <span class="at">classes =</span> classes_daan, <span class="at">classes.color =</span> class.col,</span>
<span id="cb8-337"><a href="source-code.html#cb8-337" tabindex="-1"></a>                         <span class="at">file =</span> <span class="fu">paste0</span>(path, <span class="st">&#39;/&#39;</span>, name, <span class="st">&#39;_&#39;</span>, platename, <span class="st">&#39;.pdf&#39;</span>)))</span>
<span id="cb8-338"><a href="source-code.html#cb8-338" tabindex="-1"></a>  </span>
<span id="cb8-339"><a href="source-code.html#cb8-339" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">print</span>(<span class="fu">paste0</span>(name, <span class="st">&#39; done.&#39;</span>)))</span>
<span id="cb8-340"><a href="source-code.html#cb8-340" tabindex="-1"></a>  </span>
<span id="cb8-341"><a href="source-code.html#cb8-341" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="fisher_test-script-perform_fisher-function" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> fisher_test script: ‘perform_fisher’ function<a href="source-code.html#fisher_test-script-perform_fisher-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="source-code.html#cb9-1" tabindex="-1"></a>perform_fisher <span class="ot">&lt;-</span> <span class="cf">function</span>(dataframe,</span>
<span id="cb9-2"><a href="source-code.html#cb9-2" tabindex="-1"></a>                           population1,</span>
<span id="cb9-3"><a href="source-code.html#cb9-3" tabindex="-1"></a>                           population2) {</span>
<span id="cb9-4"><a href="source-code.html#cb9-4" tabindex="-1"></a>  <span class="co">#&#39; @param </span></span>
<span id="cb9-5"><a href="source-code.html#cb9-5" tabindex="-1"></a>  <span class="co">#&#39; dataframe is a df with columns paths and class with values &#39;hub015_rad_a&#39;, &#39;hub015_prerad_a&#39; etc.</span></span>
<span id="cb9-6"><a href="source-code.html#cb9-6" tabindex="-1"></a>  <span class="co">#&#39; population 1 and population two are the two pops you want to compare, for example prerad vs rad</span></span>
<span id="cb9-7"><a href="source-code.html#cb9-7" tabindex="-1"></a>  <span class="co">#&#39; or prerad versus rad cycle 2</span></span>
<span id="cb9-8"><a href="source-code.html#cb9-8" tabindex="-1"></a>  <span class="co">#&#39; @return</span></span>
<span id="cb9-9"><a href="source-code.html#cb9-9" tabindex="-1"></a>  <span class="co">#&#39; returns the p value of a Fisher&#39;s exact test. If the p value is above 0.05, the distributions</span></span>
<span id="cb9-10"><a href="source-code.html#cb9-10" tabindex="-1"></a>  <span class="co">#&#39; pre and post rad are the same. </span></span>
<span id="cb9-11"><a href="source-code.html#cb9-11" tabindex="-1"></a>  <span class="co">#&#39; </span></span>
<span id="cb9-12"><a href="source-code.html#cb9-12" tabindex="-1"></a>  </span>
<span id="cb9-13"><a href="source-code.html#cb9-13" tabindex="-1"></a>  dataframe<span class="sc">$</span>rad <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population1, <span class="st">&#39;pop1&#39;</span>,</span>
<span id="cb9-14"><a href="source-code.html#cb9-14" tabindex="-1"></a>                          <span class="fu">ifelse</span>(dataframe<span class="sc">$</span>class <span class="sc">%in%</span> population2, <span class="st">&#39;pop2&#39;</span>,</span>
<span id="cb9-15"><a href="source-code.html#cb9-15" tabindex="-1"></a>                                 <span class="cn">NA</span>))</span>
<span id="cb9-16"><a href="source-code.html#cb9-16" tabindex="-1"></a>  </span>
<span id="cb9-17"><a href="source-code.html#cb9-17" tabindex="-1"></a>  org_id <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">str_extract</span>(<span class="fu">unique</span>(dataframe<span class="sc">$</span>class), <span class="st">&#39;hub</span><span class="sc">\\</span><span class="st">d{3}&#39;</span>))</span>
<span id="cb9-18"><a href="source-code.html#cb9-18" tabindex="-1"></a>  </span>
<span id="cb9-19"><a href="source-code.html#cb9-19" tabindex="-1"></a>  <span class="co"># remove na</span></span>
<span id="cb9-20"><a href="source-code.html#cb9-20" tabindex="-1"></a>  dataframe <span class="ot">&lt;-</span> dataframe[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(dataframe<span class="sc">$</span>rad)),]</span>
<span id="cb9-21"><a href="source-code.html#cb9-21" tabindex="-1"></a>  </span>
<span id="cb9-22"><a href="source-code.html#cb9-22" tabindex="-1"></a>  simple_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(string) {</span>
<span id="cb9-23"><a href="source-code.html#cb9-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">tail</span>(string,<span class="dv">1</span>))</span>
<span id="cb9-24"><a href="source-code.html#cb9-24" tabindex="-1"></a>  }</span>
<span id="cb9-25"><a href="source-code.html#cb9-25" tabindex="-1"></a>  </span>
<span id="cb9-26"><a href="source-code.html#cb9-26" tabindex="-1"></a>  dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(stringr<span class="sc">::</span><span class="fu">str_split</span>(dataframe<span class="sc">$</span>class, <span class="at">pattern =</span> <span class="st">&#39;_&#39;</span>), simple_fun))</span>
<span id="cb9-27"><a href="source-code.html#cb9-27" tabindex="-1"></a>  </span>
<span id="cb9-28"><a href="source-code.html#cb9-28" tabindex="-1"></a>  <span class="co"># in hub015 and hub005, we defined subclones. These are actually part of the same clone, so should be </span></span>
<span id="cb9-29"><a href="source-code.html#cb9-29" tabindex="-1"></a>  <span class="co"># analysed together</span></span>
<span id="cb9-30"><a href="source-code.html#cb9-30" tabindex="-1"></a>  <span class="cf">if</span> (org_id <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;hub005&#39;</span>, <span class="st">&#39;hub015&#39;</span>)){</span>
<span id="cb9-31"><a href="source-code.html#cb9-31" tabindex="-1"></a>    dataframe<span class="sc">$</span>clone <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(dataframe<span class="sc">$</span>clone, <span class="at">pattern =</span> <span class="st">&#39;a.a|a.b&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;a&#39;</span>)</span>
<span id="cb9-32"><a href="source-code.html#cb9-32" tabindex="-1"></a>    </span>
<span id="cb9-33"><a href="source-code.html#cb9-33" tabindex="-1"></a>  }</span>
<span id="cb9-34"><a href="source-code.html#cb9-34" tabindex="-1"></a>  </span>
<span id="cb9-35"><a href="source-code.html#cb9-35" tabindex="-1"></a>  tab <span class="ot">&lt;-</span> <span class="fu">table</span>(dataframe<span class="sc">$</span>clone, dataframe<span class="sc">$</span>rad)</span>
<span id="cb9-36"><a href="source-code.html#cb9-36" tabindex="-1"></a>  </span>
<span id="cb9-37"><a href="source-code.html#cb9-37" tabindex="-1"></a>  </span>
<span id="cb9-38"><a href="source-code.html#cb9-38" tabindex="-1"></a>  <span class="co"># df &lt;- data.frame(cline &lt;- org_id, p_val &lt;- fisher.test(tab)$p.value)</span></span>
<span id="cb9-39"><a href="source-code.html#cb9-39" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb9-40"><a href="source-code.html#cb9-40" tabindex="-1"></a>  <span class="co"># return(df) </span></span>
<span id="cb9-41"><a href="source-code.html#cb9-41" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">fisher.test</span>(tab)<span class="sc">$</span>p.value)</span>
<span id="cb9-42"><a href="source-code.html#cb9-42" tabindex="-1"></a>  </span>
<span id="cb9-43"><a href="source-code.html#cb9-43" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="find_specific_cna_sc-script-find_specific_cna_sc-function" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> find_specific_cna_sc script: ‘find_specific_cna_sc’ function<a href="source-code.html#find_specific_cna_sc-script-find_specific_cna_sc-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="source-code.html#cb10-1" tabindex="-1"></a>find_specific_cna_sc <span class="ot">&lt;-</span></span>
<span id="cb10-2"><a href="source-code.html#cb10-2" tabindex="-1"></a>  <span class="cf">function</span>(list_resistant,</span>
<span id="cb10-3"><a href="source-code.html#cb10-3" tabindex="-1"></a>           list_sensitive,</span>
<span id="cb10-4"><a href="source-code.html#cb10-4" tabindex="-1"></a>           perc_cutoff_within_subclone_resistant,</span>
<span id="cb10-5"><a href="source-code.html#cb10-5" tabindex="-1"></a>           perc_cutoff_within_subclone_sensitive,</span>
<span id="cb10-6"><a href="source-code.html#cb10-6" tabindex="-1"></a>           per_cutoff_across_subclone_resistant,</span>
<span id="cb10-7"><a href="source-code.html#cb10-7" tabindex="-1"></a>           per_cutoff_across_subclone_sensitive,</span>
<span id="cb10-8"><a href="source-code.html#cb10-8" tabindex="-1"></a>           <span class="at">return_cnas_per_subclone =</span> F) {</span>
<span id="cb10-9"><a href="source-code.html#cb10-9" tabindex="-1"></a>    <span class="co">#&#39; @param list_resistant a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb10-10"><a href="source-code.html#cb10-10" tabindex="-1"></a>    <span class="co">#&#39; @param list_sensitive a list subclone names which itself contains a list of paths of paths to aneuHMM .Rda files. </span></span>
<span id="cb10-11"><a href="source-code.html#cb10-11" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_resistant user defined percentage cutoff to use (e.g. 0.6 means a certain CNA is shared by 60% of the cells</span></span>
<span id="cb10-12"><a href="source-code.html#cb10-12" tabindex="-1"></a>    <span class="co">#&#39; within a subclone).</span></span>
<span id="cb10-13"><a href="source-code.html#cb10-13" tabindex="-1"></a>    <span class="co">#&#39; @param perc_cutoff_within_subclone_senstive same as in perc_cutoff_within_subclone_resistant but for the sensitive group</span></span>
<span id="cb10-14"><a href="source-code.html#cb10-14" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_resistant user defined percentage cutoff to use (e.g. 0.7 means a certain CNA is shared in 70% of all subclones)</span></span>
<span id="cb10-15"><a href="source-code.html#cb10-15" tabindex="-1"></a>    <span class="co">#&#39; @param return_cnas_per_subclone boolean. If TRUE will return the shared by perc_cutoff_within_subclone CNAs for each list of each item within </span></span>
<span id="cb10-16"><a href="source-code.html#cb10-16" tabindex="-1"></a>    <span class="co">#&#39; list_resistant. Note list_sensitive should be set to NULL for proper results!!</span></span>
<span id="cb10-17"><a href="source-code.html#cb10-17" tabindex="-1"></a>    <span class="co">#&#39; @param per_cutoff_across_subclone_sensitive same as resistant but for the sensitive group</span></span>
<span id="cb10-18"><a href="source-code.html#cb10-18" tabindex="-1"></a>    <span class="co">#&#39; @example perc_cutoff_within_subclone = 0.6 and per_cutoff_across_subclone = 0.7 means the algorithm will find CNAs that are shared </span></span>
<span id="cb10-19"><a href="source-code.html#cb10-19" tabindex="-1"></a>    <span class="co">#&#39; by 60% of single cells within each subclone, and by 70% of all subclones. Important, a cutoff of 50% is the same as computing the median.</span></span>
<span id="cb10-20"><a href="source-code.html#cb10-20" tabindex="-1"></a>    <span class="co">#&#39; @return a list of GRanges showing unique to list_resistant amplifications and deletions and unique to list_sensitive</span></span>
<span id="cb10-21"><a href="source-code.html#cb10-21" tabindex="-1"></a>    <span class="co">#&#39; amplifications and deletions or, if return_cnas_per_subclone see param return_cnas_per_subclone above.</span></span>
<span id="cb10-22"><a href="source-code.html#cb10-22" tabindex="-1"></a>    <span class="co">#&#39; @details Note that this algorithm is NOT sensitive to the number of single cells within each subclone. </span></span>
<span id="cb10-23"><a href="source-code.html#cb10-23" tabindex="-1"></a>    </span>
<span id="cb10-24"><a href="source-code.html#cb10-24" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb10-25"><a href="source-code.html#cb10-25" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb10-26"><a href="source-code.html#cb10-26" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb10-27"><a href="source-code.html#cb10-27" tabindex="-1"></a>    <span class="fu">require</span>(plyranges)</span>
<span id="cb10-28"><a href="source-code.html#cb10-28" tabindex="-1"></a>    <span class="fu">require</span>(AneuFinder)</span>
<span id="cb10-29"><a href="source-code.html#cb10-29" tabindex="-1"></a>    </span>
<span id="cb10-30"><a href="source-code.html#cb10-30" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-31"><a href="source-code.html#cb10-31" tabindex="-1"></a>    <span class="co"># Part 1. #</span></span>
<span id="cb10-32"><a href="source-code.html#cb10-32" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-33"><a href="source-code.html#cb10-33" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb10-34"><a href="source-code.html#cb10-34" tabindex="-1"></a>    </span>
<span id="cb10-35"><a href="source-code.html#cb10-35" tabindex="-1"></a>    resistant_sc <span class="ot">&lt;-</span></span>
<span id="cb10-36"><a href="source-code.html#cb10-36" tabindex="-1"></a>      <span class="fu">lapply</span>(list_resistant,</span>
<span id="cb10-37"><a href="source-code.html#cb10-37" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb10-38"><a href="source-code.html#cb10-38" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb10-39"><a href="source-code.html#cb10-39" tabindex="-1"></a>    sensitive_sc <span class="ot">&lt;-</span></span>
<span id="cb10-40"><a href="source-code.html#cb10-40" tabindex="-1"></a>      <span class="fu">lapply</span>(list_sensitive,</span>
<span id="cb10-41"><a href="source-code.html#cb10-41" tabindex="-1"></a>             loadFromFiles,</span>
<span id="cb10-42"><a href="source-code.html#cb10-42" tabindex="-1"></a>             <span class="at">check.class =</span> <span class="fu">c</span>(<span class="st">&quot;aneuHMM&quot;</span>, <span class="st">&quot;aneuBiHMM&quot;</span>))</span>
<span id="cb10-43"><a href="source-code.html#cb10-43" tabindex="-1"></a>    </span>
<span id="cb10-44"><a href="source-code.html#cb10-44" tabindex="-1"></a>    <span class="co"># find specific CNAs needs a list of GRanges where each GRange has columns</span></span>
<span id="cb10-45"><a href="source-code.html#cb10-45" tabindex="-1"></a>    <span class="co"># seqnames, ranges, strand (not necessary) and cn.</span></span>
<span id="cb10-46"><a href="source-code.html#cb10-46" tabindex="-1"></a>    wrangle_sc <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff) {</span>
<span id="cb10-47"><a href="source-code.html#cb10-47" tabindex="-1"></a>      <span class="co"># segments is the reduced bins GRange (so all equal CN bins put together)</span></span>
<span id="cb10-48"><a href="source-code.html#cb10-48" tabindex="-1"></a>      wrangle_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_lff_within) {</span>
<span id="cb10-49"><a href="source-code.html#cb10-49" tabindex="-1"></a>        segments <span class="ot">&lt;-</span> list_lff_within<span class="sc">$</span>segments</span>
<span id="cb10-50"><a href="source-code.html#cb10-50" tabindex="-1"></a>        <span class="co"># ugly way of adding a cn column</span></span>
<span id="cb10-51"><a href="source-code.html#cb10-51" tabindex="-1"></a>        segments<span class="sc">$</span>cn <span class="ot">&lt;-</span> segments<span class="sc">$</span>copy.number</span>
<span id="cb10-52"><a href="source-code.html#cb10-52" tabindex="-1"></a>        </span>
<span id="cb10-53"><a href="source-code.html#cb10-53" tabindex="-1"></a>        <span class="fu">return</span>(segments)</span>
<span id="cb10-54"><a href="source-code.html#cb10-54" tabindex="-1"></a>        </span>
<span id="cb10-55"><a href="source-code.html#cb10-55" tabindex="-1"></a>        <span class="co"># </span><span class="re">END</span><span class="co"> of wrange_within function</span></span>
<span id="cb10-56"><a href="source-code.html#cb10-56" tabindex="-1"></a>        </span>
<span id="cb10-57"><a href="source-code.html#cb10-57" tabindex="-1"></a>      }</span>
<span id="cb10-58"><a href="source-code.html#cb10-58" tabindex="-1"></a>      within_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_lff, wrangle_within)</span>
<span id="cb10-59"><a href="source-code.html#cb10-59" tabindex="-1"></a>      </span>
<span id="cb10-60"><a href="source-code.html#cb10-60" tabindex="-1"></a>      <span class="co"># </span><span class="re">END</span><span class="co"> OF wrangle_sc function</span></span>
<span id="cb10-61"><a href="source-code.html#cb10-61" tabindex="-1"></a>      </span>
<span id="cb10-62"><a href="source-code.html#cb10-62" tabindex="-1"></a>    }</span>
<span id="cb10-63"><a href="source-code.html#cb10-63" tabindex="-1"></a></span>
<span id="cb10-64"><a href="source-code.html#cb10-64" tabindex="-1"></a>    list_resistant <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_sc, wrangle_sc)</span>
<span id="cb10-65"><a href="source-code.html#cb10-65" tabindex="-1"></a>    list_sensitive <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_sc, wrangle_sc)</span>
<span id="cb10-66"><a href="source-code.html#cb10-66" tabindex="-1"></a>    </span>
<span id="cb10-67"><a href="source-code.html#cb10-67" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Extracted segment info from files&#39;</span>)</span>
<span id="cb10-68"><a href="source-code.html#cb10-68" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb10-69"><a href="source-code.html#cb10-69" tabindex="-1"></a>    </span>
<span id="cb10-70"><a href="source-code.html#cb10-70" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-71"><a href="source-code.html#cb10-71" tabindex="-1"></a>    <span class="co"># Part 2. #</span></span>
<span id="cb10-72"><a href="source-code.html#cb10-72" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-73"><a href="source-code.html#cb10-73" tabindex="-1"></a>    <span class="co"># The Granges are now still arranged in bins. We are first going to</span></span>
<span id="cb10-74"><a href="source-code.html#cb10-74" tabindex="-1"></a>    <span class="co"># combine bins with similar copy number.</span></span>
<span id="cb10-75"><a href="source-code.html#cb10-75" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (this may take some minutes)...&#39;</span>)</span>
<span id="cb10-76"><a href="source-code.html#cb10-76" tabindex="-1"></a>    combine_indels <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus) {</span>
<span id="cb10-77"><a href="source-code.html#cb10-77" tabindex="-1"></a>      <span class="co">#&#39; @param subclone_consensus_list is a GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb10-78"><a href="source-code.html#cb10-78" tabindex="-1"></a>      <span class="co">#&#39; @returns list of Granges where each Granges has only CNA of cn2 or cn&gt;2</span></span>
<span id="cb10-79"><a href="source-code.html#cb10-79" tabindex="-1"></a>      </span>
<span id="cb10-80"><a href="source-code.html#cb10-80" tabindex="-1"></a>      combine_within <span class="ot">&lt;-</span> <span class="cf">function</span>(subclone_consensus_within) {</span>
<span id="cb10-81"><a href="source-code.html#cb10-81" tabindex="-1"></a>        <span class="co"># subset subclone_consensus in cn &lt;2, cn = 2 and cn &gt; 2</span></span>
<span id="cb10-82"><a href="source-code.html#cb10-82" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span></span>
<span id="cb10-83"><a href="source-code.html#cb10-83" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&lt;</span> <span class="dv">2</span>,]</span>
<span id="cb10-84"><a href="source-code.html#cb10-84" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span></span>
<span id="cb10-85"><a href="source-code.html#cb10-85" tabindex="-1"></a>          subclone_consensus_within[subclone_consensus_within<span class="sc">$</span>cn <span class="sc">&gt;</span> <span class="dv">2</span>,]</span>
<span id="cb10-86"><a href="source-code.html#cb10-86" tabindex="-1"></a>        </span>
<span id="cb10-87"><a href="source-code.html#cb10-87" tabindex="-1"></a>        <span class="co"># reduce each GRanges (joins consecutive ranges together)</span></span>
<span id="cb10-88"><a href="source-code.html#cb10-88" tabindex="-1"></a>        cn_less_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_less_2)</span>
<span id="cb10-89"><a href="source-code.html#cb10-89" tabindex="-1"></a>        cn_more_2 <span class="ot">&lt;-</span> <span class="fu">reduce</span>(cn_more_2)</span>
<span id="cb10-90"><a href="source-code.html#cb10-90" tabindex="-1"></a>        </span>
<span id="cb10-91"><a href="source-code.html#cb10-91" tabindex="-1"></a>        <span class="co"># reduce deletes metadata.</span></span>
<span id="cb10-92"><a href="source-code.html#cb10-92" tabindex="-1"></a>        <span class="co"># adding metadata</span></span>
<span id="cb10-93"><a href="source-code.html#cb10-93" tabindex="-1"></a>        cn_less_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_less_2))</span>
<span id="cb10-94"><a href="source-code.html#cb10-94" tabindex="-1"></a>        </span>
<span id="cb10-95"><a href="source-code.html#cb10-95" tabindex="-1"></a>        cn_more_2<span class="sc">$</span>cn <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="fu">length</span>(cn_more_2))</span>
<span id="cb10-96"><a href="source-code.html#cb10-96" tabindex="-1"></a>        </span>
<span id="cb10-97"><a href="source-code.html#cb10-97" tabindex="-1"></a>        <span class="co"># make list and rename items</span></span>
<span id="cb10-98"><a href="source-code.html#cb10-98" tabindex="-1"></a>        GR_list <span class="ot">&lt;-</span> <span class="fu">GRangesList</span>(cn_less_2, cn_more_2)</span>
<span id="cb10-99"><a href="source-code.html#cb10-99" tabindex="-1"></a>        <span class="fu">names</span>(GR_list) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;&lt;2&#39;</span>, <span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb10-100"><a href="source-code.html#cb10-100" tabindex="-1"></a>        </span>
<span id="cb10-101"><a href="source-code.html#cb10-101" tabindex="-1"></a>        <span class="co"># Return #</span></span>
<span id="cb10-102"><a href="source-code.html#cb10-102" tabindex="-1"></a>        <span class="fu">return</span>(GR_list)</span>
<span id="cb10-103"><a href="source-code.html#cb10-103" tabindex="-1"></a>      }</span>
<span id="cb10-104"><a href="source-code.html#cb10-104" tabindex="-1"></a>      </span>
<span id="cb10-105"><a href="source-code.html#cb10-105" tabindex="-1"></a>      combine_subcl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(subclone_consensus, combine_within)</span>
<span id="cb10-106"><a href="source-code.html#cb10-106" tabindex="-1"></a>      </span>
<span id="cb10-107"><a href="source-code.html#cb10-107" tabindex="-1"></a>    }</span>
<span id="cb10-108"><a href="source-code.html#cb10-108" tabindex="-1"></a>    </span>
<span id="cb10-109"><a href="source-code.html#cb10-109" tabindex="-1"></a>    <span class="co"># run an apply function here on each Granges in list_resistant versus list_sensitive</span></span>
<span id="cb10-110"><a href="source-code.html#cb10-110" tabindex="-1"></a>    resistant_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_resistant, combine_indels)</span>
<span id="cb10-111"><a href="source-code.html#cb10-111" tabindex="-1"></a>    sensitive_split <span class="ot">&lt;-</span> <span class="fu">lapply</span>(list_sensitive, combine_indels)</span>
<span id="cb10-112"><a href="source-code.html#cb10-112" tabindex="-1"></a>    </span>
<span id="cb10-113"><a href="source-code.html#cb10-113" tabindex="-1"></a>    <span class="co"># splits deletions together within each subclone</span></span>
<span id="cb10-114"><a href="source-code.html#cb10-114" tabindex="-1"></a>    grouper_del <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb10-115"><a href="source-code.html#cb10-115" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb10-116"><a href="source-code.html#cb10-116" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&lt;2&#39;</span>)</span>
<span id="cb10-117"><a href="source-code.html#cb10-117" tabindex="-1"></a>      }</span>
<span id="cb10-118"><a href="source-code.html#cb10-118" tabindex="-1"></a>      </span>
<span id="cb10-119"><a href="source-code.html#cb10-119" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb10-120"><a href="source-code.html#cb10-120" tabindex="-1"></a>    }</span>
<span id="cb10-121"><a href="source-code.html#cb10-121" tabindex="-1"></a>    </span>
<span id="cb10-122"><a href="source-code.html#cb10-122" tabindex="-1"></a>    <span class="co"># splits amplifciations together within each subclone</span></span>
<span id="cb10-123"><a href="source-code.html#cb10-123" tabindex="-1"></a>    grouper_ampl <span class="ot">&lt;-</span> <span class="cf">function</span>(gr) {</span>
<span id="cb10-124"><a href="source-code.html#cb10-124" tabindex="-1"></a>      grouper_within <span class="ot">&lt;-</span> <span class="cf">function</span>(gr_within) {</span>
<span id="cb10-125"><a href="source-code.html#cb10-125" tabindex="-1"></a>        <span class="fu">return</span>(gr_within<span class="sc">$</span><span class="st">&#39;&gt;2&#39;</span>)</span>
<span id="cb10-126"><a href="source-code.html#cb10-126" tabindex="-1"></a>      }</span>
<span id="cb10-127"><a href="source-code.html#cb10-127" tabindex="-1"></a>      </span>
<span id="cb10-128"><a href="source-code.html#cb10-128" tabindex="-1"></a>      within_grouper <span class="ot">&lt;-</span> <span class="fu">lapply</span>(gr, grouper_within)</span>
<span id="cb10-129"><a href="source-code.html#cb10-129" tabindex="-1"></a>    }</span>
<span id="cb10-130"><a href="source-code.html#cb10-130" tabindex="-1"></a>    </span>
<span id="cb10-131"><a href="source-code.html#cb10-131" tabindex="-1"></a>    <span class="co"># run apply on each subclone</span></span>
<span id="cb10-132"><a href="source-code.html#cb10-132" tabindex="-1"></a>    resistant_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_del)</span>
<span id="cb10-133"><a href="source-code.html#cb10-133" tabindex="-1"></a>    resistant_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(resistant_split, grouper_ampl)</span>
<span id="cb10-134"><a href="source-code.html#cb10-134" tabindex="-1"></a>    </span>
<span id="cb10-135"><a href="source-code.html#cb10-135" tabindex="-1"></a>    sensitive_split_del <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_del)</span>
<span id="cb10-136"><a href="source-code.html#cb10-136" tabindex="-1"></a>    sensitive_split_ampl <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sensitive_split, grouper_ampl)</span>
<span id="cb10-137"><a href="source-code.html#cb10-137" tabindex="-1"></a>    </span>
<span id="cb10-138"><a href="source-code.html#cb10-138" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-139"><a href="source-code.html#cb10-139" tabindex="-1"></a>    <span class="co"># Part 3. #</span></span>
<span id="cb10-140"><a href="source-code.html#cb10-140" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-141"><a href="source-code.html#cb10-141" tabindex="-1"></a>    <span class="co"># Part 3A #</span></span>
<span id="cb10-142"><a href="source-code.html#cb10-142" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-143"><a href="source-code.html#cb10-143" tabindex="-1"></a>    <span class="co"># Finding common CNAs within each subclone</span></span>
<span id="cb10-144"><a href="source-code.html#cb10-144" tabindex="-1"></a>    shared_cnas <span class="ot">&lt;-</span></span>
<span id="cb10-145"><a href="source-code.html#cb10-145" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb10-146"><a href="source-code.html#cb10-146" tabindex="-1"></a>               perc_cutoff_within_subclone) {</span>
<span id="cb10-147"><a href="source-code.html#cb10-147" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb10-148"><a href="source-code.html#cb10-148" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff_within_subclone user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb10-149"><a href="source-code.html#cb10-149" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb10-150"><a href="source-code.html#cb10-150" tabindex="-1"></a>        </span>
<span id="cb10-151"><a href="source-code.html#cb10-151" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb10-152"><a href="source-code.html#cb10-152" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb10-153"><a href="source-code.html#cb10-153" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb10-154"><a href="source-code.html#cb10-154" tabindex="-1"></a>        </span>
<span id="cb10-155"><a href="source-code.html#cb10-155" tabindex="-1"></a>        shared_cnas_within <span class="ot">&lt;-</span> <span class="cf">function</span>(list_GRanges_within) {</span>
<span id="cb10-156"><a href="source-code.html#cb10-156" tabindex="-1"></a>          n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges_within) <span class="co"># number of range sets</span></span>
<span id="cb10-157"><a href="source-code.html#cb10-157" tabindex="-1"></a>          shared_cna <span class="ot">&lt;-</span></span>
<span id="cb10-158"><a href="source-code.html#cb10-158" tabindex="-1"></a>            <span class="fu">bind_ranges</span>(list_GRanges_within, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-159"><a href="source-code.html#cb10-159" tabindex="-1"></a>            <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-160"><a href="source-code.html#cb10-160" tabindex="-1"></a>            <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb10-161"><a href="source-code.html#cb10-161" tabindex="-1"></a>            <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> perc_cutoff_within_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb10-162"><a href="source-code.html#cb10-162" tabindex="-1"></a>            <span class="fu">reduce_ranges</span>()</span>
<span id="cb10-163"><a href="source-code.html#cb10-163" tabindex="-1"></a>          </span>
<span id="cb10-164"><a href="source-code.html#cb10-164" tabindex="-1"></a>          <span class="fu">return</span>(shared_cna)</span>
<span id="cb10-165"><a href="source-code.html#cb10-165" tabindex="-1"></a>        }</span>
<span id="cb10-166"><a href="source-code.html#cb10-166" tabindex="-1"></a>        </span>
<span id="cb10-167"><a href="source-code.html#cb10-167" tabindex="-1"></a>        <span class="fu">lapply</span>(list_GRanges, shared_cnas_within)</span>
<span id="cb10-168"><a href="source-code.html#cb10-168" tabindex="-1"></a>      }</span>
<span id="cb10-169"><a href="source-code.html#cb10-169" tabindex="-1"></a>    </span>
<span id="cb10-170"><a href="source-code.html#cb10-170" tabindex="-1"></a>    resistant_shared_del <span class="ot">&lt;-</span></span>
<span id="cb10-171"><a href="source-code.html#cb10-171" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb10-172"><a href="source-code.html#cb10-172" tabindex="-1"></a>    resistant_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb10-173"><a href="source-code.html#cb10-173" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(resistant_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_resistant)</span>
<span id="cb10-174"><a href="source-code.html#cb10-174" tabindex="-1"></a>    sensitive_shared_del <span class="ot">&lt;-</span></span>
<span id="cb10-175"><a href="source-code.html#cb10-175" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_del, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb10-176"><a href="source-code.html#cb10-176" tabindex="-1"></a>    sensitive_shared_ampl <span class="ot">&lt;-</span></span>
<span id="cb10-177"><a href="source-code.html#cb10-177" tabindex="-1"></a>      <span class="fu">shared_cnas</span>(sensitive_split_ampl, <span class="at">perc_cutoff_within_subclone =</span> perc_cutoff_within_subclone_sensitive)</span>
<span id="cb10-178"><a href="source-code.html#cb10-178" tabindex="-1"></a>    </span>
<span id="cb10-179"><a href="source-code.html#cb10-179" tabindex="-1"></a>    <span class="co"># </span><span class="re">END</span><span class="co"> of function if return_cnas_per_subclone == T</span></span>
<span id="cb10-180"><a href="source-code.html#cb10-180" tabindex="-1"></a>    <span class="cf">if</span> (return_cnas_per_subclone <span class="sc">==</span> T) {</span>
<span id="cb10-181"><a href="source-code.html#cb10-181" tabindex="-1"></a>      cna_list <span class="ot">&lt;-</span> <span class="fu">list</span> (<span class="at">deletions =</span> <span class="fu">c</span>(resistant_shared_del, sensitive_shared_del), </span>
<span id="cb10-182"><a href="source-code.html#cb10-182" tabindex="-1"></a>                        <span class="at">amplifications =</span> <span class="fu">c</span>(resistant_shared_ampl, sensitive_shared_del))</span>
<span id="cb10-183"><a href="source-code.html#cb10-183" tabindex="-1"></a>      </span>
<span id="cb10-184"><a href="source-code.html#cb10-184" tabindex="-1"></a>      <span class="fu">print</span>(</span>
<span id="cb10-185"><a href="source-code.html#cb10-185" tabindex="-1"></a>        <span class="fu">paste0</span>(</span>
<span id="cb10-186"><a href="source-code.html#cb10-186" tabindex="-1"></a>          <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb10-187"><a href="source-code.html#cb10-187" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb10-188"><a href="source-code.html#cb10-188" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each resistant subclone, and by &#39;</span>,</span>
<span id="cb10-189"><a href="source-code.html#cb10-189" tabindex="-1"></a>          <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb10-190"><a href="source-code.html#cb10-190" tabindex="-1"></a>          <span class="st">&#39;% of single cells within each sensitive subclone&#39;</span></span>
<span id="cb10-191"><a href="source-code.html#cb10-191" tabindex="-1"></a>        )</span>
<span id="cb10-192"><a href="source-code.html#cb10-192" tabindex="-1"></a>      )</span>
<span id="cb10-193"><a href="source-code.html#cb10-193" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb10-194"><a href="source-code.html#cb10-194" tabindex="-1"></a>      <span class="co"># Unload #</span></span>
<span id="cb10-195"><a href="source-code.html#cb10-195" tabindex="-1"></a>      <span class="do">##########</span></span>
<span id="cb10-196"><a href="source-code.html#cb10-196" tabindex="-1"></a>      <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb10-197"><a href="source-code.html#cb10-197" tabindex="-1"></a>                                        detach,</span>
<span id="cb10-198"><a href="source-code.html#cb10-198" tabindex="-1"></a>                                        <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb10-199"><a href="source-code.html#cb10-199" tabindex="-1"></a>      </span>
<span id="cb10-200"><a href="source-code.html#cb10-200" tabindex="-1"></a>      <span class="fu">return</span>(cna_list)</span>
<span id="cb10-201"><a href="source-code.html#cb10-201" tabindex="-1"></a>    }</span>
<span id="cb10-202"><a href="source-code.html#cb10-202" tabindex="-1"></a>    </span>
<span id="cb10-203"><a href="source-code.html#cb10-203" tabindex="-1"></a>    <span class="co"># Part 3B #</span></span>
<span id="cb10-204"><a href="source-code.html#cb10-204" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-205"><a href="source-code.html#cb10-205" tabindex="-1"></a>    <span class="co"># Finding common CNAs across subclones</span></span>
<span id="cb10-206"><a href="source-code.html#cb10-206" tabindex="-1"></a>    shared_cnas_across_subclones <span class="ot">&lt;-</span></span>
<span id="cb10-207"><a href="source-code.html#cb10-207" tabindex="-1"></a>      <span class="cf">function</span>(list_GRanges,</span>
<span id="cb10-208"><a href="source-code.html#cb10-208" tabindex="-1"></a>               per_cutoff_across_subclone) {</span>
<span id="cb10-209"><a href="source-code.html#cb10-209" tabindex="-1"></a>        <span class="co">#&#39; @param list_GRanges is a list of GRanges with seqnames, ranges, strand and cn</span></span>
<span id="cb10-210"><a href="source-code.html#cb10-210" tabindex="-1"></a>        <span class="co">#&#39; @param perc_cutoff user defined percentage cutoff to use (e.g. 0.6)</span></span>
<span id="cb10-211"><a href="source-code.html#cb10-211" tabindex="-1"></a>        <span class="co">#&#39; @returns Granges where each Granges is shared by =&gt; perc_cutoff_within_subclone</span></span>
<span id="cb10-212"><a href="source-code.html#cb10-212" tabindex="-1"></a>        </span>
<span id="cb10-213"><a href="source-code.html#cb10-213" tabindex="-1"></a>        <span class="co"># finding overlaps in x% of regions: wait for response community:</span></span>
<span id="cb10-214"><a href="source-code.html#cb10-214" tabindex="-1"></a>        <span class="co"># https://stackoverflow.com/questions/74900085/find-ranges-that-are-shared-by-80-or-more-of-10-granges-objects</span></span>
<span id="cb10-215"><a href="source-code.html#cb10-215" tabindex="-1"></a>        <span class="co"># https://support.bioconductor.org/p/9148540/</span></span>
<span id="cb10-216"><a href="source-code.html#cb10-216" tabindex="-1"></a>        </span>
<span id="cb10-217"><a href="source-code.html#cb10-217" tabindex="-1"></a>        n <span class="ot">&lt;-</span> <span class="fu">length</span>(list_GRanges) <span class="co"># number of range sets</span></span>
<span id="cb10-218"><a href="source-code.html#cb10-218" tabindex="-1"></a>        shared_cna <span class="ot">&lt;-</span></span>
<span id="cb10-219"><a href="source-code.html#cb10-219" tabindex="-1"></a>          <span class="fu">bind_ranges</span>(list_GRanges, <span class="at">.id =</span> <span class="st">&quot;origin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-220"><a href="source-code.html#cb10-220" tabindex="-1"></a>          <span class="fu">compute_coverage</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-221"><a href="source-code.html#cb10-221" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">fraction_cov =</span> score <span class="sc">/</span> n) <span class="sc">%&gt;%</span></span>
<span id="cb10-222"><a href="source-code.html#cb10-222" tabindex="-1"></a>          <span class="fu">filter</span>(fraction_cov <span class="sc">&gt;=</span> per_cutoff_across_subclone) <span class="sc">%&gt;%</span></span>
<span id="cb10-223"><a href="source-code.html#cb10-223" tabindex="-1"></a>          <span class="fu">reduce_ranges</span>()</span>
<span id="cb10-224"><a href="source-code.html#cb10-224" tabindex="-1"></a>        </span>
<span id="cb10-225"><a href="source-code.html#cb10-225" tabindex="-1"></a>        <span class="fu">return</span>(shared_cna)</span>
<span id="cb10-226"><a href="source-code.html#cb10-226" tabindex="-1"></a>        </span>
<span id="cb10-227"><a href="source-code.html#cb10-227" tabindex="-1"></a>      }</span>
<span id="cb10-228"><a href="source-code.html#cb10-228" tabindex="-1"></a>    </span>
<span id="cb10-229"><a href="source-code.html#cb10-229" tabindex="-1"></a>    resistant_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb10-230"><a href="source-code.html#cb10-230" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb10-231"><a href="source-code.html#cb10-231" tabindex="-1"></a>    resistant_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb10-232"><a href="source-code.html#cb10-232" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(resistant_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_resistant)</span>
<span id="cb10-233"><a href="source-code.html#cb10-233" tabindex="-1"></a>    sensitive_shared_del_across <span class="ot">&lt;-</span></span>
<span id="cb10-234"><a href="source-code.html#cb10-234" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_del, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb10-235"><a href="source-code.html#cb10-235" tabindex="-1"></a>    sensitive_shared_ampl_across <span class="ot">&lt;-</span></span>
<span id="cb10-236"><a href="source-code.html#cb10-236" tabindex="-1"></a>      <span class="fu">shared_cnas_across_subclones</span>(sensitive_shared_ampl, <span class="at">per_cutoff_across_subclone =</span> per_cutoff_across_subclone_sensitive)</span>
<span id="cb10-237"><a href="source-code.html#cb10-237" tabindex="-1"></a>    </span>
<span id="cb10-238"><a href="source-code.html#cb10-238" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb10-239"><a href="source-code.html#cb10-239" tabindex="-1"></a>      <span class="fu">paste0</span>(</span>
<span id="cb10-240"><a href="source-code.html#cb10-240" tabindex="-1"></a>        <span class="st">&#39;Extracted CNAs that are shared by &#39;</span>,</span>
<span id="cb10-241"><a href="source-code.html#cb10-241" tabindex="-1"></a>        <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(perc_cutoff_within_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>,</span>
<span id="cb10-242"><a href="source-code.html#cb10-242" tabindex="-1"></a>        <span class="st">&#39;% of single cells within resistant and sensitive subclones, respectively, and that are shared by &#39;</span>,</span>
<span id="cb10-243"><a href="source-code.html#cb10-243" tabindex="-1"></a>        <span class="fu">as.numeric</span>(per_cutoff_across_subclone_resistant) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39; and &#39;</span>, <span class="fu">as.numeric</span>(per_cutoff_across_subclone_sensitive) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb10-244"><a href="source-code.html#cb10-244" tabindex="-1"></a>        <span class="st">&#39;% across resistant and sensitive subclones, respectively.&#39;</span></span>
<span id="cb10-245"><a href="source-code.html#cb10-245" tabindex="-1"></a>      )</span>
<span id="cb10-246"><a href="source-code.html#cb10-246" tabindex="-1"></a>    )</span>
<span id="cb10-247"><a href="source-code.html#cb10-247" tabindex="-1"></a>    </span>
<span id="cb10-248"><a href="source-code.html#cb10-248" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">1</span>) </span>
<span id="cb10-249"><a href="source-code.html#cb10-249" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Computing (few seconds)...&#39;</span>)</span>
<span id="cb10-250"><a href="source-code.html#cb10-250" tabindex="-1"></a>    <span class="fu">Sys.sleep</span>(<span class="dv">2</span>) </span>
<span id="cb10-251"><a href="source-code.html#cb10-251" tabindex="-1"></a>    </span>
<span id="cb10-252"><a href="source-code.html#cb10-252" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-253"><a href="source-code.html#cb10-253" tabindex="-1"></a>    <span class="co"># Part 4. #</span></span>
<span id="cb10-254"><a href="source-code.html#cb10-254" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb10-255"><a href="source-code.html#cb10-255" tabindex="-1"></a>    <span class="co"># Now that we have shared CNAs across subclones with the resistant and sensitive group, we need to extract</span></span>
<span id="cb10-256"><a href="source-code.html#cb10-256" tabindex="-1"></a>    <span class="co"># only those CNAs that are unique to the resistant group.</span></span>
<span id="cb10-257"><a href="source-code.html#cb10-257" tabindex="-1"></a>    </span>
<span id="cb10-258"><a href="source-code.html#cb10-258" tabindex="-1"></a>    <span class="co"># unique to resistant lines deletions:</span></span>
<span id="cb10-259"><a href="source-code.html#cb10-259" tabindex="-1"></a>    unique_resistant_del <span class="ot">&lt;-</span></span>
<span id="cb10-260"><a href="source-code.html#cb10-260" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_del_across, sensitive_shared_del_across)</span>
<span id="cb10-261"><a href="source-code.html#cb10-261" tabindex="-1"></a>    <span class="co"># unique to resistant lines amplifications</span></span>
<span id="cb10-262"><a href="source-code.html#cb10-262" tabindex="-1"></a>    unique_resistant_ampl <span class="ot">&lt;-</span></span>
<span id="cb10-263"><a href="source-code.html#cb10-263" tabindex="-1"></a>      <span class="fu">setdiff</span>(resistant_shared_ampl_across,</span>
<span id="cb10-264"><a href="source-code.html#cb10-264" tabindex="-1"></a>              sensitive_shared_ampl_across)</span>
<span id="cb10-265"><a href="source-code.html#cb10-265" tabindex="-1"></a>    </span>
<span id="cb10-266"><a href="source-code.html#cb10-266" tabindex="-1"></a>    <span class="co"># unique to sensitive lines deletions:</span></span>
<span id="cb10-267"><a href="source-code.html#cb10-267" tabindex="-1"></a>    unique_sensitive_del <span class="ot">&lt;-</span></span>
<span id="cb10-268"><a href="source-code.html#cb10-268" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_del_across, resistant_shared_del_across)</span>
<span id="cb10-269"><a href="source-code.html#cb10-269" tabindex="-1"></a>    <span class="co"># unique to sensitive lines amplifications:</span></span>
<span id="cb10-270"><a href="source-code.html#cb10-270" tabindex="-1"></a>    unique_sensitive_ampl <span class="ot">&lt;-</span></span>
<span id="cb10-271"><a href="source-code.html#cb10-271" tabindex="-1"></a>      <span class="fu">setdiff</span>(sensitive_shared_ampl_across,</span>
<span id="cb10-272"><a href="source-code.html#cb10-272" tabindex="-1"></a>              resistant_shared_ampl_across)</span>
<span id="cb10-273"><a href="source-code.html#cb10-273" tabindex="-1"></a>    </span>
<span id="cb10-274"><a href="source-code.html#cb10-274" tabindex="-1"></a>    <span class="fu">print</span>(</span>
<span id="cb10-275"><a href="source-code.html#cb10-275" tabindex="-1"></a>      <span class="st">&#39;Identified amplifications and deletions that are unique to resistant or sensitive subclones&#39;</span></span>
<span id="cb10-276"><a href="source-code.html#cb10-276" tabindex="-1"></a>    )</span>
<span id="cb10-277"><a href="source-code.html#cb10-277" tabindex="-1"></a>    </span>
<span id="cb10-278"><a href="source-code.html#cb10-278" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb10-279"><a href="source-code.html#cb10-279" tabindex="-1"></a>    <span class="co"># Return #</span></span>
<span id="cb10-280"><a href="source-code.html#cb10-280" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb10-281"><a href="source-code.html#cb10-281" tabindex="-1"></a>    returner <span class="ot">&lt;-</span></span>
<span id="cb10-282"><a href="source-code.html#cb10-282" tabindex="-1"></a>      <span class="fu">list</span>(</span>
<span id="cb10-283"><a href="source-code.html#cb10-283" tabindex="-1"></a>        unique_resistant_del,</span>
<span id="cb10-284"><a href="source-code.html#cb10-284" tabindex="-1"></a>        unique_resistant_ampl,</span>
<span id="cb10-285"><a href="source-code.html#cb10-285" tabindex="-1"></a>        unique_sensitive_del,</span>
<span id="cb10-286"><a href="source-code.html#cb10-286" tabindex="-1"></a>        unique_sensitive_ampl</span>
<span id="cb10-287"><a href="source-code.html#cb10-287" tabindex="-1"></a>      )</span>
<span id="cb10-288"><a href="source-code.html#cb10-288" tabindex="-1"></a>    <span class="fu">names</span>(returner) <span class="ot">&lt;-</span></span>
<span id="cb10-289"><a href="source-code.html#cb10-289" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb10-290"><a href="source-code.html#cb10-290" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_del&#39;</span>,</span>
<span id="cb10-291"><a href="source-code.html#cb10-291" tabindex="-1"></a>        <span class="st">&#39;unique_resistant_ampl&#39;</span>,</span>
<span id="cb10-292"><a href="source-code.html#cb10-292" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_del&#39;</span>,</span>
<span id="cb10-293"><a href="source-code.html#cb10-293" tabindex="-1"></a>        <span class="st">&#39;unique_sensitive_ampl&#39;</span></span>
<span id="cb10-294"><a href="source-code.html#cb10-294" tabindex="-1"></a>      )</span>
<span id="cb10-295"><a href="source-code.html#cb10-295" tabindex="-1"></a>    </span>
<span id="cb10-296"><a href="source-code.html#cb10-296" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb10-297"><a href="source-code.html#cb10-297" tabindex="-1"></a>    <span class="co"># Unload #</span></span>
<span id="cb10-298"><a href="source-code.html#cb10-298" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb10-299"><a href="source-code.html#cb10-299" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(<span class="fu">paste0</span>(<span class="st">&quot;package:&quot;</span>, <span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs)),   <span class="co"># Unload add-on packages</span></span>
<span id="cb10-300"><a href="source-code.html#cb10-300" tabindex="-1"></a>                     detach,</span>
<span id="cb10-301"><a href="source-code.html#cb10-301" tabindex="-1"></a>                     <span class="at">character.only =</span> <span class="cn">TRUE</span>, <span class="at">unload =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb10-302"><a href="source-code.html#cb10-302" tabindex="-1"></a>    </span>
<span id="cb10-303"><a href="source-code.html#cb10-303" tabindex="-1"></a>    <span class="fu">return</span>(returner)</span>
<span id="cb10-304"><a href="source-code.html#cb10-304" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="get_cytoband_coverage-script-get_cytoband_coverage-function" class="section level2 hasAnchor" number="6.11">
<h2><span class="header-section-number">6.11</span> get_cytoband_coverage script: ‘get_cytoband_coverage’ function<a href="source-code.html#get_cytoband_coverage-script-get_cytoband_coverage-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="source-code.html#cb11-1" tabindex="-1"></a>get_cytoband_coverage <span class="ot">&lt;-</span> <span class="cf">function</span>(Grange) {</span>
<span id="cb11-2"><a href="source-code.html#cb11-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges.</span></span>
<span id="cb11-3"><a href="source-code.html#cb11-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns a GRange with cytogenetic locations including</span></span>
<span id="cb11-4"><a href="source-code.html#cb11-4" tabindex="-1"></a>  <span class="co">#&#39; percentage of overlap and a dataframe containing information on how much of</span></span>
<span id="cb11-5"><a href="source-code.html#cb11-5" tabindex="-1"></a>  <span class="co">#&#39; the total arm was affected.</span></span>
<span id="cb11-6"><a href="source-code.html#cb11-6" tabindex="-1"></a>  </span>
<span id="cb11-7"><a href="source-code.html#cb11-7" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb11-8"><a href="source-code.html#cb11-8" tabindex="-1"></a>  <span class="co"># Libraries #</span></span>
<span id="cb11-9"><a href="source-code.html#cb11-9" tabindex="-1"></a>  <span class="do">#############</span></span>
<span id="cb11-10"><a href="source-code.html#cb11-10" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb11-11"><a href="source-code.html#cb11-11" tabindex="-1"></a>  <span class="fu">require</span>(plyranges)</span>
<span id="cb11-12"><a href="source-code.html#cb11-12" tabindex="-1"></a>  </span>
<span id="cb11-13"><a href="source-code.html#cb11-13" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb11-14"><a href="source-code.html#cb11-14" tabindex="-1"></a>  <span class="co"># Data #</span></span>
<span id="cb11-15"><a href="source-code.html#cb11-15" tabindex="-1"></a>  <span class="do">########</span></span>
<span id="cb11-16"><a href="source-code.html#cb11-16" tabindex="-1"></a>  <span class="co"># Uncomment me, this is to download cytogenetic coordinates</span></span>
<span id="cb11-17"><a href="source-code.html#cb11-17" tabindex="-1"></a>  <span class="co"># library(biovizBase)</span></span>
<span id="cb11-18"><a href="source-code.html#cb11-18" tabindex="-1"></a>  <span class="co"># hg38IdeogramCyto &lt;- getIdeogram(&quot;hg38&quot;, cytobands = TRUE)</span></span>
<span id="cb11-19"><a href="source-code.html#cb11-19" tabindex="-1"></a>  <span class="co"># save(hg38IdeogramCyto, file = &#39;rda/cna_analysis/hg38IdeogramCyto.rda&#39;)</span></span>
<span id="cb11-20"><a href="source-code.html#cb11-20" tabindex="-1"></a>  </span>
<span id="cb11-21"><a href="source-code.html#cb11-21" tabindex="-1"></a>  <span class="co"># !!! ADDING ../cna_analysis for publication in GitHub !!!</span></span>
<span id="cb11-22"><a href="source-code.html#cb11-22" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/hg38IdeogramCyto.rda&#39;</span>)</span>
<span id="cb11-23"><a href="source-code.html#cb11-23" tabindex="-1"></a>  </span>
<span id="cb11-24"><a href="source-code.html#cb11-24" tabindex="-1"></a>  </span>
<span id="cb11-25"><a href="source-code.html#cb11-25" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb11-26"><a href="source-code.html#cb11-26" tabindex="-1"></a>  <span class="co"># Wrangle #</span></span>
<span id="cb11-27"><a href="source-code.html#cb11-27" tabindex="-1"></a>  <span class="do">###########</span></span>
<span id="cb11-28"><a href="source-code.html#cb11-28" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb11-29"><a href="source-code.html#cb11-29" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(Grange) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb11-30"><a href="source-code.html#cb11-30" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(hg38IdeogramCyto) <span class="ot">&lt;-</span> <span class="st">&quot;NCBI&quot;</span></span>
<span id="cb11-31"><a href="source-code.html#cb11-31" tabindex="-1"></a>  </span>
<span id="cb11-32"><a href="source-code.html#cb11-32" tabindex="-1"></a>  Grange_list_cytoband <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb11-33"><a href="source-code.html#cb11-33" tabindex="-1"></a>  Grange_list_pq_affected <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb11-34"><a href="source-code.html#cb11-34" tabindex="-1"></a>  </span>
<span id="cb11-35"><a href="source-code.html#cb11-35" tabindex="-1"></a>  <span class="co"># return empty Grange and empty data.frame if Grange is empty</span></span>
<span id="cb11-36"><a href="source-code.html#cb11-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(Grange) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb11-37"><a href="source-code.html#cb11-37" tabindex="-1"></a>    <span class="co"># return empty Grange</span></span>
<span id="cb11-38"><a href="source-code.html#cb11-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="fu">GRanges</span>(</span>
<span id="cb11-39"><a href="source-code.html#cb11-39" tabindex="-1"></a>      <span class="at">seqnames =</span> <span class="fu">character</span>(<span class="dv">0</span>),</span>
<span id="cb11-40"><a href="source-code.html#cb11-40" tabindex="-1"></a>      <span class="fu">IRanges</span>(<span class="at">start =</span> <span class="fu">integer</span>(<span class="dv">0</span>), <span class="at">end =</span> <span class="fu">integer</span>(<span class="dv">0</span>)),</span>
<span id="cb11-41"><a href="source-code.html#cb11-41" tabindex="-1"></a>      <span class="at">cytobands_cov =</span>  <span class="fu">character</span>(<span class="dv">0</span>)</span>
<span id="cb11-42"><a href="source-code.html#cb11-42" tabindex="-1"></a>    ),</span>
<span id="cb11-43"><a href="source-code.html#cb11-43" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">seqnames =</span> <span class="fu">character</span>(), </span>
<span id="cb11-44"><a href="source-code.html#cb11-44" tabindex="-1"></a>               <span class="at">start_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb11-45"><a href="source-code.html#cb11-45" tabindex="-1"></a>               <span class="at">end_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb11-46"><a href="source-code.html#cb11-46" tabindex="-1"></a>               <span class="at">width_of_arm =</span> <span class="fu">numeric</span>(), </span>
<span id="cb11-47"><a href="source-code.html#cb11-47" tabindex="-1"></a>               <span class="at">strand =</span> <span class="fu">character</span>(), </span>
<span id="cb11-48"><a href="source-code.html#cb11-48" tabindex="-1"></a>               <span class="at">chromosome =</span> <span class="fu">character</span>(), </span>
<span id="cb11-49"><a href="source-code.html#cb11-49" tabindex="-1"></a>               <span class="at">arm =</span> <span class="fu">character</span>(), </span>
<span id="cb11-50"><a href="source-code.html#cb11-50" tabindex="-1"></a>               <span class="at">percentage_arm_affected =</span> <span class="fu">numeric</span>(), </span>
<span id="cb11-51"><a href="source-code.html#cb11-51" tabindex="-1"></a>               <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-52"><a href="source-code.html#cb11-52" tabindex="-1"></a>    ))</span>
<span id="cb11-53"><a href="source-code.html#cb11-53" tabindex="-1"></a>  }</span>
<span id="cb11-54"><a href="source-code.html#cb11-54" tabindex="-1"></a>  </span>
<span id="cb11-55"><a href="source-code.html#cb11-55" tabindex="-1"></a>  <span class="cf">for</span> (row <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Grange)) {</span>
<span id="cb11-56"><a href="source-code.html#cb11-56" tabindex="-1"></a>    for_range <span class="ot">&lt;-</span> Grange[row]</span>
<span id="cb11-57"><a href="source-code.html#cb11-57" tabindex="-1"></a>    </span>
<span id="cb11-58"><a href="source-code.html#cb11-58" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb11-59"><a href="source-code.html#cb11-59" tabindex="-1"></a>    <span class="co"># Part 1 #</span></span>
<span id="cb11-60"><a href="source-code.html#cb11-60" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb11-61"><a href="source-code.html#cb11-61" tabindex="-1"></a>    <span class="do">## First find specific cytoband locations</span></span>
<span id="cb11-62"><a href="source-code.html#cb11-62" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb11-63"><a href="source-code.html#cb11-63" tabindex="-1"></a>    hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, hg38IdeogramCyto)</span>
<span id="cb11-64"><a href="source-code.html#cb11-64" tabindex="-1"></a>    </span>
<span id="cb11-65"><a href="source-code.html#cb11-65" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb11-66"><a href="source-code.html#cb11-66" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb11-67"><a href="source-code.html#cb11-67" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits)], hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb11-68"><a href="source-code.html#cb11-68" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb11-69"><a href="source-code.html#cb11-69" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)])</span>
<span id="cb11-70"><a href="source-code.html#cb11-70" tabindex="-1"></a>    </span>
<span id="cb11-71"><a href="source-code.html#cb11-71" tabindex="-1"></a>    <span class="co"># extracting cytobands</span></span>
<span id="cb11-72"><a href="source-code.html#cb11-72" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb11-73"><a href="source-code.html#cb11-73" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="fu">seqnames</span>(hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]), hg38IdeogramCyto[<span class="fu">subjectHits</span>(hits)]<span class="sc">$</span>name)</span>
<span id="cb11-74"><a href="source-code.html#cb11-74" tabindex="-1"></a>    <span class="co"># replace &#39;chr&#39;</span></span>
<span id="cb11-75"><a href="source-code.html#cb11-75" tabindex="-1"></a>    cytobands <span class="ot">&lt;-</span></span>
<span id="cb11-76"><a href="source-code.html#cb11-76" tabindex="-1"></a>      <span class="fu">str_replace</span>(cytobands, <span class="at">pattern =</span> <span class="st">&#39;chr&#39;</span>, <span class="at">replacement =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb11-77"><a href="source-code.html#cb11-77" tabindex="-1"></a>    </span>
<span id="cb11-78"><a href="source-code.html#cb11-78" tabindex="-1"></a>    cytobands_cov <span class="ot">&lt;-</span></span>
<span id="cb11-79"><a href="source-code.html#cb11-79" tabindex="-1"></a>      <span class="fu">list</span>(<span class="fu">paste0</span>(cytobands, <span class="st">&#39;;&#39;</span>, <span class="fu">round</span>(percentOverlap, <span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>, <span class="st">&#39;%&#39;</span>))</span>
<span id="cb11-80"><a href="source-code.html#cb11-80" tabindex="-1"></a>    </span>
<span id="cb11-81"><a href="source-code.html#cb11-81" tabindex="-1"></a>    for_range<span class="sc">$</span>cytobands_cov <span class="ot">&lt;-</span> cytobands_cov</span>
<span id="cb11-82"><a href="source-code.html#cb11-82" tabindex="-1"></a>    </span>
<span id="cb11-83"><a href="source-code.html#cb11-83" tabindex="-1"></a>    Grange_list_cytoband[[row]] <span class="ot">&lt;-</span> for_range</span>
<span id="cb11-84"><a href="source-code.html#cb11-84" tabindex="-1"></a>    </span>
<span id="cb11-85"><a href="source-code.html#cb11-85" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb11-86"><a href="source-code.html#cb11-86" tabindex="-1"></a>    <span class="co"># Part 2 #</span></span>
<span id="cb11-87"><a href="source-code.html#cb11-87" tabindex="-1"></a>    <span class="do">##########</span></span>
<span id="cb11-88"><a href="source-code.html#cb11-88" tabindex="-1"></a>    <span class="co"># make dataframe that shows how much of an arm is affected </span></span>
<span id="cb11-89"><a href="source-code.html#cb11-89" tabindex="-1"></a>    </span>
<span id="cb11-90"><a href="source-code.html#cb11-90" tabindex="-1"></a>    <span class="co"># Grange with p and q IRanges</span></span>
<span id="cb11-91"><a href="source-code.html#cb11-91" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> hg38IdeogramCyto</span>
<span id="cb11-92"><a href="source-code.html#cb11-92" tabindex="-1"></a>    </span>
<span id="cb11-93"><a href="source-code.html#cb11-93" tabindex="-1"></a>    p_q<span class="sc">$</span>arm <span class="ot">&lt;-</span> <span class="fu">str_extract</span>(p_q<span class="sc">$</span>name, <span class="st">&#39;p|q&#39;</span>)</span>
<span id="cb11-94"><a href="source-code.html#cb11-94" tabindex="-1"></a>    </span>
<span id="cb11-95"><a href="source-code.html#cb11-95" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> <span class="fu">subset</span>(p_q, seqnames <span class="sc">%in%</span> <span class="fu">c</span>(<span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">22</span>), <span class="st">&#39;X&#39;</span>, <span class="st">&#39;Y&#39;</span>))</span>
<span id="cb11-96"><a href="source-code.html#cb11-96" tabindex="-1"></a>    </span>
<span id="cb11-97"><a href="source-code.html#cb11-97" tabindex="-1"></a>    p_q <span class="ot">&lt;-</span> p_q <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames, arm) <span class="sc">%&gt;%</span> <span class="fu">reduce_ranges</span>()</span>
<span id="cb11-98"><a href="source-code.html#cb11-98" tabindex="-1"></a>    </span>
<span id="cb11-99"><a href="source-code.html#cb11-99" tabindex="-1"></a>    <span class="co"># finding overlaps</span></span>
<span id="cb11-100"><a href="source-code.html#cb11-100" tabindex="-1"></a>    hits_pq <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(for_range, p_q)</span>
<span id="cb11-101"><a href="source-code.html#cb11-101" tabindex="-1"></a>    </span>
<span id="cb11-102"><a href="source-code.html#cb11-102" tabindex="-1"></a>    <span class="co"># computing percentage overlap</span></span>
<span id="cb11-103"><a href="source-code.html#cb11-103" tabindex="-1"></a>    overlaps <span class="ot">&lt;-</span></span>
<span id="cb11-104"><a href="source-code.html#cb11-104" tabindex="-1"></a>      <span class="fu">pintersect</span>(for_range[<span class="fu">queryHits</span>(hits_pq)], p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb11-105"><a href="source-code.html#cb11-105" tabindex="-1"></a>    percentOverlap <span class="ot">&lt;-</span></span>
<span id="cb11-106"><a href="source-code.html#cb11-106" tabindex="-1"></a>      <span class="fu">width</span>(overlaps) <span class="sc">/</span> <span class="fu">width</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb11-107"><a href="source-code.html#cb11-107" tabindex="-1"></a>    </span>
<span id="cb11-108"><a href="source-code.html#cb11-108" tabindex="-1"></a>    p_q_percentage <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(p_q[<span class="fu">subjectHits</span>(hits_pq)])</span>
<span id="cb11-109"><a href="source-code.html#cb11-109" tabindex="-1"></a>    </span>
<span id="cb11-110"><a href="source-code.html#cb11-110" tabindex="-1"></a>    p_q_percentage<span class="sc">$</span>percentage_arm_affected <span class="ot">&lt;-</span> percentOverlap <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb11-111"><a href="source-code.html#cb11-111" tabindex="-1"></a>    </span>
<span id="cb11-112"><a href="source-code.html#cb11-112" tabindex="-1"></a>    <span class="fu">colnames</span>(p_q_percentage) <span class="ot">&lt;-</span></span>
<span id="cb11-113"><a href="source-code.html#cb11-113" tabindex="-1"></a>      <span class="fu">c</span>(</span>
<span id="cb11-114"><a href="source-code.html#cb11-114" tabindex="-1"></a>        <span class="st">&#39;seqnames&#39;</span>,</span>
<span id="cb11-115"><a href="source-code.html#cb11-115" tabindex="-1"></a>        <span class="st">&#39;start_of_arm&#39;</span>,</span>
<span id="cb11-116"><a href="source-code.html#cb11-116" tabindex="-1"></a>        <span class="st">&#39;end_of_arm&#39;</span>,</span>
<span id="cb11-117"><a href="source-code.html#cb11-117" tabindex="-1"></a>        <span class="st">&#39;width_of_arm&#39;</span>,</span>
<span id="cb11-118"><a href="source-code.html#cb11-118" tabindex="-1"></a>        <span class="st">&#39;strand&#39;</span>,</span>
<span id="cb11-119"><a href="source-code.html#cb11-119" tabindex="-1"></a>        <span class="st">&#39;chromosome&#39;</span>,</span>
<span id="cb11-120"><a href="source-code.html#cb11-120" tabindex="-1"></a>        <span class="st">&#39;arm&#39;</span>,</span>
<span id="cb11-121"><a href="source-code.html#cb11-121" tabindex="-1"></a>        <span class="st">&#39;percentage_arm_affected&#39;</span></span>
<span id="cb11-122"><a href="source-code.html#cb11-122" tabindex="-1"></a>      )</span>
<span id="cb11-123"><a href="source-code.html#cb11-123" tabindex="-1"></a>    </span>
<span id="cb11-124"><a href="source-code.html#cb11-124" tabindex="-1"></a>    </span>
<span id="cb11-125"><a href="source-code.html#cb11-125" tabindex="-1"></a>    </span>
<span id="cb11-126"><a href="source-code.html#cb11-126" tabindex="-1"></a>    </span>
<span id="cb11-127"><a href="source-code.html#cb11-127" tabindex="-1"></a>    Grange_list_pq_affected[[row]] <span class="ot">&lt;-</span> p_q_percentage</span>
<span id="cb11-128"><a href="source-code.html#cb11-128" tabindex="-1"></a>    </span>
<span id="cb11-129"><a href="source-code.html#cb11-129" tabindex="-1"></a>    </span>
<span id="cb11-130"><a href="source-code.html#cb11-130" tabindex="-1"></a>  }</span>
<span id="cb11-131"><a href="source-code.html#cb11-131" tabindex="-1"></a>  </span>
<span id="cb11-132"><a href="source-code.html#cb11-132" tabindex="-1"></a>  Grange <span class="ot">&lt;-</span> <span class="fu">bind_ranges</span>(Grange_list_cytoband)</span>
<span id="cb11-133"><a href="source-code.html#cb11-133" tabindex="-1"></a>  pq_arm_affect <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(Grange_list_pq_affected)</span>
<span id="cb11-134"><a href="source-code.html#cb11-134" tabindex="-1"></a>  </span>
<span id="cb11-135"><a href="source-code.html#cb11-135" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(Grange, pq_arm_affect))</span>
<span id="cb11-136"><a href="source-code.html#cb11-136" tabindex="-1"></a>  </span>
<span id="cb11-137"><a href="source-code.html#cb11-137" tabindex="-1"></a>  <span class="do">### </span><span class="re">END</span><span class="do"> of function </span><span class="al">###</span></span>
<span id="cb11-138"><a href="source-code.html#cb11-138" tabindex="-1"></a>  </span>
<span id="cb11-139"><a href="source-code.html#cb11-139" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_genes-script-get_genes-functoin" class="section level2 hasAnchor" number="6.12">
<h2><span class="header-section-number">6.12</span> get_genes script: ‘get_genes’ functoin<a href="source-code.html#get_genes-script-get_genes-functoin" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="source-code.html#cb12-1" tabindex="-1"></a>get_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) { </span>
<span id="cb12-2"><a href="source-code.html#cb12-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb12-3"><a href="source-code.html#cb12-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb12-4"><a href="source-code.html#cb12-4" tabindex="-1"></a>  </span>
<span id="cb12-5"><a href="source-code.html#cb12-5" tabindex="-1"></a>  <span class="fu">require</span>(GenomicRanges)</span>
<span id="cb12-6"><a href="source-code.html#cb12-6" tabindex="-1"></a>  <span class="fu">require</span>(biomaRt)</span>
<span id="cb12-7"><a href="source-code.html#cb12-7" tabindex="-1"></a>  </span>
<span id="cb12-8"><a href="source-code.html#cb12-8" tabindex="-1"></a>  <span class="co"># This code should be run only once and is therefore commented. </span></span>
<span id="cb12-9"><a href="source-code.html#cb12-9" tabindex="-1"></a>  <span class="co"># code from: https://www.biostars.org/p/311199/</span></span>
<span id="cb12-10"><a href="source-code.html#cb12-10" tabindex="-1"></a>  </span>
<span id="cb12-11"><a href="source-code.html#cb12-11" tabindex="-1"></a>  <span class="co"># Set up an gene annotation template to use</span></span>
<span id="cb12-12"><a href="source-code.html#cb12-12" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb12-13"><a href="source-code.html#cb12-13" tabindex="-1"></a>  <span class="co"># mart &lt;- useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, host=&quot;www.ensembl.org&quot;, path=&quot;/biomart/martservice&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</span></span>
<span id="cb12-14"><a href="source-code.html#cb12-14" tabindex="-1"></a>  <span class="co"># genes &lt;- getBM(attributes=c(&quot;hgnc_symbol&quot;,&quot;chromosome_name&quot;,&quot;start_position&quot;,&quot;end_position&quot;), mart=mart)</span></span>
<span id="cb12-15"><a href="source-code.html#cb12-15" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[genes[,1]!=&quot;&quot; &amp; genes[,2] %in% c(1:22,&quot;X&quot;,&quot;Y&quot;),]</span></span>
<span id="cb12-16"><a href="source-code.html#cb12-16" tabindex="-1"></a>  <span class="co"># xidx &lt;- which(genes[,2]==&quot;X&quot;)</span></span>
<span id="cb12-17"><a href="source-code.html#cb12-17" tabindex="-1"></a>  <span class="co"># yidx &lt;- which(genes[,2]==&quot;Y&quot;)</span></span>
<span id="cb12-18"><a href="source-code.html#cb12-18" tabindex="-1"></a>  <span class="co"># genes[xidx, 2] &lt;- 23</span></span>
<span id="cb12-19"><a href="source-code.html#cb12-19" tabindex="-1"></a>  <span class="co"># genes[yidx, 2] &lt;- 24</span></span>
<span id="cb12-20"><a href="source-code.html#cb12-20" tabindex="-1"></a>  <span class="co"># genes[,2] &lt;- sapply(genes[,2],as.integer)</span></span>
<span id="cb12-21"><a href="source-code.html#cb12-21" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,3]),]</span></span>
<span id="cb12-22"><a href="source-code.html#cb12-22" tabindex="-1"></a>  <span class="co"># genes &lt;- genes[order(genes[,2]),]</span></span>
<span id="cb12-23"><a href="source-code.html#cb12-23" tabindex="-1"></a>  <span class="co"># save(genes, file = &#39;rda/cna_analysis/genes.rda&#39;)</span></span>
<span id="cb12-24"><a href="source-code.html#cb12-24" tabindex="-1"></a>  <span class="co"># colnames(genes) &lt;- c(&quot;GeneSymbol&quot;,&quot;Chr&quot;,&quot;Start&quot;,&quot;End&quot;)</span></span>
<span id="cb12-25"><a href="source-code.html#cb12-25" tabindex="-1"></a>  <span class="co"># genes_GR &lt;- makeGRangesFromDataFrame(genes,keep.extra.columns = TRUE)</span></span>
<span id="cb12-26"><a href="source-code.html#cb12-26" tabindex="-1"></a>  <span class="co"># save(genes_GR, file = &#39;rda/cna_analysis/genes_GR.rda&#39;)</span></span>
<span id="cb12-27"><a href="source-code.html#cb12-27" tabindex="-1"></a>  </span>
<span id="cb12-28"><a href="source-code.html#cb12-28" tabindex="-1"></a>  <span class="co"># loading genelist and genomic location</span></span>
<span id="cb12-29"><a href="source-code.html#cb12-29" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&quot;../cna_analysis/rda/cna_analysis/genes_GR.rda&quot;</span>, <span class="at">envir =</span> .GlobalEnv)</span>
<span id="cb12-30"><a href="source-code.html#cb12-30" tabindex="-1"></a>  <span class="fu">load</span>(<span class="st">&#39;../cna_analysis/rda/cna_analysis/genes.rda&#39;</span>, <span class="at">envir =</span> .GlobalEnv )</span>
<span id="cb12-31"><a href="source-code.html#cb12-31" tabindex="-1"></a>  </span>
<span id="cb12-32"><a href="source-code.html#cb12-32" tabindex="-1"></a>  <span class="co"># Convert both GRanges to similar karyogram style</span></span>
<span id="cb12-33"><a href="source-code.html#cb12-33" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb12-34"><a href="source-code.html#cb12-34" tabindex="-1"></a>  GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="fu">paste0</span>(<span class="st">&#39;chr&#39;</span>, <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">23</span>)))</span>
<span id="cb12-35"><a href="source-code.html#cb12-35" tabindex="-1"></a>  </span>
<span id="cb12-36"><a href="source-code.html#cb12-36" tabindex="-1"></a>  <span class="fu">seqlevelsStyle</span>(genes_GR) <span class="ot">&lt;-</span> <span class="st">&quot;UCSC&quot;</span></span>
<span id="cb12-37"><a href="source-code.html#cb12-37" tabindex="-1"></a>  </span>
<span id="cb12-38"><a href="source-code.html#cb12-38" tabindex="-1"></a>  <span class="co"># Finding hits with genes_GR and GRanges</span></span>
<span id="cb12-39"><a href="source-code.html#cb12-39" tabindex="-1"></a>  hits <span class="ot">&lt;-</span> <span class="fu">findOverlaps</span>(genes_GR, GRange, <span class="at">type=</span><span class="st">&quot;within&quot;</span>)</span>
<span id="cb12-40"><a href="source-code.html#cb12-40" tabindex="-1"></a>  </span>
<span id="cb12-41"><a href="source-code.html#cb12-41" tabindex="-1"></a>  <span class="co"># constructing</span></span>
<span id="cb12-42"><a href="source-code.html#cb12-42" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(GRange)[<span class="fu">subjectHits</span>(hits),],genes[<span class="fu">queryHits</span>(hits),])</span>
<span id="cb12-43"><a href="source-code.html#cb12-43" tabindex="-1"></a>  </span>
<span id="cb12-44"><a href="source-code.html#cb12-44" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb12-45"><a href="source-code.html#cb12-45" tabindex="-1"></a>  </span>
<span id="cb12-46"><a href="source-code.html#cb12-46" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="get_oncogenes-scripts-get_oncogenes-function" class="section level2 hasAnchor" number="6.13">
<h2><span class="header-section-number">6.13</span> get_oncogenes scripts: ‘get_oncogenes’ function<a href="source-code.html#get_oncogenes-scripts-get_oncogenes-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="source-code.html#cb13-1" tabindex="-1"></a>get_oncogenes <span class="ot">&lt;-</span> <span class="cf">function</span>(GRange) {</span>
<span id="cb13-2"><a href="source-code.html#cb13-2" tabindex="-1"></a>  <span class="co">#&#39; @param Grange a grange with seqnames and granges. </span></span>
<span id="cb13-3"><a href="source-code.html#cb13-3" tabindex="-1"></a>  <span class="co">#&#39; @returns returns df with all genes encoded within the GRange regions. </span></span>
<span id="cb13-4"><a href="source-code.html#cb13-4" tabindex="-1"></a>  <span class="fu">source</span>(<span class="st">&#39;R/get_genes.R&#39;</span>)</span>
<span id="cb13-5"><a href="source-code.html#cb13-5" tabindex="-1"></a>  </span>
<span id="cb13-6"><a href="source-code.html#cb13-6" tabindex="-1"></a>  <span class="co"># first finding all genes within GRange</span></span>
<span id="cb13-7"><a href="source-code.html#cb13-7" tabindex="-1"></a>  all_genes <span class="ot">&lt;-</span> <span class="fu">get_genes</span>(GRange) </span>
<span id="cb13-8"><a href="source-code.html#cb13-8" tabindex="-1"></a>  </span>
<span id="cb13-9"><a href="source-code.html#cb13-9" tabindex="-1"></a>  <span class="co"># change colnames so dataframes match</span></span>
<span id="cb13-10"><a href="source-code.html#cb13-10" tabindex="-1"></a>  <span class="fu">colnames</span>(all_genes)[<span class="fu">which</span>(<span class="fu">names</span>(all_genes) <span class="sc">==</span> <span class="st">&#39;hgnc_symbol&#39;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Gene Symbol&quot;</span>   </span>
<span id="cb13-11"><a href="source-code.html#cb13-11" tabindex="-1"></a>  </span>
<span id="cb13-12"><a href="source-code.html#cb13-12" tabindex="-1"></a>  <span class="co"># downloading oncogenes</span></span>
<span id="cb13-13"><a href="source-code.html#cb13-13" tabindex="-1"></a>  <span class="fu">require</span>(readr)</span>
<span id="cb13-14"><a href="source-code.html#cb13-14" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../cna_analysis/data/cna_analysis/Census_allMon May 30 12_43_01 2022.csv&quot;</span>)</span>
<span id="cb13-15"><a href="source-code.html#cb13-15" tabindex="-1"></a>  </span>
<span id="cb13-16"><a href="source-code.html#cb13-16" tabindex="-1"></a>  </span>
<span id="cb13-17"><a href="source-code.html#cb13-17" tabindex="-1"></a>  <span class="co"># subset df_genes so to only have oncogenes</span></span>
<span id="cb13-18"><a href="source-code.html#cb13-18" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb13-19"><a href="source-code.html#cb13-19" tabindex="-1"></a>  oncogenes <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(oncogenes, all_genes)</span>
<span id="cb13-20"><a href="source-code.html#cb13-20" tabindex="-1"></a>  </span>
<span id="cb13-21"><a href="source-code.html#cb13-21" tabindex="-1"></a>  <span class="fu">return</span>(oncogenes)</span>
<span id="cb13-22"><a href="source-code.html#cb13-22" tabindex="-1"></a>  </span>
<span id="cb13-23"><a href="source-code.html#cb13-23" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="plot_genes_karyotype-script-plot_genes_karyotype-function" class="section level2 hasAnchor" number="6.14">
<h2><span class="header-section-number">6.14</span> plot_genes_karyotype script: ‘plot_genes_karyotype’ function<a href="source-code.html#plot_genes_karyotype-script-plot_genes_karyotype-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="source-code.html#cb14-1" tabindex="-1"></a>plot_genes_karyotype <span class="ot">&lt;-</span></span>
<span id="cb14-2"><a href="source-code.html#cb14-2" tabindex="-1"></a>  <span class="cf">function</span>(GRange,</span>
<span id="cb14-3"><a href="source-code.html#cb14-3" tabindex="-1"></a>           <span class="at">dist =</span> <span class="sc">-</span><span class="dv">30</span>,</span>
<span id="cb14-4"><a href="source-code.html#cb14-4" tabindex="-1"></a>           show_genes,</span>
<span id="cb14-5"><a href="source-code.html#cb14-5" tabindex="-1"></a>           <span class="at">show_chromosomes =</span> <span class="fu">c</span>(</span>
<span id="cb14-6"><a href="source-code.html#cb14-6" tabindex="-1"></a>             <span class="st">&quot;chr1&quot;</span>,</span>
<span id="cb14-7"><a href="source-code.html#cb14-7" tabindex="-1"></a>             <span class="st">&quot;chr2&quot;</span>,</span>
<span id="cb14-8"><a href="source-code.html#cb14-8" tabindex="-1"></a>             <span class="st">&quot;chr3&quot;</span>,</span>
<span id="cb14-9"><a href="source-code.html#cb14-9" tabindex="-1"></a>             <span class="st">&quot;chr4&quot;</span>,</span>
<span id="cb14-10"><a href="source-code.html#cb14-10" tabindex="-1"></a>             <span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb14-11"><a href="source-code.html#cb14-11" tabindex="-1"></a>             <span class="st">&quot;chr6&quot;</span>,</span>
<span id="cb14-12"><a href="source-code.html#cb14-12" tabindex="-1"></a>             <span class="st">&quot;chr7&quot;</span>,</span>
<span id="cb14-13"><a href="source-code.html#cb14-13" tabindex="-1"></a>             <span class="st">&quot;chr8&quot;</span>,</span>
<span id="cb14-14"><a href="source-code.html#cb14-14" tabindex="-1"></a>             <span class="st">&quot;chr9&quot;</span>,</span>
<span id="cb14-15"><a href="source-code.html#cb14-15" tabindex="-1"></a>             <span class="st">&quot;chr10&quot;</span>,</span>
<span id="cb14-16"><a href="source-code.html#cb14-16" tabindex="-1"></a>             <span class="st">&quot;chr11&quot;</span>,</span>
<span id="cb14-17"><a href="source-code.html#cb14-17" tabindex="-1"></a>             <span class="st">&quot;chr12&quot;</span>,</span>
<span id="cb14-18"><a href="source-code.html#cb14-18" tabindex="-1"></a>             <span class="st">&quot;chr13&quot;</span>,</span>
<span id="cb14-19"><a href="source-code.html#cb14-19" tabindex="-1"></a>             <span class="st">&quot;chr14&quot;</span>,</span>
<span id="cb14-20"><a href="source-code.html#cb14-20" tabindex="-1"></a>             <span class="st">&quot;chr15&quot;</span>,</span>
<span id="cb14-21"><a href="source-code.html#cb14-21" tabindex="-1"></a>             <span class="st">&quot;chr16&quot;</span>,</span>
<span id="cb14-22"><a href="source-code.html#cb14-22" tabindex="-1"></a>             <span class="st">&quot;chr17&quot;</span>,</span>
<span id="cb14-23"><a href="source-code.html#cb14-23" tabindex="-1"></a>             <span class="st">&quot;chr18&quot;</span>,</span>
<span id="cb14-24"><a href="source-code.html#cb14-24" tabindex="-1"></a>             <span class="st">&quot;chr19&quot;</span>,</span>
<span id="cb14-25"><a href="source-code.html#cb14-25" tabindex="-1"></a>             <span class="st">&quot;chr20&quot;</span>,</span>
<span id="cb14-26"><a href="source-code.html#cb14-26" tabindex="-1"></a>             <span class="st">&quot;chr21&quot;</span>,</span>
<span id="cb14-27"><a href="source-code.html#cb14-27" tabindex="-1"></a>             <span class="st">&quot;chr22&quot;</span>,</span>
<span id="cb14-28"><a href="source-code.html#cb14-28" tabindex="-1"></a>             <span class="st">&quot;chrX&quot;</span></span>
<span id="cb14-29"><a href="source-code.html#cb14-29" tabindex="-1"></a>           )) {</span>
<span id="cb14-30"><a href="source-code.html#cb14-30" tabindex="-1"></a>    <span class="co">#&#39; @param GRange a GRange with seqnames, ranges, strand and hgnc_symbol</span></span>
<span id="cb14-31"><a href="source-code.html#cb14-31" tabindex="-1"></a>    <span class="co">#&#39; @param dist How far away the genes should be plotted from the lines.</span></span>
<span id="cb14-32"><a href="source-code.html#cb14-32" tabindex="-1"></a>    <span class="co">#&#39; Between -40 and -70 seems to do the trick most of the times.</span></span>
<span id="cb14-33"><a href="source-code.html#cb14-33" tabindex="-1"></a>    <span class="co">#&#39; @param show_genes is a vector of gene symbols to show.</span></span>
<span id="cb14-34"><a href="source-code.html#cb14-34" tabindex="-1"></a>    <span class="co">#&#39; @param show_chromosomes is a vector indicating which chromosomes to show.</span></span>
<span id="cb14-35"><a href="source-code.html#cb14-35" tabindex="-1"></a>    <span class="co">#&#39; @return a karyotype with genes labeled</span></span>
<span id="cb14-36"><a href="source-code.html#cb14-36" tabindex="-1"></a>    <span class="co">#&#39; @details. Throws an error when the GRange does not contain either</span></span>
<span id="cb14-37"><a href="source-code.html#cb14-37" tabindex="-1"></a>    <span class="co">#&#39; ampflications or deletions. This can be ignored, the resulting plot</span></span>
<span id="cb14-38"><a href="source-code.html#cb14-38" tabindex="-1"></a>    <span class="co">#&#39; is correct</span></span>
<span id="cb14-39"><a href="source-code.html#cb14-39" tabindex="-1"></a>    </span>
<span id="cb14-40"><a href="source-code.html#cb14-40" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb14-41"><a href="source-code.html#cb14-41" tabindex="-1"></a>    <span class="co"># Libraries #</span></span>
<span id="cb14-42"><a href="source-code.html#cb14-42" tabindex="-1"></a>    <span class="do">#############</span></span>
<span id="cb14-43"><a href="source-code.html#cb14-43" tabindex="-1"></a>    <span class="fu">require</span>(karyoploteR)</span>
<span id="cb14-44"><a href="source-code.html#cb14-44" tabindex="-1"></a>    <span class="fu">require</span>(stringr)</span>
<span id="cb14-45"><a href="source-code.html#cb14-45" tabindex="-1"></a>    </span>
<span id="cb14-46"><a href="source-code.html#cb14-46" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb14-47"><a href="source-code.html#cb14-47" tabindex="-1"></a>    <span class="co"># Wrangle #</span></span>
<span id="cb14-48"><a href="source-code.html#cb14-48" tabindex="-1"></a>    <span class="do">###########</span></span>
<span id="cb14-49"><a href="source-code.html#cb14-49" tabindex="-1"></a>    <span class="co"># making sure seqlevels match</span></span>
<span id="cb14-50"><a href="source-code.html#cb14-50" tabindex="-1"></a>    <span class="fu">seqlevelsStyle</span>(GRange) <span class="ot">&lt;-</span> <span class="st">&#39;UCSC&#39;</span></span>
<span id="cb14-51"><a href="source-code.html#cb14-51" tabindex="-1"></a>    </span>
<span id="cb14-52"><a href="source-code.html#cb14-52" tabindex="-1"></a>    kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(</span>
<span id="cb14-53"><a href="source-code.html#cb14-53" tabindex="-1"></a>      <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb14-54"><a href="source-code.html#cb14-54" tabindex="-1"></a>      <span class="at">plot.type =</span> <span class="dv">5</span>,</span>
<span id="cb14-55"><a href="source-code.html#cb14-55" tabindex="-1"></a>      <span class="at">labels.plotter =</span> <span class="cn">NULL</span>,</span>
<span id="cb14-56"><a href="source-code.html#cb14-56" tabindex="-1"></a>      <span class="at">main =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb14-57"><a href="source-code.html#cb14-57" tabindex="-1"></a>      <span class="at">cex =</span> <span class="dv">4</span>,</span>
<span id="cb14-58"><a href="source-code.html#cb14-58" tabindex="-1"></a>      <span class="at">chromosomes =</span> show_chromosomes</span>
<span id="cb14-59"><a href="source-code.html#cb14-59" tabindex="-1"></a>    )</span>
<span id="cb14-60"><a href="source-code.html#cb14-60" tabindex="-1"></a>    </span>
<span id="cb14-61"><a href="source-code.html#cb14-61" tabindex="-1"></a>    <span class="co">#renaming seqlevels</span></span>
<span id="cb14-62"><a href="source-code.html#cb14-62" tabindex="-1"></a>    seqlevels_grange_change <span class="ot">&lt;-</span></span>
<span id="cb14-63"><a href="source-code.html#cb14-63" tabindex="-1"></a>      <span class="fu">str_replace_all</span>(<span class="fu">seqlevels</span>(GRange),</span>
<span id="cb14-64"><a href="source-code.html#cb14-64" tabindex="-1"></a>                      <span class="at">pattern =</span> <span class="st">&#39;chr23&#39;</span>,</span>
<span id="cb14-65"><a href="source-code.html#cb14-65" tabindex="-1"></a>                      <span class="at">replacement =</span> <span class="st">&#39;chrX&#39;</span>)</span>
<span id="cb14-66"><a href="source-code.html#cb14-66" tabindex="-1"></a>    </span>
<span id="cb14-67"><a href="source-code.html#cb14-67" tabindex="-1"></a>    GRange <span class="ot">&lt;-</span> <span class="fu">renameSeqlevels</span>(GRange, <span class="at">value =</span> seqlevels_grange_change)</span>
<span id="cb14-68"><a href="source-code.html#cb14-68" tabindex="-1"></a>    </span>
<span id="cb14-69"><a href="source-code.html#cb14-69" tabindex="-1"></a>    <span class="fu">kpAddChromosomeNames</span>(</span>
<span id="cb14-70"><a href="source-code.html#cb14-70" tabindex="-1"></a>      kp,</span>
<span id="cb14-71"><a href="source-code.html#cb14-71" tabindex="-1"></a>      <span class="at">chr.names =</span> <span class="fu">str_remove_all</span>(show_chromosomes, <span class="st">&#39;chr&#39;</span>),</span>
<span id="cb14-72"><a href="source-code.html#cb14-72" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.3</span></span>
<span id="cb14-73"><a href="source-code.html#cb14-73" tabindex="-1"></a>    )</span>
<span id="cb14-74"><a href="source-code.html#cb14-74" tabindex="-1"></a>    </span>
<span id="cb14-75"><a href="source-code.html#cb14-75" tabindex="-1"></a>    <span class="co"># split into amplifications and deletions</span></span>
<span id="cb14-76"><a href="source-code.html#cb14-76" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;amplified&#39;</span>]</span>
<span id="cb14-77"><a href="source-code.html#cb14-77" tabindex="-1"></a>    ampl <span class="ot">&lt;-</span> ampl[ampl<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb14-78"><a href="source-code.html#cb14-78" tabindex="-1"></a>    </span>
<span id="cb14-79"><a href="source-code.html#cb14-79" tabindex="-1"></a>    del <span class="ot">&lt;-</span> GRange[GRange<span class="sc">$</span>cn <span class="sc">==</span> <span class="st">&#39;deleted&#39;</span>]</span>
<span id="cb14-80"><a href="source-code.html#cb14-80" tabindex="-1"></a>    del <span class="ot">&lt;-</span> del[del<span class="sc">$</span>hgnc_symbol <span class="sc">%in%</span> show_genes]</span>
<span id="cb14-81"><a href="source-code.html#cb14-81" tabindex="-1"></a>    </span>
<span id="cb14-82"><a href="source-code.html#cb14-82" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb14-83"><a href="source-code.html#cb14-83" tabindex="-1"></a>      cor <span class="ot">&lt;-</span></span>
<span id="cb14-84"><a href="source-code.html#cb14-84" tabindex="-1"></a>        <span class="fu">cor.test</span>(df<span class="sc">$</span>AUC, df<span class="sc">$</span>dependency, <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>),</span>
<span id="cb14-85"><a href="source-code.html#cb14-85" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb14-86"><a href="source-code.html#cb14-86" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb14-87"><a href="source-code.html#cb14-87" tabindex="-1"></a>    )</span>
<span id="cb14-88"><a href="source-code.html#cb14-88" tabindex="-1"></a>    </span>
<span id="cb14-89"><a href="source-code.html#cb14-89" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb14-90"><a href="source-code.html#cb14-90" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb14-91"><a href="source-code.html#cb14-91" tabindex="-1"></a>        kp,</span>
<span id="cb14-92"><a href="source-code.html#cb14-92" tabindex="-1"></a>        <span class="at">data =</span> ampl,</span>
<span id="cb14-93"><a href="source-code.html#cb14-93" tabindex="-1"></a>        <span class="at">labels =</span> ampl<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb14-94"><a href="source-code.html#cb14-94" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb14-95"><a href="source-code.html#cb14-95" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb14-96"><a href="source-code.html#cb14-96" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.85</span>,</span>
<span id="cb14-97"><a href="source-code.html#cb14-97" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb14-98"><a href="source-code.html#cb14-98" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb14-99"><a href="source-code.html#cb14-99" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb14-100"><a href="source-code.html#cb14-100" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;firebrick&#39;</span>,</span>
<span id="cb14-101"><a href="source-code.html#cb14-101" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;firebrick&#39;</span></span>
<span id="cb14-102"><a href="source-code.html#cb14-102" tabindex="-1"></a>      ),</span>
<span id="cb14-103"><a href="source-code.html#cb14-103" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb14-104"><a href="source-code.html#cb14-104" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb14-105"><a href="source-code.html#cb14-105" tabindex="-1"></a>    )</span>
<span id="cb14-106"><a href="source-code.html#cb14-106" tabindex="-1"></a>    </span>
<span id="cb14-107"><a href="source-code.html#cb14-107" tabindex="-1"></a>    </span>
<span id="cb14-108"><a href="source-code.html#cb14-108" tabindex="-1"></a>    <span class="fu">tryCatch</span>(</span>
<span id="cb14-109"><a href="source-code.html#cb14-109" tabindex="-1"></a>      <span class="fu">kpPlotMarkers</span>(</span>
<span id="cb14-110"><a href="source-code.html#cb14-110" tabindex="-1"></a>        kp,</span>
<span id="cb14-111"><a href="source-code.html#cb14-111" tabindex="-1"></a>        <span class="at">data =</span> del,</span>
<span id="cb14-112"><a href="source-code.html#cb14-112" tabindex="-1"></a>        <span class="at">labels =</span> del<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb14-113"><a href="source-code.html#cb14-113" tabindex="-1"></a>        <span class="at">ignore.chromosome.ends =</span> T,</span>
<span id="cb14-114"><a href="source-code.html#cb14-114" tabindex="-1"></a>        <span class="at">r0 =</span> <span class="dv">1</span>,</span>
<span id="cb14-115"><a href="source-code.html#cb14-115" tabindex="-1"></a>        <span class="at">r1 =</span> <span class="fl">0.4</span>,</span>
<span id="cb14-116"><a href="source-code.html#cb14-116" tabindex="-1"></a>        <span class="at">label.dist =</span> <span class="fl">0.003</span>,</span>
<span id="cb14-117"><a href="source-code.html#cb14-117" tabindex="-1"></a>        <span class="at">cex =</span> <span class="dv">8</span>,</span>
<span id="cb14-118"><a href="source-code.html#cb14-118" tabindex="-1"></a>        <span class="at">label.margin =</span> dist,</span>
<span id="cb14-119"><a href="source-code.html#cb14-119" tabindex="-1"></a>        <span class="at">marker.parts =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.1</span>),</span>
<span id="cb14-120"><a href="source-code.html#cb14-120" tabindex="-1"></a>        <span class="at">line.color =</span> <span class="st">&#39;steelblue&#39;</span>,</span>
<span id="cb14-121"><a href="source-code.html#cb14-121" tabindex="-1"></a>        <span class="at">label.color =</span> <span class="st">&#39;steelblue&#39;</span></span>
<span id="cb14-122"><a href="source-code.html#cb14-122" tabindex="-1"></a>      ),</span>
<span id="cb14-123"><a href="source-code.html#cb14-123" tabindex="-1"></a>      <span class="at">error =</span> <span class="cf">function</span>(e)</span>
<span id="cb14-124"><a href="source-code.html#cb14-124" tabindex="-1"></a>        <span class="cn">NULL</span></span>
<span id="cb14-125"><a href="source-code.html#cb14-125" tabindex="-1"></a>    )</span>
<span id="cb14-126"><a href="source-code.html#cb14-126" tabindex="-1"></a>    </span>
<span id="cb14-127"><a href="source-code.html#cb14-127" tabindex="-1"></a>    </span>
<span id="cb14-128"><a href="source-code.html#cb14-128" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="section level2 hasAnchor" number="6.15">
<h2><span class="header-section-number">6.15</span> shifters script: ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions<a href="source-code.html#shifters-script-create_cn_matrix-and-cluster_and_shift_transitions-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘create_cn_matrix’ and ‘cluster_and_shift_transitions’ functions, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="section level2 hasAnchor" number="6.16">
<h2><span class="header-section-number">6.16</span> calculate_heterogeneity_score script: ‘calculate_heterogeneity_score’ function<a href="source-code.html#calculate_heterogeneity_score-script-calculate_heterogeneity_score-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the ‘calculate_heterogeneity_score’ function, please submit a code sharing request to the corresponding author of this paper:</p>
<p>Bolhaqueiro, A.C.F., Ponsioen, B., Bakker, B. et al. Ongoing chromosomal instability and karyotype evolution in human colorectal cancer organoids. Nat Genet 51, 824–834 (2019). <a href="https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6" class="uri">https://doi-org.proxy.library.uu.nl/10.1038/s41588-019-0399-6</a></p>
</div>
<div id="cna_profiler-script-cna_profiler-function" class="section level2 hasAnchor" number="6.17">
<h2><span class="header-section-number">6.17</span> cna_profiler script: ‘cna_profiler’ function<a href="source-code.html#cna_profiler-script-cna_profiler-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="source-code.html#cb15-1" tabindex="-1"></a>cna_profiler <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb15-2"><a href="source-code.html#cb15-2" tabindex="-1"></a>  <span class="co">#&#39; @param df is a datafame containing all the copy number calls of all the files.</span></span>
<span id="cb15-3"><a href="source-code.html#cb15-3" tabindex="-1"></a>  <span class="co">#&#39;                       Columns: 01) seqnames</span></span>
<span id="cb15-4"><a href="source-code.html#cb15-4" tabindex="-1"></a>  <span class="co">#&#39;                                02) start</span></span>
<span id="cb15-5"><a href="source-code.html#cb15-5" tabindex="-1"></a>  <span class="co">#&#39;                                03) end</span></span>
<span id="cb15-6"><a href="source-code.html#cb15-6" tabindex="-1"></a>  <span class="co">#&#39;                                04) width</span></span>
<span id="cb15-7"><a href="source-code.html#cb15-7" tabindex="-1"></a>  <span class="co">#&#39;                                05) strand</span></span>
<span id="cb15-8"><a href="source-code.html#cb15-8" tabindex="-1"></a>  <span class="co">#&#39;                                06) [copy.number cell 1]</span></span>
<span id="cb15-9"><a href="source-code.html#cb15-9" tabindex="-1"></a>  <span class="co">#&#39;                                07) [copy.number cell 2], etc.</span></span>
<span id="cb15-10"><a href="source-code.html#cb15-10" tabindex="-1"></a>  <span class="co">#&#39; @return Outputs a dataframe that states the ratio (normalized to total cells) of</span></span>
<span id="cb15-11"><a href="source-code.html#cb15-11" tabindex="-1"></a>  <span class="co">#&#39; small, medium, large and whole chromosome CNAs</span></span>
<span id="cb15-12"><a href="source-code.html#cb15-12" tabindex="-1"></a></span>
<span id="cb15-13"><a href="source-code.html#cb15-13" tabindex="-1"></a>  <span class="fu">require</span>(tidyverse)</span>
<span id="cb15-14"><a href="source-code.html#cb15-14" tabindex="-1"></a>  <span class="fu">require</span>(GWASTools)</span>
<span id="cb15-15"><a href="source-code.html#cb15-15" tabindex="-1"></a>  </span>
<span id="cb15-16"><a href="source-code.html#cb15-16" tabindex="-1"></a>  <span class="co"># defining some parameters</span></span>
<span id="cb15-17"><a href="source-code.html#cb15-17" tabindex="-1"></a>  <span class="co"># number of cells </span></span>
<span id="cb15-18"><a href="source-code.html#cb15-18" tabindex="-1"></a>  n_cells <span class="ot">&lt;-</span> <span class="fu">ncol</span>(df) <span class="sc">-</span> <span class="dv">5</span></span>
<span id="cb15-19"><a href="source-code.html#cb15-19" tabindex="-1"></a>  </span>
<span id="cb15-20"><a href="source-code.html#cb15-20" tabindex="-1"></a>  <span class="co"># extract centromere positions</span></span>
<span id="cb15-21"><a href="source-code.html#cb15-21" tabindex="-1"></a>  <span class="fu">data</span>(centromeres.hg38)</span>
<span id="cb15-22"><a href="source-code.html#cb15-22" tabindex="-1"></a>  </span>
<span id="cb15-23"><a href="source-code.html#cb15-23" tabindex="-1"></a>  centromeres.hg38 <span class="ot">&lt;-</span> <span class="fu">subset</span>(centromeres.hg38, chrom <span class="sc">!=</span> <span class="st">&#39;Y&#39;</span>)</span>
<span id="cb15-24"><a href="source-code.html#cb15-24" tabindex="-1"></a>  <span class="fu">names</span>(centromeres.hg38) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>)</span>
<span id="cb15-25"><a href="source-code.html#cb15-25" tabindex="-1"></a>  </span>
<span id="cb15-26"><a href="source-code.html#cb15-26" tabindex="-1"></a>  </span>
<span id="cb15-27"><a href="source-code.html#cb15-27" tabindex="-1"></a>  <span class="co"># extract total length chromosomes</span></span>
<span id="cb15-28"><a href="source-code.html#cb15-28" tabindex="-1"></a>  total_length <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(seqnames) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>(<span class="fu">sum</span>(width))</span>
<span id="cb15-29"><a href="source-code.html#cb15-29" tabindex="-1"></a>  </span>
<span id="cb15-30"><a href="source-code.html#cb15-30" tabindex="-1"></a>  <span class="co"># construct dataframe containing chromosome name, centromere position (start and end) and total length chromosome</span></span>
<span id="cb15-31"><a href="source-code.html#cb15-31" tabindex="-1"></a>  chrom_centro_total <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(total_length, centromeres.hg38)</span>
<span id="cb15-32"><a href="source-code.html#cb15-32" tabindex="-1"></a>  </span>
<span id="cb15-33"><a href="source-code.html#cb15-33" tabindex="-1"></a>  <span class="co"># compute arm lengths</span></span>
<span id="cb15-34"><a href="source-code.html#cb15-34" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>p_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>start <span class="sc">+</span> (chrom_centro_total<span class="sc">$</span>end<span class="sc">-</span>chrom_centro_total<span class="sc">$</span>start)</span>
<span id="cb15-35"><a href="source-code.html#cb15-35" tabindex="-1"></a>  chrom_centro_total<span class="sc">$</span>q_arm <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span><span class="st">`</span><span class="at">sum(width)</span><span class="st">`</span> <span class="sc">-</span>  chrom_centro_total<span class="sc">$</span>p_arm</span>
<span id="cb15-36"><a href="source-code.html#cb15-36" tabindex="-1"></a>  </span>
<span id="cb15-37"><a href="source-code.html#cb15-37" tabindex="-1"></a>  <span class="do">##### !!!!!!!! ######</span></span>
<span id="cb15-38"><a href="source-code.html#cb15-38" tabindex="-1"></a>  <span class="co"># Check if this is still correct when using new assemblies !</span></span>
<span id="cb15-39"><a href="source-code.html#cb15-39" tabindex="-1"></a>  </span>
<span id="cb15-40"><a href="source-code.html#cb15-40" tabindex="-1"></a>  <span class="co"># now all p_arms except for chromosome 1 are correct (for some reason the p-arm is the larger arm here). </span></span>
<span id="cb15-41"><a href="source-code.html#cb15-41" tabindex="-1"></a>  <span class="co"># I am going to swithc it. </span></span>
<span id="cb15-42"><a href="source-code.html#cb15-42" tabindex="-1"></a>  p_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>q_arm[<span class="dv">1</span>]</span>
<span id="cb15-43"><a href="source-code.html#cb15-43" tabindex="-1"></a>  q_arm_1 <span class="ot">&lt;-</span> chrom_centro_total<span class="sc">$</span>p_arm[<span class="dv">1</span>]</span>
<span id="cb15-44"><a href="source-code.html#cb15-44" tabindex="-1"></a>  </span>
<span id="cb15-45"><a href="source-code.html#cb15-45" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">5</span>] <span class="ot">&lt;-</span> p_arm_1</span>
<span id="cb15-46"><a href="source-code.html#cb15-46" tabindex="-1"></a>  chrom_centro_total[<span class="dv">1</span>,<span class="dv">6</span>] <span class="ot">&lt;-</span> q_arm_1</span>
<span id="cb15-47"><a href="source-code.html#cb15-47" tabindex="-1"></a>  </span>
<span id="cb15-48"><a href="source-code.html#cb15-48" tabindex="-1"></a>  </span>
<span id="cb15-49"><a href="source-code.html#cb15-49" tabindex="-1"></a>  <span class="co"># function to extract CNA for each chromosome, run this function for each chromosome</span></span>
<span id="cb15-50"><a href="source-code.html#cb15-50" tabindex="-1"></a>  cna_extract <span class="ot">&lt;-</span> <span class="cf">function</span>(chromosome) {</span>
<span id="cb15-51"><a href="source-code.html#cb15-51" tabindex="-1"></a>    <span class="co"># @parama: chromosome is the chromosome (i.e., 1, 2,3, X, etc) </span></span>
<span id="cb15-52"><a href="source-code.html#cb15-52" tabindex="-1"></a>    df_chrom <span class="ot">&lt;-</span> <span class="fu">subset</span>(df, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb15-53"><a href="source-code.html#cb15-53" tabindex="-1"></a>    </span>
<span id="cb15-54"><a href="source-code.html#cb15-54" tabindex="-1"></a>    <span class="co"># run this function for each column in df_chrom</span></span>
<span id="cb15-55"><a href="source-code.html#cb15-55" tabindex="-1"></a>    cna_extract_cell <span class="ot">&lt;-</span> <span class="cf">function</span>(column_i) {</span>
<span id="cb15-56"><a href="source-code.html#cb15-56" tabindex="-1"></a>      <span class="co"># @ param column i is the integer of the column in df_chrom </span></span>
<span id="cb15-57"><a href="source-code.html#cb15-57" tabindex="-1"></a>      <span class="co"># from which to extract CNAs</span></span>
<span id="cb15-58"><a href="source-code.html#cb15-58" tabindex="-1"></a>      </span>
<span id="cb15-59"><a href="source-code.html#cb15-59" tabindex="-1"></a>      <span class="co"># extract rows that switch CNA value (for example from 2 to 3)</span></span>
<span id="cb15-60"><a href="source-code.html#cb15-60" tabindex="-1"></a>      CNA_change <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="fu">tail</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>) <span class="sc">!=</span> <span class="fu">head</span>(df_chrom[,column_i],<span class="sc">-</span><span class="dv">1</span>)))</span>
<span id="cb15-61"><a href="source-code.html#cb15-61" tabindex="-1"></a>      </span>
<span id="cb15-62"><a href="source-code.html#cb15-62" tabindex="-1"></a>      <span class="co"># extract indices</span></span>
<span id="cb15-63"><a href="source-code.html#cb15-63" tabindex="-1"></a>      CNA_change_indices <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">sort</span>(<span class="fu">c</span>(<span class="dv">1</span>,CNA_change, (CNA_change<span class="dv">-1</span>), <span class="fu">nrow</span>(df_chrom))))</span>
<span id="cb15-64"><a href="source-code.html#cb15-64" tabindex="-1"></a>      </span>
<span id="cb15-65"><a href="source-code.html#cb15-65" tabindex="-1"></a>      <span class="co"># construct dataframe containing end and start of each CNA</span></span>
<span id="cb15-66"><a href="source-code.html#cb15-66" tabindex="-1"></a>      CNA_change_df <span class="ot">&lt;-</span> df_chrom[CNA_change_indices,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>, column_i)]</span>
<span id="cb15-67"><a href="source-code.html#cb15-67" tabindex="-1"></a>      <span class="fu">names</span>(CNA_change_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;seqnames&#39;</span>, <span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;width&#39;</span>, <span class="st">&#39;strand&#39;</span>, <span class="st">&#39;cna&#39;</span>)</span>
<span id="cb15-68"><a href="source-code.html#cb15-68" tabindex="-1"></a>      </span>
<span id="cb15-69"><a href="source-code.html#cb15-69" tabindex="-1"></a>      start_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-70"><a href="source-code.html#cb15-70" tabindex="-1"></a>      end_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-71"><a href="source-code.html#cb15-71" tabindex="-1"></a>      </span>
<span id="cb15-72"><a href="source-code.html#cb15-72" tabindex="-1"></a>      i <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb15-73"><a href="source-code.html#cb15-73" tabindex="-1"></a>      </span>
<span id="cb15-74"><a href="source-code.html#cb15-74" tabindex="-1"></a>      <span class="cf">while</span>(i<span class="sc">&lt;=</span><span class="fu">nrow</span>(CNA_change_df)){</span>
<span id="cb15-75"><a href="source-code.html#cb15-75" tabindex="-1"></a>        </span>
<span id="cb15-76"><a href="source-code.html#cb15-76" tabindex="-1"></a>        <span class="fu">try</span>(<span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">==</span> CNA_change_df[i, <span class="dv">6</span>]){</span>
<span id="cb15-77"><a href="source-code.html#cb15-77" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb15-78"><a href="source-code.html#cb15-78" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">3</span>]</span>
<span id="cb15-79"><a href="source-code.html#cb15-79" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">2</span></span>
<span id="cb15-80"><a href="source-code.html#cb15-80" tabindex="-1"></a>        })</span>
<span id="cb15-81"><a href="source-code.html#cb15-81" tabindex="-1"></a>        </span>
<span id="cb15-82"><a href="source-code.html#cb15-82" tabindex="-1"></a>        <span class="cf">if</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>] <span class="sc">!=</span> CNA_change_df[i, <span class="dv">6</span>] <span class="sc">||</span> <span class="fu">is.na</span>(CNA_change_df[i<span class="sc">+</span><span class="dv">1</span>, <span class="dv">6</span>])){</span>
<span id="cb15-83"><a href="source-code.html#cb15-83" tabindex="-1"></a>          start_list[[(<span class="fu">length</span>(start_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i,<span class="dv">2</span>]</span>
<span id="cb15-84"><a href="source-code.html#cb15-84" tabindex="-1"></a>          end_list[[(<span class="fu">length</span>(end_list) <span class="sc">+</span> <span class="dv">1</span>)]] <span class="ot">&lt;-</span> CNA_change_df[i, <span class="dv">3</span>]</span>
<span id="cb15-85"><a href="source-code.html#cb15-85" tabindex="-1"></a>          i <span class="ot">&lt;-</span> i<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb15-86"><a href="source-code.html#cb15-86" tabindex="-1"></a>        }</span>
<span id="cb15-87"><a href="source-code.html#cb15-87" tabindex="-1"></a>        </span>
<span id="cb15-88"><a href="source-code.html#cb15-88" tabindex="-1"></a>      }</span>
<span id="cb15-89"><a href="source-code.html#cb15-89" tabindex="-1"></a>      </span>
<span id="cb15-90"><a href="source-code.html#cb15-90" tabindex="-1"></a>      <span class="co"># constructing a dataframe</span></span>
<span id="cb15-91"><a href="source-code.html#cb15-91" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">seqnames =</span> chromosome, </span>
<span id="cb15-92"><a href="source-code.html#cb15-92" tabindex="-1"></a>                                  <span class="at">start =</span> <span class="fu">unlist</span>(start_list), </span>
<span id="cb15-93"><a href="source-code.html#cb15-93" tabindex="-1"></a>                                  <span class="at">end =</span> <span class="fu">unlist</span>(end_list))</span>
<span id="cb15-94"><a href="source-code.html#cb15-94" tabindex="-1"></a>      </span>
<span id="cb15-95"><a href="source-code.html#cb15-95" tabindex="-1"></a>      <span class="co"># remove NA</span></span>
<span id="cb15-96"><a href="source-code.html#cb15-96" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(cna_dataframe)</span>
<span id="cb15-97"><a href="source-code.html#cb15-97" tabindex="-1"></a>      </span>
<span id="cb15-98"><a href="source-code.html#cb15-98" tabindex="-1"></a>      <span class="co"># add column CNA</span></span>
<span id="cb15-99"><a href="source-code.html#cb15-99" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>cna <span class="ot">&lt;-</span> (CNA_change_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(cna<span class="sc">!=</span><span class="fu">lag</span>(cna) <span class="sc">|</span> <span class="fu">is.na</span>(<span class="fu">lag</span>(cna))))<span class="sc">$</span>cna</span>
<span id="cb15-100"><a href="source-code.html#cb15-100" tabindex="-1"></a>      </span>
<span id="cb15-101"><a href="source-code.html#cb15-101" tabindex="-1"></a>      <span class="co"># remove columsn with cna == 2 (this is no CNA but normal diploid)</span></span>
<span id="cb15-102"><a href="source-code.html#cb15-102" tabindex="-1"></a>      cna_dataframe <span class="ot">&lt;-</span> <span class="fu">subset</span>(cna_dataframe, cna <span class="sc">!=</span> <span class="dv">2</span>)</span>
<span id="cb15-103"><a href="source-code.html#cb15-103" tabindex="-1"></a>      </span>
<span id="cb15-104"><a href="source-code.html#cb15-104" tabindex="-1"></a>      <span class="co"># add total number of CNAs in this cell as colunn num_cna</span></span>
<span id="cb15-105"><a href="source-code.html#cb15-105" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>num_cna <span class="ot">&lt;-</span> <span class="fu">length</span>(cna_dataframe<span class="sc">$</span>cna), <span class="at">silent =</span> T)</span>
<span id="cb15-106"><a href="source-code.html#cb15-106" tabindex="-1"></a>      </span>
<span id="cb15-107"><a href="source-code.html#cb15-107" tabindex="-1"></a>      <span class="co"># add a width column of the CNA</span></span>
<span id="cb15-108"><a href="source-code.html#cb15-108" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>width <span class="ot">&lt;-</span> cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb15-109"><a href="source-code.html#cb15-109" tabindex="-1"></a>      </span>
<span id="cb15-110"><a href="source-code.html#cb15-110" tabindex="-1"></a>      <span class="co"># extract chromosome specifics</span></span>
<span id="cb15-111"><a href="source-code.html#cb15-111" tabindex="-1"></a>      chrom_df <span class="ot">&lt;-</span> <span class="fu">subset</span>(chrom_centro_total, seqnames <span class="sc">==</span> chromosome)</span>
<span id="cb15-112"><a href="source-code.html#cb15-112" tabindex="-1"></a>      </span>
<span id="cb15-113"><a href="source-code.html#cb15-113" tabindex="-1"></a>      <span class="co"># in the next of code, we are going to compute the fraction of SCNA</span></span>
<span id="cb15-114"><a href="source-code.html#cb15-114" tabindex="-1"></a>      <span class="co"># Important: using this measure, a fraction of 1 means that a region the size of one arm</span></span>
<span id="cb15-115"><a href="source-code.html#cb15-115" tabindex="-1"></a>      <span class="co"># was affected. A fraction of 2 means the whole chromosome is affected. Note that these CNA</span></span>
<span id="cb15-116"><a href="source-code.html#cb15-116" tabindex="-1"></a>      <span class="co"># can cross centromeres, so we are unable to easily assign if it is the p-arm or q-arm that is </span></span>
<span id="cb15-117"><a href="source-code.html#cb15-117" tabindex="-1"></a>      <span class="co"># affected. </span></span>
<span id="cb15-118"><a href="source-code.html#cb15-118" tabindex="-1"></a>      cna_dataframe<span class="sc">$</span>fraction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>end <span class="sc">&lt;</span> chrom_df<span class="sc">$</span>p_arm, (cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm), </span>
<span id="cb15-119"><a href="source-code.html#cb15-119" tabindex="-1"></a>                                       <span class="fu">ifelse</span>(cna_dataframe<span class="sc">$</span>start <span class="sc">&gt;</span> chrom_df<span class="sc">$</span>p_arm, cna_dataframe<span class="sc">$</span>width <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm,</span>
<span id="cb15-120"><a href="source-code.html#cb15-120" tabindex="-1"></a>                                              (((chrom_df<span class="sc">$</span>p_arm <span class="sc">-</span> cna_dataframe<span class="sc">$</span>start) <span class="sc">/</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">+</span> ((cna_dataframe<span class="sc">$</span>end <span class="sc">-</span> chrom_df<span class="sc">$</span>p_arm) <span class="sc">/</span> chrom_df<span class="sc">$</span>q_arm))))</span>
<span id="cb15-121"><a href="source-code.html#cb15-121" tabindex="-1"></a>      </span>
<span id="cb15-122"><a href="source-code.html#cb15-122" tabindex="-1"></a>      <span class="co"># adding cellname</span></span>
<span id="cb15-123"><a href="source-code.html#cb15-123" tabindex="-1"></a>      <span class="fu">try</span>(cna_dataframe<span class="sc">$</span>cell_id <span class="ot">&lt;-</span> <span class="fu">colnames</span>(df_chrom)[column_i], <span class="at">silent =</span> T)</span>
<span id="cb15-124"><a href="source-code.html#cb15-124" tabindex="-1"></a>      </span>
<span id="cb15-125"><a href="source-code.html#cb15-125" tabindex="-1"></a>      <span class="fu">return</span>(cna_dataframe)</span>
<span id="cb15-126"><a href="source-code.html#cb15-126" tabindex="-1"></a>      </span>
<span id="cb15-127"><a href="source-code.html#cb15-127" tabindex="-1"></a>    }</span>
<span id="cb15-128"><a href="source-code.html#cb15-128" tabindex="-1"></a>    </span>
<span id="cb15-129"><a href="source-code.html#cb15-129" tabindex="-1"></a>    <span class="co"># running for each cell. </span></span>
<span id="cb15-130"><a href="source-code.html#cb15-130" tabindex="-1"></a>    cna_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">6</span><span class="sc">:</span><span class="fu">ncol</span>(df), cna_extract_cell)</span>
<span id="cb15-131"><a href="source-code.html#cb15-131" tabindex="-1"></a>    </span>
<span id="cb15-132"><a href="source-code.html#cb15-132" tabindex="-1"></a>    <span class="co"># constructing single dataframe </span></span>
<span id="cb15-133"><a href="source-code.html#cb15-133" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_list, <span class="at">.id =</span> <span class="st">&quot;cell_num_scar&quot;</span>)</span>
<span id="cb15-134"><a href="source-code.html#cb15-134" tabindex="-1"></a>    </span>
<span id="cb15-135"><a href="source-code.html#cb15-135" tabindex="-1"></a>    <span class="co"># removing cell_num scar</span></span>
<span id="cb15-136"><a href="source-code.html#cb15-136" tabindex="-1"></a>    cna_df <span class="ot">&lt;-</span> cna_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df)]</span>
<span id="cb15-137"><a href="source-code.html#cb15-137" tabindex="-1"></a>    </span>
<span id="cb15-138"><a href="source-code.html#cb15-138" tabindex="-1"></a>    <span class="fu">return</span>(cna_df)</span>
<span id="cb15-139"><a href="source-code.html#cb15-139" tabindex="-1"></a>    </span>
<span id="cb15-140"><a href="source-code.html#cb15-140" tabindex="-1"></a>  }</span>
<span id="cb15-141"><a href="source-code.html#cb15-141" tabindex="-1"></a>  </span>
<span id="cb15-142"><a href="source-code.html#cb15-142" tabindex="-1"></a>  cna_df_per_chrom_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb15-143"><a href="source-code.html#cb15-143" tabindex="-1"></a>  </span>
<span id="cb15-144"><a href="source-code.html#cb15-144" tabindex="-1"></a>  <span class="cf">for</span>(chrom <span class="cf">in</span> <span class="fu">unique</span>(df<span class="sc">$</span>seqnames)) {</span>
<span id="cb15-145"><a href="source-code.html#cb15-145" tabindex="-1"></a>    cna_df_per_chrom_list[[chrom]] <span class="ot">&lt;-</span> <span class="fu">cna_extract</span>(chrom)</span>
<span id="cb15-146"><a href="source-code.html#cb15-146" tabindex="-1"></a>  }</span>
<span id="cb15-147"><a href="source-code.html#cb15-147" tabindex="-1"></a>  </span>
<span id="cb15-148"><a href="source-code.html#cb15-148" tabindex="-1"></a>  <span class="co"># constructing single dataframe </span></span>
<span id="cb15-149"><a href="source-code.html#cb15-149" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_df_per_chrom_list, <span class="at">.id =</span> <span class="st">&quot;chrom_scar&quot;</span>)</span>
<span id="cb15-150"><a href="source-code.html#cb15-150" tabindex="-1"></a>  </span>
<span id="cb15-151"><a href="source-code.html#cb15-151" tabindex="-1"></a>  <span class="co"># removing cell_num scar</span></span>
<span id="cb15-152"><a href="source-code.html#cb15-152" tabindex="-1"></a>  cna_df_per_chrom_df <span class="ot">&lt;-</span> cna_df_per_chrom_df[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(cna_df_per_chrom_df)]</span>
<span id="cb15-153"><a href="source-code.html#cb15-153" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="cin_signature_ccle-script-run_cin_ccle-function" class="section level2 hasAnchor" number="6.18">
<h2><span class="header-section-number">6.18</span> cin_signature_ccle script: ‘run_cin_ccle’ function<a href="source-code.html#cin_signature_ccle-script-run_cin_ccle-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre class="run_cin_ccle"><code>run_cin_ccle &lt;- function(cnv, d) { 
  #&#39; @param cnv is segmental data to input in CINSignatureQuantification
  #&#39; @param d is a dataframe with sample metadata that is joined with cnv
  #&#39; @return outputs CIN signature values.
  
  #############
  # Libraries #
  #############
  require(CINSignatureQuantification)
  require(tidyverse)
  
  ###########
  # Wrangle #
  ###########
  wrangler_cin_signature_ccle &lt;- function(df) {
    #&#39; @param  df: A dataframe containing all the copy number calls of all the files.
    #&#39; @return a dataframe in the format https://github.com/markowetzlab/CINSignatureQuantification
    
    # remove unwanted column names
    keep_columns &lt;-
      c(&#39;Chromosome&#39;,
        &#39;Start&#39;,
        &#39;End&#39;,
        &#39;Segment_Mean&#39;,
        &#39;DepMap_ID&#39;)
    
    new_df &lt;- subset(df, select = keep_columns)
    
    # reorder columns
    col_order &lt;- c(&#39;Chromosome&#39;,
                   &#39;Start&#39;,
                   &#39;End&#39;,
                   &#39;Segment_Mean&#39;,
                   &#39;DepMap_ID&#39;)
    
    new_df &lt;- new_df[, col_order]
    
    # define new column names
    column_names &lt;-
      c(&#39;chromosome&#39;,
        &#39;start&#39;,
        &#39;end&#39;,
        &#39;segVal&#39;,
        &#39;sample&#39;)
    
    colnames(new_df) &lt;- column_names
    
    return(new_df)
    
  }
  
  # Subset cnv to only include samples where radiation sensitivity is known
  cnv &lt;-
    cnv[cnv$DepMap_ID %in% d$DepMap_ID,]
  
  # Wrangle
  cnv &lt;-
    wrangler_cin_signature_ccle(cnv)
  
  # The CIN algorithm (run with quantifyCNSignatures) does not support Y chromosome, we have to get rid of those.
  cnv &lt;-
    cnv[cnv$chromosome != &#39;Y&#39;, ]
  
  #######
  # Run #
  #######
  # Running the CIN scores
  cin_scores &lt;-
    quantifyCNSignatures(cnv, build = &#39;hg38&#39;)
  
  # Extracting the values.
  cin_values &lt;- getActivities(cin_scores)

  ###########
  # Wrangle #
  ###########
  cin_df &lt;- as.data.frame(cin_values)
  
  cin_df$DepMap_ID &lt;- rownames(cin_df)
  
  cin_df &lt;-
    gather(cin_df, cx_signature, cx_value, CX1:CX17, factor_key = T)
  
  # add some metadata
  cin_df &lt;- inner_join(cin_df, d)

  ##########
  # Return #
  ##########
  return(cin_df)
  
}
</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="copy-number-pattern-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/edit/main/06-source.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/dshandel/subclonal_evolution_paper/blob/main/06-source.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
